// @goauthentik/api/src/models/AppleLoginChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * Special challenge for apple-native authentication flow, which happens on the client.
 * @export
 * @interface AppleLoginChallenge
 */
export interface AppleLoginChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof AppleLoginChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof AppleLoginChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof AppleLoginChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof AppleLoginChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof AppleLoginChallenge
     */
    clientId: string;
    /**
     * 
     * @type {string}
     * @memberof AppleLoginChallenge
     */
    scope: string;
    /**
     * 
     * @type {string}
     * @memberof AppleLoginChallenge
     */
    redirectUri: string;
    /**
     * 
     * @type {string}
     * @memberof AppleLoginChallenge
     */
    state: string;
}

/**
 * Check if a given object implements the AppleLoginChallenge interface.
 */
export function instanceOfAppleLoginChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "scope" in value;
    isInstance = isInstance && "redirectUri" in value;
    isInstance = isInstance && "state" in value;

    return isInstance;
}

export function AppleLoginChallengeFromJSON(json: any): AppleLoginChallenge {
    return AppleLoginChallengeFromJSONTyped(json, false);
}

export function AppleLoginChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): AppleLoginChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'clientId': json['client_id'],
        'scope': json['scope'],
        'redirectUri': json['redirect_uri'],
        'state': json['state'],
    };
}

export function AppleLoginChallengeToJSON(value?: AppleLoginChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'client_id': value.clientId,
        'scope': value.scope,
        'redirect_uri': value.redirectUri,
        'state': value.state,
    };
}
