// @goauthentik/api/src/models/Application.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { PolicyEngineMode } from './PolicyEngineMode';
import {
    PolicyEngineModeFromJSON,
    PolicyEngineModeFromJSONTyped,
    PolicyEngineModeToJSON,
} from './PolicyEngineMode';
import type { Provider } from './Provider';
import {
    ProviderFromJSON,
    ProviderFromJSONTyped,
    ProviderToJSON,
} from './Provider';

/**
 * Application Serializer
 * @export
 * @interface Application
 */
export interface Application {
    /**
     * 
     * @type {string}
     * @memberof Application
     */
    readonly pk: string;
    /**
     * Application's display Name.
     * @type {string}
     * @memberof Application
     */
    name: string;
    /**
     * Internal application name, used in URLs.
     * @type {string}
     * @memberof Application
     */
    slug: string;
    /**
     * 
     * @type {number}
     * @memberof Application
     */
    provider?: number | null;
    /**
     * 
     * @type {Provider}
     * @memberof Application
     */
    readonly providerObj: Provider;
    /**
     * 
     * @type {Array<number>}
     * @memberof Application
     */
    backchannelProviders?: Array<number>;
    /**
     * 
     * @type {Array<Provider>}
     * @memberof Application
     */
    readonly backchannelProvidersObj: Array<Provider>;
    /**
     * Allow formatting of launch URL
     * @type {string}
     * @memberof Application
     */
    readonly launchUrl: string | null;
    /**
     * Open launch URL in a new browser tab or window.
     * @type {boolean}
     * @memberof Application
     */
    openInNewTab?: boolean;
    /**
     * 
     * @type {string}
     * @memberof Application
     */
    metaLaunchUrl?: string;
    /**
     * Get the URL to the App Icon image. If the name is /static or starts with http
     * it is returned as-is
     * @type {string}
     * @memberof Application
     */
    readonly metaIcon: string | null;
    /**
     * 
     * @type {string}
     * @memberof Application
     */
    metaDescription?: string;
    /**
     * 
     * @type {string}
     * @memberof Application
     */
    metaPublisher?: string;
    /**
     * 
     * @type {PolicyEngineMode}
     * @memberof Application
     */
    policyEngineMode?: PolicyEngineMode;
    /**
     * 
     * @type {string}
     * @memberof Application
     */
    group?: string;
}

/**
 * Check if a given object implements the Application interface.
 */
export function instanceOfApplication(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "providerObj" in value;
    isInstance = isInstance && "backchannelProvidersObj" in value;
    isInstance = isInstance && "launchUrl" in value;
    isInstance = isInstance && "metaIcon" in value;

    return isInstance;
}

export function ApplicationFromJSON(json: any): Application {
    return ApplicationFromJSONTyped(json, false);
}

export function ApplicationFromJSONTyped(json: any, ignoreDiscriminator: boolean): Application {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'slug': json['slug'],
        'provider': !exists(json, 'provider') ? undefined : json['provider'],
        'providerObj': ProviderFromJSON(json['provider_obj']),
        'backchannelProviders': !exists(json, 'backchannel_providers') ? undefined : json['backchannel_providers'],
        'backchannelProvidersObj': ((json['backchannel_providers_obj'] as Array<any>).map(ProviderFromJSON)),
        'launchUrl': json['launch_url'],
        'openInNewTab': !exists(json, 'open_in_new_tab') ? undefined : json['open_in_new_tab'],
        'metaLaunchUrl': !exists(json, 'meta_launch_url') ? undefined : json['meta_launch_url'],
        'metaIcon': json['meta_icon'],
        'metaDescription': !exists(json, 'meta_description') ? undefined : json['meta_description'],
        'metaPublisher': !exists(json, 'meta_publisher') ? undefined : json['meta_publisher'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'group': !exists(json, 'group') ? undefined : json['group'],
    };
}

export function ApplicationToJSON(value?: Application | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'slug': value.slug,
        'provider': value.provider,
        'backchannel_providers': value.backchannelProviders,
        'open_in_new_tab': value.openInNewTab,
        'meta_launch_url': value.metaLaunchUrl,
        'meta_description': value.metaDescription,
        'meta_publisher': value.metaPublisher,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'group': value.group,
    };
}
