// @goauthentik/api/src/models/AppleChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Pseudo class for plex response
 * @export
 * @interface AppleChallengeResponseRequest
 */
export interface AppleChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof AppleChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the AppleChallengeResponseRequest interface.
 */
export function instanceOfAppleChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function AppleChallengeResponseRequestFromJSON(json: any): AppleChallengeResponseRequest {
    return AppleChallengeResponseRequestFromJSONTyped(json, false);
}

export function AppleChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AppleChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function AppleChallengeResponseRequestToJSON(value?: AppleChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
