// @goauthentik/api/src/models/AuthenticatorDuoStageManualDeviceImportRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface AuthenticatorDuoStageManualDeviceImportRequest
 */
export interface AuthenticatorDuoStageManualDeviceImportRequest {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStageManualDeviceImportRequest
     */
    duoUserId: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStageManualDeviceImportRequest
     */
    username: string;
}

/**
 * Check if a given object implements the AuthenticatorDuoStageManualDeviceImportRequest interface.
 */
export function instanceOfAuthenticatorDuoStageManualDeviceImportRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "duoUserId" in value;
    isInstance = isInstance && "username" in value;

    return isInstance;
}

export function AuthenticatorDuoStageManualDeviceImportRequestFromJSON(json: any): AuthenticatorDuoStageManualDeviceImportRequest {
    return AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json, false);
}

export function AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorDuoStageManualDeviceImportRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'duoUserId': json['duo_user_id'],
        'username': json['username'],
    };
}

export function AuthenticatorDuoStageManualDeviceImportRequestToJSON(value?: AuthenticatorDuoStageManualDeviceImportRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'duo_user_id': value.duoUserId,
        'username': value.username,
    };
}
