// @goauthentik/api/src/models/AuthenticatorDuoChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * Duo Challenge
 * @export
 * @interface AuthenticatorDuoChallenge
 */
export interface AuthenticatorDuoChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof AuthenticatorDuoChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof AuthenticatorDuoChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof AuthenticatorDuoChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    pendingUser: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    pendingUserAvatar: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    activationBarcode: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    activationCode: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallenge
     */
    stageUuid: string;
}

/**
 * Check if a given object implements the AuthenticatorDuoChallenge interface.
 */
export function instanceOfAuthenticatorDuoChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "pendingUser" in value;
    isInstance = isInstance && "pendingUserAvatar" in value;
    isInstance = isInstance && "activationBarcode" in value;
    isInstance = isInstance && "activationCode" in value;
    isInstance = isInstance && "stageUuid" in value;

    return isInstance;
}

export function AuthenticatorDuoChallengeFromJSON(json: any): AuthenticatorDuoChallenge {
    return AuthenticatorDuoChallengeFromJSONTyped(json, false);
}

export function AuthenticatorDuoChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorDuoChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'pendingUser': json['pending_user'],
        'pendingUserAvatar': json['pending_user_avatar'],
        'activationBarcode': json['activation_barcode'],
        'activationCode': json['activation_code'],
        'stageUuid': json['stage_uuid'],
    };
}

export function AuthenticatorDuoChallengeToJSON(value?: AuthenticatorDuoChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'pending_user': value.pendingUser,
        'pending_user_avatar': value.pendingUserAvatar,
        'activation_barcode': value.activationBarcode,
        'activation_code': value.activationCode,
        'stage_uuid': value.stageUuid,
    };
}
