// @goauthentik/api/src/models/AuthenticatorStaticStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * AuthenticatorStaticStage Serializer
 * @export
 * @interface AuthenticatorStaticStageRequest
 */
export interface AuthenticatorStaticStageRequest {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorStaticStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof AuthenticatorStaticStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof AuthenticatorStaticStageRequest
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorStaticStageRequest
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorStaticStageRequest
     */
    tokenCount?: number;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorStaticStageRequest
     */
    tokenLength?: number;
}

/**
 * Check if a given object implements the AuthenticatorStaticStageRequest interface.
 */
export function instanceOfAuthenticatorStaticStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function AuthenticatorStaticStageRequestFromJSON(json: any): AuthenticatorStaticStageRequest {
    return AuthenticatorStaticStageRequestFromJSONTyped(json, false);
}

export function AuthenticatorStaticStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorStaticStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'tokenCount': !exists(json, 'token_count') ? undefined : json['token_count'],
        'tokenLength': !exists(json, 'token_length') ? undefined : json['token_length'],
    };
}

export function AuthenticatorStaticStageRequestToJSON(value?: AuthenticatorStaticStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'token_count': value.tokenCount,
        'token_length': value.tokenLength,
    };
}
