// @goauthentik/api/src/models/AuthenticatorTOTPStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DigitsEnum } from './DigitsEnum';
import {
    DigitsEnumFromJSON,
    DigitsEnumFromJSONTyped,
    DigitsEnumToJSON,
} from './DigitsEnum';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * AuthenticatorTOTPStage Serializer
 * @export
 * @interface AuthenticatorTOTPStage
 */
export interface AuthenticatorTOTPStage {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof AuthenticatorTOTPStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorTOTPStage
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {DigitsEnum}
     * @memberof AuthenticatorTOTPStage
     */
    digits: DigitsEnum;
}

/**
 * Check if a given object implements the AuthenticatorTOTPStage interface.
 */
export function instanceOfAuthenticatorTOTPStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "digits" in value;

    return isInstance;
}

export function AuthenticatorTOTPStageFromJSON(json: any): AuthenticatorTOTPStage {
    return AuthenticatorTOTPStageFromJSONTyped(json, false);
}

export function AuthenticatorTOTPStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorTOTPStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'digits': DigitsEnumFromJSON(json['digits']),
    };
}

export function AuthenticatorTOTPStageToJSON(value?: AuthenticatorTOTPStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'digits': DigitsEnumToJSON(value.digits),
    };
}
