// @goauthentik/api/src/models/AuthenticatorDuoStageDeviceImportResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface AuthenticatorDuoStageDeviceImportResponse
 */
export interface AuthenticatorDuoStageDeviceImportResponse {
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorDuoStageDeviceImportResponse
     */
    readonly count: number;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStageDeviceImportResponse
     */
    readonly error: string;
}

/**
 * Check if a given object implements the AuthenticatorDuoStageDeviceImportResponse interface.
 */
export function instanceOfAuthenticatorDuoStageDeviceImportResponse(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    isInstance = isInstance && "error" in value;

    return isInstance;
}

export function AuthenticatorDuoStageDeviceImportResponseFromJSON(json: any): AuthenticatorDuoStageDeviceImportResponse {
    return AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json, false);
}

export function AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorDuoStageDeviceImportResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'count': json['count'],
        'error': json['error'],
    };
}

export function AuthenticatorDuoStageDeviceImportResponseToJSON(value?: AuthenticatorDuoStageDeviceImportResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
