// @goauthentik/api/src/models/AuthenticatedSessionUserAgentOs.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * User agent os
 * @export
 * @interface AuthenticatedSessionUserAgentOs
 */
export interface AuthenticatedSessionUserAgentOs {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentOs
     */
    family: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentOs
     */
    major: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentOs
     */
    minor: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentOs
     */
    patch: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentOs
     */
    patchMinor: string;
}

/**
 * Check if a given object implements the AuthenticatedSessionUserAgentOs interface.
 */
export function instanceOfAuthenticatedSessionUserAgentOs(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "family" in value;
    isInstance = isInstance && "major" in value;
    isInstance = isInstance && "minor" in value;
    isInstance = isInstance && "patch" in value;
    isInstance = isInstance && "patchMinor" in value;

    return isInstance;
}

export function AuthenticatedSessionUserAgentOsFromJSON(json: any): AuthenticatedSessionUserAgentOs {
    return AuthenticatedSessionUserAgentOsFromJSONTyped(json, false);
}

export function AuthenticatedSessionUserAgentOsFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatedSessionUserAgentOs {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'family': json['family'],
        'major': json['major'],
        'minor': json['minor'],
        'patch': json['patch'],
        'patchMinor': json['patch_minor'],
    };
}

export function AuthenticatedSessionUserAgentOsToJSON(value?: AuthenticatedSessionUserAgentOs | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'family': value.family,
        'major': value.major,
        'minor': value.minor,
        'patch': value.patch,
        'patch_minor': value.patchMinor,
    };
}
