// @goauthentik/api/src/models/AuthenticationEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `none` - None
 * * `require_authenticated` - Require Authenticated
 * * `require_unauthenticated` - Require Unauthenticated
 * * `require_superuser` - Require Superuser
 * * `require_outpost` - Require Outpost
 * @export
 */
export const AuthenticationEnum = {
    None: 'none',
    RequireAuthenticated: 'require_authenticated',
    RequireUnauthenticated: 'require_unauthenticated',
    RequireSuperuser: 'require_superuser',
    RequireOutpost: 'require_outpost',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type AuthenticationEnum = typeof AuthenticationEnum[keyof typeof AuthenticationEnum];


export function AuthenticationEnumFromJSON(json: any): AuthenticationEnum {
    return AuthenticationEnumFromJSONTyped(json, false);
}

export function AuthenticationEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticationEnum {
    return json as AuthenticationEnum;
}

export function AuthenticationEnumToJSON(value?: AuthenticationEnum | null): any {
    return value as any;
}
