// @goauthentik/api/src/models/AuthenticatorSMSChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * SMS Setup challenge
 * @export
 * @interface AuthenticatorSMSChallenge
 */
export interface AuthenticatorSMSChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof AuthenticatorSMSChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof AuthenticatorSMSChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorSMSChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof AuthenticatorSMSChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorSMSChallenge
     */
    pendingUser: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorSMSChallenge
     */
    pendingUserAvatar: string;
    /**
     * 
     * @type {boolean}
     * @memberof AuthenticatorSMSChallenge
     */
    phoneNumberRequired?: boolean;
}

/**
 * Check if a given object implements the AuthenticatorSMSChallenge interface.
 */
export function instanceOfAuthenticatorSMSChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "pendingUser" in value;
    isInstance = isInstance && "pendingUserAvatar" in value;

    return isInstance;
}

export function AuthenticatorSMSChallengeFromJSON(json: any): AuthenticatorSMSChallenge {
    return AuthenticatorSMSChallengeFromJSONTyped(json, false);
}

export function AuthenticatorSMSChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorSMSChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'pendingUser': json['pending_user'],
        'pendingUserAvatar': json['pending_user_avatar'],
        'phoneNumberRequired': !exists(json, 'phone_number_required') ? undefined : json['phone_number_required'],
    };
}

export function AuthenticatorSMSChallengeToJSON(value?: AuthenticatorSMSChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'pending_user': value.pendingUser,
        'pending_user_avatar': value.pendingUserAvatar,
        'phone_number_required': value.phoneNumberRequired,
    };
}
