// @goauthentik/api/src/models/AuthenticatorValidateStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DeviceClassesEnum } from './DeviceClassesEnum';
import {
    DeviceClassesEnumFromJSON,
    DeviceClassesEnumFromJSONTyped,
    DeviceClassesEnumToJSON,
} from './DeviceClassesEnum';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';
import type { NotConfiguredActionEnum } from './NotConfiguredActionEnum';
import {
    NotConfiguredActionEnumFromJSON,
    NotConfiguredActionEnumFromJSONTyped,
    NotConfiguredActionEnumToJSON,
} from './NotConfiguredActionEnum';
import type { UserVerificationEnum } from './UserVerificationEnum';
import {
    UserVerificationEnumFromJSON,
    UserVerificationEnumFromJSONTyped,
    UserVerificationEnumToJSON,
} from './UserVerificationEnum';

/**
 * AuthenticatorValidateStage Serializer
 * @export
 * @interface AuthenticatorValidateStage
 */
export interface AuthenticatorValidateStage {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof AuthenticatorValidateStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * 
     * @type {NotConfiguredActionEnum}
     * @memberof AuthenticatorValidateStage
     */
    notConfiguredAction?: NotConfiguredActionEnum;
    /**
     * Device classes which can be used to authenticate
     * @type {Array<DeviceClassesEnum>}
     * @memberof AuthenticatorValidateStage
     */
    deviceClasses?: Array<DeviceClassesEnum>;
    /**
     * Stages used to configure Authenticator when user doesn't have any compatible devices. After this configuration Stage passes, the user is not prompted again.
     * @type {Array<string>}
     * @memberof AuthenticatorValidateStage
     */
    configurationStages?: Array<string>;
    /**
     * If any of the user's device has been used within this threshold, this stage will be skipped
     * @type {string}
     * @memberof AuthenticatorValidateStage
     */
    lastAuthThreshold?: string;
    /**
     * 
     * @type {UserVerificationEnum}
     * @memberof AuthenticatorValidateStage
     */
    webauthnUserVerification?: UserVerificationEnum;
}

/**
 * Check if a given object implements the AuthenticatorValidateStage interface.
 */
export function instanceOfAuthenticatorValidateStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function AuthenticatorValidateStageFromJSON(json: any): AuthenticatorValidateStage {
    return AuthenticatorValidateStageFromJSONTyped(json, false);
}

export function AuthenticatorValidateStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorValidateStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'notConfiguredAction': !exists(json, 'not_configured_action') ? undefined : NotConfiguredActionEnumFromJSON(json['not_configured_action']),
        'deviceClasses': !exists(json, 'device_classes') ? undefined : ((json['device_classes'] as Array<any>).map(DeviceClassesEnumFromJSON)),
        'configurationStages': !exists(json, 'configuration_stages') ? undefined : json['configuration_stages'],
        'lastAuthThreshold': !exists(json, 'last_auth_threshold') ? undefined : json['last_auth_threshold'],
        'webauthnUserVerification': !exists(json, 'webauthn_user_verification') ? undefined : UserVerificationEnumFromJSON(json['webauthn_user_verification']),
    };
}

export function AuthenticatorValidateStageToJSON(value?: AuthenticatorValidateStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'not_configured_action': NotConfiguredActionEnumToJSON(value.notConfiguredAction),
        'device_classes': value.deviceClasses === undefined ? undefined : ((value.deviceClasses as Array<any>).map(DeviceClassesEnumToJSON)),
        'configuration_stages': value.configurationStages,
        'last_auth_threshold': value.lastAuthThreshold,
        'webauthn_user_verification': UserVerificationEnumToJSON(value.webauthnUserVerification),
    };
}
