// @goauthentik/api/src/models/AuthenticatorAttachmentEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `platform` - Platform
 * * `cross-platform` - Cross Platform
 * @export
 */
export const AuthenticatorAttachmentEnum = {
    Platform: 'platform',
    CrossPlatform: 'cross-platform',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type AuthenticatorAttachmentEnum = typeof AuthenticatorAttachmentEnum[keyof typeof AuthenticatorAttachmentEnum];


export function AuthenticatorAttachmentEnumFromJSON(json: any): AuthenticatorAttachmentEnum {
    return AuthenticatorAttachmentEnumFromJSONTyped(json, false);
}

export function AuthenticatorAttachmentEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorAttachmentEnum {
    return json as AuthenticatorAttachmentEnum;
}

export function AuthenticatorAttachmentEnumToJSON(value?: AuthenticatorAttachmentEnum | null): any {
    return value as any;
}
