// @goauthentik/api/src/models/AuthenticatorStaticStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * AuthenticatorStaticStage Serializer
 * @export
 * @interface AuthenticatorStaticStage
 */
export interface AuthenticatorStaticStage {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof AuthenticatorStaticStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorStaticStage
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorStaticStage
     */
    tokenCount?: number;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorStaticStage
     */
    tokenLength?: number;
}

/**
 * Check if a given object implements the AuthenticatorStaticStage interface.
 */
export function instanceOfAuthenticatorStaticStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function AuthenticatorStaticStageFromJSON(json: any): AuthenticatorStaticStage {
    return AuthenticatorStaticStageFromJSONTyped(json, false);
}

export function AuthenticatorStaticStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorStaticStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'tokenCount': !exists(json, 'token_count') ? undefined : json['token_count'],
        'tokenLength': !exists(json, 'token_length') ? undefined : json['token_length'],
    };
}

export function AuthenticatorStaticStageToJSON(value?: AuthenticatorStaticStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'token_count': value.tokenCount,
        'token_length': value.tokenLength,
    };
}
