// @goauthentik/api/src/models/AuthenticatorDuoChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Pseudo class for duo response
 * @export
 * @interface AuthenticatorDuoChallengeResponseRequest
 */
export interface AuthenticatorDuoChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the AuthenticatorDuoChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorDuoChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function AuthenticatorDuoChallengeResponseRequestFromJSON(json: any): AuthenticatorDuoChallengeResponseRequest {
    return AuthenticatorDuoChallengeResponseRequestFromJSONTyped(json, false);
}

export function AuthenticatorDuoChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorDuoChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function AuthenticatorDuoChallengeResponseRequestToJSON(value?: AuthenticatorDuoChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
