// @goauthentik/api/src/models/AuthenticatedSessionGeoIp.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Get GeoIP Data
 * @export
 * @interface AuthenticatedSessionGeoIp
 */
export interface AuthenticatedSessionGeoIp {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionGeoIp
     */
    continent: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionGeoIp
     */
    country: string;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatedSessionGeoIp
     */
    lat: number;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatedSessionGeoIp
     */
    _long: number;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionGeoIp
     */
    city: string;
}

/**
 * Check if a given object implements the AuthenticatedSessionGeoIp interface.
 */
export function instanceOfAuthenticatedSessionGeoIp(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "continent" in value;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "lat" in value;
    isInstance = isInstance && "_long" in value;
    isInstance = isInstance && "city" in value;

    return isInstance;
}

export function AuthenticatedSessionGeoIpFromJSON(json: any): AuthenticatedSessionGeoIp {
    return AuthenticatedSessionGeoIpFromJSONTyped(json, false);
}

export function AuthenticatedSessionGeoIpFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatedSessionGeoIp {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'continent': json['continent'],
        'country': json['country'],
        'lat': json['lat'],
        '_long': json['long'],
        'city': json['city'],
    };
}

export function AuthenticatedSessionGeoIpToJSON(value?: AuthenticatedSessionGeoIp | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'continent': value.continent,
        'country': value.country,
        'lat': value.lat,
        'long': value._long,
        'city': value.city,
    };
}
