// @goauthentik/api/src/models/AuthenticatedSessionUserAgentUserAgent.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * User agent browser
 * @export
 * @interface AuthenticatedSessionUserAgentUserAgent
 */
export interface AuthenticatedSessionUserAgentUserAgent {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentUserAgent
     */
    family: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentUserAgent
     */
    major: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentUserAgent
     */
    minor: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionUserAgentUserAgent
     */
    patch: string;
}

/**
 * Check if a given object implements the AuthenticatedSessionUserAgentUserAgent interface.
 */
export function instanceOfAuthenticatedSessionUserAgentUserAgent(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "family" in value;
    isInstance = isInstance && "major" in value;
    isInstance = isInstance && "minor" in value;
    isInstance = isInstance && "patch" in value;

    return isInstance;
}

export function AuthenticatedSessionUserAgentUserAgentFromJSON(json: any): AuthenticatedSessionUserAgentUserAgent {
    return AuthenticatedSessionUserAgentUserAgentFromJSONTyped(json, false);
}

export function AuthenticatedSessionUserAgentUserAgentFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatedSessionUserAgentUserAgent {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'family': json['family'],
        'major': json['major'],
        'minor': json['minor'],
        'patch': json['patch'],
    };
}

export function AuthenticatedSessionUserAgentUserAgentToJSON(value?: AuthenticatedSessionUserAgentUserAgent | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'family': value.family,
        'major': value.major,
        'minor': value.minor,
        'patch': value.patch,
    };
}
