// @goauthentik/api/src/models/AuthenticatorWebAuthnChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * WebAuthn Challenge
 * @export
 * @interface AuthenticatorWebAuthnChallenge
 */
export interface AuthenticatorWebAuthnChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    pendingUser: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    pendingUserAvatar: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof AuthenticatorWebAuthnChallenge
     */
    registration: { [key: string]: any; };
}

/**
 * Check if a given object implements the AuthenticatorWebAuthnChallenge interface.
 */
export function instanceOfAuthenticatorWebAuthnChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "pendingUser" in value;
    isInstance = isInstance && "pendingUserAvatar" in value;
    isInstance = isInstance && "registration" in value;

    return isInstance;
}

export function AuthenticatorWebAuthnChallengeFromJSON(json: any): AuthenticatorWebAuthnChallenge {
    return AuthenticatorWebAuthnChallengeFromJSONTyped(json, false);
}

export function AuthenticatorWebAuthnChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorWebAuthnChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'pendingUser': json['pending_user'],
        'pendingUserAvatar': json['pending_user_avatar'],
        'registration': json['registration'],
    };
}

export function AuthenticatorWebAuthnChallengeToJSON(value?: AuthenticatorWebAuthnChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'pending_user': value.pendingUser,
        'pending_user_avatar': value.pendingUserAvatar,
        'registration': value.registration,
    };
}
