// @goauthentik/api/src/models/AuthenticatorWebAuthnChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * WebAuthn Challenge response
 * @export
 * @interface AuthenticatorWebAuthnChallengeResponseRequest
 */
export interface AuthenticatorWebAuthnChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorWebAuthnChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof AuthenticatorWebAuthnChallengeResponseRequest
     */
    response: { [key: string]: any; };
}

/**
 * Check if a given object implements the AuthenticatorWebAuthnChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorWebAuthnChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "response" in value;

    return isInstance;
}

export function AuthenticatorWebAuthnChallengeResponseRequestFromJSON(json: any): AuthenticatorWebAuthnChallengeResponseRequest {
    return AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json, false);
}

export function AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorWebAuthnChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'response': json['response'],
    };
}

export function AuthenticatorWebAuthnChallengeResponseRequestToJSON(value?: AuthenticatorWebAuthnChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'response': value.response,
    };
}
