// @goauthentik/api/src/models/AuthenticatedSessionAsn.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Get ASN Data
 * @export
 * @interface AuthenticatedSessionAsn
 */
export interface AuthenticatedSessionAsn {
    /**
     * 
     * @type {number}
     * @memberof AuthenticatedSessionAsn
     */
    asn: number;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionAsn
     */
    asOrg: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatedSessionAsn
     */
    network: string | null;
}

/**
 * Check if a given object implements the AuthenticatedSessionAsn interface.
 */
export function instanceOfAuthenticatedSessionAsn(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "asn" in value;
    isInstance = isInstance && "asOrg" in value;
    isInstance = isInstance && "network" in value;

    return isInstance;
}

export function AuthenticatedSessionAsnFromJSON(json: any): AuthenticatedSessionAsn {
    return AuthenticatedSessionAsnFromJSONTyped(json, false);
}

export function AuthenticatedSessionAsnFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatedSessionAsn {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'asn': json['asn'],
        'asOrg': json['as_org'],
        'network': json['network'],
    };
}

export function AuthenticatedSessionAsnToJSON(value?: AuthenticatedSessionAsn | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'asn': value.asn,
        'as_org': value.asOrg,
        'network': value.network,
    };
}
