// @goauthentik/api/src/models/AuthenticatorDuoStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * AuthenticatorDuoStage Serializer
 * @export
 * @interface AuthenticatorDuoStage
 */
export interface AuthenticatorDuoStage {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof AuthenticatorDuoStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    clientId: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    apiHostname: string;
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorDuoStage
     */
    adminIntegrationKey?: string;
}

/**
 * Check if a given object implements the AuthenticatorDuoStage interface.
 */
export function instanceOfAuthenticatorDuoStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "apiHostname" in value;

    return isInstance;
}

export function AuthenticatorDuoStageFromJSON(json: any): AuthenticatorDuoStage {
    return AuthenticatorDuoStageFromJSONTyped(json, false);
}

export function AuthenticatorDuoStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorDuoStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'clientId': json['client_id'],
        'apiHostname': json['api_hostname'],
        'adminIntegrationKey': !exists(json, 'admin_integration_key') ? undefined : json['admin_integration_key'],
    };
}

export function AuthenticatorDuoStageToJSON(value?: AuthenticatorDuoStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'client_id': value.clientId,
        'api_hostname': value.apiHostname,
        'admin_integration_key': value.adminIntegrationKey,
    };
}
