// @goauthentik/api/src/models/AuthenticatorTOTPChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * TOTP Challenge response, device is set by get_response_instance
 * @export
 * @interface AuthenticatorTOTPChallengeResponseRequest
 */
export interface AuthenticatorTOTPChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof AuthenticatorTOTPChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {number}
     * @memberof AuthenticatorTOTPChallengeResponseRequest
     */
    code: number;
}

/**
 * Check if a given object implements the AuthenticatorTOTPChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorTOTPChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "code" in value;

    return isInstance;
}

export function AuthenticatorTOTPChallengeResponseRequestFromJSON(json: any): AuthenticatorTOTPChallengeResponseRequest {
    return AuthenticatorTOTPChallengeResponseRequestFromJSONTyped(json, false);
}

export function AuthenticatorTOTPChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): AuthenticatorTOTPChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'code': json['code'],
    };
}

export function AuthenticatorTOTPChallengeResponseRequestToJSON(value?: AuthenticatorTOTPChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'code': value.code,
    };
}
