// @goauthentik/api/src/models/AutosubmitChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * Autosubmit challenge used to send and navigate a POST request
 * @export
 * @interface AutosubmitChallenge
 */
export interface AutosubmitChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof AutosubmitChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof AutosubmitChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof AutosubmitChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof AutosubmitChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof AutosubmitChallenge
     */
    url: string;
    /**
     * 
     * @type {{ [key: string]: string; }}
     * @memberof AutosubmitChallenge
     */
    attrs: { [key: string]: string; };
    /**
     * 
     * @type {string}
     * @memberof AutosubmitChallenge
     */
    title?: string;
}

/**
 * Check if a given object implements the AutosubmitChallenge interface.
 */
export function instanceOfAutosubmitChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "attrs" in value;

    return isInstance;
}

export function AutosubmitChallengeFromJSON(json: any): AutosubmitChallenge {
    return AutosubmitChallengeFromJSONTyped(json, false);
}

export function AutosubmitChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): AutosubmitChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'url': json['url'],
        'attrs': json['attrs'],
        'title': !exists(json, 'title') ? undefined : json['title'],
    };
}

export function AutosubmitChallengeToJSON(value?: AutosubmitChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'url': value.url,
        'attrs': value.attrs,
        'title': value.title,
    };
}
