// @goauthentik/api/src/models/BackendsEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `authentik.core.auth.InbuiltBackend` - User database + standard password
 * * `authentik.core.auth.TokenBackend` - User database + app passwords
 * * `authentik.sources.ldap.auth.LDAPBackend` - User database + LDAP password
 * @export
 */
export const BackendsEnum = {
    CoreAuthInbuiltBackend: 'authentik.core.auth.InbuiltBackend',
    CoreAuthTokenBackend: 'authentik.core.auth.TokenBackend',
    SourcesLdapAuthLdapBackend: 'authentik.sources.ldap.auth.LDAPBackend',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type BackendsEnum = typeof BackendsEnum[keyof typeof BackendsEnum];


export function BackendsEnumFromJSON(json: any): BackendsEnum {
    return BackendsEnumFromJSONTyped(json, false);
}

export function BackendsEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): BackendsEnum {
    return json as BackendsEnum;
}

export function BackendsEnumToJSON(value?: BackendsEnum | null): any {
    return value as any;
}
