// @goauthentik/api/src/models/BindingTypeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `REDIRECT` - Redirect Binding
 * * `POST` - POST Binding
 * * `POST_AUTO` - POST Binding with auto-confirmation
 * @export
 */
export const BindingTypeEnum = {
    Redirect: 'REDIRECT',
    Post: 'POST',
    PostAuto: 'POST_AUTO',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type BindingTypeEnum = typeof BindingTypeEnum[keyof typeof BindingTypeEnum];


export function BindingTypeEnumFromJSON(json: any): BindingTypeEnum {
    return BindingTypeEnumFromJSONTyped(json, false);
}

export function BindingTypeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): BindingTypeEnum {
    return json as BindingTypeEnum;
}

export function BindingTypeEnumToJSON(value?: BindingTypeEnum | null): any {
    return value as any;
}
