// @goauthentik/api/src/models/BlueprintInstanceStatusEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `successful` - Successful
 * * `warning` - Warning
 * * `error` - Error
 * * `orphaned` - Orphaned
 * * `unknown` - Unknown
 * @export
 */
export const BlueprintInstanceStatusEnum = {
    Successful: 'successful',
    Warning: 'warning',
    Error: 'error',
    Orphaned: 'orphaned',
    Unknown: 'unknown',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type BlueprintInstanceStatusEnum = typeof BlueprintInstanceStatusEnum[keyof typeof BlueprintInstanceStatusEnum];


export function BlueprintInstanceStatusEnumFromJSON(json: any): BlueprintInstanceStatusEnum {
    return BlueprintInstanceStatusEnumFromJSONTyped(json, false);
}

export function BlueprintInstanceStatusEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): BlueprintInstanceStatusEnum {
    return json as BlueprintInstanceStatusEnum;
}

export function BlueprintInstanceStatusEnumToJSON(value?: BlueprintInstanceStatusEnum | null): any {
    return value as any;
}
