// @goauthentik/api/src/models/BlueprintInstanceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Info about a single blueprint instance file
 * @export
 * @interface BlueprintInstanceRequest
 */
export interface BlueprintInstanceRequest {
    /**
     * 
     * @type {string}
     * @memberof BlueprintInstanceRequest
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof BlueprintInstanceRequest
     */
    path?: string;
    /**
     * 
     * @type {any}
     * @memberof BlueprintInstanceRequest
     */
    context?: any | null;
    /**
     * 
     * @type {boolean}
     * @memberof BlueprintInstanceRequest
     */
    enabled?: boolean;
    /**
     * 
     * @type {string}
     * @memberof BlueprintInstanceRequest
     */
    content?: string;
}

/**
 * Check if a given object implements the BlueprintInstanceRequest interface.
 */
export function instanceOfBlueprintInstanceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function BlueprintInstanceRequestFromJSON(json: any): BlueprintInstanceRequest {
    return BlueprintInstanceRequestFromJSONTyped(json, false);
}

export function BlueprintInstanceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): BlueprintInstanceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'path': !exists(json, 'path') ? undefined : json['path'],
        'context': !exists(json, 'context') ? undefined : json['context'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'content': !exists(json, 'content') ? undefined : json['content'],
    };
}

export function BlueprintInstanceRequestToJSON(value?: BlueprintInstanceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
