// @goauthentik/api/src/models/Cache.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Generic cache stats for an object
 * @export
 * @interface Cache
 */
export interface Cache {
    /**
     * 
     * @type {number}
     * @memberof Cache
     */
    readonly count: number;
}

/**
 * Check if a given object implements the Cache interface.
 */
export function instanceOfCache(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "count" in value;

    return isInstance;
}

export function CacheFromJSON(json: any): Cache {
    return CacheFromJSONTyped(json, false);
}

export function CacheFromJSONTyped(json: any, ignoreDiscriminator: boolean): Cache {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'count': json['count'],
    };
}

export function CacheToJSON(value?: Cache | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
