// @goauthentik/api/src/models/CapabilitiesEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `can_save_media` - Can Save Media
 * * `can_geo_ip` - Can Geo Ip
 * * `can_asn` - Can Asn
 * * `can_impersonate` - Can Impersonate
 * * `can_debug` - Can Debug
 * * `is_enterprise` - Is Enterprise
 * @export
 */
export const CapabilitiesEnum = {
    CanSaveMedia: 'can_save_media',
    CanGeoIp: 'can_geo_ip',
    CanAsn: 'can_asn',
    CanImpersonate: 'can_impersonate',
    CanDebug: 'can_debug',
    IsEnterprise: 'is_enterprise',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type CapabilitiesEnum = typeof CapabilitiesEnum[keyof typeof CapabilitiesEnum];


export function CapabilitiesEnumFromJSON(json: any): CapabilitiesEnum {
    return CapabilitiesEnumFromJSONTyped(json, false);
}

export function CapabilitiesEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): CapabilitiesEnum {
    return json as CapabilitiesEnum;
}

export function CapabilitiesEnumToJSON(value?: CapabilitiesEnum | null): any {
    return value as any;
}
