// @goauthentik/api/src/models/CaptchaStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * CaptchaStage Serializer
 * @export
 * @interface CaptchaStage
 */
export interface CaptchaStage {
    /**
     * 
     * @type {string}
     * @memberof CaptchaStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof CaptchaStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof CaptchaStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof CaptchaStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof CaptchaStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof CaptchaStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Public key, acquired your captcha Provider.
     * @type {string}
     * @memberof CaptchaStage
     */
    publicKey: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaStage
     */
    jsUrl?: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaStage
     */
    apiUrl?: string;
}

/**
 * Check if a given object implements the CaptchaStage interface.
 */
export function instanceOfCaptchaStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "publicKey" in value;

    return isInstance;
}

export function CaptchaStageFromJSON(json: any): CaptchaStage {
    return CaptchaStageFromJSONTyped(json, false);
}

export function CaptchaStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): CaptchaStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'publicKey': json['public_key'],
        'jsUrl': !exists(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !exists(json, 'api_url') ? undefined : json['api_url'],
    };
}

export function CaptchaStageToJSON(value?: CaptchaStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'public_key': value.publicKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
