// @goauthentik/api/src/models/CaptchaStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * CaptchaStage Serializer
 * @export
 * @interface CaptchaStageRequest
 */
export interface CaptchaStageRequest {
    /**
     * 
     * @type {string}
     * @memberof CaptchaStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof CaptchaStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Public key, acquired your captcha Provider.
     * @type {string}
     * @memberof CaptchaStageRequest
     */
    publicKey: string;
    /**
     * Private key, acquired your captcha Provider.
     * @type {string}
     * @memberof CaptchaStageRequest
     */
    privateKey: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaStageRequest
     */
    jsUrl?: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaStageRequest
     */
    apiUrl?: string;
}

/**
 * Check if a given object implements the CaptchaStageRequest interface.
 */
export function instanceOfCaptchaStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "publicKey" in value;
    isInstance = isInstance && "privateKey" in value;

    return isInstance;
}

export function CaptchaStageRequestFromJSON(json: any): CaptchaStageRequest {
    return CaptchaStageRequestFromJSONTyped(json, false);
}

export function CaptchaStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): CaptchaStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'publicKey': json['public_key'],
        'privateKey': json['private_key'],
        'jsUrl': !exists(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !exists(json, 'api_url') ? undefined : json['api_url'],
    };
}

export function CaptchaStageRequestToJSON(value?: CaptchaStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'public_key': value.publicKey,
        'private_key': value.privateKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
