// @goauthentik/api/src/models/CaptchaChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Validate captcha token
 * @export
 * @interface CaptchaChallengeResponseRequest
 */
export interface CaptchaChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof CaptchaChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {string}
     * @memberof CaptchaChallengeResponseRequest
     */
    token: string;
}

/**
 * Check if a given object implements the CaptchaChallengeResponseRequest interface.
 */
export function instanceOfCaptchaChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "token" in value;

    return isInstance;
}

export function CaptchaChallengeResponseRequestFromJSON(json: any): CaptchaChallengeResponseRequest {
    return CaptchaChallengeResponseRequestFromJSONTyped(json, false);
}

export function CaptchaChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): CaptchaChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'token': json['token'],
    };
}

export function CaptchaChallengeResponseRequestToJSON(value?: CaptchaChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'token': value.token,
    };
}
