// @goauthentik/api/src/models/CertificateData.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Get CertificateKeyPair's data
 * @export
 * @interface CertificateData
 */
export interface CertificateData {
    /**
     * 
     * @type {string}
     * @memberof CertificateData
     */
    readonly data: string;
}

/**
 * Check if a given object implements the CertificateData interface.
 */
export function instanceOfCertificateData(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "data" in value;

    return isInstance;
}

export function CertificateDataFromJSON(json: any): CertificateData {
    return CertificateDataFromJSONTyped(json, false);
}

export function CertificateDataFromJSONTyped(json: any, ignoreDiscriminator: boolean): CertificateData {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'data': json['data'],
    };
}

export function CertificateDataToJSON(value?: CertificateData | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
