// @goauthentik/api/src/models/ChallengeChoices.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `native` - NATIVE
 * * `shell` - SHELL
 * * `redirect` - REDIRECT
 * @export
 */
export const ChallengeChoices = {
    Native: 'native',
    Shell: 'shell',
    Redirect: 'redirect',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ChallengeChoices = typeof ChallengeChoices[keyof typeof ChallengeChoices];


export function ChallengeChoicesFromJSON(json: any): ChallengeChoices {
    return ChallengeChoicesFromJSONTyped(json, false);
}

export function ChallengeChoicesFromJSONTyped(json: any, ignoreDiscriminator: boolean): ChallengeChoices {
    return json as ChallengeChoices;
}

export function ChallengeChoicesToJSON(value?: ChallengeChoices | null): any {
    return value as any;
}
