// @goauthentik/api/src/models/ClientTypeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `confidential` - Confidential
 * * `public` - Public
 * @export
 */
export const ClientTypeEnum = {
    Confidential: 'confidential',
    Public: 'public',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ClientTypeEnum = typeof ClientTypeEnum[keyof typeof ClientTypeEnum];


export function ClientTypeEnumFromJSON(json: any): ClientTypeEnum {
    return ClientTypeEnumFromJSONTyped(json, false);
}

export function ClientTypeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): ClientTypeEnum {
    return json as ClientTypeEnum;
}

export function ClientTypeEnumToJSON(value?: ClientTypeEnum | null): any {
    return value as any;
}
