// @goauthentik/api/src/models/ConsentStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ConsentStageModeEnum } from './ConsentStageModeEnum';
import {
    ConsentStageModeEnumFromJSON,
    ConsentStageModeEnumFromJSONTyped,
    ConsentStageModeEnumToJSON,
} from './ConsentStageModeEnum';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * ConsentStage Serializer
 * @export
 * @interface ConsentStage
 */
export interface ConsentStage {
    /**
     * 
     * @type {string}
     * @memberof ConsentStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof ConsentStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof ConsentStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof ConsentStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof ConsentStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof ConsentStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof ConsentStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * 
     * @type {ConsentStageModeEnum}
     * @memberof ConsentStage
     */
    mode?: ConsentStageModeEnum;
    /**
     * Offset after which consent expires. (Format: hours=1;minutes=2;seconds=3).
     * @type {string}
     * @memberof ConsentStage
     */
    consentExpireIn?: string;
}

/**
 * Check if a given object implements the ConsentStage interface.
 */
export function instanceOfConsentStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function ConsentStageFromJSON(json: any): ConsentStage {
    return ConsentStageFromJSONTyped(json, false);
}

export function ConsentStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): ConsentStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'mode': !exists(json, 'mode') ? undefined : ConsentStageModeEnumFromJSON(json['mode']),
        'consentExpireIn': !exists(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}

export function ConsentStageToJSON(value?: ConsentStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'mode': ConsentStageModeEnumToJSON(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
