// @goauthentik/api/src/models/ConsentStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ConsentStageModeEnum } from './ConsentStageModeEnum';
import {
    ConsentStageModeEnumFromJSON,
    ConsentStageModeEnumFromJSONTyped,
    ConsentStageModeEnumToJSON,
} from './ConsentStageModeEnum';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * ConsentStage Serializer
 * @export
 * @interface ConsentStageRequest
 */
export interface ConsentStageRequest {
    /**
     * 
     * @type {string}
     * @memberof ConsentStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof ConsentStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {ConsentStageModeEnum}
     * @memberof ConsentStageRequest
     */
    mode?: ConsentStageModeEnum;
    /**
     * Offset after which consent expires. (Format: hours=1;minutes=2;seconds=3).
     * @type {string}
     * @memberof ConsentStageRequest
     */
    consentExpireIn?: string;
}

/**
 * Check if a given object implements the ConsentStageRequest interface.
 */
export function instanceOfConsentStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function ConsentStageRequestFromJSON(json: any): ConsentStageRequest {
    return ConsentStageRequestFromJSONTyped(json, false);
}

export function ConsentStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ConsentStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'mode': !exists(json, 'mode') ? undefined : ConsentStageModeEnumFromJSON(json['mode']),
        'consentExpireIn': !exists(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}

export function ConsentStageRequestToJSON(value?: ConsentStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'mode': ConsentStageModeEnumToJSON(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
