// @goauthentik/api/src/models/ConsentStageModeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `always_require` - Always Require
 * * `permanent` - Permanent
 * * `expiring` - Expiring
 * @export
 */
export const ConsentStageModeEnum = {
    AlwaysRequire: 'always_require',
    Permanent: 'permanent',
    Expiring: 'expiring',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ConsentStageModeEnum = typeof ConsentStageModeEnum[keyof typeof ConsentStageModeEnum];


export function ConsentStageModeEnumFromJSON(json: any): ConsentStageModeEnum {
    return ConsentStageModeEnumFromJSONTyped(json, false);
}

export function ConsentStageModeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): ConsentStageModeEnum {
    return json as ConsentStageModeEnum;
}

export function ConsentStageModeEnumToJSON(value?: ConsentStageModeEnum | null): any {
    return value as any;
}
