// @goauthentik/api/src/models/ConsentChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Consent challenge response, any valid response request is valid
 * @export
 * @interface ConsentChallengeResponseRequest
 */
export interface ConsentChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof ConsentChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {string}
     * @memberof ConsentChallengeResponseRequest
     */
    token: string;
}

/**
 * Check if a given object implements the ConsentChallengeResponseRequest interface.
 */
export function instanceOfConsentChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "token" in value;

    return isInstance;
}

export function ConsentChallengeResponseRequestFromJSON(json: any): ConsentChallengeResponseRequest {
    return ConsentChallengeResponseRequestFromJSONTyped(json, false);
}

export function ConsentChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ConsentChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'token': json['token'],
    };
}

export function ConsentChallengeResponseRequestToJSON(value?: ConsentChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'token': value.token,
    };
}
