// @goauthentik/api/src/models/ConsentPermission.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Permission used for consent
 * @export
 * @interface ConsentPermission
 */
export interface ConsentPermission {
    /**
     * 
     * @type {string}
     * @memberof ConsentPermission
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof ConsentPermission
     */
    id: string;
}

/**
 * Check if a given object implements the ConsentPermission interface.
 */
export function instanceOfConsentPermission(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "id" in value;

    return isInstance;
}

export function ConsentPermissionFromJSON(json: any): ConsentPermission {
    return ConsentPermissionFromJSONTyped(json, false);
}

export function ConsentPermissionFromJSONTyped(json: any, ignoreDiscriminator: boolean): ConsentPermission {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'id': json['id'],
    };
}

export function ConsentPermissionToJSON(value?: ConsentPermission | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'id': value.id,
    };
}
