// @goauthentik/api/src/models/ContextualFlowInfoLayoutEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `stacked` - STACKED
 * * `content_left` - CONTENT_LEFT
 * * `content_right` - CONTENT_RIGHT
 * * `sidebar_left` - SIDEBAR_LEFT
 * * `sidebar_right` - SIDEBAR_RIGHT
 * @export
 */
export const ContextualFlowInfoLayoutEnum = {
    Stacked: 'stacked',
    ContentLeft: 'content_left',
    ContentRight: 'content_right',
    SidebarLeft: 'sidebar_left',
    SidebarRight: 'sidebar_right',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ContextualFlowInfoLayoutEnum = typeof ContextualFlowInfoLayoutEnum[keyof typeof ContextualFlowInfoLayoutEnum];


export function ContextualFlowInfoLayoutEnumFromJSON(json: any): ContextualFlowInfoLayoutEnum {
    return ContextualFlowInfoLayoutEnumFromJSONTyped(json, false);
}

export function ContextualFlowInfoLayoutEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): ContextualFlowInfoLayoutEnum {
    return json as ContextualFlowInfoLayoutEnum;
}

export function ContextualFlowInfoLayoutEnumToJSON(value?: ContextualFlowInfoLayoutEnum | null): any {
    return value as any;
}
