// @goauthentik/api/src/models/ContextualFlowInfo.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ContextualFlowInfoLayoutEnum } from './ContextualFlowInfoLayoutEnum';
import {
    ContextualFlowInfoLayoutEnumFromJSON,
    ContextualFlowInfoLayoutEnumFromJSONTyped,
    ContextualFlowInfoLayoutEnumToJSON,
} from './ContextualFlowInfoLayoutEnum';

/**
 * Contextual flow information for a challenge
 * @export
 * @interface ContextualFlowInfo
 */
export interface ContextualFlowInfo {
    /**
     * 
     * @type {string}
     * @memberof ContextualFlowInfo
     */
    title?: string;
    /**
     * 
     * @type {string}
     * @memberof ContextualFlowInfo
     */
    background?: string;
    /**
     * 
     * @type {string}
     * @memberof ContextualFlowInfo
     */
    cancelUrl: string;
    /**
     * 
     * @type {ContextualFlowInfoLayoutEnum}
     * @memberof ContextualFlowInfo
     */
    layout: ContextualFlowInfoLayoutEnum;
}

/**
 * Check if a given object implements the ContextualFlowInfo interface.
 */
export function instanceOfContextualFlowInfo(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "cancelUrl" in value;
    isInstance = isInstance && "layout" in value;

    return isInstance;
}

export function ContextualFlowInfoFromJSON(json: any): ContextualFlowInfo {
    return ContextualFlowInfoFromJSONTyped(json, false);
}

export function ContextualFlowInfoFromJSONTyped(json: any, ignoreDiscriminator: boolean): ContextualFlowInfo {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'title': !exists(json, 'title') ? undefined : json['title'],
        'background': !exists(json, 'background') ? undefined : json['background'],
        'cancelUrl': json['cancel_url'],
        'layout': ContextualFlowInfoLayoutEnumFromJSON(json['layout']),
    };
}

export function ContextualFlowInfoToJSON(value?: ContextualFlowInfo | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'title': value.title,
        'background': value.background,
        'cancel_url': value.cancelUrl,
        'layout': ContextualFlowInfoLayoutEnumToJSON(value.layout),
    };
}
