// @goauthentik/api/src/models/CurrentBrand.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FooterLink } from './FooterLink';
import {
    FooterLinkFromJSON,
    FooterLinkFromJSONTyped,
    FooterLinkToJSON,
} from './FooterLink';
import type { UiThemeEnum } from './UiThemeEnum';
import {
    UiThemeEnumFromJSON,
    UiThemeEnumFromJSONTyped,
    UiThemeEnumToJSON,
} from './UiThemeEnum';

/**
 * Partial brand information for styling
 * @export
 * @interface CurrentBrand
 */
export interface CurrentBrand {
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    matchedDomain: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    brandingTitle: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    brandingLogo: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    brandingFavicon: string;
    /**
     * 
     * @type {Array<FooterLink>}
     * @memberof CurrentBrand
     */
    readonly uiFooterLinks: Array<FooterLink>;
    /**
     * 
     * @type {UiThemeEnum}
     * @memberof CurrentBrand
     */
    readonly uiTheme: UiThemeEnum;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowAuthentication?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowInvalidation?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowRecovery?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowUnenrollment?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowUserSettings?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    flowDeviceCode?: string;
    /**
     * 
     * @type {string}
     * @memberof CurrentBrand
     */
    readonly defaultLocale: string;
}

/**
 * Check if a given object implements the CurrentBrand interface.
 */
export function instanceOfCurrentBrand(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "matchedDomain" in value;
    isInstance = isInstance && "brandingTitle" in value;
    isInstance = isInstance && "brandingLogo" in value;
    isInstance = isInstance && "brandingFavicon" in value;
    isInstance = isInstance && "uiFooterLinks" in value;
    isInstance = isInstance && "uiTheme" in value;
    isInstance = isInstance && "defaultLocale" in value;

    return isInstance;
}

export function CurrentBrandFromJSON(json: any): CurrentBrand {
    return CurrentBrandFromJSONTyped(json, false);
}

export function CurrentBrandFromJSONTyped(json: any, ignoreDiscriminator: boolean): CurrentBrand {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'matchedDomain': json['matched_domain'],
        'brandingTitle': json['branding_title'],
        'brandingLogo': json['branding_logo'],
        'brandingFavicon': json['branding_favicon'],
        'uiFooterLinks': ((json['ui_footer_links'] as Array<any>).map(FooterLinkFromJSON)),
        'uiTheme': UiThemeEnumFromJSON(json['ui_theme']),
        'flowAuthentication': !exists(json, 'flow_authentication') ? undefined : json['flow_authentication'],
        'flowInvalidation': !exists(json, 'flow_invalidation') ? undefined : json['flow_invalidation'],
        'flowRecovery': !exists(json, 'flow_recovery') ? undefined : json['flow_recovery'],
        'flowUnenrollment': !exists(json, 'flow_unenrollment') ? undefined : json['flow_unenrollment'],
        'flowUserSettings': !exists(json, 'flow_user_settings') ? undefined : json['flow_user_settings'],
        'flowDeviceCode': !exists(json, 'flow_device_code') ? undefined : json['flow_device_code'],
        'defaultLocale': json['default_locale'],
    };
}

export function CurrentBrandToJSON(value?: CurrentBrand | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'matched_domain': value.matchedDomain,
        'branding_title': value.brandingTitle,
        'branding_logo': value.brandingLogo,
        'branding_favicon': value.brandingFavicon,
        'flow_authentication': value.flowAuthentication,
        'flow_invalidation': value.flowInvalidation,
        'flow_recovery': value.flowRecovery,
        'flow_unenrollment': value.flowUnenrollment,
        'flow_user_settings': value.flowUserSettings,
        'flow_device_code': value.flowDeviceCode,
    };
}
