// @goauthentik/api/src/models/DeniedActionEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `message_continue` - Message Continue
 * * `message` - Message
 * * `continue` - Continue
 * @export
 */
export const DeniedActionEnum = {
    MessageContinue: 'message_continue',
    Message: 'message',
    Continue: 'continue',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type DeniedActionEnum = typeof DeniedActionEnum[keyof typeof DeniedActionEnum];


export function DeniedActionEnumFromJSON(json: any): DeniedActionEnum {
    return DeniedActionEnumFromJSONTyped(json, false);
}

export function DeniedActionEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): DeniedActionEnum {
    return json as DeniedActionEnum;
}

export function DeniedActionEnumToJSON(value?: DeniedActionEnum | null): any {
    return value as any;
}
