// @goauthentik/api/src/models/DenyStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * DenyStage Serializer
 * @export
 * @interface DenyStageRequest
 */
export interface DenyStageRequest {
    /**
     * 
     * @type {string}
     * @memberof DenyStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof DenyStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {string}
     * @memberof DenyStageRequest
     */
    denyMessage?: string;
}

/**
 * Check if a given object implements the DenyStageRequest interface.
 */
export function instanceOfDenyStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function DenyStageRequestFromJSON(json: any): DenyStageRequest {
    return DenyStageRequestFromJSONTyped(json, false);
}

export function DenyStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): DenyStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'denyMessage': !exists(json, 'deny_message') ? undefined : json['deny_message'],
    };
}

export function DenyStageRequestToJSON(value?: DenyStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'deny_message': value.denyMessage,
    };
}
