// @goauthentik/api/src/models/DeviceClassesEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `static` - Static
 * * `totp` - TOTP
 * * `webauthn` - WebAuthn
 * * `duo` - Duo
 * * `sms` - SMS
 * @export
 */
export const DeviceClassesEnum = {
    Static: 'static',
    Totp: 'totp',
    Webauthn: 'webauthn',
    Duo: 'duo',
    Sms: 'sms',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type DeviceClassesEnum = typeof DeviceClassesEnum[keyof typeof DeviceClassesEnum];


export function DeviceClassesEnumFromJSON(json: any): DeviceClassesEnum {
    return DeviceClassesEnumFromJSONTyped(json, false);
}

export function DeviceClassesEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): DeviceClassesEnum {
    return json as DeviceClassesEnum;
}

export function DeviceClassesEnumToJSON(value?: DeviceClassesEnum | null): any {
    return value as any;
}
