// @goauthentik/api/src/models/Device.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for Duo authenticator devices
 * @export
 * @interface Device
 */
export interface Device {
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof Device
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof Device
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof Device
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {number}
     * @memberof Device
     */
    pk: number;
    /**
     * 
     * @type {string}
     * @memberof Device
     */
    name: string;
    /**
     * Get type of device
     * @type {string}
     * @memberof Device
     */
    readonly type: string;
    /**
     * 
     * @type {boolean}
     * @memberof Device
     */
    confirmed: boolean;
}

/**
 * Check if a given object implements the Device interface.
 */
export function instanceOfDevice(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "confirmed" in value;

    return isInstance;
}

export function DeviceFromJSON(json: any): Device {
    return DeviceFromJSONTyped(json, false);
}

export function DeviceFromJSONTyped(json: any, ignoreDiscriminator: boolean): Device {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'pk': json['pk'],
        'name': json['name'],
        'type': json['type'],
        'confirmed': json['confirmed'],
    };
}

export function DeviceToJSON(value?: Device | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pk': value.pk,
        'name': value.name,
        'confirmed': value.confirmed,
    };
}
