// @goauthentik/api/src/models/DeviceChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Single device challenge
 * @export
 * @interface DeviceChallenge
 */
export interface DeviceChallenge {
    /**
     * 
     * @type {string}
     * @memberof DeviceChallenge
     */
    deviceClass: string;
    /**
     * 
     * @type {string}
     * @memberof DeviceChallenge
     */
    deviceUid: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof DeviceChallenge
     */
    challenge: { [key: string]: any; };
}

/**
 * Check if a given object implements the DeviceChallenge interface.
 */
export function instanceOfDeviceChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "deviceClass" in value;
    isInstance = isInstance && "deviceUid" in value;
    isInstance = isInstance && "challenge" in value;

    return isInstance;
}

export function DeviceChallengeFromJSON(json: any): DeviceChallenge {
    return DeviceChallengeFromJSONTyped(json, false);
}

export function DeviceChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): DeviceChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'deviceClass': json['device_class'],
        'deviceUid': json['device_uid'],
        'challenge': json['challenge'],
    };
}

export function DeviceChallengeToJSON(value?: DeviceChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'device_class': value.deviceClass,
        'device_uid': value.deviceUid,
        'challenge': value.challenge,
    };
}
