// @goauthentik/api/src/models/DockerServiceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * DockerServiceConnection Serializer
 * @export
 * @interface DockerServiceConnection
 */
export interface DockerServiceConnection {
    /**
     * 
     * @type {string}
     * @memberof DockerServiceConnection
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof DockerServiceConnection
     */
    name: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof DockerServiceConnection
     */
    local?: boolean;
    /**
     * 
     * @type {string}
     * @memberof DockerServiceConnection
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof DockerServiceConnection
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof DockerServiceConnection
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof DockerServiceConnection
     */
    readonly metaModelName: string;
    /**
     * Can be in the format of 'unix://<path>' when connecting to a local docker daemon, or 'https://<hostname>:2376' when connecting to a remote system.
     * @type {string}
     * @memberof DockerServiceConnection
     */
    url: string;
    /**
     * CA which the endpoint's Certificate is verified against. Can be left empty for no validation.
     * @type {string}
     * @memberof DockerServiceConnection
     */
    tlsVerification?: string | null;
    /**
     * Certificate/Key used for authentication. Can be left empty for no authentication.
     * @type {string}
     * @memberof DockerServiceConnection
     */
    tlsAuthentication?: string | null;
}

/**
 * Check if a given object implements the DockerServiceConnection interface.
 */
export function instanceOfDockerServiceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "url" in value;

    return isInstance;
}

export function DockerServiceConnectionFromJSON(json: any): DockerServiceConnection {
    return DockerServiceConnectionFromJSONTyped(json, false);
}

export function DockerServiceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): DockerServiceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'url': json['url'],
        'tlsVerification': !exists(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !exists(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}

export function DockerServiceConnectionToJSON(value?: DockerServiceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
