// @goauthentik/api/src/models/DummyPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Dummy Policy Serializer
 * @export
 * @interface DummyPolicyRequest
 */
export interface DummyPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof DummyPolicyRequest
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof DummyPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof DummyPolicyRequest
     */
    result?: boolean;
    /**
     * 
     * @type {number}
     * @memberof DummyPolicyRequest
     */
    waitMin?: number;
    /**
     * 
     * @type {number}
     * @memberof DummyPolicyRequest
     */
    waitMax?: number;
}

/**
 * Check if a given object implements the DummyPolicyRequest interface.
 */
export function instanceOfDummyPolicyRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function DummyPolicyRequestFromJSON(json: any): DummyPolicyRequest {
    return DummyPolicyRequestFromJSONTyped(json, false);
}

export function DummyPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): DummyPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'result': !exists(json, 'result') ? undefined : json['result'],
        'waitMin': !exists(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !exists(json, 'wait_max') ? undefined : json['wait_max'],
    };
}

export function DummyPolicyRequestToJSON(value?: DummyPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
