// @goauthentik/api/src/models/DummyChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Dummy challenge response
 * @export
 * @interface DummyChallengeResponseRequest
 */
export interface DummyChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof DummyChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the DummyChallengeResponseRequest interface.
 */
export function instanceOfDummyChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function DummyChallengeResponseRequestFromJSON(json: any): DummyChallengeResponseRequest {
    return DummyChallengeResponseRequestFromJSONTyped(json, false);
}

export function DummyChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): DummyChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function DummyChallengeResponseRequestToJSON(value?: DummyChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
