// @goauthentik/api/src/models/DummyPolicy.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Dummy Policy Serializer
 * @export
 * @interface DummyPolicy
 */
export interface DummyPolicy {
    /**
     * 
     * @type {string}
     * @memberof DummyPolicy
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof DummyPolicy
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof DummyPolicy
     */
    executionLogging?: boolean;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof DummyPolicy
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof DummyPolicy
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof DummyPolicy
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof DummyPolicy
     */
    readonly metaModelName: string;
    /**
     * Return objects policy is bound to
     * @type {number}
     * @memberof DummyPolicy
     */
    readonly boundTo: number;
    /**
     * 
     * @type {boolean}
     * @memberof DummyPolicy
     */
    result?: boolean;
    /**
     * 
     * @type {number}
     * @memberof DummyPolicy
     */
    waitMin?: number;
    /**
     * 
     * @type {number}
     * @memberof DummyPolicy
     */
    waitMax?: number;
}

/**
 * Check if a given object implements the DummyPolicy interface.
 */
export function instanceOfDummyPolicy(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;

    return isInstance;
}

export function DummyPolicyFromJSON(json: any): DummyPolicy {
    return DummyPolicyFromJSONTyped(json, false);
}

export function DummyPolicyFromJSONTyped(json: any, ignoreDiscriminator: boolean): DummyPolicy {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'result': !exists(json, 'result') ? undefined : json['result'],
        'waitMin': !exists(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !exists(json, 'wait_max') ? undefined : json['wait_max'],
    };
}

export function DummyPolicyToJSON(value?: DummyPolicy | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
