// @goauthentik/api/src/models/DuoDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for Duo authenticator devices
 * @export
 * @interface DuoDeviceRequest
 */
export interface DuoDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof DuoDeviceRequest
     */
    name: string;
}

/**
 * Check if a given object implements the DuoDeviceRequest interface.
 */
export function instanceOfDuoDeviceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function DuoDeviceRequestFromJSON(json: any): DuoDeviceRequest {
    return DuoDeviceRequestFromJSONTyped(json, false);
}

export function DuoDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): DuoDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
    };
}

export function DuoDeviceRequestToJSON(value?: DuoDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
