// @goauthentik/api/src/models/DuoDeviceEnrollmentStatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DuoResponseEnum } from './DuoResponseEnum';
import {
    DuoResponseEnumFromJSON,
    DuoResponseEnumFromJSONTyped,
    DuoResponseEnumToJSON,
} from './DuoResponseEnum';

/**
 * 
 * @export
 * @interface DuoDeviceEnrollmentStatus
 */
export interface DuoDeviceEnrollmentStatus {
    /**
     * 
     * @type {DuoResponseEnum}
     * @memberof DuoDeviceEnrollmentStatus
     */
    duoResponse: DuoResponseEnum;
}

/**
 * Check if a given object implements the DuoDeviceEnrollmentStatus interface.
 */
export function instanceOfDuoDeviceEnrollmentStatus(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "duoResponse" in value;

    return isInstance;
}

export function DuoDeviceEnrollmentStatusFromJSON(json: any): DuoDeviceEnrollmentStatus {
    return DuoDeviceEnrollmentStatusFromJSONTyped(json, false);
}

export function DuoDeviceEnrollmentStatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): DuoDeviceEnrollmentStatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'duoResponse': DuoResponseEnumFromJSON(json['duo_response']),
    };
}

export function DuoDeviceEnrollmentStatusToJSON(value?: DuoDeviceEnrollmentStatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'duo_response': DuoResponseEnumToJSON(value.duoResponse),
    };
}
