// @goauthentik/api/src/models/DuoDevice.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for Duo authenticator devices
 * @export
 * @interface DuoDevice
 */
export interface DuoDevice {
    /**
     * 
     * @type {number}
     * @memberof DuoDevice
     */
    readonly pk: number;
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof DuoDevice
     */
    name: string;
}

/**
 * Check if a given object implements the DuoDevice interface.
 */
export function instanceOfDuoDevice(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function DuoDeviceFromJSON(json: any): DuoDevice {
    return DuoDeviceFromJSONTyped(json, false);
}

export function DuoDeviceFromJSONTyped(json: any, ignoreDiscriminator: boolean): DuoDevice {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
    };
}

export function DuoDeviceToJSON(value?: DuoDevice | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
