// @goauthentik/api/src/models/DuoResponseEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `success` - Success
 * * `waiting` - Waiting
 * * `invalid` - Invalid
 * @export
 */
export const DuoResponseEnum = {
    Success: 'success',
    Waiting: 'waiting',
    Invalid: 'invalid',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type DuoResponseEnum = typeof DuoResponseEnum[keyof typeof DuoResponseEnum];


export function DuoResponseEnumFromJSON(json: any): DuoResponseEnum {
    return DuoResponseEnumFromJSONTyped(json, false);
}

export function DuoResponseEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): DuoResponseEnum {
    return json as DuoResponseEnum;
}

export function DuoResponseEnumToJSON(value?: DuoResponseEnum | null): any {
    return value as any;
}
