// @goauthentik/api/src/models/EmailStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * EmailStage Serializer
 * @export
 * @interface EmailStageRequest
 */
export interface EmailStageRequest {
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof EmailStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * When enabled, global Email connection settings will be used and connection settings below will be ignored.
     * @type {boolean}
     * @memberof EmailStageRequest
     */
    useGlobalSettings?: boolean;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    host?: string;
    /**
     * 
     * @type {number}
     * @memberof EmailStageRequest
     */
    port?: number;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    username?: string;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    password?: string;
    /**
     * 
     * @type {boolean}
     * @memberof EmailStageRequest
     */
    useTls?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof EmailStageRequest
     */
    useSsl?: boolean;
    /**
     * 
     * @type {number}
     * @memberof EmailStageRequest
     */
    timeout?: number;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    fromAddress?: string;
    /**
     * Time in minutes the token sent is valid.
     * @type {number}
     * @memberof EmailStageRequest
     */
    tokenExpiry?: number;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    subject?: string;
    /**
     * 
     * @type {string}
     * @memberof EmailStageRequest
     */
    template?: string;
    /**
     * Activate users upon completion of stage.
     * @type {boolean}
     * @memberof EmailStageRequest
     */
    activateUserOnSuccess?: boolean;
}

/**
 * Check if a given object implements the EmailStageRequest interface.
 */
export function instanceOfEmailStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function EmailStageRequestFromJSON(json: any): EmailStageRequest {
    return EmailStageRequestFromJSONTyped(json, false);
}

export function EmailStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): EmailStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'useGlobalSettings': !exists(json, 'use_global_settings') ? undefined : json['use_global_settings'],
        'host': !exists(json, 'host') ? undefined : json['host'],
        'port': !exists(json, 'port') ? undefined : json['port'],
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'useTls': !exists(json, 'use_tls') ? undefined : json['use_tls'],
        'useSsl': !exists(json, 'use_ssl') ? undefined : json['use_ssl'],
        'timeout': !exists(json, 'timeout') ? undefined : json['timeout'],
        'fromAddress': !exists(json, 'from_address') ? undefined : json['from_address'],
        'tokenExpiry': !exists(json, 'token_expiry') ? undefined : json['token_expiry'],
        'subject': !exists(json, 'subject') ? undefined : json['subject'],
        'template': !exists(json, 'template') ? undefined : json['template'],
        'activateUserOnSuccess': !exists(json, 'activate_user_on_success') ? undefined : json['activate_user_on_success'],
    };
}

export function EmailStageRequestToJSON(value?: EmailStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'use_global_settings': value.useGlobalSettings,
        'host': value.host,
        'port': value.port,
        'username': value.username,
        'password': value.password,
        'use_tls': value.useTls,
        'use_ssl': value.useSsl,
        'timeout': value.timeout,
        'from_address': value.fromAddress,
        'token_expiry': value.tokenExpiry,
        'subject': value.subject,
        'template': value.template,
        'activate_user_on_success': value.activateUserOnSuccess,
    };
}
