// @goauthentik/api/src/models/EmailChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Email challenge resposen. No fields. This challenge is
 * always declared invalid to give the user a chance to retry
 * @export
 * @interface EmailChallengeResponseRequest
 */
export interface EmailChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof EmailChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the EmailChallengeResponseRequest interface.
 */
export function instanceOfEmailChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function EmailChallengeResponseRequestFromJSON(json: any): EmailChallengeResponseRequest {
    return EmailChallengeResponseRequestFromJSONTyped(json, false);
}

export function EmailChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): EmailChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function EmailChallengeResponseRequestToJSON(value?: EmailChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
