// @goauthentik/api/src/models/ErrorDetail.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for rest_framework's error messages
 * @export
 * @interface ErrorDetail
 */
export interface ErrorDetail {
    /**
     * 
     * @type {string}
     * @memberof ErrorDetail
     */
    string: string;
    /**
     * 
     * @type {string}
     * @memberof ErrorDetail
     */
    code: string;
}

/**
 * Check if a given object implements the ErrorDetail interface.
 */
export function instanceOfErrorDetail(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "string" in value;
    isInstance = isInstance && "code" in value;

    return isInstance;
}

export function ErrorDetailFromJSON(json: any): ErrorDetail {
    return ErrorDetailFromJSONTyped(json, false);
}

export function ErrorDetailFromJSONTyped(json: any, ignoreDiscriminator: boolean): ErrorDetail {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'string': json['string'],
        'code': json['code'],
    };
}

export function ErrorDetailToJSON(value?: ErrorDetail | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'string': value.string,
        'code': value.code,
    };
}
