// @goauthentik/api/src/models/ErrorReportingConfig.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Config for error reporting
 * @export
 * @interface ErrorReportingConfig
 */
export interface ErrorReportingConfig {
    /**
     * 
     * @type {boolean}
     * @memberof ErrorReportingConfig
     */
    readonly enabled: boolean;
    /**
     * 
     * @type {string}
     * @memberof ErrorReportingConfig
     */
    readonly sentryDsn: string;
    /**
     * 
     * @type {string}
     * @memberof ErrorReportingConfig
     */
    readonly environment: string;
    /**
     * 
     * @type {boolean}
     * @memberof ErrorReportingConfig
     */
    readonly sendPii: boolean;
    /**
     * 
     * @type {number}
     * @memberof ErrorReportingConfig
     */
    readonly tracesSampleRate: number;
}

/**
 * Check if a given object implements the ErrorReportingConfig interface.
 */
export function instanceOfErrorReportingConfig(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "sentryDsn" in value;
    isInstance = isInstance && "environment" in value;
    isInstance = isInstance && "sendPii" in value;
    isInstance = isInstance && "tracesSampleRate" in value;

    return isInstance;
}

export function ErrorReportingConfigFromJSON(json: any): ErrorReportingConfig {
    return ErrorReportingConfigFromJSONTyped(json, false);
}

export function ErrorReportingConfigFromJSONTyped(json: any, ignoreDiscriminator: boolean): ErrorReportingConfig {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'enabled': json['enabled'],
        'sentryDsn': json['sentry_dsn'],
        'environment': json['environment'],
        'sendPii': json['send_pii'],
        'tracesSampleRate': json['traces_sample_rate'],
    };
}

export function ErrorReportingConfigToJSON(value?: ErrorReportingConfig | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
