// @goauthentik/api/src/models/EventMatcherPolicy.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AppEnum } from './AppEnum';
import {
    AppEnumFromJSON,
    AppEnumFromJSONTyped,
    AppEnumToJSON,
} from './AppEnum';
import type { EventActions } from './EventActions';
import {
    EventActionsFromJSON,
    EventActionsFromJSONTyped,
    EventActionsToJSON,
} from './EventActions';
import type { ModelEnum } from './ModelEnum';
import {
    ModelEnumFromJSON,
    ModelEnumFromJSONTyped,
    ModelEnumToJSON,
} from './ModelEnum';

/**
 * Event Matcher Policy Serializer
 * @export
 * @interface EventMatcherPolicy
 */
export interface EventMatcherPolicy {
    /**
     * 
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof EventMatcherPolicy
     */
    executionLogging?: boolean;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    readonly metaModelName: string;
    /**
     * Return objects policy is bound to
     * @type {number}
     * @memberof EventMatcherPolicy
     */
    readonly boundTo: number;
    /**
     * 
     * @type {EventActions}
     * @memberof EventMatcherPolicy
     */
    action?: EventActions | null;
    /**
     * Matches Event's Client IP (strict matching, for network matching use an Expression Policy)
     * @type {string}
     * @memberof EventMatcherPolicy
     */
    clientIp?: string | null;
    /**
     * 
     * @type {AppEnum}
     * @memberof EventMatcherPolicy
     */
    app?: AppEnum | null;
    /**
     * 
     * @type {ModelEnum}
     * @memberof EventMatcherPolicy
     */
    model?: ModelEnum | null;
}

/**
 * Check if a given object implements the EventMatcherPolicy interface.
 */
export function instanceOfEventMatcherPolicy(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;

    return isInstance;
}

export function EventMatcherPolicyFromJSON(json: any): EventMatcherPolicy {
    return EventMatcherPolicyFromJSONTyped(json, false);
}

export function EventMatcherPolicyFromJSONTyped(json: any, ignoreDiscriminator: boolean): EventMatcherPolicy {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'action': !exists(json, 'action') ? undefined : EventActionsFromJSON(json['action']),
        'clientIp': !exists(json, 'client_ip') ? undefined : json['client_ip'],
        'app': !exists(json, 'app') ? undefined : AppEnumFromJSON(json['app']),
        'model': !exists(json, 'model') ? undefined : ModelEnumFromJSON(json['model']),
    };
}

export function EventMatcherPolicyToJSON(value?: EventMatcherPolicy | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'action': EventActionsToJSON(value.action),
        'client_ip': value.clientIp,
        'app': AppEnumToJSON(value.app),
        'model': ModelEnumToJSON(value.model),
    };
}
