// @goauthentik/api/src/models/EventRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { EventActions } from './EventActions';
import {
    EventActionsFromJSON,
    EventActionsFromJSONTyped,
    EventActionsToJSON,
} from './EventActions';

/**
 * Event Serializer
 * @export
 * @interface EventRequest
 */
export interface EventRequest {
    /**
     * 
     * @type {any}
     * @memberof EventRequest
     */
    user?: any | null;
    /**
     * 
     * @type {EventActions}
     * @memberof EventRequest
     */
    action: EventActions;
    /**
     * 
     * @type {string}
     * @memberof EventRequest
     */
    app: string;
    /**
     * 
     * @type {any}
     * @memberof EventRequest
     */
    context?: any | null;
    /**
     * 
     * @type {string}
     * @memberof EventRequest
     */
    clientIp?: string | null;
    /**
     * 
     * @type {Date}
     * @memberof EventRequest
     */
    expires?: Date;
    /**
     * 
     * @type {any}
     * @memberof EventRequest
     */
    brand?: any | null;
}

/**
 * Check if a given object implements the EventRequest interface.
 */
export function instanceOfEventRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "action" in value;
    isInstance = isInstance && "app" in value;

    return isInstance;
}

export function EventRequestFromJSON(json: any): EventRequest {
    return EventRequestFromJSONTyped(json, false);
}

export function EventRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): EventRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': !exists(json, 'user') ? undefined : json['user'],
        'action': EventActionsFromJSON(json['action']),
        'app': json['app'],
        'context': !exists(json, 'context') ? undefined : json['context'],
        'clientIp': !exists(json, 'client_ip') ? undefined : json['client_ip'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'brand': !exists(json, 'brand') ? undefined : json['brand'],
    };
}

export function EventRequestToJSON(value?: EventRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'action': EventActionsToJSON(value.action),
        'app': value.app,
        'context': value.context,
        'client_ip': value.clientIp,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'brand': value.brand,
    };
}
