// @goauthentik/api/src/models/EventActions.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `login` - Login
 * * `login_failed` - Login Failed
 * * `logout` - Logout
 * * `user_write` - User Write
 * * `suspicious_request` - Suspicious Request
 * * `password_set` - Password Set
 * * `secret_view` - Secret View
 * * `secret_rotate` - Secret Rotate
 * * `invitation_used` - Invite Used
 * * `authorize_application` - Authorize Application
 * * `source_linked` - Source Linked
 * * `impersonation_started` - Impersonation Started
 * * `impersonation_ended` - Impersonation Ended
 * * `flow_execution` - Flow Execution
 * * `policy_execution` - Policy Execution
 * * `policy_exception` - Policy Exception
 * * `property_mapping_exception` - Property Mapping Exception
 * * `system_task_execution` - System Task Execution
 * * `system_task_exception` - System Task Exception
 * * `system_exception` - System Exception
 * * `configuration_error` - Configuration Error
 * * `model_created` - Model Created
 * * `model_updated` - Model Updated
 * * `model_deleted` - Model Deleted
 * * `email_sent` - Email Sent
 * * `update_available` - Update Available
 * * `custom_` - Custom Prefix
 * @export
 */
export const EventActions = {
    Login: 'login',
    LoginFailed: 'login_failed',
    Logout: 'logout',
    UserWrite: 'user_write',
    SuspiciousRequest: 'suspicious_request',
    PasswordSet: 'password_set',
    SecretView: 'secret_view',
    SecretRotate: 'secret_rotate',
    InvitationUsed: 'invitation_used',
    AuthorizeApplication: 'authorize_application',
    SourceLinked: 'source_linked',
    ImpersonationStarted: 'impersonation_started',
    ImpersonationEnded: 'impersonation_ended',
    FlowExecution: 'flow_execution',
    PolicyExecution: 'policy_execution',
    PolicyException: 'policy_exception',
    PropertyMappingException: 'property_mapping_exception',
    SystemTaskExecution: 'system_task_execution',
    SystemTaskException: 'system_task_exception',
    SystemException: 'system_exception',
    ConfigurationError: 'configuration_error',
    ModelCreated: 'model_created',
    ModelUpdated: 'model_updated',
    ModelDeleted: 'model_deleted',
    EmailSent: 'email_sent',
    UpdateAvailable: 'update_available',
    Custom: 'custom_',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type EventActions = typeof EventActions[keyof typeof EventActions];


export function EventActionsFromJSON(json: any): EventActions {
    return EventActionsFromJSONTyped(json, false);
}

export function EventActionsFromJSONTyped(json: any, ignoreDiscriminator: boolean): EventActions {
    return json as EventActions;
}

export function EventActionsToJSON(value?: EventActions | null): any {
    return value as any;
}
