// @goauthentik/api/src/models/ExpressionPolicy.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Group Membership Policy Serializer
 * @export
 * @interface ExpressionPolicy
 */
export interface ExpressionPolicy {
    /**
     * 
     * @type {string}
     * @memberof ExpressionPolicy
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof ExpressionPolicy
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof ExpressionPolicy
     */
    executionLogging?: boolean;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof ExpressionPolicy
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof ExpressionPolicy
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof ExpressionPolicy
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof ExpressionPolicy
     */
    readonly metaModelName: string;
    /**
     * Return objects policy is bound to
     * @type {number}
     * @memberof ExpressionPolicy
     */
    readonly boundTo: number;
    /**
     * 
     * @type {string}
     * @memberof ExpressionPolicy
     */
    expression: string;
}

/**
 * Check if a given object implements the ExpressionPolicy interface.
 */
export function instanceOfExpressionPolicy(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;
    isInstance = isInstance && "expression" in value;

    return isInstance;
}

export function ExpressionPolicyFromJSON(json: any): ExpressionPolicy {
    return ExpressionPolicyFromJSONTyped(json, false);
}

export function ExpressionPolicyFromJSONTyped(json: any, ignoreDiscriminator: boolean): ExpressionPolicy {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'expression': json['expression'],
    };
}

export function ExpressionPolicyToJSON(value?: ExpressionPolicy | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'expression': value.expression,
    };
}
