// @goauthentik/api/src/models/ExpressionPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Group Membership Policy Serializer
 * @export
 * @interface ExpressionPolicyRequest
 */
export interface ExpressionPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof ExpressionPolicyRequest
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof ExpressionPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {string}
     * @memberof ExpressionPolicyRequest
     */
    expression: string;
}

/**
 * Check if a given object implements the ExpressionPolicyRequest interface.
 */
export function instanceOfExpressionPolicyRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;

    return isInstance;
}

export function ExpressionPolicyRequestFromJSON(json: any): ExpressionPolicyRequest {
    return ExpressionPolicyRequestFromJSONTyped(json, false);
}

export function ExpressionPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ExpressionPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'expression': json['expression'],
    };
}

export function ExpressionPolicyRequestToJSON(value?: ExpressionPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'expression': value.expression,
    };
}
