// @goauthentik/api/src/models/ExtAuthStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * ExtAuthStage Serializer
 * @export
 * @interface ExtAuthStageRequest
 */
export interface ExtAuthStageRequest {
    /**
     * 
     * @type {string}
     * @memberof ExtAuthStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof ExtAuthStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
}

/**
 * Check if a given object implements the ExtAuthStageRequest interface.
 */
export function instanceOfExtAuthStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function ExtAuthStageRequestFromJSON(json: any): ExtAuthStageRequest {
    return ExtAuthStageRequestFromJSONTyped(json, false);
}

export function ExtAuthStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ExtAuthStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
    };
}

export function ExtAuthStageRequestToJSON(value?: ExtAuthStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
    };
}
