// @goauthentik/api/src/models/ExtraRoleObjectPermission.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * User permission with additional object-related data
 * @export
 * @interface ExtraRoleObjectPermission
 */
export interface ExtraRoleObjectPermission {
    /**
     * 
     * @type {number}
     * @memberof ExtraRoleObjectPermission
     */
    readonly id: number;
    /**
     * 
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly codename: string;
    /**
     * 
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly model: string;
    /**
     * 
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly appLabel: string;
    /**
     * 
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly objectPk: string;
    /**
     * 
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly name: string;
    /**
     * Get app label from permission's model
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly appLabelVerbose: string;
    /**
     * Get model label from permission's model
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly modelVerbose: string;
    /**
     * Get model description from attached model. This operation takes at least
     * one additional query, and the description is only shown if the user/role has the
     * view_ permission on the object
     * @type {string}
     * @memberof ExtraRoleObjectPermission
     */
    readonly objectDescription: string | null;
}

/**
 * Check if a given object implements the ExtraRoleObjectPermission interface.
 */
export function instanceOfExtraRoleObjectPermission(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "objectPk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "appLabelVerbose" in value;
    isInstance = isInstance && "modelVerbose" in value;
    isInstance = isInstance && "objectDescription" in value;

    return isInstance;
}

export function ExtraRoleObjectPermissionFromJSON(json: any): ExtraRoleObjectPermission {
    return ExtraRoleObjectPermissionFromJSONTyped(json, false);
}

export function ExtraRoleObjectPermissionFromJSONTyped(json: any, ignoreDiscriminator: boolean): ExtraRoleObjectPermission {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': json['id'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'objectPk': json['object_pk'],
        'name': json['name'],
        'appLabelVerbose': json['app_label_verbose'],
        'modelVerbose': json['model_verbose'],
        'objectDescription': json['object_description'],
    };
}

export function ExtraRoleObjectPermissionToJSON(value?: ExtraRoleObjectPermission | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
