// @goauthentik/api/src/models/FilePathRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer to upload file
 * @export
 * @interface FilePathRequest
 */
export interface FilePathRequest {
    /**
     * 
     * @type {string}
     * @memberof FilePathRequest
     */
    url: string;
}

/**
 * Check if a given object implements the FilePathRequest interface.
 */
export function instanceOfFilePathRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "url" in value;

    return isInstance;
}

export function FilePathRequestFromJSON(json: any): FilePathRequest {
    return FilePathRequestFromJSONTyped(json, false);
}

export function FilePathRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): FilePathRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'url': json['url'],
    };
}

export function FilePathRequestToJSON(value?: FilePathRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'url': value.url,
    };
}
