// @goauthentik/api/src/models/FlowSet.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DeniedActionEnum } from './DeniedActionEnum';
import {
    DeniedActionEnumFromJSON,
    DeniedActionEnumFromJSONTyped,
    DeniedActionEnumToJSON,
} from './DeniedActionEnum';
import type { FlowDesignationEnum } from './FlowDesignationEnum';
import {
    FlowDesignationEnumFromJSON,
    FlowDesignationEnumFromJSONTyped,
    FlowDesignationEnumToJSON,
} from './FlowDesignationEnum';
import type { FlowLayoutEnum } from './FlowLayoutEnum';
import {
    FlowLayoutEnumFromJSON,
    FlowLayoutEnumFromJSONTyped,
    FlowLayoutEnumToJSON,
} from './FlowLayoutEnum';
import type { PolicyEngineMode } from './PolicyEngineMode';
import {
    PolicyEngineModeFromJSON,
    PolicyEngineModeFromJSONTyped,
    PolicyEngineModeToJSON,
} from './PolicyEngineMode';

/**
 * Stripped down flow serializer
 * @export
 * @interface FlowSet
 */
export interface FlowSet {
    /**
     * 
     * @type {string}
     * @memberof FlowSet
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof FlowSet
     */
    readonly policybindingmodelPtrId: string;
    /**
     * 
     * @type {string}
     * @memberof FlowSet
     */
    name: string;
    /**
     * Visible in the URL.
     * @type {string}
     * @memberof FlowSet
     */
    slug: string;
    /**
     * Shown as the Title in Flow pages.
     * @type {string}
     * @memberof FlowSet
     */
    title: string;
    /**
     * 
     * @type {FlowDesignationEnum}
     * @memberof FlowSet
     */
    designation: FlowDesignationEnum;
    /**
     * Get the URL to the background image. If the name is /static or starts with http
     * it is returned as-is
     * @type {string}
     * @memberof FlowSet
     */
    readonly background: string;
    /**
     * 
     * @type {PolicyEngineMode}
     * @memberof FlowSet
     */
    policyEngineMode?: PolicyEngineMode;
    /**
     * Enable compatibility mode, increases compatibility with password managers on mobile devices.
     * @type {boolean}
     * @memberof FlowSet
     */
    compatibilityMode?: boolean;
    /**
     * Get export URL for flow
     * @type {string}
     * @memberof FlowSet
     */
    readonly exportUrl: string;
    /**
     * 
     * @type {FlowLayoutEnum}
     * @memberof FlowSet
     */
    layout?: FlowLayoutEnum;
    /**
     * 
     * @type {DeniedActionEnum}
     * @memberof FlowSet
     */
    deniedAction?: DeniedActionEnum;
}

/**
 * Check if a given object implements the FlowSet interface.
 */
export function instanceOfFlowSet(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "policybindingmodelPtrId" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "title" in value;
    isInstance = isInstance && "designation" in value;
    isInstance = isInstance && "background" in value;
    isInstance = isInstance && "exportUrl" in value;

    return isInstance;
}

export function FlowSetFromJSON(json: any): FlowSet {
    return FlowSetFromJSONTyped(json, false);
}

export function FlowSetFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowSet {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'policybindingmodelPtrId': json['policybindingmodel_ptr_id'],
        'name': json['name'],
        'slug': json['slug'],
        'title': json['title'],
        'designation': FlowDesignationEnumFromJSON(json['designation']),
        'background': json['background'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'compatibilityMode': !exists(json, 'compatibility_mode') ? undefined : json['compatibility_mode'],
        'exportUrl': json['export_url'],
        'layout': !exists(json, 'layout') ? undefined : FlowLayoutEnumFromJSON(json['layout']),
        'deniedAction': !exists(json, 'denied_action') ? undefined : DeniedActionEnumFromJSON(json['denied_action']),
    };
}

export function FlowSetToJSON(value?: FlowSet | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'slug': value.slug,
        'title': value.title,
        'designation': FlowDesignationEnumToJSON(value.designation),
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'compatibility_mode': value.compatibilityMode,
        'layout': FlowLayoutEnumToJSON(value.layout),
        'denied_action': DeniedActionEnumToJSON(value.deniedAction),
    };
}
