// @goauthentik/api/src/models/FlowDiagram.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * response of the flow's diagram action
 * @export
 * @interface FlowDiagram
 */
export interface FlowDiagram {
    /**
     * 
     * @type {string}
     * @memberof FlowDiagram
     */
    readonly diagram: string;
}

/**
 * Check if a given object implements the FlowDiagram interface.
 */
export function instanceOfFlowDiagram(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "diagram" in value;

    return isInstance;
}

export function FlowDiagramFromJSON(json: any): FlowDiagram {
    return FlowDiagramFromJSONTyped(json, false);
}

export function FlowDiagramFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowDiagram {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'diagram': json['diagram'],
    };
}

export function FlowDiagramToJSON(value?: FlowDiagram | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
