// @goauthentik/api/src/models/FlowLayoutEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `stacked` - Stacked
 * * `content_left` - Content Left
 * * `content_right` - Content Right
 * * `sidebar_left` - Sidebar Left
 * * `sidebar_right` - Sidebar Right
 * @export
 */
export const FlowLayoutEnum = {
    Stacked: 'stacked',
    ContentLeft: 'content_left',
    ContentRight: 'content_right',
    SidebarLeft: 'sidebar_left',
    SidebarRight: 'sidebar_right',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type FlowLayoutEnum = typeof FlowLayoutEnum[keyof typeof FlowLayoutEnum];


export function FlowLayoutEnumFromJSON(json: any): FlowLayoutEnum {
    return FlowLayoutEnumFromJSONTyped(json, false);
}

export function FlowLayoutEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowLayoutEnum {
    return json as FlowLayoutEnum;
}

export function FlowLayoutEnumToJSON(value?: FlowLayoutEnum | null): any {
    return value as any;
}
