// @goauthentik/api/src/models/FlowInspection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowInspectorPlan } from './FlowInspectorPlan';
import {
    FlowInspectorPlanFromJSON,
    FlowInspectorPlanFromJSONTyped,
    FlowInspectorPlanToJSON,
} from './FlowInspectorPlan';

/**
 * Serializer for inspect endpoint
 * @export
 * @interface FlowInspection
 */
export interface FlowInspection {
    /**
     * 
     * @type {Array<FlowInspectorPlan>}
     * @memberof FlowInspection
     */
    plans: Array<FlowInspectorPlan>;
    /**
     * 
     * @type {FlowInspectorPlan}
     * @memberof FlowInspection
     */
    currentPlan?: FlowInspectorPlan;
    /**
     * 
     * @type {boolean}
     * @memberof FlowInspection
     */
    isCompleted: boolean;
}

/**
 * Check if a given object implements the FlowInspection interface.
 */
export function instanceOfFlowInspection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "plans" in value;
    isInstance = isInstance && "isCompleted" in value;

    return isInstance;
}

export function FlowInspectionFromJSON(json: any): FlowInspection {
    return FlowInspectionFromJSONTyped(json, false);
}

export function FlowInspectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowInspection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'plans': ((json['plans'] as Array<any>).map(FlowInspectorPlanFromJSON)),
        'currentPlan': !exists(json, 'current_plan') ? undefined : FlowInspectorPlanFromJSON(json['current_plan']),
        'isCompleted': json['is_completed'],
    };
}

export function FlowInspectionToJSON(value?: FlowInspection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'plans': ((value.plans as Array<any>).map(FlowInspectorPlanToJSON)),
        'current_plan': FlowInspectorPlanToJSON(value.currentPlan),
        'is_completed': value.isCompleted,
    };
}
