// @goauthentik/api/src/models/FlowImportResult.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Logs of an attempted flow import
 * @export
 * @interface FlowImportResult
 */
export interface FlowImportResult {
    /**
     * 
     * @type {Array<{ [key: string]: any; }>}
     * @memberof FlowImportResult
     */
    readonly logs: Array<{ [key: string]: any; }>;
    /**
     * 
     * @type {boolean}
     * @memberof FlowImportResult
     */
    readonly success: boolean;
}

/**
 * Check if a given object implements the FlowImportResult interface.
 */
export function instanceOfFlowImportResult(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "logs" in value;
    isInstance = isInstance && "success" in value;

    return isInstance;
}

export function FlowImportResultFromJSON(json: any): FlowImportResult {
    return FlowImportResultFromJSONTyped(json, false);
}

export function FlowImportResultFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowImportResult {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'logs': json['logs'],
        'success': json['success'],
    };
}

export function FlowImportResultToJSON(value?: FlowImportResult | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
