// @goauthentik/api/src/models/FlowErrorChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * Challenge class when an unhandled error occurs during a stage. Normal users
 * are shown an error message, superusers are shown a full stacktrace.
 * @export
 * @interface FlowErrorChallenge
 */
export interface FlowErrorChallenge {
    /**
     * 
     * @type {string}
     * @memberof FlowErrorChallenge
     */
    type?: string;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof FlowErrorChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof FlowErrorChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof FlowErrorChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof FlowErrorChallenge
     */
    requestId: string;
    /**
     * 
     * @type {string}
     * @memberof FlowErrorChallenge
     */
    error?: string;
    /**
     * 
     * @type {string}
     * @memberof FlowErrorChallenge
     */
    traceback?: string;
}

/**
 * Check if a given object implements the FlowErrorChallenge interface.
 */
export function instanceOfFlowErrorChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "requestId" in value;

    return isInstance;
}

export function FlowErrorChallengeFromJSON(json: any): FlowErrorChallenge {
    return FlowErrorChallengeFromJSONTyped(json, false);
}

export function FlowErrorChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): FlowErrorChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': !exists(json, 'type') ? undefined : json['type'],
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'requestId': json['request_id'],
        'error': !exists(json, 'error') ? undefined : json['error'],
        'traceback': !exists(json, 'traceback') ? undefined : json['traceback'],
    };
}

export function FlowErrorChallengeToJSON(value?: FlowErrorChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': value.type,
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'request_id': value.requestId,
        'error': value.error,
        'traceback': value.traceback,
    };
}
