// @goauthentik/api/src/models/FooterLink.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Links returned in Config API
 * @export
 * @interface FooterLink
 */
export interface FooterLink {
    /**
     * 
     * @type {string}
     * @memberof FooterLink
     */
    readonly href: string;
    /**
     * 
     * @type {string}
     * @memberof FooterLink
     */
    readonly name: string;
}

/**
 * Check if a given object implements the FooterLink interface.
 */
export function instanceOfFooterLink(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "href" in value;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function FooterLinkFromJSON(json: any): FooterLink {
    return FooterLinkFromJSONTyped(json, false);
}

export function FooterLinkFromJSONTyped(json: any, ignoreDiscriminator: boolean): FooterLink {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'href': json['href'],
        'name': json['name'],
    };
}

export function FooterLinkToJSON(value?: FooterLink | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
