// @goauthentik/api/src/models/GroupMember.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Stripped down user serializer to show relevant users for groups
 * @export
 * @interface GroupMember
 */
export interface GroupMember {
    /**
     * 
     * @type {number}
     * @memberof GroupMember
     */
    readonly pk: number;
    /**
     * Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.
     * @type {string}
     * @memberof GroupMember
     */
    username: string;
    /**
     * User's display name.
     * @type {string}
     * @memberof GroupMember
     */
    name: string;
    /**
     * Designates whether this user should be treated as active. Unselect this instead of deleting accounts.
     * @type {boolean}
     * @memberof GroupMember
     */
    isActive?: boolean;
    /**
     * 
     * @type {Date}
     * @memberof GroupMember
     */
    lastLogin?: Date | null;
    /**
     * 
     * @type {string}
     * @memberof GroupMember
     */
    email?: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof GroupMember
     */
    attributes?: { [key: string]: any; };
    /**
     * 
     * @type {string}
     * @memberof GroupMember
     */
    readonly uid: string;
}

/**
 * Check if a given object implements the GroupMember interface.
 */
export function instanceOfGroupMember(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "uid" in value;

    return isInstance;
}

export function GroupMemberFromJSON(json: any): GroupMember {
    return GroupMemberFromJSONTyped(json, false);
}

export function GroupMemberFromJSONTyped(json: any, ignoreDiscriminator: boolean): GroupMember {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !exists(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !exists(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
    };
}

export function GroupMemberToJSON(value?: GroupMember | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'email': value.email,
        'attributes': value.attributes,
    };
}
