// @goauthentik/api/src/models/Group.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { GroupMember } from './GroupMember';
import {
    GroupMemberFromJSON,
    GroupMemberFromJSONTyped,
    GroupMemberToJSON,
} from './GroupMember';
import type { Role } from './Role';
import {
    RoleFromJSON,
    RoleFromJSONTyped,
    RoleToJSON,
} from './Role';

/**
 * Group Serializer
 * @export
 * @interface Group
 */
export interface Group {
    /**
     * 
     * @type {string}
     * @memberof Group
     */
    readonly pk: string;
    /**
     * 
     * @type {number}
     * @memberof Group
     */
    readonly numPk: number;
    /**
     * 
     * @type {string}
     * @memberof Group
     */
    name: string;
    /**
     * Users added to this group will be superusers.
     * @type {boolean}
     * @memberof Group
     */
    isSuperuser?: boolean;
    /**
     * 
     * @type {string}
     * @memberof Group
     */
    parent?: string | null;
    /**
     * 
     * @type {string}
     * @memberof Group
     */
    readonly parentName: string | null;
    /**
     * 
     * @type {Array<number>}
     * @memberof Group
     */
    users?: Array<number>;
    /**
     * 
     * @type {Array<GroupMember>}
     * @memberof Group
     */
    readonly usersObj: Array<GroupMember>;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof Group
     */
    attributes?: { [key: string]: any; };
    /**
     * 
     * @type {Array<string>}
     * @memberof Group
     */
    roles?: Array<string>;
    /**
     * 
     * @type {Array<Role>}
     * @memberof Group
     */
    readonly rolesObj: Array<Role>;
}

/**
 * Check if a given object implements the Group interface.
 */
export function instanceOfGroup(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "numPk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "parentName" in value;
    isInstance = isInstance && "usersObj" in value;
    isInstance = isInstance && "rolesObj" in value;

    return isInstance;
}

export function GroupFromJSON(json: any): Group {
    return GroupFromJSONTyped(json, false);
}

export function GroupFromJSONTyped(json: any, ignoreDiscriminator: boolean): Group {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'numPk': json['num_pk'],
        'name': json['name'],
        'isSuperuser': !exists(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !exists(json, 'parent') ? undefined : json['parent'],
        'parentName': json['parent_name'],
        'users': !exists(json, 'users') ? undefined : json['users'],
        'usersObj': ((json['users_obj'] as Array<any>).map(GroupMemberFromJSON)),
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'roles': !exists(json, 'roles') ? undefined : json['roles'],
        'rolesObj': ((json['roles_obj'] as Array<any>).map(RoleFromJSON)),
    };
}

export function GroupToJSON(value?: Group | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'users': value.users,
        'attributes': value.attributes,
        'roles': value.roles,
    };
}
