// @goauthentik/api/src/models/IdentificationChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Identification challenge
 * @export
 * @interface IdentificationChallengeResponseRequest
 */
export interface IdentificationChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof IdentificationChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {string}
     * @memberof IdentificationChallengeResponseRequest
     */
    uidField: string;
    /**
     * 
     * @type {string}
     * @memberof IdentificationChallengeResponseRequest
     */
    password?: string | null;
}

/**
 * Check if a given object implements the IdentificationChallengeResponseRequest interface.
 */
export function instanceOfIdentificationChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "uidField" in value;

    return isInstance;
}

export function IdentificationChallengeResponseRequestFromJSON(json: any): IdentificationChallengeResponseRequest {
    return IdentificationChallengeResponseRequestFromJSONTyped(json, false);
}

export function IdentificationChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): IdentificationChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'uidField': json['uid_field'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
}

export function IdentificationChallengeResponseRequestToJSON(value?: IdentificationChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'uid_field': value.uidField,
        'password': value.password,
    };
}
