// @goauthentik/api/src/models/InstallID.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface InstallID
 */
export interface InstallID {
    /**
     * 
     * @type {string}
     * @memberof InstallID
     */
    installId: string;
}

/**
 * Check if a given object implements the InstallID interface.
 */
export function instanceOfInstallID(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "installId" in value;

    return isInstance;
}

export function InstallIDFromJSON(json: any): InstallID {
    return InstallIDFromJSONTyped(json, false);
}

export function InstallIDFromJSONTyped(json: any, ignoreDiscriminator: boolean): InstallID {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'installId': json['install_id'],
    };
}

export function InstallIDToJSON(value?: InstallID | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'install_id': value.installId,
    };
}
