// @goauthentik/api/src/models/IntentEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `verification` - Intent Verification
 * * `api` - Intent Api
 * * `recovery` - Intent Recovery
 * * `app_password` - Intent App Password
 * @export
 */
export const IntentEnum = {
    Verification: 'verification',
    Api: 'api',
    Recovery: 'recovery',
    AppPassword: 'app_password',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type IntentEnum = typeof IntentEnum[keyof typeof IntentEnum];


export function IntentEnumFromJSON(json: any): IntentEnum {
    return IntentEnumFromJSONTyped(json, false);
}

export function IntentEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): IntentEnum {
    return json as IntentEnum;
}

export function IntentEnumToJSON(value?: IntentEnum | null): any {
    return value as any;
}
