// @goauthentik/api/src/models/InvalidResponseActionEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `retry` - Retry
 * * `restart` - Restart
 * * `restart_with_context` - Restart With Context
 * @export
 */
export const InvalidResponseActionEnum = {
    Retry: 'retry',
    Restart: 'restart',
    RestartWithContext: 'restart_with_context',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type InvalidResponseActionEnum = typeof InvalidResponseActionEnum[keyof typeof InvalidResponseActionEnum];


export function InvalidResponseActionEnumFromJSON(json: any): InvalidResponseActionEnum {
    return InvalidResponseActionEnumFromJSONTyped(json, false);
}

export function InvalidResponseActionEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): InvalidResponseActionEnum {
    return json as InvalidResponseActionEnum;
}

export function InvalidResponseActionEnumToJSON(value?: InvalidResponseActionEnum | null): any {
    return value as any;
}
