// @goauthentik/api/src/models/Invitation.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Flow } from './Flow';
import {
    FlowFromJSON,
    FlowFromJSONTyped,
    FlowToJSON,
} from './Flow';
import type { GroupMember } from './GroupMember';
import {
    GroupMemberFromJSON,
    GroupMemberFromJSONTyped,
    GroupMemberToJSON,
} from './GroupMember';

/**
 * Invitation Serializer
 * @export
 * @interface Invitation
 */
export interface Invitation {
    /**
     * 
     * @type {string}
     * @memberof Invitation
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof Invitation
     */
    name: string;
    /**
     * 
     * @type {Date}
     * @memberof Invitation
     */
    expires?: Date;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof Invitation
     */
    fixedData?: { [key: string]: any; };
    /**
     * 
     * @type {GroupMember}
     * @memberof Invitation
     */
    readonly createdBy: GroupMember;
    /**
     * When enabled, the invitation will be deleted after usage.
     * @type {boolean}
     * @memberof Invitation
     */
    singleUse?: boolean;
    /**
     * When set, only the configured flow can use this invitation.
     * @type {string}
     * @memberof Invitation
     */
    flow?: string | null;
    /**
     * 
     * @type {Flow}
     * @memberof Invitation
     */
    readonly flowObj: Flow;
}

/**
 * Check if a given object implements the Invitation interface.
 */
export function instanceOfInvitation(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "flowObj" in value;

    return isInstance;
}

export function InvitationFromJSON(json: any): Invitation {
    return InvitationFromJSONTyped(json, false);
}

export function InvitationFromJSONTyped(json: any, ignoreDiscriminator: boolean): Invitation {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !exists(json, 'fixed_data') ? undefined : json['fixed_data'],
        'createdBy': GroupMemberFromJSON(json['created_by']),
        'singleUse': !exists(json, 'single_use') ? undefined : json['single_use'],
        'flow': !exists(json, 'flow') ? undefined : json['flow'],
        'flowObj': FlowFromJSON(json['flow_obj']),
    };
}

export function InvitationToJSON(value?: Invitation | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
