// @goauthentik/api/src/models/IssuerModeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `global` - Same identifier is used for all providers
 * * `per_provider` - Each provider has a different issuer, based on the application slug.
 * @export
 */
export const IssuerModeEnum = {
    Global: 'global',
    PerProvider: 'per_provider',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type IssuerModeEnum = typeof IssuerModeEnum[keyof typeof IssuerModeEnum];


export function IssuerModeEnumFromJSON(json: any): IssuerModeEnum {
    return IssuerModeEnumFromJSONTyped(json, false);
}

export function IssuerModeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): IssuerModeEnum {
    return json as IssuerModeEnum;
}

export function IssuerModeEnumToJSON(value?: IssuerModeEnum | null): any {
    return value as any;
}
