// @goauthentik/api/src/models/KubernetesServiceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * KubernetesServiceConnection Serializer
 * @export
 * @interface KubernetesServiceConnection
 */
export interface KubernetesServiceConnection {
    /**
     * 
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    name: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof KubernetesServiceConnection
     */
    local?: boolean;
    /**
     * 
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof KubernetesServiceConnection
     */
    readonly metaModelName: string;
    /**
     * Paste your kubeconfig here. authentik will automatically use the currently selected context.
     * @type {any}
     * @memberof KubernetesServiceConnection
     */
    kubeconfig?: any | null;
    /**
     * Verify SSL Certificates of the Kubernetes API endpoint
     * @type {boolean}
     * @memberof KubernetesServiceConnection
     */
    verifySsl?: boolean;
}

/**
 * Check if a given object implements the KubernetesServiceConnection interface.
 */
export function instanceOfKubernetesServiceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function KubernetesServiceConnectionFromJSON(json: any): KubernetesServiceConnection {
    return KubernetesServiceConnectionFromJSONTyped(json, false);
}

export function KubernetesServiceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): KubernetesServiceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'kubeconfig': !exists(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !exists(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}

export function KubernetesServiceConnectionToJSON(value?: KubernetesServiceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
