// @goauthentik/api/src/models/KubernetesServiceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * KubernetesServiceConnection Serializer
 * @export
 * @interface KubernetesServiceConnectionRequest
 */
export interface KubernetesServiceConnectionRequest {
    /**
     * 
     * @type {string}
     * @memberof KubernetesServiceConnectionRequest
     */
    name: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof KubernetesServiceConnectionRequest
     */
    local?: boolean;
    /**
     * Paste your kubeconfig here. authentik will automatically use the currently selected context.
     * @type {any}
     * @memberof KubernetesServiceConnectionRequest
     */
    kubeconfig?: any | null;
    /**
     * Verify SSL Certificates of the Kubernetes API endpoint
     * @type {boolean}
     * @memberof KubernetesServiceConnectionRequest
     */
    verifySsl?: boolean;
}

/**
 * Check if a given object implements the KubernetesServiceConnectionRequest interface.
 */
export function instanceOfKubernetesServiceConnectionRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function KubernetesServiceConnectionRequestFromJSON(json: any): KubernetesServiceConnectionRequest {
    return KubernetesServiceConnectionRequestFromJSONTyped(json, false);
}

export function KubernetesServiceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): KubernetesServiceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'kubeconfig': !exists(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !exists(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}

export function KubernetesServiceConnectionRequestToJSON(value?: KubernetesServiceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
