// @goauthentik/api/src/models/LDAPOutpostConfig.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPAPIAccessMode } from './LDAPAPIAccessMode';
import {
    LDAPAPIAccessModeFromJSON,
    LDAPAPIAccessModeFromJSONTyped,
    LDAPAPIAccessModeToJSON,
} from './LDAPAPIAccessMode';

/**
 * LDAPProvider Serializer
 * @export
 * @interface LDAPOutpostConfig
 */
export interface LDAPOutpostConfig {
    /**
     * 
     * @type {number}
     * @memberof LDAPOutpostConfig
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    name: string;
    /**
     * DN under which objects are accessible.
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    baseDn?: string;
    /**
     * 
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    bindFlowSlug: string;
    /**
     * Prioritise backchannel slug over direct application slug
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    readonly applicationSlug: string;
    /**
     * Users in this group can do search queries. If not set, every user can execute search queries.
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    searchGroup?: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    certificate?: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPOutpostConfig
     */
    tlsServerName?: string;
    /**
     * The start for uidNumbers, this number is added to the user.pk to make sure that the numbers aren't too low for POSIX users. Default is 2000 to ensure that we don't collide with local users uidNumber
     * @type {number}
     * @memberof LDAPOutpostConfig
     */
    uidStartNumber?: number;
    /**
     * The start for gidNumbers, this number is added to a number generated from the group.pk to make sure that the numbers aren't too low for POSIX groups. Default is 4000 to ensure that we don't collide with local groups or users primary groups gidNumber
     * @type {number}
     * @memberof LDAPOutpostConfig
     */
    gidStartNumber?: number;
    /**
     * 
     * @type {LDAPAPIAccessMode}
     * @memberof LDAPOutpostConfig
     */
    searchMode?: LDAPAPIAccessMode;
    /**
     * 
     * @type {LDAPAPIAccessMode}
     * @memberof LDAPOutpostConfig
     */
    bindMode?: LDAPAPIAccessMode;
    /**
     * When enabled, code-based multi-factor authentication can be used by appending a semicolon and the TOTP code to the password. This should only be enabled if all users that will bind to this provider have a TOTP device configured, as otherwise a password may incorrectly be rejected if it contains a semicolon.
     * @type {boolean}
     * @memberof LDAPOutpostConfig
     */
    mfaSupport?: boolean;
}

/**
 * Check if a given object implements the LDAPOutpostConfig interface.
 */
export function instanceOfLDAPOutpostConfig(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "bindFlowSlug" in value;
    isInstance = isInstance && "applicationSlug" in value;

    return isInstance;
}

export function LDAPOutpostConfigFromJSON(json: any): LDAPOutpostConfig {
    return LDAPOutpostConfigFromJSONTyped(json, false);
}

export function LDAPOutpostConfigFromJSONTyped(json: any, ignoreDiscriminator: boolean): LDAPOutpostConfig {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'baseDn': !exists(json, 'base_dn') ? undefined : json['base_dn'],
        'bindFlowSlug': json['bind_flow_slug'],
        'applicationSlug': json['application_slug'],
        'searchGroup': !exists(json, 'search_group') ? undefined : json['search_group'],
        'certificate': !exists(json, 'certificate') ? undefined : json['certificate'],
        'tlsServerName': !exists(json, 'tls_server_name') ? undefined : json['tls_server_name'],
        'uidStartNumber': !exists(json, 'uid_start_number') ? undefined : json['uid_start_number'],
        'gidStartNumber': !exists(json, 'gid_start_number') ? undefined : json['gid_start_number'],
        'searchMode': !exists(json, 'search_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['search_mode']),
        'bindMode': !exists(json, 'bind_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['bind_mode']),
        'mfaSupport': !exists(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}

export function LDAPOutpostConfigToJSON(value?: LDAPOutpostConfig | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'base_dn': value.baseDn,
        'bind_flow_slug': value.bindFlowSlug,
        'search_group': value.searchGroup,
        'certificate': value.certificate,
        'tls_server_name': value.tlsServerName,
        'uid_start_number': value.uidStartNumber,
        'gid_start_number': value.gidStartNumber,
        'search_mode': LDAPAPIAccessModeToJSON(value.searchMode),
        'bind_mode': LDAPAPIAccessModeToJSON(value.bindMode),
        'mfa_support': value.mfaSupport,
    };
}
