// @goauthentik/api/src/models/LDAPPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * LDAP PropertyMapping Serializer
 * @export
 * @interface LDAPPropertyMappingRequest
 */
export interface LDAPPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof LDAPPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPPropertyMappingRequest
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof LDAPPropertyMappingRequest
     */
    expression: string;
    /**
     * 
     * @type {string}
     * @memberof LDAPPropertyMappingRequest
     */
    objectField: string;
}

/**
 * Check if a given object implements the LDAPPropertyMappingRequest interface.
 */
export function instanceOfLDAPPropertyMappingRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "objectField" in value;

    return isInstance;
}

export function LDAPPropertyMappingRequestFromJSON(json: any): LDAPPropertyMappingRequest {
    return LDAPPropertyMappingRequestFromJSONTyped(json, false);
}

export function LDAPPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): LDAPPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'objectField': json['object_field'],
    };
}

export function LDAPPropertyMappingRequestToJSON(value?: LDAPPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
