// @goauthentik/api/src/models/LDAPSource.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { PolicyEngineMode } from './PolicyEngineMode';
import {
    PolicyEngineModeFromJSON,
    PolicyEngineModeFromJSONTyped,
    PolicyEngineModeToJSON,
} from './PolicyEngineMode';
import type { UserMatchingModeEnum } from './UserMatchingModeEnum';
import {
    UserMatchingModeEnumFromJSON,
    UserMatchingModeEnumFromJSONTyped,
    UserMatchingModeEnumToJSON,
} from './UserMatchingModeEnum';

/**
 * LDAP Source Serializer
 * @export
 * @interface LDAPSource
 */
export interface LDAPSource {
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    readonly pk: string;
    /**
     * Source's display Name.
     * @type {string}
     * @memberof LDAPSource
     */
    name: string;
    /**
     * Internal source name, used in URLs.
     * @type {string}
     * @memberof LDAPSource
     */
    slug: string;
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSource
     */
    enabled?: boolean;
    /**
     * Flow to use when authenticating existing users.
     * @type {string}
     * @memberof LDAPSource
     */
    authenticationFlow?: string | null;
    /**
     * Flow to use when enrolling new users.
     * @type {string}
     * @memberof LDAPSource
     */
    enrollmentFlow?: string | null;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof LDAPSource
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof LDAPSource
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof LDAPSource
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof LDAPSource
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {PolicyEngineMode}
     * @memberof LDAPSource
     */
    policyEngineMode?: PolicyEngineMode;
    /**
     * 
     * @type {UserMatchingModeEnum}
     * @memberof LDAPSource
     */
    userMatchingMode?: UserMatchingModeEnum;
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof LDAPSource
     */
    readonly managed: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    userPathTemplate?: string;
    /**
     * Get the URL to the Icon. If the name is /static or
     * starts with http it is returned as-is
     * @type {string}
     * @memberof LDAPSource
     */
    readonly icon: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    serverUri: string;
    /**
     * Optionally verify the LDAP Server's Certificate against the CA Chain in this keypair.
     * @type {string}
     * @memberof LDAPSource
     */
    peerCertificate?: string | null;
    /**
     * Client certificate to authenticate against the LDAP Server's Certificate.
     * @type {string}
     * @memberof LDAPSource
     */
    clientCertificate?: string | null;
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    bindCn?: string;
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSource
     */
    startTls?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSource
     */
    sni?: boolean;
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    baseDn: string;
    /**
     * Prepended to Base DN for User-queries.
     * @type {string}
     * @memberof LDAPSource
     */
    additionalUserDn?: string;
    /**
     * Prepended to Base DN for Group-queries.
     * @type {string}
     * @memberof LDAPSource
     */
    additionalGroupDn?: string;
    /**
     * Consider Objects matching this filter to be Users.
     * @type {string}
     * @memberof LDAPSource
     */
    userObjectFilter?: string;
    /**
     * Consider Objects matching this filter to be Groups.
     * @type {string}
     * @memberof LDAPSource
     */
    groupObjectFilter?: string;
    /**
     * Field which contains members of a group.
     * @type {string}
     * @memberof LDAPSource
     */
    groupMembershipField?: string;
    /**
     * Field which contains a unique Identifier.
     * @type {string}
     * @memberof LDAPSource
     */
    objectUniquenessField?: string;
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSource
     */
    syncUsers?: boolean;
    /**
     * When a user changes their password, sync it back to LDAP. This can only be enabled on a single LDAP source.
     * @type {boolean}
     * @memberof LDAPSource
     */
    syncUsersPassword?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSource
     */
    syncGroups?: boolean;
    /**
     * 
     * @type {string}
     * @memberof LDAPSource
     */
    syncParentGroup?: string | null;
    /**
     * 
     * @type {Array<string>}
     * @memberof LDAPSource
     */
    propertyMappings?: Array<string>;
    /**
     * Property mappings used for group creation/updating.
     * @type {Array<string>}
     * @memberof LDAPSource
     */
    propertyMappingsGroup?: Array<string>;
    /**
     * Get cached source connectivity
     * @type {{ [key: string]: { [key: string]: string; }; }}
     * @memberof LDAPSource
     */
    readonly connectivity: { [key: string]: { [key: string]: string; }; } | null;
}

/**
 * Check if a given object implements the LDAPSource interface.
 */
export function instanceOfLDAPSource(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "managed" in value;
    isInstance = isInstance && "icon" in value;
    isInstance = isInstance && "serverUri" in value;
    isInstance = isInstance && "baseDn" in value;
    isInstance = isInstance && "connectivity" in value;

    return isInstance;
}

export function LDAPSourceFromJSON(json: any): LDAPSource {
    return LDAPSourceFromJSONTyped(json, false);
}

export function LDAPSourceFromJSONTyped(json: any, ignoreDiscriminator: boolean): LDAPSource {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'slug': json['slug'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'enrollmentFlow': !exists(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'userMatchingMode': !exists(json, 'user_matching_mode') ? undefined : UserMatchingModeEnumFromJSON(json['user_matching_mode']),
        'managed': json['managed'],
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
        'icon': json['icon'],
        'serverUri': json['server_uri'],
        'peerCertificate': !exists(json, 'peer_certificate') ? undefined : json['peer_certificate'],
        'clientCertificate': !exists(json, 'client_certificate') ? undefined : json['client_certificate'],
        'bindCn': !exists(json, 'bind_cn') ? undefined : json['bind_cn'],
        'startTls': !exists(json, 'start_tls') ? undefined : json['start_tls'],
        'sni': !exists(json, 'sni') ? undefined : json['sni'],
        'baseDn': json['base_dn'],
        'additionalUserDn': !exists(json, 'additional_user_dn') ? undefined : json['additional_user_dn'],
        'additionalGroupDn': !exists(json, 'additional_group_dn') ? undefined : json['additional_group_dn'],
        'userObjectFilter': !exists(json, 'user_object_filter') ? undefined : json['user_object_filter'],
        'groupObjectFilter': !exists(json, 'group_object_filter') ? undefined : json['group_object_filter'],
        'groupMembershipField': !exists(json, 'group_membership_field') ? undefined : json['group_membership_field'],
        'objectUniquenessField': !exists(json, 'object_uniqueness_field') ? undefined : json['object_uniqueness_field'],
        'syncUsers': !exists(json, 'sync_users') ? undefined : json['sync_users'],
        'syncUsersPassword': !exists(json, 'sync_users_password') ? undefined : json['sync_users_password'],
        'syncGroups': !exists(json, 'sync_groups') ? undefined : json['sync_groups'],
        'syncParentGroup': !exists(json, 'sync_parent_group') ? undefined : json['sync_parent_group'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !exists(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'connectivity': json['connectivity'],
    };
}

export function LDAPSourceToJSON(value?: LDAPSource | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'slug': value.slug,
        'enabled': value.enabled,
        'authentication_flow': value.authenticationFlow,
        'enrollment_flow': value.enrollmentFlow,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'user_matching_mode': UserMatchingModeEnumToJSON(value.userMatchingMode),
        'user_path_template': value.userPathTemplate,
        'server_uri': value.serverUri,
        'peer_certificate': value.peerCertificate,
        'client_certificate': value.clientCertificate,
        'bind_cn': value.bindCn,
        'start_tls': value.startTls,
        'sni': value.sni,
        'base_dn': value.baseDn,
        'additional_user_dn': value.additionalUserDn,
        'additional_group_dn': value.additionalGroupDn,
        'user_object_filter': value.userObjectFilter,
        'group_object_filter': value.groupObjectFilter,
        'group_membership_field': value.groupMembershipField,
        'object_uniqueness_field': value.objectUniquenessField,
        'sync_users': value.syncUsers,
        'sync_users_password': value.syncUsersPassword,
        'sync_groups': value.syncGroups,
        'sync_parent_group': value.syncParentGroup,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
    };
}
