// @goauthentik/api/src/models/LDAPAPIAccessMode.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `direct` - Direct
 * * `cached` - Cached
 * @export
 */
export const LDAPAPIAccessMode = {
    Direct: 'direct',
    Cached: 'cached',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type LDAPAPIAccessMode = typeof LDAPAPIAccessMode[keyof typeof LDAPAPIAccessMode];


export function LDAPAPIAccessModeFromJSON(json: any): LDAPAPIAccessMode {
    return LDAPAPIAccessModeFromJSONTyped(json, false);
}

export function LDAPAPIAccessModeFromJSONTyped(json: any, ignoreDiscriminator: boolean): LDAPAPIAccessMode {
    return json as LDAPAPIAccessMode;
}

export function LDAPAPIAccessModeToJSON(value?: LDAPAPIAccessMode | null): any {
    return value as any;
}
