// @goauthentik/api/src/models/LDAPSyncStatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { SystemTask } from './SystemTask';
import {
    SystemTaskFromJSON,
    SystemTaskFromJSONTyped,
    SystemTaskToJSON,
} from './SystemTask';

/**
 * LDAP Source sync status
 * @export
 * @interface LDAPSyncStatus
 */
export interface LDAPSyncStatus {
    /**
     * 
     * @type {boolean}
     * @memberof LDAPSyncStatus
     */
    readonly isRunning: boolean;
    /**
     * 
     * @type {Array<SystemTask>}
     * @memberof LDAPSyncStatus
     */
    readonly tasks: Array<SystemTask>;
}

/**
 * Check if a given object implements the LDAPSyncStatus interface.
 */
export function instanceOfLDAPSyncStatus(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "isRunning" in value;
    isInstance = isInstance && "tasks" in value;

    return isInstance;
}

export function LDAPSyncStatusFromJSON(json: any): LDAPSyncStatus {
    return LDAPSyncStatusFromJSONTyped(json, false);
}

export function LDAPSyncStatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): LDAPSyncStatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'isRunning': json['is_running'],
        'tasks': ((json['tasks'] as Array<any>).map(SystemTaskFromJSON)),
    };
}

export function LDAPSyncStatusToJSON(value?: LDAPSyncStatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
