// @goauthentik/api/src/models/LicenseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * License Serializer
 * @export
 * @interface LicenseRequest
 */
export interface LicenseRequest {
    /**
     * 
     * @type {string}
     * @memberof LicenseRequest
     */
    key: string;
}

/**
 * Check if a given object implements the LicenseRequest interface.
 */
export function instanceOfLicenseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "key" in value;

    return isInstance;
}

export function LicenseRequestFromJSON(json: any): LicenseRequest {
    return LicenseRequestFromJSONTyped(json, false);
}

export function LicenseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): LicenseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'key': json['key'],
    };
}

export function LicenseRequestToJSON(value?: LicenseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'key': value.key,
    };
}
