// @goauthentik/api/src/models/License.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * License Serializer
 * @export
 * @interface License
 */
export interface License {
    /**
     * 
     * @type {string}
     * @memberof License
     */
    readonly licenseUuid: string;
    /**
     * 
     * @type {string}
     * @memberof License
     */
    readonly name: string;
    /**
     * 
     * @type {string}
     * @memberof License
     */
    key: string;
    /**
     * 
     * @type {Date}
     * @memberof License
     */
    readonly expiry: Date;
    /**
     * 
     * @type {number}
     * @memberof License
     */
    readonly internalUsers: number;
    /**
     * 
     * @type {number}
     * @memberof License
     */
    readonly externalUsers: number;
}

/**
 * Check if a given object implements the License interface.
 */
export function instanceOfLicense(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "licenseUuid" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "expiry" in value;
    isInstance = isInstance && "internalUsers" in value;
    isInstance = isInstance && "externalUsers" in value;

    return isInstance;
}

export function LicenseFromJSON(json: any): License {
    return LicenseFromJSONTyped(json, false);
}

export function LicenseFromJSONTyped(json: any, ignoreDiscriminator: boolean): License {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'licenseUuid': json['license_uuid'],
        'name': json['name'],
        'key': json['key'],
        'expiry': (new Date(json['expiry'])),
        'internalUsers': json['internal_users'],
        'externalUsers': json['external_users'],
    };
}

export function LicenseToJSON(value?: License | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'key': value.key,
    };
}
