// @goauthentik/api/src/models/NameIdPolicyEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress` - Email
 * * `urn:oasis:names:tc:SAML:2.0:nameid-format:persistent` - Persistent
 * * `urn:oasis:names:tc:SAML:2.0:nameid-format:X509SubjectName` - X509
 * * `urn:oasis:names:tc:SAML:2.0:nameid-format:WindowsDomainQualifiedName` - Windows
 * * `urn:oasis:names:tc:SAML:2.0:nameid-format:transient` - Transient
 * @export
 */
export const NameIdPolicyEnum = {
    _11nameidFormatemailAddress: 'urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress',
    _20nameidFormatpersistent: 'urn:oasis:names:tc:SAML:2.0:nameid-format:persistent',
    _20nameidFormatX509SubjectName: 'urn:oasis:names:tc:SAML:2.0:nameid-format:X509SubjectName',
    _20nameidFormatWindowsDomainQualifiedName: 'urn:oasis:names:tc:SAML:2.0:nameid-format:WindowsDomainQualifiedName',
    _20nameidFormattransient: 'urn:oasis:names:tc:SAML:2.0:nameid-format:transient',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type NameIdPolicyEnum = typeof NameIdPolicyEnum[keyof typeof NameIdPolicyEnum];


export function NameIdPolicyEnumFromJSON(json: any): NameIdPolicyEnum {
    return NameIdPolicyEnumFromJSONTyped(json, false);
}

export function NameIdPolicyEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): NameIdPolicyEnum {
    return json as NameIdPolicyEnum;
}

export function NameIdPolicyEnumToJSON(value?: NameIdPolicyEnum | null): any {
    return value as any;
}
