// @goauthentik/api/src/models/NetworkBindingEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `no_binding` - No Binding
 * * `bind_asn` - Bind Asn
 * * `bind_asn_network` - Bind Asn Network
 * * `bind_asn_network_ip` - Bind Asn Network Ip
 * @export
 */
export const NetworkBindingEnum = {
    NoBinding: 'no_binding',
    BindAsn: 'bind_asn',
    BindAsnNetwork: 'bind_asn_network',
    BindAsnNetworkIp: 'bind_asn_network_ip',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type NetworkBindingEnum = typeof NetworkBindingEnum[keyof typeof NetworkBindingEnum];


export function NetworkBindingEnumFromJSON(json: any): NetworkBindingEnum {
    return NetworkBindingEnumFromJSONTyped(json, false);
}

export function NetworkBindingEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): NetworkBindingEnum {
    return json as NetworkBindingEnum;
}

export function NetworkBindingEnumToJSON(value?: NetworkBindingEnum | null): any {
    return value as any;
}
