// @goauthentik/api/src/models/NotConfiguredActionEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `skip` - Skip
 * * `deny` - Deny
 * * `configure` - Configure
 * @export
 */
export const NotConfiguredActionEnum = {
    Skip: 'skip',
    Deny: 'deny',
    Configure: 'configure',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type NotConfiguredActionEnum = typeof NotConfiguredActionEnum[keyof typeof NotConfiguredActionEnum];


export function NotConfiguredActionEnumFromJSON(json: any): NotConfiguredActionEnum {
    return NotConfiguredActionEnumFromJSONTyped(json, false);
}

export function NotConfiguredActionEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotConfiguredActionEnum {
    return json as NotConfiguredActionEnum;
}

export function NotConfiguredActionEnumToJSON(value?: NotConfiguredActionEnum | null): any {
    return value as any;
}
