// @goauthentik/api/src/models/NotificationWebhookMapping.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * NotificationWebhookMapping Serializer
 * @export
 * @interface NotificationWebhookMapping
 */
export interface NotificationWebhookMapping {
    /**
     * 
     * @type {string}
     * @memberof NotificationWebhookMapping
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof NotificationWebhookMapping
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof NotificationWebhookMapping
     */
    expression: string;
}

/**
 * Check if a given object implements the NotificationWebhookMapping interface.
 */
export function instanceOfNotificationWebhookMapping(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;

    return isInstance;
}

export function NotificationWebhookMappingFromJSON(json: any): NotificationWebhookMapping {
    return NotificationWebhookMappingFromJSONTyped(json, false);
}

export function NotificationWebhookMappingFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationWebhookMapping {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'expression': json['expression'],
    };
}

export function NotificationWebhookMappingToJSON(value?: NotificationWebhookMapping | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'expression': value.expression,
    };
}
