// @goauthentik/api/src/models/NotificationWebhookMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * NotificationWebhookMapping Serializer
 * @export
 * @interface NotificationWebhookMappingRequest
 */
export interface NotificationWebhookMappingRequest {
    /**
     * 
     * @type {string}
     * @memberof NotificationWebhookMappingRequest
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof NotificationWebhookMappingRequest
     */
    expression: string;
}

/**
 * Check if a given object implements the NotificationWebhookMappingRequest interface.
 */
export function instanceOfNotificationWebhookMappingRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;

    return isInstance;
}

export function NotificationWebhookMappingRequestFromJSON(json: any): NotificationWebhookMappingRequest {
    return NotificationWebhookMappingRequestFromJSONTyped(json, false);
}

export function NotificationWebhookMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationWebhookMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'expression': json['expression'],
    };
}

export function NotificationWebhookMappingRequestToJSON(value?: NotificationWebhookMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'expression': value.expression,
    };
}
