// @goauthentik/api/src/models/NotificationRule.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Group } from './Group';
import {
    GroupFromJSON,
    GroupFromJSONTyped,
    GroupToJSON,
} from './Group';
import type { SeverityEnum } from './SeverityEnum';
import {
    SeverityEnumFromJSON,
    SeverityEnumFromJSONTyped,
    SeverityEnumToJSON,
} from './SeverityEnum';

/**
 * NotificationRule Serializer
 * @export
 * @interface NotificationRule
 */
export interface NotificationRule {
    /**
     * 
     * @type {string}
     * @memberof NotificationRule
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof NotificationRule
     */
    name: string;
    /**
     * Select which transports should be used to notify the user. If none are selected, the notification will only be shown in the authentik UI.
     * @type {Array<string>}
     * @memberof NotificationRule
     */
    transports?: Array<string>;
    /**
     * 
     * @type {SeverityEnum}
     * @memberof NotificationRule
     */
    severity?: SeverityEnum;
    /**
     * Define which group of users this notification should be sent and shown to. If left empty, Notification won't ben sent.
     * @type {string}
     * @memberof NotificationRule
     */
    group?: string | null;
    /**
     * 
     * @type {Group}
     * @memberof NotificationRule
     */
    readonly groupObj: Group;
}

/**
 * Check if a given object implements the NotificationRule interface.
 */
export function instanceOfNotificationRule(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "groupObj" in value;

    return isInstance;
}

export function NotificationRuleFromJSON(json: any): NotificationRule {
    return NotificationRuleFromJSONTyped(json, false);
}

export function NotificationRuleFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationRule {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'transports': !exists(json, 'transports') ? undefined : json['transports'],
        'severity': !exists(json, 'severity') ? undefined : SeverityEnumFromJSON(json['severity']),
        'group': !exists(json, 'group') ? undefined : json['group'],
        'groupObj': GroupFromJSON(json['group_obj']),
    };
}

export function NotificationRuleToJSON(value?: NotificationRule | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'transports': value.transports,
        'severity': SeverityEnumToJSON(value.severity),
        'group': value.group,
    };
}
