// @goauthentik/api/src/models/NotificationRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { EventRequest } from './EventRequest';
import {
    EventRequestFromJSON,
    EventRequestFromJSONTyped,
    EventRequestToJSON,
} from './EventRequest';

/**
 * Notification Serializer
 * @export
 * @interface NotificationRequest
 */
export interface NotificationRequest {
    /**
     * 
     * @type {EventRequest}
     * @memberof NotificationRequest
     */
    event?: EventRequest;
    /**
     * 
     * @type {boolean}
     * @memberof NotificationRequest
     */
    seen?: boolean;
}

/**
 * Check if a given object implements the NotificationRequest interface.
 */
export function instanceOfNotificationRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function NotificationRequestFromJSON(json: any): NotificationRequest {
    return NotificationRequestFromJSONTyped(json, false);
}

export function NotificationRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'event': !exists(json, 'event') ? undefined : EventRequestFromJSON(json['event']),
        'seen': !exists(json, 'seen') ? undefined : json['seen'],
    };
}

export function NotificationRequestToJSON(value?: NotificationRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'event': EventRequestToJSON(value.event),
        'seen': value.seen,
    };
}
