// @goauthentik/api/src/models/NotificationRuleRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { SeverityEnum } from './SeverityEnum';
import {
    SeverityEnumFromJSON,
    SeverityEnumFromJSONTyped,
    SeverityEnumToJSON,
} from './SeverityEnum';

/**
 * NotificationRule Serializer
 * @export
 * @interface NotificationRuleRequest
 */
export interface NotificationRuleRequest {
    /**
     * 
     * @type {string}
     * @memberof NotificationRuleRequest
     */
    name: string;
    /**
     * Select which transports should be used to notify the user. If none are selected, the notification will only be shown in the authentik UI.
     * @type {Array<string>}
     * @memberof NotificationRuleRequest
     */
    transports?: Array<string>;
    /**
     * 
     * @type {SeverityEnum}
     * @memberof NotificationRuleRequest
     */
    severity?: SeverityEnum;
    /**
     * Define which group of users this notification should be sent and shown to. If left empty, Notification won't ben sent.
     * @type {string}
     * @memberof NotificationRuleRequest
     */
    group?: string | null;
}

/**
 * Check if a given object implements the NotificationRuleRequest interface.
 */
export function instanceOfNotificationRuleRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function NotificationRuleRequestFromJSON(json: any): NotificationRuleRequest {
    return NotificationRuleRequestFromJSONTyped(json, false);
}

export function NotificationRuleRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationRuleRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'transports': !exists(json, 'transports') ? undefined : json['transports'],
        'severity': !exists(json, 'severity') ? undefined : SeverityEnumFromJSON(json['severity']),
        'group': !exists(json, 'group') ? undefined : json['group'],
    };
}

export function NotificationRuleRequestToJSON(value?: NotificationRuleRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'transports': value.transports,
        'severity': SeverityEnumToJSON(value.severity),
        'group': value.group,
    };
}
