// @goauthentik/api/src/models/Notification.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Event } from './Event';
import {
    EventFromJSON,
    EventFromJSONTyped,
    EventToJSON,
} from './Event';
import type { SeverityEnum } from './SeverityEnum';
import {
    SeverityEnumFromJSON,
    SeverityEnumFromJSONTyped,
    SeverityEnumToJSON,
} from './SeverityEnum';

/**
 * Notification Serializer
 * @export
 * @interface Notification
 */
export interface Notification {
    /**
     * 
     * @type {string}
     * @memberof Notification
     */
    readonly pk: string;
    /**
     * 
     * @type {SeverityEnum}
     * @memberof Notification
     */
    readonly severity: SeverityEnum;
    /**
     * 
     * @type {string}
     * @memberof Notification
     */
    readonly body: string;
    /**
     * 
     * @type {Date}
     * @memberof Notification
     */
    readonly created: Date;
    /**
     * 
     * @type {Event}
     * @memberof Notification
     */
    event?: Event;
    /**
     * 
     * @type {boolean}
     * @memberof Notification
     */
    seen?: boolean;
}

/**
 * Check if a given object implements the Notification interface.
 */
export function instanceOfNotification(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "severity" in value;
    isInstance = isInstance && "body" in value;
    isInstance = isInstance && "created" in value;

    return isInstance;
}

export function NotificationFromJSON(json: any): Notification {
    return NotificationFromJSONTyped(json, false);
}

export function NotificationFromJSONTyped(json: any, ignoreDiscriminator: boolean): Notification {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'severity': SeverityEnumFromJSON(json['severity']),
        'body': json['body'],
        'created': (new Date(json['created'])),
        'event': !exists(json, 'event') ? undefined : EventFromJSON(json['event']),
        'seen': !exists(json, 'seen') ? undefined : json['seen'],
    };
}

export function NotificationToJSON(value?: Notification | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'event': EventToJSON(value.event),
        'seen': value.seen,
    };
}
