// @goauthentik/api/src/models/NotificationTransportTest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Notification test serializer
 * @export
 * @interface NotificationTransportTest
 */
export interface NotificationTransportTest {
    /**
     * 
     * @type {Array<string>}
     * @memberof NotificationTransportTest
     */
    messages: Array<string>;
}

/**
 * Check if a given object implements the NotificationTransportTest interface.
 */
export function instanceOfNotificationTransportTest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "messages" in value;

    return isInstance;
}

export function NotificationTransportTestFromJSON(json: any): NotificationTransportTest {
    return NotificationTransportTestFromJSONTyped(json, false);
}

export function NotificationTransportTestFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationTransportTest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'messages': json['messages'],
    };
}

export function NotificationTransportTestToJSON(value?: NotificationTransportTest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'messages': value.messages,
    };
}
