// @goauthentik/api/src/models/NotificationTransportModeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `local` - authentik inbuilt notifications
 * * `webhook` - Generic Webhook
 * * `webhook_slack` - Slack Webhook (Slack/Discord)
 * * `email` - Email
 * @export
 */
export const NotificationTransportModeEnum = {
    Local: 'local',
    Webhook: 'webhook',
    WebhookSlack: 'webhook_slack',
    Email: 'email',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type NotificationTransportModeEnum = typeof NotificationTransportModeEnum[keyof typeof NotificationTransportModeEnum];


export function NotificationTransportModeEnumFromJSON(json: any): NotificationTransportModeEnum {
    return NotificationTransportModeEnumFromJSONTyped(json, false);
}

export function NotificationTransportModeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): NotificationTransportModeEnum {
    return json as NotificationTransportModeEnum;
}

export function NotificationTransportModeEnumToJSON(value?: NotificationTransportModeEnum | null): any {
    return value as any;
}
