// @goauthentik/api/src/models/OAuthDeviceCodeChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Response that includes the user-entered device code
 * @export
 * @interface OAuthDeviceCodeChallengeResponseRequest
 */
export interface OAuthDeviceCodeChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof OAuthDeviceCodeChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {number}
     * @memberof OAuthDeviceCodeChallengeResponseRequest
     */
    code: number;
}

/**
 * Check if a given object implements the OAuthDeviceCodeChallengeResponseRequest interface.
 */
export function instanceOfOAuthDeviceCodeChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "code" in value;

    return isInstance;
}

export function OAuthDeviceCodeChallengeResponseRequestFromJSON(json: any): OAuthDeviceCodeChallengeResponseRequest {
    return OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json, false);
}

export function OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): OAuthDeviceCodeChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'code': json['code'],
    };
}

export function OAuthDeviceCodeChallengeResponseRequestToJSON(value?: OAuthDeviceCodeChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'code': value.code,
    };
}
