// @goauthentik/api/src/models/OAuthDeviceCodeFinishChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Response that device has been authenticated and tab can be closed
 * @export
 * @interface OAuthDeviceCodeFinishChallengeResponseRequest
 */
export interface OAuthDeviceCodeFinishChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof OAuthDeviceCodeFinishChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the OAuthDeviceCodeFinishChallengeResponseRequest interface.
 */
export function instanceOfOAuthDeviceCodeFinishChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function OAuthDeviceCodeFinishChallengeResponseRequestFromJSON(json: any): OAuthDeviceCodeFinishChallengeResponseRequest {
    return OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped(json, false);
}

export function OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): OAuthDeviceCodeFinishChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function OAuthDeviceCodeFinishChallengeResponseRequestToJSON(value?: OAuthDeviceCodeFinishChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
