// @goauthentik/api/src/models/OutpostTypeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `proxy` - Proxy
 * * `ldap` - Ldap
 * * `radius` - Radius
 * * `rac` - Rac
 * @export
 */
export const OutpostTypeEnum = {
    Proxy: 'proxy',
    Ldap: 'ldap',
    Radius: 'radius',
    Rac: 'rac',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type OutpostTypeEnum = typeof OutpostTypeEnum[keyof typeof OutpostTypeEnum];


export function OutpostTypeEnumFromJSON(json: any): OutpostTypeEnum {
    return OutpostTypeEnumFromJSONTyped(json, false);
}

export function OutpostTypeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): OutpostTypeEnum {
    return json as OutpostTypeEnum;
}

export function OutpostTypeEnumToJSON(value?: OutpostTypeEnum | null): any {
    return value as any;
}
