// @goauthentik/api/src/models/OutpostDefaultConfig.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Global default outpost config
 * @export
 * @interface OutpostDefaultConfig
 */
export interface OutpostDefaultConfig {
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof OutpostDefaultConfig
     */
    readonly config: { [key: string]: any; };
}

/**
 * Check if a given object implements the OutpostDefaultConfig interface.
 */
export function instanceOfOutpostDefaultConfig(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "config" in value;

    return isInstance;
}

export function OutpostDefaultConfigFromJSON(json: any): OutpostDefaultConfig {
    return OutpostDefaultConfigFromJSONTyped(json, false);
}

export function OutpostDefaultConfigFromJSONTyped(json: any, ignoreDiscriminator: boolean): OutpostDefaultConfig {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'config': json['config'],
    };
}

export function OutpostDefaultConfigToJSON(value?: OutpostDefaultConfig | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
