// @goauthentik/api/src/models/PaginatedPromptStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PromptStage } from './PromptStage';
import {
    PromptStageFromJSON,
    PromptStageFromJSONTyped,
    PromptStageToJSON,
} from './PromptStage';

/**
 * 
 * @export
 * @interface PaginatedPromptStageList
 */
export interface PaginatedPromptStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPromptStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PromptStage>}
     * @memberof PaginatedPromptStageList
     */
    results: Array<PromptStage>;
}

/**
 * Check if a given object implements the PaginatedPromptStageList interface.
 */
export function instanceOfPaginatedPromptStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPromptStageListFromJSON(json: any): PaginatedPromptStageList {
    return PaginatedPromptStageListFromJSONTyped(json, false);
}

export function PaginatedPromptStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPromptStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PromptStageFromJSON)),
    };
}

export function PaginatedPromptStageListToJSON(value?: PaginatedPromptStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PromptStageToJSON)),
    };
}
