// @goauthentik/api/src/models/PaginatedPlexSourceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PlexSourceConnection } from './PlexSourceConnection';
import {
    PlexSourceConnectionFromJSON,
    PlexSourceConnectionFromJSONTyped,
    PlexSourceConnectionToJSON,
} from './PlexSourceConnection';

/**
 * 
 * @export
 * @interface PaginatedPlexSourceConnectionList
 */
export interface PaginatedPlexSourceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPlexSourceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PlexSourceConnection>}
     * @memberof PaginatedPlexSourceConnectionList
     */
    results: Array<PlexSourceConnection>;
}

/**
 * Check if a given object implements the PaginatedPlexSourceConnectionList interface.
 */
export function instanceOfPaginatedPlexSourceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPlexSourceConnectionListFromJSON(json: any): PaginatedPlexSourceConnectionList {
    return PaginatedPlexSourceConnectionListFromJSONTyped(json, false);
}

export function PaginatedPlexSourceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPlexSourceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PlexSourceConnectionFromJSON)),
    };
}

export function PaginatedPlexSourceConnectionListToJSON(value?: PaginatedPlexSourceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PlexSourceConnectionToJSON)),
    };
}
