// @goauthentik/api/src/models/PaginatedEventMatcherPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { EventMatcherPolicy } from './EventMatcherPolicy';
import {
    EventMatcherPolicyFromJSON,
    EventMatcherPolicyFromJSONTyped,
    EventMatcherPolicyToJSON,
} from './EventMatcherPolicy';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedEventMatcherPolicyList
 */
export interface PaginatedEventMatcherPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedEventMatcherPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<EventMatcherPolicy>}
     * @memberof PaginatedEventMatcherPolicyList
     */
    results: Array<EventMatcherPolicy>;
}

/**
 * Check if a given object implements the PaginatedEventMatcherPolicyList interface.
 */
export function instanceOfPaginatedEventMatcherPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedEventMatcherPolicyListFromJSON(json: any): PaginatedEventMatcherPolicyList {
    return PaginatedEventMatcherPolicyListFromJSONTyped(json, false);
}

export function PaginatedEventMatcherPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedEventMatcherPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(EventMatcherPolicyFromJSON)),
    };
}

export function PaginatedEventMatcherPolicyListToJSON(value?: PaginatedEventMatcherPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(EventMatcherPolicyToJSON)),
    };
}
