// @goauthentik/api/src/models/PaginatedOAuth2ProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { OAuth2Provider } from './OAuth2Provider';
import {
    OAuth2ProviderFromJSON,
    OAuth2ProviderFromJSONTyped,
    OAuth2ProviderToJSON,
} from './OAuth2Provider';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedOAuth2ProviderList
 */
export interface PaginatedOAuth2ProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedOAuth2ProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<OAuth2Provider>}
     * @memberof PaginatedOAuth2ProviderList
     */
    results: Array<OAuth2Provider>;
}

/**
 * Check if a given object implements the PaginatedOAuth2ProviderList interface.
 */
export function instanceOfPaginatedOAuth2ProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedOAuth2ProviderListFromJSON(json: any): PaginatedOAuth2ProviderList {
    return PaginatedOAuth2ProviderListFromJSONTyped(json, false);
}

export function PaginatedOAuth2ProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedOAuth2ProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(OAuth2ProviderFromJSON)),
    };
}

export function PaginatedOAuth2ProviderListToJSON(value?: PaginatedOAuth2ProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(OAuth2ProviderToJSON)),
    };
}
