// @goauthentik/api/src/models/PaginatedPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Policy } from './Policy';
import {
    PolicyFromJSON,
    PolicyFromJSONTyped,
    PolicyToJSON,
} from './Policy';

/**
 * 
 * @export
 * @interface PaginatedPolicyList
 */
export interface PaginatedPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Policy>}
     * @memberof PaginatedPolicyList
     */
    results: Array<Policy>;
}

/**
 * Check if a given object implements the PaginatedPolicyList interface.
 */
export function instanceOfPaginatedPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPolicyListFromJSON(json: any): PaginatedPolicyList {
    return PaginatedPolicyListFromJSONTyped(json, false);
}

export function PaginatedPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PolicyFromJSON)),
    };
}

export function PaginatedPolicyListToJSON(value?: PaginatedPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PolicyToJSON)),
    };
}
