// @goauthentik/api/src/models/PaginatedTokenModelList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { TokenModel } from './TokenModel';
import {
    TokenModelFromJSON,
    TokenModelFromJSONTyped,
    TokenModelToJSON,
} from './TokenModel';

/**
 * 
 * @export
 * @interface PaginatedTokenModelList
 */
export interface PaginatedTokenModelList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedTokenModelList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<TokenModel>}
     * @memberof PaginatedTokenModelList
     */
    results: Array<TokenModel>;
}

/**
 * Check if a given object implements the PaginatedTokenModelList interface.
 */
export function instanceOfPaginatedTokenModelList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedTokenModelListFromJSON(json: any): PaginatedTokenModelList {
    return PaginatedTokenModelListFromJSONTyped(json, false);
}

export function PaginatedTokenModelListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedTokenModelList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(TokenModelFromJSON)),
    };
}

export function PaginatedTokenModelListToJSON(value?: PaginatedTokenModelList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(TokenModelToJSON)),
    };
}
