// @goauthentik/api/src/models/PaginatedExtraRoleObjectPermissionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ExtraRoleObjectPermission } from './ExtraRoleObjectPermission';
import {
    ExtraRoleObjectPermissionFromJSON,
    ExtraRoleObjectPermissionFromJSONTyped,
    ExtraRoleObjectPermissionToJSON,
} from './ExtraRoleObjectPermission';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedExtraRoleObjectPermissionList
 */
export interface PaginatedExtraRoleObjectPermissionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedExtraRoleObjectPermissionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ExtraRoleObjectPermission>}
     * @memberof PaginatedExtraRoleObjectPermissionList
     */
    results: Array<ExtraRoleObjectPermission>;
}

/**
 * Check if a given object implements the PaginatedExtraRoleObjectPermissionList interface.
 */
export function instanceOfPaginatedExtraRoleObjectPermissionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedExtraRoleObjectPermissionListFromJSON(json: any): PaginatedExtraRoleObjectPermissionList {
    return PaginatedExtraRoleObjectPermissionListFromJSONTyped(json, false);
}

export function PaginatedExtraRoleObjectPermissionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedExtraRoleObjectPermissionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ExtraRoleObjectPermissionFromJSON)),
    };
}

export function PaginatedExtraRoleObjectPermissionListToJSON(value?: PaginatedExtraRoleObjectPermissionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ExtraRoleObjectPermissionToJSON)),
    };
}
