// @goauthentik/api/src/models/PaginatedServiceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { ServiceConnection } from './ServiceConnection';
import {
    ServiceConnectionFromJSON,
    ServiceConnectionFromJSONTyped,
    ServiceConnectionToJSON,
} from './ServiceConnection';

/**
 * 
 * @export
 * @interface PaginatedServiceConnectionList
 */
export interface PaginatedServiceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedServiceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ServiceConnection>}
     * @memberof PaginatedServiceConnectionList
     */
    results: Array<ServiceConnection>;
}

/**
 * Check if a given object implements the PaginatedServiceConnectionList interface.
 */
export function instanceOfPaginatedServiceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedServiceConnectionListFromJSON(json: any): PaginatedServiceConnectionList {
    return PaginatedServiceConnectionListFromJSONTyped(json, false);
}

export function PaginatedServiceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedServiceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ServiceConnectionFromJSON)),
    };
}

export function PaginatedServiceConnectionListToJSON(value?: PaginatedServiceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ServiceConnectionToJSON)),
    };
}
