// @goauthentik/api/src/models/PaginatedUserAssignedObjectPermissionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserAssignedObjectPermission } from './UserAssignedObjectPermission';
import {
    UserAssignedObjectPermissionFromJSON,
    UserAssignedObjectPermissionFromJSONTyped,
    UserAssignedObjectPermissionToJSON,
} from './UserAssignedObjectPermission';

/**
 * 
 * @export
 * @interface PaginatedUserAssignedObjectPermissionList
 */
export interface PaginatedUserAssignedObjectPermissionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserAssignedObjectPermissionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserAssignedObjectPermission>}
     * @memberof PaginatedUserAssignedObjectPermissionList
     */
    results: Array<UserAssignedObjectPermission>;
}

/**
 * Check if a given object implements the PaginatedUserAssignedObjectPermissionList interface.
 */
export function instanceOfPaginatedUserAssignedObjectPermissionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserAssignedObjectPermissionListFromJSON(json: any): PaginatedUserAssignedObjectPermissionList {
    return PaginatedUserAssignedObjectPermissionListFromJSONTyped(json, false);
}

export function PaginatedUserAssignedObjectPermissionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserAssignedObjectPermissionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserAssignedObjectPermissionFromJSON)),
    };
}

export function PaginatedUserAssignedObjectPermissionListToJSON(value?: PaginatedUserAssignedObjectPermissionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserAssignedObjectPermissionToJSON)),
    };
}
