// @goauthentik/api/src/models/PaginatedApplicationList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Application } from './Application';
import {
    ApplicationFromJSON,
    ApplicationFromJSONTyped,
    ApplicationToJSON,
} from './Application';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedApplicationList
 */
export interface PaginatedApplicationList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedApplicationList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Application>}
     * @memberof PaginatedApplicationList
     */
    results: Array<Application>;
}

/**
 * Check if a given object implements the PaginatedApplicationList interface.
 */
export function instanceOfPaginatedApplicationList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedApplicationListFromJSON(json: any): PaginatedApplicationList {
    return PaginatedApplicationListFromJSONTyped(json, false);
}

export function PaginatedApplicationListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedApplicationList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ApplicationFromJSON)),
    };
}

export function PaginatedApplicationListToJSON(value?: PaginatedApplicationList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ApplicationToJSON)),
    };
}
