// @goauthentik/api/src/models/PaginatedAuthenticatedSessionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatedSession } from './AuthenticatedSession';
import {
    AuthenticatedSessionFromJSON,
    AuthenticatedSessionFromJSONTyped,
    AuthenticatedSessionToJSON,
} from './AuthenticatedSession';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatedSessionList
 */
export interface PaginatedAuthenticatedSessionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatedSessionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatedSession>}
     * @memberof PaginatedAuthenticatedSessionList
     */
    results: Array<AuthenticatedSession>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatedSessionList interface.
 */
export function instanceOfPaginatedAuthenticatedSessionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatedSessionListFromJSON(json: any): PaginatedAuthenticatedSessionList {
    return PaginatedAuthenticatedSessionListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatedSessionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatedSessionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatedSessionFromJSON)),
    };
}

export function PaginatedAuthenticatedSessionListToJSON(value?: PaginatedAuthenticatedSessionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatedSessionToJSON)),
    };
}
