// @goauthentik/api/src/models/PaginatedEmailStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { EmailStage } from './EmailStage';
import {
    EmailStageFromJSON,
    EmailStageFromJSONTyped,
    EmailStageToJSON,
} from './EmailStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedEmailStageList
 */
export interface PaginatedEmailStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedEmailStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<EmailStage>}
     * @memberof PaginatedEmailStageList
     */
    results: Array<EmailStage>;
}

/**
 * Check if a given object implements the PaginatedEmailStageList interface.
 */
export function instanceOfPaginatedEmailStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedEmailStageListFromJSON(json: any): PaginatedEmailStageList {
    return PaginatedEmailStageListFromJSONTyped(json, false);
}

export function PaginatedEmailStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedEmailStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(EmailStageFromJSON)),
    };
}

export function PaginatedEmailStageListToJSON(value?: PaginatedEmailStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(EmailStageToJSON)),
    };
}
