// @goauthentik/api/src/models/PaginatedAuthenticatorStaticStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorStaticStage } from './AuthenticatorStaticStage';
import {
    AuthenticatorStaticStageFromJSON,
    AuthenticatorStaticStageFromJSONTyped,
    AuthenticatorStaticStageToJSON,
} from './AuthenticatorStaticStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatorStaticStageList
 */
export interface PaginatedAuthenticatorStaticStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatorStaticStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatorStaticStage>}
     * @memberof PaginatedAuthenticatorStaticStageList
     */
    results: Array<AuthenticatorStaticStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatorStaticStageList interface.
 */
export function instanceOfPaginatedAuthenticatorStaticStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatorStaticStageListFromJSON(json: any): PaginatedAuthenticatorStaticStageList {
    return PaginatedAuthenticatorStaticStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatorStaticStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatorStaticStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatorStaticStageFromJSON)),
    };
}

export function PaginatedAuthenticatorStaticStageListToJSON(value?: PaginatedAuthenticatorStaticStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatorStaticStageToJSON)),
    };
}
