// @goauthentik/api/src/models/PaginatedReputationList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Reputation } from './Reputation';
import {
    ReputationFromJSON,
    ReputationFromJSONTyped,
    ReputationToJSON,
} from './Reputation';

/**
 * 
 * @export
 * @interface PaginatedReputationList
 */
export interface PaginatedReputationList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedReputationList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Reputation>}
     * @memberof PaginatedReputationList
     */
    results: Array<Reputation>;
}

/**
 * Check if a given object implements the PaginatedReputationList interface.
 */
export function instanceOfPaginatedReputationList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedReputationListFromJSON(json: any): PaginatedReputationList {
    return PaginatedReputationListFromJSONTyped(json, false);
}

export function PaginatedReputationListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedReputationList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ReputationFromJSON)),
    };
}

export function PaginatedReputationListToJSON(value?: PaginatedReputationList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ReputationToJSON)),
    };
}
