// @goauthentik/api/src/models/PaginatedSMSDeviceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SMSDevice } from './SMSDevice';
import {
    SMSDeviceFromJSON,
    SMSDeviceFromJSONTyped,
    SMSDeviceToJSON,
} from './SMSDevice';

/**
 * 
 * @export
 * @interface PaginatedSMSDeviceList
 */
export interface PaginatedSMSDeviceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSMSDeviceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SMSDevice>}
     * @memberof PaginatedSMSDeviceList
     */
    results: Array<SMSDevice>;
}

/**
 * Check if a given object implements the PaginatedSMSDeviceList interface.
 */
export function instanceOfPaginatedSMSDeviceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSMSDeviceListFromJSON(json: any): PaginatedSMSDeviceList {
    return PaginatedSMSDeviceListFromJSONTyped(json, false);
}

export function PaginatedSMSDeviceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSMSDeviceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SMSDeviceFromJSON)),
    };
}

export function PaginatedSMSDeviceListToJSON(value?: PaginatedSMSDeviceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SMSDeviceToJSON)),
    };
}
