// @goauthentik/api/src/models/PaginatedConnectionTokenList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ConnectionToken } from './ConnectionToken';
import {
    ConnectionTokenFromJSON,
    ConnectionTokenFromJSONTyped,
    ConnectionTokenToJSON,
} from './ConnectionToken';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedConnectionTokenList
 */
export interface PaginatedConnectionTokenList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedConnectionTokenList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ConnectionToken>}
     * @memberof PaginatedConnectionTokenList
     */
    results: Array<ConnectionToken>;
}

/**
 * Check if a given object implements the PaginatedConnectionTokenList interface.
 */
export function instanceOfPaginatedConnectionTokenList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedConnectionTokenListFromJSON(json: any): PaginatedConnectionTokenList {
    return PaginatedConnectionTokenListFromJSONTyped(json, false);
}

export function PaginatedConnectionTokenListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedConnectionTokenList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ConnectionTokenFromJSON)),
    };
}

export function PaginatedConnectionTokenListToJSON(value?: PaginatedConnectionTokenList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ConnectionTokenToJSON)),
    };
}
