// @goauthentik/api/src/models/PaginatedInvitationStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { InvitationStage } from './InvitationStage';
import {
    InvitationStageFromJSON,
    InvitationStageFromJSONTyped,
    InvitationStageToJSON,
} from './InvitationStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedInvitationStageList
 */
export interface PaginatedInvitationStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedInvitationStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<InvitationStage>}
     * @memberof PaginatedInvitationStageList
     */
    results: Array<InvitationStage>;
}

/**
 * Check if a given object implements the PaginatedInvitationStageList interface.
 */
export function instanceOfPaginatedInvitationStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedInvitationStageListFromJSON(json: any): PaginatedInvitationStageList {
    return PaginatedInvitationStageListFromJSONTyped(json, false);
}

export function PaginatedInvitationStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedInvitationStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(InvitationStageFromJSON)),
    };
}

export function PaginatedInvitationStageListToJSON(value?: PaginatedInvitationStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(InvitationStageToJSON)),
    };
}
