// @goauthentik/api/src/models/PaginatedPromptList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Prompt } from './Prompt';
import {
    PromptFromJSON,
    PromptFromJSONTyped,
    PromptToJSON,
} from './Prompt';

/**
 * 
 * @export
 * @interface PaginatedPromptList
 */
export interface PaginatedPromptList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPromptList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Prompt>}
     * @memberof PaginatedPromptList
     */
    results: Array<Prompt>;
}

/**
 * Check if a given object implements the PaginatedPromptList interface.
 */
export function instanceOfPaginatedPromptList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPromptListFromJSON(json: any): PaginatedPromptList {
    return PaginatedPromptListFromJSONTyped(json, false);
}

export function PaginatedPromptListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPromptList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PromptFromJSON)),
    };
}

export function PaginatedPromptListToJSON(value?: PaginatedPromptList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PromptToJSON)),
    };
}
