// @goauthentik/api/src/models/PaginatedAuthenticatorDuoStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorDuoStage } from './AuthenticatorDuoStage';
import {
    AuthenticatorDuoStageFromJSON,
    AuthenticatorDuoStageFromJSONTyped,
    AuthenticatorDuoStageToJSON,
} from './AuthenticatorDuoStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatorDuoStageList
 */
export interface PaginatedAuthenticatorDuoStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatorDuoStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatorDuoStage>}
     * @memberof PaginatedAuthenticatorDuoStageList
     */
    results: Array<AuthenticatorDuoStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatorDuoStageList interface.
 */
export function instanceOfPaginatedAuthenticatorDuoStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatorDuoStageListFromJSON(json: any): PaginatedAuthenticatorDuoStageList {
    return PaginatedAuthenticatorDuoStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatorDuoStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatorDuoStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatorDuoStageFromJSON)),
    };
}

export function PaginatedAuthenticatorDuoStageListToJSON(value?: PaginatedAuthenticatorDuoStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatorDuoStageToJSON)),
    };
}
