// @goauthentik/api/src/models/PaginatedUserSAMLSourceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserSAMLSourceConnection } from './UserSAMLSourceConnection';
import {
    UserSAMLSourceConnectionFromJSON,
    UserSAMLSourceConnectionFromJSONTyped,
    UserSAMLSourceConnectionToJSON,
} from './UserSAMLSourceConnection';

/**
 * 
 * @export
 * @interface PaginatedUserSAMLSourceConnectionList
 */
export interface PaginatedUserSAMLSourceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserSAMLSourceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserSAMLSourceConnection>}
     * @memberof PaginatedUserSAMLSourceConnectionList
     */
    results: Array<UserSAMLSourceConnection>;
}

/**
 * Check if a given object implements the PaginatedUserSAMLSourceConnectionList interface.
 */
export function instanceOfPaginatedUserSAMLSourceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserSAMLSourceConnectionListFromJSON(json: any): PaginatedUserSAMLSourceConnectionList {
    return PaginatedUserSAMLSourceConnectionListFromJSONTyped(json, false);
}

export function PaginatedUserSAMLSourceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserSAMLSourceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserSAMLSourceConnectionFromJSON)),
    };
}

export function PaginatedUserSAMLSourceConnectionListToJSON(value?: PaginatedUserSAMLSourceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserSAMLSourceConnectionToJSON)),
    };
}
