// @goauthentik/api/src/models/PaginatedScopeMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { ScopeMapping } from './ScopeMapping';
import {
    ScopeMappingFromJSON,
    ScopeMappingFromJSONTyped,
    ScopeMappingToJSON,
} from './ScopeMapping';

/**
 * 
 * @export
 * @interface PaginatedScopeMappingList
 */
export interface PaginatedScopeMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedScopeMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ScopeMapping>}
     * @memberof PaginatedScopeMappingList
     */
    results: Array<ScopeMapping>;
}

/**
 * Check if a given object implements the PaginatedScopeMappingList interface.
 */
export function instanceOfPaginatedScopeMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedScopeMappingListFromJSON(json: any): PaginatedScopeMappingList {
    return PaginatedScopeMappingListFromJSONTyped(json, false);
}

export function PaginatedScopeMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedScopeMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ScopeMappingFromJSON)),
    };
}

export function PaginatedScopeMappingListToJSON(value?: PaginatedScopeMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ScopeMappingToJSON)),
    };
}
