// @goauthentik/api/src/models/PaginatedUserLoginStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserLoginStage } from './UserLoginStage';
import {
    UserLoginStageFromJSON,
    UserLoginStageFromJSONTyped,
    UserLoginStageToJSON,
} from './UserLoginStage';

/**
 * 
 * @export
 * @interface PaginatedUserLoginStageList
 */
export interface PaginatedUserLoginStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserLoginStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserLoginStage>}
     * @memberof PaginatedUserLoginStageList
     */
    results: Array<UserLoginStage>;
}

/**
 * Check if a given object implements the PaginatedUserLoginStageList interface.
 */
export function instanceOfPaginatedUserLoginStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserLoginStageListFromJSON(json: any): PaginatedUserLoginStageList {
    return PaginatedUserLoginStageListFromJSONTyped(json, false);
}

export function PaginatedUserLoginStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserLoginStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserLoginStageFromJSON)),
    };
}

export function PaginatedUserLoginStageListToJSON(value?: PaginatedUserLoginStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserLoginStageToJSON)),
    };
}
