// @goauthentik/api/src/models/PaginatedLDAPOutpostConfigList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPOutpostConfig } from './LDAPOutpostConfig';
import {
    LDAPOutpostConfigFromJSON,
    LDAPOutpostConfigFromJSONTyped,
    LDAPOutpostConfigToJSON,
} from './LDAPOutpostConfig';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedLDAPOutpostConfigList
 */
export interface PaginatedLDAPOutpostConfigList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedLDAPOutpostConfigList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<LDAPOutpostConfig>}
     * @memberof PaginatedLDAPOutpostConfigList
     */
    results: Array<LDAPOutpostConfig>;
}

/**
 * Check if a given object implements the PaginatedLDAPOutpostConfigList interface.
 */
export function instanceOfPaginatedLDAPOutpostConfigList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedLDAPOutpostConfigListFromJSON(json: any): PaginatedLDAPOutpostConfigList {
    return PaginatedLDAPOutpostConfigListFromJSONTyped(json, false);
}

export function PaginatedLDAPOutpostConfigListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedLDAPOutpostConfigList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(LDAPOutpostConfigFromJSON)),
    };
}

export function PaginatedLDAPOutpostConfigListToJSON(value?: PaginatedLDAPOutpostConfigList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(LDAPOutpostConfigToJSON)),
    };
}
