// @goauthentik/api/src/models/PaginatedPermissionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Permission } from './Permission';
import {
    PermissionFromJSON,
    PermissionFromJSONTyped,
    PermissionToJSON,
} from './Permission';

/**
 * 
 * @export
 * @interface PaginatedPermissionList
 */
export interface PaginatedPermissionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPermissionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Permission>}
     * @memberof PaginatedPermissionList
     */
    results: Array<Permission>;
}

/**
 * Check if a given object implements the PaginatedPermissionList interface.
 */
export function instanceOfPaginatedPermissionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPermissionListFromJSON(json: any): PaginatedPermissionList {
    return PaginatedPermissionListFromJSONTyped(json, false);
}

export function PaginatedPermissionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPermissionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PermissionFromJSON)),
    };
}

export function PaginatedPermissionListToJSON(value?: PaginatedPermissionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PermissionToJSON)),
    };
}
