// @goauthentik/api/src/models/PaginatedAuthenticatorSMSStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorSMSStage } from './AuthenticatorSMSStage';
import {
    AuthenticatorSMSStageFromJSON,
    AuthenticatorSMSStageFromJSONTyped,
    AuthenticatorSMSStageToJSON,
} from './AuthenticatorSMSStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatorSMSStageList
 */
export interface PaginatedAuthenticatorSMSStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatorSMSStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatorSMSStage>}
     * @memberof PaginatedAuthenticatorSMSStageList
     */
    results: Array<AuthenticatorSMSStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatorSMSStageList interface.
 */
export function instanceOfPaginatedAuthenticatorSMSStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatorSMSStageListFromJSON(json: any): PaginatedAuthenticatorSMSStageList {
    return PaginatedAuthenticatorSMSStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatorSMSStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatorSMSStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatorSMSStageFromJSON)),
    };
}

export function PaginatedAuthenticatorSMSStageListToJSON(value?: PaginatedAuthenticatorSMSStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatorSMSStageToJSON)),
    };
}
