// @goauthentik/api/src/models/PaginatedAuthenticatorTOTPStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorTOTPStage } from './AuthenticatorTOTPStage';
import {
    AuthenticatorTOTPStageFromJSON,
    AuthenticatorTOTPStageFromJSONTyped,
    AuthenticatorTOTPStageToJSON,
} from './AuthenticatorTOTPStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatorTOTPStageList
 */
export interface PaginatedAuthenticatorTOTPStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatorTOTPStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatorTOTPStage>}
     * @memberof PaginatedAuthenticatorTOTPStageList
     */
    results: Array<AuthenticatorTOTPStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatorTOTPStageList interface.
 */
export function instanceOfPaginatedAuthenticatorTOTPStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatorTOTPStageListFromJSON(json: any): PaginatedAuthenticatorTOTPStageList {
    return PaginatedAuthenticatorTOTPStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatorTOTPStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatorTOTPStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatorTOTPStageFromJSON)),
    };
}

export function PaginatedAuthenticatorTOTPStageListToJSON(value?: PaginatedAuthenticatorTOTPStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatorTOTPStageToJSON)),
    };
}
