// @goauthentik/api/src/models/PaginatedEventList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Event } from './Event';
import {
    EventFromJSON,
    EventFromJSONTyped,
    EventToJSON,
} from './Event';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedEventList
 */
export interface PaginatedEventList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedEventList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Event>}
     * @memberof PaginatedEventList
     */
    results: Array<Event>;
}

/**
 * Check if a given object implements the PaginatedEventList interface.
 */
export function instanceOfPaginatedEventList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedEventListFromJSON(json: any): PaginatedEventList {
    return PaginatedEventListFromJSONTyped(json, false);
}

export function PaginatedEventListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedEventList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(EventFromJSON)),
    };
}

export function PaginatedEventListToJSON(value?: PaginatedEventList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(EventToJSON)),
    };
}
