// @goauthentik/api/src/models/PaginatedUserList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { User } from './User';
import {
    UserFromJSON,
    UserFromJSONTyped,
    UserToJSON,
} from './User';

/**
 * 
 * @export
 * @interface PaginatedUserList
 */
export interface PaginatedUserList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<User>}
     * @memberof PaginatedUserList
     */
    results: Array<User>;
}

/**
 * Check if a given object implements the PaginatedUserList interface.
 */
export function instanceOfPaginatedUserList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserListFromJSON(json: any): PaginatedUserList {
    return PaginatedUserListFromJSONTyped(json, false);
}

export function PaginatedUserListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserFromJSON)),
    };
}

export function PaginatedUserListToJSON(value?: PaginatedUserList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserToJSON)),
    };
}
