// @goauthentik/api/src/models/PaginatedExtraUserObjectPermissionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ExtraUserObjectPermission } from './ExtraUserObjectPermission';
import {
    ExtraUserObjectPermissionFromJSON,
    ExtraUserObjectPermissionFromJSONTyped,
    ExtraUserObjectPermissionToJSON,
} from './ExtraUserObjectPermission';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedExtraUserObjectPermissionList
 */
export interface PaginatedExtraUserObjectPermissionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedExtraUserObjectPermissionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ExtraUserObjectPermission>}
     * @memberof PaginatedExtraUserObjectPermissionList
     */
    results: Array<ExtraUserObjectPermission>;
}

/**
 * Check if a given object implements the PaginatedExtraUserObjectPermissionList interface.
 */
export function instanceOfPaginatedExtraUserObjectPermissionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedExtraUserObjectPermissionListFromJSON(json: any): PaginatedExtraUserObjectPermissionList {
    return PaginatedExtraUserObjectPermissionListFromJSONTyped(json, false);
}

export function PaginatedExtraUserObjectPermissionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedExtraUserObjectPermissionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ExtraUserObjectPermissionFromJSON)),
    };
}

export function PaginatedExtraUserObjectPermissionListToJSON(value?: PaginatedExtraUserObjectPermissionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ExtraUserObjectPermissionToJSON)),
    };
}
