// @goauthentik/api/src/models/PaginatedExpiringBaseGrantModelList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ExpiringBaseGrantModel } from './ExpiringBaseGrantModel';
import {
    ExpiringBaseGrantModelFromJSON,
    ExpiringBaseGrantModelFromJSONTyped,
    ExpiringBaseGrantModelToJSON,
} from './ExpiringBaseGrantModel';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedExpiringBaseGrantModelList
 */
export interface PaginatedExpiringBaseGrantModelList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedExpiringBaseGrantModelList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ExpiringBaseGrantModel>}
     * @memberof PaginatedExpiringBaseGrantModelList
     */
    results: Array<ExpiringBaseGrantModel>;
}

/**
 * Check if a given object implements the PaginatedExpiringBaseGrantModelList interface.
 */
export function instanceOfPaginatedExpiringBaseGrantModelList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedExpiringBaseGrantModelListFromJSON(json: any): PaginatedExpiringBaseGrantModelList {
    return PaginatedExpiringBaseGrantModelListFromJSONTyped(json, false);
}

export function PaginatedExpiringBaseGrantModelListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedExpiringBaseGrantModelList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ExpiringBaseGrantModelFromJSON)),
    };
}

export function PaginatedExpiringBaseGrantModelListToJSON(value?: PaginatedExpiringBaseGrantModelList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ExpiringBaseGrantModelToJSON)),
    };
}
