// @goauthentik/api/src/models/PaginatedLicenseList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { License } from './License';
import {
    LicenseFromJSON,
    LicenseFromJSONTyped,
    LicenseToJSON,
} from './License';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedLicenseList
 */
export interface PaginatedLicenseList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedLicenseList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<License>}
     * @memberof PaginatedLicenseList
     */
    results: Array<License>;
}

/**
 * Check if a given object implements the PaginatedLicenseList interface.
 */
export function instanceOfPaginatedLicenseList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedLicenseListFromJSON(json: any): PaginatedLicenseList {
    return PaginatedLicenseListFromJSONTyped(json, false);
}

export function PaginatedLicenseListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedLicenseList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(LicenseFromJSON)),
    };
}

export function PaginatedLicenseListToJSON(value?: PaginatedLicenseList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(LicenseToJSON)),
    };
}
