// @goauthentik/api/src/models/PaginatedGroupList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Group } from './Group';
import {
    GroupFromJSON,
    GroupFromJSONTyped,
    GroupToJSON,
} from './Group';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedGroupList
 */
export interface PaginatedGroupList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedGroupList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Group>}
     * @memberof PaginatedGroupList
     */
    results: Array<Group>;
}

/**
 * Check if a given object implements the PaginatedGroupList interface.
 */
export function instanceOfPaginatedGroupList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedGroupListFromJSON(json: any): PaginatedGroupList {
    return PaginatedGroupListFromJSONTyped(json, false);
}

export function PaginatedGroupListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedGroupList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(GroupFromJSON)),
    };
}

export function PaginatedGroupListToJSON(value?: PaginatedGroupList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(GroupToJSON)),
    };
}
