// @goauthentik/api/src/models/PaginatedBlueprintInstanceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { BlueprintInstance } from './BlueprintInstance';
import {
    BlueprintInstanceFromJSON,
    BlueprintInstanceFromJSONTyped,
    BlueprintInstanceToJSON,
} from './BlueprintInstance';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedBlueprintInstanceList
 */
export interface PaginatedBlueprintInstanceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedBlueprintInstanceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<BlueprintInstance>}
     * @memberof PaginatedBlueprintInstanceList
     */
    results: Array<BlueprintInstance>;
}

/**
 * Check if a given object implements the PaginatedBlueprintInstanceList interface.
 */
export function instanceOfPaginatedBlueprintInstanceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedBlueprintInstanceListFromJSON(json: any): PaginatedBlueprintInstanceList {
    return PaginatedBlueprintInstanceListFromJSONTyped(json, false);
}

export function PaginatedBlueprintInstanceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedBlueprintInstanceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(BlueprintInstanceFromJSON)),
    };
}

export function PaginatedBlueprintInstanceListToJSON(value?: PaginatedBlueprintInstanceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(BlueprintInstanceToJSON)),
    };
}
