// @goauthentik/api/src/models/PaginatedKubernetesServiceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { KubernetesServiceConnection } from './KubernetesServiceConnection';
import {
    KubernetesServiceConnectionFromJSON,
    KubernetesServiceConnectionFromJSONTyped,
    KubernetesServiceConnectionToJSON,
} from './KubernetesServiceConnection';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedKubernetesServiceConnectionList
 */
export interface PaginatedKubernetesServiceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedKubernetesServiceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<KubernetesServiceConnection>}
     * @memberof PaginatedKubernetesServiceConnectionList
     */
    results: Array<KubernetesServiceConnection>;
}

/**
 * Check if a given object implements the PaginatedKubernetesServiceConnectionList interface.
 */
export function instanceOfPaginatedKubernetesServiceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedKubernetesServiceConnectionListFromJSON(json: any): PaginatedKubernetesServiceConnectionList {
    return PaginatedKubernetesServiceConnectionListFromJSONTyped(json, false);
}

export function PaginatedKubernetesServiceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedKubernetesServiceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(KubernetesServiceConnectionFromJSON)),
    };
}

export function PaginatedKubernetesServiceConnectionListToJSON(value?: PaginatedKubernetesServiceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(KubernetesServiceConnectionToJSON)),
    };
}
