// @goauthentik/api/src/models/PaginatedLDAPPropertyMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPPropertyMapping } from './LDAPPropertyMapping';
import {
    LDAPPropertyMappingFromJSON,
    LDAPPropertyMappingFromJSONTyped,
    LDAPPropertyMappingToJSON,
} from './LDAPPropertyMapping';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedLDAPPropertyMappingList
 */
export interface PaginatedLDAPPropertyMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedLDAPPropertyMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<LDAPPropertyMapping>}
     * @memberof PaginatedLDAPPropertyMappingList
     */
    results: Array<LDAPPropertyMapping>;
}

/**
 * Check if a given object implements the PaginatedLDAPPropertyMappingList interface.
 */
export function instanceOfPaginatedLDAPPropertyMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedLDAPPropertyMappingListFromJSON(json: any): PaginatedLDAPPropertyMappingList {
    return PaginatedLDAPPropertyMappingListFromJSONTyped(json, false);
}

export function PaginatedLDAPPropertyMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedLDAPPropertyMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(LDAPPropertyMappingFromJSON)),
    };
}

export function PaginatedLDAPPropertyMappingListToJSON(value?: PaginatedLDAPPropertyMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(LDAPPropertyMappingToJSON)),
    };
}
