// @goauthentik/api/src/models/PaginatedSAMLPropertyMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SAMLPropertyMapping } from './SAMLPropertyMapping';
import {
    SAMLPropertyMappingFromJSON,
    SAMLPropertyMappingFromJSONTyped,
    SAMLPropertyMappingToJSON,
} from './SAMLPropertyMapping';

/**
 * 
 * @export
 * @interface PaginatedSAMLPropertyMappingList
 */
export interface PaginatedSAMLPropertyMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSAMLPropertyMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SAMLPropertyMapping>}
     * @memberof PaginatedSAMLPropertyMappingList
     */
    results: Array<SAMLPropertyMapping>;
}

/**
 * Check if a given object implements the PaginatedSAMLPropertyMappingList interface.
 */
export function instanceOfPaginatedSAMLPropertyMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSAMLPropertyMappingListFromJSON(json: any): PaginatedSAMLPropertyMappingList {
    return PaginatedSAMLPropertyMappingListFromJSONTyped(json, false);
}

export function PaginatedSAMLPropertyMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSAMLPropertyMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SAMLPropertyMappingFromJSON)),
    };
}

export function PaginatedSAMLPropertyMappingListToJSON(value?: PaginatedSAMLPropertyMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SAMLPropertyMappingToJSON)),
    };
}
