// @goauthentik/api/src/models/PaginatedPasswordPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PasswordPolicy } from './PasswordPolicy';
import {
    PasswordPolicyFromJSON,
    PasswordPolicyFromJSONTyped,
    PasswordPolicyToJSON,
} from './PasswordPolicy';

/**
 * 
 * @export
 * @interface PaginatedPasswordPolicyList
 */
export interface PaginatedPasswordPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPasswordPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PasswordPolicy>}
     * @memberof PaginatedPasswordPolicyList
     */
    results: Array<PasswordPolicy>;
}

/**
 * Check if a given object implements the PaginatedPasswordPolicyList interface.
 */
export function instanceOfPaginatedPasswordPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPasswordPolicyListFromJSON(json: any): PaginatedPasswordPolicyList {
    return PaginatedPasswordPolicyListFromJSONTyped(json, false);
}

export function PaginatedPasswordPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPasswordPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PasswordPolicyFromJSON)),
    };
}

export function PaginatedPasswordPolicyListToJSON(value?: PaginatedPasswordPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PasswordPolicyToJSON)),
    };
}
