// @goauthentik/api/src/models/PaginatedExtAuthStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ExtAuthStage } from './ExtAuthStage';
import {
    ExtAuthStageFromJSON,
    ExtAuthStageFromJSONTyped,
    ExtAuthStageToJSON,
} from './ExtAuthStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedExtAuthStageList
 */
export interface PaginatedExtAuthStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedExtAuthStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ExtAuthStage>}
     * @memberof PaginatedExtAuthStageList
     */
    results: Array<ExtAuthStage>;
}

/**
 * Check if a given object implements the PaginatedExtAuthStageList interface.
 */
export function instanceOfPaginatedExtAuthStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedExtAuthStageListFromJSON(json: any): PaginatedExtAuthStageList {
    return PaginatedExtAuthStageListFromJSONTyped(json, false);
}

export function PaginatedExtAuthStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedExtAuthStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ExtAuthStageFromJSON)),
    };
}

export function PaginatedExtAuthStageListToJSON(value?: PaginatedExtAuthStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ExtAuthStageToJSON)),
    };
}
