// @goauthentik/api/src/models/PaginatedRACPropertyMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { RACPropertyMapping } from './RACPropertyMapping';
import {
    RACPropertyMappingFromJSON,
    RACPropertyMappingFromJSONTyped,
    RACPropertyMappingToJSON,
} from './RACPropertyMapping';

/**
 * 
 * @export
 * @interface PaginatedRACPropertyMappingList
 */
export interface PaginatedRACPropertyMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRACPropertyMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<RACPropertyMapping>}
     * @memberof PaginatedRACPropertyMappingList
     */
    results: Array<RACPropertyMapping>;
}

/**
 * Check if a given object implements the PaginatedRACPropertyMappingList interface.
 */
export function instanceOfPaginatedRACPropertyMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRACPropertyMappingListFromJSON(json: any): PaginatedRACPropertyMappingList {
    return PaginatedRACPropertyMappingListFromJSONTyped(json, false);
}

export function PaginatedRACPropertyMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRACPropertyMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RACPropertyMappingFromJSON)),
    };
}

export function PaginatedRACPropertyMappingListToJSON(value?: PaginatedRACPropertyMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RACPropertyMappingToJSON)),
    };
}
