// @goauthentik/api/src/models/PaginatedNotificationTransportList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { NotificationTransport } from './NotificationTransport';
import {
    NotificationTransportFromJSON,
    NotificationTransportFromJSONTyped,
    NotificationTransportToJSON,
} from './NotificationTransport';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedNotificationTransportList
 */
export interface PaginatedNotificationTransportList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedNotificationTransportList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<NotificationTransport>}
     * @memberof PaginatedNotificationTransportList
     */
    results: Array<NotificationTransport>;
}

/**
 * Check if a given object implements the PaginatedNotificationTransportList interface.
 */
export function instanceOfPaginatedNotificationTransportList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedNotificationTransportListFromJSON(json: any): PaginatedNotificationTransportList {
    return PaginatedNotificationTransportListFromJSONTyped(json, false);
}

export function PaginatedNotificationTransportListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedNotificationTransportList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(NotificationTransportFromJSON)),
    };
}

export function PaginatedNotificationTransportListToJSON(value?: PaginatedNotificationTransportList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(NotificationTransportToJSON)),
    };
}
