// @goauthentik/api/src/models/PaginatedProxyOutpostConfigList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { ProxyOutpostConfig } from './ProxyOutpostConfig';
import {
    ProxyOutpostConfigFromJSON,
    ProxyOutpostConfigFromJSONTyped,
    ProxyOutpostConfigToJSON,
} from './ProxyOutpostConfig';

/**
 * 
 * @export
 * @interface PaginatedProxyOutpostConfigList
 */
export interface PaginatedProxyOutpostConfigList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedProxyOutpostConfigList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ProxyOutpostConfig>}
     * @memberof PaginatedProxyOutpostConfigList
     */
    results: Array<ProxyOutpostConfig>;
}

/**
 * Check if a given object implements the PaginatedProxyOutpostConfigList interface.
 */
export function instanceOfPaginatedProxyOutpostConfigList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedProxyOutpostConfigListFromJSON(json: any): PaginatedProxyOutpostConfigList {
    return PaginatedProxyOutpostConfigListFromJSONTyped(json, false);
}

export function PaginatedProxyOutpostConfigListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedProxyOutpostConfigList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ProxyOutpostConfigFromJSON)),
    };
}

export function PaginatedProxyOutpostConfigListToJSON(value?: PaginatedProxyOutpostConfigList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ProxyOutpostConfigToJSON)),
    };
}
