// @goauthentik/api/src/models/PaginatedNotificationWebhookMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { NotificationWebhookMapping } from './NotificationWebhookMapping';
import {
    NotificationWebhookMappingFromJSON,
    NotificationWebhookMappingFromJSONTyped,
    NotificationWebhookMappingToJSON,
} from './NotificationWebhookMapping';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedNotificationWebhookMappingList
 */
export interface PaginatedNotificationWebhookMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedNotificationWebhookMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<NotificationWebhookMapping>}
     * @memberof PaginatedNotificationWebhookMappingList
     */
    results: Array<NotificationWebhookMapping>;
}

/**
 * Check if a given object implements the PaginatedNotificationWebhookMappingList interface.
 */
export function instanceOfPaginatedNotificationWebhookMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedNotificationWebhookMappingListFromJSON(json: any): PaginatedNotificationWebhookMappingList {
    return PaginatedNotificationWebhookMappingListFromJSONTyped(json, false);
}

export function PaginatedNotificationWebhookMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedNotificationWebhookMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(NotificationWebhookMappingFromJSON)),
    };
}

export function PaginatedNotificationWebhookMappingListToJSON(value?: PaginatedNotificationWebhookMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(NotificationWebhookMappingToJSON)),
    };
}
