// @goauthentik/api/src/models/PaginatedReputationPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { ReputationPolicy } from './ReputationPolicy';
import {
    ReputationPolicyFromJSON,
    ReputationPolicyFromJSONTyped,
    ReputationPolicyToJSON,
} from './ReputationPolicy';

/**
 * 
 * @export
 * @interface PaginatedReputationPolicyList
 */
export interface PaginatedReputationPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedReputationPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ReputationPolicy>}
     * @memberof PaginatedReputationPolicyList
     */
    results: Array<ReputationPolicy>;
}

/**
 * Check if a given object implements the PaginatedReputationPolicyList interface.
 */
export function instanceOfPaginatedReputationPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedReputationPolicyListFromJSON(json: any): PaginatedReputationPolicyList {
    return PaginatedReputationPolicyListFromJSONTyped(json, false);
}

export function PaginatedReputationPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedReputationPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ReputationPolicyFromJSON)),
    };
}

export function PaginatedReputationPolicyListToJSON(value?: PaginatedReputationPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ReputationPolicyToJSON)),
    };
}
