// @goauthentik/api/src/models/PaginatedNotificationList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Notification } from './Notification';
import {
    NotificationFromJSON,
    NotificationFromJSONTyped,
    NotificationToJSON,
} from './Notification';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedNotificationList
 */
export interface PaginatedNotificationList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedNotificationList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Notification>}
     * @memberof PaginatedNotificationList
     */
    results: Array<Notification>;
}

/**
 * Check if a given object implements the PaginatedNotificationList interface.
 */
export function instanceOfPaginatedNotificationList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedNotificationListFromJSON(json: any): PaginatedNotificationList {
    return PaginatedNotificationListFromJSONTyped(json, false);
}

export function PaginatedNotificationListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedNotificationList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(NotificationFromJSON)),
    };
}

export function PaginatedNotificationListToJSON(value?: PaginatedNotificationList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(NotificationToJSON)),
    };
}
