// @goauthentik/api/src/models/PaginatedOAuthSourceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { OAuthSource } from './OAuthSource';
import {
    OAuthSourceFromJSON,
    OAuthSourceFromJSONTyped,
    OAuthSourceToJSON,
} from './OAuthSource';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedOAuthSourceList
 */
export interface PaginatedOAuthSourceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedOAuthSourceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<OAuthSource>}
     * @memberof PaginatedOAuthSourceList
     */
    results: Array<OAuthSource>;
}

/**
 * Check if a given object implements the PaginatedOAuthSourceList interface.
 */
export function instanceOfPaginatedOAuthSourceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedOAuthSourceListFromJSON(json: any): PaginatedOAuthSourceList {
    return PaginatedOAuthSourceListFromJSONTyped(json, false);
}

export function PaginatedOAuthSourceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedOAuthSourceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(OAuthSourceFromJSON)),
    };
}

export function PaginatedOAuthSourceListToJSON(value?: PaginatedOAuthSourceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(OAuthSourceToJSON)),
    };
}
