// @goauthentik/api/src/models/PaginatedStaticDeviceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { StaticDevice } from './StaticDevice';
import {
    StaticDeviceFromJSON,
    StaticDeviceFromJSONTyped,
    StaticDeviceToJSON,
} from './StaticDevice';

/**
 * 
 * @export
 * @interface PaginatedStaticDeviceList
 */
export interface PaginatedStaticDeviceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedStaticDeviceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<StaticDevice>}
     * @memberof PaginatedStaticDeviceList
     */
    results: Array<StaticDevice>;
}

/**
 * Check if a given object implements the PaginatedStaticDeviceList interface.
 */
export function instanceOfPaginatedStaticDeviceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedStaticDeviceListFromJSON(json: any): PaginatedStaticDeviceList {
    return PaginatedStaticDeviceListFromJSONTyped(json, false);
}

export function PaginatedStaticDeviceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedStaticDeviceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(StaticDeviceFromJSON)),
    };
}

export function PaginatedStaticDeviceListToJSON(value?: PaginatedStaticDeviceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(StaticDeviceToJSON)),
    };
}
