// @goauthentik/api/src/models/PaginatedDummyStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DummyStage } from './DummyStage';
import {
    DummyStageFromJSON,
    DummyStageFromJSONTyped,
    DummyStageToJSON,
} from './DummyStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDummyStageList
 */
export interface PaginatedDummyStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDummyStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<DummyStage>}
     * @memberof PaginatedDummyStageList
     */
    results: Array<DummyStage>;
}

/**
 * Check if a given object implements the PaginatedDummyStageList interface.
 */
export function instanceOfPaginatedDummyStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDummyStageListFromJSON(json: any): PaginatedDummyStageList {
    return PaginatedDummyStageListFromJSONTyped(json, false);
}

export function PaginatedDummyStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDummyStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DummyStageFromJSON)),
    };
}

export function PaginatedDummyStageListToJSON(value?: PaginatedDummyStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DummyStageToJSON)),
    };
}
