// @goauthentik/api/src/models/PaginatedRACProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { RACProvider } from './RACProvider';
import {
    RACProviderFromJSON,
    RACProviderFromJSONTyped,
    RACProviderToJSON,
} from './RACProvider';

/**
 * 
 * @export
 * @interface PaginatedRACProviderList
 */
export interface PaginatedRACProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRACProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<RACProvider>}
     * @memberof PaginatedRACProviderList
     */
    results: Array<RACProvider>;
}

/**
 * Check if a given object implements the PaginatedRACProviderList interface.
 */
export function instanceOfPaginatedRACProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRACProviderListFromJSON(json: any): PaginatedRACProviderList {
    return PaginatedRACProviderListFromJSONTyped(json, false);
}

export function PaginatedRACProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRACProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RACProviderFromJSON)),
    };
}

export function PaginatedRACProviderListToJSON(value?: PaginatedRACProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RACProviderToJSON)),
    };
}
