// @goauthentik/api/src/models/PaginatedSystemTaskList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SystemTask } from './SystemTask';
import {
    SystemTaskFromJSON,
    SystemTaskFromJSONTyped,
    SystemTaskToJSON,
} from './SystemTask';

/**
 * 
 * @export
 * @interface PaginatedSystemTaskList
 */
export interface PaginatedSystemTaskList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSystemTaskList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SystemTask>}
     * @memberof PaginatedSystemTaskList
     */
    results: Array<SystemTask>;
}

/**
 * Check if a given object implements the PaginatedSystemTaskList interface.
 */
export function instanceOfPaginatedSystemTaskList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSystemTaskListFromJSON(json: any): PaginatedSystemTaskList {
    return PaginatedSystemTaskListFromJSONTyped(json, false);
}

export function PaginatedSystemTaskListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSystemTaskList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SystemTaskFromJSON)),
    };
}

export function PaginatedSystemTaskListToJSON(value?: PaginatedSystemTaskList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SystemTaskToJSON)),
    };
}
