// @goauthentik/api/src/models/PaginatedPropertyMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PropertyMapping } from './PropertyMapping';
import {
    PropertyMappingFromJSON,
    PropertyMappingFromJSONTyped,
    PropertyMappingToJSON,
} from './PropertyMapping';

/**
 * 
 * @export
 * @interface PaginatedPropertyMappingList
 */
export interface PaginatedPropertyMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPropertyMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PropertyMapping>}
     * @memberof PaginatedPropertyMappingList
     */
    results: Array<PropertyMapping>;
}

/**
 * Check if a given object implements the PaginatedPropertyMappingList interface.
 */
export function instanceOfPaginatedPropertyMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPropertyMappingListFromJSON(json: any): PaginatedPropertyMappingList {
    return PaginatedPropertyMappingListFromJSONTyped(json, false);
}

export function PaginatedPropertyMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPropertyMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PropertyMappingFromJSON)),
    };
}

export function PaginatedPropertyMappingListToJSON(value?: PaginatedPropertyMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PropertyMappingToJSON)),
    };
}
