// @goauthentik/api/src/models/PaginatedOutpostList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Outpost } from './Outpost';
import {
    OutpostFromJSON,
    OutpostFromJSONTyped,
    OutpostToJSON,
} from './Outpost';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedOutpostList
 */
export interface PaginatedOutpostList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedOutpostList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Outpost>}
     * @memberof PaginatedOutpostList
     */
    results: Array<Outpost>;
}

/**
 * Check if a given object implements the PaginatedOutpostList interface.
 */
export function instanceOfPaginatedOutpostList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedOutpostListFromJSON(json: any): PaginatedOutpostList {
    return PaginatedOutpostListFromJSONTyped(json, false);
}

export function PaginatedOutpostListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedOutpostList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(OutpostFromJSON)),
    };
}

export function PaginatedOutpostListToJSON(value?: PaginatedOutpostList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(OutpostToJSON)),
    };
}
