// @goauthentik/api/src/models/PaginatedRadiusOutpostConfigList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { RadiusOutpostConfig } from './RadiusOutpostConfig';
import {
    RadiusOutpostConfigFromJSON,
    RadiusOutpostConfigFromJSONTyped,
    RadiusOutpostConfigToJSON,
} from './RadiusOutpostConfig';

/**
 * 
 * @export
 * @interface PaginatedRadiusOutpostConfigList
 */
export interface PaginatedRadiusOutpostConfigList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRadiusOutpostConfigList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<RadiusOutpostConfig>}
     * @memberof PaginatedRadiusOutpostConfigList
     */
    results: Array<RadiusOutpostConfig>;
}

/**
 * Check if a given object implements the PaginatedRadiusOutpostConfigList interface.
 */
export function instanceOfPaginatedRadiusOutpostConfigList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRadiusOutpostConfigListFromJSON(json: any): PaginatedRadiusOutpostConfigList {
    return PaginatedRadiusOutpostConfigListFromJSONTyped(json, false);
}

export function PaginatedRadiusOutpostConfigListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRadiusOutpostConfigList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RadiusOutpostConfigFromJSON)),
    };
}

export function PaginatedRadiusOutpostConfigListToJSON(value?: PaginatedRadiusOutpostConfigList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RadiusOutpostConfigToJSON)),
    };
}
