// @goauthentik/api/src/models/PaginatedDenyStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DenyStage } from './DenyStage';
import {
    DenyStageFromJSON,
    DenyStageFromJSONTyped,
    DenyStageToJSON,
} from './DenyStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDenyStageList
 */
export interface PaginatedDenyStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDenyStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<DenyStage>}
     * @memberof PaginatedDenyStageList
     */
    results: Array<DenyStage>;
}

/**
 * Check if a given object implements the PaginatedDenyStageList interface.
 */
export function instanceOfPaginatedDenyStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDenyStageListFromJSON(json: any): PaginatedDenyStageList {
    return PaginatedDenyStageListFromJSONTyped(json, false);
}

export function PaginatedDenyStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDenyStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DenyStageFromJSON)),
    };
}

export function PaginatedDenyStageListToJSON(value?: PaginatedDenyStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DenyStageToJSON)),
    };
}
