// @goauthentik/api/src/models/PaginatedTokenList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Token } from './Token';
import {
    TokenFromJSON,
    TokenFromJSONTyped,
    TokenToJSON,
} from './Token';

/**
 * 
 * @export
 * @interface PaginatedTokenList
 */
export interface PaginatedTokenList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedTokenList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Token>}
     * @memberof PaginatedTokenList
     */
    results: Array<Token>;
}

/**
 * Check if a given object implements the PaginatedTokenList interface.
 */
export function instanceOfPaginatedTokenList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedTokenListFromJSON(json: any): PaginatedTokenList {
    return PaginatedTokenListFromJSONTyped(json, false);
}

export function PaginatedTokenListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedTokenList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(TokenFromJSON)),
    };
}

export function PaginatedTokenListToJSON(value?: PaginatedTokenList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(TokenToJSON)),
    };
}
