// @goauthentik/api/src/models/PaginatedExpressionPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ExpressionPolicy } from './ExpressionPolicy';
import {
    ExpressionPolicyFromJSON,
    ExpressionPolicyFromJSONTyped,
    ExpressionPolicyToJSON,
} from './ExpressionPolicy';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedExpressionPolicyList
 */
export interface PaginatedExpressionPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedExpressionPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ExpressionPolicy>}
     * @memberof PaginatedExpressionPolicyList
     */
    results: Array<ExpressionPolicy>;
}

/**
 * Check if a given object implements the PaginatedExpressionPolicyList interface.
 */
export function instanceOfPaginatedExpressionPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedExpressionPolicyListFromJSON(json: any): PaginatedExpressionPolicyList {
    return PaginatedExpressionPolicyListFromJSONTyped(json, false);
}

export function PaginatedExpressionPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedExpressionPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ExpressionPolicyFromJSON)),
    };
}

export function PaginatedExpressionPolicyListToJSON(value?: PaginatedExpressionPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ExpressionPolicyToJSON)),
    };
}
