// @goauthentik/api/src/models/PaginatedLDAPSourceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPSource } from './LDAPSource';
import {
    LDAPSourceFromJSON,
    LDAPSourceFromJSONTyped,
    LDAPSourceToJSON,
} from './LDAPSource';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedLDAPSourceList
 */
export interface PaginatedLDAPSourceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedLDAPSourceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<LDAPSource>}
     * @memberof PaginatedLDAPSourceList
     */
    results: Array<LDAPSource>;
}

/**
 * Check if a given object implements the PaginatedLDAPSourceList interface.
 */
export function instanceOfPaginatedLDAPSourceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedLDAPSourceListFromJSON(json: any): PaginatedLDAPSourceList {
    return PaginatedLDAPSourceListFromJSONTyped(json, false);
}

export function PaginatedLDAPSourceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedLDAPSourceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(LDAPSourceFromJSON)),
    };
}

export function PaginatedLDAPSourceListToJSON(value?: PaginatedLDAPSourceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(LDAPSourceToJSON)),
    };
}
