// @goauthentik/api/src/models/PaginatedSCIMMappingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SCIMMapping } from './SCIMMapping';
import {
    SCIMMappingFromJSON,
    SCIMMappingFromJSONTyped,
    SCIMMappingToJSON,
} from './SCIMMapping';

/**
 * 
 * @export
 * @interface PaginatedSCIMMappingList
 */
export interface PaginatedSCIMMappingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSCIMMappingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SCIMMapping>}
     * @memberof PaginatedSCIMMappingList
     */
    results: Array<SCIMMapping>;
}

/**
 * Check if a given object implements the PaginatedSCIMMappingList interface.
 */
export function instanceOfPaginatedSCIMMappingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSCIMMappingListFromJSON(json: any): PaginatedSCIMMappingList {
    return PaginatedSCIMMappingListFromJSONTyped(json, false);
}

export function PaginatedSCIMMappingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSCIMMappingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SCIMMappingFromJSON)),
    };
}

export function PaginatedSCIMMappingListToJSON(value?: PaginatedSCIMMappingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SCIMMappingToJSON)),
    };
}
