// @goauthentik/api/src/models/PaginatedLDAPProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPProvider } from './LDAPProvider';
import {
    LDAPProviderFromJSON,
    LDAPProviderFromJSONTyped,
    LDAPProviderToJSON,
} from './LDAPProvider';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedLDAPProviderList
 */
export interface PaginatedLDAPProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedLDAPProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<LDAPProvider>}
     * @memberof PaginatedLDAPProviderList
     */
    results: Array<LDAPProvider>;
}

/**
 * Check if a given object implements the PaginatedLDAPProviderList interface.
 */
export function instanceOfPaginatedLDAPProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedLDAPProviderListFromJSON(json: any): PaginatedLDAPProviderList {
    return PaginatedLDAPProviderListFromJSONTyped(json, false);
}

export function PaginatedLDAPProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedLDAPProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(LDAPProviderFromJSON)),
    };
}

export function PaginatedLDAPProviderListToJSON(value?: PaginatedLDAPProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(LDAPProviderToJSON)),
    };
}
