// @goauthentik/api/src/models/PaginatedFlowList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Flow } from './Flow';
import {
    FlowFromJSON,
    FlowFromJSONTyped,
    FlowToJSON,
} from './Flow';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedFlowList
 */
export interface PaginatedFlowList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedFlowList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Flow>}
     * @memberof PaginatedFlowList
     */
    results: Array<Flow>;
}

/**
 * Check if a given object implements the PaginatedFlowList interface.
 */
export function instanceOfPaginatedFlowList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedFlowListFromJSON(json: any): PaginatedFlowList {
    return PaginatedFlowListFromJSONTyped(json, false);
}

export function PaginatedFlowListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedFlowList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(FlowFromJSON)),
    };
}

export function PaginatedFlowListToJSON(value?: PaginatedFlowList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(FlowToJSON)),
    };
}
