// @goauthentik/api/src/models/PaginatedConsentStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ConsentStage } from './ConsentStage';
import {
    ConsentStageFromJSON,
    ConsentStageFromJSONTyped,
    ConsentStageToJSON,
} from './ConsentStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedConsentStageList
 */
export interface PaginatedConsentStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedConsentStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<ConsentStage>}
     * @memberof PaginatedConsentStageList
     */
    results: Array<ConsentStage>;
}

/**
 * Check if a given object implements the PaginatedConsentStageList interface.
 */
export function instanceOfPaginatedConsentStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedConsentStageListFromJSON(json: any): PaginatedConsentStageList {
    return PaginatedConsentStageListFromJSONTyped(json, false);
}

export function PaginatedConsentStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedConsentStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(ConsentStageFromJSON)),
    };
}

export function PaginatedConsentStageListToJSON(value?: PaginatedConsentStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(ConsentStageToJSON)),
    };
}
