// @goauthentik/api/src/models/PaginatedFlowStageBindingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowStageBinding } from './FlowStageBinding';
import {
    FlowStageBindingFromJSON,
    FlowStageBindingFromJSONTyped,
    FlowStageBindingToJSON,
} from './FlowStageBinding';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedFlowStageBindingList
 */
export interface PaginatedFlowStageBindingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedFlowStageBindingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<FlowStageBinding>}
     * @memberof PaginatedFlowStageBindingList
     */
    results: Array<FlowStageBinding>;
}

/**
 * Check if a given object implements the PaginatedFlowStageBindingList interface.
 */
export function instanceOfPaginatedFlowStageBindingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedFlowStageBindingListFromJSON(json: any): PaginatedFlowStageBindingList {
    return PaginatedFlowStageBindingListFromJSONTyped(json, false);
}

export function PaginatedFlowStageBindingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedFlowStageBindingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(FlowStageBindingFromJSON)),
    };
}

export function PaginatedFlowStageBindingListToJSON(value?: PaginatedFlowStageBindingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(FlowStageBindingToJSON)),
    };
}
