// @goauthentik/api/src/models/PaginatedUserOAuthSourceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserOAuthSourceConnection } from './UserOAuthSourceConnection';
import {
    UserOAuthSourceConnectionFromJSON,
    UserOAuthSourceConnectionFromJSONTyped,
    UserOAuthSourceConnectionToJSON,
} from './UserOAuthSourceConnection';

/**
 * 
 * @export
 * @interface PaginatedUserOAuthSourceConnectionList
 */
export interface PaginatedUserOAuthSourceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserOAuthSourceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserOAuthSourceConnection>}
     * @memberof PaginatedUserOAuthSourceConnectionList
     */
    results: Array<UserOAuthSourceConnection>;
}

/**
 * Check if a given object implements the PaginatedUserOAuthSourceConnectionList interface.
 */
export function instanceOfPaginatedUserOAuthSourceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserOAuthSourceConnectionListFromJSON(json: any): PaginatedUserOAuthSourceConnectionList {
    return PaginatedUserOAuthSourceConnectionListFromJSONTyped(json, false);
}

export function PaginatedUserOAuthSourceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserOAuthSourceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserOAuthSourceConnectionFromJSON)),
    };
}

export function PaginatedUserOAuthSourceConnectionListToJSON(value?: PaginatedUserOAuthSourceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserOAuthSourceConnectionToJSON)),
    };
}
