// @goauthentik/api/src/models/PaginatedCertificateKeyPairList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { CertificateKeyPair } from './CertificateKeyPair';
import {
    CertificateKeyPairFromJSON,
    CertificateKeyPairFromJSONTyped,
    CertificateKeyPairToJSON,
} from './CertificateKeyPair';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedCertificateKeyPairList
 */
export interface PaginatedCertificateKeyPairList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedCertificateKeyPairList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<CertificateKeyPair>}
     * @memberof PaginatedCertificateKeyPairList
     */
    results: Array<CertificateKeyPair>;
}

/**
 * Check if a given object implements the PaginatedCertificateKeyPairList interface.
 */
export function instanceOfPaginatedCertificateKeyPairList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedCertificateKeyPairListFromJSON(json: any): PaginatedCertificateKeyPairList {
    return PaginatedCertificateKeyPairListFromJSONTyped(json, false);
}

export function PaginatedCertificateKeyPairListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedCertificateKeyPairList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(CertificateKeyPairFromJSON)),
    };
}

export function PaginatedCertificateKeyPairListToJSON(value?: PaginatedCertificateKeyPairList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(CertificateKeyPairToJSON)),
    };
}
