// @goauthentik/api/src/models/PaginatedTOTPDeviceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { TOTPDevice } from './TOTPDevice';
import {
    TOTPDeviceFromJSON,
    TOTPDeviceFromJSONTyped,
    TOTPDeviceToJSON,
} from './TOTPDevice';

/**
 * 
 * @export
 * @interface PaginatedTOTPDeviceList
 */
export interface PaginatedTOTPDeviceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedTOTPDeviceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<TOTPDevice>}
     * @memberof PaginatedTOTPDeviceList
     */
    results: Array<TOTPDevice>;
}

/**
 * Check if a given object implements the PaginatedTOTPDeviceList interface.
 */
export function instanceOfPaginatedTOTPDeviceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedTOTPDeviceListFromJSON(json: any): PaginatedTOTPDeviceList {
    return PaginatedTOTPDeviceListFromJSONTyped(json, false);
}

export function PaginatedTOTPDeviceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedTOTPDeviceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(TOTPDeviceFromJSON)),
    };
}

export function PaginatedTOTPDeviceListToJSON(value?: PaginatedTOTPDeviceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(TOTPDeviceToJSON)),
    };
}
