// @goauthentik/api/src/models/PaginatedPolicyBindingList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PolicyBinding } from './PolicyBinding';
import {
    PolicyBindingFromJSON,
    PolicyBindingFromJSONTyped,
    PolicyBindingToJSON,
} from './PolicyBinding';

/**
 * 
 * @export
 * @interface PaginatedPolicyBindingList
 */
export interface PaginatedPolicyBindingList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPolicyBindingList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PolicyBinding>}
     * @memberof PaginatedPolicyBindingList
     */
    results: Array<PolicyBinding>;
}

/**
 * Check if a given object implements the PaginatedPolicyBindingList interface.
 */
export function instanceOfPaginatedPolicyBindingList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPolicyBindingListFromJSON(json: any): PaginatedPolicyBindingList {
    return PaginatedPolicyBindingListFromJSONTyped(json, false);
}

export function PaginatedPolicyBindingListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPolicyBindingList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PolicyBindingFromJSON)),
    };
}

export function PaginatedPolicyBindingListToJSON(value?: PaginatedPolicyBindingList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PolicyBindingToJSON)),
    };
}
