// @goauthentik/api/src/models/PaginatedWebAuthnDeviceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { WebAuthnDevice } from './WebAuthnDevice';
import {
    WebAuthnDeviceFromJSON,
    WebAuthnDeviceFromJSONTyped,
    WebAuthnDeviceToJSON,
} from './WebAuthnDevice';

/**
 * 
 * @export
 * @interface PaginatedWebAuthnDeviceList
 */
export interface PaginatedWebAuthnDeviceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedWebAuthnDeviceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<WebAuthnDevice>}
     * @memberof PaginatedWebAuthnDeviceList
     */
    results: Array<WebAuthnDevice>;
}

/**
 * Check if a given object implements the PaginatedWebAuthnDeviceList interface.
 */
export function instanceOfPaginatedWebAuthnDeviceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedWebAuthnDeviceListFromJSON(json: any): PaginatedWebAuthnDeviceList {
    return PaginatedWebAuthnDeviceListFromJSONTyped(json, false);
}

export function PaginatedWebAuthnDeviceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedWebAuthnDeviceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(WebAuthnDeviceFromJSON)),
    };
}

export function PaginatedWebAuthnDeviceListToJSON(value?: PaginatedWebAuthnDeviceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(WebAuthnDeviceToJSON)),
    };
}
