// @goauthentik/api/src/models/PaginatedTenantList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Tenant } from './Tenant';
import {
    TenantFromJSON,
    TenantFromJSONTyped,
    TenantToJSON,
} from './Tenant';

/**
 * 
 * @export
 * @interface PaginatedTenantList
 */
export interface PaginatedTenantList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedTenantList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Tenant>}
     * @memberof PaginatedTenantList
     */
    results: Array<Tenant>;
}

/**
 * Check if a given object implements the PaginatedTenantList interface.
 */
export function instanceOfPaginatedTenantList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedTenantListFromJSON(json: any): PaginatedTenantList {
    return PaginatedTenantListFromJSONTyped(json, false);
}

export function PaginatedTenantListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedTenantList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(TenantFromJSON)),
    };
}

export function PaginatedTenantListToJSON(value?: PaginatedTenantList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(TenantToJSON)),
    };
}
