// @goauthentik/api/src/models/PaginatedRoleList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Role } from './Role';
import {
    RoleFromJSON,
    RoleFromJSONTyped,
    RoleToJSON,
} from './Role';

/**
 * 
 * @export
 * @interface PaginatedRoleList
 */
export interface PaginatedRoleList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRoleList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Role>}
     * @memberof PaginatedRoleList
     */
    results: Array<Role>;
}

/**
 * Check if a given object implements the PaginatedRoleList interface.
 */
export function instanceOfPaginatedRoleList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRoleListFromJSON(json: any): PaginatedRoleList {
    return PaginatedRoleListFromJSONTyped(json, false);
}

export function PaginatedRoleListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRoleList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RoleFromJSON)),
    };
}

export function PaginatedRoleListToJSON(value?: PaginatedRoleList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RoleToJSON)),
    };
}
