// @goauthentik/api/src/models/PaginatedUserLogoutStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserLogoutStage } from './UserLogoutStage';
import {
    UserLogoutStageFromJSON,
    UserLogoutStageFromJSONTyped,
    UserLogoutStageToJSON,
} from './UserLogoutStage';

/**
 * 
 * @export
 * @interface PaginatedUserLogoutStageList
 */
export interface PaginatedUserLogoutStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserLogoutStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserLogoutStage>}
     * @memberof PaginatedUserLogoutStageList
     */
    results: Array<UserLogoutStage>;
}

/**
 * Check if a given object implements the PaginatedUserLogoutStageList interface.
 */
export function instanceOfPaginatedUserLogoutStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserLogoutStageListFromJSON(json: any): PaginatedUserLogoutStageList {
    return PaginatedUserLogoutStageListFromJSONTyped(json, false);
}

export function PaginatedUserLogoutStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserLogoutStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserLogoutStageFromJSON)),
    };
}

export function PaginatedUserLogoutStageListToJSON(value?: PaginatedUserLogoutStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserLogoutStageToJSON)),
    };
}
