// @goauthentik/api/src/models/PaginatedRoleAssignedObjectPermissionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { RoleAssignedObjectPermission } from './RoleAssignedObjectPermission';
import {
    RoleAssignedObjectPermissionFromJSON,
    RoleAssignedObjectPermissionFromJSONTyped,
    RoleAssignedObjectPermissionToJSON,
} from './RoleAssignedObjectPermission';

/**
 * 
 * @export
 * @interface PaginatedRoleAssignedObjectPermissionList
 */
export interface PaginatedRoleAssignedObjectPermissionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRoleAssignedObjectPermissionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<RoleAssignedObjectPermission>}
     * @memberof PaginatedRoleAssignedObjectPermissionList
     */
    results: Array<RoleAssignedObjectPermission>;
}

/**
 * Check if a given object implements the PaginatedRoleAssignedObjectPermissionList interface.
 */
export function instanceOfPaginatedRoleAssignedObjectPermissionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRoleAssignedObjectPermissionListFromJSON(json: any): PaginatedRoleAssignedObjectPermissionList {
    return PaginatedRoleAssignedObjectPermissionListFromJSONTyped(json, false);
}

export function PaginatedRoleAssignedObjectPermissionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRoleAssignedObjectPermissionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RoleAssignedObjectPermissionFromJSON)),
    };
}

export function PaginatedRoleAssignedObjectPermissionListToJSON(value?: PaginatedRoleAssignedObjectPermissionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RoleAssignedObjectPermissionToJSON)),
    };
}
