// @goauthentik/api/src/models/PaginatedSCIMProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SCIMProvider } from './SCIMProvider';
import {
    SCIMProviderFromJSON,
    SCIMProviderFromJSONTyped,
    SCIMProviderToJSON,
} from './SCIMProvider';

/**
 * 
 * @export
 * @interface PaginatedSCIMProviderList
 */
export interface PaginatedSCIMProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSCIMProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SCIMProvider>}
     * @memberof PaginatedSCIMProviderList
     */
    results: Array<SCIMProvider>;
}

/**
 * Check if a given object implements the PaginatedSCIMProviderList interface.
 */
export function instanceOfPaginatedSCIMProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSCIMProviderListFromJSON(json: any): PaginatedSCIMProviderList {
    return PaginatedSCIMProviderListFromJSONTyped(json, false);
}

export function PaginatedSCIMProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSCIMProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SCIMProviderFromJSON)),
    };
}

export function PaginatedSCIMProviderListToJSON(value?: PaginatedSCIMProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SCIMProviderToJSON)),
    };
}
