// @goauthentik/api/src/models/PaginatedUserConsentList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserConsent } from './UserConsent';
import {
    UserConsentFromJSON,
    UserConsentFromJSONTyped,
    UserConsentToJSON,
} from './UserConsent';

/**
 * 
 * @export
 * @interface PaginatedUserConsentList
 */
export interface PaginatedUserConsentList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserConsentList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserConsent>}
     * @memberof PaginatedUserConsentList
     */
    results: Array<UserConsent>;
}

/**
 * Check if a given object implements the PaginatedUserConsentList interface.
 */
export function instanceOfPaginatedUserConsentList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserConsentListFromJSON(json: any): PaginatedUserConsentList {
    return PaginatedUserConsentListFromJSONTyped(json, false);
}

export function PaginatedUserConsentListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserConsentList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserConsentFromJSON)),
    };
}

export function PaginatedUserConsentListToJSON(value?: PaginatedUserConsentList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserConsentToJSON)),
    };
}
