// @goauthentik/api/src/models/PaginatedAuthenticatorValidateStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorValidateStage } from './AuthenticatorValidateStage';
import {
    AuthenticatorValidateStageFromJSON,
    AuthenticatorValidateStageFromJSONTyped,
    AuthenticatorValidateStageToJSON,
} from './AuthenticatorValidateStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticatorValidateStageList
 */
export interface PaginatedAuthenticatorValidateStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticatorValidateStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticatorValidateStage>}
     * @memberof PaginatedAuthenticatorValidateStageList
     */
    results: Array<AuthenticatorValidateStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticatorValidateStageList interface.
 */
export function instanceOfPaginatedAuthenticatorValidateStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticatorValidateStageListFromJSON(json: any): PaginatedAuthenticatorValidateStageList {
    return PaginatedAuthenticatorValidateStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticatorValidateStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticatorValidateStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticatorValidateStageFromJSON)),
    };
}

export function PaginatedAuthenticatorValidateStageListToJSON(value?: PaginatedAuthenticatorValidateStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticatorValidateStageToJSON)),
    };
}
