// @goauthentik/api/src/models/PaginatedDockerServiceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DockerServiceConnection } from './DockerServiceConnection';
import {
    DockerServiceConnectionFromJSON,
    DockerServiceConnectionFromJSONTyped,
    DockerServiceConnectionToJSON,
} from './DockerServiceConnection';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDockerServiceConnectionList
 */
export interface PaginatedDockerServiceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDockerServiceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<DockerServiceConnection>}
     * @memberof PaginatedDockerServiceConnectionList
     */
    results: Array<DockerServiceConnection>;
}

/**
 * Check if a given object implements the PaginatedDockerServiceConnectionList interface.
 */
export function instanceOfPaginatedDockerServiceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDockerServiceConnectionListFromJSON(json: any): PaginatedDockerServiceConnectionList {
    return PaginatedDockerServiceConnectionListFromJSONTyped(json, false);
}

export function PaginatedDockerServiceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDockerServiceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DockerServiceConnectionFromJSON)),
    };
}

export function PaginatedDockerServiceConnectionListToJSON(value?: PaginatedDockerServiceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DockerServiceConnectionToJSON)),
    };
}
