// @goauthentik/api/src/models/PaginatedDomainList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Domain } from './Domain';
import {
    DomainFromJSON,
    DomainFromJSONTyped,
    DomainToJSON,
} from './Domain';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDomainList
 */
export interface PaginatedDomainList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDomainList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Domain>}
     * @memberof PaginatedDomainList
     */
    results: Array<Domain>;
}

/**
 * Check if a given object implements the PaginatedDomainList interface.
 */
export function instanceOfPaginatedDomainList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDomainListFromJSON(json: any): PaginatedDomainList {
    return PaginatedDomainListFromJSONTyped(json, false);
}

export function PaginatedDomainListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDomainList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DomainFromJSON)),
    };
}

export function PaginatedDomainListToJSON(value?: PaginatedDomainList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DomainToJSON)),
    };
}
