// @goauthentik/api/src/models/PaginatedSAMLSourceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SAMLSource } from './SAMLSource';
import {
    SAMLSourceFromJSON,
    SAMLSourceFromJSONTyped,
    SAMLSourceToJSON,
} from './SAMLSource';

/**
 * 
 * @export
 * @interface PaginatedSAMLSourceList
 */
export interface PaginatedSAMLSourceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSAMLSourceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SAMLSource>}
     * @memberof PaginatedSAMLSourceList
     */
    results: Array<SAMLSource>;
}

/**
 * Check if a given object implements the PaginatedSAMLSourceList interface.
 */
export function instanceOfPaginatedSAMLSourceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSAMLSourceListFromJSON(json: any): PaginatedSAMLSourceList {
    return PaginatedSAMLSourceListFromJSONTyped(json, false);
}

export function PaginatedSAMLSourceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSAMLSourceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SAMLSourceFromJSON)),
    };
}

export function PaginatedSAMLSourceListToJSON(value?: PaginatedSAMLSourceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SAMLSourceToJSON)),
    };
}
