// @goauthentik/api/src/models/PaginatedPasswordStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PasswordStage } from './PasswordStage';
import {
    PasswordStageFromJSON,
    PasswordStageFromJSONTyped,
    PasswordStageToJSON,
} from './PasswordStage';

/**
 * 
 * @export
 * @interface PaginatedPasswordStageList
 */
export interface PaginatedPasswordStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPasswordStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PasswordStage>}
     * @memberof PaginatedPasswordStageList
     */
    results: Array<PasswordStage>;
}

/**
 * Check if a given object implements the PaginatedPasswordStageList interface.
 */
export function instanceOfPaginatedPasswordStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPasswordStageListFromJSON(json: any): PaginatedPasswordStageList {
    return PaginatedPasswordStageListFromJSONTyped(json, false);
}

export function PaginatedPasswordStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPasswordStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PasswordStageFromJSON)),
    };
}

export function PaginatedPasswordStageListToJSON(value?: PaginatedPasswordStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PasswordStageToJSON)),
    };
}
