// @goauthentik/api/src/models/PaginatedSourceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Source } from './Source';
import {
    SourceFromJSON,
    SourceFromJSONTyped,
    SourceToJSON,
} from './Source';

/**
 * 
 * @export
 * @interface PaginatedSourceList
 */
export interface PaginatedSourceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSourceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Source>}
     * @memberof PaginatedSourceList
     */
    results: Array<Source>;
}

/**
 * Check if a given object implements the PaginatedSourceList interface.
 */
export function instanceOfPaginatedSourceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSourceListFromJSON(json: any): PaginatedSourceList {
    return PaginatedSourceListFromJSONTyped(json, false);
}

export function PaginatedSourceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSourceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SourceFromJSON)),
    };
}

export function PaginatedSourceListToJSON(value?: PaginatedSourceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SourceToJSON)),
    };
}
