// @goauthentik/api/src/models/PaginatedPasswordExpiryPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PasswordExpiryPolicy } from './PasswordExpiryPolicy';
import {
    PasswordExpiryPolicyFromJSON,
    PasswordExpiryPolicyFromJSONTyped,
    PasswordExpiryPolicyToJSON,
} from './PasswordExpiryPolicy';

/**
 * 
 * @export
 * @interface PaginatedPasswordExpiryPolicyList
 */
export interface PaginatedPasswordExpiryPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPasswordExpiryPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PasswordExpiryPolicy>}
     * @memberof PaginatedPasswordExpiryPolicyList
     */
    results: Array<PasswordExpiryPolicy>;
}

/**
 * Check if a given object implements the PaginatedPasswordExpiryPolicyList interface.
 */
export function instanceOfPaginatedPasswordExpiryPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPasswordExpiryPolicyListFromJSON(json: any): PaginatedPasswordExpiryPolicyList {
    return PaginatedPasswordExpiryPolicyListFromJSONTyped(json, false);
}

export function PaginatedPasswordExpiryPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPasswordExpiryPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PasswordExpiryPolicyFromJSON)),
    };
}

export function PaginatedPasswordExpiryPolicyListToJSON(value?: PaginatedPasswordExpiryPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PasswordExpiryPolicyToJSON)),
    };
}
