// @goauthentik/api/src/models/PaginatedDuoDeviceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DuoDevice } from './DuoDevice';
import {
    DuoDeviceFromJSON,
    DuoDeviceFromJSONTyped,
    DuoDeviceToJSON,
} from './DuoDevice';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDuoDeviceList
 */
export interface PaginatedDuoDeviceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDuoDeviceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<DuoDevice>}
     * @memberof PaginatedDuoDeviceList
     */
    results: Array<DuoDevice>;
}

/**
 * Check if a given object implements the PaginatedDuoDeviceList interface.
 */
export function instanceOfPaginatedDuoDeviceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDuoDeviceListFromJSON(json: any): PaginatedDuoDeviceList {
    return PaginatedDuoDeviceListFromJSONTyped(json, false);
}

export function PaginatedDuoDeviceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDuoDeviceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DuoDeviceFromJSON)),
    };
}

export function PaginatedDuoDeviceListToJSON(value?: PaginatedDuoDeviceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DuoDeviceToJSON)),
    };
}
