// @goauthentik/api/src/models/PaginatedUserSourceConnectionList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserSourceConnection } from './UserSourceConnection';
import {
    UserSourceConnectionFromJSON,
    UserSourceConnectionFromJSONTyped,
    UserSourceConnectionToJSON,
} from './UserSourceConnection';

/**
 * 
 * @export
 * @interface PaginatedUserSourceConnectionList
 */
export interface PaginatedUserSourceConnectionList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserSourceConnectionList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserSourceConnection>}
     * @memberof PaginatedUserSourceConnectionList
     */
    results: Array<UserSourceConnection>;
}

/**
 * Check if a given object implements the PaginatedUserSourceConnectionList interface.
 */
export function instanceOfPaginatedUserSourceConnectionList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserSourceConnectionListFromJSON(json: any): PaginatedUserSourceConnectionList {
    return PaginatedUserSourceConnectionListFromJSONTyped(json, false);
}

export function PaginatedUserSourceConnectionListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserSourceConnectionList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserSourceConnectionFromJSON)),
    };
}

export function PaginatedUserSourceConnectionListToJSON(value?: PaginatedUserSourceConnectionList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserSourceConnectionToJSON)),
    };
}
