// @goauthentik/api/src/models/PaginatedDummyPolicyList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DummyPolicy } from './DummyPolicy';
import {
    DummyPolicyFromJSON,
    DummyPolicyFromJSONTyped,
    DummyPolicyToJSON,
} from './DummyPolicy';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedDummyPolicyList
 */
export interface PaginatedDummyPolicyList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedDummyPolicyList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<DummyPolicy>}
     * @memberof PaginatedDummyPolicyList
     */
    results: Array<DummyPolicy>;
}

/**
 * Check if a given object implements the PaginatedDummyPolicyList interface.
 */
export function instanceOfPaginatedDummyPolicyList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedDummyPolicyListFromJSON(json: any): PaginatedDummyPolicyList {
    return PaginatedDummyPolicyListFromJSONTyped(json, false);
}

export function PaginatedDummyPolicyListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedDummyPolicyList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(DummyPolicyFromJSON)),
    };
}

export function PaginatedDummyPolicyListToJSON(value?: PaginatedDummyPolicyList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(DummyPolicyToJSON)),
    };
}
