// @goauthentik/api/src/models/PaginatedUserDeleteStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserDeleteStage } from './UserDeleteStage';
import {
    UserDeleteStageFromJSON,
    UserDeleteStageFromJSONTyped,
    UserDeleteStageToJSON,
} from './UserDeleteStage';

/**
 * 
 * @export
 * @interface PaginatedUserDeleteStageList
 */
export interface PaginatedUserDeleteStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserDeleteStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserDeleteStage>}
     * @memberof PaginatedUserDeleteStageList
     */
    results: Array<UserDeleteStage>;
}

/**
 * Check if a given object implements the PaginatedUserDeleteStageList interface.
 */
export function instanceOfPaginatedUserDeleteStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserDeleteStageListFromJSON(json: any): PaginatedUserDeleteStageList {
    return PaginatedUserDeleteStageListFromJSONTyped(json, false);
}

export function PaginatedUserDeleteStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserDeleteStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserDeleteStageFromJSON)),
    };
}

export function PaginatedUserDeleteStageListToJSON(value?: PaginatedUserDeleteStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserDeleteStageToJSON)),
    };
}
