// @goauthentik/api/src/models/PaginatedEndpointList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Endpoint } from './Endpoint';
import {
    EndpointFromJSON,
    EndpointFromJSONTyped,
    EndpointToJSON,
} from './Endpoint';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedEndpointList
 */
export interface PaginatedEndpointList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedEndpointList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Endpoint>}
     * @memberof PaginatedEndpointList
     */
    results: Array<Endpoint>;
}

/**
 * Check if a given object implements the PaginatedEndpointList interface.
 */
export function instanceOfPaginatedEndpointList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedEndpointListFromJSON(json: any): PaginatedEndpointList {
    return PaginatedEndpointListFromJSONTyped(json, false);
}

export function PaginatedEndpointListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedEndpointList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(EndpointFromJSON)),
    };
}

export function PaginatedEndpointListToJSON(value?: PaginatedEndpointList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(EndpointToJSON)),
    };
}
