// @goauthentik/api/src/models/PaginatedAuthenticateWebAuthnStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticateWebAuthnStage } from './AuthenticateWebAuthnStage';
import {
    AuthenticateWebAuthnStageFromJSON,
    AuthenticateWebAuthnStageFromJSONTyped,
    AuthenticateWebAuthnStageToJSON,
} from './AuthenticateWebAuthnStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedAuthenticateWebAuthnStageList
 */
export interface PaginatedAuthenticateWebAuthnStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedAuthenticateWebAuthnStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<AuthenticateWebAuthnStage>}
     * @memberof PaginatedAuthenticateWebAuthnStageList
     */
    results: Array<AuthenticateWebAuthnStage>;
}

/**
 * Check if a given object implements the PaginatedAuthenticateWebAuthnStageList interface.
 */
export function instanceOfPaginatedAuthenticateWebAuthnStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedAuthenticateWebAuthnStageListFromJSON(json: any): PaginatedAuthenticateWebAuthnStageList {
    return PaginatedAuthenticateWebAuthnStageListFromJSONTyped(json, false);
}

export function PaginatedAuthenticateWebAuthnStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedAuthenticateWebAuthnStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(AuthenticateWebAuthnStageFromJSON)),
    };
}

export function PaginatedAuthenticateWebAuthnStageListToJSON(value?: PaginatedAuthenticateWebAuthnStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(AuthenticateWebAuthnStageToJSON)),
    };
}
