// @goauthentik/api/src/models/PaginatedBrandList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Brand } from './Brand';
import {
    BrandFromJSON,
    BrandFromJSONTyped,
    BrandToJSON,
} from './Brand';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedBrandList
 */
export interface PaginatedBrandList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedBrandList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Brand>}
     * @memberof PaginatedBrandList
     */
    results: Array<Brand>;
}

/**
 * Check if a given object implements the PaginatedBrandList interface.
 */
export function instanceOfPaginatedBrandList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedBrandListFromJSON(json: any): PaginatedBrandList {
    return PaginatedBrandListFromJSONTyped(json, false);
}

export function PaginatedBrandListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedBrandList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(BrandFromJSON)),
    };
}

export function PaginatedBrandListToJSON(value?: PaginatedBrandList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(BrandToJSON)),
    };
}
