// @goauthentik/api/src/models/PaginatedStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { Stage } from './Stage';
import {
    StageFromJSON,
    StageFromJSONTyped,
    StageToJSON,
} from './Stage';

/**
 * 
 * @export
 * @interface PaginatedStageList
 */
export interface PaginatedStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<Stage>}
     * @memberof PaginatedStageList
     */
    results: Array<Stage>;
}

/**
 * Check if a given object implements the PaginatedStageList interface.
 */
export function instanceOfPaginatedStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedStageListFromJSON(json: any): PaginatedStageList {
    return PaginatedStageListFromJSONTyped(json, false);
}

export function PaginatedStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(StageFromJSON)),
    };
}

export function PaginatedStageListToJSON(value?: PaginatedStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(StageToJSON)),
    };
}
