// @goauthentik/api/src/models/PaginatedPlexSourceList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { PlexSource } from './PlexSource';
import {
    PlexSourceFromJSON,
    PlexSourceFromJSONTyped,
    PlexSourceToJSON,
} from './PlexSource';

/**
 * 
 * @export
 * @interface PaginatedPlexSourceList
 */
export interface PaginatedPlexSourceList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedPlexSourceList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<PlexSource>}
     * @memberof PaginatedPlexSourceList
     */
    results: Array<PlexSource>;
}

/**
 * Check if a given object implements the PaginatedPlexSourceList interface.
 */
export function instanceOfPaginatedPlexSourceList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedPlexSourceListFromJSON(json: any): PaginatedPlexSourceList {
    return PaginatedPlexSourceListFromJSONTyped(json, false);
}

export function PaginatedPlexSourceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedPlexSourceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(PlexSourceFromJSON)),
    };
}

export function PaginatedPlexSourceListToJSON(value?: PaginatedPlexSourceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(PlexSourceToJSON)),
    };
}
