// @goauthentik/api/src/models/PaginatedCaptchaStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { CaptchaStage } from './CaptchaStage';
import {
    CaptchaStageFromJSON,
    CaptchaStageFromJSONTyped,
    CaptchaStageToJSON,
} from './CaptchaStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedCaptchaStageList
 */
export interface PaginatedCaptchaStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedCaptchaStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<CaptchaStage>}
     * @memberof PaginatedCaptchaStageList
     */
    results: Array<CaptchaStage>;
}

/**
 * Check if a given object implements the PaginatedCaptchaStageList interface.
 */
export function instanceOfPaginatedCaptchaStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedCaptchaStageListFromJSON(json: any): PaginatedCaptchaStageList {
    return PaginatedCaptchaStageListFromJSONTyped(json, false);
}

export function PaginatedCaptchaStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedCaptchaStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(CaptchaStageFromJSON)),
    };
}

export function PaginatedCaptchaStageListToJSON(value?: PaginatedCaptchaStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(CaptchaStageToJSON)),
    };
}
