// @goauthentik/api/src/models/PaginatedRadiusProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { RadiusProvider } from './RadiusProvider';
import {
    RadiusProviderFromJSON,
    RadiusProviderFromJSONTyped,
    RadiusProviderToJSON,
} from './RadiusProvider';

/**
 * 
 * @export
 * @interface PaginatedRadiusProviderList
 */
export interface PaginatedRadiusProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedRadiusProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<RadiusProvider>}
     * @memberof PaginatedRadiusProviderList
     */
    results: Array<RadiusProvider>;
}

/**
 * Check if a given object implements the PaginatedRadiusProviderList interface.
 */
export function instanceOfPaginatedRadiusProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedRadiusProviderListFromJSON(json: any): PaginatedRadiusProviderList {
    return PaginatedRadiusProviderListFromJSONTyped(json, false);
}

export function PaginatedRadiusProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedRadiusProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(RadiusProviderFromJSON)),
    };
}

export function PaginatedRadiusProviderListToJSON(value?: PaginatedRadiusProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(RadiusProviderToJSON)),
    };
}
