// @goauthentik/api/src/models/PaginatedSAMLProviderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { SAMLProvider } from './SAMLProvider';
import {
    SAMLProviderFromJSON,
    SAMLProviderFromJSONTyped,
    SAMLProviderToJSON,
} from './SAMLProvider';

/**
 * 
 * @export
 * @interface PaginatedSAMLProviderList
 */
export interface PaginatedSAMLProviderList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedSAMLProviderList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<SAMLProvider>}
     * @memberof PaginatedSAMLProviderList
     */
    results: Array<SAMLProvider>;
}

/**
 * Check if a given object implements the PaginatedSAMLProviderList interface.
 */
export function instanceOfPaginatedSAMLProviderList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedSAMLProviderListFromJSON(json: any): PaginatedSAMLProviderList {
    return PaginatedSAMLProviderListFromJSONTyped(json, false);
}

export function PaginatedSAMLProviderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedSAMLProviderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(SAMLProviderFromJSON)),
    };
}

export function PaginatedSAMLProviderListToJSON(value?: PaginatedSAMLProviderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(SAMLProviderToJSON)),
    };
}
