// @goauthentik/api/src/models/PaginatedUserWriteStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';
import type { UserWriteStage } from './UserWriteStage';
import {
    UserWriteStageFromJSON,
    UserWriteStageFromJSONTyped,
    UserWriteStageToJSON,
} from './UserWriteStage';

/**
 * 
 * @export
 * @interface PaginatedUserWriteStageList
 */
export interface PaginatedUserWriteStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedUserWriteStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<UserWriteStage>}
     * @memberof PaginatedUserWriteStageList
     */
    results: Array<UserWriteStage>;
}

/**
 * Check if a given object implements the PaginatedUserWriteStageList interface.
 */
export function instanceOfPaginatedUserWriteStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedUserWriteStageListFromJSON(json: any): PaginatedUserWriteStageList {
    return PaginatedUserWriteStageListFromJSONTyped(json, false);
}

export function PaginatedUserWriteStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedUserWriteStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(UserWriteStageFromJSON)),
    };
}

export function PaginatedUserWriteStageListToJSON(value?: PaginatedUserWriteStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(UserWriteStageToJSON)),
    };
}
