// @goauthentik/api/src/models/PaginatedIdentificationStageList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { IdentificationStage } from './IdentificationStage';
import {
    IdentificationStageFromJSON,
    IdentificationStageFromJSONTyped,
    IdentificationStageToJSON,
} from './IdentificationStage';
import type { Pagination } from './Pagination';
import {
    PaginationFromJSON,
    PaginationFromJSONTyped,
    PaginationToJSON,
} from './Pagination';

/**
 * 
 * @export
 * @interface PaginatedIdentificationStageList
 */
export interface PaginatedIdentificationStageList {
    /**
     * 
     * @type {Pagination}
     * @memberof PaginatedIdentificationStageList
     */
    pagination: Pagination;
    /**
     * 
     * @type {Array<IdentificationStage>}
     * @memberof PaginatedIdentificationStageList
     */
    results: Array<IdentificationStage>;
}

/**
 * Check if a given object implements the PaginatedIdentificationStageList interface.
 */
export function instanceOfPaginatedIdentificationStageList(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;

    return isInstance;
}

export function PaginatedIdentificationStageListFromJSON(json: any): PaginatedIdentificationStageList {
    return PaginatedIdentificationStageListFromJSONTyped(json, false);
}

export function PaginatedIdentificationStageListFromJSONTyped(json: any, ignoreDiscriminator: boolean): PaginatedIdentificationStageList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pagination': PaginationFromJSON(json['pagination']),
        'results': ((json['results'] as Array<any>).map(IdentificationStageFromJSON)),
    };
}

export function PaginatedIdentificationStageListToJSON(value?: PaginatedIdentificationStageList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pagination': PaginationToJSON(value.pagination),
        'results': ((value.results as Array<any>).map(IdentificationStageToJSON)),
    };
}
