// @goauthentik/api/src/models/PasswordChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Password challenge response
 * @export
 * @interface PasswordChallengeResponseRequest
 */
export interface PasswordChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof PasswordChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {string}
     * @memberof PasswordChallengeResponseRequest
     */
    password: string;
}

/**
 * Check if a given object implements the PasswordChallengeResponseRequest interface.
 */
export function instanceOfPasswordChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "password" in value;

    return isInstance;
}

export function PasswordChallengeResponseRequestFromJSON(json: any): PasswordChallengeResponseRequest {
    return PasswordChallengeResponseRequestFromJSONTyped(json, false);
}

export function PasswordChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PasswordChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'password': json['password'],
    };
}

export function PasswordChallengeResponseRequestToJSON(value?: PasswordChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'password': value.password,
    };
}
