// @goauthentik/api/src/models/PasswordStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { BackendsEnum } from './BackendsEnum';
import {
    BackendsEnumFromJSON,
    BackendsEnumFromJSONTyped,
    BackendsEnumToJSON,
} from './BackendsEnum';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * PasswordStage Serializer
 * @export
 * @interface PasswordStage
 */
export interface PasswordStage {
    /**
     * 
     * @type {string}
     * @memberof PasswordStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof PasswordStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof PasswordStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof PasswordStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof PasswordStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof PasswordStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof PasswordStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Selection of backends to test the password against.
     * @type {Array<BackendsEnum>}
     * @memberof PasswordStage
     */
    backends: Array<BackendsEnum>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof PasswordStage
     */
    configureFlow?: string | null;
    /**
     * How many attempts a user has before the flow is canceled. To lock the user out, use a reputation policy and a user_write stage.
     * @type {number}
     * @memberof PasswordStage
     */
    failedAttemptsBeforeCancel?: number;
}

/**
 * Check if a given object implements the PasswordStage interface.
 */
export function instanceOfPasswordStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "backends" in value;

    return isInstance;
}

export function PasswordStageFromJSON(json: any): PasswordStage {
    return PasswordStageFromJSONTyped(json, false);
}

export function PasswordStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): PasswordStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'backends': ((json['backends'] as Array<any>).map(BackendsEnumFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'failedAttemptsBeforeCancel': !exists(json, 'failed_attempts_before_cancel') ? undefined : json['failed_attempts_before_cancel'],
    };
}

export function PasswordStageToJSON(value?: PasswordStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'backends': ((value.backends as Array<any>).map(BackendsEnumToJSON)),
        'configure_flow': value.configureFlow,
        'failed_attempts_before_cancel': value.failedAttemptsBeforeCancel,
    };
}
