// @goauthentik/api/src/models/PasswordPolicy.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Password Policy Serializer
 * @export
 * @interface PasswordPolicy
 */
export interface PasswordPolicy {
    /**
     * 
     * @type {string}
     * @memberof PasswordPolicy
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof PasswordPolicy
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof PasswordPolicy
     */
    executionLogging?: boolean;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof PasswordPolicy
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof PasswordPolicy
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof PasswordPolicy
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof PasswordPolicy
     */
    readonly metaModelName: string;
    /**
     * Return objects policy is bound to
     * @type {number}
     * @memberof PasswordPolicy
     */
    readonly boundTo: number;
    /**
     * Field key to check, field keys defined in Prompt stages are available.
     * @type {string}
     * @memberof PasswordPolicy
     */
    passwordField?: string;
    /**
     * 
     * @type {number}
     * @memberof PasswordPolicy
     */
    amountDigits?: number;
    /**
     * 
     * @type {number}
     * @memberof PasswordPolicy
     */
    amountUppercase?: number;
    /**
     * 
     * @type {number}
     * @memberof PasswordPolicy
     */
    amountLowercase?: number;
    /**
     * 
     * @type {number}
     * @memberof PasswordPolicy
     */
    amountSymbols?: number;
    /**
     * 
     * @type {number}
     * @memberof PasswordPolicy
     */
    lengthMin?: number;
    /**
     * 
     * @type {string}
     * @memberof PasswordPolicy
     */
    symbolCharset?: string;
    /**
     * 
     * @type {string}
     * @memberof PasswordPolicy
     */
    errorMessage?: string;
    /**
     * 
     * @type {boolean}
     * @memberof PasswordPolicy
     */
    checkStaticRules?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PasswordPolicy
     */
    checkHaveIBeenPwned?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PasswordPolicy
     */
    checkZxcvbn?: boolean;
    /**
     * How many times the password hash is allowed to be on haveibeenpwned
     * @type {number}
     * @memberof PasswordPolicy
     */
    hibpAllowedCount?: number;
    /**
     * If the zxcvbn score is equal or less than this value, the policy will fail.
     * @type {number}
     * @memberof PasswordPolicy
     */
    zxcvbnScoreThreshold?: number;
}

/**
 * Check if a given object implements the PasswordPolicy interface.
 */
export function instanceOfPasswordPolicy(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;

    return isInstance;
}

export function PasswordPolicyFromJSON(json: any): PasswordPolicy {
    return PasswordPolicyFromJSONTyped(json, false);
}

export function PasswordPolicyFromJSONTyped(json: any, ignoreDiscriminator: boolean): PasswordPolicy {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'passwordField': !exists(json, 'password_field') ? undefined : json['password_field'],
        'amountDigits': !exists(json, 'amount_digits') ? undefined : json['amount_digits'],
        'amountUppercase': !exists(json, 'amount_uppercase') ? undefined : json['amount_uppercase'],
        'amountLowercase': !exists(json, 'amount_lowercase') ? undefined : json['amount_lowercase'],
        'amountSymbols': !exists(json, 'amount_symbols') ? undefined : json['amount_symbols'],
        'lengthMin': !exists(json, 'length_min') ? undefined : json['length_min'],
        'symbolCharset': !exists(json, 'symbol_charset') ? undefined : json['symbol_charset'],
        'errorMessage': !exists(json, 'error_message') ? undefined : json['error_message'],
        'checkStaticRules': !exists(json, 'check_static_rules') ? undefined : json['check_static_rules'],
        'checkHaveIBeenPwned': !exists(json, 'check_have_i_been_pwned') ? undefined : json['check_have_i_been_pwned'],
        'checkZxcvbn': !exists(json, 'check_zxcvbn') ? undefined : json['check_zxcvbn'],
        'hibpAllowedCount': !exists(json, 'hibp_allowed_count') ? undefined : json['hibp_allowed_count'],
        'zxcvbnScoreThreshold': !exists(json, 'zxcvbn_score_threshold') ? undefined : json['zxcvbn_score_threshold'],
    };
}

export function PasswordPolicyToJSON(value?: PasswordPolicy | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'password_field': value.passwordField,
        'amount_digits': value.amountDigits,
        'amount_uppercase': value.amountUppercase,
        'amount_lowercase': value.amountLowercase,
        'amount_symbols': value.amountSymbols,
        'length_min': value.lengthMin,
        'symbol_charset': value.symbolCharset,
        'error_message': value.errorMessage,
        'check_static_rules': value.checkStaticRules,
        'check_have_i_been_pwned': value.checkHaveIBeenPwned,
        'check_zxcvbn': value.checkZxcvbn,
        'hibp_allowed_count': value.hibpAllowedCount,
        'zxcvbn_score_threshold': value.zxcvbnScoreThreshold,
    };
}
