// @goauthentik/api/src/models/PasswordExpiryPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Password Expiry Policy Serializer
 * @export
 * @interface PasswordExpiryPolicyRequest
 */
export interface PasswordExpiryPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof PasswordExpiryPolicyRequest
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof PasswordExpiryPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {number}
     * @memberof PasswordExpiryPolicyRequest
     */
    days: number;
    /**
     * 
     * @type {boolean}
     * @memberof PasswordExpiryPolicyRequest
     */
    denyOnly?: boolean;
}

/**
 * Check if a given object implements the PasswordExpiryPolicyRequest interface.
 */
export function instanceOfPasswordExpiryPolicyRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "days" in value;

    return isInstance;
}

export function PasswordExpiryPolicyRequestFromJSON(json: any): PasswordExpiryPolicyRequest {
    return PasswordExpiryPolicyRequestFromJSONTyped(json, false);
}

export function PasswordExpiryPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PasswordExpiryPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'days': json['days'],
        'denyOnly': !exists(json, 'deny_only') ? undefined : json['deny_only'],
    };
}

export function PasswordExpiryPolicyRequestToJSON(value?: PasswordExpiryPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
