// @goauthentik/api/src/models/PatchedAuthenticateWebAuthnStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthenticatorAttachmentEnum } from './AuthenticatorAttachmentEnum';
import {
    AuthenticatorAttachmentEnumFromJSON,
    AuthenticatorAttachmentEnumFromJSONTyped,
    AuthenticatorAttachmentEnumToJSON,
} from './AuthenticatorAttachmentEnum';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';
import type { ResidentKeyRequirementEnum } from './ResidentKeyRequirementEnum';
import {
    ResidentKeyRequirementEnumFromJSON,
    ResidentKeyRequirementEnumFromJSONTyped,
    ResidentKeyRequirementEnumToJSON,
} from './ResidentKeyRequirementEnum';
import type { UserVerificationEnum } from './UserVerificationEnum';
import {
    UserVerificationEnumFromJSON,
    UserVerificationEnumFromJSONTyped,
    UserVerificationEnumToJSON,
} from './UserVerificationEnum';

/**
 * AuthenticateWebAuthnStage Serializer
 * @export
 * @interface PatchedAuthenticateWebAuthnStageRequest
 */
export interface PatchedAuthenticateWebAuthnStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {UserVerificationEnum}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    userVerification?: UserVerificationEnum;
    /**
     * 
     * @type {AuthenticatorAttachmentEnum}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    authenticatorAttachment?: AuthenticatorAttachmentEnum | null;
    /**
     * 
     * @type {ResidentKeyRequirementEnum}
     * @memberof PatchedAuthenticateWebAuthnStageRequest
     */
    residentKeyRequirement?: ResidentKeyRequirementEnum;
}

/**
 * Check if a given object implements the PatchedAuthenticateWebAuthnStageRequest interface.
 */
export function instanceOfPatchedAuthenticateWebAuthnStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedAuthenticateWebAuthnStageRequestFromJSON(json: any): PatchedAuthenticateWebAuthnStageRequest {
    return PatchedAuthenticateWebAuthnStageRequestFromJSONTyped(json, false);
}

export function PatchedAuthenticateWebAuthnStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedAuthenticateWebAuthnStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'userVerification': !exists(json, 'user_verification') ? undefined : UserVerificationEnumFromJSON(json['user_verification']),
        'authenticatorAttachment': !exists(json, 'authenticator_attachment') ? undefined : AuthenticatorAttachmentEnumFromJSON(json['authenticator_attachment']),
        'residentKeyRequirement': !exists(json, 'resident_key_requirement') ? undefined : ResidentKeyRequirementEnumFromJSON(json['resident_key_requirement']),
    };
}

export function PatchedAuthenticateWebAuthnStageRequestToJSON(value?: PatchedAuthenticateWebAuthnStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'user_verification': UserVerificationEnumToJSON(value.userVerification),
        'authenticator_attachment': AuthenticatorAttachmentEnumToJSON(value.authenticatorAttachment),
        'resident_key_requirement': ResidentKeyRequirementEnumToJSON(value.residentKeyRequirement),
    };
}
