// @goauthentik/api/src/models/PatchedBrandRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Brand Serializer
 * @export
 * @interface PatchedBrandRequest
 */
export interface PatchedBrandRequest {
    /**
     * Domain that activates this brand. Can be a superset, i.e. `a.b` for `aa.b` and `ba.b`
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    domain?: string;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedBrandRequest
     */
    _default?: boolean;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    brandingTitle?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    brandingLogo?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    brandingFavicon?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowAuthentication?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowInvalidation?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowRecovery?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowUnenrollment?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowUserSettings?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    flowDeviceCode?: string | null;
    /**
     * Web Certificate used by the authentik Core webserver.
     * @type {string}
     * @memberof PatchedBrandRequest
     */
    webCertificate?: string | null;
    /**
     * 
     * @type {any}
     * @memberof PatchedBrandRequest
     */
    attributes?: any | null;
}

/**
 * Check if a given object implements the PatchedBrandRequest interface.
 */
export function instanceOfPatchedBrandRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedBrandRequestFromJSON(json: any): PatchedBrandRequest {
    return PatchedBrandRequestFromJSONTyped(json, false);
}

export function PatchedBrandRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedBrandRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'domain': !exists(json, 'domain') ? undefined : json['domain'],
        '_default': !exists(json, 'default') ? undefined : json['default'],
        'brandingTitle': !exists(json, 'branding_title') ? undefined : json['branding_title'],
        'brandingLogo': !exists(json, 'branding_logo') ? undefined : json['branding_logo'],
        'brandingFavicon': !exists(json, 'branding_favicon') ? undefined : json['branding_favicon'],
        'flowAuthentication': !exists(json, 'flow_authentication') ? undefined : json['flow_authentication'],
        'flowInvalidation': !exists(json, 'flow_invalidation') ? undefined : json['flow_invalidation'],
        'flowRecovery': !exists(json, 'flow_recovery') ? undefined : json['flow_recovery'],
        'flowUnenrollment': !exists(json, 'flow_unenrollment') ? undefined : json['flow_unenrollment'],
        'flowUserSettings': !exists(json, 'flow_user_settings') ? undefined : json['flow_user_settings'],
        'flowDeviceCode': !exists(json, 'flow_device_code') ? undefined : json['flow_device_code'],
        'webCertificate': !exists(json, 'web_certificate') ? undefined : json['web_certificate'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
    };
}

export function PatchedBrandRequestToJSON(value?: PatchedBrandRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'domain': value.domain,
        'default': value._default,
        'branding_title': value.brandingTitle,
        'branding_logo': value.brandingLogo,
        'branding_favicon': value.brandingFavicon,
        'flow_authentication': value.flowAuthentication,
        'flow_invalidation': value.flowInvalidation,
        'flow_recovery': value.flowRecovery,
        'flow_unenrollment': value.flowUnenrollment,
        'flow_user_settings': value.flowUserSettings,
        'flow_device_code': value.flowDeviceCode,
        'web_certificate': value.webCertificate,
        'attributes': value.attributes,
    };
}
