// @goauthentik/api/src/models/PatchedConsentStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ConsentStageModeEnum } from './ConsentStageModeEnum';
import {
    ConsentStageModeEnumFromJSON,
    ConsentStageModeEnumFromJSONTyped,
    ConsentStageModeEnumToJSON,
} from './ConsentStageModeEnum';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * ConsentStage Serializer
 * @export
 * @interface PatchedConsentStageRequest
 */
export interface PatchedConsentStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedConsentStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedConsentStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {ConsentStageModeEnum}
     * @memberof PatchedConsentStageRequest
     */
    mode?: ConsentStageModeEnum;
    /**
     * Offset after which consent expires. (Format: hours=1;minutes=2;seconds=3).
     * @type {string}
     * @memberof PatchedConsentStageRequest
     */
    consentExpireIn?: string;
}

/**
 * Check if a given object implements the PatchedConsentStageRequest interface.
 */
export function instanceOfPatchedConsentStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedConsentStageRequestFromJSON(json: any): PatchedConsentStageRequest {
    return PatchedConsentStageRequestFromJSONTyped(json, false);
}

export function PatchedConsentStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedConsentStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'mode': !exists(json, 'mode') ? undefined : ConsentStageModeEnumFromJSON(json['mode']),
        'consentExpireIn': !exists(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}

export function PatchedConsentStageRequestToJSON(value?: PatchedConsentStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'mode': ConsentStageModeEnumToJSON(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
