// @goauthentik/api/src/models/PatchedRoleRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Role serializer
 * @export
 * @interface PatchedRoleRequest
 */
export interface PatchedRoleRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedRoleRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedRoleRequest interface.
 */
export function instanceOfPatchedRoleRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedRoleRequestFromJSON(json: any): PatchedRoleRequest {
    return PatchedRoleRequestFromJSONTyped(json, false);
}

export function PatchedRoleRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedRoleRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedRoleRequestToJSON(value?: PatchedRoleRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
