// @goauthentik/api/src/models/PatchedAuthenticatorSMSStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { AuthTypeEnum } from './AuthTypeEnum';
import {
    AuthTypeEnumFromJSON,
    AuthTypeEnumFromJSONTyped,
    AuthTypeEnumToJSON,
} from './AuthTypeEnum';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';
import type { ProviderEnum } from './ProviderEnum';
import {
    ProviderEnumFromJSON,
    ProviderEnumFromJSONTyped,
    ProviderEnumToJSON,
} from './ProviderEnum';

/**
 * AuthenticatorSMSStage Serializer
 * @export
 * @interface PatchedAuthenticatorSMSStageRequest
 */
export interface PatchedAuthenticatorSMSStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {ProviderEnum}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    provider?: ProviderEnum;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    fromNumber?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    accountSid?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    auth?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    authPassword?: string;
    /**
     * 
     * @type {AuthTypeEnum}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    authType?: AuthTypeEnum;
    /**
     * When enabled, the Phone number is only used during enrollment to verify the users authenticity. Only a hash of the phone number is saved to ensure it is not reused in the future.
     * @type {boolean}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    verifyOnly?: boolean;
    /**
     * Optionally modify the payload being sent to custom providers.
     * @type {string}
     * @memberof PatchedAuthenticatorSMSStageRequest
     */
    mapping?: string | null;
}

/**
 * Check if a given object implements the PatchedAuthenticatorSMSStageRequest interface.
 */
export function instanceOfPatchedAuthenticatorSMSStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedAuthenticatorSMSStageRequestFromJSON(json: any): PatchedAuthenticatorSMSStageRequest {
    return PatchedAuthenticatorSMSStageRequestFromJSONTyped(json, false);
}

export function PatchedAuthenticatorSMSStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedAuthenticatorSMSStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'provider': !exists(json, 'provider') ? undefined : ProviderEnumFromJSON(json['provider']),
        'fromNumber': !exists(json, 'from_number') ? undefined : json['from_number'],
        'accountSid': !exists(json, 'account_sid') ? undefined : json['account_sid'],
        'auth': !exists(json, 'auth') ? undefined : json['auth'],
        'authPassword': !exists(json, 'auth_password') ? undefined : json['auth_password'],
        'authType': !exists(json, 'auth_type') ? undefined : AuthTypeEnumFromJSON(json['auth_type']),
        'verifyOnly': !exists(json, 'verify_only') ? undefined : json['verify_only'],
        'mapping': !exists(json, 'mapping') ? undefined : json['mapping'],
    };
}

export function PatchedAuthenticatorSMSStageRequestToJSON(value?: PatchedAuthenticatorSMSStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'provider': ProviderEnumToJSON(value.provider),
        'from_number': value.fromNumber,
        'account_sid': value.accountSid,
        'auth': value.auth,
        'auth_password': value.authPassword,
        'auth_type': AuthTypeEnumToJSON(value.authType),
        'verify_only': value.verifyOnly,
        'mapping': value.mapping,
    };
}
