// @goauthentik/api/src/models/PatchedAuthenticatorDuoStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * AuthenticatorDuoStage Serializer
 * @export
 * @interface PatchedAuthenticatorDuoStageRequest
 */
export interface PatchedAuthenticatorDuoStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    clientId?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    clientSecret?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    apiHostname?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    adminIntegrationKey?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorDuoStageRequest
     */
    adminSecretKey?: string;
}

/**
 * Check if a given object implements the PatchedAuthenticatorDuoStageRequest interface.
 */
export function instanceOfPatchedAuthenticatorDuoStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedAuthenticatorDuoStageRequestFromJSON(json: any): PatchedAuthenticatorDuoStageRequest {
    return PatchedAuthenticatorDuoStageRequestFromJSONTyped(json, false);
}

export function PatchedAuthenticatorDuoStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedAuthenticatorDuoStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'clientId': !exists(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !exists(json, 'client_secret') ? undefined : json['client_secret'],
        'apiHostname': !exists(json, 'api_hostname') ? undefined : json['api_hostname'],
        'adminIntegrationKey': !exists(json, 'admin_integration_key') ? undefined : json['admin_integration_key'],
        'adminSecretKey': !exists(json, 'admin_secret_key') ? undefined : json['admin_secret_key'],
    };
}

export function PatchedAuthenticatorDuoStageRequestToJSON(value?: PatchedAuthenticatorDuoStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'api_hostname': value.apiHostname,
        'admin_integration_key': value.adminIntegrationKey,
        'admin_secret_key': value.adminSecretKey,
    };
}
