// @goauthentik/api/src/models/PatchedPermissionAssignRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ModelEnum } from './ModelEnum';
import {
    ModelEnumFromJSON,
    ModelEnumFromJSONTyped,
    ModelEnumToJSON,
} from './ModelEnum';

/**
 * Request to assign a new permission
 * @export
 * @interface PatchedPermissionAssignRequest
 */
export interface PatchedPermissionAssignRequest {
    /**
     * 
     * @type {Array<string>}
     * @memberof PatchedPermissionAssignRequest
     */
    permissions?: Array<string>;
    /**
     * 
     * @type {ModelEnum}
     * @memberof PatchedPermissionAssignRequest
     */
    model?: ModelEnum;
    /**
     * 
     * @type {string}
     * @memberof PatchedPermissionAssignRequest
     */
    objectPk?: string;
}

/**
 * Check if a given object implements the PatchedPermissionAssignRequest interface.
 */
export function instanceOfPatchedPermissionAssignRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedPermissionAssignRequestFromJSON(json: any): PatchedPermissionAssignRequest {
    return PatchedPermissionAssignRequestFromJSONTyped(json, false);
}

export function PatchedPermissionAssignRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedPermissionAssignRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'permissions': !exists(json, 'permissions') ? undefined : json['permissions'],
        'model': !exists(json, 'model') ? undefined : ModelEnumFromJSON(json['model']),
        'objectPk': !exists(json, 'object_pk') ? undefined : json['object_pk'],
    };
}

export function PatchedPermissionAssignRequestToJSON(value?: PatchedPermissionAssignRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'permissions': value.permissions,
        'model': ModelEnumToJSON(value.model),
        'object_pk': value.objectPk,
    };
}
