// @goauthentik/api/src/models/PatchedNotificationWebhookMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * NotificationWebhookMapping Serializer
 * @export
 * @interface PatchedNotificationWebhookMappingRequest
 */
export interface PatchedNotificationWebhookMappingRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedNotificationWebhookMappingRequest
     */
    name?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedNotificationWebhookMappingRequest
     */
    expression?: string;
}

/**
 * Check if a given object implements the PatchedNotificationWebhookMappingRequest interface.
 */
export function instanceOfPatchedNotificationWebhookMappingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedNotificationWebhookMappingRequestFromJSON(json: any): PatchedNotificationWebhookMappingRequest {
    return PatchedNotificationWebhookMappingRequestFromJSONTyped(json, false);
}

export function PatchedNotificationWebhookMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedNotificationWebhookMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
    };
}

export function PatchedNotificationWebhookMappingRequestToJSON(value?: PatchedNotificationWebhookMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'expression': value.expression,
    };
}
