// @goauthentik/api/src/models/PatchedSMSDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for sms authenticator devices
 * @export
 * @interface PatchedSMSDeviceRequest
 */
export interface PatchedSMSDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof PatchedSMSDeviceRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedSMSDeviceRequest interface.
 */
export function instanceOfPatchedSMSDeviceRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedSMSDeviceRequestFromJSON(json: any): PatchedSMSDeviceRequest {
    return PatchedSMSDeviceRequestFromJSONTyped(json, false);
}

export function PatchedSMSDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedSMSDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedSMSDeviceRequestToJSON(value?: PatchedSMSDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
