// @goauthentik/api/src/models/PatchedUserDeleteStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * UserDeleteStage Serializer
 * @export
 * @interface PatchedUserDeleteStageRequest
 */
export interface PatchedUserDeleteStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedUserDeleteStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedUserDeleteStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
}

/**
 * Check if a given object implements the PatchedUserDeleteStageRequest interface.
 */
export function instanceOfPatchedUserDeleteStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedUserDeleteStageRequestFromJSON(json: any): PatchedUserDeleteStageRequest {
    return PatchedUserDeleteStageRequestFromJSONTyped(json, false);
}

export function PatchedUserDeleteStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedUserDeleteStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
    };
}

export function PatchedUserDeleteStageRequestToJSON(value?: PatchedUserDeleteStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
    };
}
