// @goauthentik/api/src/models/PatchedFlowStageBindingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { InvalidResponseActionEnum } from './InvalidResponseActionEnum';
import {
    InvalidResponseActionEnumFromJSON,
    InvalidResponseActionEnumFromJSONTyped,
    InvalidResponseActionEnumToJSON,
} from './InvalidResponseActionEnum';
import type { PolicyEngineMode } from './PolicyEngineMode';
import {
    PolicyEngineModeFromJSON,
    PolicyEngineModeFromJSONTyped,
    PolicyEngineModeToJSON,
} from './PolicyEngineMode';

/**
 * FlowStageBinding Serializer
 * @export
 * @interface PatchedFlowStageBindingRequest
 */
export interface PatchedFlowStageBindingRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedFlowStageBindingRequest
     */
    target?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedFlowStageBindingRequest
     */
    stage?: string;
    /**
     * Evaluate policies during the Flow planning process.
     * @type {boolean}
     * @memberof PatchedFlowStageBindingRequest
     */
    evaluateOnPlan?: boolean;
    /**
     * Evaluate policies when the Stage is present to the user.
     * @type {boolean}
     * @memberof PatchedFlowStageBindingRequest
     */
    reEvaluatePolicies?: boolean;
    /**
     * 
     * @type {number}
     * @memberof PatchedFlowStageBindingRequest
     */
    order?: number;
    /**
     * 
     * @type {PolicyEngineMode}
     * @memberof PatchedFlowStageBindingRequest
     */
    policyEngineMode?: PolicyEngineMode;
    /**
     * 
     * @type {InvalidResponseActionEnum}
     * @memberof PatchedFlowStageBindingRequest
     */
    invalidResponseAction?: InvalidResponseActionEnum;
}

/**
 * Check if a given object implements the PatchedFlowStageBindingRequest interface.
 */
export function instanceOfPatchedFlowStageBindingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedFlowStageBindingRequestFromJSON(json: any): PatchedFlowStageBindingRequest {
    return PatchedFlowStageBindingRequestFromJSONTyped(json, false);
}

export function PatchedFlowStageBindingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedFlowStageBindingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'target': !exists(json, 'target') ? undefined : json['target'],
        'stage': !exists(json, 'stage') ? undefined : json['stage'],
        'evaluateOnPlan': !exists(json, 'evaluate_on_plan') ? undefined : json['evaluate_on_plan'],
        'reEvaluatePolicies': !exists(json, 're_evaluate_policies') ? undefined : json['re_evaluate_policies'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'invalidResponseAction': !exists(json, 'invalid_response_action') ? undefined : InvalidResponseActionEnumFromJSON(json['invalid_response_action']),
    };
}

export function PatchedFlowStageBindingRequestToJSON(value?: PatchedFlowStageBindingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'target': value.target,
        'stage': value.stage,
        'evaluate_on_plan': value.evaluateOnPlan,
        're_evaluate_policies': value.reEvaluatePolicies,
        'order': value.order,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'invalid_response_action': InvalidResponseActionEnumToJSON(value.invalidResponseAction),
    };
}
