// @goauthentik/api/src/models/PatchedLDAPPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * LDAP PropertyMapping Serializer
 * @export
 * @interface PatchedLDAPPropertyMappingRequest
 */
export interface PatchedLDAPPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof PatchedLDAPPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPPropertyMappingRequest
     */
    name?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPPropertyMappingRequest
     */
    expression?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPPropertyMappingRequest
     */
    objectField?: string;
}

/**
 * Check if a given object implements the PatchedLDAPPropertyMappingRequest interface.
 */
export function instanceOfPatchedLDAPPropertyMappingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedLDAPPropertyMappingRequestFromJSON(json: any): PatchedLDAPPropertyMappingRequest {
    return PatchedLDAPPropertyMappingRequestFromJSONTyped(json, false);
}

export function PatchedLDAPPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedLDAPPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'objectField': !exists(json, 'object_field') ? undefined : json['object_field'],
    };
}

export function PatchedLDAPPropertyMappingRequestToJSON(value?: PatchedLDAPPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
