// @goauthentik/api/src/models/PatchedDenyStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * DenyStage Serializer
 * @export
 * @interface PatchedDenyStageRequest
 */
export interface PatchedDenyStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedDenyStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedDenyStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {string}
     * @memberof PatchedDenyStageRequest
     */
    denyMessage?: string;
}

/**
 * Check if a given object implements the PatchedDenyStageRequest interface.
 */
export function instanceOfPatchedDenyStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedDenyStageRequestFromJSON(json: any): PatchedDenyStageRequest {
    return PatchedDenyStageRequestFromJSONTyped(json, false);
}

export function PatchedDenyStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedDenyStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'denyMessage': !exists(json, 'deny_message') ? undefined : json['deny_message'],
    };
}

export function PatchedDenyStageRequestToJSON(value?: PatchedDenyStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'deny_message': value.denyMessage,
    };
}
