// @goauthentik/api/src/models/PatchedDuoDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for Duo authenticator devices
 * @export
 * @interface PatchedDuoDeviceRequest
 */
export interface PatchedDuoDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof PatchedDuoDeviceRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedDuoDeviceRequest interface.
 */
export function instanceOfPatchedDuoDeviceRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedDuoDeviceRequestFromJSON(json: any): PatchedDuoDeviceRequest {
    return PatchedDuoDeviceRequestFromJSONTyped(json, false);
}

export function PatchedDuoDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedDuoDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedDuoDeviceRequestToJSON(value?: PatchedDuoDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
