// @goauthentik/api/src/models/PatchedDummyStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * DummyStage Serializer
 * @export
 * @interface PatchedDummyStageRequest
 */
export interface PatchedDummyStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedDummyStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedDummyStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedDummyStageRequest
     */
    throwError?: boolean;
}

/**
 * Check if a given object implements the PatchedDummyStageRequest interface.
 */
export function instanceOfPatchedDummyStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedDummyStageRequestFromJSON(json: any): PatchedDummyStageRequest {
    return PatchedDummyStageRequestFromJSONTyped(json, false);
}

export function PatchedDummyStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedDummyStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'throwError': !exists(json, 'throw_error') ? undefined : json['throw_error'],
    };
}

export function PatchedDummyStageRequestToJSON(value?: PatchedDummyStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'throw_error': value.throwError,
    };
}
