// @goauthentik/api/src/models/PatchedSCIMMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SCIMMapping Serializer
 * @export
 * @interface PatchedSCIMMappingRequest
 */
export interface PatchedSCIMMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof PatchedSCIMMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedSCIMMappingRequest
     */
    name?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedSCIMMappingRequest
     */
    expression?: string;
}

/**
 * Check if a given object implements the PatchedSCIMMappingRequest interface.
 */
export function instanceOfPatchedSCIMMappingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedSCIMMappingRequestFromJSON(json: any): PatchedSCIMMappingRequest {
    return PatchedSCIMMappingRequestFromJSONTyped(json, false);
}

export function PatchedSCIMMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedSCIMMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
    };
}

export function PatchedSCIMMappingRequestToJSON(value?: PatchedSCIMMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
    };
}
