// @goauthentik/api/src/models/PatchedPasswordExpiryPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Password Expiry Policy Serializer
 * @export
 * @interface PatchedPasswordExpiryPolicyRequest
 */
export interface PatchedPasswordExpiryPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedPasswordExpiryPolicyRequest
     */
    name?: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof PatchedPasswordExpiryPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordExpiryPolicyRequest
     */
    days?: number;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedPasswordExpiryPolicyRequest
     */
    denyOnly?: boolean;
}

/**
 * Check if a given object implements the PatchedPasswordExpiryPolicyRequest interface.
 */
export function instanceOfPatchedPasswordExpiryPolicyRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedPasswordExpiryPolicyRequestFromJSON(json: any): PatchedPasswordExpiryPolicyRequest {
    return PatchedPasswordExpiryPolicyRequestFromJSONTyped(json, false);
}

export function PatchedPasswordExpiryPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedPasswordExpiryPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'days': !exists(json, 'days') ? undefined : json['days'],
        'denyOnly': !exists(json, 'deny_only') ? undefined : json['deny_only'],
    };
}

export function PatchedPasswordExpiryPolicyRequestToJSON(value?: PatchedPasswordExpiryPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
