// @goauthentik/api/src/models/PatchedLicenseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * License Serializer
 * @export
 * @interface PatchedLicenseRequest
 */
export interface PatchedLicenseRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedLicenseRequest
     */
    key?: string;
}

/**
 * Check if a given object implements the PatchedLicenseRequest interface.
 */
export function instanceOfPatchedLicenseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedLicenseRequestFromJSON(json: any): PatchedLicenseRequest {
    return PatchedLicenseRequestFromJSONTyped(json, false);
}

export function PatchedLicenseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedLicenseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'key': !exists(json, 'key') ? undefined : json['key'],
    };
}

export function PatchedLicenseRequestToJSON(value?: PatchedLicenseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'key': value.key,
    };
}
