// @goauthentik/api/src/models/PatchedLDAPProviderRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { LDAPAPIAccessMode } from './LDAPAPIAccessMode';
import {
    LDAPAPIAccessModeFromJSON,
    LDAPAPIAccessModeFromJSONTyped,
    LDAPAPIAccessModeToJSON,
} from './LDAPAPIAccessMode';

/**
 * LDAPProvider Serializer
 * @export
 * @interface PatchedLDAPProviderRequest
 */
export interface PatchedLDAPProviderRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    name?: string;
    /**
     * Flow used for authentication when the associated application is accessed by an un-authenticated user.
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    authenticationFlow?: string | null;
    /**
     * Flow used when authorizing this provider.
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    authorizationFlow?: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof PatchedLDAPProviderRequest
     */
    propertyMappings?: Array<string>;
    /**
     * DN under which objects are accessible.
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    baseDn?: string;
    /**
     * Users in this group can do search queries. If not set, every user can execute search queries.
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    searchGroup?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    certificate?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedLDAPProviderRequest
     */
    tlsServerName?: string;
    /**
     * The start for uidNumbers, this number is added to the user.pk to make sure that the numbers aren't too low for POSIX users. Default is 2000 to ensure that we don't collide with local users uidNumber
     * @type {number}
     * @memberof PatchedLDAPProviderRequest
     */
    uidStartNumber?: number;
    /**
     * The start for gidNumbers, this number is added to a number generated from the group.pk to make sure that the numbers aren't too low for POSIX groups. Default is 4000 to ensure that we don't collide with local groups or users primary groups gidNumber
     * @type {number}
     * @memberof PatchedLDAPProviderRequest
     */
    gidStartNumber?: number;
    /**
     * 
     * @type {LDAPAPIAccessMode}
     * @memberof PatchedLDAPProviderRequest
     */
    searchMode?: LDAPAPIAccessMode;
    /**
     * 
     * @type {LDAPAPIAccessMode}
     * @memberof PatchedLDAPProviderRequest
     */
    bindMode?: LDAPAPIAccessMode;
    /**
     * When enabled, code-based multi-factor authentication can be used by appending a semicolon and the TOTP code to the password. This should only be enabled if all users that will bind to this provider have a TOTP device configured, as otherwise a password may incorrectly be rejected if it contains a semicolon.
     * @type {boolean}
     * @memberof PatchedLDAPProviderRequest
     */
    mfaSupport?: boolean;
}

/**
 * Check if a given object implements the PatchedLDAPProviderRequest interface.
 */
export function instanceOfPatchedLDAPProviderRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedLDAPProviderRequestFromJSON(json: any): PatchedLDAPProviderRequest {
    return PatchedLDAPProviderRequestFromJSONTyped(json, false);
}

export function PatchedLDAPProviderRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedLDAPProviderRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': !exists(json, 'authorization_flow') ? undefined : json['authorization_flow'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'baseDn': !exists(json, 'base_dn') ? undefined : json['base_dn'],
        'searchGroup': !exists(json, 'search_group') ? undefined : json['search_group'],
        'certificate': !exists(json, 'certificate') ? undefined : json['certificate'],
        'tlsServerName': !exists(json, 'tls_server_name') ? undefined : json['tls_server_name'],
        'uidStartNumber': !exists(json, 'uid_start_number') ? undefined : json['uid_start_number'],
        'gidStartNumber': !exists(json, 'gid_start_number') ? undefined : json['gid_start_number'],
        'searchMode': !exists(json, 'search_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['search_mode']),
        'bindMode': !exists(json, 'bind_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['bind_mode']),
        'mfaSupport': !exists(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}

export function PatchedLDAPProviderRequestToJSON(value?: PatchedLDAPProviderRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'base_dn': value.baseDn,
        'search_group': value.searchGroup,
        'certificate': value.certificate,
        'tls_server_name': value.tlsServerName,
        'uid_start_number': value.uidStartNumber,
        'gid_start_number': value.gidStartNumber,
        'search_mode': LDAPAPIAccessModeToJSON(value.searchMode),
        'bind_mode': LDAPAPIAccessModeToJSON(value.bindMode),
        'mfa_support': value.mfaSupport,
    };
}
