// @goauthentik/api/src/models/PatchedTOTPDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for totp authenticator devices
 * @export
 * @interface PatchedTOTPDeviceRequest
 */
export interface PatchedTOTPDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof PatchedTOTPDeviceRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedTOTPDeviceRequest interface.
 */
export function instanceOfPatchedTOTPDeviceRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedTOTPDeviceRequestFromJSON(json: any): PatchedTOTPDeviceRequest {
    return PatchedTOTPDeviceRequestFromJSONTyped(json, false);
}

export function PatchedTOTPDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedTOTPDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedTOTPDeviceRequestToJSON(value?: PatchedTOTPDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
