// @goauthentik/api/src/models/PatchedExtAuthStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * ExtAuthStage Serializer
 * @export
 * @interface PatchedExtAuthStageRequest
 */
export interface PatchedExtAuthStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedExtAuthStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedExtAuthStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
}

/**
 * Check if a given object implements the PatchedExtAuthStageRequest interface.
 */
export function instanceOfPatchedExtAuthStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedExtAuthStageRequestFromJSON(json: any): PatchedExtAuthStageRequest {
    return PatchedExtAuthStageRequestFromJSONTyped(json, false);
}

export function PatchedExtAuthStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedExtAuthStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
    };
}

export function PatchedExtAuthStageRequestToJSON(value?: PatchedExtAuthStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
    };
}
