// @goauthentik/api/src/models/PatchedCaptchaStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * CaptchaStage Serializer
 * @export
 * @interface PatchedCaptchaStageRequest
 */
export interface PatchedCaptchaStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedCaptchaStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedCaptchaStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Public key, acquired your captcha Provider.
     * @type {string}
     * @memberof PatchedCaptchaStageRequest
     */
    publicKey?: string;
    /**
     * Private key, acquired your captcha Provider.
     * @type {string}
     * @memberof PatchedCaptchaStageRequest
     */
    privateKey?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedCaptchaStageRequest
     */
    jsUrl?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedCaptchaStageRequest
     */
    apiUrl?: string;
}

/**
 * Check if a given object implements the PatchedCaptchaStageRequest interface.
 */
export function instanceOfPatchedCaptchaStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedCaptchaStageRequestFromJSON(json: any): PatchedCaptchaStageRequest {
    return PatchedCaptchaStageRequestFromJSONTyped(json, false);
}

export function PatchedCaptchaStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedCaptchaStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'publicKey': !exists(json, 'public_key') ? undefined : json['public_key'],
        'privateKey': !exists(json, 'private_key') ? undefined : json['private_key'],
        'jsUrl': !exists(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !exists(json, 'api_url') ? undefined : json['api_url'],
    };
}

export function PatchedCaptchaStageRequestToJSON(value?: PatchedCaptchaStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'public_key': value.publicKey,
        'private_key': value.privateKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
