// @goauthentik/api/src/models/PatchedGroupRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Group Serializer
 * @export
 * @interface PatchedGroupRequest
 */
export interface PatchedGroupRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedGroupRequest
     */
    name?: string;
    /**
     * Users added to this group will be superusers.
     * @type {boolean}
     * @memberof PatchedGroupRequest
     */
    isSuperuser?: boolean;
    /**
     * 
     * @type {string}
     * @memberof PatchedGroupRequest
     */
    parent?: string | null;
    /**
     * 
     * @type {Array<number>}
     * @memberof PatchedGroupRequest
     */
    users?: Array<number>;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof PatchedGroupRequest
     */
    attributes?: { [key: string]: any; };
    /**
     * 
     * @type {Array<string>}
     * @memberof PatchedGroupRequest
     */
    roles?: Array<string>;
}

/**
 * Check if a given object implements the PatchedGroupRequest interface.
 */
export function instanceOfPatchedGroupRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedGroupRequestFromJSON(json: any): PatchedGroupRequest {
    return PatchedGroupRequestFromJSONTyped(json, false);
}

export function PatchedGroupRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedGroupRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'isSuperuser': !exists(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !exists(json, 'parent') ? undefined : json['parent'],
        'users': !exists(json, 'users') ? undefined : json['users'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'roles': !exists(json, 'roles') ? undefined : json['roles'],
    };
}

export function PatchedGroupRequestToJSON(value?: PatchedGroupRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'users': value.users,
        'attributes': value.attributes,
        'roles': value.roles,
    };
}
