// @goauthentik/api/src/models/PatchedUserSAMLSourceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SAML Source Serializer
 * @export
 * @interface PatchedUserSAMLSourceConnectionRequest
 */
export interface PatchedUserSAMLSourceConnectionRequest {
    /**
     * 
     * @type {number}
     * @memberof PatchedUserSAMLSourceConnectionRequest
     */
    user?: number;
    /**
     * 
     * @type {string}
     * @memberof PatchedUserSAMLSourceConnectionRequest
     */
    identifier?: string;
}

/**
 * Check if a given object implements the PatchedUserSAMLSourceConnectionRequest interface.
 */
export function instanceOfPatchedUserSAMLSourceConnectionRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedUserSAMLSourceConnectionRequestFromJSON(json: any): PatchedUserSAMLSourceConnectionRequest {
    return PatchedUserSAMLSourceConnectionRequestFromJSONTyped(json, false);
}

export function PatchedUserSAMLSourceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedUserSAMLSourceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': !exists(json, 'user') ? undefined : json['user'],
        'identifier': !exists(json, 'identifier') ? undefined : json['identifier'],
    };
}

export function PatchedUserSAMLSourceConnectionRequestToJSON(value?: PatchedUserSAMLSourceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'identifier': value.identifier,
    };
}
