// @goauthentik/api/src/models/PatchedRACPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * RACPropertyMapping Serializer
 * @export
 * @interface PatchedRACPropertyMappingRequest
 */
export interface PatchedRACPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof PatchedRACPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedRACPropertyMappingRequest
     */
    name?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedRACPropertyMappingRequest
     */
    expression?: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof PatchedRACPropertyMappingRequest
     */
    staticSettings?: { [key: string]: any; };
}

/**
 * Check if a given object implements the PatchedRACPropertyMappingRequest interface.
 */
export function instanceOfPatchedRACPropertyMappingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedRACPropertyMappingRequestFromJSON(json: any): PatchedRACPropertyMappingRequest {
    return PatchedRACPropertyMappingRequestFromJSONTyped(json, false);
}

export function PatchedRACPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedRACPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'staticSettings': !exists(json, 'static_settings') ? undefined : json['static_settings'],
    };
}

export function PatchedRACPropertyMappingRequestToJSON(value?: PatchedRACPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'static_settings': value.staticSettings,
    };
}
