// @goauthentik/api/src/models/PatchedAuthenticatorTOTPStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { DigitsEnum } from './DigitsEnum';
import {
    DigitsEnumFromJSON,
    DigitsEnumFromJSONTyped,
    DigitsEnumToJSON,
} from './DigitsEnum';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * AuthenticatorTOTPStage Serializer
 * @export
 * @interface PatchedAuthenticatorTOTPStageRequest
 */
export interface PatchedAuthenticatorTOTPStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorTOTPStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedAuthenticatorTOTPStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Flow used by an authenticated user to configure this Stage. If empty, user will not be able to configure this stage.
     * @type {string}
     * @memberof PatchedAuthenticatorTOTPStageRequest
     */
    configureFlow?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedAuthenticatorTOTPStageRequest
     */
    friendlyName?: string | null;
    /**
     * 
     * @type {DigitsEnum}
     * @memberof PatchedAuthenticatorTOTPStageRequest
     */
    digits?: DigitsEnum;
}

/**
 * Check if a given object implements the PatchedAuthenticatorTOTPStageRequest interface.
 */
export function instanceOfPatchedAuthenticatorTOTPStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedAuthenticatorTOTPStageRequestFromJSON(json: any): PatchedAuthenticatorTOTPStageRequest {
    return PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json, false);
}

export function PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedAuthenticatorTOTPStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'digits': !exists(json, 'digits') ? undefined : DigitsEnumFromJSON(json['digits']),
    };
}

export function PatchedAuthenticatorTOTPStageRequestToJSON(value?: PatchedAuthenticatorTOTPStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'digits': DigitsEnumToJSON(value.digits),
    };
}
