// @goauthentik/api/src/models/PatchedKubernetesServiceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * KubernetesServiceConnection Serializer
 * @export
 * @interface PatchedKubernetesServiceConnectionRequest
 */
export interface PatchedKubernetesServiceConnectionRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedKubernetesServiceConnectionRequest
     */
    name?: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof PatchedKubernetesServiceConnectionRequest
     */
    local?: boolean;
    /**
     * Paste your kubeconfig here. authentik will automatically use the currently selected context.
     * @type {any}
     * @memberof PatchedKubernetesServiceConnectionRequest
     */
    kubeconfig?: any | null;
    /**
     * Verify SSL Certificates of the Kubernetes API endpoint
     * @type {boolean}
     * @memberof PatchedKubernetesServiceConnectionRequest
     */
    verifySsl?: boolean;
}

/**
 * Check if a given object implements the PatchedKubernetesServiceConnectionRequest interface.
 */
export function instanceOfPatchedKubernetesServiceConnectionRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedKubernetesServiceConnectionRequestFromJSON(json: any): PatchedKubernetesServiceConnectionRequest {
    return PatchedKubernetesServiceConnectionRequestFromJSONTyped(json, false);
}

export function PatchedKubernetesServiceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedKubernetesServiceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'kubeconfig': !exists(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !exists(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}

export function PatchedKubernetesServiceConnectionRequestToJSON(value?: PatchedKubernetesServiceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
