// @goauthentik/api/src/models/PatchedSettingsRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Settings Serializer
 * @export
 * @interface PatchedSettingsRequest
 */
export interface PatchedSettingsRequest {
    /**
     * Configure how authentik should show avatars for users.
     * @type {string}
     * @memberof PatchedSettingsRequest
     */
    avatars?: string;
    /**
     * Enable the ability for users to change their name.
     * @type {boolean}
     * @memberof PatchedSettingsRequest
     */
    defaultUserChangeName?: boolean;
    /**
     * Enable the ability for users to change their email address.
     * @type {boolean}
     * @memberof PatchedSettingsRequest
     */
    defaultUserChangeEmail?: boolean;
    /**
     * Enable the ability for users to change their username.
     * @type {boolean}
     * @memberof PatchedSettingsRequest
     */
    defaultUserChangeUsername?: boolean;
    /**
     * Events will be deleted after this duration.(Format: weeks=3;days=2;hours=3,seconds=2).
     * @type {string}
     * @memberof PatchedSettingsRequest
     */
    eventRetention?: string;
    /**
     * The option configures the footer links on the flow executor pages.
     * @type {any}
     * @memberof PatchedSettingsRequest
     */
    footerLinks?: any | null;
    /**
     * When enabled, all the events caused by a user will be deleted upon the user's deletion.
     * @type {boolean}
     * @memberof PatchedSettingsRequest
     */
    gdprCompliance?: boolean;
    /**
     * Globally enable/disable impersonation.
     * @type {boolean}
     * @memberof PatchedSettingsRequest
     */
    impersonation?: boolean;
}

/**
 * Check if a given object implements the PatchedSettingsRequest interface.
 */
export function instanceOfPatchedSettingsRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedSettingsRequestFromJSON(json: any): PatchedSettingsRequest {
    return PatchedSettingsRequestFromJSONTyped(json, false);
}

export function PatchedSettingsRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedSettingsRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'avatars': !exists(json, 'avatars') ? undefined : json['avatars'],
        'defaultUserChangeName': !exists(json, 'default_user_change_name') ? undefined : json['default_user_change_name'],
        'defaultUserChangeEmail': !exists(json, 'default_user_change_email') ? undefined : json['default_user_change_email'],
        'defaultUserChangeUsername': !exists(json, 'default_user_change_username') ? undefined : json['default_user_change_username'],
        'eventRetention': !exists(json, 'event_retention') ? undefined : json['event_retention'],
        'footerLinks': !exists(json, 'footer_links') ? undefined : json['footer_links'],
        'gdprCompliance': !exists(json, 'gdpr_compliance') ? undefined : json['gdpr_compliance'],
        'impersonation': !exists(json, 'impersonation') ? undefined : json['impersonation'],
    };
}

export function PatchedSettingsRequestToJSON(value?: PatchedSettingsRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'avatars': value.avatars,
        'default_user_change_name': value.defaultUserChangeName,
        'default_user_change_email': value.defaultUserChangeEmail,
        'default_user_change_username': value.defaultUserChangeUsername,
        'event_retention': value.eventRetention,
        'footer_links': value.footerLinks,
        'gdpr_compliance': value.gdprCompliance,
        'impersonation': value.impersonation,
    };
}
