// @goauthentik/api/src/models/PatchedInvitationStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * InvitationStage Serializer
 * @export
 * @interface PatchedInvitationStageRequest
 */
export interface PatchedInvitationStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedInvitationStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedInvitationStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * If this flag is set, this Stage will jump to the next Stage when no Invitation is given. By default this Stage will cancel the Flow when no invitation is given.
     * @type {boolean}
     * @memberof PatchedInvitationStageRequest
     */
    continueFlowWithoutInvitation?: boolean;
}

/**
 * Check if a given object implements the PatchedInvitationStageRequest interface.
 */
export function instanceOfPatchedInvitationStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedInvitationStageRequestFromJSON(json: any): PatchedInvitationStageRequest {
    return PatchedInvitationStageRequestFromJSONTyped(json, false);
}

export function PatchedInvitationStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedInvitationStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'continueFlowWithoutInvitation': !exists(json, 'continue_flow_without_invitation') ? undefined : json['continue_flow_without_invitation'],
    };
}

export function PatchedInvitationStageRequestToJSON(value?: PatchedInvitationStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'continue_flow_without_invitation': value.continueFlowWithoutInvitation,
    };
}
