// @goauthentik/api/src/models/PatchedTenantRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant Serializer
 * @export
 * @interface PatchedTenantRequest
 */
export interface PatchedTenantRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedTenantRequest
     */
    schemaName?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedTenantRequest
     */
    name?: string;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedTenantRequest
     */
    ready?: boolean;
}

/**
 * Check if a given object implements the PatchedTenantRequest interface.
 */
export function instanceOfPatchedTenantRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedTenantRequestFromJSON(json: any): PatchedTenantRequest {
    return PatchedTenantRequestFromJSONTyped(json, false);
}

export function PatchedTenantRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedTenantRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'schemaName': !exists(json, 'schema_name') ? undefined : json['schema_name'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'ready': !exists(json, 'ready') ? undefined : json['ready'],
    };
}

export function PatchedTenantRequestToJSON(value?: PatchedTenantRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
