// @goauthentik/api/src/models/PatchedSAMLPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SAMLPropertyMapping Serializer
 * @export
 * @interface PatchedSAMLPropertyMappingRequest
 */
export interface PatchedSAMLPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof PatchedSAMLPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PatchedSAMLPropertyMappingRequest
     */
    name?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedSAMLPropertyMappingRequest
     */
    expression?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedSAMLPropertyMappingRequest
     */
    samlName?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedSAMLPropertyMappingRequest
     */
    friendlyName?: string | null;
}

/**
 * Check if a given object implements the PatchedSAMLPropertyMappingRequest interface.
 */
export function instanceOfPatchedSAMLPropertyMappingRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedSAMLPropertyMappingRequestFromJSON(json: any): PatchedSAMLPropertyMappingRequest {
    return PatchedSAMLPropertyMappingRequestFromJSONTyped(json, false);
}

export function PatchedSAMLPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedSAMLPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'samlName': !exists(json, 'saml_name') ? undefined : json['saml_name'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}

export function PatchedSAMLPropertyMappingRequestToJSON(value?: PatchedSAMLPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
