// @goauthentik/api/src/models/PatchedStaticDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for static authenticator devices
 * @export
 * @interface PatchedStaticDeviceRequest
 */
export interface PatchedStaticDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof PatchedStaticDeviceRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedStaticDeviceRequest interface.
 */
export function instanceOfPatchedStaticDeviceRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedStaticDeviceRequestFromJSON(json: any): PatchedStaticDeviceRequest {
    return PatchedStaticDeviceRequestFromJSONTyped(json, false);
}

export function PatchedStaticDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedStaticDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedStaticDeviceRequestToJSON(value?: PatchedStaticDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
