// @goauthentik/api/src/models/PatchedNotificationRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { EventRequest } from './EventRequest';
import {
    EventRequestFromJSON,
    EventRequestFromJSONTyped,
    EventRequestToJSON,
} from './EventRequest';

/**
 * Notification Serializer
 * @export
 * @interface PatchedNotificationRequest
 */
export interface PatchedNotificationRequest {
    /**
     * 
     * @type {EventRequest}
     * @memberof PatchedNotificationRequest
     */
    event?: EventRequest;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedNotificationRequest
     */
    seen?: boolean;
}

/**
 * Check if a given object implements the PatchedNotificationRequest interface.
 */
export function instanceOfPatchedNotificationRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedNotificationRequestFromJSON(json: any): PatchedNotificationRequest {
    return PatchedNotificationRequestFromJSONTyped(json, false);
}

export function PatchedNotificationRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedNotificationRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'event': !exists(json, 'event') ? undefined : EventRequestFromJSON(json['event']),
        'seen': !exists(json, 'seen') ? undefined : json['seen'],
    };
}

export function PatchedNotificationRequestToJSON(value?: PatchedNotificationRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'event': EventRequestToJSON(value.event),
        'seen': value.seen,
    };
}
