// @goauthentik/api/src/models/PatchedPromptStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * PromptStage Serializer
 * @export
 * @interface PatchedPromptStageRequest
 */
export interface PatchedPromptStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedPromptStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedPromptStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PatchedPromptStageRequest
     */
    fields?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PatchedPromptStageRequest
     */
    validationPolicies?: Array<string>;
}

/**
 * Check if a given object implements the PatchedPromptStageRequest interface.
 */
export function instanceOfPatchedPromptStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedPromptStageRequestFromJSON(json: any): PatchedPromptStageRequest {
    return PatchedPromptStageRequestFromJSONTyped(json, false);
}

export function PatchedPromptStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedPromptStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'fields': !exists(json, 'fields') ? undefined : json['fields'],
        'validationPolicies': !exists(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}

export function PatchedPromptStageRequestToJSON(value?: PatchedPromptStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
