// @goauthentik/api/src/models/PatchedWebAuthnDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for WebAuthn authenticator devices
 * @export
 * @interface PatchedWebAuthnDeviceRequest
 */
export interface PatchedWebAuthnDeviceRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedWebAuthnDeviceRequest
     */
    name?: string;
}

/**
 * Check if a given object implements the PatchedWebAuthnDeviceRequest interface.
 */
export function instanceOfPatchedWebAuthnDeviceRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedWebAuthnDeviceRequestFromJSON(json: any): PatchedWebAuthnDeviceRequest {
    return PatchedWebAuthnDeviceRequestFromJSONTyped(json, false);
}

export function PatchedWebAuthnDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedWebAuthnDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function PatchedWebAuthnDeviceRequestToJSON(value?: PatchedWebAuthnDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
