// @goauthentik/api/src/models/PatchedReputationPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Reputation Policy Serializer
 * @export
 * @interface PatchedReputationPolicyRequest
 */
export interface PatchedReputationPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedReputationPolicyRequest
     */
    name?: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof PatchedReputationPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedReputationPolicyRequest
     */
    checkIp?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedReputationPolicyRequest
     */
    checkUsername?: boolean;
    /**
     * 
     * @type {number}
     * @memberof PatchedReputationPolicyRequest
     */
    threshold?: number;
}

/**
 * Check if a given object implements the PatchedReputationPolicyRequest interface.
 */
export function instanceOfPatchedReputationPolicyRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedReputationPolicyRequestFromJSON(json: any): PatchedReputationPolicyRequest {
    return PatchedReputationPolicyRequestFromJSONTyped(json, false);
}

export function PatchedReputationPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedReputationPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'checkIp': !exists(json, 'check_ip') ? undefined : json['check_ip'],
        'checkUsername': !exists(json, 'check_username') ? undefined : json['check_username'],
        'threshold': !exists(json, 'threshold') ? undefined : json['threshold'],
    };
}

export function PatchedReputationPolicyRequestToJSON(value?: PatchedReputationPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'check_ip': value.checkIp,
        'check_username': value.checkUsername,
        'threshold': value.threshold,
    };
}
