// @goauthentik/api/src/models/PatchedDomainRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Domain Serializer
 * @export
 * @interface PatchedDomainRequest
 */
export interface PatchedDomainRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedDomainRequest
     */
    domain?: string;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedDomainRequest
     */
    isPrimary?: boolean;
    /**
     * 
     * @type {string}
     * @memberof PatchedDomainRequest
     */
    tenant?: string;
}

/**
 * Check if a given object implements the PatchedDomainRequest interface.
 */
export function instanceOfPatchedDomainRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedDomainRequestFromJSON(json: any): PatchedDomainRequest {
    return PatchedDomainRequestFromJSONTyped(json, false);
}

export function PatchedDomainRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedDomainRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'domain': !exists(json, 'domain') ? undefined : json['domain'],
        'isPrimary': !exists(json, 'is_primary') ? undefined : json['is_primary'],
        'tenant': !exists(json, 'tenant') ? undefined : json['tenant'],
    };
}

export function PatchedDomainRequestToJSON(value?: PatchedDomainRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'domain': value.domain,
        'is_primary': value.isPrimary,
        'tenant': value.tenant,
    };
}
