// @goauthentik/api/src/models/PatchedPasswordPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Password Policy Serializer
 * @export
 * @interface PatchedPasswordPolicyRequest
 */
export interface PatchedPasswordPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedPasswordPolicyRequest
     */
    name?: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof PatchedPasswordPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * Field key to check, field keys defined in Prompt stages are available.
     * @type {string}
     * @memberof PatchedPasswordPolicyRequest
     */
    passwordField?: string;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    amountDigits?: number;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    amountUppercase?: number;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    amountLowercase?: number;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    amountSymbols?: number;
    /**
     * 
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    lengthMin?: number;
    /**
     * 
     * @type {string}
     * @memberof PatchedPasswordPolicyRequest
     */
    symbolCharset?: string;
    /**
     * 
     * @type {string}
     * @memberof PatchedPasswordPolicyRequest
     */
    errorMessage?: string;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedPasswordPolicyRequest
     */
    checkStaticRules?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedPasswordPolicyRequest
     */
    checkHaveIBeenPwned?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PatchedPasswordPolicyRequest
     */
    checkZxcvbn?: boolean;
    /**
     * How many times the password hash is allowed to be on haveibeenpwned
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    hibpAllowedCount?: number;
    /**
     * If the zxcvbn score is equal or less than this value, the policy will fail.
     * @type {number}
     * @memberof PatchedPasswordPolicyRequest
     */
    zxcvbnScoreThreshold?: number;
}

/**
 * Check if a given object implements the PatchedPasswordPolicyRequest interface.
 */
export function instanceOfPatchedPasswordPolicyRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedPasswordPolicyRequestFromJSON(json: any): PatchedPasswordPolicyRequest {
    return PatchedPasswordPolicyRequestFromJSONTyped(json, false);
}

export function PatchedPasswordPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedPasswordPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'passwordField': !exists(json, 'password_field') ? undefined : json['password_field'],
        'amountDigits': !exists(json, 'amount_digits') ? undefined : json['amount_digits'],
        'amountUppercase': !exists(json, 'amount_uppercase') ? undefined : json['amount_uppercase'],
        'amountLowercase': !exists(json, 'amount_lowercase') ? undefined : json['amount_lowercase'],
        'amountSymbols': !exists(json, 'amount_symbols') ? undefined : json['amount_symbols'],
        'lengthMin': !exists(json, 'length_min') ? undefined : json['length_min'],
        'symbolCharset': !exists(json, 'symbol_charset') ? undefined : json['symbol_charset'],
        'errorMessage': !exists(json, 'error_message') ? undefined : json['error_message'],
        'checkStaticRules': !exists(json, 'check_static_rules') ? undefined : json['check_static_rules'],
        'checkHaveIBeenPwned': !exists(json, 'check_have_i_been_pwned') ? undefined : json['check_have_i_been_pwned'],
        'checkZxcvbn': !exists(json, 'check_zxcvbn') ? undefined : json['check_zxcvbn'],
        'hibpAllowedCount': !exists(json, 'hibp_allowed_count') ? undefined : json['hibp_allowed_count'],
        'zxcvbnScoreThreshold': !exists(json, 'zxcvbn_score_threshold') ? undefined : json['zxcvbn_score_threshold'],
    };
}

export function PatchedPasswordPolicyRequestToJSON(value?: PatchedPasswordPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'password_field': value.passwordField,
        'amount_digits': value.amountDigits,
        'amount_uppercase': value.amountUppercase,
        'amount_lowercase': value.amountLowercase,
        'amount_symbols': value.amountSymbols,
        'length_min': value.lengthMin,
        'symbol_charset': value.symbolCharset,
        'error_message': value.errorMessage,
        'check_static_rules': value.checkStaticRules,
        'check_have_i_been_pwned': value.checkHaveIBeenPwned,
        'check_zxcvbn': value.checkZxcvbn,
        'hibp_allowed_count': value.hibpAllowedCount,
        'zxcvbn_score_threshold': value.zxcvbnScoreThreshold,
    };
}
