// @goauthentik/api/src/models/PatchedUserLoginStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';
import type { GeoipBindingEnum } from './GeoipBindingEnum';
import {
    GeoipBindingEnumFromJSON,
    GeoipBindingEnumFromJSONTyped,
    GeoipBindingEnumToJSON,
} from './GeoipBindingEnum';
import type { NetworkBindingEnum } from './NetworkBindingEnum';
import {
    NetworkBindingEnumFromJSON,
    NetworkBindingEnumFromJSONTyped,
    NetworkBindingEnumToJSON,
} from './NetworkBindingEnum';

/**
 * UserLoginStage Serializer
 * @export
 * @interface PatchedUserLoginStageRequest
 */
export interface PatchedUserLoginStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedUserLoginStageRequest
     */
    name?: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PatchedUserLoginStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * Determines how long a session lasts. Default of 0 means that the sessions lasts until the browser is closed. (Format: hours=-1;minutes=-2;seconds=-3)
     * @type {string}
     * @memberof PatchedUserLoginStageRequest
     */
    sessionDuration?: string;
    /**
     * Terminate all other sessions of the user logging in.
     * @type {boolean}
     * @memberof PatchedUserLoginStageRequest
     */
    terminateOtherSessions?: boolean;
    /**
     * Offset the session will be extended by when the user picks the remember me option. Default of 0 means that the remember me option will not be shown. (Format: hours=-1;minutes=-2;seconds=-3)
     * @type {string}
     * @memberof PatchedUserLoginStageRequest
     */
    rememberMeOffset?: string;
    /**
     * 
     * @type {NetworkBindingEnum}
     * @memberof PatchedUserLoginStageRequest
     */
    networkBinding?: NetworkBindingEnum;
    /**
     * 
     * @type {GeoipBindingEnum}
     * @memberof PatchedUserLoginStageRequest
     */
    geoipBinding?: GeoipBindingEnum;
}

/**
 * Check if a given object implements the PatchedUserLoginStageRequest interface.
 */
export function instanceOfPatchedUserLoginStageRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedUserLoginStageRequestFromJSON(json: any): PatchedUserLoginStageRequest {
    return PatchedUserLoginStageRequestFromJSONTyped(json, false);
}

export function PatchedUserLoginStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedUserLoginStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'sessionDuration': !exists(json, 'session_duration') ? undefined : json['session_duration'],
        'terminateOtherSessions': !exists(json, 'terminate_other_sessions') ? undefined : json['terminate_other_sessions'],
        'rememberMeOffset': !exists(json, 'remember_me_offset') ? undefined : json['remember_me_offset'],
        'networkBinding': !exists(json, 'network_binding') ? undefined : NetworkBindingEnumFromJSON(json['network_binding']),
        'geoipBinding': !exists(json, 'geoip_binding') ? undefined : GeoipBindingEnumFromJSON(json['geoip_binding']),
    };
}

export function PatchedUserLoginStageRequestToJSON(value?: PatchedUserLoginStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'session_duration': value.sessionDuration,
        'terminate_other_sessions': value.terminateOtherSessions,
        'remember_me_offset': value.rememberMeOffset,
        'network_binding': NetworkBindingEnumToJSON(value.networkBinding),
        'geoip_binding': GeoipBindingEnumToJSON(value.geoipBinding),
    };
}
