// @goauthentik/api/src/models/PatchedConnectionTokenRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * ConnectionToken Serializer
 * @export
 * @interface PatchedConnectionTokenRequest
 */
export interface PatchedConnectionTokenRequest {
    /**
     * 
     * @type {number}
     * @memberof PatchedConnectionTokenRequest
     */
    provider?: number;
}

/**
 * Check if a given object implements the PatchedConnectionTokenRequest interface.
 */
export function instanceOfPatchedConnectionTokenRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedConnectionTokenRequestFromJSON(json: any): PatchedConnectionTokenRequest {
    return PatchedConnectionTokenRequestFromJSONTyped(json, false);
}

export function PatchedConnectionTokenRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedConnectionTokenRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'provider': !exists(json, 'provider') ? undefined : json['provider'],
    };
}

export function PatchedConnectionTokenRequestToJSON(value?: PatchedConnectionTokenRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'provider': value.provider,
    };
}
