// @goauthentik/api/src/models/PatchedInvitationRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Invitation Serializer
 * @export
 * @interface PatchedInvitationRequest
 */
export interface PatchedInvitationRequest {
    /**
     * 
     * @type {string}
     * @memberof PatchedInvitationRequest
     */
    name?: string;
    /**
     * 
     * @type {Date}
     * @memberof PatchedInvitationRequest
     */
    expires?: Date;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof PatchedInvitationRequest
     */
    fixedData?: { [key: string]: any; };
    /**
     * When enabled, the invitation will be deleted after usage.
     * @type {boolean}
     * @memberof PatchedInvitationRequest
     */
    singleUse?: boolean;
    /**
     * When set, only the configured flow can use this invitation.
     * @type {string}
     * @memberof PatchedInvitationRequest
     */
    flow?: string | null;
}

/**
 * Check if a given object implements the PatchedInvitationRequest interface.
 */
export function instanceOfPatchedInvitationRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PatchedInvitationRequestFromJSON(json: any): PatchedInvitationRequest {
    return PatchedInvitationRequestFromJSONTyped(json, false);
}

export function PatchedInvitationRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PatchedInvitationRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !exists(json, 'fixed_data') ? undefined : json['fixed_data'],
        'singleUse': !exists(json, 'single_use') ? undefined : json['single_use'],
        'flow': !exists(json, 'flow') ? undefined : json['flow'],
    };
}

export function PatchedInvitationRequestToJSON(value?: PatchedInvitationRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
