// @goauthentik/api/src/models/PermissionAssignRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ModelEnum } from './ModelEnum';
import {
    ModelEnumFromJSON,
    ModelEnumFromJSONTyped,
    ModelEnumToJSON,
} from './ModelEnum';

/**
 * Request to assign a new permission
 * @export
 * @interface PermissionAssignRequest
 */
export interface PermissionAssignRequest {
    /**
     * 
     * @type {Array<string>}
     * @memberof PermissionAssignRequest
     */
    permissions: Array<string>;
    /**
     * 
     * @type {ModelEnum}
     * @memberof PermissionAssignRequest
     */
    model?: ModelEnum;
    /**
     * 
     * @type {string}
     * @memberof PermissionAssignRequest
     */
    objectPk?: string;
}

/**
 * Check if a given object implements the PermissionAssignRequest interface.
 */
export function instanceOfPermissionAssignRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "permissions" in value;

    return isInstance;
}

export function PermissionAssignRequestFromJSON(json: any): PermissionAssignRequest {
    return PermissionAssignRequestFromJSONTyped(json, false);
}

export function PermissionAssignRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PermissionAssignRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'permissions': json['permissions'],
        'model': !exists(json, 'model') ? undefined : ModelEnumFromJSON(json['model']),
        'objectPk': !exists(json, 'object_pk') ? undefined : json['object_pk'],
    };
}

export function PermissionAssignRequestToJSON(value?: PermissionAssignRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'permissions': value.permissions,
        'model': ModelEnumToJSON(value.model),
        'object_pk': value.objectPk,
    };
}
