// @goauthentik/api/src/models/Permission.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Global permission
 * @export
 * @interface Permission
 */
export interface Permission {
    /**
     * 
     * @type {number}
     * @memberof Permission
     */
    readonly id: number;
    /**
     * 
     * @type {string}
     * @memberof Permission
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof Permission
     */
    codename: string;
    /**
     * 
     * @type {string}
     * @memberof Permission
     */
    readonly model: string;
    /**
     * 
     * @type {string}
     * @memberof Permission
     */
    readonly appLabel: string;
    /**
     * Human-readable app label
     * @type {string}
     * @memberof Permission
     */
    readonly appLabelVerbose: string;
    /**
     * Human-readable model name
     * @type {string}
     * @memberof Permission
     */
    readonly modelVerbose: string;
}

/**
 * Check if a given object implements the Permission interface.
 */
export function instanceOfPermission(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "appLabelVerbose" in value;
    isInstance = isInstance && "modelVerbose" in value;

    return isInstance;
}

export function PermissionFromJSON(json: any): Permission {
    return PermissionFromJSONTyped(json, false);
}

export function PermissionFromJSONTyped(json: any, ignoreDiscriminator: boolean): Permission {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': json['id'],
        'name': json['name'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'appLabelVerbose': json['app_label_verbose'],
        'modelVerbose': json['model_verbose'],
    };
}

export function PermissionToJSON(value?: Permission | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'codename': value.codename,
    };
}
