// @goauthentik/api/src/models/PlexTokenRedeemRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer to redeem a plex token
 * @export
 * @interface PlexTokenRedeemRequest
 */
export interface PlexTokenRedeemRequest {
    /**
     * 
     * @type {string}
     * @memberof PlexTokenRedeemRequest
     */
    plexToken: string;
}

/**
 * Check if a given object implements the PlexTokenRedeemRequest interface.
 */
export function instanceOfPlexTokenRedeemRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "plexToken" in value;

    return isInstance;
}

export function PlexTokenRedeemRequestFromJSON(json: any): PlexTokenRedeemRequest {
    return PlexTokenRedeemRequestFromJSONTyped(json, false);
}

export function PlexTokenRedeemRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PlexTokenRedeemRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'plexToken': json['plex_token'],
    };
}

export function PlexTokenRedeemRequestToJSON(value?: PlexTokenRedeemRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'plex_token': value.plexToken,
    };
}
