// @goauthentik/api/src/models/PlexAuthenticationChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';

/**
 * Challenge shown to the user in identification stage
 * @export
 * @interface PlexAuthenticationChallenge
 */
export interface PlexAuthenticationChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof PlexAuthenticationChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof PlexAuthenticationChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof PlexAuthenticationChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof PlexAuthenticationChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {string}
     * @memberof PlexAuthenticationChallenge
     */
    clientId: string;
    /**
     * 
     * @type {string}
     * @memberof PlexAuthenticationChallenge
     */
    slug: string;
}

/**
 * Check if a given object implements the PlexAuthenticationChallenge interface.
 */
export function instanceOfPlexAuthenticationChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "slug" in value;

    return isInstance;
}

export function PlexAuthenticationChallengeFromJSON(json: any): PlexAuthenticationChallenge {
    return PlexAuthenticationChallengeFromJSONTyped(json, false);
}

export function PlexAuthenticationChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): PlexAuthenticationChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'clientId': json['client_id'],
        'slug': json['slug'],
    };
}

export function PlexAuthenticationChallengeToJSON(value?: PlexAuthenticationChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'client_id': value.clientId,
        'slug': value.slug,
    };
}
