// @goauthentik/api/src/models/PlexAuthenticationChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Pseudo class for plex response
 * @export
 * @interface PlexAuthenticationChallengeResponseRequest
 */
export interface PlexAuthenticationChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof PlexAuthenticationChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the PlexAuthenticationChallengeResponseRequest interface.
 */
export function instanceOfPlexAuthenticationChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PlexAuthenticationChallengeResponseRequestFromJSON(json: any): PlexAuthenticationChallengeResponseRequest {
    return PlexAuthenticationChallengeResponseRequestFromJSONTyped(json, false);
}

export function PlexAuthenticationChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PlexAuthenticationChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function PlexAuthenticationChallengeResponseRequestToJSON(value?: PlexAuthenticationChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
    };
}
