// @goauthentik/api/src/models/PlexSourceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Plex Source connection Serializer
 * @export
 * @interface PlexSourceConnectionRequest
 */
export interface PlexSourceConnectionRequest {
    /**
     * 
     * @type {string}
     * @memberof PlexSourceConnectionRequest
     */
    identifier: string;
    /**
     * 
     * @type {string}
     * @memberof PlexSourceConnectionRequest
     */
    plexToken: string;
}

/**
 * Check if a given object implements the PlexSourceConnectionRequest interface.
 */
export function instanceOfPlexSourceConnectionRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "plexToken" in value;

    return isInstance;
}

export function PlexSourceConnectionRequestFromJSON(json: any): PlexSourceConnectionRequest {
    return PlexSourceConnectionRequestFromJSONTyped(json, false);
}

export function PlexSourceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PlexSourceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'identifier': json['identifier'],
        'plexToken': json['plex_token'],
    };
}

export function PlexSourceConnectionRequestToJSON(value?: PlexSourceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
