// @goauthentik/api/src/models/PlexSourceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Source } from './Source';
import {
    SourceFromJSON,
    SourceFromJSONTyped,
    SourceToJSON,
} from './Source';

/**
 * Plex Source connection Serializer
 * @export
 * @interface PlexSourceConnection
 */
export interface PlexSourceConnection {
    /**
     * 
     * @type {number}
     * @memberof PlexSourceConnection
     */
    readonly pk: number;
    /**
     * 
     * @type {number}
     * @memberof PlexSourceConnection
     */
    readonly user: number;
    /**
     * 
     * @type {Source}
     * @memberof PlexSourceConnection
     */
    readonly source: Source;
    /**
     * 
     * @type {string}
     * @memberof PlexSourceConnection
     */
    identifier: string;
    /**
     * 
     * @type {string}
     * @memberof PlexSourceConnection
     */
    plexToken: string;
}

/**
 * Check if a given object implements the PlexSourceConnection interface.
 */
export function instanceOfPlexSourceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "plexToken" in value;

    return isInstance;
}

export function PlexSourceConnectionFromJSON(json: any): PlexSourceConnection {
    return PlexSourceConnectionFromJSONTyped(json, false);
}

export function PlexSourceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): PlexSourceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'identifier': json['identifier'],
        'plexToken': json['plex_token'],
    };
}

export function PlexSourceConnectionToJSON(value?: PlexSourceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
