// @goauthentik/api/src/models/PolicyTestRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Test policy execution for a user with context
 * @export
 * @interface PolicyTestRequest
 */
export interface PolicyTestRequest {
    /**
     * 
     * @type {number}
     * @memberof PolicyTestRequest
     */
    user: number;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof PolicyTestRequest
     */
    context?: { [key: string]: any; };
}

/**
 * Check if a given object implements the PolicyTestRequest interface.
 */
export function instanceOfPolicyTestRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "user" in value;

    return isInstance;
}

export function PolicyTestRequestFromJSON(json: any): PolicyTestRequest {
    return PolicyTestRequestFromJSONTyped(json, false);
}

export function PolicyTestRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PolicyTestRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': json['user'],
        'context': !exists(json, 'context') ? undefined : json['context'],
    };
}

export function PolicyTestRequestToJSON(value?: PolicyTestRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'context': value.context,
    };
}
