// @goauthentik/api/src/models/PolicyEngineMode.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `all` - all, all policies must pass
 * * `any` - any, any policy must pass
 * @export
 */
export const PolicyEngineMode = {
    All: 'all',
    Any: 'any',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type PolicyEngineMode = typeof PolicyEngineMode[keyof typeof PolicyEngineMode];


export function PolicyEngineModeFromJSON(json: any): PolicyEngineMode {
    return PolicyEngineModeFromJSONTyped(json, false);
}

export function PolicyEngineModeFromJSONTyped(json: any, ignoreDiscriminator: boolean): PolicyEngineMode {
    return json as PolicyEngineMode;
}

export function PolicyEngineModeToJSON(value?: PolicyEngineMode | null): any {
    return value as any;
}
