// @goauthentik/api/src/models/PolicyTestResult.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * result of a policy test
 * @export
 * @interface PolicyTestResult
 */
export interface PolicyTestResult {
    /**
     * 
     * @type {boolean}
     * @memberof PolicyTestResult
     */
    passing: boolean;
    /**
     * 
     * @type {Array<string>}
     * @memberof PolicyTestResult
     */
    readonly messages: Array<string>;
    /**
     * 
     * @type {Array<{ [key: string]: any; }>}
     * @memberof PolicyTestResult
     */
    readonly logMessages: Array<{ [key: string]: any; }>;
}

/**
 * Check if a given object implements the PolicyTestResult interface.
 */
export function instanceOfPolicyTestResult(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "passing" in value;
    isInstance = isInstance && "messages" in value;
    isInstance = isInstance && "logMessages" in value;

    return isInstance;
}

export function PolicyTestResultFromJSON(json: any): PolicyTestResult {
    return PolicyTestResultFromJSONTyped(json, false);
}

export function PolicyTestResultFromJSONTyped(json: any, ignoreDiscriminator: boolean): PolicyTestResult {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'passing': json['passing'],
        'messages': json['messages'],
        'logMessages': json['log_messages'],
    };
}

export function PolicyTestResultToJSON(value?: PolicyTestResult | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'passing': value.passing,
    };
}
