// @goauthentik/api/src/models/PolicyBinding.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Group } from './Group';
import {
    GroupFromJSON,
    GroupFromJSONTyped,
    GroupToJSON,
} from './Group';
import type { Policy } from './Policy';
import {
    PolicyFromJSON,
    PolicyFromJSONTyped,
    PolicyToJSON,
} from './Policy';
import type { User } from './User';
import {
    UserFromJSON,
    UserFromJSONTyped,
    UserToJSON,
} from './User';

/**
 * PolicyBinding Serializer
 * @export
 * @interface PolicyBinding
 */
export interface PolicyBinding {
    /**
     * 
     * @type {string}
     * @memberof PolicyBinding
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof PolicyBinding
     */
    policy?: string | null;
    /**
     * 
     * @type {string}
     * @memberof PolicyBinding
     */
    group?: string | null;
    /**
     * 
     * @type {number}
     * @memberof PolicyBinding
     */
    user?: number | null;
    /**
     * 
     * @type {Policy}
     * @memberof PolicyBinding
     */
    readonly policyObj: Policy;
    /**
     * 
     * @type {Group}
     * @memberof PolicyBinding
     */
    readonly groupObj: Group;
    /**
     * 
     * @type {User}
     * @memberof PolicyBinding
     */
    readonly userObj: User;
    /**
     * 
     * @type {string}
     * @memberof PolicyBinding
     */
    target: string;
    /**
     * Negates the outcome of the policy. Messages are unaffected.
     * @type {boolean}
     * @memberof PolicyBinding
     */
    negate?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof PolicyBinding
     */
    enabled?: boolean;
    /**
     * 
     * @type {number}
     * @memberof PolicyBinding
     */
    order: number;
    /**
     * Timeout after which Policy execution is terminated.
     * @type {number}
     * @memberof PolicyBinding
     */
    timeout?: number;
    /**
     * Result if the Policy execution fails.
     * @type {boolean}
     * @memberof PolicyBinding
     */
    failureResult?: boolean;
}

/**
 * Check if a given object implements the PolicyBinding interface.
 */
export function instanceOfPolicyBinding(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "policyObj" in value;
    isInstance = isInstance && "groupObj" in value;
    isInstance = isInstance && "userObj" in value;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "order" in value;

    return isInstance;
}

export function PolicyBindingFromJSON(json: any): PolicyBinding {
    return PolicyBindingFromJSONTyped(json, false);
}

export function PolicyBindingFromJSONTyped(json: any, ignoreDiscriminator: boolean): PolicyBinding {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'policy': !exists(json, 'policy') ? undefined : json['policy'],
        'group': !exists(json, 'group') ? undefined : json['group'],
        'user': !exists(json, 'user') ? undefined : json['user'],
        'policyObj': PolicyFromJSON(json['policy_obj']),
        'groupObj': GroupFromJSON(json['group_obj']),
        'userObj': UserFromJSON(json['user_obj']),
        'target': json['target'],
        'negate': !exists(json, 'negate') ? undefined : json['negate'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'order': json['order'],
        'timeout': !exists(json, 'timeout') ? undefined : json['timeout'],
        'failureResult': !exists(json, 'failure_result') ? undefined : json['failure_result'],
    };
}

export function PolicyBindingToJSON(value?: PolicyBinding | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'policy': value.policy,
        'group': value.group,
        'user': value.user,
        'target': value.target,
        'negate': value.negate,
        'enabled': value.enabled,
        'order': value.order,
        'timeout': value.timeout,
        'failure_result': value.failureResult,
    };
}
