// @goauthentik/api/src/models/PromptStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';

/**
 * PromptStage Serializer
 * @export
 * @interface PromptStage
 */
export interface PromptStage {
    /**
     * 
     * @type {string}
     * @memberof PromptStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof PromptStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof PromptStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof PromptStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof PromptStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof PromptStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof PromptStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PromptStage
     */
    fields: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PromptStage
     */
    validationPolicies?: Array<string>;
}

/**
 * Check if a given object implements the PromptStage interface.
 */
export function instanceOfPromptStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "fields" in value;

    return isInstance;
}

export function PromptStageFromJSON(json: any): PromptStage {
    return PromptStageFromJSONTyped(json, false);
}

export function PromptStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): PromptStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'fields': json['fields'],
        'validationPolicies': !exists(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}

export function PromptStageToJSON(value?: PromptStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
