// @goauthentik/api/src/models/PromptChallenge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { ChallengeChoices } from './ChallengeChoices';
import {
    ChallengeChoicesFromJSON,
    ChallengeChoicesFromJSONTyped,
    ChallengeChoicesToJSON,
} from './ChallengeChoices';
import type { ContextualFlowInfo } from './ContextualFlowInfo';
import {
    ContextualFlowInfoFromJSON,
    ContextualFlowInfoFromJSONTyped,
    ContextualFlowInfoToJSON,
} from './ContextualFlowInfo';
import type { ErrorDetail } from './ErrorDetail';
import {
    ErrorDetailFromJSON,
    ErrorDetailFromJSONTyped,
    ErrorDetailToJSON,
} from './ErrorDetail';
import type { StagePrompt } from './StagePrompt';
import {
    StagePromptFromJSON,
    StagePromptFromJSONTyped,
    StagePromptToJSON,
} from './StagePrompt';

/**
 * Initial challenge being sent, define fields
 * @export
 * @interface PromptChallenge
 */
export interface PromptChallenge {
    /**
     * 
     * @type {ChallengeChoices}
     * @memberof PromptChallenge
     */
    type: ChallengeChoices;
    /**
     * 
     * @type {ContextualFlowInfo}
     * @memberof PromptChallenge
     */
    flowInfo?: ContextualFlowInfo;
    /**
     * 
     * @type {string}
     * @memberof PromptChallenge
     */
    component?: string;
    /**
     * 
     * @type {{ [key: string]: Array<ErrorDetail>; }}
     * @memberof PromptChallenge
     */
    responseErrors?: { [key: string]: Array<ErrorDetail>; };
    /**
     * 
     * @type {Array<StagePrompt>}
     * @memberof PromptChallenge
     */
    fields: Array<StagePrompt>;
}

/**
 * Check if a given object implements the PromptChallenge interface.
 */
export function instanceOfPromptChallenge(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "fields" in value;

    return isInstance;
}

export function PromptChallengeFromJSON(json: any): PromptChallenge {
    return PromptChallengeFromJSONTyped(json, false);
}

export function PromptChallengeFromJSONTyped(json: any, ignoreDiscriminator: boolean): PromptChallenge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'fields': ((json['fields'] as Array<any>).map(StagePromptFromJSON)),
    };
}

export function PromptChallengeToJSON(value?: PromptChallenge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'fields': ((value.fields as Array<any>).map(StagePromptToJSON)),
    };
}
