// @goauthentik/api/src/models/PromptChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Validate response, fields are dynamically created based
 * on the stage
 * @export
 * @interface PromptChallengeResponseRequest
 */
export interface PromptChallengeResponseRequest {
    [key: string]: any | any;
    /**
     * 
     * @type {string}
     * @memberof PromptChallengeResponseRequest
     */
    component?: string;
}

/**
 * Check if a given object implements the PromptChallengeResponseRequest interface.
 */
export function instanceOfPromptChallengeResponseRequest(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function PromptChallengeResponseRequestFromJSON(json: any): PromptChallengeResponseRequest {
    return PromptChallengeResponseRequestFromJSONTyped(json, false);
}

export function PromptChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PromptChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
            ...json,
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}

export function PromptChallengeResponseRequestToJSON(value?: PromptChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
            ...value,
        'component': value.component,
    };
}
