// @goauthentik/api/src/models/PromptStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * PromptStage Serializer
 * @export
 * @interface PromptStageRequest
 */
export interface PromptStageRequest {
    /**
     * 
     * @type {string}
     * @memberof PromptStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof PromptStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PromptStageRequest
     */
    fields: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof PromptStageRequest
     */
    validationPolicies?: Array<string>;
}

/**
 * Check if a given object implements the PromptStageRequest interface.
 */
export function instanceOfPromptStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "fields" in value;

    return isInstance;
}

export function PromptStageRequestFromJSON(json: any): PromptStageRequest {
    return PromptStageRequestFromJSONTyped(json, false);
}

export function PromptStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): PromptStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'fields': json['fields'],
        'validationPolicies': !exists(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}

export function PromptStageRequestToJSON(value?: PromptStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
