// @goauthentik/api/src/models/PropertyMappingTestResult.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Result of a Property-mapping test
 * @export
 * @interface PropertyMappingTestResult
 */
export interface PropertyMappingTestResult {
    /**
     * 
     * @type {string}
     * @memberof PropertyMappingTestResult
     */
    readonly result: string;
    /**
     * 
     * @type {boolean}
     * @memberof PropertyMappingTestResult
     */
    readonly successful: boolean;
}

/**
 * Check if a given object implements the PropertyMappingTestResult interface.
 */
export function instanceOfPropertyMappingTestResult(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "result" in value;
    isInstance = isInstance && "successful" in value;

    return isInstance;
}

export function PropertyMappingTestResultFromJSON(json: any): PropertyMappingTestResult {
    return PropertyMappingTestResultFromJSONTyped(json, false);
}

export function PropertyMappingTestResultFromJSONTyped(json: any, ignoreDiscriminator: boolean): PropertyMappingTestResult {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'result': json['result'],
        'successful': json['successful'],
    };
}

export function PropertyMappingTestResultToJSON(value?: PropertyMappingTestResult | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
