// @goauthentik/api/src/models/PropertyMappingPreview.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Preview how the current user is mapped via the property mappings selected in a provider
 * @export
 * @interface PropertyMappingPreview
 */
export interface PropertyMappingPreview {
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof PropertyMappingPreview
     */
    readonly preview: { [key: string]: any; };
}

/**
 * Check if a given object implements the PropertyMappingPreview interface.
 */
export function instanceOfPropertyMappingPreview(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "preview" in value;

    return isInstance;
}

export function PropertyMappingPreviewFromJSON(json: any): PropertyMappingPreview {
    return PropertyMappingPreviewFromJSONTyped(json, false);
}

export function PropertyMappingPreviewFromJSONTyped(json: any, ignoreDiscriminator: boolean): PropertyMappingPreview {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'preview': json['preview'],
    };
}

export function PropertyMappingPreviewToJSON(value?: PropertyMappingPreview | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
