// @goauthentik/api/src/models/ProviderModelEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `authentik_providers_ldap.ldapprovider` - authentik_providers_ldap.ldapprovider
 * * `authentik_providers_oauth2.oauth2provider` - authentik_providers_oauth2.oauth2provider
 * * `authentik_providers_proxy.proxyprovider` - authentik_providers_proxy.proxyprovider
 * * `authentik_providers_rac.racprovider` - authentik_providers_rac.racprovider
 * * `authentik_providers_radius.radiusprovider` - authentik_providers_radius.radiusprovider
 * * `authentik_providers_saml.samlprovider` - authentik_providers_saml.samlprovider
 * * `authentik_providers_scim.scimprovider` - authentik_providers_scim.scimprovider
 * @export
 */
export const ProviderModelEnum = {
    LdapLdapprovider: 'authentik_providers_ldap.ldapprovider',
    Oauth2Oauth2provider: 'authentik_providers_oauth2.oauth2provider',
    ProxyProxyprovider: 'authentik_providers_proxy.proxyprovider',
    RacRacprovider: 'authentik_providers_rac.racprovider',
    RadiusRadiusprovider: 'authentik_providers_radius.radiusprovider',
    SamlSamlprovider: 'authentik_providers_saml.samlprovider',
    ScimScimprovider: 'authentik_providers_scim.scimprovider',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ProviderModelEnum = typeof ProviderModelEnum[keyof typeof ProviderModelEnum];


export function ProviderModelEnumFromJSON(json: any): ProviderModelEnum {
    return ProviderModelEnumFromJSONTyped(json, false);
}

export function ProviderModelEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): ProviderModelEnum {
    return json as ProviderModelEnum;
}

export function ProviderModelEnumToJSON(value?: ProviderModelEnum | null): any {
    return value as any;
}
