// @goauthentik/api/src/models/ProxyMode.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `proxy` - Proxy
 * * `forward_single` - Forward Single
 * * `forward_domain` - Forward Domain
 * @export
 */
export const ProxyMode = {
    Proxy: 'proxy',
    ForwardSingle: 'forward_single',
    ForwardDomain: 'forward_domain',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ProxyMode = typeof ProxyMode[keyof typeof ProxyMode];


export function ProxyModeFromJSON(json: any): ProxyMode {
    return ProxyModeFromJSONTyped(json, false);
}

export function ProxyModeFromJSONTyped(json: any, ignoreDiscriminator: boolean): ProxyMode {
    return json as ProxyMode;
}

export function ProxyModeToJSON(value?: ProxyMode | null): any {
    return value as any;
}
