// @goauthentik/api/src/models/RACProvider.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * RACProvider Serializer
 * @export
 * @interface RACProvider
 */
export interface RACProvider {
    /**
     * 
     * @type {number}
     * @memberof RACProvider
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof RACProvider
     */
    name: string;
    /**
     * Flow used for authentication when the associated application is accessed by an un-authenticated user.
     * @type {string}
     * @memberof RACProvider
     */
    authenticationFlow?: string | null;
    /**
     * Flow used when authorizing this provider.
     * @type {string}
     * @memberof RACProvider
     */
    authorizationFlow: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof RACProvider
     */
    propertyMappings?: Array<string>;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof RACProvider
     */
    readonly component: string;
    /**
     * Internal application name, used in URLs.
     * @type {string}
     * @memberof RACProvider
     */
    readonly assignedApplicationSlug: string;
    /**
     * Application's display Name.
     * @type {string}
     * @memberof RACProvider
     */
    readonly assignedApplicationName: string;
    /**
     * Internal application name, used in URLs.
     * @type {string}
     * @memberof RACProvider
     */
    readonly assignedBackchannelApplicationSlug: string;
    /**
     * Application's display Name.
     * @type {string}
     * @memberof RACProvider
     */
    readonly assignedBackchannelApplicationName: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof RACProvider
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof RACProvider
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof RACProvider
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {any}
     * @memberof RACProvider
     */
    settings?: any | null;
    /**
     * 
     * @type {Array<string>}
     * @memberof RACProvider
     */
    readonly outpostSet: Array<string>;
    /**
     * Determines how long a session lasts. Default of 0 means that the sessions lasts until the browser is closed. (Format: hours=-1;minutes=-2;seconds=-3)
     * @type {string}
     * @memberof RACProvider
     */
    connectionExpiry?: string;
    /**
     * When set to true, connection tokens will be deleted upon disconnect.
     * @type {boolean}
     * @memberof RACProvider
     */
    deleteTokenOnDisconnect?: boolean;
}

/**
 * Check if a given object implements the RACProvider interface.
 */
export function instanceOfRACProvider(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "authorizationFlow" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "assignedApplicationSlug" in value;
    isInstance = isInstance && "assignedApplicationName" in value;
    isInstance = isInstance && "assignedBackchannelApplicationSlug" in value;
    isInstance = isInstance && "assignedBackchannelApplicationName" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "outpostSet" in value;

    return isInstance;
}

export function RACProviderFromJSON(json: any): RACProvider {
    return RACProviderFromJSONTyped(json, false);
}

export function RACProviderFromJSONTyped(json: any, ignoreDiscriminator: boolean): RACProvider {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': json['authorization_flow'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'component': json['component'],
        'assignedApplicationSlug': json['assigned_application_slug'],
        'assignedApplicationName': json['assigned_application_name'],
        'assignedBackchannelApplicationSlug': json['assigned_backchannel_application_slug'],
        'assignedBackchannelApplicationName': json['assigned_backchannel_application_name'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'settings': !exists(json, 'settings') ? undefined : json['settings'],
        'outpostSet': json['outpost_set'],
        'connectionExpiry': !exists(json, 'connection_expiry') ? undefined : json['connection_expiry'],
        'deleteTokenOnDisconnect': !exists(json, 'delete_token_on_disconnect') ? undefined : json['delete_token_on_disconnect'],
    };
}

export function RACProviderToJSON(value?: RACProvider | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'settings': value.settings,
        'connection_expiry': value.connectionExpiry,
        'delete_token_on_disconnect': value.deleteTokenOnDisconnect,
    };
}
