// @goauthentik/api/src/models/RACPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * RACPropertyMapping Serializer
 * @export
 * @interface RACPropertyMappingRequest
 */
export interface RACPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof RACPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof RACPropertyMappingRequest
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof RACPropertyMappingRequest
     */
    expression?: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof RACPropertyMappingRequest
     */
    staticSettings: { [key: string]: any; };
}

/**
 * Check if a given object implements the RACPropertyMappingRequest interface.
 */
export function instanceOfRACPropertyMappingRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "staticSettings" in value;

    return isInstance;
}

export function RACPropertyMappingRequestFromJSON(json: any): RACPropertyMappingRequest {
    return RACPropertyMappingRequestFromJSONTyped(json, false);
}

export function RACPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): RACPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'staticSettings': json['static_settings'],
    };
}

export function RACPropertyMappingRequestToJSON(value?: RACPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'static_settings': value.staticSettings,
    };
}
