// @goauthentik/api/src/models/Reputation.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Reputation Serializer
 * @export
 * @interface Reputation
 */
export interface Reputation {
    /**
     * 
     * @type {string}
     * @memberof Reputation
     */
    pk?: string;
    /**
     * 
     * @type {string}
     * @memberof Reputation
     */
    identifier: string;
    /**
     * 
     * @type {string}
     * @memberof Reputation
     */
    ip: string;
    /**
     * 
     * @type {any}
     * @memberof Reputation
     */
    ipGeoData?: any | null;
    /**
     * 
     * @type {any}
     * @memberof Reputation
     */
    ipAsnData?: any | null;
    /**
     * 
     * @type {number}
     * @memberof Reputation
     */
    score?: number;
    /**
     * 
     * @type {Date}
     * @memberof Reputation
     */
    readonly updated: Date;
}

/**
 * Check if a given object implements the Reputation interface.
 */
export function instanceOfReputation(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "ip" in value;
    isInstance = isInstance && "updated" in value;

    return isInstance;
}

export function ReputationFromJSON(json: any): Reputation {
    return ReputationFromJSONTyped(json, false);
}

export function ReputationFromJSONTyped(json: any, ignoreDiscriminator: boolean): Reputation {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': !exists(json, 'pk') ? undefined : json['pk'],
        'identifier': json['identifier'],
        'ip': json['ip'],
        'ipGeoData': !exists(json, 'ip_geo_data') ? undefined : json['ip_geo_data'],
        'ipAsnData': !exists(json, 'ip_asn_data') ? undefined : json['ip_asn_data'],
        'score': !exists(json, 'score') ? undefined : json['score'],
        'updated': (new Date(json['updated'])),
    };
}

export function ReputationToJSON(value?: Reputation | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pk': value.pk,
        'identifier': value.identifier,
        'ip': value.ip,
        'ip_geo_data': value.ipGeoData,
        'ip_asn_data': value.ipAsnData,
        'score': value.score,
    };
}
