// @goauthentik/api/src/models/ReputationPolicyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Reputation Policy Serializer
 * @export
 * @interface ReputationPolicyRequest
 */
export interface ReputationPolicyRequest {
    /**
     * 
     * @type {string}
     * @memberof ReputationPolicyRequest
     */
    name: string;
    /**
     * When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.
     * @type {boolean}
     * @memberof ReputationPolicyRequest
     */
    executionLogging?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof ReputationPolicyRequest
     */
    checkIp?: boolean;
    /**
     * 
     * @type {boolean}
     * @memberof ReputationPolicyRequest
     */
    checkUsername?: boolean;
    /**
     * 
     * @type {number}
     * @memberof ReputationPolicyRequest
     */
    threshold?: number;
}

/**
 * Check if a given object implements the ReputationPolicyRequest interface.
 */
export function instanceOfReputationPolicyRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function ReputationPolicyRequestFromJSON(json: any): ReputationPolicyRequest {
    return ReputationPolicyRequestFromJSONTyped(json, false);
}

export function ReputationPolicyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ReputationPolicyRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'checkIp': !exists(json, 'check_ip') ? undefined : json['check_ip'],
        'checkUsername': !exists(json, 'check_username') ? undefined : json['check_username'],
        'threshold': !exists(json, 'threshold') ? undefined : json['threshold'],
    };
}

export function ReputationPolicyRequestToJSON(value?: ReputationPolicyRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'execution_logging': value.executionLogging,
        'check_ip': value.checkIp,
        'check_username': value.checkUsername,
        'threshold': value.threshold,
    };
}
