// @goauthentik/api/src/models/ResidentKeyRequirementEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `discouraged` - Discouraged
 * * `preferred` - Preferred
 * * `required` - Required
 * @export
 */
export const ResidentKeyRequirementEnum = {
    Discouraged: 'discouraged',
    Preferred: 'preferred',
    Required: 'required',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type ResidentKeyRequirementEnum = typeof ResidentKeyRequirementEnum[keyof typeof ResidentKeyRequirementEnum];


export function ResidentKeyRequirementEnumFromJSON(json: any): ResidentKeyRequirementEnum {
    return ResidentKeyRequirementEnumFromJSONTyped(json, false);
}

export function ResidentKeyRequirementEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): ResidentKeyRequirementEnum {
    return json as ResidentKeyRequirementEnum;
}

export function ResidentKeyRequirementEnumToJSON(value?: ResidentKeyRequirementEnum | null): any {
    return value as any;
}
