// @goauthentik/api/src/models/RoleObjectPermission.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Role-bound object level permission
 * @export
 * @interface RoleObjectPermission
 */
export interface RoleObjectPermission {
    /**
     * 
     * @type {number}
     * @memberof RoleObjectPermission
     */
    readonly id: number;
    /**
     * 
     * @type {string}
     * @memberof RoleObjectPermission
     */
    readonly codename: string;
    /**
     * 
     * @type {string}
     * @memberof RoleObjectPermission
     */
    readonly model: string;
    /**
     * 
     * @type {string}
     * @memberof RoleObjectPermission
     */
    readonly appLabel: string;
    /**
     * 
     * @type {string}
     * @memberof RoleObjectPermission
     */
    readonly objectPk: string;
    /**
     * 
     * @type {string}
     * @memberof RoleObjectPermission
     */
    readonly name: string;
}

/**
 * Check if a given object implements the RoleObjectPermission interface.
 */
export function instanceOfRoleObjectPermission(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "objectPk" in value;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function RoleObjectPermissionFromJSON(json: any): RoleObjectPermission {
    return RoleObjectPermissionFromJSONTyped(json, false);
}

export function RoleObjectPermissionFromJSONTyped(json: any, ignoreDiscriminator: boolean): RoleObjectPermission {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': json['id'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'objectPk': json['object_pk'],
        'name': json['name'],
    };
}

export function RoleObjectPermissionToJSON(value?: RoleObjectPermission | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
