// @goauthentik/api/src/models/SAMLPropertyMappingRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SAMLPropertyMapping Serializer
 * @export
 * @interface SAMLPropertyMappingRequest
 */
export interface SAMLPropertyMappingRequest {
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof SAMLPropertyMappingRequest
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof SAMLPropertyMappingRequest
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof SAMLPropertyMappingRequest
     */
    expression: string;
    /**
     * 
     * @type {string}
     * @memberof SAMLPropertyMappingRequest
     */
    samlName: string;
    /**
     * 
     * @type {string}
     * @memberof SAMLPropertyMappingRequest
     */
    friendlyName?: string | null;
}

/**
 * Check if a given object implements the SAMLPropertyMappingRequest interface.
 */
export function instanceOfSAMLPropertyMappingRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "samlName" in value;

    return isInstance;
}

export function SAMLPropertyMappingRequestFromJSON(json: any): SAMLPropertyMappingRequest {
    return SAMLPropertyMappingRequestFromJSONTyped(json, false);
}

export function SAMLPropertyMappingRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): SAMLPropertyMappingRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'samlName': json['saml_name'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}

export function SAMLPropertyMappingRequestToJSON(value?: SAMLPropertyMappingRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
