// @goauthentik/api/src/models/SAMLMetadata.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SAML Provider Metadata serializer
 * @export
 * @interface SAMLMetadata
 */
export interface SAMLMetadata {
    /**
     * 
     * @type {string}
     * @memberof SAMLMetadata
     */
    readonly metadata: string;
    /**
     * 
     * @type {string}
     * @memberof SAMLMetadata
     */
    readonly downloadUrl: string;
}

/**
 * Check if a given object implements the SAMLMetadata interface.
 */
export function instanceOfSAMLMetadata(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "metadata" in value;
    isInstance = isInstance && "downloadUrl" in value;

    return isInstance;
}

export function SAMLMetadataFromJSON(json: any): SAMLMetadata {
    return SAMLMetadataFromJSONTyped(json, false);
}

export function SAMLMetadataFromJSONTyped(json: any, ignoreDiscriminator: boolean): SAMLMetadata {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'metadata': json['metadata'],
        'downloadUrl': json['download_url'],
    };
}

export function SAMLMetadataToJSON(value?: SAMLMetadata | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
