// @goauthentik/api/src/models/SCIMProviderRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SCIMProvider Serializer
 * @export
 * @interface SCIMProviderRequest
 */
export interface SCIMProviderRequest {
    /**
     * 
     * @type {string}
     * @memberof SCIMProviderRequest
     */
    name: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof SCIMProviderRequest
     */
    propertyMappings?: Array<string>;
    /**
     * Property mappings used for group creation/updating.
     * @type {Array<string>}
     * @memberof SCIMProviderRequest
     */
    propertyMappingsGroup?: Array<string>;
    /**
     * Base URL to SCIM requests, usually ends in /v2
     * @type {string}
     * @memberof SCIMProviderRequest
     */
    url: string;
    /**
     * Authentication token
     * @type {string}
     * @memberof SCIMProviderRequest
     */
    token: string;
    /**
     * 
     * @type {boolean}
     * @memberof SCIMProviderRequest
     */
    excludeUsersServiceAccount?: boolean;
    /**
     * 
     * @type {string}
     * @memberof SCIMProviderRequest
     */
    filterGroup?: string | null;
}

/**
 * Check if a given object implements the SCIMProviderRequest interface.
 */
export function instanceOfSCIMProviderRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "token" in value;

    return isInstance;
}

export function SCIMProviderRequestFromJSON(json: any): SCIMProviderRequest {
    return SCIMProviderRequestFromJSONTyped(json, false);
}

export function SCIMProviderRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): SCIMProviderRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !exists(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'url': json['url'],
        'token': json['token'],
        'excludeUsersServiceAccount': !exists(json, 'exclude_users_service_account') ? undefined : json['exclude_users_service_account'],
        'filterGroup': !exists(json, 'filter_group') ? undefined : json['filter_group'],
    };
}

export function SCIMProviderRequestToJSON(value?: SCIMProviderRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
        'url': value.url,
        'token': value.token,
        'exclude_users_service_account': value.excludeUsersServiceAccount,
        'filter_group': value.filterGroup,
    };
}
