// @goauthentik/api/src/models/SCIMSyncStatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { SystemTask } from './SystemTask';
import {
    SystemTaskFromJSON,
    SystemTaskFromJSONTyped,
    SystemTaskToJSON,
} from './SystemTask';

/**
 * SCIM Provider sync status
 * @export
 * @interface SCIMSyncStatus
 */
export interface SCIMSyncStatus {
    /**
     * 
     * @type {boolean}
     * @memberof SCIMSyncStatus
     */
    readonly isRunning: boolean;
    /**
     * 
     * @type {Array<SystemTask>}
     * @memberof SCIMSyncStatus
     */
    readonly tasks: Array<SystemTask>;
}

/**
 * Check if a given object implements the SCIMSyncStatus interface.
 */
export function instanceOfSCIMSyncStatus(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "isRunning" in value;
    isInstance = isInstance && "tasks" in value;

    return isInstance;
}

export function SCIMSyncStatusFromJSON(json: any): SCIMSyncStatus {
    return SCIMSyncStatusFromJSONTyped(json, false);
}

export function SCIMSyncStatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): SCIMSyncStatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'isRunning': json['is_running'],
        'tasks': ((json['tasks'] as Array<any>).map(SystemTaskFromJSON)),
    };
}

export function SCIMSyncStatusToJSON(value?: SCIMSyncStatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
