// @goauthentik/api/src/models/SCIMProvider.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SCIMProvider Serializer
 * @export
 * @interface SCIMProvider
 */
export interface SCIMProvider {
    /**
     * 
     * @type {number}
     * @memberof SCIMProvider
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof SCIMProvider
     */
    name: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof SCIMProvider
     */
    propertyMappings?: Array<string>;
    /**
     * Property mappings used for group creation/updating.
     * @type {Array<string>}
     * @memberof SCIMProvider
     */
    propertyMappingsGroup?: Array<string>;
    /**
     * Get object component so that we know how to edit the object
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly component: string;
    /**
     * Internal application name, used in URLs.
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly assignedBackchannelApplicationSlug: string;
    /**
     * Application's display Name.
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly assignedBackchannelApplicationName: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof SCIMProvider
     */
    readonly metaModelName: string;
    /**
     * Base URL to SCIM requests, usually ends in /v2
     * @type {string}
     * @memberof SCIMProvider
     */
    url: string;
    /**
     * Authentication token
     * @type {string}
     * @memberof SCIMProvider
     */
    token: string;
    /**
     * 
     * @type {boolean}
     * @memberof SCIMProvider
     */
    excludeUsersServiceAccount?: boolean;
    /**
     * 
     * @type {string}
     * @memberof SCIMProvider
     */
    filterGroup?: string | null;
}

/**
 * Check if a given object implements the SCIMProvider interface.
 */
export function instanceOfSCIMProvider(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "assignedBackchannelApplicationSlug" in value;
    isInstance = isInstance && "assignedBackchannelApplicationName" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "token" in value;

    return isInstance;
}

export function SCIMProviderFromJSON(json: any): SCIMProvider {
    return SCIMProviderFromJSONTyped(json, false);
}

export function SCIMProviderFromJSONTyped(json: any, ignoreDiscriminator: boolean): SCIMProvider {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !exists(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'component': json['component'],
        'assignedBackchannelApplicationSlug': json['assigned_backchannel_application_slug'],
        'assignedBackchannelApplicationName': json['assigned_backchannel_application_name'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'url': json['url'],
        'token': json['token'],
        'excludeUsersServiceAccount': !exists(json, 'exclude_users_service_account') ? undefined : json['exclude_users_service_account'],
        'filterGroup': !exists(json, 'filter_group') ? undefined : json['filter_group'],
    };
}

export function SCIMProviderToJSON(value?: SCIMProvider | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
        'url': value.url,
        'token': value.token,
        'exclude_users_service_account': value.excludeUsersServiceAccount,
        'filter_group': value.filterGroup,
    };
}
