// @goauthentik/api/src/models/ScopeMapping.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * ScopeMapping Serializer
 * @export
 * @interface ScopeMapping
 */
export interface ScopeMapping {
    /**
     * 
     * @type {string}
     * @memberof ScopeMapping
     */
    readonly pk: string;
    /**
     * Objects that are managed by authentik. These objects are created and updated automatically. This flag only indicates that an object can be overwritten by migrations. You can still modify the objects via the API, but expect changes to be overwritten in a later update.
     * @type {string}
     * @memberof ScopeMapping
     */
    managed?: string | null;
    /**
     * 
     * @type {string}
     * @memberof ScopeMapping
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof ScopeMapping
     */
    expression: string;
    /**
     * Get object's component so that we know how to edit the object
     * @type {string}
     * @memberof ScopeMapping
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof ScopeMapping
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof ScopeMapping
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof ScopeMapping
     */
    readonly metaModelName: string;
    /**
     * Scope name requested by the client
     * @type {string}
     * @memberof ScopeMapping
     */
    scopeName: string;
    /**
     * Description shown to the user when consenting. If left empty, the user won't be informed.
     * @type {string}
     * @memberof ScopeMapping
     */
    description?: string;
}

/**
 * Check if a given object implements the ScopeMapping interface.
 */
export function instanceOfScopeMapping(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "scopeName" in value;

    return isInstance;
}

export function ScopeMappingFromJSON(json: any): ScopeMapping {
    return ScopeMappingFromJSONTyped(json, false);
}

export function ScopeMappingFromJSONTyped(json: any, ignoreDiscriminator: boolean): ScopeMapping {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'scopeName': json['scope_name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
    };
}

export function ScopeMappingToJSON(value?: ScopeMapping | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
