// @goauthentik/api/src/models/SelectableStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for stages which can be selected by users
 * @export
 * @interface SelectableStage
 */
export interface SelectableStage {
    /**
     * 
     * @type {string}
     * @memberof SelectableStage
     */
    pk: string;
    /**
     * 
     * @type {string}
     * @memberof SelectableStage
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof SelectableStage
     */
    verboseName: string;
    /**
     * 
     * @type {string}
     * @memberof SelectableStage
     */
    metaModelName: string;
}

/**
 * Check if a given object implements the SelectableStage interface.
 */
export function instanceOfSelectableStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function SelectableStageFromJSON(json: any): SelectableStage {
    return SelectableStageFromJSONTyped(json, false);
}

export function SelectableStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): SelectableStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'verboseName': json['verbose_name'],
        'metaModelName': json['meta_model_name'],
    };
}

export function SelectableStageToJSON(value?: SelectableStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pk': value.pk,
        'name': value.name,
        'verbose_name': value.verboseName,
        'meta_model_name': value.metaModelName,
    };
}
