// @goauthentik/api/src/models/ServiceConnectionState.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for Service connection state
 * @export
 * @interface ServiceConnectionState
 */
export interface ServiceConnectionState {
    /**
     * 
     * @type {boolean}
     * @memberof ServiceConnectionState
     */
    readonly healthy: boolean;
    /**
     * 
     * @type {string}
     * @memberof ServiceConnectionState
     */
    readonly version: string;
}

/**
 * Check if a given object implements the ServiceConnectionState interface.
 */
export function instanceOfServiceConnectionState(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "healthy" in value;
    isInstance = isInstance && "version" in value;

    return isInstance;
}

export function ServiceConnectionStateFromJSON(json: any): ServiceConnectionState {
    return ServiceConnectionStateFromJSONTyped(json, false);
}

export function ServiceConnectionStateFromJSONTyped(json: any, ignoreDiscriminator: boolean): ServiceConnectionState {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'healthy': json['healthy'],
        'version': json['version'],
    };
}

export function ServiceConnectionStateToJSON(value?: ServiceConnectionState | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
