// @goauthentik/api/src/models/ServiceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * ServiceConnection Serializer
 * @export
 * @interface ServiceConnection
 */
export interface ServiceConnection {
    /**
     * 
     * @type {string}
     * @memberof ServiceConnection
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof ServiceConnection
     */
    name: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof ServiceConnection
     */
    local?: boolean;
    /**
     * Return component used to edit this object
     * @type {string}
     * @memberof ServiceConnection
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof ServiceConnection
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof ServiceConnection
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof ServiceConnection
     */
    readonly metaModelName: string;
}

/**
 * Check if a given object implements the ServiceConnection interface.
 */
export function instanceOfServiceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function ServiceConnectionFromJSON(json: any): ServiceConnection {
    return ServiceConnectionFromJSONTyped(json, false);
}

export function ServiceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): ServiceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
    };
}

export function ServiceConnectionToJSON(value?: ServiceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
    };
}
