// @goauthentik/api/src/models/ServiceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * ServiceConnection Serializer
 * @export
 * @interface ServiceConnectionRequest
 */
export interface ServiceConnectionRequest {
    /**
     * 
     * @type {string}
     * @memberof ServiceConnectionRequest
     */
    name: string;
    /**
     * If enabled, use the local connection. Required Docker socket/Kubernetes Integration
     * @type {boolean}
     * @memberof ServiceConnectionRequest
     */
    local?: boolean;
}

/**
 * Check if a given object implements the ServiceConnectionRequest interface.
 */
export function instanceOfServiceConnectionRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function ServiceConnectionRequestFromJSON(json: any): ServiceConnectionRequest {
    return ServiceConnectionRequestFromJSONTyped(json, false);
}

export function ServiceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): ServiceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
    };
}

export function ServiceConnectionRequestToJSON(value?: ServiceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'local': value.local,
    };
}
