// @goauthentik/api/src/models/SessionUser.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { UserSelf } from './UserSelf';
import {
    UserSelfFromJSON,
    UserSelfFromJSONTyped,
    UserSelfToJSON,
} from './UserSelf';

/**
 * Response for the /user/me endpoint, returns the currently active user (as `user` property)
 * and, if this user is being impersonated, the original user in the `original` property.
 * @export
 * @interface SessionUser
 */
export interface SessionUser {
    /**
     * 
     * @type {UserSelf}
     * @memberof SessionUser
     */
    user: UserSelf;
    /**
     * 
     * @type {UserSelf}
     * @memberof SessionUser
     */
    original?: UserSelf;
}

/**
 * Check if a given object implements the SessionUser interface.
 */
export function instanceOfSessionUser(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "user" in value;

    return isInstance;
}

export function SessionUserFromJSON(json: any): SessionUser {
    return SessionUserFromJSONTyped(json, false);
}

export function SessionUserFromJSONTyped(json: any, ignoreDiscriminator: boolean): SessionUser {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': UserSelfFromJSON(json['user']),
        'original': !exists(json, 'original') ? undefined : UserSelfFromJSON(json['original']),
    };
}

export function SessionUserToJSON(value?: SessionUser | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': UserSelfToJSON(value.user),
        'original': UserSelfToJSON(value.original),
    };
}
