// @goauthentik/api/src/models/SeverityEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `notice` - Notice
 * * `warning` - Warning
 * * `alert` - Alert
 * @export
 */
export const SeverityEnum = {
    Notice: 'notice',
    Warning: 'warning',
    Alert: 'alert',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type SeverityEnum = typeof SeverityEnum[keyof typeof SeverityEnum];


export function SeverityEnumFromJSON(json: any): SeverityEnum {
    return SeverityEnumFromJSONTyped(json, false);
}

export function SeverityEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): SeverityEnum {
    return json as SeverityEnum;
}

export function SeverityEnumToJSON(value?: SeverityEnum | null): any {
    return value as any;
}
