// @goauthentik/api/src/models/SMSDevice.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for sms authenticator devices
 * @export
 * @interface SMSDevice
 */
export interface SMSDevice {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof SMSDevice
     */
    name: string;
    /**
     * 
     * @type {number}
     * @memberof SMSDevice
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof SMSDevice
     */
    readonly phoneNumber: string;
}

/**
 * Check if a given object implements the SMSDevice interface.
 */
export function instanceOfSMSDevice(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "phoneNumber" in value;

    return isInstance;
}

export function SMSDeviceFromJSON(json: any): SMSDevice {
    return SMSDeviceFromJSONTyped(json, false);
}

export function SMSDeviceFromJSONTyped(json: any, ignoreDiscriminator: boolean): SMSDevice {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'pk': json['pk'],
        'phoneNumber': json['phone_number'],
    };
}

export function SMSDeviceToJSON(value?: SMSDevice | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
