// @goauthentik/api/src/models/SMSDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for sms authenticator devices
 * @export
 * @interface SMSDeviceRequest
 */
export interface SMSDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof SMSDeviceRequest
     */
    name: string;
}

/**
 * Check if a given object implements the SMSDeviceRequest interface.
 */
export function instanceOfSMSDeviceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function SMSDeviceRequestFromJSON(json: any): SMSDeviceRequest {
    return SMSDeviceRequestFromJSONTyped(json, false);
}

export function SMSDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): SMSDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
    };
}

export function SMSDeviceRequestToJSON(value?: SMSDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
