// @goauthentik/api/src/models/StagePrompt.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { PromptTypeEnum } from './PromptTypeEnum';
import {
    PromptTypeEnumFromJSON,
    PromptTypeEnumFromJSONTyped,
    PromptTypeEnumToJSON,
} from './PromptTypeEnum';

/**
 * Serializer for a single Prompt field
 * @export
 * @interface StagePrompt
 */
export interface StagePrompt {
    /**
     * 
     * @type {string}
     * @memberof StagePrompt
     */
    fieldKey: string;
    /**
     * 
     * @type {string}
     * @memberof StagePrompt
     */
    label: string;
    /**
     * 
     * @type {PromptTypeEnum}
     * @memberof StagePrompt
     */
    type: PromptTypeEnum;
    /**
     * 
     * @type {boolean}
     * @memberof StagePrompt
     */
    required: boolean;
    /**
     * 
     * @type {string}
     * @memberof StagePrompt
     */
    placeholder: string;
    /**
     * 
     * @type {string}
     * @memberof StagePrompt
     */
    initialValue: string;
    /**
     * 
     * @type {number}
     * @memberof StagePrompt
     */
    order: number;
    /**
     * 
     * @type {string}
     * @memberof StagePrompt
     */
    subText: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof StagePrompt
     */
    choices: Array<string> | null;
}

/**
 * Check if a given object implements the StagePrompt interface.
 */
export function instanceOfStagePrompt(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "fieldKey" in value;
    isInstance = isInstance && "label" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "required" in value;
    isInstance = isInstance && "placeholder" in value;
    isInstance = isInstance && "initialValue" in value;
    isInstance = isInstance && "order" in value;
    isInstance = isInstance && "subText" in value;
    isInstance = isInstance && "choices" in value;

    return isInstance;
}

export function StagePromptFromJSON(json: any): StagePrompt {
    return StagePromptFromJSONTyped(json, false);
}

export function StagePromptFromJSONTyped(json: any, ignoreDiscriminator: boolean): StagePrompt {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'fieldKey': json['field_key'],
        'label': json['label'],
        'type': PromptTypeEnumFromJSON(json['type']),
        'required': json['required'],
        'placeholder': json['placeholder'],
        'initialValue': json['initial_value'],
        'order': json['order'],
        'subText': json['sub_text'],
        'choices': json['choices'],
    };
}

export function StagePromptToJSON(value?: StagePrompt | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'field_key': value.fieldKey,
        'label': value.label,
        'type': PromptTypeEnumToJSON(value.type),
        'required': value.required,
        'placeholder': value.placeholder,
        'initial_value': value.initialValue,
        'order': value.order,
        'sub_text': value.subText,
        'choices': value.choices,
    };
}
