// @goauthentik/api/src/models/StageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';

/**
 * Stage Serializer
 * @export
 * @interface StageRequest
 */
export interface StageRequest {
    /**
     * 
     * @type {string}
     * @memberof StageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof StageRequest
     */
    flowSet?: Array<FlowSetRequest>;
}

/**
 * Check if a given object implements the StageRequest interface.
 */
export function instanceOfStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function StageRequestFromJSON(json: any): StageRequest {
    return StageRequestFromJSONTyped(json, false);
}

export function StageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): StageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
    };
}

export function StageRequestToJSON(value?: StageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
    };
}
