// @goauthentik/api/src/models/StaticDeviceTokenRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for static device's tokens
 * @export
 * @interface StaticDeviceTokenRequest
 */
export interface StaticDeviceTokenRequest {
    /**
     * 
     * @type {string}
     * @memberof StaticDeviceTokenRequest
     */
    token: string;
}

/**
 * Check if a given object implements the StaticDeviceTokenRequest interface.
 */
export function instanceOfStaticDeviceTokenRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "token" in value;

    return isInstance;
}

export function StaticDeviceTokenRequestFromJSON(json: any): StaticDeviceTokenRequest {
    return StaticDeviceTokenRequestFromJSONTyped(json, false);
}

export function StaticDeviceTokenRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): StaticDeviceTokenRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'token': json['token'],
    };
}

export function StaticDeviceTokenRequestToJSON(value?: StaticDeviceTokenRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'token': value.token,
    };
}
