// @goauthentik/api/src/models/StaticDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for static authenticator devices
 * @export
 * @interface StaticDeviceRequest
 */
export interface StaticDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof StaticDeviceRequest
     */
    name: string;
}

/**
 * Check if a given object implements the StaticDeviceRequest interface.
 */
export function instanceOfStaticDeviceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function StaticDeviceRequestFromJSON(json: any): StaticDeviceRequest {
    return StaticDeviceRequestFromJSONTyped(json, false);
}

export function StaticDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): StaticDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
    };
}

export function StaticDeviceRequestToJSON(value?: StaticDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
