// @goauthentik/api/src/models/SystemTaskStatusEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `unknown` - UNKNOWN
 * * `successful` - SUCCESSFUL
 * * `warning` - WARNING
 * * `error` - ERROR
 * @export
 */
export const SystemTaskStatusEnum = {
    Unknown: 'unknown',
    Successful: 'successful',
    Warning: 'warning',
    Error: 'error',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type SystemTaskStatusEnum = typeof SystemTaskStatusEnum[keyof typeof SystemTaskStatusEnum];


export function SystemTaskStatusEnumFromJSON(json: any): SystemTaskStatusEnum {
    return SystemTaskStatusEnumFromJSONTyped(json, false);
}

export function SystemTaskStatusEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): SystemTaskStatusEnum {
    return json as SystemTaskStatusEnum;
}

export function SystemTaskStatusEnumToJSON(value?: SystemTaskStatusEnum | null): any {
    return value as any;
}
