// @goauthentik/api/src/models/SystemInfoRuntime.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Get versions
 * @export
 * @interface SystemInfoRuntime
 */
export interface SystemInfoRuntime {
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    pythonVersion: string;
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    gunicornVersion: string;
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    environment: string;
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    architecture: string;
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    platform: string;
    /**
     * 
     * @type {string}
     * @memberof SystemInfoRuntime
     */
    uname: string;
}

/**
 * Check if a given object implements the SystemInfoRuntime interface.
 */
export function instanceOfSystemInfoRuntime(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pythonVersion" in value;
    isInstance = isInstance && "gunicornVersion" in value;
    isInstance = isInstance && "environment" in value;
    isInstance = isInstance && "architecture" in value;
    isInstance = isInstance && "platform" in value;
    isInstance = isInstance && "uname" in value;

    return isInstance;
}

export function SystemInfoRuntimeFromJSON(json: any): SystemInfoRuntime {
    return SystemInfoRuntimeFromJSONTyped(json, false);
}

export function SystemInfoRuntimeFromJSONTyped(json: any, ignoreDiscriminator: boolean): SystemInfoRuntime {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pythonVersion': json['python_version'],
        'gunicornVersion': json['gunicorn_version'],
        'environment': json['environment'],
        'architecture': json['architecture'],
        'platform': json['platform'],
        'uname': json['uname'],
    };
}

export function SystemInfoRuntimeToJSON(value?: SystemInfoRuntime | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'python_version': value.pythonVersion,
        'gunicorn_version': value.gunicornVersion,
        'environment': value.environment,
        'architecture': value.architecture,
        'platform': value.platform,
        'uname': value.uname,
    };
}
