// @goauthentik/api/src/models/Tenant.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant Serializer
 * @export
 * @interface Tenant
 */
export interface Tenant {
    /**
     * 
     * @type {string}
     * @memberof Tenant
     */
    readonly tenantUuid: string;
    /**
     * 
     * @type {string}
     * @memberof Tenant
     */
    schemaName: string;
    /**
     * 
     * @type {string}
     * @memberof Tenant
     */
    name: string;
    /**
     * 
     * @type {boolean}
     * @memberof Tenant
     */
    ready?: boolean;
}

/**
 * Check if a given object implements the Tenant interface.
 */
export function instanceOfTenant(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "tenantUuid" in value;
    isInstance = isInstance && "schemaName" in value;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function TenantFromJSON(json: any): Tenant {
    return TenantFromJSONTyped(json, false);
}

export function TenantFromJSONTyped(json: any, ignoreDiscriminator: boolean): Tenant {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'tenantUuid': json['tenant_uuid'],
        'schemaName': json['schema_name'],
        'name': json['name'],
        'ready': !exists(json, 'ready') ? undefined : json['ready'],
    };
}

export function TenantToJSON(value?: Tenant | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
