// @goauthentik/api/src/models/TenantAdminGroupRequestRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant admin group creation request serializer
 * @export
 * @interface TenantAdminGroupRequestRequest
 */
export interface TenantAdminGroupRequestRequest {
    /**
     * 
     * @type {string}
     * @memberof TenantAdminGroupRequestRequest
     */
    user: string;
}

/**
 * Check if a given object implements the TenantAdminGroupRequestRequest interface.
 */
export function instanceOfTenantAdminGroupRequestRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "user" in value;

    return isInstance;
}

export function TenantAdminGroupRequestRequestFromJSON(json: any): TenantAdminGroupRequestRequest {
    return TenantAdminGroupRequestRequestFromJSONTyped(json, false);
}

export function TenantAdminGroupRequestRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): TenantAdminGroupRequestRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': json['user'],
    };
}

export function TenantAdminGroupRequestRequestToJSON(value?: TenantAdminGroupRequestRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
    };
}
