// @goauthentik/api/src/models/TenantRecoveryKeyRequestRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant recovery key creation request serializer
 * @export
 * @interface TenantRecoveryKeyRequestRequest
 */
export interface TenantRecoveryKeyRequestRequest {
    /**
     * 
     * @type {string}
     * @memberof TenantRecoveryKeyRequestRequest
     */
    user: string;
    /**
     * 
     * @type {number}
     * @memberof TenantRecoveryKeyRequestRequest
     */
    durationDays: number;
}

/**
 * Check if a given object implements the TenantRecoveryKeyRequestRequest interface.
 */
export function instanceOfTenantRecoveryKeyRequestRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "durationDays" in value;

    return isInstance;
}

export function TenantRecoveryKeyRequestRequestFromJSON(json: any): TenantRecoveryKeyRequestRequest {
    return TenantRecoveryKeyRequestRequestFromJSONTyped(json, false);
}

export function TenantRecoveryKeyRequestRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): TenantRecoveryKeyRequestRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': json['user'],
        'durationDays': json['duration_days'],
    };
}

export function TenantRecoveryKeyRequestRequestToJSON(value?: TenantRecoveryKeyRequestRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'duration_days': value.durationDays,
    };
}
