// @goauthentik/api/src/models/TenantRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant Serializer
 * @export
 * @interface TenantRequest
 */
export interface TenantRequest {
    /**
     * 
     * @type {string}
     * @memberof TenantRequest
     */
    schemaName: string;
    /**
     * 
     * @type {string}
     * @memberof TenantRequest
     */
    name: string;
    /**
     * 
     * @type {boolean}
     * @memberof TenantRequest
     */
    ready?: boolean;
}

/**
 * Check if a given object implements the TenantRequest interface.
 */
export function instanceOfTenantRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "schemaName" in value;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function TenantRequestFromJSON(json: any): TenantRequest {
    return TenantRequestFromJSONTyped(json, false);
}

export function TenantRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): TenantRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'schemaName': json['schema_name'],
        'name': json['name'],
        'ready': !exists(json, 'ready') ? undefined : json['ready'],
    };
}

export function TenantRequestToJSON(value?: TenantRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
