// @goauthentik/api/src/models/TenantRecoveryKeyResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Tenant recovery key creation response serializer
 * @export
 * @interface TenantRecoveryKeyResponse
 */
export interface TenantRecoveryKeyResponse {
    /**
     * 
     * @type {Date}
     * @memberof TenantRecoveryKeyResponse
     */
    expiry: Date;
    /**
     * 
     * @type {string}
     * @memberof TenantRecoveryKeyResponse
     */
    url: string;
}

/**
 * Check if a given object implements the TenantRecoveryKeyResponse interface.
 */
export function instanceOfTenantRecoveryKeyResponse(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "expiry" in value;
    isInstance = isInstance && "url" in value;

    return isInstance;
}

export function TenantRecoveryKeyResponseFromJSON(json: any): TenantRecoveryKeyResponse {
    return TenantRecoveryKeyResponseFromJSONTyped(json, false);
}

export function TenantRecoveryKeyResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): TenantRecoveryKeyResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'expiry': (new Date(json['expiry'])),
        'url': json['url'],
    };
}

export function TenantRecoveryKeyResponseToJSON(value?: TenantRecoveryKeyResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'expiry': (value.expiry.toISOString()),
        'url': value.url,
    };
}
