// @goauthentik/api/src/models/TokenView.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Show token's current key
 * @export
 * @interface TokenView
 */
export interface TokenView {
    /**
     * 
     * @type {string}
     * @memberof TokenView
     */
    readonly key: string;
}

/**
 * Check if a given object implements the TokenView interface.
 */
export function instanceOfTokenView(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "key" in value;

    return isInstance;
}

export function TokenViewFromJSON(json: any): TokenView {
    return TokenViewFromJSONTyped(json, false);
}

export function TokenViewFromJSONTyped(json: any, ignoreDiscriminator: boolean): TokenView {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'key': json['key'],
    };
}

export function TokenViewToJSON(value?: TokenView | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
