// @goauthentik/api/src/models/TOTPDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for totp authenticator devices
 * @export
 * @interface TOTPDeviceRequest
 */
export interface TOTPDeviceRequest {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof TOTPDeviceRequest
     */
    name: string;
}

/**
 * Check if a given object implements the TOTPDeviceRequest interface.
 */
export function instanceOfTOTPDeviceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function TOTPDeviceRequestFromJSON(json: any): TOTPDeviceRequest {
    return TOTPDeviceRequestFromJSONTyped(json, false);
}

export function TOTPDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): TOTPDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
    };
}

export function TOTPDeviceRequestToJSON(value?: TOTPDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
