// @goauthentik/api/src/models/TOTPDevice.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for totp authenticator devices
 * @export
 * @interface TOTPDevice
 */
export interface TOTPDevice {
    /**
     * The human-readable name of this device.
     * @type {string}
     * @memberof TOTPDevice
     */
    name: string;
    /**
     * 
     * @type {number}
     * @memberof TOTPDevice
     */
    readonly pk: number;
}

/**
 * Check if a given object implements the TOTPDevice interface.
 */
export function instanceOfTOTPDevice(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;

    return isInstance;
}

export function TOTPDeviceFromJSON(json: any): TOTPDevice {
    return TOTPDeviceFromJSONTyped(json, false);
}

export function TOTPDeviceFromJSONTyped(json: any, ignoreDiscriminator: boolean): TOTPDevice {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'pk': json['pk'],
    };
}

export function TOTPDeviceToJSON(value?: TOTPDevice | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
