// @goauthentik/api/src/models/TransactionApplicationResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Transactional creation response
 * @export
 * @interface TransactionApplicationResponse
 */
export interface TransactionApplicationResponse {
    /**
     * 
     * @type {boolean}
     * @memberof TransactionApplicationResponse
     */
    applied: boolean;
    /**
     * 
     * @type {Array<string>}
     * @memberof TransactionApplicationResponse
     */
    logs: Array<string>;
}

/**
 * Check if a given object implements the TransactionApplicationResponse interface.
 */
export function instanceOfTransactionApplicationResponse(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "applied" in value;
    isInstance = isInstance && "logs" in value;

    return isInstance;
}

export function TransactionApplicationResponseFromJSON(json: any): TransactionApplicationResponse {
    return TransactionApplicationResponseFromJSONTyped(json, false);
}

export function TransactionApplicationResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): TransactionApplicationResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'applied': json['applied'],
        'logs': json['logs'],
    };
}

export function TransactionApplicationResponseToJSON(value?: TransactionApplicationResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'applied': value.applied,
        'logs': value.logs,
    };
}
