// @goauthentik/api/src/models/TypeCreate.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Types of an object that can be created
 * @export
 * @interface TypeCreate
 */
export interface TypeCreate {
    /**
     * 
     * @type {string}
     * @memberof TypeCreate
     */
    name: string;
    /**
     * 
     * @type {string}
     * @memberof TypeCreate
     */
    description: string;
    /**
     * 
     * @type {string}
     * @memberof TypeCreate
     */
    component: string;
    /**
     * 
     * @type {string}
     * @memberof TypeCreate
     */
    modelName: string;
    /**
     * 
     * @type {boolean}
     * @memberof TypeCreate
     */
    requiresEnterprise?: boolean;
}

/**
 * Check if a given object implements the TypeCreate interface.
 */
export function instanceOfTypeCreate(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "modelName" in value;

    return isInstance;
}

export function TypeCreateFromJSON(json: any): TypeCreate {
    return TypeCreateFromJSONTyped(json, false);
}

export function TypeCreateFromJSONTyped(json: any, ignoreDiscriminator: boolean): TypeCreate {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'description': json['description'],
        'component': json['component'],
        'modelName': json['model_name'],
        'requiresEnterprise': !exists(json, 'requires_enterprise') ? undefined : json['requires_enterprise'],
    };
}

export function TypeCreateToJSON(value?: TypeCreate | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'description': value.description,
        'component': value.component,
        'model_name': value.modelName,
        'requires_enterprise': value.requiresEnterprise,
    };
}
