// @goauthentik/api/src/models/UsedByActionEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `cascade` - CASCADE
 * * `cascade_many` - CASCADE_MANY
 * * `set_null` - SET_NULL
 * * `set_default` - SET_DEFAULT
 * @export
 */
export const UsedByActionEnum = {
    Cascade: 'cascade',
    CascadeMany: 'cascade_many',
    SetNull: 'set_null',
    SetDefault: 'set_default',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type UsedByActionEnum = typeof UsedByActionEnum[keyof typeof UsedByActionEnum];


export function UsedByActionEnumFromJSON(json: any): UsedByActionEnum {
    return UsedByActionEnumFromJSONTyped(json, false);
}

export function UsedByActionEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): UsedByActionEnum {
    return json as UsedByActionEnum;
}

export function UsedByActionEnumToJSON(value?: UsedByActionEnum | null): any {
    return value as any;
}
