// @goauthentik/api/src/models/UsedBy.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { UsedByActionEnum } from './UsedByActionEnum';
import {
    UsedByActionEnumFromJSON,
    UsedByActionEnumFromJSONTyped,
    UsedByActionEnumToJSON,
} from './UsedByActionEnum';

/**
 * A list of all objects referencing the queried object
 * @export
 * @interface UsedBy
 */
export interface UsedBy {
    /**
     * 
     * @type {string}
     * @memberof UsedBy
     */
    app: string;
    /**
     * 
     * @type {string}
     * @memberof UsedBy
     */
    modelName: string;
    /**
     * 
     * @type {string}
     * @memberof UsedBy
     */
    pk: string;
    /**
     * 
     * @type {string}
     * @memberof UsedBy
     */
    name: string;
    /**
     * 
     * @type {UsedByActionEnum}
     * @memberof UsedBy
     */
    action: UsedByActionEnum;
}

/**
 * Check if a given object implements the UsedBy interface.
 */
export function instanceOfUsedBy(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    isInstance = isInstance && "modelName" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "action" in value;

    return isInstance;
}

export function UsedByFromJSON(json: any): UsedBy {
    return UsedByFromJSONTyped(json, false);
}

export function UsedByFromJSONTyped(json: any, ignoreDiscriminator: boolean): UsedBy {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'app': json['app'],
        'modelName': json['model_name'],
        'pk': json['pk'],
        'name': json['name'],
        'action': UsedByActionEnumFromJSON(json['action']),
    };
}

export function UsedByToJSON(value?: UsedBy | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'app': value.app,
        'model_name': value.modelName,
        'pk': value.pk,
        'name': value.name,
        'action': UsedByActionEnumToJSON(value.action),
    };
}
