// @goauthentik/api/src/models/UserWriteStageRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSetRequest } from './FlowSetRequest';
import {
    FlowSetRequestFromJSON,
    FlowSetRequestFromJSONTyped,
    FlowSetRequestToJSON,
} from './FlowSetRequest';
import type { UserCreationModeEnum } from './UserCreationModeEnum';
import {
    UserCreationModeEnumFromJSON,
    UserCreationModeEnumFromJSONTyped,
    UserCreationModeEnumToJSON,
} from './UserCreationModeEnum';
import type { UserTypeEnum } from './UserTypeEnum';
import {
    UserTypeEnumFromJSON,
    UserTypeEnumFromJSONTyped,
    UserTypeEnumToJSON,
} from './UserTypeEnum';

/**
 * UserWriteStage Serializer
 * @export
 * @interface UserWriteStageRequest
 */
export interface UserWriteStageRequest {
    /**
     * 
     * @type {string}
     * @memberof UserWriteStageRequest
     */
    name: string;
    /**
     * 
     * @type {Array<FlowSetRequest>}
     * @memberof UserWriteStageRequest
     */
    flowSet?: Array<FlowSetRequest>;
    /**
     * 
     * @type {UserCreationModeEnum}
     * @memberof UserWriteStageRequest
     */
    userCreationMode?: UserCreationModeEnum;
    /**
     * When set, newly created users are inactive and cannot login.
     * @type {boolean}
     * @memberof UserWriteStageRequest
     */
    createUsersAsInactive?: boolean;
    /**
     * Optionally add newly created users to this group.
     * @type {string}
     * @memberof UserWriteStageRequest
     */
    createUsersGroup?: string | null;
    /**
     * 
     * @type {UserTypeEnum}
     * @memberof UserWriteStageRequest
     */
    userType?: UserTypeEnum;
    /**
     * 
     * @type {string}
     * @memberof UserWriteStageRequest
     */
    userPathTemplate?: string;
}

/**
 * Check if a given object implements the UserWriteStageRequest interface.
 */
export function instanceOfUserWriteStageRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function UserWriteStageRequestFromJSON(json: any): UserWriteStageRequest {
    return UserWriteStageRequestFromJSONTyped(json, false);
}

export function UserWriteStageRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserWriteStageRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetRequestFromJSON)),
        'userCreationMode': !exists(json, 'user_creation_mode') ? undefined : UserCreationModeEnumFromJSON(json['user_creation_mode']),
        'createUsersAsInactive': !exists(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !exists(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !exists(json, 'user_type') ? undefined : UserTypeEnumFromJSON(json['user_type']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}

export function UserWriteStageRequestToJSON(value?: UserWriteStageRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetRequestToJSON)),
        'user_creation_mode': UserCreationModeEnumToJSON(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': UserTypeEnumToJSON(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
