// @goauthentik/api/src/models/UserVerificationEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `required` - Required
 * * `preferred` - Preferred
 * * `discouraged` - Discouraged
 * @export
 */
export const UserVerificationEnum = {
    Required: 'required',
    Preferred: 'preferred',
    Discouraged: 'discouraged',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type UserVerificationEnum = typeof UserVerificationEnum[keyof typeof UserVerificationEnum];


export function UserVerificationEnumFromJSON(json: any): UserVerificationEnum {
    return UserVerificationEnumFromJSONTyped(json, false);
}

export function UserVerificationEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserVerificationEnum {
    return json as UserVerificationEnum;
}

export function UserVerificationEnumToJSON(value?: UserVerificationEnum | null): any {
    return value as any;
}
