// @goauthentik/api/src/models/UserSelfGroups.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface UserSelfGroups
 */
export interface UserSelfGroups {
    /**
     * 
     * @type {string}
     * @memberof UserSelfGroups
     */
    readonly name: string;
    /**
     * 
     * @type {string}
     * @memberof UserSelfGroups
     */
    readonly pk: string;
}

/**
 * Check if a given object implements the UserSelfGroups interface.
 */
export function instanceOfUserSelfGroups(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;

    return isInstance;
}

export function UserSelfGroupsFromJSON(json: any): UserSelfGroups {
    return UserSelfGroupsFromJSONTyped(json, false);
}

export function UserSelfGroupsFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserSelfGroups {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'pk': json['pk'],
    };
}

export function UserSelfGroupsToJSON(value?: UserSelfGroups | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
