// @goauthentik/api/src/models/UserLoginChallengeResponseRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * User login challenge
 * @export
 * @interface UserLoginChallengeResponseRequest
 */
export interface UserLoginChallengeResponseRequest {
    /**
     * 
     * @type {string}
     * @memberof UserLoginChallengeResponseRequest
     */
    component?: string;
    /**
     * 
     * @type {boolean}
     * @memberof UserLoginChallengeResponseRequest
     */
    rememberMe: boolean;
}

/**
 * Check if a given object implements the UserLoginChallengeResponseRequest interface.
 */
export function instanceOfUserLoginChallengeResponseRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "rememberMe" in value;

    return isInstance;
}

export function UserLoginChallengeResponseRequestFromJSON(json: any): UserLoginChallengeResponseRequest {
    return UserLoginChallengeResponseRequestFromJSONTyped(json, false);
}

export function UserLoginChallengeResponseRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserLoginChallengeResponseRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'component': !exists(json, 'component') ? undefined : json['component'],
        'rememberMe': json['remember_me'],
    };
}

export function UserLoginChallengeResponseRequestToJSON(value?: UserLoginChallengeResponseRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'component': value.component,
        'remember_me': value.rememberMe,
    };
}
