// @goauthentik/api/src/models/UserLoginStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';
import type { GeoipBindingEnum } from './GeoipBindingEnum';
import {
    GeoipBindingEnumFromJSON,
    GeoipBindingEnumFromJSONTyped,
    GeoipBindingEnumToJSON,
} from './GeoipBindingEnum';
import type { NetworkBindingEnum } from './NetworkBindingEnum';
import {
    NetworkBindingEnumFromJSON,
    NetworkBindingEnumFromJSONTyped,
    NetworkBindingEnumToJSON,
} from './NetworkBindingEnum';

/**
 * UserLoginStage Serializer
 * @export
 * @interface UserLoginStage
 */
export interface UserLoginStage {
    /**
     * 
     * @type {string}
     * @memberof UserLoginStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof UserLoginStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof UserLoginStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof UserLoginStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof UserLoginStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof UserLoginStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof UserLoginStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * Determines how long a session lasts. Default of 0 means that the sessions lasts until the browser is closed. (Format: hours=-1;minutes=-2;seconds=-3)
     * @type {string}
     * @memberof UserLoginStage
     */
    sessionDuration?: string;
    /**
     * Terminate all other sessions of the user logging in.
     * @type {boolean}
     * @memberof UserLoginStage
     */
    terminateOtherSessions?: boolean;
    /**
     * Offset the session will be extended by when the user picks the remember me option. Default of 0 means that the remember me option will not be shown. (Format: hours=-1;minutes=-2;seconds=-3)
     * @type {string}
     * @memberof UserLoginStage
     */
    rememberMeOffset?: string;
    /**
     * 
     * @type {NetworkBindingEnum}
     * @memberof UserLoginStage
     */
    networkBinding?: NetworkBindingEnum;
    /**
     * 
     * @type {GeoipBindingEnum}
     * @memberof UserLoginStage
     */
    geoipBinding?: GeoipBindingEnum;
}

/**
 * Check if a given object implements the UserLoginStage interface.
 */
export function instanceOfUserLoginStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function UserLoginStageFromJSON(json: any): UserLoginStage {
    return UserLoginStageFromJSONTyped(json, false);
}

export function UserLoginStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserLoginStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'sessionDuration': !exists(json, 'session_duration') ? undefined : json['session_duration'],
        'terminateOtherSessions': !exists(json, 'terminate_other_sessions') ? undefined : json['terminate_other_sessions'],
        'rememberMeOffset': !exists(json, 'remember_me_offset') ? undefined : json['remember_me_offset'],
        'networkBinding': !exists(json, 'network_binding') ? undefined : NetworkBindingEnumFromJSON(json['network_binding']),
        'geoipBinding': !exists(json, 'geoip_binding') ? undefined : GeoipBindingEnumFromJSON(json['geoip_binding']),
    };
}

export function UserLoginStageToJSON(value?: UserLoginStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'session_duration': value.sessionDuration,
        'terminate_other_sessions': value.terminateOtherSessions,
        'remember_me_offset': value.rememberMeOffset,
        'network_binding': NetworkBindingEnumToJSON(value.networkBinding),
        'geoip_binding': GeoipBindingEnumToJSON(value.geoipBinding),
    };
}
