// @goauthentik/api/src/models/UserGroupRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Simplified Group Serializer for user's groups
 * @export
 * @interface UserGroupRequest
 */
export interface UserGroupRequest {
    /**
     * 
     * @type {string}
     * @memberof UserGroupRequest
     */
    name: string;
    /**
     * Users added to this group will be superusers.
     * @type {boolean}
     * @memberof UserGroupRequest
     */
    isSuperuser?: boolean;
    /**
     * 
     * @type {string}
     * @memberof UserGroupRequest
     */
    parent?: string | null;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof UserGroupRequest
     */
    attributes?: { [key: string]: any; };
}

/**
 * Check if a given object implements the UserGroupRequest interface.
 */
export function instanceOfUserGroupRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function UserGroupRequestFromJSON(json: any): UserGroupRequest {
    return UserGroupRequestFromJSONTyped(json, false);
}

export function UserGroupRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserGroupRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
        'isSuperuser': !exists(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !exists(json, 'parent') ? undefined : json['parent'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
    };
}

export function UserGroupRequestToJSON(value?: UserGroupRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'attributes': value.attributes,
    };
}
