// @goauthentik/api/src/models/UserMatchingModeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `identifier` - Use the source-specific identifier
 * * `email_link` - Link to a user with identical email address. Can have security implications when a source doesn't validate email addresses.
 * * `email_deny` - Use the user's email address, but deny enrollment when the email address already exists.
 * * `username_link` - Link to a user with identical username. Can have security implications when a username is used with another source.
 * * `username_deny` - Use the user's username, but deny enrollment when the username already exists.
 * @export
 */
export const UserMatchingModeEnum = {
    Identifier: 'identifier',
    EmailLink: 'email_link',
    EmailDeny: 'email_deny',
    UsernameLink: 'username_link',
    UsernameDeny: 'username_deny',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type UserMatchingModeEnum = typeof UserMatchingModeEnum[keyof typeof UserMatchingModeEnum];


export function UserMatchingModeEnumFromJSON(json: any): UserMatchingModeEnum {
    return UserMatchingModeEnumFromJSONTyped(json, false);
}

export function UserMatchingModeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserMatchingModeEnum {
    return json as UserMatchingModeEnum;
}

export function UserMatchingModeEnumToJSON(value?: UserMatchingModeEnum | null): any {
    return value as any;
}
