// @goauthentik/api/src/models/UserPath.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface UserPath
 */
export interface UserPath {
    /**
     * 
     * @type {Array<string>}
     * @memberof UserPath
     */
    readonly paths: Array<string>;
}

/**
 * Check if a given object implements the UserPath interface.
 */
export function instanceOfUserPath(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "paths" in value;

    return isInstance;
}

export function UserPathFromJSON(json: any): UserPath {
    return UserPathFromJSONTyped(json, false);
}

export function UserPathFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserPath {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'paths': json['paths'],
    };
}

export function UserPathToJSON(value?: UserPath | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
