// @goauthentik/api/src/models/UserTypeEnum.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


/**
 * * `internal` - Internal
 * * `external` - External
 * * `service_account` - Service Account
 * * `internal_service_account` - Internal Service Account
 * @export
 */
export const UserTypeEnum = {
    Internal: 'internal',
    External: 'external',
    ServiceAccount: 'service_account',
    InternalServiceAccount: 'internal_service_account',
    UnknownDefaultOpenApi: '11184809'
} as const;
export type UserTypeEnum = typeof UserTypeEnum[keyof typeof UserTypeEnum];


export function UserTypeEnumFromJSON(json: any): UserTypeEnum {
    return UserTypeEnumFromJSONTyped(json, false);
}

export function UserTypeEnumFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserTypeEnum {
    return json as UserTypeEnum;
}

export function UserTypeEnumToJSON(value?: UserTypeEnum | null): any {
    return value as any;
}
