// @goauthentik/api/src/models/User.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { UserGroup } from './UserGroup';
import {
    UserGroupFromJSON,
    UserGroupFromJSONTyped,
    UserGroupToJSON,
} from './UserGroup';
import type { UserTypeEnum } from './UserTypeEnum';
import {
    UserTypeEnumFromJSON,
    UserTypeEnumFromJSONTyped,
    UserTypeEnumToJSON,
} from './UserTypeEnum';

/**
 * User Serializer
 * @export
 * @interface User
 */
export interface User {
    /**
     * 
     * @type {number}
     * @memberof User
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof User
     */
    username: string;
    /**
     * User's display name.
     * @type {string}
     * @memberof User
     */
    name: string;
    /**
     * Designates whether this user should be treated as active. Unselect this instead of deleting accounts.
     * @type {boolean}
     * @memberof User
     */
    isActive?: boolean;
    /**
     * 
     * @type {Date}
     * @memberof User
     */
    lastLogin?: Date | null;
    /**
     * 
     * @type {boolean}
     * @memberof User
     */
    readonly isSuperuser: boolean;
    /**
     * 
     * @type {Array<string>}
     * @memberof User
     */
    groups?: Array<string>;
    /**
     * 
     * @type {Array<UserGroup>}
     * @memberof User
     */
    readonly groupsObj: Array<UserGroup>;
    /**
     * 
     * @type {string}
     * @memberof User
     */
    email?: string;
    /**
     * User's avatar, either a http/https URL or a data URI
     * @type {string}
     * @memberof User
     */
    readonly avatar: string;
    /**
     * 
     * @type {{ [key: string]: any; }}
     * @memberof User
     */
    attributes?: { [key: string]: any; };
    /**
     * 
     * @type {string}
     * @memberof User
     */
    readonly uid: string;
    /**
     * 
     * @type {string}
     * @memberof User
     */
    path?: string;
    /**
     * 
     * @type {UserTypeEnum}
     * @memberof User
     */
    type?: UserTypeEnum;
    /**
     * 
     * @type {string}
     * @memberof User
     */
    readonly uuid: string;
}

/**
 * Check if a given object implements the User interface.
 */
export function instanceOfUser(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isSuperuser" in value;
    isInstance = isInstance && "groupsObj" in value;
    isInstance = isInstance && "avatar" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "uuid" in value;

    return isInstance;
}

export function UserFromJSON(json: any): User {
    return UserFromJSONTyped(json, false);
}

export function UserFromJSONTyped(json: any, ignoreDiscriminator: boolean): User {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !exists(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !exists(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'isSuperuser': json['is_superuser'],
        'groups': !exists(json, 'groups') ? undefined : json['groups'],
        'groupsObj': ((json['groups_obj'] as Array<any>).map(UserGroupFromJSON)),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'avatar': json['avatar'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
        'path': !exists(json, 'path') ? undefined : json['path'],
        'type': !exists(json, 'type') ? undefined : UserTypeEnumFromJSON(json['type']),
        'uuid': json['uuid'],
    };
}

export function UserToJSON(value?: User | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'groups': value.groups,
        'email': value.email,
        'attributes': value.attributes,
        'path': value.path,
        'type': UserTypeEnumToJSON(value.type),
    };
}
