// @goauthentik/api/src/models/UserSourceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Source } from './Source';
import {
    SourceFromJSON,
    SourceFromJSONTyped,
    SourceToJSON,
} from './Source';

/**
 * OAuth Source Serializer
 * @export
 * @interface UserSourceConnection
 */
export interface UserSourceConnection {
    /**
     * 
     * @type {number}
     * @memberof UserSourceConnection
     */
    readonly pk: number;
    /**
     * 
     * @type {number}
     * @memberof UserSourceConnection
     */
    readonly user: number;
    /**
     * 
     * @type {Source}
     * @memberof UserSourceConnection
     */
    readonly source: Source;
    /**
     * 
     * @type {Date}
     * @memberof UserSourceConnection
     */
    readonly created: Date;
}

/**
 * Check if a given object implements the UserSourceConnection interface.
 */
export function instanceOfUserSourceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "created" in value;

    return isInstance;
}

export function UserSourceConnectionFromJSON(json: any): UserSourceConnection {
    return UserSourceConnectionFromJSONTyped(json, false);
}

export function UserSourceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserSourceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'created': (new Date(json['created'])),
    };
}

export function UserSourceConnectionToJSON(value?: UserSourceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
    };
}
