// @goauthentik/api/src/models/UserSAMLSourceConnectionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * SAML Source Serializer
 * @export
 * @interface UserSAMLSourceConnectionRequest
 */
export interface UserSAMLSourceConnectionRequest {
    /**
     * 
     * @type {number}
     * @memberof UserSAMLSourceConnectionRequest
     */
    user: number;
    /**
     * 
     * @type {string}
     * @memberof UserSAMLSourceConnectionRequest
     */
    identifier: string;
}

/**
 * Check if a given object implements the UserSAMLSourceConnectionRequest interface.
 */
export function instanceOfUserSAMLSourceConnectionRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "identifier" in value;

    return isInstance;
}

export function UserSAMLSourceConnectionRequestFromJSON(json: any): UserSAMLSourceConnectionRequest {
    return UserSAMLSourceConnectionRequestFromJSONTyped(json, false);
}

export function UserSAMLSourceConnectionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserSAMLSourceConnectionRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'user': json['user'],
        'identifier': json['identifier'],
    };
}

export function UserSAMLSourceConnectionRequestToJSON(value?: UserSAMLSourceConnectionRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'identifier': value.identifier,
    };
}
