// @goauthentik/api/src/models/UserWriteStage.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { FlowSet } from './FlowSet';
import {
    FlowSetFromJSON,
    FlowSetFromJSONTyped,
    FlowSetToJSON,
} from './FlowSet';
import type { UserCreationModeEnum } from './UserCreationModeEnum';
import {
    UserCreationModeEnumFromJSON,
    UserCreationModeEnumFromJSONTyped,
    UserCreationModeEnumToJSON,
} from './UserCreationModeEnum';
import type { UserTypeEnum } from './UserTypeEnum';
import {
    UserTypeEnumFromJSON,
    UserTypeEnumFromJSONTyped,
    UserTypeEnumToJSON,
} from './UserTypeEnum';

/**
 * UserWriteStage Serializer
 * @export
 * @interface UserWriteStage
 */
export interface UserWriteStage {
    /**
     * 
     * @type {string}
     * @memberof UserWriteStage
     */
    readonly pk: string;
    /**
     * 
     * @type {string}
     * @memberof UserWriteStage
     */
    name: string;
    /**
     * Get object type so that we know how to edit the object
     * @type {string}
     * @memberof UserWriteStage
     */
    readonly component: string;
    /**
     * Return object's verbose_name
     * @type {string}
     * @memberof UserWriteStage
     */
    readonly verboseName: string;
    /**
     * Return object's plural verbose_name
     * @type {string}
     * @memberof UserWriteStage
     */
    readonly verboseNamePlural: string;
    /**
     * Return internal model name
     * @type {string}
     * @memberof UserWriteStage
     */
    readonly metaModelName: string;
    /**
     * 
     * @type {Array<FlowSet>}
     * @memberof UserWriteStage
     */
    flowSet?: Array<FlowSet>;
    /**
     * 
     * @type {UserCreationModeEnum}
     * @memberof UserWriteStage
     */
    userCreationMode?: UserCreationModeEnum;
    /**
     * When set, newly created users are inactive and cannot login.
     * @type {boolean}
     * @memberof UserWriteStage
     */
    createUsersAsInactive?: boolean;
    /**
     * Optionally add newly created users to this group.
     * @type {string}
     * @memberof UserWriteStage
     */
    createUsersGroup?: string | null;
    /**
     * 
     * @type {UserTypeEnum}
     * @memberof UserWriteStage
     */
    userType?: UserTypeEnum;
    /**
     * 
     * @type {string}
     * @memberof UserWriteStage
     */
    userPathTemplate?: string;
}

/**
 * Check if a given object implements the UserWriteStage interface.
 */
export function instanceOfUserWriteStage(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;

    return isInstance;
}

export function UserWriteStageFromJSON(json: any): UserWriteStage {
    return UserWriteStageFromJSONTyped(json, false);
}

export function UserWriteStageFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserWriteStage {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : ((json['flow_set'] as Array<any>).map(FlowSetFromJSON)),
        'userCreationMode': !exists(json, 'user_creation_mode') ? undefined : UserCreationModeEnumFromJSON(json['user_creation_mode']),
        'createUsersAsInactive': !exists(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !exists(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !exists(json, 'user_type') ? undefined : UserTypeEnumFromJSON(json['user_type']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}

export function UserWriteStageToJSON(value?: UserWriteStage | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : ((value.flowSet as Array<any>).map(FlowSetToJSON)),
        'user_creation_mode': UserCreationModeEnumToJSON(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': UserTypeEnumToJSON(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
