// @goauthentik/api/src/models/UserAccountRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Account adding/removing operations
 * @export
 * @interface UserAccountRequest
 */
export interface UserAccountRequest {
    /**
     * 
     * @type {number}
     * @memberof UserAccountRequest
     */
    pk: number;
}

/**
 * Check if a given object implements the UserAccountRequest interface.
 */
export function instanceOfUserAccountRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;

    return isInstance;
}

export function UserAccountRequestFromJSON(json: any): UserAccountRequest {
    return UserAccountRequestFromJSONTyped(json, false);
}

export function UserAccountRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserAccountRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
    };
}

export function UserAccountRequestToJSON(value?: UserAccountRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'pk': value.pk,
    };
}
