// @goauthentik/api/src/models/UserServiceAccountResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface UserServiceAccountResponse
 */
export interface UserServiceAccountResponse {
    /**
     * 
     * @type {string}
     * @memberof UserServiceAccountResponse
     */
    username: string;
    /**
     * 
     * @type {string}
     * @memberof UserServiceAccountResponse
     */
    token: string;
    /**
     * 
     * @type {string}
     * @memberof UserServiceAccountResponse
     */
    userUid: string;
    /**
     * 
     * @type {number}
     * @memberof UserServiceAccountResponse
     */
    userPk: number;
    /**
     * 
     * @type {string}
     * @memberof UserServiceAccountResponse
     */
    groupPk?: string;
}

/**
 * Check if a given object implements the UserServiceAccountResponse interface.
 */
export function instanceOfUserServiceAccountResponse(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "token" in value;
    isInstance = isInstance && "userUid" in value;
    isInstance = isInstance && "userPk" in value;

    return isInstance;
}

export function UserServiceAccountResponseFromJSON(json: any): UserServiceAccountResponse {
    return UserServiceAccountResponseFromJSONTyped(json, false);
}

export function UserServiceAccountResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserServiceAccountResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'username': json['username'],
        'token': json['token'],
        'userUid': json['user_uid'],
        'userPk': json['user_pk'],
        'groupPk': !exists(json, 'group_pk') ? undefined : json['group_pk'],
    };
}

export function UserServiceAccountResponseToJSON(value?: UserServiceAccountResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'username': value.username,
        'token': value.token,
        'user_uid': value.userUid,
        'user_pk': value.userPk,
        'group_pk': value.groupPk,
    };
}
