// @goauthentik/api/src/models/UserPasswordSetRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface UserPasswordSetRequest
 */
export interface UserPasswordSetRequest {
    /**
     * 
     * @type {string}
     * @memberof UserPasswordSetRequest
     */
    password: string;
}

/**
 * Check if a given object implements the UserPasswordSetRequest interface.
 */
export function instanceOfUserPasswordSetRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "password" in value;

    return isInstance;
}

export function UserPasswordSetRequestFromJSON(json: any): UserPasswordSetRequest {
    return UserPasswordSetRequestFromJSONTyped(json, false);
}

export function UserPasswordSetRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserPasswordSetRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'password': json['password'],
    };
}

export function UserPasswordSetRequestToJSON(value?: UserPasswordSetRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'password': value.password,
    };
}
