// @goauthentik/api/src/models/UserOAuthSourceConnection.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import type { Source } from './Source';
import {
    SourceFromJSON,
    SourceFromJSONTyped,
    SourceToJSON,
} from './Source';

/**
 * OAuth Source Serializer
 * @export
 * @interface UserOAuthSourceConnection
 */
export interface UserOAuthSourceConnection {
    /**
     * 
     * @type {number}
     * @memberof UserOAuthSourceConnection
     */
    readonly pk: number;
    /**
     * 
     * @type {number}
     * @memberof UserOAuthSourceConnection
     */
    user: number;
    /**
     * 
     * @type {Source}
     * @memberof UserOAuthSourceConnection
     */
    readonly source: Source;
    /**
     * 
     * @type {string}
     * @memberof UserOAuthSourceConnection
     */
    identifier: string;
}

/**
 * Check if a given object implements the UserOAuthSourceConnection interface.
 */
export function instanceOfUserOAuthSourceConnection(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "identifier" in value;

    return isInstance;
}

export function UserOAuthSourceConnectionFromJSON(json: any): UserOAuthSourceConnection {
    return UserOAuthSourceConnectionFromJSONTyped(json, false);
}

export function UserOAuthSourceConnectionFromJSONTyped(json: any, ignoreDiscriminator: boolean): UserOAuthSourceConnection {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'identifier': json['identifier'],
    };
}

export function UserOAuthSourceConnectionToJSON(value?: UserOAuthSourceConnection | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'user': value.user,
        'identifier': value.identifier,
    };
}
