// @goauthentik/api/src/models/ValidationError.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Validation Error
 * @export
 * @interface ValidationError
 */
export interface ValidationError {
    [key: string]: any | any;
    /**
     * 
     * @type {Array<string>}
     * @memberof ValidationError
     */
    nonFieldErrors?: Array<string>;
    /**
     * 
     * @type {string}
     * @memberof ValidationError
     */
    code?: string;
}

/**
 * Check if a given object implements the ValidationError interface.
 */
export function instanceOfValidationError(value: object): boolean {
    let isInstance = true;

    return isInstance;
}

export function ValidationErrorFromJSON(json: any): ValidationError {
    return ValidationErrorFromJSONTyped(json, false);
}

export function ValidationErrorFromJSONTyped(json: any, ignoreDiscriminator: boolean): ValidationError {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
            ...json,
        'nonFieldErrors': !exists(json, 'non_field_errors') ? undefined : json['non_field_errors'],
        'code': !exists(json, 'code') ? undefined : json['code'],
    };
}

export function ValidationErrorToJSON(value?: ValidationError | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
            ...value,
        'non_field_errors': value.nonFieldErrors,
        'code': value.code,
    };
}
