// @goauthentik/api/src/models/WebAuthnDeviceRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for WebAuthn authenticator devices
 * @export
 * @interface WebAuthnDeviceRequest
 */
export interface WebAuthnDeviceRequest {
    /**
     * 
     * @type {string}
     * @memberof WebAuthnDeviceRequest
     */
    name: string;
}

/**
 * Check if a given object implements the WebAuthnDeviceRequest interface.
 */
export function instanceOfWebAuthnDeviceRequest(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "name" in value;

    return isInstance;
}

export function WebAuthnDeviceRequestFromJSON(json: any): WebAuthnDeviceRequest {
    return WebAuthnDeviceRequestFromJSONTyped(json, false);
}

export function WebAuthnDeviceRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): WebAuthnDeviceRequest {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'name': json['name'],
    };
}

export function WebAuthnDeviceRequestToJSON(value?: WebAuthnDeviceRequest | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
