// @goauthentik/api/src/models/WebAuthnDevice.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Serializer for WebAuthn authenticator devices
 * @export
 * @interface WebAuthnDevice
 */
export interface WebAuthnDevice {
    /**
     * 
     * @type {number}
     * @memberof WebAuthnDevice
     */
    readonly pk: number;
    /**
     * 
     * @type {string}
     * @memberof WebAuthnDevice
     */
    name: string;
    /**
     * 
     * @type {Date}
     * @memberof WebAuthnDevice
     */
    readonly createdOn: Date;
}

/**
 * Check if a given object implements the WebAuthnDevice interface.
 */
export function instanceOfWebAuthnDevice(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdOn" in value;

    return isInstance;
}

export function WebAuthnDeviceFromJSON(json: any): WebAuthnDevice {
    return WebAuthnDeviceFromJSONTyped(json, false);
}

export function WebAuthnDeviceFromJSONTyped(json: any, ignoreDiscriminator: boolean): WebAuthnDevice {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pk': json['pk'],
        'name': json['name'],
        'createdOn': (new Date(json['created_on'])),
    };
}

export function WebAuthnDeviceToJSON(value?: WebAuthnDevice | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'name': value.name,
    };
}
