// @goauthentik/api/src/models/Workers.ts
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface Workers
 */
export interface Workers {
    /**
     * 
     * @type {number}
     * @memberof Workers
     */
    count: number;
}

/**
 * Check if a given object implements the Workers interface.
 */
export function instanceOfWorkers(value: object): boolean {
    let isInstance = true;
    isInstance = isInstance && "count" in value;

    return isInstance;
}

export function WorkersFromJSON(json: any): Workers {
    return WorkersFromJSONTyped(json, false);
}

export function WorkersFromJSONTyped(json: any, ignoreDiscriminator: boolean): Workers {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'count': json['count'],
    };
}

export function WorkersToJSON(value?: Workers | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'count': value.count,
    };
}
