// @goauthentik/api/dist/esm/models/AppleLoginChallenge.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ChallengeChoicesFromJSON, ChallengeChoicesToJSON, } from './ChallengeChoices';
import { ContextualFlowInfoFromJSON, ContextualFlowInfoToJSON, } from './ContextualFlowInfo';
/**
 * Check if a given object implements the AppleLoginChallenge interface.
 */
export function instanceOfAppleLoginChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "scope" in value;
    isInstance = isInstance && "redirectUri" in value;
    isInstance = isInstance && "state" in value;
    return isInstance;
}
export function AppleLoginChallengeFromJSON(json) {
    return AppleLoginChallengeFromJSONTyped(json, false);
}
export function AppleLoginChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'clientId': json['client_id'],
        'scope': json['scope'],
        'redirectUri': json['redirect_uri'],
        'state': json['state'],
    };
}
export function AppleLoginChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'client_id': value.clientId,
        'scope': value.scope,
        'redirect_uri': value.redirectUri,
        'state': value.state,
    };
}
