// @goauthentik/api/dist/esm/models/AuthenticatorSMSChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the AuthenticatorSMSChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorSMSChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function AuthenticatorSMSChallengeResponseRequestFromJSON(json) {
    return AuthenticatorSMSChallengeResponseRequestFromJSONTyped(json, false);
}
export function AuthenticatorSMSChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'code': !exists(json, 'code') ? undefined : json['code'],
        'phoneNumber': !exists(json, 'phone_number') ? undefined : json['phone_number'],
    };
}
export function AuthenticatorSMSChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'code': value.code,
        'phone_number': value.phoneNumber,
    };
}
