// @goauthentik/api/dist/esm/models/AuthenticateWebAuthnStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { AuthenticatorAttachmentEnumFromJSON, AuthenticatorAttachmentEnumToJSON, } from './AuthenticatorAttachmentEnum';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
import { ResidentKeyRequirementEnumFromJSON, ResidentKeyRequirementEnumToJSON, } from './ResidentKeyRequirementEnum';
import { UserVerificationEnumFromJSON, UserVerificationEnumToJSON, } from './UserVerificationEnum';
/**
 * Check if a given object implements the AuthenticateWebAuthnStage interface.
 */
export function instanceOfAuthenticateWebAuthnStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function AuthenticateWebAuthnStageFromJSON(json) {
    return AuthenticateWebAuthnStageFromJSONTyped(json, false);
}
export function AuthenticateWebAuthnStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'userVerification': !exists(json, 'user_verification') ? undefined : UserVerificationEnumFromJSON(json['user_verification']),
        'authenticatorAttachment': !exists(json, 'authenticator_attachment') ? undefined : AuthenticatorAttachmentEnumFromJSON(json['authenticator_attachment']),
        'residentKeyRequirement': !exists(json, 'resident_key_requirement') ? undefined : ResidentKeyRequirementEnumFromJSON(json['resident_key_requirement']),
    };
}
export function AuthenticateWebAuthnStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'user_verification': UserVerificationEnumToJSON(value.userVerification),
        'authenticator_attachment': AuthenticatorAttachmentEnumToJSON(value.authenticatorAttachment),
        'resident_key_requirement': ResidentKeyRequirementEnumToJSON(value.residentKeyRequirement),
    };
}
