// @goauthentik/api/dist/esm/models/AuthenticatorDuoStageDeviceImportResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the AuthenticatorDuoStageDeviceImportResponse interface.
 */
export function instanceOfAuthenticatorDuoStageDeviceImportResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
export function AuthenticatorDuoStageDeviceImportResponseFromJSON(json) {
    return AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json, false);
}
export function AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
        'error': json['error'],
    };
}
export function AuthenticatorDuoStageDeviceImportResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
