// @goauthentik/api/dist/esm/models/AuthenticatedSessionUserAgent.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { AuthenticatedSessionUserAgentDeviceFromJSON, AuthenticatedSessionUserAgentDeviceToJSON, } from './AuthenticatedSessionUserAgentDevice';
import { AuthenticatedSessionUserAgentOsFromJSON, AuthenticatedSessionUserAgentOsToJSON, } from './AuthenticatedSessionUserAgentOs';
import { AuthenticatedSessionUserAgentUserAgentFromJSON, AuthenticatedSessionUserAgentUserAgentToJSON, } from './AuthenticatedSessionUserAgentUserAgent';
/**
 * Check if a given object implements the AuthenticatedSessionUserAgent interface.
 */
export function instanceOfAuthenticatedSessionUserAgent(value) {
    let isInstance = true;
    isInstance = isInstance && "device" in value;
    isInstance = isInstance && "os" in value;
    isInstance = isInstance && "userAgent" in value;
    isInstance = isInstance && "string" in value;
    return isInstance;
}
export function AuthenticatedSessionUserAgentFromJSON(json) {
    return AuthenticatedSessionUserAgentFromJSONTyped(json, false);
}
export function AuthenticatedSessionUserAgentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'device': AuthenticatedSessionUserAgentDeviceFromJSON(json['device']),
        'os': AuthenticatedSessionUserAgentOsFromJSON(json['os']),
        'userAgent': AuthenticatedSessionUserAgentUserAgentFromJSON(json['user_agent']),
        'string': json['string'],
    };
}
export function AuthenticatedSessionUserAgentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'device': AuthenticatedSessionUserAgentDeviceToJSON(value.device),
        'os': AuthenticatedSessionUserAgentOsToJSON(value.os),
        'user_agent': AuthenticatedSessionUserAgentUserAgentToJSON(value.userAgent),
        'string': value.string,
    };
}
