// @goauthentik/api/dist/esm/models/AuthenticatorStaticChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the AuthenticatorStaticChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorStaticChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function AuthenticatorStaticChallengeResponseRequestFromJSON(json) {
    return AuthenticatorStaticChallengeResponseRequestFromJSONTyped(json, false);
}
export function AuthenticatorStaticChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
    };
}
export function AuthenticatorStaticChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
    };
}
