// @goauthentik/api/dist/esm/models/AuthenticatorDuoStageManualDeviceImportRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the AuthenticatorDuoStageManualDeviceImportRequest interface.
 */
export function instanceOfAuthenticatorDuoStageManualDeviceImportRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "duoUserId" in value;
    isInstance = isInstance && "username" in value;
    return isInstance;
}
export function AuthenticatorDuoStageManualDeviceImportRequestFromJSON(json) {
    return AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json, false);
}
export function AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'duoUserId': json['duo_user_id'],
        'username': json['username'],
    };
}
export function AuthenticatorDuoStageManualDeviceImportRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'duo_user_id': value.duoUserId,
        'username': value.username,
    };
}
