// @goauthentik/api/dist/esm/models/AuthenticationEnum.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * * `none` - None
 * * `require_authenticated` - Require Authenticated
 * * `require_unauthenticated` - Require Unauthenticated
 * * `require_superuser` - Require Superuser
 * * `require_outpost` - Require Outpost
 * @export
 */
export const AuthenticationEnum = {
    None: 'none',
    RequireAuthenticated: 'require_authenticated',
    RequireUnauthenticated: 'require_unauthenticated',
    RequireSuperuser: 'require_superuser',
    RequireOutpost: 'require_outpost',
    UnknownDefaultOpenApi: '11184809'
};
export function AuthenticationEnumFromJSON(json) {
    return AuthenticationEnumFromJSONTyped(json, false);
}
export function AuthenticationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function AuthenticationEnumToJSON(value) {
    return value;
}
