// @goauthentik/api/dist/esm/models/AuthenticatorValidateStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { DeviceClassesEnumFromJSON, DeviceClassesEnumToJSON, } from './DeviceClassesEnum';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
import { NotConfiguredActionEnumFromJSON, NotConfiguredActionEnumToJSON, } from './NotConfiguredActionEnum';
import { UserVerificationEnumFromJSON, UserVerificationEnumToJSON, } from './UserVerificationEnum';
/**
 * Check if a given object implements the AuthenticatorValidateStage interface.
 */
export function instanceOfAuthenticatorValidateStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function AuthenticatorValidateStageFromJSON(json) {
    return AuthenticatorValidateStageFromJSONTyped(json, false);
}
export function AuthenticatorValidateStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'notConfiguredAction': !exists(json, 'not_configured_action') ? undefined : NotConfiguredActionEnumFromJSON(json['not_configured_action']),
        'deviceClasses': !exists(json, 'device_classes') ? undefined : (json['device_classes'].map(DeviceClassesEnumFromJSON)),
        'configurationStages': !exists(json, 'configuration_stages') ? undefined : json['configuration_stages'],
        'lastAuthThreshold': !exists(json, 'last_auth_threshold') ? undefined : json['last_auth_threshold'],
        'webauthnUserVerification': !exists(json, 'webauthn_user_verification') ? undefined : UserVerificationEnumFromJSON(json['webauthn_user_verification']),
    };
}
export function AuthenticatorValidateStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'not_configured_action': NotConfiguredActionEnumToJSON(value.notConfiguredAction),
        'device_classes': value.deviceClasses === undefined ? undefined : (value.deviceClasses.map(DeviceClassesEnumToJSON)),
        'configuration_stages': value.configurationStages,
        'last_auth_threshold': value.lastAuthThreshold,
        'webauthn_user_verification': UserVerificationEnumToJSON(value.webauthnUserVerification),
    };
}
