// @goauthentik/api/dist/esm/models/AuthenticatorWebAuthnChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the AuthenticatorWebAuthnChallengeResponseRequest interface.
 */
export function instanceOfAuthenticatorWebAuthnChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "response" in value;
    return isInstance;
}
export function AuthenticatorWebAuthnChallengeResponseRequestFromJSON(json) {
    return AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json, false);
}
export function AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'response': json['response'],
    };
}
export function AuthenticatorWebAuthnChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'response': value.response,
    };
}
