// @goauthentik/api/dist/esm/models/AuthenticatorTOTPStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { DigitsEnumFromJSON, DigitsEnumToJSON, } from './DigitsEnum';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
/**
 * Check if a given object implements the AuthenticatorTOTPStage interface.
 */
export function instanceOfAuthenticatorTOTPStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "digits" in value;
    return isInstance;
}
export function AuthenticatorTOTPStageFromJSON(json) {
    return AuthenticatorTOTPStageFromJSONTyped(json, false);
}
export function AuthenticatorTOTPStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'digits': DigitsEnumFromJSON(json['digits']),
    };
}
export function AuthenticatorTOTPStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'digits': DigitsEnumToJSON(value.digits),
    };
}
