// @goauthentik/api/dist/esm/models/AuthenticatedSessionUserAgentDevice.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the AuthenticatedSessionUserAgentDevice interface.
 */
export function instanceOfAuthenticatedSessionUserAgentDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "brand" in value;
    isInstance = isInstance && "family" in value;
    isInstance = isInstance && "model" in value;
    return isInstance;
}
export function AuthenticatedSessionUserAgentDeviceFromJSON(json) {
    return AuthenticatedSessionUserAgentDeviceFromJSONTyped(json, false);
}
export function AuthenticatedSessionUserAgentDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'brand': json['brand'],
        'family': json['family'],
        'model': json['model'],
    };
}
export function AuthenticatedSessionUserAgentDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'brand': value.brand,
        'family': value.family,
        'model': value.model,
    };
}
