// @goauthentik/api/dist/esm/models/BlueprintInstance.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { BlueprintInstanceStatusEnumFromJSON, } from './BlueprintInstanceStatusEnum';
/**
 * Check if a given object implements the BlueprintInstance interface.
 */
export function instanceOfBlueprintInstance(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "lastApplied" in value;
    isInstance = isInstance && "lastAppliedHash" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "managedModels" in value;
    isInstance = isInstance && "metadata" in value;
    return isInstance;
}
export function BlueprintInstanceFromJSON(json) {
    return BlueprintInstanceFromJSONTyped(json, false);
}
export function BlueprintInstanceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'path': !exists(json, 'path') ? undefined : json['path'],
        'context': !exists(json, 'context') ? undefined : json['context'],
        'lastApplied': (new Date(json['last_applied'])),
        'lastAppliedHash': json['last_applied_hash'],
        'status': BlueprintInstanceStatusEnumFromJSON(json['status']),
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'managedModels': json['managed_models'],
        'metadata': json['metadata'],
        'content': !exists(json, 'content') ? undefined : json['content'],
    };
}
export function BlueprintInstanceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
