// @goauthentik/api/dist/esm/models/BlueprintFile.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { MetadataFromJSON, } from './Metadata';
/**
 * Check if a given object implements the BlueprintFile interface.
 */
export function instanceOfBlueprintFile(value) {
    let isInstance = true;
    isInstance = isInstance && "path" in value;
    isInstance = isInstance && "lastM" in value;
    isInstance = isInstance && "hash" in value;
    isInstance = isInstance && "meta" in value;
    return isInstance;
}
export function BlueprintFileFromJSON(json) {
    return BlueprintFileFromJSONTyped(json, false);
}
export function BlueprintFileFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'path': json['path'],
        'lastM': (new Date(json['last_m'])),
        'hash': json['hash'],
        'meta': MetadataFromJSON(json['meta']),
    };
}
export function BlueprintFileToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'path': value.path,
        'last_m': (value.lastM.toISOString()),
        'hash': value.hash,
    };
}
