// @goauthentik/api/dist/esm/models/Cache.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Cache interface.
 */
export function instanceOfCache(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    return isInstance;
}
export function CacheFromJSON(json) {
    return CacheFromJSONTyped(json, false);
}
export function CacheFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
    };
}
export function CacheToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
