// @goauthentik/api/dist/esm/models/CaptchaStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the CaptchaStageRequest interface.
 */
export function instanceOfCaptchaStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "publicKey" in value;
    isInstance = isInstance && "privateKey" in value;
    return isInstance;
}
export function CaptchaStageRequestFromJSON(json) {
    return CaptchaStageRequestFromJSONTyped(json, false);
}
export function CaptchaStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'publicKey': json['public_key'],
        'privateKey': json['private_key'],
        'jsUrl': !exists(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !exists(json, 'api_url') ? undefined : json['api_url'],
    };
}
export function CaptchaStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'public_key': value.publicKey,
        'private_key': value.privateKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
