// @goauthentik/api/dist/esm/models/CertificateData.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the CertificateData interface.
 */
export function instanceOfCertificateData(value) {
    let isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
export function CertificateDataFromJSON(json) {
    return CertificateDataFromJSONTyped(json, false);
}
export function CertificateDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': json['data'],
    };
}
export function CertificateDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
