// @goauthentik/api/dist/esm/models/CertificateGenerationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the CertificateGenerationRequest interface.
 */
export function instanceOfCertificateGenerationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "commonName" in value;
    isInstance = isInstance && "validityDays" in value;
    return isInstance;
}
export function CertificateGenerationRequestFromJSON(json) {
    return CertificateGenerationRequestFromJSONTyped(json, false);
}
export function CertificateGenerationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'commonName': json['common_name'],
        'subjectAltName': !exists(json, 'subject_alt_name') ? undefined : json['subject_alt_name'],
        'validityDays': json['validity_days'],
    };
}
export function CertificateGenerationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'common_name': value.commonName,
        'subject_alt_name': value.subjectAltName,
        'validity_days': value.validityDays,
    };
}
