// @goauthentik/api/dist/esm/models/Config.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { CapabilitiesEnumFromJSON, CapabilitiesEnumToJSON, } from './CapabilitiesEnum';
import { ErrorReportingConfigFromJSON, ErrorReportingConfigToJSON, } from './ErrorReportingConfig';
/**
 * Check if a given object implements the Config interface.
 */
export function instanceOfConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "errorReporting" in value;
    isInstance = isInstance && "capabilities" in value;
    isInstance = isInstance && "cacheTimeout" in value;
    isInstance = isInstance && "cacheTimeoutFlows" in value;
    isInstance = isInstance && "cacheTimeoutPolicies" in value;
    isInstance = isInstance && "cacheTimeoutReputation" in value;
    return isInstance;
}
export function ConfigFromJSON(json) {
    return ConfigFromJSONTyped(json, false);
}
export function ConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorReporting': ErrorReportingConfigFromJSON(json['error_reporting']),
        'capabilities': (json['capabilities'].map(CapabilitiesEnumFromJSON)),
        'cacheTimeout': json['cache_timeout'],
        'cacheTimeoutFlows': json['cache_timeout_flows'],
        'cacheTimeoutPolicies': json['cache_timeout_policies'],
        'cacheTimeoutReputation': json['cache_timeout_reputation'],
    };
}
export function ConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_reporting': ErrorReportingConfigToJSON(value.errorReporting),
        'capabilities': (value.capabilities.map(CapabilitiesEnumToJSON)),
        'cache_timeout': value.cacheTimeout,
        'cache_timeout_flows': value.cacheTimeoutFlows,
        'cache_timeout_policies': value.cacheTimeoutPolicies,
        'cache_timeout_reputation': value.cacheTimeoutReputation,
    };
}
