// @goauthentik/api/dist/esm/models/ConnectionToken.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { EndpointFromJSON, } from './Endpoint';
import { GroupMemberFromJSON, } from './GroupMember';
import { RACProviderFromJSON, } from './RACProvider';
/**
 * Check if a given object implements the ConnectionToken interface.
 */
export function instanceOfConnectionToken(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "providerObj" in value;
    isInstance = isInstance && "endpoint" in value;
    isInstance = isInstance && "endpointObj" in value;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
export function ConnectionTokenFromJSON(json) {
    return ConnectionTokenFromJSONTyped(json, false);
}
export function ConnectionTokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'provider': json['provider'],
        'providerObj': RACProviderFromJSON(json['provider_obj']),
        'endpoint': json['endpoint'],
        'endpointObj': EndpointFromJSON(json['endpoint_obj']),
        'user': GroupMemberFromJSON(json['user']),
    };
}
export function ConnectionTokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
