// @goauthentik/api/dist/esm/models/ConnectionTokenRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the ConnectionTokenRequest interface.
 */
export function instanceOfConnectionTokenRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "provider" in value;
    return isInstance;
}
export function ConnectionTokenRequestFromJSON(json) {
    return ConnectionTokenRequestFromJSONTyped(json, false);
}
export function ConnectionTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'provider': json['provider'],
    };
}
export function ConnectionTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
