// @goauthentik/api/dist/esm/models/ConsentChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the ConsentChallengeResponseRequest interface.
 */
export function instanceOfConsentChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
export function ConsentChallengeResponseRequestFromJSON(json) {
    return ConsentChallengeResponseRequestFromJSONTyped(json, false);
}
export function ConsentChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'token': json['token'],
    };
}
export function ConsentChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'token': value.token,
    };
}
