// @goauthentik/api/dist/esm/models/ConsentStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ConsentStageModeEnumFromJSON, ConsentStageModeEnumToJSON, } from './ConsentStageModeEnum';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
/**
 * Check if a given object implements the ConsentStage interface.
 */
export function instanceOfConsentStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function ConsentStageFromJSON(json) {
    return ConsentStageFromJSONTyped(json, false);
}
export function ConsentStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'mode': !exists(json, 'mode') ? undefined : ConsentStageModeEnumFromJSON(json['mode']),
        'consentExpireIn': !exists(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}
export function ConsentStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'mode': ConsentStageModeEnumToJSON(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
