// @goauthentik/api/dist/esm/models/ConsentChallenge.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ChallengeChoicesFromJSON, ChallengeChoicesToJSON, } from './ChallengeChoices';
import { ConsentPermissionFromJSON, ConsentPermissionToJSON, } from './ConsentPermission';
import { ContextualFlowInfoFromJSON, ContextualFlowInfoToJSON, } from './ContextualFlowInfo';
/**
 * Check if a given object implements the ConsentChallenge interface.
 */
export function instanceOfConsentChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "pendingUser" in value;
    isInstance = isInstance && "pendingUserAvatar" in value;
    isInstance = isInstance && "permissions" in value;
    isInstance = isInstance && "additionalPermissions" in value;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
export function ConsentChallengeFromJSON(json) {
    return ConsentChallengeFromJSONTyped(json, false);
}
export function ConsentChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'pendingUser': json['pending_user'],
        'pendingUserAvatar': json['pending_user_avatar'],
        'headerText': !exists(json, 'header_text') ? undefined : json['header_text'],
        'permissions': (json['permissions'].map(ConsentPermissionFromJSON)),
        'additionalPermissions': (json['additional_permissions'].map(ConsentPermissionFromJSON)),
        'token': json['token'],
    };
}
export function ConsentChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'pending_user': value.pendingUser,
        'pending_user_avatar': value.pendingUserAvatar,
        'header_text': value.headerText,
        'permissions': (value.permissions.map(ConsentPermissionToJSON)),
        'additional_permissions': (value.additionalPermissions.map(ConsentPermissionToJSON)),
        'token': value.token,
    };
}
