// @goauthentik/api/dist/esm/models/ContextualFlowInfo.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ContextualFlowInfoLayoutEnumFromJSON, ContextualFlowInfoLayoutEnumToJSON, } from './ContextualFlowInfoLayoutEnum';
/**
 * Check if a given object implements the ContextualFlowInfo interface.
 */
export function instanceOfContextualFlowInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "cancelUrl" in value;
    isInstance = isInstance && "layout" in value;
    return isInstance;
}
export function ContextualFlowInfoFromJSON(json) {
    return ContextualFlowInfoFromJSONTyped(json, false);
}
export function ContextualFlowInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'title': !exists(json, 'title') ? undefined : json['title'],
        'background': !exists(json, 'background') ? undefined : json['background'],
        'cancelUrl': json['cancel_url'],
        'layout': ContextualFlowInfoLayoutEnumFromJSON(json['layout']),
    };
}
export function ContextualFlowInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'title': value.title,
        'background': value.background,
        'cancel_url': value.cancelUrl,
        'layout': ContextualFlowInfoLayoutEnumToJSON(value.layout),
    };
}
