// @goauthentik/api/dist/esm/models/Coordinate.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Coordinate interface.
 */
export function instanceOfCoordinate(value) {
    let isInstance = true;
    isInstance = isInstance && "xCord" in value;
    isInstance = isInstance && "yCord" in value;
    return isInstance;
}
export function CoordinateFromJSON(json) {
    return CoordinateFromJSONTyped(json, false);
}
export function CoordinateFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'xCord': json['x_cord'],
        'yCord': json['y_cord'],
    };
}
export function CoordinateToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
