// @goauthentik/api/dist/esm/models/CurrentBrand.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FooterLinkFromJSON, } from './FooterLink';
import { UiThemeEnumFromJSON, } from './UiThemeEnum';
/**
 * Check if a given object implements the CurrentBrand interface.
 */
export function instanceOfCurrentBrand(value) {
    let isInstance = true;
    isInstance = isInstance && "matchedDomain" in value;
    isInstance = isInstance && "brandingTitle" in value;
    isInstance = isInstance && "brandingLogo" in value;
    isInstance = isInstance && "brandingFavicon" in value;
    isInstance = isInstance && "uiFooterLinks" in value;
    isInstance = isInstance && "uiTheme" in value;
    isInstance = isInstance && "defaultLocale" in value;
    return isInstance;
}
export function CurrentBrandFromJSON(json) {
    return CurrentBrandFromJSONTyped(json, false);
}
export function CurrentBrandFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'matchedDomain': json['matched_domain'],
        'brandingTitle': json['branding_title'],
        'brandingLogo': json['branding_logo'],
        'brandingFavicon': json['branding_favicon'],
        'uiFooterLinks': (json['ui_footer_links'].map(FooterLinkFromJSON)),
        'uiTheme': UiThemeEnumFromJSON(json['ui_theme']),
        'flowAuthentication': !exists(json, 'flow_authentication') ? undefined : json['flow_authentication'],
        'flowInvalidation': !exists(json, 'flow_invalidation') ? undefined : json['flow_invalidation'],
        'flowRecovery': !exists(json, 'flow_recovery') ? undefined : json['flow_recovery'],
        'flowUnenrollment': !exists(json, 'flow_unenrollment') ? undefined : json['flow_unenrollment'],
        'flowUserSettings': !exists(json, 'flow_user_settings') ? undefined : json['flow_user_settings'],
        'flowDeviceCode': !exists(json, 'flow_device_code') ? undefined : json['flow_device_code'],
        'defaultLocale': json['default_locale'],
    };
}
export function CurrentBrandToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'matched_domain': value.matchedDomain,
        'branding_title': value.brandingTitle,
        'branding_logo': value.brandingLogo,
        'branding_favicon': value.brandingFavicon,
        'flow_authentication': value.flowAuthentication,
        'flow_invalidation': value.flowInvalidation,
        'flow_recovery': value.flowRecovery,
        'flow_unenrollment': value.flowUnenrollment,
        'flow_user_settings': value.flowUserSettings,
        'flow_device_code': value.flowDeviceCode,
    };
}
