// @goauthentik/api/dist/esm/models/DenyStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the DenyStageRequest interface.
 */
export function instanceOfDenyStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function DenyStageRequestFromJSON(json) {
    return DenyStageRequestFromJSONTyped(json, false);
}
export function DenyStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'denyMessage': !exists(json, 'deny_message') ? undefined : json['deny_message'],
    };
}
export function DenyStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'deny_message': value.denyMessage,
    };
}
