// @goauthentik/api/dist/esm/models/DockerServiceConnectionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the DockerServiceConnectionRequest interface.
 */
export function instanceOfDockerServiceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
export function DockerServiceConnectionRequestFromJSON(json) {
    return DockerServiceConnectionRequestFromJSONTyped(json, false);
}
export function DockerServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'url': json['url'],
        'tlsVerification': !exists(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !exists(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}
export function DockerServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
