// @goauthentik/api/dist/esm/models/DomainRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the DomainRequest interface.
 */
export function instanceOfDomainRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "domain" in value;
    isInstance = isInstance && "tenant" in value;
    return isInstance;
}
export function DomainRequestFromJSON(json) {
    return DomainRequestFromJSONTyped(json, false);
}
export function DomainRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'domain': json['domain'],
        'isPrimary': !exists(json, 'is_primary') ? undefined : json['is_primary'],
        'tenant': json['tenant'],
    };
}
export function DomainRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'domain': value.domain,
        'is_primary': value.isPrimary,
        'tenant': value.tenant,
    };
}
