// @goauthentik/api/dist/esm/models/EmailStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the EmailStageRequest interface.
 */
export function instanceOfEmailStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function EmailStageRequestFromJSON(json) {
    return EmailStageRequestFromJSONTyped(json, false);
}
export function EmailStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'useGlobalSettings': !exists(json, 'use_global_settings') ? undefined : json['use_global_settings'],
        'host': !exists(json, 'host') ? undefined : json['host'],
        'port': !exists(json, 'port') ? undefined : json['port'],
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'useTls': !exists(json, 'use_tls') ? undefined : json['use_tls'],
        'useSsl': !exists(json, 'use_ssl') ? undefined : json['use_ssl'],
        'timeout': !exists(json, 'timeout') ? undefined : json['timeout'],
        'fromAddress': !exists(json, 'from_address') ? undefined : json['from_address'],
        'tokenExpiry': !exists(json, 'token_expiry') ? undefined : json['token_expiry'],
        'subject': !exists(json, 'subject') ? undefined : json['subject'],
        'template': !exists(json, 'template') ? undefined : json['template'],
        'activateUserOnSuccess': !exists(json, 'activate_user_on_success') ? undefined : json['activate_user_on_success'],
    };
}
export function EmailStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'use_global_settings': value.useGlobalSettings,
        'host': value.host,
        'port': value.port,
        'username': value.username,
        'password': value.password,
        'use_tls': value.useTls,
        'use_ssl': value.useSsl,
        'timeout': value.timeout,
        'from_address': value.fromAddress,
        'token_expiry': value.tokenExpiry,
        'subject': value.subject,
        'template': value.template,
        'activate_user_on_success': value.activateUserOnSuccess,
    };
}
