// @goauthentik/api/dist/esm/models/ErrorDetail.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the ErrorDetail interface.
 */
export function instanceOfErrorDetail(value) {
    let isInstance = true;
    isInstance = isInstance && "string" in value;
    isInstance = isInstance && "code" in value;
    return isInstance;
}
export function ErrorDetailFromJSON(json) {
    return ErrorDetailFromJSONTyped(json, false);
}
export function ErrorDetailFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'string': json['string'],
        'code': json['code'],
    };
}
export function ErrorDetailToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'string': value.string,
        'code': value.code,
    };
}
