// @goauthentik/api/dist/esm/models/EventMatcherPolicyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { AppEnumFromJSON, AppEnumToJSON, } from './AppEnum';
import { EventActionsFromJSON, EventActionsToJSON, } from './EventActions';
import { ModelEnumFromJSON, ModelEnumToJSON, } from './ModelEnum';
/**
 * Check if a given object implements the EventMatcherPolicyRequest interface.
 */
export function instanceOfEventMatcherPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function EventMatcherPolicyRequestFromJSON(json) {
    return EventMatcherPolicyRequestFromJSONTyped(json, false);
}
export function EventMatcherPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'action': !exists(json, 'action') ? undefined : EventActionsFromJSON(json['action']),
        'clientIp': !exists(json, 'client_ip') ? undefined : json['client_ip'],
        'app': !exists(json, 'app') ? undefined : AppEnumFromJSON(json['app']),
        'model': !exists(json, 'model') ? undefined : ModelEnumFromJSON(json['model']),
    };
}
export function EventMatcherPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'action': EventActionsToJSON(value.action),
        'client_ip': value.clientIp,
        'app': AppEnumToJSON(value.app),
        'model': ModelEnumToJSON(value.model),
    };
}
