// @goauthentik/api/dist/esm/models/Event.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { EventActionsFromJSON, EventActionsToJSON, } from './EventActions';
/**
 * Check if a given object implements the Event interface.
 */
export function instanceOfEvent(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "action" in value;
    isInstance = isInstance && "app" in value;
    isInstance = isInstance && "created" in value;
    return isInstance;
}
export function EventFromJSON(json) {
    return EventFromJSONTyped(json, false);
}
export function EventFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': !exists(json, 'user') ? undefined : json['user'],
        'action': EventActionsFromJSON(json['action']),
        'app': json['app'],
        'context': !exists(json, 'context') ? undefined : json['context'],
        'clientIp': !exists(json, 'client_ip') ? undefined : json['client_ip'],
        'created': (new Date(json['created'])),
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'brand': !exists(json, 'brand') ? undefined : json['brand'],
    };
}
export function EventToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'action': EventActionsToJSON(value.action),
        'app': value.app,
        'context': value.context,
        'client_ip': value.clientIp,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'brand': value.brand,
    };
}
