// @goauthentik/api/dist/esm/models/ExpiringBaseGrantModel.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { OAuth2ProviderFromJSON, OAuth2ProviderToJSON, } from './OAuth2Provider';
import { UserFromJSON, UserToJSON, } from './User';
/**
 * Check if a given object implements the ExpiringBaseGrantModel interface.
 */
export function instanceOfExpiringBaseGrantModel(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "isExpired" in value;
    isInstance = isInstance && "scope" in value;
    return isInstance;
}
export function ExpiringBaseGrantModelFromJSON(json) {
    return ExpiringBaseGrantModelFromJSONTyped(json, false);
}
export function ExpiringBaseGrantModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'provider': OAuth2ProviderFromJSON(json['provider']),
        'user': UserFromJSON(json['user']),
        'isExpired': json['is_expired'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'scope': json['scope'],
    };
}
export function ExpiringBaseGrantModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': OAuth2ProviderToJSON(value.provider),
        'user': UserToJSON(value.user),
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'scope': value.scope,
    };
}
