// @goauthentik/api/dist/esm/models/FilePathRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the FilePathRequest interface.
 */
export function instanceOfFilePathRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
export function FilePathRequestFromJSON(json) {
    return FilePathRequestFromJSONTyped(json, false);
}
export function FilePathRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'url': json['url'],
    };
}
export function FilePathRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'url': value.url,
    };
}
