// @goauthentik/api/dist/esm/models/FlowInspection.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowInspectorPlanFromJSON, FlowInspectorPlanToJSON, } from './FlowInspectorPlan';
/**
 * Check if a given object implements the FlowInspection interface.
 */
export function instanceOfFlowInspection(value) {
    let isInstance = true;
    isInstance = isInstance && "plans" in value;
    isInstance = isInstance && "isCompleted" in value;
    return isInstance;
}
export function FlowInspectionFromJSON(json) {
    return FlowInspectionFromJSONTyped(json, false);
}
export function FlowInspectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'plans': (json['plans'].map(FlowInspectorPlanFromJSON)),
        'currentPlan': !exists(json, 'current_plan') ? undefined : FlowInspectorPlanFromJSON(json['current_plan']),
        'isCompleted': json['is_completed'],
    };
}
export function FlowInspectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'plans': (value.plans.map(FlowInspectorPlanToJSON)),
        'current_plan': FlowInspectorPlanToJSON(value.currentPlan),
        'is_completed': value.isCompleted,
    };
}
