// @goauthentik/api/dist/esm/models/FlowImportResult.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the FlowImportResult interface.
 */
export function instanceOfFlowImportResult(value) {
    let isInstance = true;
    isInstance = isInstance && "logs" in value;
    isInstance = isInstance && "success" in value;
    return isInstance;
}
export function FlowImportResultFromJSON(json) {
    return FlowImportResultFromJSONTyped(json, false);
}
export function FlowImportResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'logs': json['logs'],
        'success': json['success'],
    };
}
export function FlowImportResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
