// @goauthentik/api/dist/esm/models/FlowDiagram.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the FlowDiagram interface.
 */
export function instanceOfFlowDiagram(value) {
    let isInstance = true;
    isInstance = isInstance && "diagram" in value;
    return isInstance;
}
export function FlowDiagramFromJSON(json) {
    return FlowDiagramFromJSONTyped(json, false);
}
export function FlowDiagramFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'diagram': json['diagram'],
    };
}
export function FlowDiagramToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
