// @goauthentik/api/dist/esm/models/FlowDesignationEnum.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * * `authentication` - Authentication
 * * `authorization` - Authorization
 * * `invalidation` - Invalidation
 * * `enrollment` - Enrollment
 * * `unenrollment` - Unrenollment
 * * `recovery` - Recovery
 * * `stage_configuration` - Stage Configuration
 * @export
 */
export const FlowDesignationEnum = {
    Authentication: 'authentication',
    Authorization: 'authorization',
    Invalidation: 'invalidation',
    Enrollment: 'enrollment',
    Unenrollment: 'unenrollment',
    Recovery: 'recovery',
    StageConfiguration: 'stage_configuration',
    UnknownDefaultOpenApi: '11184809'
};
export function FlowDesignationEnumFromJSON(json) {
    return FlowDesignationEnumFromJSONTyped(json, false);
}
export function FlowDesignationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function FlowDesignationEnumToJSON(value) {
    return value;
}
