// @goauthentik/api/dist/esm/models/FlowInspectorPlan.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { FlowStageBindingFromJSON, } from './FlowStageBinding';
/**
 * Check if a given object implements the FlowInspectorPlan interface.
 */
export function instanceOfFlowInspectorPlan(value) {
    let isInstance = true;
    isInstance = isInstance && "currentStage" in value;
    isInstance = isInstance && "nextPlannedStage" in value;
    isInstance = isInstance && "planContext" in value;
    isInstance = isInstance && "sessionId" in value;
    return isInstance;
}
export function FlowInspectorPlanFromJSON(json) {
    return FlowInspectorPlanFromJSONTyped(json, false);
}
export function FlowInspectorPlanFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'currentStage': FlowStageBindingFromJSON(json['current_stage']),
        'nextPlannedStage': FlowStageBindingFromJSON(json['next_planned_stage']),
        'planContext': json['plan_context'],
        'sessionId': json['session_id'],
    };
}
export function FlowInspectorPlanToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
