// @goauthentik/api/dist/esm/models/FlowErrorChallenge.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ContextualFlowInfoFromJSON, ContextualFlowInfoToJSON, } from './ContextualFlowInfo';
/**
 * Check if a given object implements the FlowErrorChallenge interface.
 */
export function instanceOfFlowErrorChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "requestId" in value;
    return isInstance;
}
export function FlowErrorChallengeFromJSON(json) {
    return FlowErrorChallengeFromJSONTyped(json, false);
}
export function FlowErrorChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !exists(json, 'type') ? undefined : json['type'],
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'requestId': json['request_id'],
        'error': !exists(json, 'error') ? undefined : json['error'],
        'traceback': !exists(json, 'traceback') ? undefined : json['traceback'],
    };
}
export function FlowErrorChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'request_id': value.requestId,
        'error': value.error,
        'traceback': value.traceback,
    };
}
