// @goauthentik/api/dist/esm/models/FooterLink.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the FooterLink interface.
 */
export function instanceOfFooterLink(value) {
    let isInstance = true;
    isInstance = isInstance && "href" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function FooterLinkFromJSON(json) {
    return FooterLinkFromJSONTyped(json, false);
}
export function FooterLinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': json['href'],
        'name': json['name'],
    };
}
export function FooterLinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
