// @goauthentik/api/dist/esm/models/GenericError.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the GenericError interface.
 */
export function instanceOfGenericError(value) {
    let isInstance = true;
    isInstance = isInstance && "detail" in value;
    return isInstance;
}
export function GenericErrorFromJSON(json) {
    return GenericErrorFromJSONTyped(json, false);
}
export function GenericErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'detail': json['detail'],
        'code': !exists(json, 'code') ? undefined : json['code'],
    };
}
export function GenericErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'detail': value.detail,
        'code': value.code,
    };
}
