// @goauthentik/api/dist/esm/models/GeoipBindingEnum.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * * `no_binding` - No Binding
 * * `bind_continent` - Bind Continent
 * * `bind_continent_country` - Bind Continent Country
 * * `bind_continent_country_city` - Bind Continent Country City
 * @export
 */
export const GeoipBindingEnum = {
    NoBinding: 'no_binding',
    BindContinent: 'bind_continent',
    BindContinentCountry: 'bind_continent_country',
    BindContinentCountryCity: 'bind_continent_country_city',
    UnknownDefaultOpenApi: '11184809'
};
export function GeoipBindingEnumFromJSON(json) {
    return GeoipBindingEnumFromJSONTyped(json, false);
}
export function GeoipBindingEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function GeoipBindingEnumToJSON(value) {
    return value;
}
