// @goauthentik/api/dist/esm/models/GroupRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the GroupRequest interface.
 */
export function instanceOfGroupRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function GroupRequestFromJSON(json) {
    return GroupRequestFromJSONTyped(json, false);
}
export function GroupRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'isSuperuser': !exists(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !exists(json, 'parent') ? undefined : json['parent'],
        'users': !exists(json, 'users') ? undefined : json['users'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'roles': !exists(json, 'roles') ? undefined : json['roles'],
    };
}
export function GroupRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'users': value.users,
        'attributes': value.attributes,
        'roles': value.roles,
    };
}
