// @goauthentik/api/dist/esm/models/Group.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { GroupMemberFromJSON, } from './GroupMember';
import { RoleFromJSON, } from './Role';
/**
 * Check if a given object implements the Group interface.
 */
export function instanceOfGroup(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "numPk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "parentName" in value;
    isInstance = isInstance && "usersObj" in value;
    isInstance = isInstance && "rolesObj" in value;
    return isInstance;
}
export function GroupFromJSON(json) {
    return GroupFromJSONTyped(json, false);
}
export function GroupFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'numPk': json['num_pk'],
        'name': json['name'],
        'isSuperuser': !exists(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !exists(json, 'parent') ? undefined : json['parent'],
        'parentName': json['parent_name'],
        'users': !exists(json, 'users') ? undefined : json['users'],
        'usersObj': (json['users_obj'].map(GroupMemberFromJSON)),
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'roles': !exists(json, 'roles') ? undefined : json['roles'],
        'rolesObj': (json['roles_obj'].map(RoleFromJSON)),
    };
}
export function GroupToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'users': value.users,
        'attributes': value.attributes,
        'roles': value.roles,
    };
}
