// @goauthentik/api/dist/esm/models/IdentificationChallenge.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ChallengeChoicesFromJSON, ChallengeChoicesToJSON, } from './ChallengeChoices';
import { ContextualFlowInfoFromJSON, ContextualFlowInfoToJSON, } from './ContextualFlowInfo';
import { LoginSourceFromJSON, LoginSourceToJSON, } from './LoginSource';
/**
 * Check if a given object implements the IdentificationChallenge interface.
 */
export function instanceOfIdentificationChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "userFields" in value;
    isInstance = isInstance && "passwordFields" in value;
    isInstance = isInstance && "primaryAction" in value;
    isInstance = isInstance && "showSourceLabels" in value;
    return isInstance;
}
export function IdentificationChallengeFromJSON(json) {
    return IdentificationChallengeFromJSONTyped(json, false);
}
export function IdentificationChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'userFields': json['user_fields'],
        'passwordFields': json['password_fields'],
        'applicationPre': !exists(json, 'application_pre') ? undefined : json['application_pre'],
        'enrollUrl': !exists(json, 'enroll_url') ? undefined : json['enroll_url'],
        'recoveryUrl': !exists(json, 'recovery_url') ? undefined : json['recovery_url'],
        'passwordlessUrl': !exists(json, 'passwordless_url') ? undefined : json['passwordless_url'],
        'primaryAction': json['primary_action'],
        'sources': !exists(json, 'sources') ? undefined : (json['sources'].map(LoginSourceFromJSON)),
        'showSourceLabels': json['show_source_labels'],
    };
}
export function IdentificationChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'user_fields': value.userFields,
        'password_fields': value.passwordFields,
        'application_pre': value.applicationPre,
        'enroll_url': value.enrollUrl,
        'recovery_url': value.recoveryUrl,
        'passwordless_url': value.passwordlessUrl,
        'primary_action': value.primaryAction,
        'sources': value.sources === undefined ? undefined : (value.sources.map(LoginSourceToJSON)),
        'show_source_labels': value.showSourceLabels,
    };
}
