// @goauthentik/api/dist/esm/models/IdentificationStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
import { UserFieldsEnumFromJSON, UserFieldsEnumToJSON, } from './UserFieldsEnum';
/**
 * Check if a given object implements the IdentificationStage interface.
 */
export function instanceOfIdentificationStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function IdentificationStageFromJSON(json) {
    return IdentificationStageFromJSONTyped(json, false);
}
export function IdentificationStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'userFields': !exists(json, 'user_fields') ? undefined : (json['user_fields'].map(UserFieldsEnumFromJSON)),
        'passwordStage': !exists(json, 'password_stage') ? undefined : json['password_stage'],
        'caseInsensitiveMatching': !exists(json, 'case_insensitive_matching') ? undefined : json['case_insensitive_matching'],
        'showMatchedUser': !exists(json, 'show_matched_user') ? undefined : json['show_matched_user'],
        'enrollmentFlow': !exists(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'recoveryFlow': !exists(json, 'recovery_flow') ? undefined : json['recovery_flow'],
        'passwordlessFlow': !exists(json, 'passwordless_flow') ? undefined : json['passwordless_flow'],
        'sources': !exists(json, 'sources') ? undefined : json['sources'],
        'showSourceLabels': !exists(json, 'show_source_labels') ? undefined : json['show_source_labels'],
        'pretendUserExists': !exists(json, 'pretend_user_exists') ? undefined : json['pretend_user_exists'],
    };
}
export function IdentificationStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'user_fields': value.userFields === undefined ? undefined : (value.userFields.map(UserFieldsEnumToJSON)),
        'password_stage': value.passwordStage,
        'case_insensitive_matching': value.caseInsensitiveMatching,
        'show_matched_user': value.showMatchedUser,
        'enrollment_flow': value.enrollmentFlow,
        'recovery_flow': value.recoveryFlow,
        'passwordless_flow': value.passwordlessFlow,
        'sources': value.sources,
        'show_source_labels': value.showSourceLabels,
        'pretend_user_exists': value.pretendUserExists,
    };
}
