// @goauthentik/api/dist/esm/models/Invitation.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowFromJSON, } from './Flow';
import { GroupMemberFromJSON, } from './GroupMember';
/**
 * Check if a given object implements the Invitation interface.
 */
export function instanceOfInvitation(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "flowObj" in value;
    return isInstance;
}
export function InvitationFromJSON(json) {
    return InvitationFromJSONTyped(json, false);
}
export function InvitationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !exists(json, 'fixed_data') ? undefined : json['fixed_data'],
        'createdBy': GroupMemberFromJSON(json['created_by']),
        'singleUse': !exists(json, 'single_use') ? undefined : json['single_use'],
        'flow': !exists(json, 'flow') ? undefined : json['flow'],
        'flowObj': FlowFromJSON(json['flow_obj']),
    };
}
export function InvitationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
