// @goauthentik/api/dist/esm/models/KubernetesServiceConnection.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the KubernetesServiceConnection interface.
 */
export function instanceOfKubernetesServiceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function KubernetesServiceConnectionFromJSON(json) {
    return KubernetesServiceConnectionFromJSONTyped(json, false);
}
export function KubernetesServiceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'kubeconfig': !exists(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !exists(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}
export function KubernetesServiceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
