// @goauthentik/api/dist/esm/models/LDAPDebug.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LDAPDebug interface.
 */
export function instanceOfLDAPDebug(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "group" in value;
    isInstance = isInstance && "membership" in value;
    return isInstance;
}
export function LDAPDebugFromJSON(json) {
    return LDAPDebugFromJSONTyped(json, false);
}
export function LDAPDebugFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'group': json['group'],
        'membership': json['membership'],
    };
}
export function LDAPDebugToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
