// @goauthentik/api/dist/esm/models/LDAPOutpostConfig.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { LDAPAPIAccessModeFromJSON, LDAPAPIAccessModeToJSON, } from './LDAPAPIAccessMode';
/**
 * Check if a given object implements the LDAPOutpostConfig interface.
 */
export function instanceOfLDAPOutpostConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "bindFlowSlug" in value;
    isInstance = isInstance && "applicationSlug" in value;
    return isInstance;
}
export function LDAPOutpostConfigFromJSON(json) {
    return LDAPOutpostConfigFromJSONTyped(json, false);
}
export function LDAPOutpostConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'baseDn': !exists(json, 'base_dn') ? undefined : json['base_dn'],
        'bindFlowSlug': json['bind_flow_slug'],
        'applicationSlug': json['application_slug'],
        'searchGroup': !exists(json, 'search_group') ? undefined : json['search_group'],
        'certificate': !exists(json, 'certificate') ? undefined : json['certificate'],
        'tlsServerName': !exists(json, 'tls_server_name') ? undefined : json['tls_server_name'],
        'uidStartNumber': !exists(json, 'uid_start_number') ? undefined : json['uid_start_number'],
        'gidStartNumber': !exists(json, 'gid_start_number') ? undefined : json['gid_start_number'],
        'searchMode': !exists(json, 'search_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['search_mode']),
        'bindMode': !exists(json, 'bind_mode') ? undefined : LDAPAPIAccessModeFromJSON(json['bind_mode']),
        'mfaSupport': !exists(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}
export function LDAPOutpostConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'base_dn': value.baseDn,
        'bind_flow_slug': value.bindFlowSlug,
        'search_group': value.searchGroup,
        'certificate': value.certificate,
        'tls_server_name': value.tlsServerName,
        'uid_start_number': value.uidStartNumber,
        'gid_start_number': value.gidStartNumber,
        'search_mode': LDAPAPIAccessModeToJSON(value.searchMode),
        'bind_mode': LDAPAPIAccessModeToJSON(value.bindMode),
        'mfa_support': value.mfaSupport,
    };
}
