// @goauthentik/api/dist/esm/models/LicenseForecast.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LicenseForecast interface.
 */
export function instanceOfLicenseForecast(value) {
    let isInstance = true;
    isInstance = isInstance && "internalUsers" in value;
    isInstance = isInstance && "externalUsers" in value;
    isInstance = isInstance && "forecastedInternalUsers" in value;
    isInstance = isInstance && "forecastedExternalUsers" in value;
    return isInstance;
}
export function LicenseForecastFromJSON(json) {
    return LicenseForecastFromJSONTyped(json, false);
}
export function LicenseForecastFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'internalUsers': json['internal_users'],
        'externalUsers': json['external_users'],
        'forecastedInternalUsers': json['forecasted_internal_users'],
        'forecastedExternalUsers': json['forecasted_external_users'],
    };
}
export function LicenseForecastToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'internal_users': value.internalUsers,
        'external_users': value.externalUsers,
        'forecasted_internal_users': value.forecastedInternalUsers,
        'forecasted_external_users': value.forecastedExternalUsers,
    };
}
