// @goauthentik/api/dist/esm/models/Link.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Link interface.
 */
export function instanceOfLink(value) {
    let isInstance = true;
    isInstance = isInstance && "link" in value;
    return isInstance;
}
export function LinkFromJSON(json) {
    return LinkFromJSONTyped(json, false);
}
export function LinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'link': json['link'],
    };
}
export function LinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'link': value.link,
    };
}
