// @goauthentik/api/dist/esm/models/LoginSource.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { LoginChallengeTypesFromJSON, LoginChallengeTypesToJSON, } from './LoginChallengeTypes';
/**
 * Check if a given object implements the LoginSource interface.
 */
export function instanceOfLoginSource(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "challenge" in value;
    return isInstance;
}
export function LoginSourceFromJSON(json) {
    return LoginSourceFromJSONTyped(json, false);
}
export function LoginSourceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'iconUrl': !exists(json, 'icon_url') ? undefined : json['icon_url'],
        'challenge': LoginChallengeTypesFromJSON(json['challenge']),
    };
}
export function LoginSourceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'icon_url': value.iconUrl,
        'challenge': LoginChallengeTypesToJSON(value.challenge),
    };
}
