// @goauthentik/api/dist/esm/models/LoginMetrics.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { CoordinateFromJSON, } from './Coordinate';
/**
 * Check if a given object implements the LoginMetrics interface.
 */
export function instanceOfLoginMetrics(value) {
    let isInstance = true;
    isInstance = isInstance && "logins" in value;
    isInstance = isInstance && "loginsFailed" in value;
    isInstance = isInstance && "authorizations" in value;
    return isInstance;
}
export function LoginMetricsFromJSON(json) {
    return LoginMetricsFromJSONTyped(json, false);
}
export function LoginMetricsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'logins': (json['logins'].map(CoordinateFromJSON)),
        'loginsFailed': (json['logins_failed'].map(CoordinateFromJSON)),
        'authorizations': (json['authorizations'].map(CoordinateFromJSON)),
    };
}
export function LoginMetricsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
