// @goauthentik/api/dist/esm/models/Notification.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { EventFromJSON, EventToJSON, } from './Event';
import { SeverityEnumFromJSON, } from './SeverityEnum';
/**
 * Check if a given object implements the Notification interface.
 */
export function instanceOfNotification(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "severity" in value;
    isInstance = isInstance && "body" in value;
    isInstance = isInstance && "created" in value;
    return isInstance;
}
export function NotificationFromJSON(json) {
    return NotificationFromJSONTyped(json, false);
}
export function NotificationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'severity': SeverityEnumFromJSON(json['severity']),
        'body': json['body'],
        'created': (new Date(json['created'])),
        'event': !exists(json, 'event') ? undefined : EventFromJSON(json['event']),
        'seen': !exists(json, 'seen') ? undefined : json['seen'],
    };
}
export function NotificationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'event': EventToJSON(value.event),
        'seen': value.seen,
    };
}
