// @goauthentik/api/dist/esm/models/NotificationTransportModeEnum.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * * `local` - authentik inbuilt notifications
 * * `webhook` - Generic Webhook
 * * `webhook_slack` - Slack Webhook (Slack/Discord)
 * * `email` - Email
 * @export
 */
export const NotificationTransportModeEnum = {
    Local: 'local',
    Webhook: 'webhook',
    WebhookSlack: 'webhook_slack',
    Email: 'email',
    UnknownDefaultOpenApi: '11184809'
};
export function NotificationTransportModeEnumFromJSON(json) {
    return NotificationTransportModeEnumFromJSONTyped(json, false);
}
export function NotificationTransportModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
export function NotificationTransportModeEnumToJSON(value) {
    return value;
}
