// @goauthentik/api/dist/esm/models/NotificationTransportTest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the NotificationTransportTest interface.
 */
export function instanceOfNotificationTransportTest(value) {
    let isInstance = true;
    isInstance = isInstance && "messages" in value;
    return isInstance;
}
export function NotificationTransportTestFromJSON(json) {
    return NotificationTransportTestFromJSONTyped(json, false);
}
export function NotificationTransportTestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'messages': json['messages'],
    };
}
export function NotificationTransportTestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'messages': value.messages,
    };
}
