// @goauthentik/api/dist/esm/models/NotificationRule.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { GroupFromJSON, } from './Group';
import { SeverityEnumFromJSON, SeverityEnumToJSON, } from './SeverityEnum';
/**
 * Check if a given object implements the NotificationRule interface.
 */
export function instanceOfNotificationRule(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "groupObj" in value;
    return isInstance;
}
export function NotificationRuleFromJSON(json) {
    return NotificationRuleFromJSONTyped(json, false);
}
export function NotificationRuleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'transports': !exists(json, 'transports') ? undefined : json['transports'],
        'severity': !exists(json, 'severity') ? undefined : SeverityEnumFromJSON(json['severity']),
        'group': !exists(json, 'group') ? undefined : json['group'],
        'groupObj': GroupFromJSON(json['group_obj']),
    };
}
export function NotificationRuleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'transports': value.transports,
        'severity': SeverityEnumToJSON(value.severity),
        'group': value.group,
    };
}
