// @goauthentik/api/dist/esm/models/OAuthDeviceCodeChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the OAuthDeviceCodeChallengeResponseRequest interface.
 */
export function instanceOfOAuthDeviceCodeChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    return isInstance;
}
export function OAuthDeviceCodeChallengeResponseRequestFromJSON(json) {
    return OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json, false);
}
export function OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'code': json['code'],
    };
}
export function OAuthDeviceCodeChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'code': value.code,
    };
}
