// @goauthentik/api/dist/esm/models/OutpostDefaultConfig.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the OutpostDefaultConfig interface.
 */
export function instanceOfOutpostDefaultConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
export function OutpostDefaultConfigFromJSON(json) {
    return OutpostDefaultConfigFromJSONTyped(json, false);
}
export function OutpostDefaultConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'config': json['config'],
    };
}
export function OutpostDefaultConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
