// @goauthentik/api/dist/esm/models/OutpostRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { OutpostTypeEnumFromJSON, OutpostTypeEnumToJSON, } from './OutpostTypeEnum';
/**
 * Check if a given object implements the OutpostRequest interface.
 */
export function instanceOfOutpostRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "providers" in value;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
export function OutpostRequestFromJSON(json) {
    return OutpostRequestFromJSONTyped(json, false);
}
export function OutpostRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'type': OutpostTypeEnumFromJSON(json['type']),
        'providers': json['providers'],
        'serviceConnection': !exists(json, 'service_connection') ? undefined : json['service_connection'],
        'config': json['config'],
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
    };
}
export function OutpostRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': OutpostTypeEnumToJSON(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
