// @goauthentik/api/dist/esm/models/Outpost.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { OutpostTypeEnumFromJSON, OutpostTypeEnumToJSON, } from './OutpostTypeEnum';
import { ProviderFromJSON, } from './Provider';
import { ServiceConnectionFromJSON, } from './ServiceConnection';
/**
 * Check if a given object implements the Outpost interface.
 */
export function instanceOfOutpost(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "providers" in value;
    isInstance = isInstance && "providersObj" in value;
    isInstance = isInstance && "serviceConnectionObj" in value;
    isInstance = isInstance && "tokenIdentifier" in value;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
export function OutpostFromJSON(json) {
    return OutpostFromJSONTyped(json, false);
}
export function OutpostFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'type': OutpostTypeEnumFromJSON(json['type']),
        'providers': json['providers'],
        'providersObj': (json['providers_obj'].map(ProviderFromJSON)),
        'serviceConnection': !exists(json, 'service_connection') ? undefined : json['service_connection'],
        'serviceConnectionObj': ServiceConnectionFromJSON(json['service_connection_obj']),
        'tokenIdentifier': json['token_identifier'],
        'config': json['config'],
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
    };
}
export function OutpostToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': OutpostTypeEnumToJSON(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
