// @goauthentik/api/dist/esm/models/PaginatedNotificationWebhookMappingList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { NotificationWebhookMappingFromJSON, NotificationWebhookMappingToJSON, } from './NotificationWebhookMapping';
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
/**
 * Check if a given object implements the PaginatedNotificationWebhookMappingList interface.
 */
export function instanceOfPaginatedNotificationWebhookMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedNotificationWebhookMappingListFromJSON(json) {
    return PaginatedNotificationWebhookMappingListFromJSONTyped(json, false);
}
export function PaginatedNotificationWebhookMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(NotificationWebhookMappingFromJSON)),
    };
}
export function PaginatedNotificationWebhookMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(NotificationWebhookMappingToJSON)),
    };
}
