// @goauthentik/api/dist/esm/models/PaginatedSystemTaskList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
import { SystemTaskFromJSON, SystemTaskToJSON, } from './SystemTask';
/**
 * Check if a given object implements the PaginatedSystemTaskList interface.
 */
export function instanceOfPaginatedSystemTaskList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedSystemTaskListFromJSON(json) {
    return PaginatedSystemTaskListFromJSONTyped(json, false);
}
export function PaginatedSystemTaskListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(SystemTaskFromJSON)),
    };
}
export function PaginatedSystemTaskListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(SystemTaskToJSON)),
    };
}
