// @goauthentik/api/dist/esm/models/PaginatedOAuth2ProviderList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { OAuth2ProviderFromJSON, OAuth2ProviderToJSON, } from './OAuth2Provider';
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
/**
 * Check if a given object implements the PaginatedOAuth2ProviderList interface.
 */
export function instanceOfPaginatedOAuth2ProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedOAuth2ProviderListFromJSON(json) {
    return PaginatedOAuth2ProviderListFromJSONTyped(json, false);
}
export function PaginatedOAuth2ProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(OAuth2ProviderFromJSON)),
    };
}
export function PaginatedOAuth2ProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(OAuth2ProviderToJSON)),
    };
}
