// @goauthentik/api/dist/esm/models/PaginatedRoleList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
import { RoleFromJSON, RoleToJSON, } from './Role';
/**
 * Check if a given object implements the PaginatedRoleList interface.
 */
export function instanceOfPaginatedRoleList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedRoleListFromJSON(json) {
    return PaginatedRoleListFromJSONTyped(json, false);
}
export function PaginatedRoleListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(RoleFromJSON)),
    };
}
export function PaginatedRoleListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(RoleToJSON)),
    };
}
