// @goauthentik/api/dist/esm/models/PaginatedTOTPDeviceList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
import { TOTPDeviceFromJSON, TOTPDeviceToJSON, } from './TOTPDevice';
/**
 * Check if a given object implements the PaginatedTOTPDeviceList interface.
 */
export function instanceOfPaginatedTOTPDeviceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedTOTPDeviceListFromJSON(json) {
    return PaginatedTOTPDeviceListFromJSONTyped(json, false);
}
export function PaginatedTOTPDeviceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(TOTPDeviceFromJSON)),
    };
}
export function PaginatedTOTPDeviceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(TOTPDeviceToJSON)),
    };
}
