// @goauthentik/api/dist/esm/models/PaginatedCaptchaStageList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { CaptchaStageFromJSON, CaptchaStageToJSON, } from './CaptchaStage';
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
/**
 * Check if a given object implements the PaginatedCaptchaStageList interface.
 */
export function instanceOfPaginatedCaptchaStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedCaptchaStageListFromJSON(json) {
    return PaginatedCaptchaStageListFromJSONTyped(json, false);
}
export function PaginatedCaptchaStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(CaptchaStageFromJSON)),
    };
}
export function PaginatedCaptchaStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(CaptchaStageToJSON)),
    };
}
