// @goauthentik/api/dist/esm/models/Pagination.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Pagination interface.
 */
export function instanceOfPagination(value) {
    let isInstance = true;
    isInstance = isInstance && "next" in value;
    isInstance = isInstance && "previous" in value;
    isInstance = isInstance && "count" in value;
    isInstance = isInstance && "current" in value;
    isInstance = isInstance && "totalPages" in value;
    isInstance = isInstance && "startIndex" in value;
    isInstance = isInstance && "endIndex" in value;
    return isInstance;
}
export function PaginationFromJSON(json) {
    return PaginationFromJSONTyped(json, false);
}
export function PaginationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'next': json['next'],
        'previous': json['previous'],
        'count': json['count'],
        'current': json['current'],
        'totalPages': json['total_pages'],
        'startIndex': json['start_index'],
        'endIndex': json['end_index'],
    };
}
export function PaginationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'next': value.next,
        'previous': value.previous,
        'count': value.count,
        'current': value.current,
        'total_pages': value.totalPages,
        'start_index': value.startIndex,
        'end_index': value.endIndex,
    };
}
