// @goauthentik/api/dist/esm/models/PaginatedConsentStageList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { ConsentStageFromJSON, ConsentStageToJSON, } from './ConsentStage';
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
/**
 * Check if a given object implements the PaginatedConsentStageList interface.
 */
export function instanceOfPaginatedConsentStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedConsentStageListFromJSON(json) {
    return PaginatedConsentStageListFromJSONTyped(json, false);
}
export function PaginatedConsentStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(ConsentStageFromJSON)),
    };
}
export function PaginatedConsentStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(ConsentStageToJSON)),
    };
}
