// @goauthentik/api/dist/esm/models/PaginatedPlexSourceList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
import { PlexSourceFromJSON, PlexSourceToJSON, } from './PlexSource';
/**
 * Check if a given object implements the PaginatedPlexSourceList interface.
 */
export function instanceOfPaginatedPlexSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedPlexSourceListFromJSON(json) {
    return PaginatedPlexSourceListFromJSONTyped(json, false);
}
export function PaginatedPlexSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(PlexSourceFromJSON)),
    };
}
export function PaginatedPlexSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(PlexSourceToJSON)),
    };
}
