// @goauthentik/api/dist/esm/models/PaginatedEventMatcherPolicyList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { EventMatcherPolicyFromJSON, EventMatcherPolicyToJSON, } from './EventMatcherPolicy';
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
/**
 * Check if a given object implements the PaginatedEventMatcherPolicyList interface.
 */
export function instanceOfPaginatedEventMatcherPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedEventMatcherPolicyListFromJSON(json) {
    return PaginatedEventMatcherPolicyListFromJSONTyped(json, false);
}
export function PaginatedEventMatcherPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(EventMatcherPolicyFromJSON)),
    };
}
export function PaginatedEventMatcherPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(EventMatcherPolicyToJSON)),
    };
}
