// @goauthentik/api/dist/esm/models/PaginatedProxyProviderList.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PaginationFromJSON, PaginationToJSON, } from './Pagination';
import { ProxyProviderFromJSON, ProxyProviderToJSON, } from './ProxyProvider';
/**
 * Check if a given object implements the PaginatedProxyProviderList interface.
 */
export function instanceOfPaginatedProxyProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
export function PaginatedProxyProviderListFromJSON(json) {
    return PaginatedProxyProviderListFromJSONTyped(json, false);
}
export function PaginatedProxyProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': PaginationFromJSON(json['pagination']),
        'results': (json['results'].map(ProxyProviderFromJSON)),
    };
}
export function PaginatedProxyProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': PaginationToJSON(value.pagination),
        'results': (value.results.map(ProxyProviderToJSON)),
    };
}
