// @goauthentik/api/dist/esm/models/PasswordStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { BackendsEnumFromJSON, BackendsEnumToJSON, } from './BackendsEnum';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
/**
 * Check if a given object implements the PasswordStage interface.
 */
export function instanceOfPasswordStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "backends" in value;
    return isInstance;
}
export function PasswordStageFromJSON(json) {
    return PasswordStageFromJSONTyped(json, false);
}
export function PasswordStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'backends': (json['backends'].map(BackendsEnumFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'failedAttemptsBeforeCancel': !exists(json, 'failed_attempts_before_cancel') ? undefined : json['failed_attempts_before_cancel'],
    };
}
export function PasswordStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'backends': (value.backends.map(BackendsEnumToJSON)),
        'configure_flow': value.configureFlow,
        'failed_attempts_before_cancel': value.failedAttemptsBeforeCancel,
    };
}
