// @goauthentik/api/dist/esm/models/PasswordChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PasswordChallengeResponseRequest interface.
 */
export function instanceOfPasswordChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "password" in value;
    return isInstance;
}
export function PasswordChallengeResponseRequestFromJSON(json) {
    return PasswordChallengeResponseRequestFromJSONTyped(json, false);
}
export function PasswordChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'password': json['password'],
    };
}
export function PasswordChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'password': value.password,
    };
}
