// @goauthentik/api/dist/esm/models/PasswordExpiryPolicyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PasswordExpiryPolicyRequest interface.
 */
export function instanceOfPasswordExpiryPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "days" in value;
    return isInstance;
}
export function PasswordExpiryPolicyRequestFromJSON(json) {
    return PasswordExpiryPolicyRequestFromJSONTyped(json, false);
}
export function PasswordExpiryPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'days': json['days'],
        'denyOnly': !exists(json, 'deny_only') ? undefined : json['deny_only'],
    };
}
export function PasswordExpiryPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
