// @goauthentik/api/dist/esm/models/PasswordStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { BackendsEnumFromJSON, BackendsEnumToJSON, } from './BackendsEnum';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the PasswordStageRequest interface.
 */
export function instanceOfPasswordStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "backends" in value;
    return isInstance;
}
export function PasswordStageRequestFromJSON(json) {
    return PasswordStageRequestFromJSONTyped(json, false);
}
export function PasswordStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'backends': (json['backends'].map(BackendsEnumFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'failedAttemptsBeforeCancel': !exists(json, 'failed_attempts_before_cancel') ? undefined : json['failed_attempts_before_cancel'],
    };
}
export function PasswordStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'backends': (value.backends.map(BackendsEnumToJSON)),
        'configure_flow': value.configureFlow,
        'failed_attempts_before_cancel': value.failedAttemptsBeforeCancel,
    };
}
