// @goauthentik/api/dist/esm/models/PatchedFlowRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { AuthenticationEnumFromJSON, AuthenticationEnumToJSON, } from './AuthenticationEnum';
import { DeniedActionEnumFromJSON, DeniedActionEnumToJSON, } from './DeniedActionEnum';
import { FlowDesignationEnumFromJSON, FlowDesignationEnumToJSON, } from './FlowDesignationEnum';
import { FlowLayoutEnumFromJSON, FlowLayoutEnumToJSON, } from './FlowLayoutEnum';
import { PolicyEngineModeFromJSON, PolicyEngineModeToJSON, } from './PolicyEngineMode';
/**
 * Check if a given object implements the PatchedFlowRequest interface.
 */
export function instanceOfPatchedFlowRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedFlowRequestFromJSON(json) {
    return PatchedFlowRequestFromJSONTyped(json, false);
}
export function PatchedFlowRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'slug': !exists(json, 'slug') ? undefined : json['slug'],
        'title': !exists(json, 'title') ? undefined : json['title'],
        'designation': !exists(json, 'designation') ? undefined : FlowDesignationEnumFromJSON(json['designation']),
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'compatibilityMode': !exists(json, 'compatibility_mode') ? undefined : json['compatibility_mode'],
        'layout': !exists(json, 'layout') ? undefined : FlowLayoutEnumFromJSON(json['layout']),
        'deniedAction': !exists(json, 'denied_action') ? undefined : DeniedActionEnumFromJSON(json['denied_action']),
        'authentication': !exists(json, 'authentication') ? undefined : AuthenticationEnumFromJSON(json['authentication']),
    };
}
export function PatchedFlowRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'title': value.title,
        'designation': FlowDesignationEnumToJSON(value.designation),
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'compatibility_mode': value.compatibilityMode,
        'layout': FlowLayoutEnumToJSON(value.layout),
        'denied_action': DeniedActionEnumToJSON(value.deniedAction),
        'authentication': AuthenticationEnumToJSON(value.authentication),
    };
}
