// @goauthentik/api/dist/esm/models/PatchedNotificationRuleRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { SeverityEnumFromJSON, SeverityEnumToJSON, } from './SeverityEnum';
/**
 * Check if a given object implements the PatchedNotificationRuleRequest interface.
 */
export function instanceOfPatchedNotificationRuleRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedNotificationRuleRequestFromJSON(json) {
    return PatchedNotificationRuleRequestFromJSONTyped(json, false);
}
export function PatchedNotificationRuleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'transports': !exists(json, 'transports') ? undefined : json['transports'],
        'severity': !exists(json, 'severity') ? undefined : SeverityEnumFromJSON(json['severity']),
        'group': !exists(json, 'group') ? undefined : json['group'],
    };
}
export function PatchedNotificationRuleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'transports': value.transports,
        'severity': SeverityEnumToJSON(value.severity),
        'group': value.group,
    };
}
