// @goauthentik/api/dist/esm/models/PatchedOAuth2ProviderRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ClientTypeEnumFromJSON, ClientTypeEnumToJSON, } from './ClientTypeEnum';
import { IssuerModeEnumFromJSON, IssuerModeEnumToJSON, } from './IssuerModeEnum';
import { SubModeEnumFromJSON, SubModeEnumToJSON, } from './SubModeEnum';
/**
 * Check if a given object implements the PatchedOAuth2ProviderRequest interface.
 */
export function instanceOfPatchedOAuth2ProviderRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedOAuth2ProviderRequestFromJSON(json) {
    return PatchedOAuth2ProviderRequestFromJSONTyped(json, false);
}
export function PatchedOAuth2ProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': !exists(json, 'authorization_flow') ? undefined : json['authorization_flow'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'clientType': !exists(json, 'client_type') ? undefined : ClientTypeEnumFromJSON(json['client_type']),
        'clientId': !exists(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !exists(json, 'client_secret') ? undefined : json['client_secret'],
        'accessCodeValidity': !exists(json, 'access_code_validity') ? undefined : json['access_code_validity'],
        'accessTokenValidity': !exists(json, 'access_token_validity') ? undefined : json['access_token_validity'],
        'refreshTokenValidity': !exists(json, 'refresh_token_validity') ? undefined : json['refresh_token_validity'],
        'includeClaimsInIdToken': !exists(json, 'include_claims_in_id_token') ? undefined : json['include_claims_in_id_token'],
        'signingKey': !exists(json, 'signing_key') ? undefined : json['signing_key'],
        'redirectUris': !exists(json, 'redirect_uris') ? undefined : json['redirect_uris'],
        'subMode': !exists(json, 'sub_mode') ? undefined : SubModeEnumFromJSON(json['sub_mode']),
        'issuerMode': !exists(json, 'issuer_mode') ? undefined : IssuerModeEnumFromJSON(json['issuer_mode']),
        'jwksSources': !exists(json, 'jwks_sources') ? undefined : json['jwks_sources'],
    };
}
export function PatchedOAuth2ProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'client_type': ClientTypeEnumToJSON(value.clientType),
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'access_code_validity': value.accessCodeValidity,
        'access_token_validity': value.accessTokenValidity,
        'refresh_token_validity': value.refreshTokenValidity,
        'include_claims_in_id_token': value.includeClaimsInIdToken,
        'signing_key': value.signingKey,
        'redirect_uris': value.redirectUris,
        'sub_mode': SubModeEnumToJSON(value.subMode),
        'issuer_mode': IssuerModeEnumToJSON(value.issuerMode),
        'jwks_sources': value.jwksSources,
    };
}
