// @goauthentik/api/dist/esm/models/PatchedNotificationTransportRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { NotificationTransportModeEnumFromJSON, NotificationTransportModeEnumToJSON, } from './NotificationTransportModeEnum';
/**
 * Check if a given object implements the PatchedNotificationTransportRequest interface.
 */
export function instanceOfPatchedNotificationTransportRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedNotificationTransportRequestFromJSON(json) {
    return PatchedNotificationTransportRequestFromJSONTyped(json, false);
}
export function PatchedNotificationTransportRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'mode': !exists(json, 'mode') ? undefined : NotificationTransportModeEnumFromJSON(json['mode']),
        'webhookUrl': !exists(json, 'webhook_url') ? undefined : json['webhook_url'],
        'webhookMapping': !exists(json, 'webhook_mapping') ? undefined : json['webhook_mapping'],
        'sendOnce': !exists(json, 'send_once') ? undefined : json['send_once'],
    };
}
export function PatchedNotificationTransportRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'mode': NotificationTransportModeEnumToJSON(value.mode),
        'webhook_url': value.webhookUrl,
        'webhook_mapping': value.webhookMapping,
        'send_once': value.sendOnce,
    };
}
