// @goauthentik/api/dist/esm/models/PatchedOAuthSourceRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { PolicyEngineModeFromJSON, PolicyEngineModeToJSON, } from './PolicyEngineMode';
import { ProviderTypeEnumFromJSON, ProviderTypeEnumToJSON, } from './ProviderTypeEnum';
import { UserMatchingModeEnumFromJSON, UserMatchingModeEnumToJSON, } from './UserMatchingModeEnum';
/**
 * Check if a given object implements the PatchedOAuthSourceRequest interface.
 */
export function instanceOfPatchedOAuthSourceRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedOAuthSourceRequestFromJSON(json) {
    return PatchedOAuthSourceRequestFromJSONTyped(json, false);
}
export function PatchedOAuthSourceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'slug': !exists(json, 'slug') ? undefined : json['slug'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'enrollmentFlow': !exists(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'userMatchingMode': !exists(json, 'user_matching_mode') ? undefined : UserMatchingModeEnumFromJSON(json['user_matching_mode']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
        'providerType': !exists(json, 'provider_type') ? undefined : ProviderTypeEnumFromJSON(json['provider_type']),
        'requestTokenUrl': !exists(json, 'request_token_url') ? undefined : json['request_token_url'],
        'authorizationUrl': !exists(json, 'authorization_url') ? undefined : json['authorization_url'],
        'accessTokenUrl': !exists(json, 'access_token_url') ? undefined : json['access_token_url'],
        'profileUrl': !exists(json, 'profile_url') ? undefined : json['profile_url'],
        'consumerKey': !exists(json, 'consumer_key') ? undefined : json['consumer_key'],
        'consumerSecret': !exists(json, 'consumer_secret') ? undefined : json['consumer_secret'],
        'additionalScopes': !exists(json, 'additional_scopes') ? undefined : json['additional_scopes'],
        'oidcWellKnownUrl': !exists(json, 'oidc_well_known_url') ? undefined : json['oidc_well_known_url'],
        'oidcJwksUrl': !exists(json, 'oidc_jwks_url') ? undefined : json['oidc_jwks_url'],
        'oidcJwks': !exists(json, 'oidc_jwks') ? undefined : json['oidc_jwks'],
    };
}
export function PatchedOAuthSourceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'enabled': value.enabled,
        'authentication_flow': value.authenticationFlow,
        'enrollment_flow': value.enrollmentFlow,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'user_matching_mode': UserMatchingModeEnumToJSON(value.userMatchingMode),
        'user_path_template': value.userPathTemplate,
        'provider_type': ProviderTypeEnumToJSON(value.providerType),
        'request_token_url': value.requestTokenUrl,
        'authorization_url': value.authorizationUrl,
        'access_token_url': value.accessTokenUrl,
        'profile_url': value.profileUrl,
        'consumer_key': value.consumerKey,
        'consumer_secret': value.consumerSecret,
        'additional_scopes': value.additionalScopes,
        'oidc_well_known_url': value.oidcWellKnownUrl,
        'oidc_jwks_url': value.oidcJwksUrl,
        'oidc_jwks': value.oidcJwks,
    };
}
