// @goauthentik/api/dist/esm/models/PatchedBlueprintInstanceRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedBlueprintInstanceRequest interface.
 */
export function instanceOfPatchedBlueprintInstanceRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedBlueprintInstanceRequestFromJSON(json) {
    return PatchedBlueprintInstanceRequestFromJSONTyped(json, false);
}
export function PatchedBlueprintInstanceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'path': !exists(json, 'path') ? undefined : json['path'],
        'context': !exists(json, 'context') ? undefined : json['context'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'content': !exists(json, 'content') ? undefined : json['content'],
    };
}
export function PatchedBlueprintInstanceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
