// @goauthentik/api/dist/esm/models/PatchedApplicationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { PolicyEngineModeFromJSON, PolicyEngineModeToJSON, } from './PolicyEngineMode';
/**
 * Check if a given object implements the PatchedApplicationRequest interface.
 */
export function instanceOfPatchedApplicationRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedApplicationRequestFromJSON(json) {
    return PatchedApplicationRequestFromJSONTyped(json, false);
}
export function PatchedApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'slug': !exists(json, 'slug') ? undefined : json['slug'],
        'provider': !exists(json, 'provider') ? undefined : json['provider'],
        'backchannelProviders': !exists(json, 'backchannel_providers') ? undefined : json['backchannel_providers'],
        'openInNewTab': !exists(json, 'open_in_new_tab') ? undefined : json['open_in_new_tab'],
        'metaLaunchUrl': !exists(json, 'meta_launch_url') ? undefined : json['meta_launch_url'],
        'metaDescription': !exists(json, 'meta_description') ? undefined : json['meta_description'],
        'metaPublisher': !exists(json, 'meta_publisher') ? undefined : json['meta_publisher'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'group': !exists(json, 'group') ? undefined : json['group'],
    };
}
export function PatchedApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'provider': value.provider,
        'backchannel_providers': value.backchannelProviders,
        'open_in_new_tab': value.openInNewTab,
        'meta_launch_url': value.metaLaunchUrl,
        'meta_description': value.metaDescription,
        'meta_publisher': value.metaPublisher,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'group': value.group,
    };
}
