// @goauthentik/api/dist/esm/models/PatchedSAMLPropertyMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedSAMLPropertyMappingRequest interface.
 */
export function instanceOfPatchedSAMLPropertyMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedSAMLPropertyMappingRequestFromJSON(json) {
    return PatchedSAMLPropertyMappingRequestFromJSONTyped(json, false);
}
export function PatchedSAMLPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'samlName': !exists(json, 'saml_name') ? undefined : json['saml_name'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}
export function PatchedSAMLPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
