// @goauthentik/api/dist/esm/models/PatchedDummyPolicyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedDummyPolicyRequest interface.
 */
export function instanceOfPatchedDummyPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedDummyPolicyRequestFromJSON(json) {
    return PatchedDummyPolicyRequestFromJSONTyped(json, false);
}
export function PatchedDummyPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'result': !exists(json, 'result') ? undefined : json['result'],
        'waitMin': !exists(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !exists(json, 'wait_max') ? undefined : json['wait_max'],
    };
}
export function PatchedDummyPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
