// @goauthentik/api/dist/esm/models/PatchedPlexSourceConnectionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedPlexSourceConnectionRequest interface.
 */
export function instanceOfPatchedPlexSourceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedPlexSourceConnectionRequestFromJSON(json) {
    return PatchedPlexSourceConnectionRequestFromJSONTyped(json, false);
}
export function PatchedPlexSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'identifier': !exists(json, 'identifier') ? undefined : json['identifier'],
        'plexToken': !exists(json, 'plex_token') ? undefined : json['plex_token'],
    };
}
export function PatchedPlexSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
