// @goauthentik/api/dist/esm/models/PatchedCertificateKeyPairRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedCertificateKeyPairRequest interface.
 */
export function instanceOfPatchedCertificateKeyPairRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedCertificateKeyPairRequestFromJSON(json) {
    return PatchedCertificateKeyPairRequestFromJSONTyped(json, false);
}
export function PatchedCertificateKeyPairRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'certificateData': !exists(json, 'certificate_data') ? undefined : json['certificate_data'],
        'keyData': !exists(json, 'key_data') ? undefined : json['key_data'],
    };
}
export function PatchedCertificateKeyPairRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'certificate_data': value.certificateData,
        'key_data': value.keyData,
    };
}
