// @goauthentik/api/dist/esm/models/PatchedRACPropertyMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedRACPropertyMappingRequest interface.
 */
export function instanceOfPatchedRACPropertyMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedRACPropertyMappingRequestFromJSON(json) {
    return PatchedRACPropertyMappingRequestFromJSONTyped(json, false);
}
export function PatchedRACPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'staticSettings': !exists(json, 'static_settings') ? undefined : json['static_settings'],
    };
}
export function PatchedRACPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'static_settings': value.staticSettings,
    };
}
