// @goauthentik/api/dist/esm/models/PatchedSMSDeviceRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedSMSDeviceRequest interface.
 */
export function instanceOfPatchedSMSDeviceRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedSMSDeviceRequestFromJSON(json) {
    return PatchedSMSDeviceRequestFromJSONTyped(json, false);
}
export function PatchedSMSDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}
export function PatchedSMSDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
