// @goauthentik/api/dist/esm/models/PatchedTenantRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedTenantRequest interface.
 */
export function instanceOfPatchedTenantRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedTenantRequestFromJSON(json) {
    return PatchedTenantRequestFromJSONTyped(json, false);
}
export function PatchedTenantRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'schemaName': !exists(json, 'schema_name') ? undefined : json['schema_name'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'ready': !exists(json, 'ready') ? undefined : json['ready'],
    };
}
export function PatchedTenantRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
