// @goauthentik/api/dist/esm/models/PatchedOutpostRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { OutpostTypeEnumFromJSON, OutpostTypeEnumToJSON, } from './OutpostTypeEnum';
/**
 * Check if a given object implements the PatchedOutpostRequest interface.
 */
export function instanceOfPatchedOutpostRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedOutpostRequestFromJSON(json) {
    return PatchedOutpostRequestFromJSONTyped(json, false);
}
export function PatchedOutpostRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'type': !exists(json, 'type') ? undefined : OutpostTypeEnumFromJSON(json['type']),
        'providers': !exists(json, 'providers') ? undefined : json['providers'],
        'serviceConnection': !exists(json, 'service_connection') ? undefined : json['service_connection'],
        'config': !exists(json, 'config') ? undefined : json['config'],
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
    };
}
export function PatchedOutpostRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': OutpostTypeEnumToJSON(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
