// @goauthentik/api/dist/esm/models/PatchedExpressionPolicyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedExpressionPolicyRequest interface.
 */
export function instanceOfPatchedExpressionPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedExpressionPolicyRequestFromJSON(json) {
    return PatchedExpressionPolicyRequestFromJSONTyped(json, false);
}
export function PatchedExpressionPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'executionLogging': !exists(json, 'execution_logging') ? undefined : json['execution_logging'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
    };
}
export function PatchedExpressionPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'expression': value.expression,
    };
}
