// @goauthentik/api/dist/esm/models/PatchedAuthenticatorStaticStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the PatchedAuthenticatorStaticStageRequest interface.
 */
export function instanceOfPatchedAuthenticatorStaticStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedAuthenticatorStaticStageRequestFromJSON(json) {
    return PatchedAuthenticatorStaticStageRequestFromJSONTyped(json, false);
}
export function PatchedAuthenticatorStaticStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'tokenCount': !exists(json, 'token_count') ? undefined : json['token_count'],
        'tokenLength': !exists(json, 'token_length') ? undefined : json['token_length'],
    };
}
export function PatchedAuthenticatorStaticStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'token_count': value.tokenCount,
        'token_length': value.tokenLength,
    };
}
