// @goauthentik/api/dist/esm/models/PatchedConnectionTokenRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedConnectionTokenRequest interface.
 */
export function instanceOfPatchedConnectionTokenRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedConnectionTokenRequestFromJSON(json) {
    return PatchedConnectionTokenRequestFromJSONTyped(json, false);
}
export function PatchedConnectionTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'provider': !exists(json, 'provider') ? undefined : json['provider'],
    };
}
export function PatchedConnectionTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
