// @goauthentik/api/dist/esm/models/PatchedConsentStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ConsentStageModeEnumFromJSON, ConsentStageModeEnumToJSON, } from './ConsentStageModeEnum';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the PatchedConsentStageRequest interface.
 */
export function instanceOfPatchedConsentStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedConsentStageRequestFromJSON(json) {
    return PatchedConsentStageRequestFromJSONTyped(json, false);
}
export function PatchedConsentStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'mode': !exists(json, 'mode') ? undefined : ConsentStageModeEnumFromJSON(json['mode']),
        'consentExpireIn': !exists(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}
export function PatchedConsentStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'mode': ConsentStageModeEnumToJSON(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
