// @goauthentik/api/dist/esm/models/PatchedLDAPPropertyMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedLDAPPropertyMappingRequest interface.
 */
export function instanceOfPatchedLDAPPropertyMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedLDAPPropertyMappingRequestFromJSON(json) {
    return PatchedLDAPPropertyMappingRequestFromJSONTyped(json, false);
}
export function PatchedLDAPPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'objectField': !exists(json, 'object_field') ? undefined : json['object_field'],
    };
}
export function PatchedLDAPPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
