// @goauthentik/api/dist/esm/models/PatchedScopeMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedScopeMappingRequest interface.
 */
export function instanceOfPatchedScopeMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedScopeMappingRequestFromJSON(json) {
    return PatchedScopeMappingRequestFromJSONTyped(json, false);
}
export function PatchedScopeMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'scopeName': !exists(json, 'scope_name') ? undefined : json['scope_name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
    };
}
export function PatchedScopeMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
