// @goauthentik/api/dist/esm/models/PatchedPromptStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the PatchedPromptStageRequest interface.
 */
export function instanceOfPatchedPromptStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedPromptStageRequestFromJSON(json) {
    return PatchedPromptStageRequestFromJSONTyped(json, false);
}
export function PatchedPromptStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'fields': !exists(json, 'fields') ? undefined : json['fields'],
        'validationPolicies': !exists(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}
export function PatchedPromptStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
