// @goauthentik/api/dist/esm/models/PatchedEndpointRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { AuthModeEnumFromJSON, AuthModeEnumToJSON, } from './AuthModeEnum';
import { ProtocolEnumFromJSON, ProtocolEnumToJSON, } from './ProtocolEnum';
/**
 * Check if a given object implements the PatchedEndpointRequest interface.
 */
export function instanceOfPatchedEndpointRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedEndpointRequestFromJSON(json) {
    return PatchedEndpointRequestFromJSONTyped(json, false);
}
export function PatchedEndpointRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'provider': !exists(json, 'provider') ? undefined : json['provider'],
        'protocol': !exists(json, 'protocol') ? undefined : ProtocolEnumFromJSON(json['protocol']),
        'host': !exists(json, 'host') ? undefined : json['host'],
        'settings': !exists(json, 'settings') ? undefined : json['settings'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'authMode': !exists(json, 'auth_mode') ? undefined : AuthModeEnumFromJSON(json['auth_mode']),
        'maximumConnections': !exists(json, 'maximum_connections') ? undefined : json['maximum_connections'],
    };
}
export function PatchedEndpointRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'provider': value.provider,
        'protocol': ProtocolEnumToJSON(value.protocol),
        'host': value.host,
        'settings': value.settings,
        'property_mappings': value.propertyMappings,
        'auth_mode': AuthModeEnumToJSON(value.authMode),
        'maximum_connections': value.maximumConnections,
    };
}
