// @goauthentik/api/dist/esm/models/PatchedSCIMProviderRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedSCIMProviderRequest interface.
 */
export function instanceOfPatchedSCIMProviderRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedSCIMProviderRequestFromJSON(json) {
    return PatchedSCIMProviderRequestFromJSONTyped(json, false);
}
export function PatchedSCIMProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'propertyMappings': !exists(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !exists(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'url': !exists(json, 'url') ? undefined : json['url'],
        'token': !exists(json, 'token') ? undefined : json['token'],
        'excludeUsersServiceAccount': !exists(json, 'exclude_users_service_account') ? undefined : json['exclude_users_service_account'],
        'filterGroup': !exists(json, 'filter_group') ? undefined : json['filter_group'],
    };
}
export function PatchedSCIMProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
        'url': value.url,
        'token': value.token,
        'exclude_users_service_account': value.excludeUsersServiceAccount,
        'filter_group': value.filterGroup,
    };
}
