// @goauthentik/api/dist/esm/models/PatchedFlowStageBindingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { InvalidResponseActionEnumFromJSON, InvalidResponseActionEnumToJSON, } from './InvalidResponseActionEnum';
import { PolicyEngineModeFromJSON, PolicyEngineModeToJSON, } from './PolicyEngineMode';
/**
 * Check if a given object implements the PatchedFlowStageBindingRequest interface.
 */
export function instanceOfPatchedFlowStageBindingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedFlowStageBindingRequestFromJSON(json) {
    return PatchedFlowStageBindingRequestFromJSONTyped(json, false);
}
export function PatchedFlowStageBindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'target': !exists(json, 'target') ? undefined : json['target'],
        'stage': !exists(json, 'stage') ? undefined : json['stage'],
        'evaluateOnPlan': !exists(json, 'evaluate_on_plan') ? undefined : json['evaluate_on_plan'],
        'reEvaluatePolicies': !exists(json, 're_evaluate_policies') ? undefined : json['re_evaluate_policies'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'invalidResponseAction': !exists(json, 'invalid_response_action') ? undefined : InvalidResponseActionEnumFromJSON(json['invalid_response_action']),
    };
}
export function PatchedFlowStageBindingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'target': value.target,
        'stage': value.stage,
        'evaluate_on_plan': value.evaluateOnPlan,
        're_evaluate_policies': value.reEvaluatePolicies,
        'order': value.order,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'invalid_response_action': InvalidResponseActionEnumToJSON(value.invalidResponseAction),
    };
}
