// @goauthentik/api/dist/esm/models/PatchedPermissionAssignRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ModelEnumFromJSON, ModelEnumToJSON, } from './ModelEnum';
/**
 * Check if a given object implements the PatchedPermissionAssignRequest interface.
 */
export function instanceOfPatchedPermissionAssignRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedPermissionAssignRequestFromJSON(json) {
    return PatchedPermissionAssignRequestFromJSONTyped(json, false);
}
export function PatchedPermissionAssignRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissions': !exists(json, 'permissions') ? undefined : json['permissions'],
        'model': !exists(json, 'model') ? undefined : ModelEnumFromJSON(json['model']),
        'objectPk': !exists(json, 'object_pk') ? undefined : json['object_pk'],
    };
}
export function PatchedPermissionAssignRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': value.permissions,
        'model': ModelEnumToJSON(value.model),
        'object_pk': value.objectPk,
    };
}
