// @goauthentik/api/dist/esm/models/PatchedPolicyBindingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedPolicyBindingRequest interface.
 */
export function instanceOfPatchedPolicyBindingRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedPolicyBindingRequestFromJSON(json) {
    return PatchedPolicyBindingRequestFromJSONTyped(json, false);
}
export function PatchedPolicyBindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'policy': !exists(json, 'policy') ? undefined : json['policy'],
        'group': !exists(json, 'group') ? undefined : json['group'],
        'user': !exists(json, 'user') ? undefined : json['user'],
        'target': !exists(json, 'target') ? undefined : json['target'],
        'negate': !exists(json, 'negate') ? undefined : json['negate'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'timeout': !exists(json, 'timeout') ? undefined : json['timeout'],
        'failureResult': !exists(json, 'failure_result') ? undefined : json['failure_result'],
    };
}
export function PatchedPolicyBindingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'policy': value.policy,
        'group': value.group,
        'user': value.user,
        'target': value.target,
        'negate': value.negate,
        'enabled': value.enabled,
        'order': value.order,
        'timeout': value.timeout,
        'failure_result': value.failureResult,
    };
}
