// @goauthentik/api/dist/esm/models/PatchedUserWriteStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
import { UserCreationModeEnumFromJSON, UserCreationModeEnumToJSON, } from './UserCreationModeEnum';
import { UserTypeEnumFromJSON, UserTypeEnumToJSON, } from './UserTypeEnum';
/**
 * Check if a given object implements the PatchedUserWriteStageRequest interface.
 */
export function instanceOfPatchedUserWriteStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedUserWriteStageRequestFromJSON(json) {
    return PatchedUserWriteStageRequestFromJSONTyped(json, false);
}
export function PatchedUserWriteStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'userCreationMode': !exists(json, 'user_creation_mode') ? undefined : UserCreationModeEnumFromJSON(json['user_creation_mode']),
        'createUsersAsInactive': !exists(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !exists(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !exists(json, 'user_type') ? undefined : UserTypeEnumFromJSON(json['user_type']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}
export function PatchedUserWriteStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'user_creation_mode': UserCreationModeEnumToJSON(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': UserTypeEnumToJSON(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
