// @goauthentik/api/dist/esm/models/PatchedSAMLSourceRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { BindingTypeEnumFromJSON, BindingTypeEnumToJSON, } from './BindingTypeEnum';
import { DigestAlgorithmEnumFromJSON, DigestAlgorithmEnumToJSON, } from './DigestAlgorithmEnum';
import { NameIdPolicyEnumFromJSON, NameIdPolicyEnumToJSON, } from './NameIdPolicyEnum';
import { PolicyEngineModeFromJSON, PolicyEngineModeToJSON, } from './PolicyEngineMode';
import { SignatureAlgorithmEnumFromJSON, SignatureAlgorithmEnumToJSON, } from './SignatureAlgorithmEnum';
import { UserMatchingModeEnumFromJSON, UserMatchingModeEnumToJSON, } from './UserMatchingModeEnum';
/**
 * Check if a given object implements the PatchedSAMLSourceRequest interface.
 */
export function instanceOfPatchedSAMLSourceRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedSAMLSourceRequestFromJSON(json) {
    return PatchedSAMLSourceRequestFromJSONTyped(json, false);
}
export function PatchedSAMLSourceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'slug': !exists(json, 'slug') ? undefined : json['slug'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'authenticationFlow': !exists(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'enrollmentFlow': !exists(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'policyEngineMode': !exists(json, 'policy_engine_mode') ? undefined : PolicyEngineModeFromJSON(json['policy_engine_mode']),
        'userMatchingMode': !exists(json, 'user_matching_mode') ? undefined : UserMatchingModeEnumFromJSON(json['user_matching_mode']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
        'preAuthenticationFlow': !exists(json, 'pre_authentication_flow') ? undefined : json['pre_authentication_flow'],
        'issuer': !exists(json, 'issuer') ? undefined : json['issuer'],
        'ssoUrl': !exists(json, 'sso_url') ? undefined : json['sso_url'],
        'sloUrl': !exists(json, 'slo_url') ? undefined : json['slo_url'],
        'allowIdpInitiated': !exists(json, 'allow_idp_initiated') ? undefined : json['allow_idp_initiated'],
        'nameIdPolicy': !exists(json, 'name_id_policy') ? undefined : NameIdPolicyEnumFromJSON(json['name_id_policy']),
        'bindingType': !exists(json, 'binding_type') ? undefined : BindingTypeEnumFromJSON(json['binding_type']),
        'verificationKp': !exists(json, 'verification_kp') ? undefined : json['verification_kp'],
        'signingKp': !exists(json, 'signing_kp') ? undefined : json['signing_kp'],
        'digestAlgorithm': !exists(json, 'digest_algorithm') ? undefined : DigestAlgorithmEnumFromJSON(json['digest_algorithm']),
        'signatureAlgorithm': !exists(json, 'signature_algorithm') ? undefined : SignatureAlgorithmEnumFromJSON(json['signature_algorithm']),
        'temporaryUserDeleteAfter': !exists(json, 'temporary_user_delete_after') ? undefined : json['temporary_user_delete_after'],
    };
}
export function PatchedSAMLSourceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'enabled': value.enabled,
        'authentication_flow': value.authenticationFlow,
        'enrollment_flow': value.enrollmentFlow,
        'policy_engine_mode': PolicyEngineModeToJSON(value.policyEngineMode),
        'user_matching_mode': UserMatchingModeEnumToJSON(value.userMatchingMode),
        'user_path_template': value.userPathTemplate,
        'pre_authentication_flow': value.preAuthenticationFlow,
        'issuer': value.issuer,
        'sso_url': value.ssoUrl,
        'slo_url': value.sloUrl,
        'allow_idp_initiated': value.allowIdpInitiated,
        'name_id_policy': NameIdPolicyEnumToJSON(value.nameIdPolicy),
        'binding_type': BindingTypeEnumToJSON(value.bindingType),
        'verification_kp': value.verificationKp,
        'signing_kp': value.signingKp,
        'digest_algorithm': DigestAlgorithmEnumToJSON(value.digestAlgorithm),
        'signature_algorithm': SignatureAlgorithmEnumToJSON(value.signatureAlgorithm),
        'temporary_user_delete_after': value.temporaryUserDeleteAfter,
    };
}
