// @goauthentik/api/dist/esm/models/PatchedDockerServiceConnectionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedDockerServiceConnectionRequest interface.
 */
export function instanceOfPatchedDockerServiceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedDockerServiceConnectionRequestFromJSON(json) {
    return PatchedDockerServiceConnectionRequestFromJSONTyped(json, false);
}
export function PatchedDockerServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'url': !exists(json, 'url') ? undefined : json['url'],
        'tlsVerification': !exists(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !exists(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}
export function PatchedDockerServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
