// @goauthentik/api/dist/esm/models/PatchedAuthenticatorTOTPStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { DigitsEnumFromJSON, DigitsEnumToJSON, } from './DigitsEnum';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
/**
 * Check if a given object implements the PatchedAuthenticatorTOTPStageRequest interface.
 */
export function instanceOfPatchedAuthenticatorTOTPStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedAuthenticatorTOTPStageRequestFromJSON(json) {
    return PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json, false);
}
export function PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'configureFlow': !exists(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !exists(json, 'friendly_name') ? undefined : json['friendly_name'],
        'digits': !exists(json, 'digits') ? undefined : DigitsEnumFromJSON(json['digits']),
    };
}
export function PatchedAuthenticatorTOTPStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'digits': DigitsEnumToJSON(value.digits),
    };
}
