// @goauthentik/api/dist/esm/models/PatchedUserOAuthSourceConnectionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedUserOAuthSourceConnectionRequest interface.
 */
export function instanceOfPatchedUserOAuthSourceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedUserOAuthSourceConnectionRequestFromJSON(json) {
    return PatchedUserOAuthSourceConnectionRequestFromJSONTyped(json, false);
}
export function PatchedUserOAuthSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': !exists(json, 'user') ? undefined : json['user'],
        'identifier': !exists(json, 'identifier') ? undefined : json['identifier'],
        'accessToken': !exists(json, 'access_token') ? undefined : json['access_token'],
    };
}
export function PatchedUserOAuthSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
        'access_token': value.accessToken,
    };
}
