// @goauthentik/api/dist/esm/models/PatchedKubernetesServiceConnectionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PatchedKubernetesServiceConnectionRequest interface.
 */
export function instanceOfPatchedKubernetesServiceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedKubernetesServiceConnectionRequestFromJSON(json) {
    return PatchedKubernetesServiceConnectionRequestFromJSONTyped(json, false);
}
export function PatchedKubernetesServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'local': !exists(json, 'local') ? undefined : json['local'],
        'kubeconfig': !exists(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !exists(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}
export function PatchedKubernetesServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
