// @goauthentik/api/dist/esm/models/PatchedNotificationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { EventRequestFromJSON, EventRequestToJSON, } from './EventRequest';
/**
 * Check if a given object implements the PatchedNotificationRequest interface.
 */
export function instanceOfPatchedNotificationRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PatchedNotificationRequestFromJSON(json) {
    return PatchedNotificationRequestFromJSONTyped(json, false);
}
export function PatchedNotificationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'event': !exists(json, 'event') ? undefined : EventRequestFromJSON(json['event']),
        'seen': !exists(json, 'seen') ? undefined : json['seen'],
    };
}
export function PatchedNotificationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'event': EventRequestToJSON(value.event),
        'seen': value.seen,
    };
}
