// @goauthentik/api/dist/esm/models/PlexSourceConnection.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { SourceFromJSON, } from './Source';
/**
 * Check if a given object implements the PlexSourceConnection interface.
 */
export function instanceOfPlexSourceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "plexToken" in value;
    return isInstance;
}
export function PlexSourceConnectionFromJSON(json) {
    return PlexSourceConnectionFromJSONTyped(json, false);
}
export function PlexSourceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'identifier': json['identifier'],
        'plexToken': json['plex_token'],
    };
}
export function PlexSourceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
