// @goauthentik/api/dist/esm/models/PlexTokenRedeemRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the PlexTokenRedeemRequest interface.
 */
export function instanceOfPlexTokenRedeemRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "plexToken" in value;
    return isInstance;
}
export function PlexTokenRedeemRequestFromJSON(json) {
    return PlexTokenRedeemRequestFromJSONTyped(json, false);
}
export function PlexTokenRedeemRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'plexToken': json['plex_token'],
    };
}
export function PlexTokenRedeemRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'plex_token': value.plexToken,
    };
}
