// @goauthentik/api/dist/esm/models/PolicyTestResult.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the PolicyTestResult interface.
 */
export function instanceOfPolicyTestResult(value) {
    let isInstance = true;
    isInstance = isInstance && "passing" in value;
    isInstance = isInstance && "messages" in value;
    isInstance = isInstance && "logMessages" in value;
    return isInstance;
}
export function PolicyTestResultFromJSON(json) {
    return PolicyTestResultFromJSONTyped(json, false);
}
export function PolicyTestResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passing': json['passing'],
        'messages': json['messages'],
        'logMessages': json['log_messages'],
    };
}
export function PolicyTestResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passing': value.passing,
    };
}
