// @goauthentik/api/dist/esm/models/PolicyBinding.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { GroupFromJSON, } from './Group';
import { PolicyFromJSON, } from './Policy';
import { UserFromJSON, } from './User';
/**
 * Check if a given object implements the PolicyBinding interface.
 */
export function instanceOfPolicyBinding(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "policyObj" in value;
    isInstance = isInstance && "groupObj" in value;
    isInstance = isInstance && "userObj" in value;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "order" in value;
    return isInstance;
}
export function PolicyBindingFromJSON(json) {
    return PolicyBindingFromJSONTyped(json, false);
}
export function PolicyBindingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'policy': !exists(json, 'policy') ? undefined : json['policy'],
        'group': !exists(json, 'group') ? undefined : json['group'],
        'user': !exists(json, 'user') ? undefined : json['user'],
        'policyObj': PolicyFromJSON(json['policy_obj']),
        'groupObj': GroupFromJSON(json['group_obj']),
        'userObj': UserFromJSON(json['user_obj']),
        'target': json['target'],
        'negate': !exists(json, 'negate') ? undefined : json['negate'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
        'order': json['order'],
        'timeout': !exists(json, 'timeout') ? undefined : json['timeout'],
        'failureResult': !exists(json, 'failure_result') ? undefined : json['failure_result'],
    };
}
export function PolicyBindingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'policy': value.policy,
        'group': value.group,
        'user': value.user,
        'target': value.target,
        'negate': value.negate,
        'enabled': value.enabled,
        'order': value.order,
        'timeout': value.timeout,
        'failure_result': value.failureResult,
    };
}
