// @goauthentik/api/dist/esm/models/PromptRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { PromptTypeEnumFromJSON, PromptTypeEnumToJSON, } from './PromptTypeEnum';
import { StageRequestFromJSON, StageRequestToJSON, } from './StageRequest';
/**
 * Check if a given object implements the PromptRequest interface.
 */
export function instanceOfPromptRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "fieldKey" in value;
    isInstance = isInstance && "label" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
export function PromptRequestFromJSON(json) {
    return PromptRequestFromJSONTyped(json, false);
}
export function PromptRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'fieldKey': json['field_key'],
        'label': json['label'],
        'type': PromptTypeEnumFromJSON(json['type']),
        'required': !exists(json, 'required') ? undefined : json['required'],
        'placeholder': !exists(json, 'placeholder') ? undefined : json['placeholder'],
        'initialValue': !exists(json, 'initial_value') ? undefined : json['initial_value'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'promptstageSet': !exists(json, 'promptstage_set') ? undefined : (json['promptstage_set'].map(StageRequestFromJSON)),
        'subText': !exists(json, 'sub_text') ? undefined : json['sub_text'],
        'placeholderExpression': !exists(json, 'placeholder_expression') ? undefined : json['placeholder_expression'],
        'initialValueExpression': !exists(json, 'initial_value_expression') ? undefined : json['initial_value_expression'],
    };
}
export function PromptRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'field_key': value.fieldKey,
        'label': value.label,
        'type': PromptTypeEnumToJSON(value.type),
        'required': value.required,
        'placeholder': value.placeholder,
        'initial_value': value.initialValue,
        'order': value.order,
        'promptstage_set': value.promptstageSet === undefined ? undefined : (value.promptstageSet.map(StageRequestToJSON)),
        'sub_text': value.subText,
        'placeholder_expression': value.placeholderExpression,
        'initial_value_expression': value.initialValueExpression,
    };
}
