// @goauthentik/api/dist/esm/models/PromptChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the PromptChallengeResponseRequest interface.
 */
export function instanceOfPromptChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
export function PromptChallengeResponseRequestFromJSON(json) {
    return PromptChallengeResponseRequestFromJSONTyped(json, false);
}
export function PromptChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return Object.assign(Object.assign({}, json), { 'component': !exists(json, 'component') ? undefined : json['component'] });
}
export function PromptChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return Object.assign(Object.assign({}, value), { 'component': value.component });
}
