// @goauthentik/api/dist/esm/models/PropertyMappingPreview.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the PropertyMappingPreview interface.
 */
export function instanceOfPropertyMappingPreview(value) {
    let isInstance = true;
    isInstance = isInstance && "preview" in value;
    return isInstance;
}
export function PropertyMappingPreviewFromJSON(json) {
    return PropertyMappingPreviewFromJSONTyped(json, false);
}
export function PropertyMappingPreviewFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'preview': json['preview'],
    };
}
export function PropertyMappingPreviewToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
