// @goauthentik/api/dist/esm/models/RACPropertyMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the RACPropertyMappingRequest interface.
 */
export function instanceOfRACPropertyMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "staticSettings" in value;
    return isInstance;
}
export function RACPropertyMappingRequestFromJSON(json) {
    return RACPropertyMappingRequestFromJSONTyped(json, false);
}
export function RACPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': !exists(json, 'expression') ? undefined : json['expression'],
        'staticSettings': json['static_settings'],
    };
}
export function RACPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'static_settings': value.staticSettings,
    };
}
