// @goauthentik/api/dist/esm/models/RoleAssignedObjectPermission.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { RoleObjectPermissionFromJSON, RoleObjectPermissionToJSON, } from './RoleObjectPermission';
/**
 * Check if a given object implements the RoleAssignedObjectPermission interface.
 */
export function instanceOfRoleAssignedObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "rolePk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "permissions" in value;
    return isInstance;
}
export function RoleAssignedObjectPermissionFromJSON(json) {
    return RoleAssignedObjectPermissionFromJSONTyped(json, false);
}
export function RoleAssignedObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'rolePk': json['role_pk'],
        'name': json['name'],
        'permissions': (json['permissions'].map(RoleObjectPermissionFromJSON)),
    };
}
export function RoleAssignedObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': (value.permissions.map(RoleObjectPermissionToJSON)),
    };
}
