// @goauthentik/api/dist/esm/models/SAMLMetadata.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the SAMLMetadata interface.
 */
export function instanceOfSAMLMetadata(value) {
    let isInstance = true;
    isInstance = isInstance && "metadata" in value;
    isInstance = isInstance && "downloadUrl" in value;
    return isInstance;
}
export function SAMLMetadataFromJSON(json) {
    return SAMLMetadataFromJSONTyped(json, false);
}
export function SAMLMetadataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'metadata': json['metadata'],
        'downloadUrl': json['download_url'],
    };
}
export function SAMLMetadataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
