// @goauthentik/api/dist/esm/models/SCIMSyncStatus.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { SystemTaskFromJSON, } from './SystemTask';
/**
 * Check if a given object implements the SCIMSyncStatus interface.
 */
export function instanceOfSCIMSyncStatus(value) {
    let isInstance = true;
    isInstance = isInstance && "isRunning" in value;
    isInstance = isInstance && "tasks" in value;
    return isInstance;
}
export function SCIMSyncStatusFromJSON(json) {
    return SCIMSyncStatusFromJSONTyped(json, false);
}
export function SCIMSyncStatusFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'isRunning': json['is_running'],
        'tasks': (json['tasks'].map(SystemTaskFromJSON)),
    };
}
export function SCIMSyncStatusToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
