// @goauthentik/api/dist/esm/models/ScopeMappingRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the ScopeMappingRequest interface.
 */
export function instanceOfScopeMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "scopeName" in value;
    return isInstance;
}
export function ScopeMappingRequestFromJSON(json) {
    return ScopeMappingRequestFromJSONTyped(json, false);
}
export function ScopeMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !exists(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'scopeName': json['scope_name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
    };
}
export function ScopeMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
