// @goauthentik/api/dist/esm/models/SelectableStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the SelectableStage interface.
 */
export function instanceOfSelectableStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function SelectableStageFromJSON(json) {
    return SelectableStageFromJSONTyped(json, false);
}
export function SelectableStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'verboseName': json['verbose_name'],
        'metaModelName': json['meta_model_name'],
    };
}
export function SelectableStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pk': value.pk,
        'name': value.name,
        'verbose_name': value.verboseName,
        'meta_model_name': value.metaModelName,
    };
}
