// @goauthentik/api/dist/esm/models/ServiceConnectionState.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the ServiceConnectionState interface.
 */
export function instanceOfServiceConnectionState(value) {
    let isInstance = true;
    isInstance = isInstance && "healthy" in value;
    isInstance = isInstance && "version" in value;
    return isInstance;
}
export function ServiceConnectionStateFromJSON(json) {
    return ServiceConnectionStateFromJSONTyped(json, false);
}
export function ServiceConnectionStateFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'healthy': json['healthy'],
        'version': json['version'],
    };
}
export function ServiceConnectionStateToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
