// @goauthentik/api/dist/esm/models/SessionUser.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { UserSelfFromJSON, UserSelfToJSON, } from './UserSelf';
/**
 * Check if a given object implements the SessionUser interface.
 */
export function instanceOfSessionUser(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
export function SessionUserFromJSON(json) {
    return SessionUserFromJSONTyped(json, false);
}
export function SessionUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': UserSelfFromJSON(json['user']),
        'original': !exists(json, 'original') ? undefined : UserSelfFromJSON(json['original']),
    };
}
export function SessionUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': UserSelfToJSON(value.user),
        'original': UserSelfToJSON(value.original),
    };
}
