// @goauthentik/api/dist/esm/models/Settings.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the Settings interface.
 */
export function instanceOfSettings(value) {
    let isInstance = true;
    return isInstance;
}
export function SettingsFromJSON(json) {
    return SettingsFromJSONTyped(json, false);
}
export function SettingsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'avatars': !exists(json, 'avatars') ? undefined : json['avatars'],
        'defaultUserChangeName': !exists(json, 'default_user_change_name') ? undefined : json['default_user_change_name'],
        'defaultUserChangeEmail': !exists(json, 'default_user_change_email') ? undefined : json['default_user_change_email'],
        'defaultUserChangeUsername': !exists(json, 'default_user_change_username') ? undefined : json['default_user_change_username'],
        'eventRetention': !exists(json, 'event_retention') ? undefined : json['event_retention'],
        'footerLinks': !exists(json, 'footer_links') ? undefined : json['footer_links'],
        'gdprCompliance': !exists(json, 'gdpr_compliance') ? undefined : json['gdpr_compliance'],
        'impersonation': !exists(json, 'impersonation') ? undefined : json['impersonation'],
    };
}
export function SettingsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'avatars': value.avatars,
        'default_user_change_name': value.defaultUserChangeName,
        'default_user_change_email': value.defaultUserChangeEmail,
        'default_user_change_username': value.defaultUserChangeUsername,
        'event_retention': value.eventRetention,
        'footer_links': value.footerLinks,
        'gdpr_compliance': value.gdprCompliance,
        'impersonation': value.impersonation,
    };
}
