// @goauthentik/api/dist/esm/models/ShellChallenge.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ChallengeChoicesFromJSON, ChallengeChoicesToJSON, } from './ChallengeChoices';
import { ContextualFlowInfoFromJSON, ContextualFlowInfoToJSON, } from './ContextualFlowInfo';
/**
 * Check if a given object implements the ShellChallenge interface.
 */
export function instanceOfShellChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "body" in value;
    return isInstance;
}
export function ShellChallengeFromJSON(json) {
    return ShellChallengeFromJSONTyped(json, false);
}
export function ShellChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': ChallengeChoicesFromJSON(json['type']),
        'flowInfo': !exists(json, 'flow_info') ? undefined : ContextualFlowInfoFromJSON(json['flow_info']),
        'component': !exists(json, 'component') ? undefined : json['component'],
        'responseErrors': !exists(json, 'response_errors') ? undefined : json['response_errors'],
        'body': json['body'],
    };
}
export function ShellChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': ChallengeChoicesToJSON(value.type),
        'flow_info': ContextualFlowInfoToJSON(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'body': value.body,
    };
}
