// @goauthentik/api/dist/esm/models/SMSDevice.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the SMSDevice interface.
 */
export function instanceOfSMSDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "phoneNumber" in value;
    return isInstance;
}
export function SMSDeviceFromJSON(json) {
    return SMSDeviceFromJSONTyped(json, false);
}
export function SMSDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'pk': json['pk'],
        'phoneNumber': json['phone_number'],
    };
}
export function SMSDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
