// @goauthentik/api/dist/esm/models/SourceType.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the SourceType interface.
 */
export function instanceOfSourceType(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "urlsCustomizable" in value;
    isInstance = isInstance && "requestTokenUrl" in value;
    isInstance = isInstance && "authorizationUrl" in value;
    isInstance = isInstance && "accessTokenUrl" in value;
    isInstance = isInstance && "profileUrl" in value;
    isInstance = isInstance && "oidcWellKnownUrl" in value;
    isInstance = isInstance && "oidcJwksUrl" in value;
    return isInstance;
}
export function SourceTypeFromJSON(json) {
    return SourceTypeFromJSONTyped(json, false);
}
export function SourceTypeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'verboseName': json['verbose_name'],
        'urlsCustomizable': json['urls_customizable'],
        'requestTokenUrl': json['request_token_url'],
        'authorizationUrl': json['authorization_url'],
        'accessTokenUrl': json['access_token_url'],
        'profileUrl': json['profile_url'],
        'oidcWellKnownUrl': json['oidc_well_known_url'],
        'oidcJwksUrl': json['oidc_jwks_url'],
    };
}
export function SourceTypeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'verbose_name': value.verboseName,
        'urls_customizable': value.urlsCustomizable,
    };
}
