// @goauthentik/api/dist/esm/models/StaticDevice.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { StaticDeviceTokenFromJSON, } from './StaticDeviceToken';
/**
 * Check if a given object implements the StaticDevice interface.
 */
export function instanceOfStaticDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "tokenSet" in value;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
export function StaticDeviceFromJSON(json) {
    return StaticDeviceFromJSONTyped(json, false);
}
export function StaticDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'tokenSet': (json['token_set'].map(StaticDeviceTokenFromJSON)),
        'pk': json['pk'],
    };
}
export function StaticDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
