// @goauthentik/api/dist/esm/models/StaticDeviceToken.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the StaticDeviceToken interface.
 */
export function instanceOfStaticDeviceToken(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
export function StaticDeviceTokenFromJSON(json) {
    return StaticDeviceTokenFromJSONTyped(json, false);
}
export function StaticDeviceTokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'token': json['token'],
    };
}
export function StaticDeviceTokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token': value.token,
    };
}
