// @goauthentik/api/dist/esm/models/TenantAdminGroupRequestRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the TenantAdminGroupRequestRequest interface.
 */
export function instanceOfTenantAdminGroupRequestRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
export function TenantAdminGroupRequestRequestFromJSON(json) {
    return TenantAdminGroupRequestRequestFromJSONTyped(json, false);
}
export function TenantAdminGroupRequestRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
    };
}
export function TenantAdminGroupRequestRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
    };
}
