// @goauthentik/api/dist/esm/models/TokenView.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the TokenView interface.
 */
export function instanceOfTokenView(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
export function TokenViewFromJSON(json) {
    return TokenViewFromJSONTyped(json, false);
}
export function TokenViewFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
export function TokenViewToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
