// @goauthentik/api/dist/esm/models/TransactionApplicationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { ApplicationRequestFromJSON, ApplicationRequestToJSON, } from './ApplicationRequest';
import { ModelRequestFromJSON, ModelRequestToJSON, } from './ModelRequest';
import { ProviderModelEnumFromJSON, ProviderModelEnumToJSON, } from './ProviderModelEnum';
/**
 * Check if a given object implements the TransactionApplicationRequest interface.
 */
export function instanceOfTransactionApplicationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    isInstance = isInstance && "providerModel" in value;
    isInstance = isInstance && "provider" in value;
    return isInstance;
}
export function TransactionApplicationRequestFromJSON(json) {
    return TransactionApplicationRequestFromJSONTyped(json, false);
}
export function TransactionApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'app': ApplicationRequestFromJSON(json['app']),
        'providerModel': ProviderModelEnumFromJSON(json['provider_model']),
        'provider': ModelRequestFromJSON(json['provider']),
    };
}
export function TransactionApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'app': ApplicationRequestToJSON(value.app),
        'provider_model': ProviderModelEnumToJSON(value.providerModel),
        'provider': ModelRequestToJSON(value.provider),
    };
}
