// @goauthentik/api/dist/esm/models/TransactionApplicationResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the TransactionApplicationResponse interface.
 */
export function instanceOfTransactionApplicationResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "applied" in value;
    isInstance = isInstance && "logs" in value;
    return isInstance;
}
export function TransactionApplicationResponseFromJSON(json) {
    return TransactionApplicationResponseFromJSONTyped(json, false);
}
export function TransactionApplicationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'applied': json['applied'],
        'logs': json['logs'],
    };
}
export function TransactionApplicationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'applied': value.applied,
        'logs': value.logs,
    };
}
