// @goauthentik/api/dist/esm/models/UsedBy.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { UsedByActionEnumFromJSON, UsedByActionEnumToJSON, } from './UsedByActionEnum';
/**
 * Check if a given object implements the UsedBy interface.
 */
export function instanceOfUsedBy(value) {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    isInstance = isInstance && "modelName" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "action" in value;
    return isInstance;
}
export function UsedByFromJSON(json) {
    return UsedByFromJSONTyped(json, false);
}
export function UsedByFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'app': json['app'],
        'modelName': json['model_name'],
        'pk': json['pk'],
        'name': json['name'],
        'action': UsedByActionEnumFromJSON(json['action']),
    };
}
export function UsedByToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'app': value.app,
        'model_name': value.modelName,
        'pk': value.pk,
        'name': value.name,
        'action': UsedByActionEnumToJSON(value.action),
    };
}
