// @goauthentik/api/dist/esm/models/UserOAuthSourceConnection.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { SourceFromJSON, } from './Source';
/**
 * Check if a given object implements the UserOAuthSourceConnection interface.
 */
export function instanceOfUserOAuthSourceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "identifier" in value;
    return isInstance;
}
export function UserOAuthSourceConnectionFromJSON(json) {
    return UserOAuthSourceConnectionFromJSONTyped(json, false);
}
export function UserOAuthSourceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'identifier': json['identifier'],
    };
}
export function UserOAuthSourceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
    };
}
