// @goauthentik/api/dist/esm/models/UserConsent.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ApplicationFromJSON, ApplicationToJSON, } from './Application';
import { UserFromJSON, UserToJSON, } from './User';
/**
 * Check if a given object implements the UserConsent interface.
 */
export function instanceOfUserConsent(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "application" in value;
    return isInstance;
}
export function UserConsentFromJSON(json) {
    return UserConsentFromJSONTyped(json, false);
}
export function UserConsentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'expires': !exists(json, 'expires') ? undefined : (new Date(json['expires'])),
        'expiring': !exists(json, 'expiring') ? undefined : json['expiring'],
        'user': UserFromJSON(json['user']),
        'application': ApplicationFromJSON(json['application']),
        'permissions': !exists(json, 'permissions') ? undefined : json['permissions'],
    };
}
export function UserConsentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'expiring': value.expiring,
        'user': UserToJSON(value.user),
        'application': ApplicationToJSON(value.application),
        'permissions': value.permissions,
    };
}
