// @goauthentik/api/dist/esm/models/UserPasswordSetRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UserPasswordSetRequest interface.
 */
export function instanceOfUserPasswordSetRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "password" in value;
    return isInstance;
}
export function UserPasswordSetRequestFromJSON(json) {
    return UserPasswordSetRequestFromJSONTyped(json, false);
}
export function UserPasswordSetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'password': json['password'],
    };
}
export function UserPasswordSetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'password': value.password,
    };
}
