// @goauthentik/api/dist/esm/models/UserWriteStage.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetFromJSON, FlowSetToJSON, } from './FlowSet';
import { UserCreationModeEnumFromJSON, UserCreationModeEnumToJSON, } from './UserCreationModeEnum';
import { UserTypeEnumFromJSON, UserTypeEnumToJSON, } from './UserTypeEnum';
/**
 * Check if a given object implements the UserWriteStage interface.
 */
export function instanceOfUserWriteStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
export function UserWriteStageFromJSON(json) {
    return UserWriteStageFromJSONTyped(json, false);
}
export function UserWriteStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetFromJSON)),
        'userCreationMode': !exists(json, 'user_creation_mode') ? undefined : UserCreationModeEnumFromJSON(json['user_creation_mode']),
        'createUsersAsInactive': !exists(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !exists(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !exists(json, 'user_type') ? undefined : UserTypeEnumFromJSON(json['user_type']),
        'userPathTemplate': !exists(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}
export function UserWriteStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetToJSON)),
        'user_creation_mode': UserCreationModeEnumToJSON(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': UserTypeEnumToJSON(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
