// @goauthentik/api/dist/esm/models/UserObjectPermission.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UserObjectPermission interface.
 */
export function instanceOfUserObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "objectPk" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function UserObjectPermissionFromJSON(json) {
    return UserObjectPermissionFromJSONTyped(json, false);
}
export function UserObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'objectPk': json['object_pk'],
        'name': json['name'],
    };
}
export function UserObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
