// @goauthentik/api/dist/esm/models/UserSelf.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { UserSelfGroupsFromJSON, } from './UserSelfGroups';
import { UserTypeEnumFromJSON, UserTypeEnumToJSON, } from './UserTypeEnum';
/**
 * Check if a given object implements the UserSelf interface.
 */
export function instanceOfUserSelf(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isActive" in value;
    isInstance = isInstance && "isSuperuser" in value;
    isInstance = isInstance && "groups" in value;
    isInstance = isInstance && "avatar" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "settings" in value;
    isInstance = isInstance && "systemPermissions" in value;
    return isInstance;
}
export function UserSelfFromJSON(json) {
    return UserSelfFromJSONTyped(json, false);
}
export function UserSelfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': json['is_active'],
        'isSuperuser': json['is_superuser'],
        'groups': (json['groups'].map(UserSelfGroupsFromJSON)),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'avatar': json['avatar'],
        'uid': json['uid'],
        'settings': json['settings'],
        'type': !exists(json, 'type') ? undefined : UserTypeEnumFromJSON(json['type']),
        'systemPermissions': json['system_permissions'],
    };
}
export function UserSelfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'email': value.email,
        'type': UserTypeEnumToJSON(value.type),
    };
}
