// @goauthentik/api/dist/esm/models/UserLoginChallengeResponseRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the UserLoginChallengeResponseRequest interface.
 */
export function instanceOfUserLoginChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "rememberMe" in value;
    return isInstance;
}
export function UserLoginChallengeResponseRequestFromJSON(json) {
    return UserLoginChallengeResponseRequestFromJSONTyped(json, false);
}
export function UserLoginChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !exists(json, 'component') ? undefined : json['component'],
        'rememberMe': json['remember_me'],
    };
}
export function UserLoginChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'remember_me': value.rememberMe,
    };
}
