// @goauthentik/api/dist/esm/models/UserLoginStageRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { FlowSetRequestFromJSON, FlowSetRequestToJSON, } from './FlowSetRequest';
import { GeoipBindingEnumFromJSON, GeoipBindingEnumToJSON, } from './GeoipBindingEnum';
import { NetworkBindingEnumFromJSON, NetworkBindingEnumToJSON, } from './NetworkBindingEnum';
/**
 * Check if a given object implements the UserLoginStageRequest interface.
 */
export function instanceOfUserLoginStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function UserLoginStageRequestFromJSON(json) {
    return UserLoginStageRequestFromJSONTyped(json, false);
}
export function UserLoginStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !exists(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequestFromJSON)),
        'sessionDuration': !exists(json, 'session_duration') ? undefined : json['session_duration'],
        'terminateOtherSessions': !exists(json, 'terminate_other_sessions') ? undefined : json['terminate_other_sessions'],
        'rememberMeOffset': !exists(json, 'remember_me_offset') ? undefined : json['remember_me_offset'],
        'networkBinding': !exists(json, 'network_binding') ? undefined : NetworkBindingEnumFromJSON(json['network_binding']),
        'geoipBinding': !exists(json, 'geoip_binding') ? undefined : GeoipBindingEnumFromJSON(json['geoip_binding']),
    };
}
export function UserLoginStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequestToJSON)),
        'session_duration': value.sessionDuration,
        'terminate_other_sessions': value.terminateOtherSessions,
        'remember_me_offset': value.rememberMeOffset,
        'network_binding': NetworkBindingEnumToJSON(value.networkBinding),
        'geoip_binding': GeoipBindingEnumToJSON(value.geoipBinding),
    };
}
