// @goauthentik/api/dist/esm/models/UserAccountRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UserAccountRequest interface.
 */
export function instanceOfUserAccountRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
export function UserAccountRequestFromJSON(json) {
    return UserAccountRequestFromJSONTyped(json, false);
}
export function UserAccountRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
    };
}
export function UserAccountRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pk': value.pk,
    };
}
