// @goauthentik/api/dist/esm/models/User.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { UserGroupFromJSON, } from './UserGroup';
import { UserTypeEnumFromJSON, UserTypeEnumToJSON, } from './UserTypeEnum';
/**
 * Check if a given object implements the User interface.
 */
export function instanceOfUser(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isSuperuser" in value;
    isInstance = isInstance && "groupsObj" in value;
    isInstance = isInstance && "avatar" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "uuid" in value;
    return isInstance;
}
export function UserFromJSON(json) {
    return UserFromJSONTyped(json, false);
}
export function UserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !exists(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !exists(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'isSuperuser': json['is_superuser'],
        'groups': !exists(json, 'groups') ? undefined : json['groups'],
        'groupsObj': (json['groups_obj'].map(UserGroupFromJSON)),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'avatar': json['avatar'],
        'attributes': !exists(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
        'path': !exists(json, 'path') ? undefined : json['path'],
        'type': !exists(json, 'type') ? undefined : UserTypeEnumFromJSON(json['type']),
        'uuid': json['uuid'],
    };
}
export function UserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'groups': value.groups,
        'email': value.email,
        'attributes': value.attributes,
        'path': value.path,
        'type': UserTypeEnumToJSON(value.type),
    };
}
