// @goauthentik/api/dist/esm/models/UserPath.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UserPath interface.
 */
export function instanceOfUserPath(value) {
    let isInstance = true;
    isInstance = isInstance && "paths" in value;
    return isInstance;
}
export function UserPathFromJSON(json) {
    return UserPathFromJSONTyped(json, false);
}
export function UserPathFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'paths': json['paths'],
    };
}
export function UserPathToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
