// @goauthentik/api/dist/esm/models/UserServiceAccountResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the UserServiceAccountResponse interface.
 */
export function instanceOfUserServiceAccountResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "token" in value;
    isInstance = isInstance && "userUid" in value;
    isInstance = isInstance && "userPk" in value;
    return isInstance;
}
export function UserServiceAccountResponseFromJSON(json) {
    return UserServiceAccountResponseFromJSONTyped(json, false);
}
export function UserServiceAccountResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'username': json['username'],
        'token': json['token'],
        'userUid': json['user_uid'],
        'userPk': json['user_pk'],
        'groupPk': !exists(json, 'group_pk') ? undefined : json['group_pk'],
    };
}
export function UserServiceAccountResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'token': value.token,
        'user_uid': value.userUid,
        'user_pk': value.userPk,
        'group_pk': value.groupPk,
    };
}
