// @goauthentik/api/dist/esm/models/UserSourceConnection.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { SourceFromJSON, } from './Source';
/**
 * Check if a given object implements the UserSourceConnection interface.
 */
export function instanceOfUserSourceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "created" in value;
    return isInstance;
}
export function UserSourceConnectionFromJSON(json) {
    return UserSourceConnectionFromJSONTyped(json, false);
}
export function UserSourceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': json['user'],
        'source': SourceFromJSON(json['source']),
        'created': (new Date(json['created'])),
    };
}
export function UserSourceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
