// @goauthentik/api/dist/esm/models/ValidationError.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
/**
 * Check if a given object implements the ValidationError interface.
 */
export function instanceOfValidationError(value) {
    let isInstance = true;
    return isInstance;
}
export function ValidationErrorFromJSON(json) {
    return ValidationErrorFromJSONTyped(json, false);
}
export function ValidationErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return Object.assign(Object.assign({}, json), { 'nonFieldErrors': !exists(json, 'non_field_errors') ? undefined : json['non_field_errors'], 'code': !exists(json, 'code') ? undefined : json['code'] });
}
export function ValidationErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return Object.assign(Object.assign({}, value), { 'non_field_errors': value.nonFieldErrors, 'code': value.code });
}
