// @goauthentik/api/dist/esm/models/Workers.js
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Workers interface.
 */
export function instanceOfWorkers(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    return isInstance;
}
export function WorkersFromJSON(json) {
    return WorkersFromJSONTyped(json, false);
}
export function WorkersFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
    };
}
export function WorkersToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'count': value.count,
    };
}
