// @goauthentik/api/dist/models/App.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppToJSON = exports.AppFromJSONTyped = exports.AppFromJSON = exports.instanceOfApp = void 0;
/**
 * Check if a given object implements the App interface.
 */
function instanceOfApp(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "label" in value;
    return isInstance;
}
exports.instanceOfApp = instanceOfApp;
function AppFromJSON(json) {
    return AppFromJSONTyped(json, false);
}
exports.AppFromJSON = AppFromJSON;
function AppFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'label': json['label'],
    };
}
exports.AppFromJSONTyped = AppFromJSONTyped;
function AppToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'label': value.label,
    };
}
exports.AppToJSON = AppToJSON;
