// @goauthentik/api/dist/models/ApplicationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationRequestToJSON = exports.ApplicationRequestFromJSONTyped = exports.ApplicationRequestFromJSON = exports.instanceOfApplicationRequest = void 0;
const runtime_1 = require("../runtime");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
/**
 * Check if a given object implements the ApplicationRequest interface.
 */
function instanceOfApplicationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    return isInstance;
}
exports.instanceOfApplicationRequest = instanceOfApplicationRequest;
function ApplicationRequestFromJSON(json) {
    return ApplicationRequestFromJSONTyped(json, false);
}
exports.ApplicationRequestFromJSON = ApplicationRequestFromJSON;
function ApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'slug': json['slug'],
        'provider': !(0, runtime_1.exists)(json, 'provider') ? undefined : json['provider'],
        'backchannelProviders': !(0, runtime_1.exists)(json, 'backchannel_providers') ? undefined : json['backchannel_providers'],
        'openInNewTab': !(0, runtime_1.exists)(json, 'open_in_new_tab') ? undefined : json['open_in_new_tab'],
        'metaLaunchUrl': !(0, runtime_1.exists)(json, 'meta_launch_url') ? undefined : json['meta_launch_url'],
        'metaDescription': !(0, runtime_1.exists)(json, 'meta_description') ? undefined : json['meta_description'],
        'metaPublisher': !(0, runtime_1.exists)(json, 'meta_publisher') ? undefined : json['meta_publisher'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
    };
}
exports.ApplicationRequestFromJSONTyped = ApplicationRequestFromJSONTyped;
function ApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'provider': value.provider,
        'backchannel_providers': value.backchannelProviders,
        'open_in_new_tab': value.openInNewTab,
        'meta_launch_url': value.metaLaunchUrl,
        'meta_description': value.metaDescription,
        'meta_publisher': value.metaPublisher,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'group': value.group,
    };
}
exports.ApplicationRequestToJSON = ApplicationRequestToJSON;
