// @goauthentik/api/dist/models/AppleLoginChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppleLoginChallengeToJSON = exports.AppleLoginChallengeFromJSONTyped = exports.AppleLoginChallengeFromJSON = exports.instanceOfAppleLoginChallenge = void 0;
const runtime_1 = require("../runtime");
const ChallengeChoices_1 = require("./ChallengeChoices");
const ContextualFlowInfo_1 = require("./ContextualFlowInfo");
/**
 * Check if a given object implements the AppleLoginChallenge interface.
 */
function instanceOfAppleLoginChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "scope" in value;
    isInstance = isInstance && "redirectUri" in value;
    isInstance = isInstance && "state" in value;
    return isInstance;
}
exports.instanceOfAppleLoginChallenge = instanceOfAppleLoginChallenge;
function AppleLoginChallengeFromJSON(json) {
    return AppleLoginChallengeFromJSONTyped(json, false);
}
exports.AppleLoginChallengeFromJSON = AppleLoginChallengeFromJSON;
function AppleLoginChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesFromJSON)(json['type']),
        'flowInfo': !(0, runtime_1.exists)(json, 'flow_info') ? undefined : (0, ContextualFlowInfo_1.ContextualFlowInfoFromJSON)(json['flow_info']),
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'responseErrors': !(0, runtime_1.exists)(json, 'response_errors') ? undefined : json['response_errors'],
        'clientId': json['client_id'],
        'scope': json['scope'],
        'redirectUri': json['redirect_uri'],
        'state': json['state'],
    };
}
exports.AppleLoginChallengeFromJSONTyped = AppleLoginChallengeFromJSONTyped;
function AppleLoginChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesToJSON)(value.type),
        'flow_info': (0, ContextualFlowInfo_1.ContextualFlowInfoToJSON)(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'client_id': value.clientId,
        'scope': value.scope,
        'redirect_uri': value.redirectUri,
        'state': value.state,
    };
}
exports.AppleLoginChallengeToJSON = AppleLoginChallengeToJSON;
