// @goauthentik/api/dist/models/AppleChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppleChallengeResponseRequestToJSON = exports.AppleChallengeResponseRequestFromJSONTyped = exports.AppleChallengeResponseRequestFromJSON = exports.instanceOfAppleChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AppleChallengeResponseRequest interface.
 */
function instanceOfAppleChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfAppleChallengeResponseRequest = instanceOfAppleChallengeResponseRequest;
function AppleChallengeResponseRequestFromJSON(json) {
    return AppleChallengeResponseRequestFromJSONTyped(json, false);
}
exports.AppleChallengeResponseRequestFromJSON = AppleChallengeResponseRequestFromJSON;
function AppleChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
    };
}
exports.AppleChallengeResponseRequestFromJSONTyped = AppleChallengeResponseRequestFromJSONTyped;
function AppleChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
    };
}
exports.AppleChallengeResponseRequestToJSON = AppleChallengeResponseRequestToJSON;
