// @goauthentik/api/dist/models/Application.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationToJSON = exports.ApplicationFromJSONTyped = exports.ApplicationFromJSON = exports.instanceOfApplication = void 0;
const runtime_1 = require("../runtime");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
const Provider_1 = require("./Provider");
/**
 * Check if a given object implements the Application interface.
 */
function instanceOfApplication(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "providerObj" in value;
    isInstance = isInstance && "backchannelProvidersObj" in value;
    isInstance = isInstance && "launchUrl" in value;
    isInstance = isInstance && "metaIcon" in value;
    return isInstance;
}
exports.instanceOfApplication = instanceOfApplication;
function ApplicationFromJSON(json) {
    return ApplicationFromJSONTyped(json, false);
}
exports.ApplicationFromJSON = ApplicationFromJSON;
function ApplicationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'slug': json['slug'],
        'provider': !(0, runtime_1.exists)(json, 'provider') ? undefined : json['provider'],
        'providerObj': (0, Provider_1.ProviderFromJSON)(json['provider_obj']),
        'backchannelProviders': !(0, runtime_1.exists)(json, 'backchannel_providers') ? undefined : json['backchannel_providers'],
        'backchannelProvidersObj': (json['backchannel_providers_obj'].map(Provider_1.ProviderFromJSON)),
        'launchUrl': json['launch_url'],
        'openInNewTab': !(0, runtime_1.exists)(json, 'open_in_new_tab') ? undefined : json['open_in_new_tab'],
        'metaLaunchUrl': !(0, runtime_1.exists)(json, 'meta_launch_url') ? undefined : json['meta_launch_url'],
        'metaIcon': json['meta_icon'],
        'metaDescription': !(0, runtime_1.exists)(json, 'meta_description') ? undefined : json['meta_description'],
        'metaPublisher': !(0, runtime_1.exists)(json, 'meta_publisher') ? undefined : json['meta_publisher'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
    };
}
exports.ApplicationFromJSONTyped = ApplicationFromJSONTyped;
function ApplicationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'provider': value.provider,
        'backchannel_providers': value.backchannelProviders,
        'open_in_new_tab': value.openInNewTab,
        'meta_launch_url': value.metaLaunchUrl,
        'meta_description': value.metaDescription,
        'meta_publisher': value.metaPublisher,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'group': value.group,
    };
}
exports.ApplicationToJSON = ApplicationToJSON;
