// @goauthentik/api/dist/models/AuthenticatedSessionGeoIp.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatedSessionGeoIpToJSON = exports.AuthenticatedSessionGeoIpFromJSONTyped = exports.AuthenticatedSessionGeoIpFromJSON = exports.instanceOfAuthenticatedSessionGeoIp = void 0;
/**
 * Check if a given object implements the AuthenticatedSessionGeoIp interface.
 */
function instanceOfAuthenticatedSessionGeoIp(value) {
    let isInstance = true;
    isInstance = isInstance && "continent" in value;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "lat" in value;
    isInstance = isInstance && "_long" in value;
    isInstance = isInstance && "city" in value;
    return isInstance;
}
exports.instanceOfAuthenticatedSessionGeoIp = instanceOfAuthenticatedSessionGeoIp;
function AuthenticatedSessionGeoIpFromJSON(json) {
    return AuthenticatedSessionGeoIpFromJSONTyped(json, false);
}
exports.AuthenticatedSessionGeoIpFromJSON = AuthenticatedSessionGeoIpFromJSON;
function AuthenticatedSessionGeoIpFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'continent': json['continent'],
        'country': json['country'],
        'lat': json['lat'],
        '_long': json['long'],
        'city': json['city'],
    };
}
exports.AuthenticatedSessionGeoIpFromJSONTyped = AuthenticatedSessionGeoIpFromJSONTyped;
function AuthenticatedSessionGeoIpToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'continent': value.continent,
        'country': value.country,
        'lat': value.lat,
        'long': value._long,
        'city': value.city,
    };
}
exports.AuthenticatedSessionGeoIpToJSON = AuthenticatedSessionGeoIpToJSON;
