// @goauthentik/api/dist/models/AuthenticatedSessionUserAgentOs.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatedSessionUserAgentOsToJSON = exports.AuthenticatedSessionUserAgentOsFromJSONTyped = exports.AuthenticatedSessionUserAgentOsFromJSON = exports.instanceOfAuthenticatedSessionUserAgentOs = void 0;
/**
 * Check if a given object implements the AuthenticatedSessionUserAgentOs interface.
 */
function instanceOfAuthenticatedSessionUserAgentOs(value) {
    let isInstance = true;
    isInstance = isInstance && "family" in value;
    isInstance = isInstance && "major" in value;
    isInstance = isInstance && "minor" in value;
    isInstance = isInstance && "patch" in value;
    isInstance = isInstance && "patchMinor" in value;
    return isInstance;
}
exports.instanceOfAuthenticatedSessionUserAgentOs = instanceOfAuthenticatedSessionUserAgentOs;
function AuthenticatedSessionUserAgentOsFromJSON(json) {
    return AuthenticatedSessionUserAgentOsFromJSONTyped(json, false);
}
exports.AuthenticatedSessionUserAgentOsFromJSON = AuthenticatedSessionUserAgentOsFromJSON;
function AuthenticatedSessionUserAgentOsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'family': json['family'],
        'major': json['major'],
        'minor': json['minor'],
        'patch': json['patch'],
        'patchMinor': json['patch_minor'],
    };
}
exports.AuthenticatedSessionUserAgentOsFromJSONTyped = AuthenticatedSessionUserAgentOsFromJSONTyped;
function AuthenticatedSessionUserAgentOsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'family': value.family,
        'major': value.major,
        'minor': value.minor,
        'patch': value.patch,
        'patch_minor': value.patchMinor,
    };
}
exports.AuthenticatedSessionUserAgentOsToJSON = AuthenticatedSessionUserAgentOsToJSON;
