// @goauthentik/api/dist/models/AuthenticatorSMSChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorSMSChallengeResponseRequestToJSON = exports.AuthenticatorSMSChallengeResponseRequestFromJSONTyped = exports.AuthenticatorSMSChallengeResponseRequestFromJSON = exports.instanceOfAuthenticatorSMSChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AuthenticatorSMSChallengeResponseRequest interface.
 */
function instanceOfAuthenticatorSMSChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfAuthenticatorSMSChallengeResponseRequest = instanceOfAuthenticatorSMSChallengeResponseRequest;
function AuthenticatorSMSChallengeResponseRequestFromJSON(json) {
    return AuthenticatorSMSChallengeResponseRequestFromJSONTyped(json, false);
}
exports.AuthenticatorSMSChallengeResponseRequestFromJSON = AuthenticatorSMSChallengeResponseRequestFromJSON;
function AuthenticatorSMSChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'code': !(0, runtime_1.exists)(json, 'code') ? undefined : json['code'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phone_number') ? undefined : json['phone_number'],
    };
}
exports.AuthenticatorSMSChallengeResponseRequestFromJSONTyped = AuthenticatorSMSChallengeResponseRequestFromJSONTyped;
function AuthenticatorSMSChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'code': value.code,
        'phone_number': value.phoneNumber,
    };
}
exports.AuthenticatorSMSChallengeResponseRequestToJSON = AuthenticatorSMSChallengeResponseRequestToJSON;
