// @goauthentik/api/dist/models/AuthenticatedSessionAsn.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatedSessionAsnToJSON = exports.AuthenticatedSessionAsnFromJSONTyped = exports.AuthenticatedSessionAsnFromJSON = exports.instanceOfAuthenticatedSessionAsn = void 0;
/**
 * Check if a given object implements the AuthenticatedSessionAsn interface.
 */
function instanceOfAuthenticatedSessionAsn(value) {
    let isInstance = true;
    isInstance = isInstance && "asn" in value;
    isInstance = isInstance && "asOrg" in value;
    isInstance = isInstance && "network" in value;
    return isInstance;
}
exports.instanceOfAuthenticatedSessionAsn = instanceOfAuthenticatedSessionAsn;
function AuthenticatedSessionAsnFromJSON(json) {
    return AuthenticatedSessionAsnFromJSONTyped(json, false);
}
exports.AuthenticatedSessionAsnFromJSON = AuthenticatedSessionAsnFromJSON;
function AuthenticatedSessionAsnFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'asn': json['asn'],
        'asOrg': json['as_org'],
        'network': json['network'],
    };
}
exports.AuthenticatedSessionAsnFromJSONTyped = AuthenticatedSessionAsnFromJSONTyped;
function AuthenticatedSessionAsnToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'asn': value.asn,
        'as_org': value.asOrg,
        'network': value.network,
    };
}
exports.AuthenticatedSessionAsnToJSON = AuthenticatedSessionAsnToJSON;
