// @goauthentik/api/dist/models/AuthenticatedSessionUserAgentDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatedSessionUserAgentDeviceToJSON = exports.AuthenticatedSessionUserAgentDeviceFromJSONTyped = exports.AuthenticatedSessionUserAgentDeviceFromJSON = exports.instanceOfAuthenticatedSessionUserAgentDevice = void 0;
/**
 * Check if a given object implements the AuthenticatedSessionUserAgentDevice interface.
 */
function instanceOfAuthenticatedSessionUserAgentDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "brand" in value;
    isInstance = isInstance && "family" in value;
    isInstance = isInstance && "model" in value;
    return isInstance;
}
exports.instanceOfAuthenticatedSessionUserAgentDevice = instanceOfAuthenticatedSessionUserAgentDevice;
function AuthenticatedSessionUserAgentDeviceFromJSON(json) {
    return AuthenticatedSessionUserAgentDeviceFromJSONTyped(json, false);
}
exports.AuthenticatedSessionUserAgentDeviceFromJSON = AuthenticatedSessionUserAgentDeviceFromJSON;
function AuthenticatedSessionUserAgentDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'brand': json['brand'],
        'family': json['family'],
        'model': json['model'],
    };
}
exports.AuthenticatedSessionUserAgentDeviceFromJSONTyped = AuthenticatedSessionUserAgentDeviceFromJSONTyped;
function AuthenticatedSessionUserAgentDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'brand': value.brand,
        'family': value.family,
        'model': value.model,
    };
}
exports.AuthenticatedSessionUserAgentDeviceToJSON = AuthenticatedSessionUserAgentDeviceToJSON;
