// @goauthentik/api/dist/models/AuthenticateWebAuthnStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticateWebAuthnStageToJSON = exports.AuthenticateWebAuthnStageFromJSONTyped = exports.AuthenticateWebAuthnStageFromJSON = exports.instanceOfAuthenticateWebAuthnStage = void 0;
const runtime_1 = require("../runtime");
const AuthenticatorAttachmentEnum_1 = require("./AuthenticatorAttachmentEnum");
const FlowSet_1 = require("./FlowSet");
const ResidentKeyRequirementEnum_1 = require("./ResidentKeyRequirementEnum");
const UserVerificationEnum_1 = require("./UserVerificationEnum");
/**
 * Check if a given object implements the AuthenticateWebAuthnStage interface.
 */
function instanceOfAuthenticateWebAuthnStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfAuthenticateWebAuthnStage = instanceOfAuthenticateWebAuthnStage;
function AuthenticateWebAuthnStageFromJSON(json) {
    return AuthenticateWebAuthnStageFromJSONTyped(json, false);
}
exports.AuthenticateWebAuthnStageFromJSON = AuthenticateWebAuthnStageFromJSON;
function AuthenticateWebAuthnStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'userVerification': !(0, runtime_1.exists)(json, 'user_verification') ? undefined : (0, UserVerificationEnum_1.UserVerificationEnumFromJSON)(json['user_verification']),
        'authenticatorAttachment': !(0, runtime_1.exists)(json, 'authenticator_attachment') ? undefined : (0, AuthenticatorAttachmentEnum_1.AuthenticatorAttachmentEnumFromJSON)(json['authenticator_attachment']),
        'residentKeyRequirement': !(0, runtime_1.exists)(json, 'resident_key_requirement') ? undefined : (0, ResidentKeyRequirementEnum_1.ResidentKeyRequirementEnumFromJSON)(json['resident_key_requirement']),
    };
}
exports.AuthenticateWebAuthnStageFromJSONTyped = AuthenticateWebAuthnStageFromJSONTyped;
function AuthenticateWebAuthnStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'user_verification': (0, UserVerificationEnum_1.UserVerificationEnumToJSON)(value.userVerification),
        'authenticator_attachment': (0, AuthenticatorAttachmentEnum_1.AuthenticatorAttachmentEnumToJSON)(value.authenticatorAttachment),
        'resident_key_requirement': (0, ResidentKeyRequirementEnum_1.ResidentKeyRequirementEnumToJSON)(value.residentKeyRequirement),
    };
}
exports.AuthenticateWebAuthnStageToJSON = AuthenticateWebAuthnStageToJSON;
