// @goauthentik/api/dist/models/AuthenticatorStaticStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorStaticStageToJSON = exports.AuthenticatorStaticStageFromJSONTyped = exports.AuthenticatorStaticStageFromJSON = exports.instanceOfAuthenticatorStaticStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the AuthenticatorStaticStage interface.
 */
function instanceOfAuthenticatorStaticStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorStaticStage = instanceOfAuthenticatorStaticStage;
function AuthenticatorStaticStageFromJSON(json) {
    return AuthenticatorStaticStageFromJSONTyped(json, false);
}
exports.AuthenticatorStaticStageFromJSON = AuthenticatorStaticStageFromJSON;
function AuthenticatorStaticStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'tokenCount': !(0, runtime_1.exists)(json, 'token_count') ? undefined : json['token_count'],
        'tokenLength': !(0, runtime_1.exists)(json, 'token_length') ? undefined : json['token_length'],
    };
}
exports.AuthenticatorStaticStageFromJSONTyped = AuthenticatorStaticStageFromJSONTyped;
function AuthenticatorStaticStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'token_count': value.tokenCount,
        'token_length': value.tokenLength,
    };
}
exports.AuthenticatorStaticStageToJSON = AuthenticatorStaticStageToJSON;
