// @goauthentik/api/dist/models/AuthenticationEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationEnumToJSON = exports.AuthenticationEnumFromJSONTyped = exports.AuthenticationEnumFromJSON = exports.AuthenticationEnum = void 0;
/**
 * * `none` - None
 * * `require_authenticated` - Require Authenticated
 * * `require_unauthenticated` - Require Unauthenticated
 * * `require_superuser` - Require Superuser
 * * `require_outpost` - Require Outpost
 * @export
 */
exports.AuthenticationEnum = {
    None: 'none',
    RequireAuthenticated: 'require_authenticated',
    RequireUnauthenticated: 'require_unauthenticated',
    RequireSuperuser: 'require_superuser',
    RequireOutpost: 'require_outpost',
    UnknownDefaultOpenApi: '11184809'
};
function AuthenticationEnumFromJSON(json) {
    return AuthenticationEnumFromJSONTyped(json, false);
}
exports.AuthenticationEnumFromJSON = AuthenticationEnumFromJSON;
function AuthenticationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.AuthenticationEnumFromJSONTyped = AuthenticationEnumFromJSONTyped;
function AuthenticationEnumToJSON(value) {
    return value;
}
exports.AuthenticationEnumToJSON = AuthenticationEnumToJSON;
