// @goauthentik/api/dist/models/AuthenticatorDuoChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorDuoChallengeToJSON = exports.AuthenticatorDuoChallengeFromJSONTyped = exports.AuthenticatorDuoChallengeFromJSON = exports.instanceOfAuthenticatorDuoChallenge = void 0;
const runtime_1 = require("../runtime");
const ChallengeChoices_1 = require("./ChallengeChoices");
const ContextualFlowInfo_1 = require("./ContextualFlowInfo");
/**
 * Check if a given object implements the AuthenticatorDuoChallenge interface.
 */
function instanceOfAuthenticatorDuoChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "pendingUser" in value;
    isInstance = isInstance && "pendingUserAvatar" in value;
    isInstance = isInstance && "activationBarcode" in value;
    isInstance = isInstance && "activationCode" in value;
    isInstance = isInstance && "stageUuid" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorDuoChallenge = instanceOfAuthenticatorDuoChallenge;
function AuthenticatorDuoChallengeFromJSON(json) {
    return AuthenticatorDuoChallengeFromJSONTyped(json, false);
}
exports.AuthenticatorDuoChallengeFromJSON = AuthenticatorDuoChallengeFromJSON;
function AuthenticatorDuoChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesFromJSON)(json['type']),
        'flowInfo': !(0, runtime_1.exists)(json, 'flow_info') ? undefined : (0, ContextualFlowInfo_1.ContextualFlowInfoFromJSON)(json['flow_info']),
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'responseErrors': !(0, runtime_1.exists)(json, 'response_errors') ? undefined : json['response_errors'],
        'pendingUser': json['pending_user'],
        'pendingUserAvatar': json['pending_user_avatar'],
        'activationBarcode': json['activation_barcode'],
        'activationCode': json['activation_code'],
        'stageUuid': json['stage_uuid'],
    };
}
exports.AuthenticatorDuoChallengeFromJSONTyped = AuthenticatorDuoChallengeFromJSONTyped;
function AuthenticatorDuoChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesToJSON)(value.type),
        'flow_info': (0, ContextualFlowInfo_1.ContextualFlowInfoToJSON)(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'pending_user': value.pendingUser,
        'pending_user_avatar': value.pendingUserAvatar,
        'activation_barcode': value.activationBarcode,
        'activation_code': value.activationCode,
        'stage_uuid': value.stageUuid,
    };
}
exports.AuthenticatorDuoChallengeToJSON = AuthenticatorDuoChallengeToJSON;
