// @goauthentik/api/dist/models/AuthenticatorStaticStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorStaticStageRequestToJSON = exports.AuthenticatorStaticStageRequestFromJSONTyped = exports.AuthenticatorStaticStageRequestFromJSON = exports.instanceOfAuthenticatorStaticStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the AuthenticatorStaticStageRequest interface.
 */
function instanceOfAuthenticatorStaticStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorStaticStageRequest = instanceOfAuthenticatorStaticStageRequest;
function AuthenticatorStaticStageRequestFromJSON(json) {
    return AuthenticatorStaticStageRequestFromJSONTyped(json, false);
}
exports.AuthenticatorStaticStageRequestFromJSON = AuthenticatorStaticStageRequestFromJSON;
function AuthenticatorStaticStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'tokenCount': !(0, runtime_1.exists)(json, 'token_count') ? undefined : json['token_count'],
        'tokenLength': !(0, runtime_1.exists)(json, 'token_length') ? undefined : json['token_length'],
    };
}
exports.AuthenticatorStaticStageRequestFromJSONTyped = AuthenticatorStaticStageRequestFromJSONTyped;
function AuthenticatorStaticStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'token_count': value.tokenCount,
        'token_length': value.tokenLength,
    };
}
exports.AuthenticatorStaticStageRequestToJSON = AuthenticatorStaticStageRequestToJSON;
