// @goauthentik/api/dist/models/AuthenticatorDuoStageDeviceImportResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorDuoStageDeviceImportResponseToJSON = exports.AuthenticatorDuoStageDeviceImportResponseFromJSONTyped = exports.AuthenticatorDuoStageDeviceImportResponseFromJSON = exports.instanceOfAuthenticatorDuoStageDeviceImportResponse = void 0;
/**
 * Check if a given object implements the AuthenticatorDuoStageDeviceImportResponse interface.
 */
function instanceOfAuthenticatorDuoStageDeviceImportResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorDuoStageDeviceImportResponse = instanceOfAuthenticatorDuoStageDeviceImportResponse;
function AuthenticatorDuoStageDeviceImportResponseFromJSON(json) {
    return AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json, false);
}
exports.AuthenticatorDuoStageDeviceImportResponseFromJSON = AuthenticatorDuoStageDeviceImportResponseFromJSON;
function AuthenticatorDuoStageDeviceImportResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
        'error': json['error'],
    };
}
exports.AuthenticatorDuoStageDeviceImportResponseFromJSONTyped = AuthenticatorDuoStageDeviceImportResponseFromJSONTyped;
function AuthenticatorDuoStageDeviceImportResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.AuthenticatorDuoStageDeviceImportResponseToJSON = AuthenticatorDuoStageDeviceImportResponseToJSON;
