// @goauthentik/api/dist/models/AuthenticatorAttachmentEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorAttachmentEnumToJSON = exports.AuthenticatorAttachmentEnumFromJSONTyped = exports.AuthenticatorAttachmentEnumFromJSON = exports.AuthenticatorAttachmentEnum = void 0;
/**
 * * `platform` - Platform
 * * `cross-platform` - Cross Platform
 * @export
 */
exports.AuthenticatorAttachmentEnum = {
    Platform: 'platform',
    CrossPlatform: 'cross-platform',
    UnknownDefaultOpenApi: '11184809'
};
function AuthenticatorAttachmentEnumFromJSON(json) {
    return AuthenticatorAttachmentEnumFromJSONTyped(json, false);
}
exports.AuthenticatorAttachmentEnumFromJSON = AuthenticatorAttachmentEnumFromJSON;
function AuthenticatorAttachmentEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.AuthenticatorAttachmentEnumFromJSONTyped = AuthenticatorAttachmentEnumFromJSONTyped;
function AuthenticatorAttachmentEnumToJSON(value) {
    return value;
}
exports.AuthenticatorAttachmentEnumToJSON = AuthenticatorAttachmentEnumToJSON;
