// @goauthentik/api/dist/models/AuthTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthTypeEnumToJSON = exports.AuthTypeEnumFromJSONTyped = exports.AuthTypeEnumFromJSON = exports.AuthTypeEnum = void 0;
/**
 * * `basic` - Basic
 * * `bearer` - Bearer
 * @export
 */
exports.AuthTypeEnum = {
    Basic: 'basic',
    Bearer: 'bearer',
    UnknownDefaultOpenApi: '11184809'
};
function AuthTypeEnumFromJSON(json) {
    return AuthTypeEnumFromJSONTyped(json, false);
}
exports.AuthTypeEnumFromJSON = AuthTypeEnumFromJSON;
function AuthTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.AuthTypeEnumFromJSONTyped = AuthTypeEnumFromJSONTyped;
function AuthTypeEnumToJSON(value) {
    return value;
}
exports.AuthTypeEnumToJSON = AuthTypeEnumToJSON;
