// @goauthentik/api/dist/models/AuthenticatorDuoStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorDuoStageRequestToJSON = exports.AuthenticatorDuoStageRequestFromJSONTyped = exports.AuthenticatorDuoStageRequestFromJSON = exports.instanceOfAuthenticatorDuoStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the AuthenticatorDuoStageRequest interface.
 */
function instanceOfAuthenticatorDuoStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "clientId" in value;
    isInstance = isInstance && "clientSecret" in value;
    isInstance = isInstance && "apiHostname" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorDuoStageRequest = instanceOfAuthenticatorDuoStageRequest;
function AuthenticatorDuoStageRequestFromJSON(json) {
    return AuthenticatorDuoStageRequestFromJSONTyped(json, false);
}
exports.AuthenticatorDuoStageRequestFromJSON = AuthenticatorDuoStageRequestFromJSON;
function AuthenticatorDuoStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'clientId': json['client_id'],
        'clientSecret': json['client_secret'],
        'apiHostname': json['api_hostname'],
        'adminIntegrationKey': !(0, runtime_1.exists)(json, 'admin_integration_key') ? undefined : json['admin_integration_key'],
        'adminSecretKey': !(0, runtime_1.exists)(json, 'admin_secret_key') ? undefined : json['admin_secret_key'],
    };
}
exports.AuthenticatorDuoStageRequestFromJSONTyped = AuthenticatorDuoStageRequestFromJSONTyped;
function AuthenticatorDuoStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'api_hostname': value.apiHostname,
        'admin_integration_key': value.adminIntegrationKey,
        'admin_secret_key': value.adminSecretKey,
    };
}
exports.AuthenticatorDuoStageRequestToJSON = AuthenticatorDuoStageRequestToJSON;
