// @goauthentik/api/dist/models/AuthenticatorDuoStageManualDeviceImportRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorDuoStageManualDeviceImportRequestToJSON = exports.AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped = exports.AuthenticatorDuoStageManualDeviceImportRequestFromJSON = exports.instanceOfAuthenticatorDuoStageManualDeviceImportRequest = void 0;
/**
 * Check if a given object implements the AuthenticatorDuoStageManualDeviceImportRequest interface.
 */
function instanceOfAuthenticatorDuoStageManualDeviceImportRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "duoUserId" in value;
    isInstance = isInstance && "username" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorDuoStageManualDeviceImportRequest = instanceOfAuthenticatorDuoStageManualDeviceImportRequest;
function AuthenticatorDuoStageManualDeviceImportRequestFromJSON(json) {
    return AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json, false);
}
exports.AuthenticatorDuoStageManualDeviceImportRequestFromJSON = AuthenticatorDuoStageManualDeviceImportRequestFromJSON;
function AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'duoUserId': json['duo_user_id'],
        'username': json['username'],
    };
}
exports.AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped = AuthenticatorDuoStageManualDeviceImportRequestFromJSONTyped;
function AuthenticatorDuoStageManualDeviceImportRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'duo_user_id': value.duoUserId,
        'username': value.username,
    };
}
exports.AuthenticatorDuoStageManualDeviceImportRequestToJSON = AuthenticatorDuoStageManualDeviceImportRequestToJSON;
