// @goauthentik/api/dist/models/AuthModeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthModeEnumToJSON = exports.AuthModeEnumFromJSONTyped = exports.AuthModeEnumFromJSON = exports.AuthModeEnum = void 0;
/**
 * * `static` - Static
 * * `prompt` - Prompt
 * @export
 */
exports.AuthModeEnum = {
    Static: 'static',
    Prompt: 'prompt',
    UnknownDefaultOpenApi: '11184809'
};
function AuthModeEnumFromJSON(json) {
    return AuthModeEnumFromJSONTyped(json, false);
}
exports.AuthModeEnumFromJSON = AuthModeEnumFromJSON;
function AuthModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.AuthModeEnumFromJSONTyped = AuthModeEnumFromJSONTyped;
function AuthModeEnumToJSON(value) {
    return value;
}
exports.AuthModeEnumToJSON = AuthModeEnumToJSON;
