// @goauthentik/api/dist/models/AuthenticatorTOTPChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorTOTPChallengeResponseRequestToJSON = exports.AuthenticatorTOTPChallengeResponseRequestFromJSONTyped = exports.AuthenticatorTOTPChallengeResponseRequestFromJSON = exports.instanceOfAuthenticatorTOTPChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AuthenticatorTOTPChallengeResponseRequest interface.
 */
function instanceOfAuthenticatorTOTPChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorTOTPChallengeResponseRequest = instanceOfAuthenticatorTOTPChallengeResponseRequest;
function AuthenticatorTOTPChallengeResponseRequestFromJSON(json) {
    return AuthenticatorTOTPChallengeResponseRequestFromJSONTyped(json, false);
}
exports.AuthenticatorTOTPChallengeResponseRequestFromJSON = AuthenticatorTOTPChallengeResponseRequestFromJSON;
function AuthenticatorTOTPChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'code': json['code'],
    };
}
exports.AuthenticatorTOTPChallengeResponseRequestFromJSONTyped = AuthenticatorTOTPChallengeResponseRequestFromJSONTyped;
function AuthenticatorTOTPChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'code': value.code,
    };
}
exports.AuthenticatorTOTPChallengeResponseRequestToJSON = AuthenticatorTOTPChallengeResponseRequestToJSON;
