// @goauthentik/api/dist/models/AuthenticatorWebAuthnChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticatorWebAuthnChallengeResponseRequestToJSON = exports.AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped = exports.AuthenticatorWebAuthnChallengeResponseRequestFromJSON = exports.instanceOfAuthenticatorWebAuthnChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AuthenticatorWebAuthnChallengeResponseRequest interface.
 */
function instanceOfAuthenticatorWebAuthnChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "response" in value;
    return isInstance;
}
exports.instanceOfAuthenticatorWebAuthnChallengeResponseRequest = instanceOfAuthenticatorWebAuthnChallengeResponseRequest;
function AuthenticatorWebAuthnChallengeResponseRequestFromJSON(json) {
    return AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json, false);
}
exports.AuthenticatorWebAuthnChallengeResponseRequestFromJSON = AuthenticatorWebAuthnChallengeResponseRequestFromJSON;
function AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'response': json['response'],
    };
}
exports.AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped = AuthenticatorWebAuthnChallengeResponseRequestFromJSONTyped;
function AuthenticatorWebAuthnChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'response': value.response,
    };
}
exports.AuthenticatorWebAuthnChallengeResponseRequestToJSON = AuthenticatorWebAuthnChallengeResponseRequestToJSON;
