// @goauthentik/api/dist/models/AuthenticateWebAuthnStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticateWebAuthnStageRequestToJSON = exports.AuthenticateWebAuthnStageRequestFromJSONTyped = exports.AuthenticateWebAuthnStageRequestFromJSON = exports.instanceOfAuthenticateWebAuthnStageRequest = void 0;
const runtime_1 = require("../runtime");
const AuthenticatorAttachmentEnum_1 = require("./AuthenticatorAttachmentEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
const ResidentKeyRequirementEnum_1 = require("./ResidentKeyRequirementEnum");
const UserVerificationEnum_1 = require("./UserVerificationEnum");
/**
 * Check if a given object implements the AuthenticateWebAuthnStageRequest interface.
 */
function instanceOfAuthenticateWebAuthnStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfAuthenticateWebAuthnStageRequest = instanceOfAuthenticateWebAuthnStageRequest;
function AuthenticateWebAuthnStageRequestFromJSON(json) {
    return AuthenticateWebAuthnStageRequestFromJSONTyped(json, false);
}
exports.AuthenticateWebAuthnStageRequestFromJSON = AuthenticateWebAuthnStageRequestFromJSON;
function AuthenticateWebAuthnStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'userVerification': !(0, runtime_1.exists)(json, 'user_verification') ? undefined : (0, UserVerificationEnum_1.UserVerificationEnumFromJSON)(json['user_verification']),
        'authenticatorAttachment': !(0, runtime_1.exists)(json, 'authenticator_attachment') ? undefined : (0, AuthenticatorAttachmentEnum_1.AuthenticatorAttachmentEnumFromJSON)(json['authenticator_attachment']),
        'residentKeyRequirement': !(0, runtime_1.exists)(json, 'resident_key_requirement') ? undefined : (0, ResidentKeyRequirementEnum_1.ResidentKeyRequirementEnumFromJSON)(json['resident_key_requirement']),
    };
}
exports.AuthenticateWebAuthnStageRequestFromJSONTyped = AuthenticateWebAuthnStageRequestFromJSONTyped;
function AuthenticateWebAuthnStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'user_verification': (0, UserVerificationEnum_1.UserVerificationEnumToJSON)(value.userVerification),
        'authenticator_attachment': (0, AuthenticatorAttachmentEnum_1.AuthenticatorAttachmentEnumToJSON)(value.authenticatorAttachment),
        'resident_key_requirement': (0, ResidentKeyRequirementEnum_1.ResidentKeyRequirementEnumToJSON)(value.residentKeyRequirement),
    };
}
exports.AuthenticateWebAuthnStageRequestToJSON = AuthenticateWebAuthnStageRequestToJSON;
