// @goauthentik/api/dist/models/BackendsEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendsEnumToJSON = exports.BackendsEnumFromJSONTyped = exports.BackendsEnumFromJSON = exports.BackendsEnum = void 0;
/**
 * * `authentik.core.auth.InbuiltBackend` - User database + standard password
 * * `authentik.core.auth.TokenBackend` - User database + app passwords
 * * `authentik.sources.ldap.auth.LDAPBackend` - User database + LDAP password
 * @export
 */
exports.BackendsEnum = {
    CoreAuthInbuiltBackend: 'authentik.core.auth.InbuiltBackend',
    CoreAuthTokenBackend: 'authentik.core.auth.TokenBackend',
    SourcesLdapAuthLdapBackend: 'authentik.sources.ldap.auth.LDAPBackend',
    UnknownDefaultOpenApi: '11184809'
};
function BackendsEnumFromJSON(json) {
    return BackendsEnumFromJSONTyped(json, false);
}
exports.BackendsEnumFromJSON = BackendsEnumFromJSON;
function BackendsEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.BackendsEnumFromJSONTyped = BackendsEnumFromJSONTyped;
function BackendsEnumToJSON(value) {
    return value;
}
exports.BackendsEnumToJSON = BackendsEnumToJSON;
