// @goauthentik/api/dist/models/BindingTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BindingTypeEnumToJSON = exports.BindingTypeEnumFromJSONTyped = exports.BindingTypeEnumFromJSON = exports.BindingTypeEnum = void 0;
/**
 * * `REDIRECT` - Redirect Binding
 * * `POST` - POST Binding
 * * `POST_AUTO` - POST Binding with auto-confirmation
 * @export
 */
exports.BindingTypeEnum = {
    Redirect: 'REDIRECT',
    Post: 'POST',
    PostAuto: 'POST_AUTO',
    UnknownDefaultOpenApi: '11184809'
};
function BindingTypeEnumFromJSON(json) {
    return BindingTypeEnumFromJSONTyped(json, false);
}
exports.BindingTypeEnumFromJSON = BindingTypeEnumFromJSON;
function BindingTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.BindingTypeEnumFromJSONTyped = BindingTypeEnumFromJSONTyped;
function BindingTypeEnumToJSON(value) {
    return value;
}
exports.BindingTypeEnumToJSON = BindingTypeEnumToJSON;
