// @goauthentik/api/dist/models/BlueprintInstanceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintInstanceRequestToJSON = exports.BlueprintInstanceRequestFromJSONTyped = exports.BlueprintInstanceRequestFromJSON = exports.instanceOfBlueprintInstanceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the BlueprintInstanceRequest interface.
 */
function instanceOfBlueprintInstanceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfBlueprintInstanceRequest = instanceOfBlueprintInstanceRequest;
function BlueprintInstanceRequestFromJSON(json) {
    return BlueprintInstanceRequestFromJSONTyped(json, false);
}
exports.BlueprintInstanceRequestFromJSON = BlueprintInstanceRequestFromJSON;
function BlueprintInstanceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'context': !(0, runtime_1.exists)(json, 'context') ? undefined : json['context'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'content': !(0, runtime_1.exists)(json, 'content') ? undefined : json['content'],
    };
}
exports.BlueprintInstanceRequestFromJSONTyped = BlueprintInstanceRequestFromJSONTyped;
function BlueprintInstanceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
exports.BlueprintInstanceRequestToJSON = BlueprintInstanceRequestToJSON;
