// @goauthentik/api/dist/models/BlueprintInstanceStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintInstanceStatusEnumToJSON = exports.BlueprintInstanceStatusEnumFromJSONTyped = exports.BlueprintInstanceStatusEnumFromJSON = exports.BlueprintInstanceStatusEnum = void 0;
/**
 * * `successful` - Successful
 * * `warning` - Warning
 * * `error` - Error
 * * `orphaned` - Orphaned
 * * `unknown` - Unknown
 * @export
 */
exports.BlueprintInstanceStatusEnum = {
    Successful: 'successful',
    Warning: 'warning',
    Error: 'error',
    Orphaned: 'orphaned',
    Unknown: 'unknown',
    UnknownDefaultOpenApi: '11184809'
};
function BlueprintInstanceStatusEnumFromJSON(json) {
    return BlueprintInstanceStatusEnumFromJSONTyped(json, false);
}
exports.BlueprintInstanceStatusEnumFromJSON = BlueprintInstanceStatusEnumFromJSON;
function BlueprintInstanceStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.BlueprintInstanceStatusEnumFromJSONTyped = BlueprintInstanceStatusEnumFromJSONTyped;
function BlueprintInstanceStatusEnumToJSON(value) {
    return value;
}
exports.BlueprintInstanceStatusEnumToJSON = BlueprintInstanceStatusEnumToJSON;
