// @goauthentik/api/dist/models/BlueprintFile.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintFileToJSON = exports.BlueprintFileFromJSONTyped = exports.BlueprintFileFromJSON = exports.instanceOfBlueprintFile = void 0;
const Metadata_1 = require("./Metadata");
/**
 * Check if a given object implements the BlueprintFile interface.
 */
function instanceOfBlueprintFile(value) {
    let isInstance = true;
    isInstance = isInstance && "path" in value;
    isInstance = isInstance && "lastM" in value;
    isInstance = isInstance && "hash" in value;
    isInstance = isInstance && "meta" in value;
    return isInstance;
}
exports.instanceOfBlueprintFile = instanceOfBlueprintFile;
function BlueprintFileFromJSON(json) {
    return BlueprintFileFromJSONTyped(json, false);
}
exports.BlueprintFileFromJSON = BlueprintFileFromJSON;
function BlueprintFileFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'path': json['path'],
        'lastM': (new Date(json['last_m'])),
        'hash': json['hash'],
        'meta': (0, Metadata_1.MetadataFromJSON)(json['meta']),
    };
}
exports.BlueprintFileFromJSONTyped = BlueprintFileFromJSONTyped;
function BlueprintFileToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'path': value.path,
        'last_m': (value.lastM.toISOString()),
        'hash': value.hash,
    };
}
exports.BlueprintFileToJSON = BlueprintFileToJSON;
