// @goauthentik/api/dist/models/BlueprintInstance.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintInstanceToJSON = exports.BlueprintInstanceFromJSONTyped = exports.BlueprintInstanceFromJSON = exports.instanceOfBlueprintInstance = void 0;
const runtime_1 = require("../runtime");
const BlueprintInstanceStatusEnum_1 = require("./BlueprintInstanceStatusEnum");
/**
 * Check if a given object implements the BlueprintInstance interface.
 */
function instanceOfBlueprintInstance(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "lastApplied" in value;
    isInstance = isInstance && "lastAppliedHash" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "managedModels" in value;
    isInstance = isInstance && "metadata" in value;
    return isInstance;
}
exports.instanceOfBlueprintInstance = instanceOfBlueprintInstance;
function BlueprintInstanceFromJSON(json) {
    return BlueprintInstanceFromJSONTyped(json, false);
}
exports.BlueprintInstanceFromJSON = BlueprintInstanceFromJSON;
function BlueprintInstanceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'context': !(0, runtime_1.exists)(json, 'context') ? undefined : json['context'],
        'lastApplied': (new Date(json['last_applied'])),
        'lastAppliedHash': json['last_applied_hash'],
        'status': (0, BlueprintInstanceStatusEnum_1.BlueprintInstanceStatusEnumFromJSON)(json['status']),
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'managedModels': json['managed_models'],
        'metadata': json['metadata'],
        'content': !(0, runtime_1.exists)(json, 'content') ? undefined : json['content'],
    };
}
exports.BlueprintInstanceFromJSONTyped = BlueprintInstanceFromJSONTyped;
function BlueprintInstanceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
exports.BlueprintInstanceToJSON = BlueprintInstanceToJSON;
