// @goauthentik/api/dist/models/Cache.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheToJSON = exports.CacheFromJSONTyped = exports.CacheFromJSON = exports.instanceOfCache = void 0;
/**
 * Check if a given object implements the Cache interface.
 */
function instanceOfCache(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    return isInstance;
}
exports.instanceOfCache = instanceOfCache;
function CacheFromJSON(json) {
    return CacheFromJSONTyped(json, false);
}
exports.CacheFromJSON = CacheFromJSON;
function CacheFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
    };
}
exports.CacheFromJSONTyped = CacheFromJSONTyped;
function CacheToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.CacheToJSON = CacheToJSON;
