// @goauthentik/api/dist/models/CapabilitiesEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapabilitiesEnumToJSON = exports.CapabilitiesEnumFromJSONTyped = exports.CapabilitiesEnumFromJSON = exports.CapabilitiesEnum = void 0;
/**
 * * `can_save_media` - Can Save Media
 * * `can_geo_ip` - Can Geo Ip
 * * `can_asn` - Can Asn
 * * `can_impersonate` - Can Impersonate
 * * `can_debug` - Can Debug
 * * `is_enterprise` - Is Enterprise
 * @export
 */
exports.CapabilitiesEnum = {
    CanSaveMedia: 'can_save_media',
    CanGeoIp: 'can_geo_ip',
    CanAsn: 'can_asn',
    CanImpersonate: 'can_impersonate',
    CanDebug: 'can_debug',
    IsEnterprise: 'is_enterprise',
    UnknownDefaultOpenApi: '11184809'
};
function CapabilitiesEnumFromJSON(json) {
    return CapabilitiesEnumFromJSONTyped(json, false);
}
exports.CapabilitiesEnumFromJSON = CapabilitiesEnumFromJSON;
function CapabilitiesEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.CapabilitiesEnumFromJSONTyped = CapabilitiesEnumFromJSONTyped;
function CapabilitiesEnumToJSON(value) {
    return value;
}
exports.CapabilitiesEnumToJSON = CapabilitiesEnumToJSON;
