// @goauthentik/api/dist/models/CaptchaChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptchaChallengeResponseRequestToJSON = exports.CaptchaChallengeResponseRequestFromJSONTyped = exports.CaptchaChallengeResponseRequestFromJSON = exports.instanceOfCaptchaChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CaptchaChallengeResponseRequest interface.
 */
function instanceOfCaptchaChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfCaptchaChallengeResponseRequest = instanceOfCaptchaChallengeResponseRequest;
function CaptchaChallengeResponseRequestFromJSON(json) {
    return CaptchaChallengeResponseRequestFromJSONTyped(json, false);
}
exports.CaptchaChallengeResponseRequestFromJSON = CaptchaChallengeResponseRequestFromJSON;
function CaptchaChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'token': json['token'],
    };
}
exports.CaptchaChallengeResponseRequestFromJSONTyped = CaptchaChallengeResponseRequestFromJSONTyped;
function CaptchaChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'token': value.token,
    };
}
exports.CaptchaChallengeResponseRequestToJSON = CaptchaChallengeResponseRequestToJSON;
