// @goauthentik/api/dist/models/CaptchaStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptchaStageToJSON = exports.CaptchaStageFromJSONTyped = exports.CaptchaStageFromJSON = exports.instanceOfCaptchaStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the CaptchaStage interface.
 */
function instanceOfCaptchaStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "publicKey" in value;
    return isInstance;
}
exports.instanceOfCaptchaStage = instanceOfCaptchaStage;
function CaptchaStageFromJSON(json) {
    return CaptchaStageFromJSONTyped(json, false);
}
exports.CaptchaStageFromJSON = CaptchaStageFromJSON;
function CaptchaStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'publicKey': json['public_key'],
        'jsUrl': !(0, runtime_1.exists)(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !(0, runtime_1.exists)(json, 'api_url') ? undefined : json['api_url'],
    };
}
exports.CaptchaStageFromJSONTyped = CaptchaStageFromJSONTyped;
function CaptchaStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'public_key': value.publicKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
exports.CaptchaStageToJSON = CaptchaStageToJSON;
