// @goauthentik/api/dist/models/CaptchaStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptchaStageRequestToJSON = exports.CaptchaStageRequestFromJSONTyped = exports.CaptchaStageRequestFromJSON = exports.instanceOfCaptchaStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the CaptchaStageRequest interface.
 */
function instanceOfCaptchaStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "publicKey" in value;
    isInstance = isInstance && "privateKey" in value;
    return isInstance;
}
exports.instanceOfCaptchaStageRequest = instanceOfCaptchaStageRequest;
function CaptchaStageRequestFromJSON(json) {
    return CaptchaStageRequestFromJSONTyped(json, false);
}
exports.CaptchaStageRequestFromJSON = CaptchaStageRequestFromJSON;
function CaptchaStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'publicKey': json['public_key'],
        'privateKey': json['private_key'],
        'jsUrl': !(0, runtime_1.exists)(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !(0, runtime_1.exists)(json, 'api_url') ? undefined : json['api_url'],
    };
}
exports.CaptchaStageRequestFromJSONTyped = CaptchaStageRequestFromJSONTyped;
function CaptchaStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'public_key': value.publicKey,
        'private_key': value.privateKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
exports.CaptchaStageRequestToJSON = CaptchaStageRequestToJSON;
