// @goauthentik/api/dist/models/CertificateGenerationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateGenerationRequestToJSON = exports.CertificateGenerationRequestFromJSONTyped = exports.CertificateGenerationRequestFromJSON = exports.instanceOfCertificateGenerationRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CertificateGenerationRequest interface.
 */
function instanceOfCertificateGenerationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "commonName" in value;
    isInstance = isInstance && "validityDays" in value;
    return isInstance;
}
exports.instanceOfCertificateGenerationRequest = instanceOfCertificateGenerationRequest;
function CertificateGenerationRequestFromJSON(json) {
    return CertificateGenerationRequestFromJSONTyped(json, false);
}
exports.CertificateGenerationRequestFromJSON = CertificateGenerationRequestFromJSON;
function CertificateGenerationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'commonName': json['common_name'],
        'subjectAltName': !(0, runtime_1.exists)(json, 'subject_alt_name') ? undefined : json['subject_alt_name'],
        'validityDays': json['validity_days'],
    };
}
exports.CertificateGenerationRequestFromJSONTyped = CertificateGenerationRequestFromJSONTyped;
function CertificateGenerationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'common_name': value.commonName,
        'subject_alt_name': value.subjectAltName,
        'validity_days': value.validityDays,
    };
}
exports.CertificateGenerationRequestToJSON = CertificateGenerationRequestToJSON;
