// @goauthentik/api/dist/models/CertificateData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateDataToJSON = exports.CertificateDataFromJSONTyped = exports.CertificateDataFromJSON = exports.instanceOfCertificateData = void 0;
/**
 * Check if a given object implements the CertificateData interface.
 */
function instanceOfCertificateData(value) {
    let isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfCertificateData = instanceOfCertificateData;
function CertificateDataFromJSON(json) {
    return CertificateDataFromJSONTyped(json, false);
}
exports.CertificateDataFromJSON = CertificateDataFromJSON;
function CertificateDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': json['data'],
    };
}
exports.CertificateDataFromJSONTyped = CertificateDataFromJSONTyped;
function CertificateDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.CertificateDataToJSON = CertificateDataToJSON;
