// @goauthentik/api/dist/models/ChallengeChoices.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChallengeChoicesToJSON = exports.ChallengeChoicesFromJSONTyped = exports.ChallengeChoicesFromJSON = exports.ChallengeChoices = void 0;
/**
 * * `native` - NATIVE
 * * `shell` - SHELL
 * * `redirect` - REDIRECT
 * @export
 */
exports.ChallengeChoices = {
    Native: 'native',
    Shell: 'shell',
    Redirect: 'redirect',
    UnknownDefaultOpenApi: '11184809'
};
function ChallengeChoicesFromJSON(json) {
    return ChallengeChoicesFromJSONTyped(json, false);
}
exports.ChallengeChoicesFromJSON = ChallengeChoicesFromJSON;
function ChallengeChoicesFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ChallengeChoicesFromJSONTyped = ChallengeChoicesFromJSONTyped;
function ChallengeChoicesToJSON(value) {
    return value;
}
exports.ChallengeChoicesToJSON = ChallengeChoicesToJSON;
