// @goauthentik/api/dist/models/ClientTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientTypeEnumToJSON = exports.ClientTypeEnumFromJSONTyped = exports.ClientTypeEnumFromJSON = exports.ClientTypeEnum = void 0;
/**
 * * `confidential` - Confidential
 * * `public` - Public
 * @export
 */
exports.ClientTypeEnum = {
    Confidential: 'confidential',
    Public: 'public',
    UnknownDefaultOpenApi: '11184809'
};
function ClientTypeEnumFromJSON(json) {
    return ClientTypeEnumFromJSONTyped(json, false);
}
exports.ClientTypeEnumFromJSON = ClientTypeEnumFromJSON;
function ClientTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ClientTypeEnumFromJSONTyped = ClientTypeEnumFromJSONTyped;
function ClientTypeEnumToJSON(value) {
    return value;
}
exports.ClientTypeEnumToJSON = ClientTypeEnumToJSON;
