// @goauthentik/api/dist/models/ConnectionTokenRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionTokenRequestToJSON = exports.ConnectionTokenRequestFromJSONTyped = exports.ConnectionTokenRequestFromJSON = exports.instanceOfConnectionTokenRequest = void 0;
/**
 * Check if a given object implements the ConnectionTokenRequest interface.
 */
function instanceOfConnectionTokenRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "provider" in value;
    return isInstance;
}
exports.instanceOfConnectionTokenRequest = instanceOfConnectionTokenRequest;
function ConnectionTokenRequestFromJSON(json) {
    return ConnectionTokenRequestFromJSONTyped(json, false);
}
exports.ConnectionTokenRequestFromJSON = ConnectionTokenRequestFromJSON;
function ConnectionTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'provider': json['provider'],
    };
}
exports.ConnectionTokenRequestFromJSONTyped = ConnectionTokenRequestFromJSONTyped;
function ConnectionTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
exports.ConnectionTokenRequestToJSON = ConnectionTokenRequestToJSON;
