// @goauthentik/api/dist/models/ConnectionToken.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionTokenToJSON = exports.ConnectionTokenFromJSONTyped = exports.ConnectionTokenFromJSON = exports.instanceOfConnectionToken = void 0;
const Endpoint_1 = require("./Endpoint");
const GroupMember_1 = require("./GroupMember");
const RACProvider_1 = require("./RACProvider");
/**
 * Check if a given object implements the ConnectionToken interface.
 */
function instanceOfConnectionToken(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "providerObj" in value;
    isInstance = isInstance && "endpoint" in value;
    isInstance = isInstance && "endpointObj" in value;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfConnectionToken = instanceOfConnectionToken;
function ConnectionTokenFromJSON(json) {
    return ConnectionTokenFromJSONTyped(json, false);
}
exports.ConnectionTokenFromJSON = ConnectionTokenFromJSON;
function ConnectionTokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'provider': json['provider'],
        'providerObj': (0, RACProvider_1.RACProviderFromJSON)(json['provider_obj']),
        'endpoint': json['endpoint'],
        'endpointObj': (0, Endpoint_1.EndpointFromJSON)(json['endpoint_obj']),
        'user': (0, GroupMember_1.GroupMemberFromJSON)(json['user']),
    };
}
exports.ConnectionTokenFromJSONTyped = ConnectionTokenFromJSONTyped;
function ConnectionTokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
exports.ConnectionTokenToJSON = ConnectionTokenToJSON;
