// @goauthentik/api/dist/models/ConsentStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsentStageToJSON = exports.ConsentStageFromJSONTyped = exports.ConsentStageFromJSON = exports.instanceOfConsentStage = void 0;
const runtime_1 = require("../runtime");
const ConsentStageModeEnum_1 = require("./ConsentStageModeEnum");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the ConsentStage interface.
 */
function instanceOfConsentStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfConsentStage = instanceOfConsentStage;
function ConsentStageFromJSON(json) {
    return ConsentStageFromJSONTyped(json, false);
}
exports.ConsentStageFromJSON = ConsentStageFromJSON;
function ConsentStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, ConsentStageModeEnum_1.ConsentStageModeEnumFromJSON)(json['mode']),
        'consentExpireIn': !(0, runtime_1.exists)(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}
exports.ConsentStageFromJSONTyped = ConsentStageFromJSONTyped;
function ConsentStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'mode': (0, ConsentStageModeEnum_1.ConsentStageModeEnumToJSON)(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
exports.ConsentStageToJSON = ConsentStageToJSON;
