// @goauthentik/api/dist/models/ConsentStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsentStageRequestToJSON = exports.ConsentStageRequestFromJSONTyped = exports.ConsentStageRequestFromJSON = exports.instanceOfConsentStageRequest = void 0;
const runtime_1 = require("../runtime");
const ConsentStageModeEnum_1 = require("./ConsentStageModeEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the ConsentStageRequest interface.
 */
function instanceOfConsentStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfConsentStageRequest = instanceOfConsentStageRequest;
function ConsentStageRequestFromJSON(json) {
    return ConsentStageRequestFromJSONTyped(json, false);
}
exports.ConsentStageRequestFromJSON = ConsentStageRequestFromJSON;
function ConsentStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, ConsentStageModeEnum_1.ConsentStageModeEnumFromJSON)(json['mode']),
        'consentExpireIn': !(0, runtime_1.exists)(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}
exports.ConsentStageRequestFromJSONTyped = ConsentStageRequestFromJSONTyped;
function ConsentStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'mode': (0, ConsentStageModeEnum_1.ConsentStageModeEnumToJSON)(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
exports.ConsentStageRequestToJSON = ConsentStageRequestToJSON;
