// @goauthentik/api/dist/models/ConsentStageModeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsentStageModeEnumToJSON = exports.ConsentStageModeEnumFromJSONTyped = exports.ConsentStageModeEnumFromJSON = exports.ConsentStageModeEnum = void 0;
/**
 * * `always_require` - Always Require
 * * `permanent` - Permanent
 * * `expiring` - Expiring
 * @export
 */
exports.ConsentStageModeEnum = {
    AlwaysRequire: 'always_require',
    Permanent: 'permanent',
    Expiring: 'expiring',
    UnknownDefaultOpenApi: '11184809'
};
function ConsentStageModeEnumFromJSON(json) {
    return ConsentStageModeEnumFromJSONTyped(json, false);
}
exports.ConsentStageModeEnumFromJSON = ConsentStageModeEnumFromJSON;
function ConsentStageModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ConsentStageModeEnumFromJSONTyped = ConsentStageModeEnumFromJSONTyped;
function ConsentStageModeEnumToJSON(value) {
    return value;
}
exports.ConsentStageModeEnumToJSON = ConsentStageModeEnumToJSON;
