// @goauthentik/api/dist/models/ConsentChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsentChallengeResponseRequestToJSON = exports.ConsentChallengeResponseRequestFromJSONTyped = exports.ConsentChallengeResponseRequestFromJSON = exports.instanceOfConsentChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ConsentChallengeResponseRequest interface.
 */
function instanceOfConsentChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfConsentChallengeResponseRequest = instanceOfConsentChallengeResponseRequest;
function ConsentChallengeResponseRequestFromJSON(json) {
    return ConsentChallengeResponseRequestFromJSONTyped(json, false);
}
exports.ConsentChallengeResponseRequestFromJSON = ConsentChallengeResponseRequestFromJSON;
function ConsentChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'token': json['token'],
    };
}
exports.ConsentChallengeResponseRequestFromJSONTyped = ConsentChallengeResponseRequestFromJSONTyped;
function ConsentChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'token': value.token,
    };
}
exports.ConsentChallengeResponseRequestToJSON = ConsentChallengeResponseRequestToJSON;
