// @goauthentik/api/dist/models/ConsentPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsentPermissionToJSON = exports.ConsentPermissionFromJSONTyped = exports.ConsentPermissionFromJSON = exports.instanceOfConsentPermission = void 0;
/**
 * Check if a given object implements the ConsentPermission interface.
 */
function instanceOfConsentPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "id" in value;
    return isInstance;
}
exports.instanceOfConsentPermission = instanceOfConsentPermission;
function ConsentPermissionFromJSON(json) {
    return ConsentPermissionFromJSONTyped(json, false);
}
exports.ConsentPermissionFromJSON = ConsentPermissionFromJSON;
function ConsentPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'id': json['id'],
    };
}
exports.ConsentPermissionFromJSONTyped = ConsentPermissionFromJSONTyped;
function ConsentPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'id': value.id,
    };
}
exports.ConsentPermissionToJSON = ConsentPermissionToJSON;
