// @goauthentik/api/dist/models/ContextualFlowInfoLayoutEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextualFlowInfoLayoutEnumToJSON = exports.ContextualFlowInfoLayoutEnumFromJSONTyped = exports.ContextualFlowInfoLayoutEnumFromJSON = exports.ContextualFlowInfoLayoutEnum = void 0;
/**
 * * `stacked` - STACKED
 * * `content_left` - CONTENT_LEFT
 * * `content_right` - CONTENT_RIGHT
 * * `sidebar_left` - SIDEBAR_LEFT
 * * `sidebar_right` - SIDEBAR_RIGHT
 * @export
 */
exports.ContextualFlowInfoLayoutEnum = {
    Stacked: 'stacked',
    ContentLeft: 'content_left',
    ContentRight: 'content_right',
    SidebarLeft: 'sidebar_left',
    SidebarRight: 'sidebar_right',
    UnknownDefaultOpenApi: '11184809'
};
function ContextualFlowInfoLayoutEnumFromJSON(json) {
    return ContextualFlowInfoLayoutEnumFromJSONTyped(json, false);
}
exports.ContextualFlowInfoLayoutEnumFromJSON = ContextualFlowInfoLayoutEnumFromJSON;
function ContextualFlowInfoLayoutEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ContextualFlowInfoLayoutEnumFromJSONTyped = ContextualFlowInfoLayoutEnumFromJSONTyped;
function ContextualFlowInfoLayoutEnumToJSON(value) {
    return value;
}
exports.ContextualFlowInfoLayoutEnumToJSON = ContextualFlowInfoLayoutEnumToJSON;
