// @goauthentik/api/dist/models/ContextualFlowInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContextualFlowInfoToJSON = exports.ContextualFlowInfoFromJSONTyped = exports.ContextualFlowInfoFromJSON = exports.instanceOfContextualFlowInfo = void 0;
const runtime_1 = require("../runtime");
const ContextualFlowInfoLayoutEnum_1 = require("./ContextualFlowInfoLayoutEnum");
/**
 * Check if a given object implements the ContextualFlowInfo interface.
 */
function instanceOfContextualFlowInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "cancelUrl" in value;
    isInstance = isInstance && "layout" in value;
    return isInstance;
}
exports.instanceOfContextualFlowInfo = instanceOfContextualFlowInfo;
function ContextualFlowInfoFromJSON(json) {
    return ContextualFlowInfoFromJSONTyped(json, false);
}
exports.ContextualFlowInfoFromJSON = ContextualFlowInfoFromJSON;
function ContextualFlowInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'title': !(0, runtime_1.exists)(json, 'title') ? undefined : json['title'],
        'background': !(0, runtime_1.exists)(json, 'background') ? undefined : json['background'],
        'cancelUrl': json['cancel_url'],
        'layout': (0, ContextualFlowInfoLayoutEnum_1.ContextualFlowInfoLayoutEnumFromJSON)(json['layout']),
    };
}
exports.ContextualFlowInfoFromJSONTyped = ContextualFlowInfoFromJSONTyped;
function ContextualFlowInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'title': value.title,
        'background': value.background,
        'cancel_url': value.cancelUrl,
        'layout': (0, ContextualFlowInfoLayoutEnum_1.ContextualFlowInfoLayoutEnumToJSON)(value.layout),
    };
}
exports.ContextualFlowInfoToJSON = ContextualFlowInfoToJSON;
