// @goauthentik/api/dist/models/Coordinate.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoordinateToJSON = exports.CoordinateFromJSONTyped = exports.CoordinateFromJSON = exports.instanceOfCoordinate = void 0;
/**
 * Check if a given object implements the Coordinate interface.
 */
function instanceOfCoordinate(value) {
    let isInstance = true;
    isInstance = isInstance && "xCord" in value;
    isInstance = isInstance && "yCord" in value;
    return isInstance;
}
exports.instanceOfCoordinate = instanceOfCoordinate;
function CoordinateFromJSON(json) {
    return CoordinateFromJSONTyped(json, false);
}
exports.CoordinateFromJSON = CoordinateFromJSON;
function CoordinateFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'xCord': json['x_cord'],
        'yCord': json['y_cord'],
    };
}
exports.CoordinateFromJSONTyped = CoordinateFromJSONTyped;
function CoordinateToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.CoordinateToJSON = CoordinateToJSON;
