// @goauthentik/api/dist/models/CurrentBrand.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrentBrandToJSON = exports.CurrentBrandFromJSONTyped = exports.CurrentBrandFromJSON = exports.instanceOfCurrentBrand = void 0;
const runtime_1 = require("../runtime");
const FooterLink_1 = require("./FooterLink");
const UiThemeEnum_1 = require("./UiThemeEnum");
/**
 * Check if a given object implements the CurrentBrand interface.
 */
function instanceOfCurrentBrand(value) {
    let isInstance = true;
    isInstance = isInstance && "matchedDomain" in value;
    isInstance = isInstance && "brandingTitle" in value;
    isInstance = isInstance && "brandingLogo" in value;
    isInstance = isInstance && "brandingFavicon" in value;
    isInstance = isInstance && "uiFooterLinks" in value;
    isInstance = isInstance && "uiTheme" in value;
    isInstance = isInstance && "defaultLocale" in value;
    return isInstance;
}
exports.instanceOfCurrentBrand = instanceOfCurrentBrand;
function CurrentBrandFromJSON(json) {
    return CurrentBrandFromJSONTyped(json, false);
}
exports.CurrentBrandFromJSON = CurrentBrandFromJSON;
function CurrentBrandFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'matchedDomain': json['matched_domain'],
        'brandingTitle': json['branding_title'],
        'brandingLogo': json['branding_logo'],
        'brandingFavicon': json['branding_favicon'],
        'uiFooterLinks': (json['ui_footer_links'].map(FooterLink_1.FooterLinkFromJSON)),
        'uiTheme': (0, UiThemeEnum_1.UiThemeEnumFromJSON)(json['ui_theme']),
        'flowAuthentication': !(0, runtime_1.exists)(json, 'flow_authentication') ? undefined : json['flow_authentication'],
        'flowInvalidation': !(0, runtime_1.exists)(json, 'flow_invalidation') ? undefined : json['flow_invalidation'],
        'flowRecovery': !(0, runtime_1.exists)(json, 'flow_recovery') ? undefined : json['flow_recovery'],
        'flowUnenrollment': !(0, runtime_1.exists)(json, 'flow_unenrollment') ? undefined : json['flow_unenrollment'],
        'flowUserSettings': !(0, runtime_1.exists)(json, 'flow_user_settings') ? undefined : json['flow_user_settings'],
        'flowDeviceCode': !(0, runtime_1.exists)(json, 'flow_device_code') ? undefined : json['flow_device_code'],
        'defaultLocale': json['default_locale'],
    };
}
exports.CurrentBrandFromJSONTyped = CurrentBrandFromJSONTyped;
function CurrentBrandToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'matched_domain': value.matchedDomain,
        'branding_title': value.brandingTitle,
        'branding_logo': value.brandingLogo,
        'branding_favicon': value.brandingFavicon,
        'flow_authentication': value.flowAuthentication,
        'flow_invalidation': value.flowInvalidation,
        'flow_recovery': value.flowRecovery,
        'flow_unenrollment': value.flowUnenrollment,
        'flow_user_settings': value.flowUserSettings,
        'flow_device_code': value.flowDeviceCode,
    };
}
exports.CurrentBrandToJSON = CurrentBrandToJSON;
