// @goauthentik/api/dist/models/DeniedActionEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeniedActionEnumToJSON = exports.DeniedActionEnumFromJSONTyped = exports.DeniedActionEnumFromJSON = exports.DeniedActionEnum = void 0;
/**
 * * `message_continue` - Message Continue
 * * `message` - Message
 * * `continue` - Continue
 * @export
 */
exports.DeniedActionEnum = {
    MessageContinue: 'message_continue',
    Message: 'message',
    Continue: 'continue',
    UnknownDefaultOpenApi: '11184809'
};
function DeniedActionEnumFromJSON(json) {
    return DeniedActionEnumFromJSONTyped(json, false);
}
exports.DeniedActionEnumFromJSON = DeniedActionEnumFromJSON;
function DeniedActionEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.DeniedActionEnumFromJSONTyped = DeniedActionEnumFromJSONTyped;
function DeniedActionEnumToJSON(value) {
    return value;
}
exports.DeniedActionEnumToJSON = DeniedActionEnumToJSON;
