// @goauthentik/api/dist/models/DeviceChallengeRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceChallengeRequestToJSON = exports.DeviceChallengeRequestFromJSONTyped = exports.DeviceChallengeRequestFromJSON = exports.instanceOfDeviceChallengeRequest = void 0;
/**
 * Check if a given object implements the DeviceChallengeRequest interface.
 */
function instanceOfDeviceChallengeRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "deviceClass" in value;
    isInstance = isInstance && "deviceUid" in value;
    isInstance = isInstance && "challenge" in value;
    return isInstance;
}
exports.instanceOfDeviceChallengeRequest = instanceOfDeviceChallengeRequest;
function DeviceChallengeRequestFromJSON(json) {
    return DeviceChallengeRequestFromJSONTyped(json, false);
}
exports.DeviceChallengeRequestFromJSON = DeviceChallengeRequestFromJSON;
function DeviceChallengeRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deviceClass': json['device_class'],
        'deviceUid': json['device_uid'],
        'challenge': json['challenge'],
    };
}
exports.DeviceChallengeRequestFromJSONTyped = DeviceChallengeRequestFromJSONTyped;
function DeviceChallengeRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'device_class': value.deviceClass,
        'device_uid': value.deviceUid,
        'challenge': value.challenge,
    };
}
exports.DeviceChallengeRequestToJSON = DeviceChallengeRequestToJSON;
