// @goauthentik/api/dist/models/DeviceClassesEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceClassesEnumToJSON = exports.DeviceClassesEnumFromJSONTyped = exports.DeviceClassesEnumFromJSON = exports.DeviceClassesEnum = void 0;
/**
 * * `static` - Static
 * * `totp` - TOTP
 * * `webauthn` - WebAuthn
 * * `duo` - Duo
 * * `sms` - SMS
 * @export
 */
exports.DeviceClassesEnum = {
    Static: 'static',
    Totp: 'totp',
    Webauthn: 'webauthn',
    Duo: 'duo',
    Sms: 'sms',
    UnknownDefaultOpenApi: '11184809'
};
function DeviceClassesEnumFromJSON(json) {
    return DeviceClassesEnumFromJSONTyped(json, false);
}
exports.DeviceClassesEnumFromJSON = DeviceClassesEnumFromJSON;
function DeviceClassesEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.DeviceClassesEnumFromJSONTyped = DeviceClassesEnumFromJSONTyped;
function DeviceClassesEnumToJSON(value) {
    return value;
}
exports.DeviceClassesEnumToJSON = DeviceClassesEnumToJSON;
