// @goauthentik/api/dist/models/DeviceChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceChallengeToJSON = exports.DeviceChallengeFromJSONTyped = exports.DeviceChallengeFromJSON = exports.instanceOfDeviceChallenge = void 0;
/**
 * Check if a given object implements the DeviceChallenge interface.
 */
function instanceOfDeviceChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "deviceClass" in value;
    isInstance = isInstance && "deviceUid" in value;
    isInstance = isInstance && "challenge" in value;
    return isInstance;
}
exports.instanceOfDeviceChallenge = instanceOfDeviceChallenge;
function DeviceChallengeFromJSON(json) {
    return DeviceChallengeFromJSONTyped(json, false);
}
exports.DeviceChallengeFromJSON = DeviceChallengeFromJSON;
function DeviceChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deviceClass': json['device_class'],
        'deviceUid': json['device_uid'],
        'challenge': json['challenge'],
    };
}
exports.DeviceChallengeFromJSONTyped = DeviceChallengeFromJSONTyped;
function DeviceChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'device_class': value.deviceClass,
        'device_uid': value.deviceUid,
        'challenge': value.challenge,
    };
}
exports.DeviceChallengeToJSON = DeviceChallengeToJSON;
