// @goauthentik/api/dist/models/Device.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceToJSON = exports.DeviceFromJSONTyped = exports.DeviceFromJSON = exports.instanceOfDevice = void 0;
/**
 * Check if a given object implements the Device interface.
 */
function instanceOfDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "confirmed" in value;
    return isInstance;
}
exports.instanceOfDevice = instanceOfDevice;
function DeviceFromJSON(json) {
    return DeviceFromJSONTyped(json, false);
}
exports.DeviceFromJSON = DeviceFromJSON;
function DeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'pk': json['pk'],
        'name': json['name'],
        'type': json['type'],
        'confirmed': json['confirmed'],
    };
}
exports.DeviceFromJSONTyped = DeviceFromJSONTyped;
function DeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pk': value.pk,
        'name': value.name,
        'confirmed': value.confirmed,
    };
}
exports.DeviceToJSON = DeviceToJSON;
