// @goauthentik/api/dist/models/DigitsEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DigitsEnumToJSON = exports.DigitsEnumFromJSONTyped = exports.DigitsEnumFromJSON = exports.DigitsEnum = void 0;
/**
 * * `6` - 6 digits, widely compatible
 * * `8` - 8 digits, not compatible with apps like Google Authenticator
 * @export
 */
exports.DigitsEnum = {
    _6: '6',
    _8: '8',
    UnknownDefaultOpenApi: '11184809'
};
function DigitsEnumFromJSON(json) {
    return DigitsEnumFromJSONTyped(json, false);
}
exports.DigitsEnumFromJSON = DigitsEnumFromJSON;
function DigitsEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.DigitsEnumFromJSONTyped = DigitsEnumFromJSONTyped;
function DigitsEnumToJSON(value) {
    return value;
}
exports.DigitsEnumToJSON = DigitsEnumToJSON;
