// @goauthentik/api/dist/models/DockerServiceConnection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DockerServiceConnectionToJSON = exports.DockerServiceConnectionFromJSONTyped = exports.DockerServiceConnectionFromJSON = exports.instanceOfDockerServiceConnection = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DockerServiceConnection interface.
 */
function instanceOfDockerServiceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfDockerServiceConnection = instanceOfDockerServiceConnection;
function DockerServiceConnectionFromJSON(json) {
    return DockerServiceConnectionFromJSONTyped(json, false);
}
exports.DockerServiceConnectionFromJSON = DockerServiceConnectionFromJSON;
function DockerServiceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'url': json['url'],
        'tlsVerification': !(0, runtime_1.exists)(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !(0, runtime_1.exists)(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}
exports.DockerServiceConnectionFromJSONTyped = DockerServiceConnectionFromJSONTyped;
function DockerServiceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
exports.DockerServiceConnectionToJSON = DockerServiceConnectionToJSON;
