// @goauthentik/api/dist/models/DockerServiceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DockerServiceConnectionRequestToJSON = exports.DockerServiceConnectionRequestFromJSONTyped = exports.DockerServiceConnectionRequestFromJSON = exports.instanceOfDockerServiceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DockerServiceConnectionRequest interface.
 */
function instanceOfDockerServiceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfDockerServiceConnectionRequest = instanceOfDockerServiceConnectionRequest;
function DockerServiceConnectionRequestFromJSON(json) {
    return DockerServiceConnectionRequestFromJSONTyped(json, false);
}
exports.DockerServiceConnectionRequestFromJSON = DockerServiceConnectionRequestFromJSON;
function DockerServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'url': json['url'],
        'tlsVerification': !(0, runtime_1.exists)(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !(0, runtime_1.exists)(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}
exports.DockerServiceConnectionRequestFromJSONTyped = DockerServiceConnectionRequestFromJSONTyped;
function DockerServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
exports.DockerServiceConnectionRequestToJSON = DockerServiceConnectionRequestToJSON;
