// @goauthentik/api/dist/models/DomainRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainRequestToJSON = exports.DomainRequestFromJSONTyped = exports.DomainRequestFromJSON = exports.instanceOfDomainRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DomainRequest interface.
 */
function instanceOfDomainRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "domain" in value;
    isInstance = isInstance && "tenant" in value;
    return isInstance;
}
exports.instanceOfDomainRequest = instanceOfDomainRequest;
function DomainRequestFromJSON(json) {
    return DomainRequestFromJSONTyped(json, false);
}
exports.DomainRequestFromJSON = DomainRequestFromJSON;
function DomainRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'domain': json['domain'],
        'isPrimary': !(0, runtime_1.exists)(json, 'is_primary') ? undefined : json['is_primary'],
        'tenant': json['tenant'],
    };
}
exports.DomainRequestFromJSONTyped = DomainRequestFromJSONTyped;
function DomainRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'domain': value.domain,
        'is_primary': value.isPrimary,
        'tenant': value.tenant,
    };
}
exports.DomainRequestToJSON = DomainRequestToJSON;
