// @goauthentik/api/dist/models/Domain.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainToJSON = exports.DomainFromJSONTyped = exports.DomainFromJSON = exports.instanceOfDomain = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the Domain interface.
 */
function instanceOfDomain(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "domain" in value;
    isInstance = isInstance && "tenant" in value;
    return isInstance;
}
exports.instanceOfDomain = instanceOfDomain;
function DomainFromJSON(json) {
    return DomainFromJSONTyped(json, false);
}
exports.DomainFromJSON = DomainFromJSON;
function DomainFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'domain': json['domain'],
        'isPrimary': !(0, runtime_1.exists)(json, 'is_primary') ? undefined : json['is_primary'],
        'tenant': json['tenant'],
    };
}
exports.DomainFromJSONTyped = DomainFromJSONTyped;
function DomainToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'domain': value.domain,
        'is_primary': value.isPrimary,
        'tenant': value.tenant,
    };
}
exports.DomainToJSON = DomainToJSON;
