// @goauthentik/api/dist/models/DummyPolicy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DummyPolicyToJSON = exports.DummyPolicyFromJSONTyped = exports.DummyPolicyFromJSON = exports.instanceOfDummyPolicy = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DummyPolicy interface.
 */
function instanceOfDummyPolicy(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;
    return isInstance;
}
exports.instanceOfDummyPolicy = instanceOfDummyPolicy;
function DummyPolicyFromJSON(json) {
    return DummyPolicyFromJSONTyped(json, false);
}
exports.DummyPolicyFromJSON = DummyPolicyFromJSON;
function DummyPolicyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'result': !(0, runtime_1.exists)(json, 'result') ? undefined : json['result'],
        'waitMin': !(0, runtime_1.exists)(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !(0, runtime_1.exists)(json, 'wait_max') ? undefined : json['wait_max'],
    };
}
exports.DummyPolicyFromJSONTyped = DummyPolicyFromJSONTyped;
function DummyPolicyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
exports.DummyPolicyToJSON = DummyPolicyToJSON;
