// @goauthentik/api/dist/models/DummyPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DummyPolicyRequestToJSON = exports.DummyPolicyRequestFromJSONTyped = exports.DummyPolicyRequestFromJSON = exports.instanceOfDummyPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DummyPolicyRequest interface.
 */
function instanceOfDummyPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDummyPolicyRequest = instanceOfDummyPolicyRequest;
function DummyPolicyRequestFromJSON(json) {
    return DummyPolicyRequestFromJSONTyped(json, false);
}
exports.DummyPolicyRequestFromJSON = DummyPolicyRequestFromJSON;
function DummyPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'result': !(0, runtime_1.exists)(json, 'result') ? undefined : json['result'],
        'waitMin': !(0, runtime_1.exists)(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !(0, runtime_1.exists)(json, 'wait_max') ? undefined : json['wait_max'],
    };
}
exports.DummyPolicyRequestFromJSONTyped = DummyPolicyRequestFromJSONTyped;
function DummyPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
exports.DummyPolicyRequestToJSON = DummyPolicyRequestToJSON;
