// @goauthentik/api/dist/models/DummyStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DummyStageRequestToJSON = exports.DummyStageRequestFromJSONTyped = exports.DummyStageRequestFromJSON = exports.instanceOfDummyStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the DummyStageRequest interface.
 */
function instanceOfDummyStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDummyStageRequest = instanceOfDummyStageRequest;
function DummyStageRequestFromJSON(json) {
    return DummyStageRequestFromJSONTyped(json, false);
}
exports.DummyStageRequestFromJSON = DummyStageRequestFromJSON;
function DummyStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'throwError': !(0, runtime_1.exists)(json, 'throw_error') ? undefined : json['throw_error'],
    };
}
exports.DummyStageRequestFromJSONTyped = DummyStageRequestFromJSONTyped;
function DummyStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'throw_error': value.throwError,
    };
}
exports.DummyStageRequestToJSON = DummyStageRequestToJSON;
