// @goauthentik/api/dist/models/DummyChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DummyChallengeResponseRequestToJSON = exports.DummyChallengeResponseRequestFromJSONTyped = exports.DummyChallengeResponseRequestFromJSON = exports.instanceOfDummyChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DummyChallengeResponseRequest interface.
 */
function instanceOfDummyChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfDummyChallengeResponseRequest = instanceOfDummyChallengeResponseRequest;
function DummyChallengeResponseRequestFromJSON(json) {
    return DummyChallengeResponseRequestFromJSONTyped(json, false);
}
exports.DummyChallengeResponseRequestFromJSON = DummyChallengeResponseRequestFromJSON;
function DummyChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
    };
}
exports.DummyChallengeResponseRequestFromJSONTyped = DummyChallengeResponseRequestFromJSONTyped;
function DummyChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
    };
}
exports.DummyChallengeResponseRequestToJSON = DummyChallengeResponseRequestToJSON;
