// @goauthentik/api/dist/models/DuoDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DuoDeviceRequestToJSON = exports.DuoDeviceRequestFromJSONTyped = exports.DuoDeviceRequestFromJSON = exports.instanceOfDuoDeviceRequest = void 0;
/**
 * Check if a given object implements the DuoDeviceRequest interface.
 */
function instanceOfDuoDeviceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDuoDeviceRequest = instanceOfDuoDeviceRequest;
function DuoDeviceRequestFromJSON(json) {
    return DuoDeviceRequestFromJSONTyped(json, false);
}
exports.DuoDeviceRequestFromJSON = DuoDeviceRequestFromJSON;
function DuoDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.DuoDeviceRequestFromJSONTyped = DuoDeviceRequestFromJSONTyped;
function DuoDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.DuoDeviceRequestToJSON = DuoDeviceRequestToJSON;
