// @goauthentik/api/dist/models/DuoDeviceEnrollmentStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DuoDeviceEnrollmentStatusToJSON = exports.DuoDeviceEnrollmentStatusFromJSONTyped = exports.DuoDeviceEnrollmentStatusFromJSON = exports.instanceOfDuoDeviceEnrollmentStatus = void 0;
const DuoResponseEnum_1 = require("./DuoResponseEnum");
/**
 * Check if a given object implements the DuoDeviceEnrollmentStatus interface.
 */
function instanceOfDuoDeviceEnrollmentStatus(value) {
    let isInstance = true;
    isInstance = isInstance && "duoResponse" in value;
    return isInstance;
}
exports.instanceOfDuoDeviceEnrollmentStatus = instanceOfDuoDeviceEnrollmentStatus;
function DuoDeviceEnrollmentStatusFromJSON(json) {
    return DuoDeviceEnrollmentStatusFromJSONTyped(json, false);
}
exports.DuoDeviceEnrollmentStatusFromJSON = DuoDeviceEnrollmentStatusFromJSON;
function DuoDeviceEnrollmentStatusFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'duoResponse': (0, DuoResponseEnum_1.DuoResponseEnumFromJSON)(json['duo_response']),
    };
}
exports.DuoDeviceEnrollmentStatusFromJSONTyped = DuoDeviceEnrollmentStatusFromJSONTyped;
function DuoDeviceEnrollmentStatusToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'duo_response': (0, DuoResponseEnum_1.DuoResponseEnumToJSON)(value.duoResponse),
    };
}
exports.DuoDeviceEnrollmentStatusToJSON = DuoDeviceEnrollmentStatusToJSON;
