// @goauthentik/api/dist/models/DuoDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DuoDeviceToJSON = exports.DuoDeviceFromJSONTyped = exports.DuoDeviceFromJSON = exports.instanceOfDuoDevice = void 0;
/**
 * Check if a given object implements the DuoDevice interface.
 */
function instanceOfDuoDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDuoDevice = instanceOfDuoDevice;
function DuoDeviceFromJSON(json) {
    return DuoDeviceFromJSONTyped(json, false);
}
exports.DuoDeviceFromJSON = DuoDeviceFromJSON;
function DuoDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
    };
}
exports.DuoDeviceFromJSONTyped = DuoDeviceFromJSONTyped;
function DuoDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.DuoDeviceToJSON = DuoDeviceToJSON;
