// @goauthentik/api/dist/models/DuoResponseEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DuoResponseEnumToJSON = exports.DuoResponseEnumFromJSONTyped = exports.DuoResponseEnumFromJSON = exports.DuoResponseEnum = void 0;
/**
 * * `success` - Success
 * * `waiting` - Waiting
 * * `invalid` - Invalid
 * @export
 */
exports.DuoResponseEnum = {
    Success: 'success',
    Waiting: 'waiting',
    Invalid: 'invalid',
    UnknownDefaultOpenApi: '11184809'
};
function DuoResponseEnumFromJSON(json) {
    return DuoResponseEnumFromJSONTyped(json, false);
}
exports.DuoResponseEnumFromJSON = DuoResponseEnumFromJSON;
function DuoResponseEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.DuoResponseEnumFromJSONTyped = DuoResponseEnumFromJSONTyped;
function DuoResponseEnumToJSON(value) {
    return value;
}
exports.DuoResponseEnumToJSON = DuoResponseEnumToJSON;
