// @goauthentik/api/dist/models/EmailStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailStageToJSON = exports.EmailStageFromJSONTyped = exports.EmailStageFromJSON = exports.instanceOfEmailStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the EmailStage interface.
 */
function instanceOfEmailStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfEmailStage = instanceOfEmailStage;
function EmailStageFromJSON(json) {
    return EmailStageFromJSONTyped(json, false);
}
exports.EmailStageFromJSON = EmailStageFromJSON;
function EmailStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'useGlobalSettings': !(0, runtime_1.exists)(json, 'use_global_settings') ? undefined : json['use_global_settings'],
        'host': !(0, runtime_1.exists)(json, 'host') ? undefined : json['host'],
        'port': !(0, runtime_1.exists)(json, 'port') ? undefined : json['port'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'useTls': !(0, runtime_1.exists)(json, 'use_tls') ? undefined : json['use_tls'],
        'useSsl': !(0, runtime_1.exists)(json, 'use_ssl') ? undefined : json['use_ssl'],
        'timeout': !(0, runtime_1.exists)(json, 'timeout') ? undefined : json['timeout'],
        'fromAddress': !(0, runtime_1.exists)(json, 'from_address') ? undefined : json['from_address'],
        'tokenExpiry': !(0, runtime_1.exists)(json, 'token_expiry') ? undefined : json['token_expiry'],
        'subject': !(0, runtime_1.exists)(json, 'subject') ? undefined : json['subject'],
        'template': !(0, runtime_1.exists)(json, 'template') ? undefined : json['template'],
        'activateUserOnSuccess': !(0, runtime_1.exists)(json, 'activate_user_on_success') ? undefined : json['activate_user_on_success'],
    };
}
exports.EmailStageFromJSONTyped = EmailStageFromJSONTyped;
function EmailStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'use_global_settings': value.useGlobalSettings,
        'host': value.host,
        'port': value.port,
        'username': value.username,
        'use_tls': value.useTls,
        'use_ssl': value.useSsl,
        'timeout': value.timeout,
        'from_address': value.fromAddress,
        'token_expiry': value.tokenExpiry,
        'subject': value.subject,
        'template': value.template,
        'activate_user_on_success': value.activateUserOnSuccess,
    };
}
exports.EmailStageToJSON = EmailStageToJSON;
