// @goauthentik/api/dist/models/EndpointRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointRequestToJSON = exports.EndpointRequestFromJSONTyped = exports.EndpointRequestFromJSON = exports.instanceOfEndpointRequest = void 0;
const runtime_1 = require("../runtime");
const AuthModeEnum_1 = require("./AuthModeEnum");
const ProtocolEnum_1 = require("./ProtocolEnum");
/**
 * Check if a given object implements the EndpointRequest interface.
 */
function instanceOfEndpointRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "protocol" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "authMode" in value;
    return isInstance;
}
exports.instanceOfEndpointRequest = instanceOfEndpointRequest;
function EndpointRequestFromJSON(json) {
    return EndpointRequestFromJSONTyped(json, false);
}
exports.EndpointRequestFromJSON = EndpointRequestFromJSON;
function EndpointRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'provider': json['provider'],
        'protocol': (0, ProtocolEnum_1.ProtocolEnumFromJSON)(json['protocol']),
        'host': json['host'],
        'settings': !(0, runtime_1.exists)(json, 'settings') ? undefined : json['settings'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'authMode': (0, AuthModeEnum_1.AuthModeEnumFromJSON)(json['auth_mode']),
        'maximumConnections': !(0, runtime_1.exists)(json, 'maximum_connections') ? undefined : json['maximum_connections'],
    };
}
exports.EndpointRequestFromJSONTyped = EndpointRequestFromJSONTyped;
function EndpointRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'provider': value.provider,
        'protocol': (0, ProtocolEnum_1.ProtocolEnumToJSON)(value.protocol),
        'host': value.host,
        'settings': value.settings,
        'property_mappings': value.propertyMappings,
        'auth_mode': (0, AuthModeEnum_1.AuthModeEnumToJSON)(value.authMode),
        'maximum_connections': value.maximumConnections,
    };
}
exports.EndpointRequestToJSON = EndpointRequestToJSON;
