// @goauthentik/api/dist/models/Endpoint.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointToJSON = exports.EndpointFromJSONTyped = exports.EndpointFromJSON = exports.instanceOfEndpoint = void 0;
const runtime_1 = require("../runtime");
const AuthModeEnum_1 = require("./AuthModeEnum");
const ProtocolEnum_1 = require("./ProtocolEnum");
const RACProvider_1 = require("./RACProvider");
/**
 * Check if a given object implements the Endpoint interface.
 */
function instanceOfEndpoint(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "providerObj" in value;
    isInstance = isInstance && "protocol" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "authMode" in value;
    isInstance = isInstance && "launchUrl" in value;
    return isInstance;
}
exports.instanceOfEndpoint = instanceOfEndpoint;
function EndpointFromJSON(json) {
    return EndpointFromJSONTyped(json, false);
}
exports.EndpointFromJSON = EndpointFromJSON;
function EndpointFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'provider': json['provider'],
        'providerObj': (0, RACProvider_1.RACProviderFromJSON)(json['provider_obj']),
        'protocol': (0, ProtocolEnum_1.ProtocolEnumFromJSON)(json['protocol']),
        'host': json['host'],
        'settings': !(0, runtime_1.exists)(json, 'settings') ? undefined : json['settings'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'authMode': (0, AuthModeEnum_1.AuthModeEnumFromJSON)(json['auth_mode']),
        'launchUrl': json['launch_url'],
        'maximumConnections': !(0, runtime_1.exists)(json, 'maximum_connections') ? undefined : json['maximum_connections'],
    };
}
exports.EndpointFromJSONTyped = EndpointFromJSONTyped;
function EndpointToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'provider': value.provider,
        'protocol': (0, ProtocolEnum_1.ProtocolEnumToJSON)(value.protocol),
        'host': value.host,
        'settings': value.settings,
        'property_mappings': value.propertyMappings,
        'auth_mode': (0, AuthModeEnum_1.AuthModeEnumToJSON)(value.authMode),
        'maximum_connections': value.maximumConnections,
    };
}
exports.EndpointToJSON = EndpointToJSON;
