// @goauthentik/api/dist/models/ErrorReportingConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorReportingConfigToJSON = exports.ErrorReportingConfigFromJSONTyped = exports.ErrorReportingConfigFromJSON = exports.instanceOfErrorReportingConfig = void 0;
/**
 * Check if a given object implements the ErrorReportingConfig interface.
 */
function instanceOfErrorReportingConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "sentryDsn" in value;
    isInstance = isInstance && "environment" in value;
    isInstance = isInstance && "sendPii" in value;
    isInstance = isInstance && "tracesSampleRate" in value;
    return isInstance;
}
exports.instanceOfErrorReportingConfig = instanceOfErrorReportingConfig;
function ErrorReportingConfigFromJSON(json) {
    return ErrorReportingConfigFromJSONTyped(json, false);
}
exports.ErrorReportingConfigFromJSON = ErrorReportingConfigFromJSON;
function ErrorReportingConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'enabled': json['enabled'],
        'sentryDsn': json['sentry_dsn'],
        'environment': json['environment'],
        'sendPii': json['send_pii'],
        'tracesSampleRate': json['traces_sample_rate'],
    };
}
exports.ErrorReportingConfigFromJSONTyped = ErrorReportingConfigFromJSONTyped;
function ErrorReportingConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.ErrorReportingConfigToJSON = ErrorReportingConfigToJSON;
