// @goauthentik/api/dist/models/EventMatcherPolicy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventMatcherPolicyToJSON = exports.EventMatcherPolicyFromJSONTyped = exports.EventMatcherPolicyFromJSON = exports.instanceOfEventMatcherPolicy = void 0;
const runtime_1 = require("../runtime");
const AppEnum_1 = require("./AppEnum");
const EventActions_1 = require("./EventActions");
const ModelEnum_1 = require("./ModelEnum");
/**
 * Check if a given object implements the EventMatcherPolicy interface.
 */
function instanceOfEventMatcherPolicy(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;
    return isInstance;
}
exports.instanceOfEventMatcherPolicy = instanceOfEventMatcherPolicy;
function EventMatcherPolicyFromJSON(json) {
    return EventMatcherPolicyFromJSONTyped(json, false);
}
exports.EventMatcherPolicyFromJSON = EventMatcherPolicyFromJSON;
function EventMatcherPolicyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'action': !(0, runtime_1.exists)(json, 'action') ? undefined : (0, EventActions_1.EventActionsFromJSON)(json['action']),
        'clientIp': !(0, runtime_1.exists)(json, 'client_ip') ? undefined : json['client_ip'],
        'app': !(0, runtime_1.exists)(json, 'app') ? undefined : (0, AppEnum_1.AppEnumFromJSON)(json['app']),
        'model': !(0, runtime_1.exists)(json, 'model') ? undefined : (0, ModelEnum_1.ModelEnumFromJSON)(json['model']),
    };
}
exports.EventMatcherPolicyFromJSONTyped = EventMatcherPolicyFromJSONTyped;
function EventMatcherPolicyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'action': (0, EventActions_1.EventActionsToJSON)(value.action),
        'client_ip': value.clientIp,
        'app': (0, AppEnum_1.AppEnumToJSON)(value.app),
        'model': (0, ModelEnum_1.ModelEnumToJSON)(value.model),
    };
}
exports.EventMatcherPolicyToJSON = EventMatcherPolicyToJSON;
