// @goauthentik/api/dist/models/EventTopPerUser.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventTopPerUserToJSON = exports.EventTopPerUserFromJSONTyped = exports.EventTopPerUserFromJSON = exports.instanceOfEventTopPerUser = void 0;
/**
 * Check if a given object implements the EventTopPerUser interface.
 */
function instanceOfEventTopPerUser(value) {
    let isInstance = true;
    isInstance = isInstance && "application" in value;
    isInstance = isInstance && "countedEvents" in value;
    isInstance = isInstance && "uniqueUsers" in value;
    return isInstance;
}
exports.instanceOfEventTopPerUser = instanceOfEventTopPerUser;
function EventTopPerUserFromJSON(json) {
    return EventTopPerUserFromJSONTyped(json, false);
}
exports.EventTopPerUserFromJSON = EventTopPerUserFromJSON;
function EventTopPerUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'application': json['application'],
        'countedEvents': json['counted_events'],
        'uniqueUsers': json['unique_users'],
    };
}
exports.EventTopPerUserFromJSONTyped = EventTopPerUserFromJSONTyped;
function EventTopPerUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'application': value.application,
        'counted_events': value.countedEvents,
        'unique_users': value.uniqueUsers,
    };
}
exports.EventTopPerUserToJSON = EventTopPerUserToJSON;
