// @goauthentik/api/dist/models/ExpiringBaseGrantModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpiringBaseGrantModelToJSON = exports.ExpiringBaseGrantModelFromJSONTyped = exports.ExpiringBaseGrantModelFromJSON = exports.instanceOfExpiringBaseGrantModel = void 0;
const runtime_1 = require("../runtime");
const OAuth2Provider_1 = require("./OAuth2Provider");
const User_1 = require("./User");
/**
 * Check if a given object implements the ExpiringBaseGrantModel interface.
 */
function instanceOfExpiringBaseGrantModel(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "provider" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "isExpired" in value;
    isInstance = isInstance && "scope" in value;
    return isInstance;
}
exports.instanceOfExpiringBaseGrantModel = instanceOfExpiringBaseGrantModel;
function ExpiringBaseGrantModelFromJSON(json) {
    return ExpiringBaseGrantModelFromJSONTyped(json, false);
}
exports.ExpiringBaseGrantModelFromJSON = ExpiringBaseGrantModelFromJSON;
function ExpiringBaseGrantModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'provider': (0, OAuth2Provider_1.OAuth2ProviderFromJSON)(json['provider']),
        'user': (0, User_1.UserFromJSON)(json['user']),
        'isExpired': json['is_expired'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'scope': json['scope'],
    };
}
exports.ExpiringBaseGrantModelFromJSONTyped = ExpiringBaseGrantModelFromJSONTyped;
function ExpiringBaseGrantModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': (0, OAuth2Provider_1.OAuth2ProviderToJSON)(value.provider),
        'user': (0, User_1.UserToJSON)(value.user),
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'scope': value.scope,
    };
}
exports.ExpiringBaseGrantModelToJSON = ExpiringBaseGrantModelToJSON;
