// @goauthentik/api/dist/models/ExpressionPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressionPolicyRequestToJSON = exports.ExpressionPolicyRequestFromJSONTyped = exports.ExpressionPolicyRequestFromJSON = exports.instanceOfExpressionPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ExpressionPolicyRequest interface.
 */
function instanceOfExpressionPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    return isInstance;
}
exports.instanceOfExpressionPolicyRequest = instanceOfExpressionPolicyRequest;
function ExpressionPolicyRequestFromJSON(json) {
    return ExpressionPolicyRequestFromJSONTyped(json, false);
}
exports.ExpressionPolicyRequestFromJSON = ExpressionPolicyRequestFromJSON;
function ExpressionPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'expression': json['expression'],
    };
}
exports.ExpressionPolicyRequestFromJSONTyped = ExpressionPolicyRequestFromJSONTyped;
function ExpressionPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'expression': value.expression,
    };
}
exports.ExpressionPolicyRequestToJSON = ExpressionPolicyRequestToJSON;
