// @goauthentik/api/dist/models/ExtraUserObjectPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtraUserObjectPermissionToJSON = exports.ExtraUserObjectPermissionFromJSONTyped = exports.ExtraUserObjectPermissionFromJSON = exports.instanceOfExtraUserObjectPermission = void 0;
/**
 * Check if a given object implements the ExtraUserObjectPermission interface.
 */
function instanceOfExtraUserObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "objectPk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "appLabelVerbose" in value;
    isInstance = isInstance && "modelVerbose" in value;
    isInstance = isInstance && "objectDescription" in value;
    return isInstance;
}
exports.instanceOfExtraUserObjectPermission = instanceOfExtraUserObjectPermission;
function ExtraUserObjectPermissionFromJSON(json) {
    return ExtraUserObjectPermissionFromJSONTyped(json, false);
}
exports.ExtraUserObjectPermissionFromJSON = ExtraUserObjectPermissionFromJSON;
function ExtraUserObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'objectPk': json['object_pk'],
        'name': json['name'],
        'appLabelVerbose': json['app_label_verbose'],
        'modelVerbose': json['model_verbose'],
        'objectDescription': json['object_description'],
    };
}
exports.ExtraUserObjectPermissionFromJSONTyped = ExtraUserObjectPermissionFromJSONTyped;
function ExtraUserObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.ExtraUserObjectPermissionToJSON = ExtraUserObjectPermissionToJSON;
