// @goauthentik/api/dist/models/FilePathRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilePathRequestToJSON = exports.FilePathRequestFromJSONTyped = exports.FilePathRequestFromJSON = exports.instanceOfFilePathRequest = void 0;
/**
 * Check if a given object implements the FilePathRequest interface.
 */
function instanceOfFilePathRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfFilePathRequest = instanceOfFilePathRequest;
function FilePathRequestFromJSON(json) {
    return FilePathRequestFromJSONTyped(json, false);
}
exports.FilePathRequestFromJSON = FilePathRequestFromJSON;
function FilePathRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'url': json['url'],
    };
}
exports.FilePathRequestFromJSONTyped = FilePathRequestFromJSONTyped;
function FilePathRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'url': value.url,
    };
}
exports.FilePathRequestToJSON = FilePathRequestToJSON;
