// @goauthentik/api/dist/models/FlowLayoutEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowLayoutEnumToJSON = exports.FlowLayoutEnumFromJSONTyped = exports.FlowLayoutEnumFromJSON = exports.FlowLayoutEnum = void 0;
/**
 * * `stacked` - Stacked
 * * `content_left` - Content Left
 * * `content_right` - Content Right
 * * `sidebar_left` - Sidebar Left
 * * `sidebar_right` - Sidebar Right
 * @export
 */
exports.FlowLayoutEnum = {
    Stacked: 'stacked',
    ContentLeft: 'content_left',
    ContentRight: 'content_right',
    SidebarLeft: 'sidebar_left',
    SidebarRight: 'sidebar_right',
    UnknownDefaultOpenApi: '11184809'
};
function FlowLayoutEnumFromJSON(json) {
    return FlowLayoutEnumFromJSONTyped(json, false);
}
exports.FlowLayoutEnumFromJSON = FlowLayoutEnumFromJSON;
function FlowLayoutEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.FlowLayoutEnumFromJSONTyped = FlowLayoutEnumFromJSONTyped;
function FlowLayoutEnumToJSON(value) {
    return value;
}
exports.FlowLayoutEnumToJSON = FlowLayoutEnumToJSON;
