// @goauthentik/api/dist/models/FlowDesignationEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowDesignationEnumToJSON = exports.FlowDesignationEnumFromJSONTyped = exports.FlowDesignationEnumFromJSON = exports.FlowDesignationEnum = void 0;
/**
 * * `authentication` - Authentication
 * * `authorization` - Authorization
 * * `invalidation` - Invalidation
 * * `enrollment` - Enrollment
 * * `unenrollment` - Unrenollment
 * * `recovery` - Recovery
 * * `stage_configuration` - Stage Configuration
 * @export
 */
exports.FlowDesignationEnum = {
    Authentication: 'authentication',
    Authorization: 'authorization',
    Invalidation: 'invalidation',
    Enrollment: 'enrollment',
    Unenrollment: 'unenrollment',
    Recovery: 'recovery',
    StageConfiguration: 'stage_configuration',
    UnknownDefaultOpenApi: '11184809'
};
function FlowDesignationEnumFromJSON(json) {
    return FlowDesignationEnumFromJSONTyped(json, false);
}
exports.FlowDesignationEnumFromJSON = FlowDesignationEnumFromJSON;
function FlowDesignationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.FlowDesignationEnumFromJSONTyped = FlowDesignationEnumFromJSONTyped;
function FlowDesignationEnumToJSON(value) {
    return value;
}
exports.FlowDesignationEnumToJSON = FlowDesignationEnumToJSON;
