// @goauthentik/api/dist/models/FlowStageBindingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowStageBindingRequestToJSON = exports.FlowStageBindingRequestFromJSONTyped = exports.FlowStageBindingRequestFromJSON = exports.instanceOfFlowStageBindingRequest = void 0;
const runtime_1 = require("../runtime");
const InvalidResponseActionEnum_1 = require("./InvalidResponseActionEnum");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
/**
 * Check if a given object implements the FlowStageBindingRequest interface.
 */
function instanceOfFlowStageBindingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "stage" in value;
    isInstance = isInstance && "order" in value;
    return isInstance;
}
exports.instanceOfFlowStageBindingRequest = instanceOfFlowStageBindingRequest;
function FlowStageBindingRequestFromJSON(json) {
    return FlowStageBindingRequestFromJSONTyped(json, false);
}
exports.FlowStageBindingRequestFromJSON = FlowStageBindingRequestFromJSON;
function FlowStageBindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'target': json['target'],
        'stage': json['stage'],
        'evaluateOnPlan': !(0, runtime_1.exists)(json, 'evaluate_on_plan') ? undefined : json['evaluate_on_plan'],
        'reEvaluatePolicies': !(0, runtime_1.exists)(json, 're_evaluate_policies') ? undefined : json['re_evaluate_policies'],
        'order': json['order'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'invalidResponseAction': !(0, runtime_1.exists)(json, 'invalid_response_action') ? undefined : (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumFromJSON)(json['invalid_response_action']),
    };
}
exports.FlowStageBindingRequestFromJSONTyped = FlowStageBindingRequestFromJSONTyped;
function FlowStageBindingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'target': value.target,
        'stage': value.stage,
        'evaluate_on_plan': value.evaluateOnPlan,
        're_evaluate_policies': value.reEvaluatePolicies,
        'order': value.order,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'invalid_response_action': (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumToJSON)(value.invalidResponseAction),
    };
}
exports.FlowStageBindingRequestToJSON = FlowStageBindingRequestToJSON;
