// @goauthentik/api/dist/models/FlowImportResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowImportResultToJSON = exports.FlowImportResultFromJSONTyped = exports.FlowImportResultFromJSON = exports.instanceOfFlowImportResult = void 0;
/**
 * Check if a given object implements the FlowImportResult interface.
 */
function instanceOfFlowImportResult(value) {
    let isInstance = true;
    isInstance = isInstance && "logs" in value;
    isInstance = isInstance && "success" in value;
    return isInstance;
}
exports.instanceOfFlowImportResult = instanceOfFlowImportResult;
function FlowImportResultFromJSON(json) {
    return FlowImportResultFromJSONTyped(json, false);
}
exports.FlowImportResultFromJSON = FlowImportResultFromJSON;
function FlowImportResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'logs': json['logs'],
        'success': json['success'],
    };
}
exports.FlowImportResultFromJSONTyped = FlowImportResultFromJSONTyped;
function FlowImportResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.FlowImportResultToJSON = FlowImportResultToJSON;
