// @goauthentik/api/dist/models/FlowStageBinding.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowStageBindingToJSON = exports.FlowStageBindingFromJSONTyped = exports.FlowStageBindingFromJSON = exports.instanceOfFlowStageBinding = void 0;
const runtime_1 = require("../runtime");
const InvalidResponseActionEnum_1 = require("./InvalidResponseActionEnum");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
const Stage_1 = require("./Stage");
/**
 * Check if a given object implements the FlowStageBinding interface.
 */
function instanceOfFlowStageBinding(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "policybindingmodelPtrId" in value;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "stage" in value;
    isInstance = isInstance && "stageObj" in value;
    isInstance = isInstance && "order" in value;
    return isInstance;
}
exports.instanceOfFlowStageBinding = instanceOfFlowStageBinding;
function FlowStageBindingFromJSON(json) {
    return FlowStageBindingFromJSONTyped(json, false);
}
exports.FlowStageBindingFromJSON = FlowStageBindingFromJSON;
function FlowStageBindingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'policybindingmodelPtrId': json['policybindingmodel_ptr_id'],
        'target': json['target'],
        'stage': json['stage'],
        'stageObj': (0, Stage_1.StageFromJSON)(json['stage_obj']),
        'evaluateOnPlan': !(0, runtime_1.exists)(json, 'evaluate_on_plan') ? undefined : json['evaluate_on_plan'],
        'reEvaluatePolicies': !(0, runtime_1.exists)(json, 're_evaluate_policies') ? undefined : json['re_evaluate_policies'],
        'order': json['order'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'invalidResponseAction': !(0, runtime_1.exists)(json, 'invalid_response_action') ? undefined : (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumFromJSON)(json['invalid_response_action']),
    };
}
exports.FlowStageBindingFromJSONTyped = FlowStageBindingFromJSONTyped;
function FlowStageBindingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'target': value.target,
        'stage': value.stage,
        'evaluate_on_plan': value.evaluateOnPlan,
        're_evaluate_policies': value.reEvaluatePolicies,
        'order': value.order,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'invalid_response_action': (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumToJSON)(value.invalidResponseAction),
    };
}
exports.FlowStageBindingToJSON = FlowStageBindingToJSON;
