// @goauthentik/api/dist/models/FlowRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowRequestToJSON = exports.FlowRequestFromJSONTyped = exports.FlowRequestFromJSON = exports.instanceOfFlowRequest = void 0;
const runtime_1 = require("../runtime");
const AuthenticationEnum_1 = require("./AuthenticationEnum");
const DeniedActionEnum_1 = require("./DeniedActionEnum");
const FlowDesignationEnum_1 = require("./FlowDesignationEnum");
const FlowLayoutEnum_1 = require("./FlowLayoutEnum");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
/**
 * Check if a given object implements the FlowRequest interface.
 */
function instanceOfFlowRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "title" in value;
    isInstance = isInstance && "designation" in value;
    return isInstance;
}
exports.instanceOfFlowRequest = instanceOfFlowRequest;
function FlowRequestFromJSON(json) {
    return FlowRequestFromJSONTyped(json, false);
}
exports.FlowRequestFromJSON = FlowRequestFromJSON;
function FlowRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'slug': json['slug'],
        'title': json['title'],
        'designation': (0, FlowDesignationEnum_1.FlowDesignationEnumFromJSON)(json['designation']),
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'compatibilityMode': !(0, runtime_1.exists)(json, 'compatibility_mode') ? undefined : json['compatibility_mode'],
        'layout': !(0, runtime_1.exists)(json, 'layout') ? undefined : (0, FlowLayoutEnum_1.FlowLayoutEnumFromJSON)(json['layout']),
        'deniedAction': !(0, runtime_1.exists)(json, 'denied_action') ? undefined : (0, DeniedActionEnum_1.DeniedActionEnumFromJSON)(json['denied_action']),
        'authentication': !(0, runtime_1.exists)(json, 'authentication') ? undefined : (0, AuthenticationEnum_1.AuthenticationEnumFromJSON)(json['authentication']),
    };
}
exports.FlowRequestFromJSONTyped = FlowRequestFromJSONTyped;
function FlowRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'title': value.title,
        'designation': (0, FlowDesignationEnum_1.FlowDesignationEnumToJSON)(value.designation),
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'compatibility_mode': value.compatibilityMode,
        'layout': (0, FlowLayoutEnum_1.FlowLayoutEnumToJSON)(value.layout),
        'denied_action': (0, DeniedActionEnum_1.DeniedActionEnumToJSON)(value.deniedAction),
        'authentication': (0, AuthenticationEnum_1.AuthenticationEnumToJSON)(value.authentication),
    };
}
exports.FlowRequestToJSON = FlowRequestToJSON;
