// @goauthentik/api/dist/models/FlowInspection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowInspectionToJSON = exports.FlowInspectionFromJSONTyped = exports.FlowInspectionFromJSON = exports.instanceOfFlowInspection = void 0;
const runtime_1 = require("../runtime");
const FlowInspectorPlan_1 = require("./FlowInspectorPlan");
/**
 * Check if a given object implements the FlowInspection interface.
 */
function instanceOfFlowInspection(value) {
    let isInstance = true;
    isInstance = isInstance && "plans" in value;
    isInstance = isInstance && "isCompleted" in value;
    return isInstance;
}
exports.instanceOfFlowInspection = instanceOfFlowInspection;
function FlowInspectionFromJSON(json) {
    return FlowInspectionFromJSONTyped(json, false);
}
exports.FlowInspectionFromJSON = FlowInspectionFromJSON;
function FlowInspectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'plans': (json['plans'].map(FlowInspectorPlan_1.FlowInspectorPlanFromJSON)),
        'currentPlan': !(0, runtime_1.exists)(json, 'current_plan') ? undefined : (0, FlowInspectorPlan_1.FlowInspectorPlanFromJSON)(json['current_plan']),
        'isCompleted': json['is_completed'],
    };
}
exports.FlowInspectionFromJSONTyped = FlowInspectionFromJSONTyped;
function FlowInspectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'plans': (value.plans.map(FlowInspectorPlan_1.FlowInspectorPlanToJSON)),
        'current_plan': (0, FlowInspectorPlan_1.FlowInspectorPlanToJSON)(value.currentPlan),
        'is_completed': value.isCompleted,
    };
}
exports.FlowInspectionToJSON = FlowInspectionToJSON;
