// @goauthentik/api/dist/models/FlowInspectorPlan.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowInspectorPlanToJSON = exports.FlowInspectorPlanFromJSONTyped = exports.FlowInspectorPlanFromJSON = exports.instanceOfFlowInspectorPlan = void 0;
const FlowStageBinding_1 = require("./FlowStageBinding");
/**
 * Check if a given object implements the FlowInspectorPlan interface.
 */
function instanceOfFlowInspectorPlan(value) {
    let isInstance = true;
    isInstance = isInstance && "currentStage" in value;
    isInstance = isInstance && "nextPlannedStage" in value;
    isInstance = isInstance && "planContext" in value;
    isInstance = isInstance && "sessionId" in value;
    return isInstance;
}
exports.instanceOfFlowInspectorPlan = instanceOfFlowInspectorPlan;
function FlowInspectorPlanFromJSON(json) {
    return FlowInspectorPlanFromJSONTyped(json, false);
}
exports.FlowInspectorPlanFromJSON = FlowInspectorPlanFromJSON;
function FlowInspectorPlanFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'currentStage': (0, FlowStageBinding_1.FlowStageBindingFromJSON)(json['current_stage']),
        'nextPlannedStage': (0, FlowStageBinding_1.FlowStageBindingFromJSON)(json['next_planned_stage']),
        'planContext': json['plan_context'],
        'sessionId': json['session_id'],
    };
}
exports.FlowInspectorPlanFromJSONTyped = FlowInspectorPlanFromJSONTyped;
function FlowInspectorPlanToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.FlowInspectorPlanToJSON = FlowInspectorPlanToJSON;
