// @goauthentik/api/dist/models/FlowErrorChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowErrorChallengeToJSON = exports.FlowErrorChallengeFromJSONTyped = exports.FlowErrorChallengeFromJSON = exports.instanceOfFlowErrorChallenge = void 0;
const runtime_1 = require("../runtime");
const ContextualFlowInfo_1 = require("./ContextualFlowInfo");
/**
 * Check if a given object implements the FlowErrorChallenge interface.
 */
function instanceOfFlowErrorChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "requestId" in value;
    return isInstance;
}
exports.instanceOfFlowErrorChallenge = instanceOfFlowErrorChallenge;
function FlowErrorChallengeFromJSON(json) {
    return FlowErrorChallengeFromJSONTyped(json, false);
}
exports.FlowErrorChallengeFromJSON = FlowErrorChallengeFromJSON;
function FlowErrorChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : json['type'],
        'flowInfo': !(0, runtime_1.exists)(json, 'flow_info') ? undefined : (0, ContextualFlowInfo_1.ContextualFlowInfoFromJSON)(json['flow_info']),
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'responseErrors': !(0, runtime_1.exists)(json, 'response_errors') ? undefined : json['response_errors'],
        'requestId': json['request_id'],
        'error': !(0, runtime_1.exists)(json, 'error') ? undefined : json['error'],
        'traceback': !(0, runtime_1.exists)(json, 'traceback') ? undefined : json['traceback'],
    };
}
exports.FlowErrorChallengeFromJSONTyped = FlowErrorChallengeFromJSONTyped;
function FlowErrorChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'flow_info': (0, ContextualFlowInfo_1.ContextualFlowInfoToJSON)(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'request_id': value.requestId,
        'error': value.error,
        'traceback': value.traceback,
    };
}
exports.FlowErrorChallengeToJSON = FlowErrorChallengeToJSON;
