// @goauthentik/api/dist/models/FlowDiagram.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowDiagramToJSON = exports.FlowDiagramFromJSONTyped = exports.FlowDiagramFromJSON = exports.instanceOfFlowDiagram = void 0;
/**
 * Check if a given object implements the FlowDiagram interface.
 */
function instanceOfFlowDiagram(value) {
    let isInstance = true;
    isInstance = isInstance && "diagram" in value;
    return isInstance;
}
exports.instanceOfFlowDiagram = instanceOfFlowDiagram;
function FlowDiagramFromJSON(json) {
    return FlowDiagramFromJSONTyped(json, false);
}
exports.FlowDiagramFromJSON = FlowDiagramFromJSON;
function FlowDiagramFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'diagram': json['diagram'],
    };
}
exports.FlowDiagramFromJSONTyped = FlowDiagramFromJSONTyped;
function FlowDiagramToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.FlowDiagramToJSON = FlowDiagramToJSON;
