// @goauthentik/api/dist/models/FooterLink.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FooterLinkToJSON = exports.FooterLinkFromJSONTyped = exports.FooterLinkFromJSON = exports.instanceOfFooterLink = void 0;
/**
 * Check if a given object implements the FooterLink interface.
 */
function instanceOfFooterLink(value) {
    let isInstance = true;
    isInstance = isInstance && "href" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfFooterLink = instanceOfFooterLink;
function FooterLinkFromJSON(json) {
    return FooterLinkFromJSONTyped(json, false);
}
exports.FooterLinkFromJSON = FooterLinkFromJSON;
function FooterLinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': json['href'],
        'name': json['name'],
    };
}
exports.FooterLinkFromJSONTyped = FooterLinkFromJSONTyped;
function FooterLinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.FooterLinkToJSON = FooterLinkToJSON;
