// @goauthentik/api/dist/models/GenericError.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericErrorToJSON = exports.GenericErrorFromJSONTyped = exports.GenericErrorFromJSON = exports.instanceOfGenericError = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the GenericError interface.
 */
function instanceOfGenericError(value) {
    let isInstance = true;
    isInstance = isInstance && "detail" in value;
    return isInstance;
}
exports.instanceOfGenericError = instanceOfGenericError;
function GenericErrorFromJSON(json) {
    return GenericErrorFromJSONTyped(json, false);
}
exports.GenericErrorFromJSON = GenericErrorFromJSON;
function GenericErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'detail': json['detail'],
        'code': !(0, runtime_1.exists)(json, 'code') ? undefined : json['code'],
    };
}
exports.GenericErrorFromJSONTyped = GenericErrorFromJSONTyped;
function GenericErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'detail': value.detail,
        'code': value.code,
    };
}
exports.GenericErrorToJSON = GenericErrorToJSON;
