// @goauthentik/api/dist/models/GroupMember.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupMemberToJSON = exports.GroupMemberFromJSONTyped = exports.GroupMemberFromJSON = exports.instanceOfGroupMember = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the GroupMember interface.
 */
function instanceOfGroupMember(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "uid" in value;
    return isInstance;
}
exports.instanceOfGroupMember = instanceOfGroupMember;
function GroupMemberFromJSON(json) {
    return GroupMemberFromJSONTyped(json, false);
}
exports.GroupMemberFromJSON = GroupMemberFromJSON;
function GroupMemberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !(0, runtime_1.exists)(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !(0, runtime_1.exists)(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
    };
}
exports.GroupMemberFromJSONTyped = GroupMemberFromJSONTyped;
function GroupMemberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'email': value.email,
        'attributes': value.attributes,
    };
}
exports.GroupMemberToJSON = GroupMemberToJSON;
