// @goauthentik/api/dist/models/IdentificationChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationChallengeResponseRequestToJSON = exports.IdentificationChallengeResponseRequestFromJSONTyped = exports.IdentificationChallengeResponseRequestFromJSON = exports.instanceOfIdentificationChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the IdentificationChallengeResponseRequest interface.
 */
function instanceOfIdentificationChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "uidField" in value;
    return isInstance;
}
exports.instanceOfIdentificationChallengeResponseRequest = instanceOfIdentificationChallengeResponseRequest;
function IdentificationChallengeResponseRequestFromJSON(json) {
    return IdentificationChallengeResponseRequestFromJSONTyped(json, false);
}
exports.IdentificationChallengeResponseRequestFromJSON = IdentificationChallengeResponseRequestFromJSON;
function IdentificationChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'uidField': json['uid_field'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
    };
}
exports.IdentificationChallengeResponseRequestFromJSONTyped = IdentificationChallengeResponseRequestFromJSONTyped;
function IdentificationChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'uid_field': value.uidField,
        'password': value.password,
    };
}
exports.IdentificationChallengeResponseRequestToJSON = IdentificationChallengeResponseRequestToJSON;
