// @goauthentik/api/dist/models/IdentificationStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationStageToJSON = exports.IdentificationStageFromJSONTyped = exports.IdentificationStageFromJSON = exports.instanceOfIdentificationStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
const UserFieldsEnum_1 = require("./UserFieldsEnum");
/**
 * Check if a given object implements the IdentificationStage interface.
 */
function instanceOfIdentificationStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfIdentificationStage = instanceOfIdentificationStage;
function IdentificationStageFromJSON(json) {
    return IdentificationStageFromJSONTyped(json, false);
}
exports.IdentificationStageFromJSON = IdentificationStageFromJSON;
function IdentificationStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'userFields': !(0, runtime_1.exists)(json, 'user_fields') ? undefined : (json['user_fields'].map(UserFieldsEnum_1.UserFieldsEnumFromJSON)),
        'passwordStage': !(0, runtime_1.exists)(json, 'password_stage') ? undefined : json['password_stage'],
        'caseInsensitiveMatching': !(0, runtime_1.exists)(json, 'case_insensitive_matching') ? undefined : json['case_insensitive_matching'],
        'showMatchedUser': !(0, runtime_1.exists)(json, 'show_matched_user') ? undefined : json['show_matched_user'],
        'enrollmentFlow': !(0, runtime_1.exists)(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'recoveryFlow': !(0, runtime_1.exists)(json, 'recovery_flow') ? undefined : json['recovery_flow'],
        'passwordlessFlow': !(0, runtime_1.exists)(json, 'passwordless_flow') ? undefined : json['passwordless_flow'],
        'sources': !(0, runtime_1.exists)(json, 'sources') ? undefined : json['sources'],
        'showSourceLabels': !(0, runtime_1.exists)(json, 'show_source_labels') ? undefined : json['show_source_labels'],
        'pretendUserExists': !(0, runtime_1.exists)(json, 'pretend_user_exists') ? undefined : json['pretend_user_exists'],
    };
}
exports.IdentificationStageFromJSONTyped = IdentificationStageFromJSONTyped;
function IdentificationStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'user_fields': value.userFields === undefined ? undefined : (value.userFields.map(UserFieldsEnum_1.UserFieldsEnumToJSON)),
        'password_stage': value.passwordStage,
        'case_insensitive_matching': value.caseInsensitiveMatching,
        'show_matched_user': value.showMatchedUser,
        'enrollment_flow': value.enrollmentFlow,
        'recovery_flow': value.recoveryFlow,
        'passwordless_flow': value.passwordlessFlow,
        'sources': value.sources,
        'show_source_labels': value.showSourceLabels,
        'pretend_user_exists': value.pretendUserExists,
    };
}
exports.IdentificationStageToJSON = IdentificationStageToJSON;
