// @goauthentik/api/dist/models/IdentificationChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationChallengeToJSON = exports.IdentificationChallengeFromJSONTyped = exports.IdentificationChallengeFromJSON = exports.instanceOfIdentificationChallenge = void 0;
const runtime_1 = require("../runtime");
const ChallengeChoices_1 = require("./ChallengeChoices");
const ContextualFlowInfo_1 = require("./ContextualFlowInfo");
const LoginSource_1 = require("./LoginSource");
/**
 * Check if a given object implements the IdentificationChallenge interface.
 */
function instanceOfIdentificationChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "userFields" in value;
    isInstance = isInstance && "passwordFields" in value;
    isInstance = isInstance && "primaryAction" in value;
    isInstance = isInstance && "showSourceLabels" in value;
    return isInstance;
}
exports.instanceOfIdentificationChallenge = instanceOfIdentificationChallenge;
function IdentificationChallengeFromJSON(json) {
    return IdentificationChallengeFromJSONTyped(json, false);
}
exports.IdentificationChallengeFromJSON = IdentificationChallengeFromJSON;
function IdentificationChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesFromJSON)(json['type']),
        'flowInfo': !(0, runtime_1.exists)(json, 'flow_info') ? undefined : (0, ContextualFlowInfo_1.ContextualFlowInfoFromJSON)(json['flow_info']),
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'responseErrors': !(0, runtime_1.exists)(json, 'response_errors') ? undefined : json['response_errors'],
        'userFields': json['user_fields'],
        'passwordFields': json['password_fields'],
        'applicationPre': !(0, runtime_1.exists)(json, 'application_pre') ? undefined : json['application_pre'],
        'enrollUrl': !(0, runtime_1.exists)(json, 'enroll_url') ? undefined : json['enroll_url'],
        'recoveryUrl': !(0, runtime_1.exists)(json, 'recovery_url') ? undefined : json['recovery_url'],
        'passwordlessUrl': !(0, runtime_1.exists)(json, 'passwordless_url') ? undefined : json['passwordless_url'],
        'primaryAction': json['primary_action'],
        'sources': !(0, runtime_1.exists)(json, 'sources') ? undefined : (json['sources'].map(LoginSource_1.LoginSourceFromJSON)),
        'showSourceLabels': json['show_source_labels'],
    };
}
exports.IdentificationChallengeFromJSONTyped = IdentificationChallengeFromJSONTyped;
function IdentificationChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesToJSON)(value.type),
        'flow_info': (0, ContextualFlowInfo_1.ContextualFlowInfoToJSON)(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'user_fields': value.userFields,
        'password_fields': value.passwordFields,
        'application_pre': value.applicationPre,
        'enroll_url': value.enrollUrl,
        'recovery_url': value.recoveryUrl,
        'passwordless_url': value.passwordlessUrl,
        'primary_action': value.primaryAction,
        'sources': value.sources === undefined ? undefined : (value.sources.map(LoginSource_1.LoginSourceToJSON)),
        'show_source_labels': value.showSourceLabels,
    };
}
exports.IdentificationChallengeToJSON = IdentificationChallengeToJSON;
