// @goauthentik/api/dist/models/InstallID.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstallIDToJSON = exports.InstallIDFromJSONTyped = exports.InstallIDFromJSON = exports.instanceOfInstallID = void 0;
/**
 * Check if a given object implements the InstallID interface.
 */
function instanceOfInstallID(value) {
    let isInstance = true;
    isInstance = isInstance && "installId" in value;
    return isInstance;
}
exports.instanceOfInstallID = instanceOfInstallID;
function InstallIDFromJSON(json) {
    return InstallIDFromJSONTyped(json, false);
}
exports.InstallIDFromJSON = InstallIDFromJSON;
function InstallIDFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'installId': json['install_id'],
    };
}
exports.InstallIDFromJSONTyped = InstallIDFromJSONTyped;
function InstallIDToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'install_id': value.installId,
    };
}
exports.InstallIDToJSON = InstallIDToJSON;
