// @goauthentik/api/dist/models/IntentEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntentEnumToJSON = exports.IntentEnumFromJSONTyped = exports.IntentEnumFromJSON = exports.IntentEnum = void 0;
/**
 * * `verification` - Intent Verification
 * * `api` - Intent Api
 * * `recovery` - Intent Recovery
 * * `app_password` - Intent App Password
 * @export
 */
exports.IntentEnum = {
    Verification: 'verification',
    Api: 'api',
    Recovery: 'recovery',
    AppPassword: 'app_password',
    UnknownDefaultOpenApi: '11184809'
};
function IntentEnumFromJSON(json) {
    return IntentEnumFromJSONTyped(json, false);
}
exports.IntentEnumFromJSON = IntentEnumFromJSON;
function IntentEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.IntentEnumFromJSONTyped = IntentEnumFromJSONTyped;
function IntentEnumToJSON(value) {
    return value;
}
exports.IntentEnumToJSON = IntentEnumToJSON;
