// @goauthentik/api/dist/models/InvalidResponseActionEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvalidResponseActionEnumToJSON = exports.InvalidResponseActionEnumFromJSONTyped = exports.InvalidResponseActionEnumFromJSON = exports.InvalidResponseActionEnum = void 0;
/**
 * * `retry` - Retry
 * * `restart` - Restart
 * * `restart_with_context` - Restart With Context
 * @export
 */
exports.InvalidResponseActionEnum = {
    Retry: 'retry',
    Restart: 'restart',
    RestartWithContext: 'restart_with_context',
    UnknownDefaultOpenApi: '11184809'
};
function InvalidResponseActionEnumFromJSON(json) {
    return InvalidResponseActionEnumFromJSONTyped(json, false);
}
exports.InvalidResponseActionEnumFromJSON = InvalidResponseActionEnumFromJSON;
function InvalidResponseActionEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.InvalidResponseActionEnumFromJSONTyped = InvalidResponseActionEnumFromJSONTyped;
function InvalidResponseActionEnumToJSON(value) {
    return value;
}
exports.InvalidResponseActionEnumToJSON = InvalidResponseActionEnumToJSON;
