// @goauthentik/api/dist/models/InvitationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvitationRequestToJSON = exports.InvitationRequestFromJSONTyped = exports.InvitationRequestFromJSON = exports.instanceOfInvitationRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the InvitationRequest interface.
 */
function instanceOfInvitationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfInvitationRequest = instanceOfInvitationRequest;
function InvitationRequestFromJSON(json) {
    return InvitationRequestFromJSONTyped(json, false);
}
exports.InvitationRequestFromJSON = InvitationRequestFromJSON;
function InvitationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !(0, runtime_1.exists)(json, 'fixed_data') ? undefined : json['fixed_data'],
        'singleUse': !(0, runtime_1.exists)(json, 'single_use') ? undefined : json['single_use'],
        'flow': !(0, runtime_1.exists)(json, 'flow') ? undefined : json['flow'],
    };
}
exports.InvitationRequestFromJSONTyped = InvitationRequestFromJSONTyped;
function InvitationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
exports.InvitationRequestToJSON = InvitationRequestToJSON;
