// @goauthentik/api/dist/models/InvitationStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvitationStageToJSON = exports.InvitationStageFromJSONTyped = exports.InvitationStageFromJSON = exports.instanceOfInvitationStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the InvitationStage interface.
 */
function instanceOfInvitationStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfInvitationStage = instanceOfInvitationStage;
function InvitationStageFromJSON(json) {
    return InvitationStageFromJSONTyped(json, false);
}
exports.InvitationStageFromJSON = InvitationStageFromJSON;
function InvitationStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'continueFlowWithoutInvitation': !(0, runtime_1.exists)(json, 'continue_flow_without_invitation') ? undefined : json['continue_flow_without_invitation'],
    };
}
exports.InvitationStageFromJSONTyped = InvitationStageFromJSONTyped;
function InvitationStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'continue_flow_without_invitation': value.continueFlowWithoutInvitation,
    };
}
exports.InvitationStageToJSON = InvitationStageToJSON;
