// @goauthentik/api/dist/models/Invitation.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvitationToJSON = exports.InvitationFromJSONTyped = exports.InvitationFromJSON = exports.instanceOfInvitation = void 0;
const runtime_1 = require("../runtime");
const Flow_1 = require("./Flow");
const GroupMember_1 = require("./GroupMember");
/**
 * Check if a given object implements the Invitation interface.
 */
function instanceOfInvitation(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "flowObj" in value;
    return isInstance;
}
exports.instanceOfInvitation = instanceOfInvitation;
function InvitationFromJSON(json) {
    return InvitationFromJSONTyped(json, false);
}
exports.InvitationFromJSON = InvitationFromJSON;
function InvitationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !(0, runtime_1.exists)(json, 'fixed_data') ? undefined : json['fixed_data'],
        'createdBy': (0, GroupMember_1.GroupMemberFromJSON)(json['created_by']),
        'singleUse': !(0, runtime_1.exists)(json, 'single_use') ? undefined : json['single_use'],
        'flow': !(0, runtime_1.exists)(json, 'flow') ? undefined : json['flow'],
        'flowObj': (0, Flow_1.FlowFromJSON)(json['flow_obj']),
    };
}
exports.InvitationFromJSONTyped = InvitationFromJSONTyped;
function InvitationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
exports.InvitationToJSON = InvitationToJSON;
