// @goauthentik/api/dist/models/IssuerModeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssuerModeEnumToJSON = exports.IssuerModeEnumFromJSONTyped = exports.IssuerModeEnumFromJSON = exports.IssuerModeEnum = void 0;
/**
 * * `global` - Same identifier is used for all providers
 * * `per_provider` - Each provider has a different issuer, based on the application slug.
 * @export
 */
exports.IssuerModeEnum = {
    Global: 'global',
    PerProvider: 'per_provider',
    UnknownDefaultOpenApi: '11184809'
};
function IssuerModeEnumFromJSON(json) {
    return IssuerModeEnumFromJSONTyped(json, false);
}
exports.IssuerModeEnumFromJSON = IssuerModeEnumFromJSON;
function IssuerModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.IssuerModeEnumFromJSONTyped = IssuerModeEnumFromJSONTyped;
function IssuerModeEnumToJSON(value) {
    return value;
}
exports.IssuerModeEnumToJSON = IssuerModeEnumToJSON;
