// @goauthentik/api/dist/models/KubernetesServiceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesServiceConnectionRequestToJSON = exports.KubernetesServiceConnectionRequestFromJSONTyped = exports.KubernetesServiceConnectionRequestFromJSON = exports.instanceOfKubernetesServiceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the KubernetesServiceConnectionRequest interface.
 */
function instanceOfKubernetesServiceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfKubernetesServiceConnectionRequest = instanceOfKubernetesServiceConnectionRequest;
function KubernetesServiceConnectionRequestFromJSON(json) {
    return KubernetesServiceConnectionRequestFromJSONTyped(json, false);
}
exports.KubernetesServiceConnectionRequestFromJSON = KubernetesServiceConnectionRequestFromJSON;
function KubernetesServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'kubeconfig': !(0, runtime_1.exists)(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !(0, runtime_1.exists)(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}
exports.KubernetesServiceConnectionRequestFromJSONTyped = KubernetesServiceConnectionRequestFromJSONTyped;
function KubernetesServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
exports.KubernetesServiceConnectionRequestToJSON = KubernetesServiceConnectionRequestToJSON;
