// @goauthentik/api/dist/models/KubernetesServiceConnection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesServiceConnectionToJSON = exports.KubernetesServiceConnectionFromJSONTyped = exports.KubernetesServiceConnectionFromJSON = exports.instanceOfKubernetesServiceConnection = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the KubernetesServiceConnection interface.
 */
function instanceOfKubernetesServiceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfKubernetesServiceConnection = instanceOfKubernetesServiceConnection;
function KubernetesServiceConnectionFromJSON(json) {
    return KubernetesServiceConnectionFromJSONTyped(json, false);
}
exports.KubernetesServiceConnectionFromJSON = KubernetesServiceConnectionFromJSON;
function KubernetesServiceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'kubeconfig': !(0, runtime_1.exists)(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !(0, runtime_1.exists)(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}
exports.KubernetesServiceConnectionFromJSONTyped = KubernetesServiceConnectionFromJSONTyped;
function KubernetesServiceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
exports.KubernetesServiceConnectionToJSON = KubernetesServiceConnectionToJSON;
