// @goauthentik/api/dist/models/LDAPSourceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPSourceRequestToJSON = exports.LDAPSourceRequestFromJSONTyped = exports.LDAPSourceRequestFromJSON = exports.instanceOfLDAPSourceRequest = void 0;
const runtime_1 = require("../runtime");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
const UserMatchingModeEnum_1 = require("./UserMatchingModeEnum");
/**
 * Check if a given object implements the LDAPSourceRequest interface.
 */
function instanceOfLDAPSourceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "serverUri" in value;
    isInstance = isInstance && "baseDn" in value;
    return isInstance;
}
exports.instanceOfLDAPSourceRequest = instanceOfLDAPSourceRequest;
function LDAPSourceRequestFromJSON(json) {
    return LDAPSourceRequestFromJSONTyped(json, false);
}
exports.LDAPSourceRequestFromJSON = LDAPSourceRequestFromJSON;
function LDAPSourceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'slug': json['slug'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'enrollmentFlow': !(0, runtime_1.exists)(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'userMatchingMode': !(0, runtime_1.exists)(json, 'user_matching_mode') ? undefined : (0, UserMatchingModeEnum_1.UserMatchingModeEnumFromJSON)(json['user_matching_mode']),
        'userPathTemplate': !(0, runtime_1.exists)(json, 'user_path_template') ? undefined : json['user_path_template'],
        'serverUri': json['server_uri'],
        'peerCertificate': !(0, runtime_1.exists)(json, 'peer_certificate') ? undefined : json['peer_certificate'],
        'clientCertificate': !(0, runtime_1.exists)(json, 'client_certificate') ? undefined : json['client_certificate'],
        'bindCn': !(0, runtime_1.exists)(json, 'bind_cn') ? undefined : json['bind_cn'],
        'bindPassword': !(0, runtime_1.exists)(json, 'bind_password') ? undefined : json['bind_password'],
        'startTls': !(0, runtime_1.exists)(json, 'start_tls') ? undefined : json['start_tls'],
        'sni': !(0, runtime_1.exists)(json, 'sni') ? undefined : json['sni'],
        'baseDn': json['base_dn'],
        'additionalUserDn': !(0, runtime_1.exists)(json, 'additional_user_dn') ? undefined : json['additional_user_dn'],
        'additionalGroupDn': !(0, runtime_1.exists)(json, 'additional_group_dn') ? undefined : json['additional_group_dn'],
        'userObjectFilter': !(0, runtime_1.exists)(json, 'user_object_filter') ? undefined : json['user_object_filter'],
        'groupObjectFilter': !(0, runtime_1.exists)(json, 'group_object_filter') ? undefined : json['group_object_filter'],
        'groupMembershipField': !(0, runtime_1.exists)(json, 'group_membership_field') ? undefined : json['group_membership_field'],
        'objectUniquenessField': !(0, runtime_1.exists)(json, 'object_uniqueness_field') ? undefined : json['object_uniqueness_field'],
        'syncUsers': !(0, runtime_1.exists)(json, 'sync_users') ? undefined : json['sync_users'],
        'syncUsersPassword': !(0, runtime_1.exists)(json, 'sync_users_password') ? undefined : json['sync_users_password'],
        'syncGroups': !(0, runtime_1.exists)(json, 'sync_groups') ? undefined : json['sync_groups'],
        'syncParentGroup': !(0, runtime_1.exists)(json, 'sync_parent_group') ? undefined : json['sync_parent_group'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !(0, runtime_1.exists)(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
    };
}
exports.LDAPSourceRequestFromJSONTyped = LDAPSourceRequestFromJSONTyped;
function LDAPSourceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'enabled': value.enabled,
        'authentication_flow': value.authenticationFlow,
        'enrollment_flow': value.enrollmentFlow,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'user_matching_mode': (0, UserMatchingModeEnum_1.UserMatchingModeEnumToJSON)(value.userMatchingMode),
        'user_path_template': value.userPathTemplate,
        'server_uri': value.serverUri,
        'peer_certificate': value.peerCertificate,
        'client_certificate': value.clientCertificate,
        'bind_cn': value.bindCn,
        'bind_password': value.bindPassword,
        'start_tls': value.startTls,
        'sni': value.sni,
        'base_dn': value.baseDn,
        'additional_user_dn': value.additionalUserDn,
        'additional_group_dn': value.additionalGroupDn,
        'user_object_filter': value.userObjectFilter,
        'group_object_filter': value.groupObjectFilter,
        'group_membership_field': value.groupMembershipField,
        'object_uniqueness_field': value.objectUniquenessField,
        'sync_users': value.syncUsers,
        'sync_users_password': value.syncUsersPassword,
        'sync_groups': value.syncGroups,
        'sync_parent_group': value.syncParentGroup,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
    };
}
exports.LDAPSourceRequestToJSON = LDAPSourceRequestToJSON;
