// @goauthentik/api/dist/models/LDAPDebug.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPDebugToJSON = exports.LDAPDebugFromJSONTyped = exports.LDAPDebugFromJSON = exports.instanceOfLDAPDebug = void 0;
/**
 * Check if a given object implements the LDAPDebug interface.
 */
function instanceOfLDAPDebug(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "group" in value;
    isInstance = isInstance && "membership" in value;
    return isInstance;
}
exports.instanceOfLDAPDebug = instanceOfLDAPDebug;
function LDAPDebugFromJSON(json) {
    return LDAPDebugFromJSONTyped(json, false);
}
exports.LDAPDebugFromJSON = LDAPDebugFromJSON;
function LDAPDebugFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'group': json['group'],
        'membership': json['membership'],
    };
}
exports.LDAPDebugFromJSONTyped = LDAPDebugFromJSONTyped;
function LDAPDebugToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.LDAPDebugToJSON = LDAPDebugToJSON;
