// @goauthentik/api/dist/models/LDAPPropertyMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPPropertyMappingRequestToJSON = exports.LDAPPropertyMappingRequestFromJSONTyped = exports.LDAPPropertyMappingRequestFromJSON = exports.instanceOfLDAPPropertyMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the LDAPPropertyMappingRequest interface.
 */
function instanceOfLDAPPropertyMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "objectField" in value;
    return isInstance;
}
exports.instanceOfLDAPPropertyMappingRequest = instanceOfLDAPPropertyMappingRequest;
function LDAPPropertyMappingRequestFromJSON(json) {
    return LDAPPropertyMappingRequestFromJSONTyped(json, false);
}
exports.LDAPPropertyMappingRequestFromJSON = LDAPPropertyMappingRequestFromJSON;
function LDAPPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'objectField': json['object_field'],
    };
}
exports.LDAPPropertyMappingRequestFromJSONTyped = LDAPPropertyMappingRequestFromJSONTyped;
function LDAPPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
exports.LDAPPropertyMappingRequestToJSON = LDAPPropertyMappingRequestToJSON;
