// @goauthentik/api/dist/models/LDAPAPIAccessMode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPAPIAccessModeToJSON = exports.LDAPAPIAccessModeFromJSONTyped = exports.LDAPAPIAccessModeFromJSON = exports.LDAPAPIAccessMode = void 0;
/**
 * * `direct` - Direct
 * * `cached` - Cached
 * @export
 */
exports.LDAPAPIAccessMode = {
    Direct: 'direct',
    Cached: 'cached',
    UnknownDefaultOpenApi: '11184809'
};
function LDAPAPIAccessModeFromJSON(json) {
    return LDAPAPIAccessModeFromJSONTyped(json, false);
}
exports.LDAPAPIAccessModeFromJSON = LDAPAPIAccessModeFromJSON;
function LDAPAPIAccessModeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.LDAPAPIAccessModeFromJSONTyped = LDAPAPIAccessModeFromJSONTyped;
function LDAPAPIAccessModeToJSON(value) {
    return value;
}
exports.LDAPAPIAccessModeToJSON = LDAPAPIAccessModeToJSON;
