// @goauthentik/api/dist/models/LDAPOutpostConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPOutpostConfigToJSON = exports.LDAPOutpostConfigFromJSONTyped = exports.LDAPOutpostConfigFromJSON = exports.instanceOfLDAPOutpostConfig = void 0;
const runtime_1 = require("../runtime");
const LDAPAPIAccessMode_1 = require("./LDAPAPIAccessMode");
/**
 * Check if a given object implements the LDAPOutpostConfig interface.
 */
function instanceOfLDAPOutpostConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "bindFlowSlug" in value;
    isInstance = isInstance && "applicationSlug" in value;
    return isInstance;
}
exports.instanceOfLDAPOutpostConfig = instanceOfLDAPOutpostConfig;
function LDAPOutpostConfigFromJSON(json) {
    return LDAPOutpostConfigFromJSONTyped(json, false);
}
exports.LDAPOutpostConfigFromJSON = LDAPOutpostConfigFromJSON;
function LDAPOutpostConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'baseDn': !(0, runtime_1.exists)(json, 'base_dn') ? undefined : json['base_dn'],
        'bindFlowSlug': json['bind_flow_slug'],
        'applicationSlug': json['application_slug'],
        'searchGroup': !(0, runtime_1.exists)(json, 'search_group') ? undefined : json['search_group'],
        'certificate': !(0, runtime_1.exists)(json, 'certificate') ? undefined : json['certificate'],
        'tlsServerName': !(0, runtime_1.exists)(json, 'tls_server_name') ? undefined : json['tls_server_name'],
        'uidStartNumber': !(0, runtime_1.exists)(json, 'uid_start_number') ? undefined : json['uid_start_number'],
        'gidStartNumber': !(0, runtime_1.exists)(json, 'gid_start_number') ? undefined : json['gid_start_number'],
        'searchMode': !(0, runtime_1.exists)(json, 'search_mode') ? undefined : (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeFromJSON)(json['search_mode']),
        'bindMode': !(0, runtime_1.exists)(json, 'bind_mode') ? undefined : (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeFromJSON)(json['bind_mode']),
        'mfaSupport': !(0, runtime_1.exists)(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}
exports.LDAPOutpostConfigFromJSONTyped = LDAPOutpostConfigFromJSONTyped;
function LDAPOutpostConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'base_dn': value.baseDn,
        'bind_flow_slug': value.bindFlowSlug,
        'search_group': value.searchGroup,
        'certificate': value.certificate,
        'tls_server_name': value.tlsServerName,
        'uid_start_number': value.uidStartNumber,
        'gid_start_number': value.gidStartNumber,
        'search_mode': (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeToJSON)(value.searchMode),
        'bind_mode': (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeToJSON)(value.bindMode),
        'mfa_support': value.mfaSupport,
    };
}
exports.LDAPOutpostConfigToJSON = LDAPOutpostConfigToJSON;
