// @goauthentik/api/dist/models/LDAPPropertyMapping.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LDAPPropertyMappingToJSON = exports.LDAPPropertyMappingFromJSONTyped = exports.LDAPPropertyMappingFromJSON = exports.instanceOfLDAPPropertyMapping = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the LDAPPropertyMapping interface.
 */
function instanceOfLDAPPropertyMapping(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "objectField" in value;
    return isInstance;
}
exports.instanceOfLDAPPropertyMapping = instanceOfLDAPPropertyMapping;
function LDAPPropertyMappingFromJSON(json) {
    return LDAPPropertyMappingFromJSONTyped(json, false);
}
exports.LDAPPropertyMappingFromJSON = LDAPPropertyMappingFromJSON;
function LDAPPropertyMappingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'objectField': json['object_field'],
    };
}
exports.LDAPPropertyMappingFromJSONTyped = LDAPPropertyMappingFromJSONTyped;
function LDAPPropertyMappingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
exports.LDAPPropertyMappingToJSON = LDAPPropertyMappingToJSON;
