// @goauthentik/api/dist/models/LicenseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseRequestToJSON = exports.LicenseRequestFromJSONTyped = exports.LicenseRequestFromJSON = exports.instanceOfLicenseRequest = void 0;
/**
 * Check if a given object implements the LicenseRequest interface.
 */
function instanceOfLicenseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
exports.instanceOfLicenseRequest = instanceOfLicenseRequest;
function LicenseRequestFromJSON(json) {
    return LicenseRequestFromJSONTyped(json, false);
}
exports.LicenseRequestFromJSON = LicenseRequestFromJSON;
function LicenseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
exports.LicenseRequestFromJSONTyped = LicenseRequestFromJSONTyped;
function LicenseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
    };
}
exports.LicenseRequestToJSON = LicenseRequestToJSON;
