// @goauthentik/api/dist/models/LoginSource.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginSourceToJSON = exports.LoginSourceFromJSONTyped = exports.LoginSourceFromJSON = exports.instanceOfLoginSource = void 0;
const runtime_1 = require("../runtime");
const LoginChallengeTypes_1 = require("./LoginChallengeTypes");
/**
 * Check if a given object implements the LoginSource interface.
 */
function instanceOfLoginSource(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "challenge" in value;
    return isInstance;
}
exports.instanceOfLoginSource = instanceOfLoginSource;
function LoginSourceFromJSON(json) {
    return LoginSourceFromJSONTyped(json, false);
}
exports.LoginSourceFromJSON = LoginSourceFromJSON;
function LoginSourceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'iconUrl': !(0, runtime_1.exists)(json, 'icon_url') ? undefined : json['icon_url'],
        'challenge': (0, LoginChallengeTypes_1.LoginChallengeTypesFromJSON)(json['challenge']),
    };
}
exports.LoginSourceFromJSONTyped = LoginSourceFromJSONTyped;
function LoginSourceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'icon_url': value.iconUrl,
        'challenge': (0, LoginChallengeTypes_1.LoginChallengeTypesToJSON)(value.challenge),
    };
}
exports.LoginSourceToJSON = LoginSourceToJSON;
