// @goauthentik/api/dist/models/LoginMetrics.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginMetricsToJSON = exports.LoginMetricsFromJSONTyped = exports.LoginMetricsFromJSON = exports.instanceOfLoginMetrics = void 0;
const Coordinate_1 = require("./Coordinate");
/**
 * Check if a given object implements the LoginMetrics interface.
 */
function instanceOfLoginMetrics(value) {
    let isInstance = true;
    isInstance = isInstance && "logins" in value;
    isInstance = isInstance && "loginsFailed" in value;
    isInstance = isInstance && "authorizations" in value;
    return isInstance;
}
exports.instanceOfLoginMetrics = instanceOfLoginMetrics;
function LoginMetricsFromJSON(json) {
    return LoginMetricsFromJSONTyped(json, false);
}
exports.LoginMetricsFromJSON = LoginMetricsFromJSON;
function LoginMetricsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'logins': (json['logins'].map(Coordinate_1.CoordinateFromJSON)),
        'loginsFailed': (json['logins_failed'].map(Coordinate_1.CoordinateFromJSON)),
        'authorizations': (json['authorizations'].map(Coordinate_1.CoordinateFromJSON)),
    };
}
exports.LoginMetricsFromJSONTyped = LoginMetricsFromJSONTyped;
function LoginMetricsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.LoginMetricsToJSON = LoginMetricsToJSON;
