// @goauthentik/api/dist/models/Metadata.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataToJSON = exports.MetadataFromJSONTyped = exports.MetadataFromJSON = exports.instanceOfMetadata = void 0;
/**
 * Check if a given object implements the Metadata interface.
 */
function instanceOfMetadata(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "labels" in value;
    return isInstance;
}
exports.instanceOfMetadata = instanceOfMetadata;
function MetadataFromJSON(json) {
    return MetadataFromJSONTyped(json, false);
}
exports.MetadataFromJSON = MetadataFromJSON;
function MetadataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'labels': json['labels'],
    };
}
exports.MetadataFromJSONTyped = MetadataFromJSONTyped;
function MetadataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'labels': value.labels,
    };
}
exports.MetadataToJSON = MetadataToJSON;
