// @goauthentik/api/dist/models/NetworkBindingEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkBindingEnumToJSON = exports.NetworkBindingEnumFromJSONTyped = exports.NetworkBindingEnumFromJSON = exports.NetworkBindingEnum = void 0;
/**
 * * `no_binding` - No Binding
 * * `bind_asn` - Bind Asn
 * * `bind_asn_network` - Bind Asn Network
 * * `bind_asn_network_ip` - Bind Asn Network Ip
 * @export
 */
exports.NetworkBindingEnum = {
    NoBinding: 'no_binding',
    BindAsn: 'bind_asn',
    BindAsnNetwork: 'bind_asn_network',
    BindAsnNetworkIp: 'bind_asn_network_ip',
    UnknownDefaultOpenApi: '11184809'
};
function NetworkBindingEnumFromJSON(json) {
    return NetworkBindingEnumFromJSONTyped(json, false);
}
exports.NetworkBindingEnumFromJSON = NetworkBindingEnumFromJSON;
function NetworkBindingEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.NetworkBindingEnumFromJSONTyped = NetworkBindingEnumFromJSONTyped;
function NetworkBindingEnumToJSON(value) {
    return value;
}
exports.NetworkBindingEnumToJSON = NetworkBindingEnumToJSON;
