// @goauthentik/api/dist/models/NotConfiguredActionEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotConfiguredActionEnumToJSON = exports.NotConfiguredActionEnumFromJSONTyped = exports.NotConfiguredActionEnumFromJSON = exports.NotConfiguredActionEnum = void 0;
/**
 * * `skip` - Skip
 * * `deny` - Deny
 * * `configure` - Configure
 * @export
 */
exports.NotConfiguredActionEnum = {
    Skip: 'skip',
    Deny: 'deny',
    Configure: 'configure',
    UnknownDefaultOpenApi: '11184809'
};
function NotConfiguredActionEnumFromJSON(json) {
    return NotConfiguredActionEnumFromJSONTyped(json, false);
}
exports.NotConfiguredActionEnumFromJSON = NotConfiguredActionEnumFromJSON;
function NotConfiguredActionEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.NotConfiguredActionEnumFromJSONTyped = NotConfiguredActionEnumFromJSONTyped;
function NotConfiguredActionEnumToJSON(value) {
    return value;
}
exports.NotConfiguredActionEnumToJSON = NotConfiguredActionEnumToJSON;
