// @goauthentik/api/dist/models/NotificationRuleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationRuleRequestToJSON = exports.NotificationRuleRequestFromJSONTyped = exports.NotificationRuleRequestFromJSON = exports.instanceOfNotificationRuleRequest = void 0;
const runtime_1 = require("../runtime");
const SeverityEnum_1 = require("./SeverityEnum");
/**
 * Check if a given object implements the NotificationRuleRequest interface.
 */
function instanceOfNotificationRuleRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfNotificationRuleRequest = instanceOfNotificationRuleRequest;
function NotificationRuleRequestFromJSON(json) {
    return NotificationRuleRequestFromJSONTyped(json, false);
}
exports.NotificationRuleRequestFromJSON = NotificationRuleRequestFromJSON;
function NotificationRuleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'transports': !(0, runtime_1.exists)(json, 'transports') ? undefined : json['transports'],
        'severity': !(0, runtime_1.exists)(json, 'severity') ? undefined : (0, SeverityEnum_1.SeverityEnumFromJSON)(json['severity']),
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
    };
}
exports.NotificationRuleRequestFromJSONTyped = NotificationRuleRequestFromJSONTyped;
function NotificationRuleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'transports': value.transports,
        'severity': (0, SeverityEnum_1.SeverityEnumToJSON)(value.severity),
        'group': value.group,
    };
}
exports.NotificationRuleRequestToJSON = NotificationRuleRequestToJSON;
