// @goauthentik/api/dist/models/NotificationWebhookMapping.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationWebhookMappingToJSON = exports.NotificationWebhookMappingFromJSONTyped = exports.NotificationWebhookMappingFromJSON = exports.instanceOfNotificationWebhookMapping = void 0;
/**
 * Check if a given object implements the NotificationWebhookMapping interface.
 */
function instanceOfNotificationWebhookMapping(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    return isInstance;
}
exports.instanceOfNotificationWebhookMapping = instanceOfNotificationWebhookMapping;
function NotificationWebhookMappingFromJSON(json) {
    return NotificationWebhookMappingFromJSONTyped(json, false);
}
exports.NotificationWebhookMappingFromJSON = NotificationWebhookMappingFromJSON;
function NotificationWebhookMappingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'expression': json['expression'],
    };
}
exports.NotificationWebhookMappingFromJSONTyped = NotificationWebhookMappingFromJSONTyped;
function NotificationWebhookMappingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expression': value.expression,
    };
}
exports.NotificationWebhookMappingToJSON = NotificationWebhookMappingToJSON;
