// @goauthentik/api/dist/models/Notification.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationToJSON = exports.NotificationFromJSONTyped = exports.NotificationFromJSON = exports.instanceOfNotification = void 0;
const runtime_1 = require("../runtime");
const Event_1 = require("./Event");
const SeverityEnum_1 = require("./SeverityEnum");
/**
 * Check if a given object implements the Notification interface.
 */
function instanceOfNotification(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "severity" in value;
    isInstance = isInstance && "body" in value;
    isInstance = isInstance && "created" in value;
    return isInstance;
}
exports.instanceOfNotification = instanceOfNotification;
function NotificationFromJSON(json) {
    return NotificationFromJSONTyped(json, false);
}
exports.NotificationFromJSON = NotificationFromJSON;
function NotificationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'severity': (0, SeverityEnum_1.SeverityEnumFromJSON)(json['severity']),
        'body': json['body'],
        'created': (new Date(json['created'])),
        'event': !(0, runtime_1.exists)(json, 'event') ? undefined : (0, Event_1.EventFromJSON)(json['event']),
        'seen': !(0, runtime_1.exists)(json, 'seen') ? undefined : json['seen'],
    };
}
exports.NotificationFromJSONTyped = NotificationFromJSONTyped;
function NotificationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'event': (0, Event_1.EventToJSON)(value.event),
        'seen': value.seen,
    };
}
exports.NotificationToJSON = NotificationToJSON;
