// @goauthentik/api/dist/models/NotificationRule.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationRuleToJSON = exports.NotificationRuleFromJSONTyped = exports.NotificationRuleFromJSON = exports.instanceOfNotificationRule = void 0;
const runtime_1 = require("../runtime");
const Group_1 = require("./Group");
const SeverityEnum_1 = require("./SeverityEnum");
/**
 * Check if a given object implements the NotificationRule interface.
 */
function instanceOfNotificationRule(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "groupObj" in value;
    return isInstance;
}
exports.instanceOfNotificationRule = instanceOfNotificationRule;
function NotificationRuleFromJSON(json) {
    return NotificationRuleFromJSONTyped(json, false);
}
exports.NotificationRuleFromJSON = NotificationRuleFromJSON;
function NotificationRuleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'transports': !(0, runtime_1.exists)(json, 'transports') ? undefined : json['transports'],
        'severity': !(0, runtime_1.exists)(json, 'severity') ? undefined : (0, SeverityEnum_1.SeverityEnumFromJSON)(json['severity']),
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
        'groupObj': (0, Group_1.GroupFromJSON)(json['group_obj']),
    };
}
exports.NotificationRuleFromJSONTyped = NotificationRuleFromJSONTyped;
function NotificationRuleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'transports': value.transports,
        'severity': (0, SeverityEnum_1.SeverityEnumToJSON)(value.severity),
        'group': value.group,
    };
}
exports.NotificationRuleToJSON = NotificationRuleToJSON;
