// @goauthentik/api/dist/models/NotificationTransportModeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationTransportModeEnumToJSON = exports.NotificationTransportModeEnumFromJSONTyped = exports.NotificationTransportModeEnumFromJSON = exports.NotificationTransportModeEnum = void 0;
/**
 * * `local` - authentik inbuilt notifications
 * * `webhook` - Generic Webhook
 * * `webhook_slack` - Slack Webhook (Slack/Discord)
 * * `email` - Email
 * @export
 */
exports.NotificationTransportModeEnum = {
    Local: 'local',
    Webhook: 'webhook',
    WebhookSlack: 'webhook_slack',
    Email: 'email',
    UnknownDefaultOpenApi: '11184809'
};
function NotificationTransportModeEnumFromJSON(json) {
    return NotificationTransportModeEnumFromJSONTyped(json, false);
}
exports.NotificationTransportModeEnumFromJSON = NotificationTransportModeEnumFromJSON;
function NotificationTransportModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.NotificationTransportModeEnumFromJSONTyped = NotificationTransportModeEnumFromJSONTyped;
function NotificationTransportModeEnumToJSON(value) {
    return value;
}
exports.NotificationTransportModeEnumToJSON = NotificationTransportModeEnumToJSON;
