// @goauthentik/api/dist/models/NotificationWebhookMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationWebhookMappingRequestToJSON = exports.NotificationWebhookMappingRequestFromJSONTyped = exports.NotificationWebhookMappingRequestFromJSON = exports.instanceOfNotificationWebhookMappingRequest = void 0;
/**
 * Check if a given object implements the NotificationWebhookMappingRequest interface.
 */
function instanceOfNotificationWebhookMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    return isInstance;
}
exports.instanceOfNotificationWebhookMappingRequest = instanceOfNotificationWebhookMappingRequest;
function NotificationWebhookMappingRequestFromJSON(json) {
    return NotificationWebhookMappingRequestFromJSONTyped(json, false);
}
exports.NotificationWebhookMappingRequestFromJSON = NotificationWebhookMappingRequestFromJSON;
function NotificationWebhookMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'expression': json['expression'],
    };
}
exports.NotificationWebhookMappingRequestFromJSONTyped = NotificationWebhookMappingRequestFromJSONTyped;
function NotificationWebhookMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expression': value.expression,
    };
}
exports.NotificationWebhookMappingRequestToJSON = NotificationWebhookMappingRequestToJSON;
