// @goauthentik/api/dist/models/NotificationTransportTest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationTransportTestToJSON = exports.NotificationTransportTestFromJSONTyped = exports.NotificationTransportTestFromJSON = exports.instanceOfNotificationTransportTest = void 0;
/**
 * Check if a given object implements the NotificationTransportTest interface.
 */
function instanceOfNotificationTransportTest(value) {
    let isInstance = true;
    isInstance = isInstance && "messages" in value;
    return isInstance;
}
exports.instanceOfNotificationTransportTest = instanceOfNotificationTransportTest;
function NotificationTransportTestFromJSON(json) {
    return NotificationTransportTestFromJSONTyped(json, false);
}
exports.NotificationTransportTestFromJSON = NotificationTransportTestFromJSON;
function NotificationTransportTestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'messages': json['messages'],
    };
}
exports.NotificationTransportTestFromJSONTyped = NotificationTransportTestFromJSONTyped;
function NotificationTransportTestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'messages': value.messages,
    };
}
exports.NotificationTransportTestToJSON = NotificationTransportTestToJSON;
