// @goauthentik/api/dist/models/NotificationTransport.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationTransportToJSON = exports.NotificationTransportFromJSONTyped = exports.NotificationTransportFromJSON = exports.instanceOfNotificationTransport = void 0;
const runtime_1 = require("../runtime");
const NotificationTransportModeEnum_1 = require("./NotificationTransportModeEnum");
/**
 * Check if a given object implements the NotificationTransport interface.
 */
function instanceOfNotificationTransport(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "modeVerbose" in value;
    return isInstance;
}
exports.instanceOfNotificationTransport = instanceOfNotificationTransport;
function NotificationTransportFromJSON(json) {
    return NotificationTransportFromJSONTyped(json, false);
}
exports.NotificationTransportFromJSON = NotificationTransportFromJSON;
function NotificationTransportFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, NotificationTransportModeEnum_1.NotificationTransportModeEnumFromJSON)(json['mode']),
        'modeVerbose': json['mode_verbose'],
        'webhookUrl': !(0, runtime_1.exists)(json, 'webhook_url') ? undefined : json['webhook_url'],
        'webhookMapping': !(0, runtime_1.exists)(json, 'webhook_mapping') ? undefined : json['webhook_mapping'],
        'sendOnce': !(0, runtime_1.exists)(json, 'send_once') ? undefined : json['send_once'],
    };
}
exports.NotificationTransportFromJSONTyped = NotificationTransportFromJSONTyped;
function NotificationTransportToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'mode': (0, NotificationTransportModeEnum_1.NotificationTransportModeEnumToJSON)(value.mode),
        'webhook_url': value.webhookUrl,
        'webhook_mapping': value.webhookMapping,
        'send_once': value.sendOnce,
    };
}
exports.NotificationTransportToJSON = NotificationTransportToJSON;
