// @goauthentik/api/dist/models/NotificationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationRequestToJSON = exports.NotificationRequestFromJSONTyped = exports.NotificationRequestFromJSON = exports.instanceOfNotificationRequest = void 0;
const runtime_1 = require("../runtime");
const EventRequest_1 = require("./EventRequest");
/**
 * Check if a given object implements the NotificationRequest interface.
 */
function instanceOfNotificationRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfNotificationRequest = instanceOfNotificationRequest;
function NotificationRequestFromJSON(json) {
    return NotificationRequestFromJSONTyped(json, false);
}
exports.NotificationRequestFromJSON = NotificationRequestFromJSON;
function NotificationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'event': !(0, runtime_1.exists)(json, 'event') ? undefined : (0, EventRequest_1.EventRequestFromJSON)(json['event']),
        'seen': !(0, runtime_1.exists)(json, 'seen') ? undefined : json['seen'],
    };
}
exports.NotificationRequestFromJSONTyped = NotificationRequestFromJSONTyped;
function NotificationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'event': (0, EventRequest_1.EventRequestToJSON)(value.event),
        'seen': value.seen,
    };
}
exports.NotificationRequestToJSON = NotificationRequestToJSON;
