// @goauthentik/api/dist/models/OAuth2Provider.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth2ProviderToJSON = exports.OAuth2ProviderFromJSONTyped = exports.OAuth2ProviderFromJSON = exports.instanceOfOAuth2Provider = void 0;
const runtime_1 = require("../runtime");
const ClientTypeEnum_1 = require("./ClientTypeEnum");
const IssuerModeEnum_1 = require("./IssuerModeEnum");
const SubModeEnum_1 = require("./SubModeEnum");
/**
 * Check if a given object implements the OAuth2Provider interface.
 */
function instanceOfOAuth2Provider(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "authorizationFlow" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "assignedApplicationSlug" in value;
    isInstance = isInstance && "assignedApplicationName" in value;
    isInstance = isInstance && "assignedBackchannelApplicationSlug" in value;
    isInstance = isInstance && "assignedBackchannelApplicationName" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfOAuth2Provider = instanceOfOAuth2Provider;
function OAuth2ProviderFromJSON(json) {
    return OAuth2ProviderFromJSONTyped(json, false);
}
exports.OAuth2ProviderFromJSON = OAuth2ProviderFromJSON;
function OAuth2ProviderFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'component': json['component'],
        'assignedApplicationSlug': json['assigned_application_slug'],
        'assignedApplicationName': json['assigned_application_name'],
        'assignedBackchannelApplicationSlug': json['assigned_backchannel_application_slug'],
        'assignedBackchannelApplicationName': json['assigned_backchannel_application_name'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'clientType': !(0, runtime_1.exists)(json, 'client_type') ? undefined : (0, ClientTypeEnum_1.ClientTypeEnumFromJSON)(json['client_type']),
        'clientId': !(0, runtime_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !(0, runtime_1.exists)(json, 'client_secret') ? undefined : json['client_secret'],
        'accessCodeValidity': !(0, runtime_1.exists)(json, 'access_code_validity') ? undefined : json['access_code_validity'],
        'accessTokenValidity': !(0, runtime_1.exists)(json, 'access_token_validity') ? undefined : json['access_token_validity'],
        'refreshTokenValidity': !(0, runtime_1.exists)(json, 'refresh_token_validity') ? undefined : json['refresh_token_validity'],
        'includeClaimsInIdToken': !(0, runtime_1.exists)(json, 'include_claims_in_id_token') ? undefined : json['include_claims_in_id_token'],
        'signingKey': !(0, runtime_1.exists)(json, 'signing_key') ? undefined : json['signing_key'],
        'redirectUris': !(0, runtime_1.exists)(json, 'redirect_uris') ? undefined : json['redirect_uris'],
        'subMode': !(0, runtime_1.exists)(json, 'sub_mode') ? undefined : (0, SubModeEnum_1.SubModeEnumFromJSON)(json['sub_mode']),
        'issuerMode': !(0, runtime_1.exists)(json, 'issuer_mode') ? undefined : (0, IssuerModeEnum_1.IssuerModeEnumFromJSON)(json['issuer_mode']),
        'jwksSources': !(0, runtime_1.exists)(json, 'jwks_sources') ? undefined : json['jwks_sources'],
    };
}
exports.OAuth2ProviderFromJSONTyped = OAuth2ProviderFromJSONTyped;
function OAuth2ProviderToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'client_type': (0, ClientTypeEnum_1.ClientTypeEnumToJSON)(value.clientType),
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'access_code_validity': value.accessCodeValidity,
        'access_token_validity': value.accessTokenValidity,
        'refresh_token_validity': value.refreshTokenValidity,
        'include_claims_in_id_token': value.includeClaimsInIdToken,
        'signing_key': value.signingKey,
        'redirect_uris': value.redirectUris,
        'sub_mode': (0, SubModeEnum_1.SubModeEnumToJSON)(value.subMode),
        'issuer_mode': (0, IssuerModeEnum_1.IssuerModeEnumToJSON)(value.issuerMode),
        'jwks_sources': value.jwksSources,
    };
}
exports.OAuth2ProviderToJSON = OAuth2ProviderToJSON;
