// @goauthentik/api/dist/models/OAuth2ProviderSetupURLs.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth2ProviderSetupURLsToJSON = exports.OAuth2ProviderSetupURLsFromJSONTyped = exports.OAuth2ProviderSetupURLsFromJSON = exports.instanceOfOAuth2ProviderSetupURLs = void 0;
/**
 * Check if a given object implements the OAuth2ProviderSetupURLs interface.
 */
function instanceOfOAuth2ProviderSetupURLs(value) {
    let isInstance = true;
    isInstance = isInstance && "issuer" in value;
    isInstance = isInstance && "authorize" in value;
    isInstance = isInstance && "token" in value;
    isInstance = isInstance && "userInfo" in value;
    isInstance = isInstance && "providerInfo" in value;
    isInstance = isInstance && "logout" in value;
    isInstance = isInstance && "jwks" in value;
    return isInstance;
}
exports.instanceOfOAuth2ProviderSetupURLs = instanceOfOAuth2ProviderSetupURLs;
function OAuth2ProviderSetupURLsFromJSON(json) {
    return OAuth2ProviderSetupURLsFromJSONTyped(json, false);
}
exports.OAuth2ProviderSetupURLsFromJSON = OAuth2ProviderSetupURLsFromJSON;
function OAuth2ProviderSetupURLsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'issuer': json['issuer'],
        'authorize': json['authorize'],
        'token': json['token'],
        'userInfo': json['user_info'],
        'providerInfo': json['provider_info'],
        'logout': json['logout'],
        'jwks': json['jwks'],
    };
}
exports.OAuth2ProviderSetupURLsFromJSONTyped = OAuth2ProviderSetupURLsFromJSONTyped;
function OAuth2ProviderSetupURLsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.OAuth2ProviderSetupURLsToJSON = OAuth2ProviderSetupURLsToJSON;
