// @goauthentik/api/dist/models/OAuthDeviceCodeFinishChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthDeviceCodeFinishChallengeResponseRequestToJSON = exports.OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped = exports.OAuthDeviceCodeFinishChallengeResponseRequestFromJSON = exports.instanceOfOAuthDeviceCodeFinishChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the OAuthDeviceCodeFinishChallengeResponseRequest interface.
 */
function instanceOfOAuthDeviceCodeFinishChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfOAuthDeviceCodeFinishChallengeResponseRequest = instanceOfOAuthDeviceCodeFinishChallengeResponseRequest;
function OAuthDeviceCodeFinishChallengeResponseRequestFromJSON(json) {
    return OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped(json, false);
}
exports.OAuthDeviceCodeFinishChallengeResponseRequestFromJSON = OAuthDeviceCodeFinishChallengeResponseRequestFromJSON;
function OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
    };
}
exports.OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped = OAuthDeviceCodeFinishChallengeResponseRequestFromJSONTyped;
function OAuthDeviceCodeFinishChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
    };
}
exports.OAuthDeviceCodeFinishChallengeResponseRequestToJSON = OAuthDeviceCodeFinishChallengeResponseRequestToJSON;
