// @goauthentik/api/dist/models/OAuthSourceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthSourceRequestToJSON = exports.OAuthSourceRequestFromJSONTyped = exports.OAuthSourceRequestFromJSON = exports.instanceOfOAuthSourceRequest = void 0;
const runtime_1 = require("../runtime");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
const ProviderTypeEnum_1 = require("./ProviderTypeEnum");
const UserMatchingModeEnum_1 = require("./UserMatchingModeEnum");
/**
 * Check if a given object implements the OAuthSourceRequest interface.
 */
function instanceOfOAuthSourceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "slug" in value;
    isInstance = isInstance && "providerType" in value;
    isInstance = isInstance && "consumerKey" in value;
    isInstance = isInstance && "consumerSecret" in value;
    return isInstance;
}
exports.instanceOfOAuthSourceRequest = instanceOfOAuthSourceRequest;
function OAuthSourceRequestFromJSON(json) {
    return OAuthSourceRequestFromJSONTyped(json, false);
}
exports.OAuthSourceRequestFromJSON = OAuthSourceRequestFromJSON;
function OAuthSourceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'slug': json['slug'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'enrollmentFlow': !(0, runtime_1.exists)(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'userMatchingMode': !(0, runtime_1.exists)(json, 'user_matching_mode') ? undefined : (0, UserMatchingModeEnum_1.UserMatchingModeEnumFromJSON)(json['user_matching_mode']),
        'userPathTemplate': !(0, runtime_1.exists)(json, 'user_path_template') ? undefined : json['user_path_template'],
        'providerType': (0, ProviderTypeEnum_1.ProviderTypeEnumFromJSON)(json['provider_type']),
        'requestTokenUrl': !(0, runtime_1.exists)(json, 'request_token_url') ? undefined : json['request_token_url'],
        'authorizationUrl': !(0, runtime_1.exists)(json, 'authorization_url') ? undefined : json['authorization_url'],
        'accessTokenUrl': !(0, runtime_1.exists)(json, 'access_token_url') ? undefined : json['access_token_url'],
        'profileUrl': !(0, runtime_1.exists)(json, 'profile_url') ? undefined : json['profile_url'],
        'consumerKey': json['consumer_key'],
        'consumerSecret': json['consumer_secret'],
        'additionalScopes': !(0, runtime_1.exists)(json, 'additional_scopes') ? undefined : json['additional_scopes'],
        'oidcWellKnownUrl': !(0, runtime_1.exists)(json, 'oidc_well_known_url') ? undefined : json['oidc_well_known_url'],
        'oidcJwksUrl': !(0, runtime_1.exists)(json, 'oidc_jwks_url') ? undefined : json['oidc_jwks_url'],
        'oidcJwks': !(0, runtime_1.exists)(json, 'oidc_jwks') ? undefined : json['oidc_jwks'],
    };
}
exports.OAuthSourceRequestFromJSONTyped = OAuthSourceRequestFromJSONTyped;
function OAuthSourceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'slug': value.slug,
        'enabled': value.enabled,
        'authentication_flow': value.authenticationFlow,
        'enrollment_flow': value.enrollmentFlow,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'user_matching_mode': (0, UserMatchingModeEnum_1.UserMatchingModeEnumToJSON)(value.userMatchingMode),
        'user_path_template': value.userPathTemplate,
        'provider_type': (0, ProviderTypeEnum_1.ProviderTypeEnumToJSON)(value.providerType),
        'request_token_url': value.requestTokenUrl,
        'authorization_url': value.authorizationUrl,
        'access_token_url': value.accessTokenUrl,
        'profile_url': value.profileUrl,
        'consumer_key': value.consumerKey,
        'consumer_secret': value.consumerSecret,
        'additional_scopes': value.additionalScopes,
        'oidc_well_known_url': value.oidcWellKnownUrl,
        'oidc_jwks_url': value.oidcJwksUrl,
        'oidc_jwks': value.oidcJwks,
    };
}
exports.OAuthSourceRequestToJSON = OAuthSourceRequestToJSON;
