// @goauthentik/api/dist/models/OAuthDeviceCodeChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthDeviceCodeChallengeResponseRequestToJSON = exports.OAuthDeviceCodeChallengeResponseRequestFromJSONTyped = exports.OAuthDeviceCodeChallengeResponseRequestFromJSON = exports.instanceOfOAuthDeviceCodeChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the OAuthDeviceCodeChallengeResponseRequest interface.
 */
function instanceOfOAuthDeviceCodeChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    return isInstance;
}
exports.instanceOfOAuthDeviceCodeChallengeResponseRequest = instanceOfOAuthDeviceCodeChallengeResponseRequest;
function OAuthDeviceCodeChallengeResponseRequestFromJSON(json) {
    return OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json, false);
}
exports.OAuthDeviceCodeChallengeResponseRequestFromJSON = OAuthDeviceCodeChallengeResponseRequestFromJSON;
function OAuthDeviceCodeChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'code': json['code'],
    };
}
exports.OAuthDeviceCodeChallengeResponseRequestFromJSONTyped = OAuthDeviceCodeChallengeResponseRequestFromJSONTyped;
function OAuthDeviceCodeChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'code': value.code,
    };
}
exports.OAuthDeviceCodeChallengeResponseRequestToJSON = OAuthDeviceCodeChallengeResponseRequestToJSON;
