// @goauthentik/api/dist/models/OutpostRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutpostRequestToJSON = exports.OutpostRequestFromJSONTyped = exports.OutpostRequestFromJSON = exports.instanceOfOutpostRequest = void 0;
const runtime_1 = require("../runtime");
const OutpostTypeEnum_1 = require("./OutpostTypeEnum");
/**
 * Check if a given object implements the OutpostRequest interface.
 */
function instanceOfOutpostRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "providers" in value;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
exports.instanceOfOutpostRequest = instanceOfOutpostRequest;
function OutpostRequestFromJSON(json) {
    return OutpostRequestFromJSONTyped(json, false);
}
exports.OutpostRequestFromJSON = OutpostRequestFromJSON;
function OutpostRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'type': (0, OutpostTypeEnum_1.OutpostTypeEnumFromJSON)(json['type']),
        'providers': json['providers'],
        'serviceConnection': !(0, runtime_1.exists)(json, 'service_connection') ? undefined : json['service_connection'],
        'config': json['config'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
    };
}
exports.OutpostRequestFromJSONTyped = OutpostRequestFromJSONTyped;
function OutpostRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': (0, OutpostTypeEnum_1.OutpostTypeEnumToJSON)(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
exports.OutpostRequestToJSON = OutpostRequestToJSON;
