// @goauthentik/api/dist/models/OutpostDefaultConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutpostDefaultConfigToJSON = exports.OutpostDefaultConfigFromJSONTyped = exports.OutpostDefaultConfigFromJSON = exports.instanceOfOutpostDefaultConfig = void 0;
/**
 * Check if a given object implements the OutpostDefaultConfig interface.
 */
function instanceOfOutpostDefaultConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
exports.instanceOfOutpostDefaultConfig = instanceOfOutpostDefaultConfig;
function OutpostDefaultConfigFromJSON(json) {
    return OutpostDefaultConfigFromJSONTyped(json, false);
}
exports.OutpostDefaultConfigFromJSON = OutpostDefaultConfigFromJSON;
function OutpostDefaultConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'config': json['config'],
    };
}
exports.OutpostDefaultConfigFromJSONTyped = OutpostDefaultConfigFromJSONTyped;
function OutpostDefaultConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.OutpostDefaultConfigToJSON = OutpostDefaultConfigToJSON;
