// @goauthentik/api/dist/models/Outpost.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutpostToJSON = exports.OutpostFromJSONTyped = exports.OutpostFromJSON = exports.instanceOfOutpost = void 0;
const runtime_1 = require("../runtime");
const OutpostTypeEnum_1 = require("./OutpostTypeEnum");
const Provider_1 = require("./Provider");
const ServiceConnection_1 = require("./ServiceConnection");
/**
 * Check if a given object implements the Outpost interface.
 */
function instanceOfOutpost(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "providers" in value;
    isInstance = isInstance && "providersObj" in value;
    isInstance = isInstance && "serviceConnectionObj" in value;
    isInstance = isInstance && "tokenIdentifier" in value;
    isInstance = isInstance && "config" in value;
    return isInstance;
}
exports.instanceOfOutpost = instanceOfOutpost;
function OutpostFromJSON(json) {
    return OutpostFromJSONTyped(json, false);
}
exports.OutpostFromJSON = OutpostFromJSON;
function OutpostFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'type': (0, OutpostTypeEnum_1.OutpostTypeEnumFromJSON)(json['type']),
        'providers': json['providers'],
        'providersObj': (json['providers_obj'].map(Provider_1.ProviderFromJSON)),
        'serviceConnection': !(0, runtime_1.exists)(json, 'service_connection') ? undefined : json['service_connection'],
        'serviceConnectionObj': (0, ServiceConnection_1.ServiceConnectionFromJSON)(json['service_connection_obj']),
        'tokenIdentifier': json['token_identifier'],
        'config': json['config'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
    };
}
exports.OutpostFromJSONTyped = OutpostFromJSONTyped;
function OutpostToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': (0, OutpostTypeEnum_1.OutpostTypeEnumToJSON)(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
exports.OutpostToJSON = OutpostToJSON;
