// @goauthentik/api/dist/models/OutpostTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutpostTypeEnumToJSON = exports.OutpostTypeEnumFromJSONTyped = exports.OutpostTypeEnumFromJSON = exports.OutpostTypeEnum = void 0;
/**
 * * `proxy` - Proxy
 * * `ldap` - Ldap
 * * `radius` - Radius
 * * `rac` - Rac
 * @export
 */
exports.OutpostTypeEnum = {
    Proxy: 'proxy',
    Ldap: 'ldap',
    Radius: 'radius',
    Rac: 'rac',
    UnknownDefaultOpenApi: '11184809'
};
function OutpostTypeEnumFromJSON(json) {
    return OutpostTypeEnumFromJSONTyped(json, false);
}
exports.OutpostTypeEnumFromJSON = OutpostTypeEnumFromJSON;
function OutpostTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.OutpostTypeEnumFromJSONTyped = OutpostTypeEnumFromJSONTyped;
function OutpostTypeEnumToJSON(value) {
    return value;
}
exports.OutpostTypeEnumToJSON = OutpostTypeEnumToJSON;
