// @goauthentik/api/dist/models/OutpostHealth.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutpostHealthToJSON = exports.OutpostHealthFromJSONTyped = exports.OutpostHealthFromJSON = exports.instanceOfOutpostHealth = void 0;
/**
 * Check if a given object implements the OutpostHealth interface.
 */
function instanceOfOutpostHealth(value) {
    let isInstance = true;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "lastSeen" in value;
    isInstance = isInstance && "version" in value;
    isInstance = isInstance && "versionShould" in value;
    isInstance = isInstance && "versionOutdated" in value;
    isInstance = isInstance && "buildHash" in value;
    isInstance = isInstance && "buildHashShould" in value;
    isInstance = isInstance && "hostname" in value;
    return isInstance;
}
exports.instanceOfOutpostHealth = instanceOfOutpostHealth;
function OutpostHealthFromJSON(json) {
    return OutpostHealthFromJSONTyped(json, false);
}
exports.OutpostHealthFromJSON = OutpostHealthFromJSON;
function OutpostHealthFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'uid': json['uid'],
        'lastSeen': (new Date(json['last_seen'])),
        'version': json['version'],
        'versionShould': json['version_should'],
        'versionOutdated': json['version_outdated'],
        'buildHash': json['build_hash'],
        'buildHashShould': json['build_hash_should'],
        'hostname': json['hostname'],
    };
}
exports.OutpostHealthFromJSONTyped = OutpostHealthFromJSONTyped;
function OutpostHealthToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.OutpostHealthToJSON = OutpostHealthToJSON;
