// @goauthentik/api/dist/models/PaginatedBlueprintInstanceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedBlueprintInstanceListToJSON = exports.PaginatedBlueprintInstanceListFromJSONTyped = exports.PaginatedBlueprintInstanceListFromJSON = exports.instanceOfPaginatedBlueprintInstanceList = void 0;
const BlueprintInstance_1 = require("./BlueprintInstance");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedBlueprintInstanceList interface.
 */
function instanceOfPaginatedBlueprintInstanceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedBlueprintInstanceList = instanceOfPaginatedBlueprintInstanceList;
function PaginatedBlueprintInstanceListFromJSON(json) {
    return PaginatedBlueprintInstanceListFromJSONTyped(json, false);
}
exports.PaginatedBlueprintInstanceListFromJSON = PaginatedBlueprintInstanceListFromJSON;
function PaginatedBlueprintInstanceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(BlueprintInstance_1.BlueprintInstanceFromJSON)),
    };
}
exports.PaginatedBlueprintInstanceListFromJSONTyped = PaginatedBlueprintInstanceListFromJSONTyped;
function PaginatedBlueprintInstanceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(BlueprintInstance_1.BlueprintInstanceToJSON)),
    };
}
exports.PaginatedBlueprintInstanceListToJSON = PaginatedBlueprintInstanceListToJSON;
