// @goauthentik/api/dist/models/PaginatedEventList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedEventListToJSON = exports.PaginatedEventListFromJSONTyped = exports.PaginatedEventListFromJSON = exports.instanceOfPaginatedEventList = void 0;
const Event_1 = require("./Event");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedEventList interface.
 */
function instanceOfPaginatedEventList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedEventList = instanceOfPaginatedEventList;
function PaginatedEventListFromJSON(json) {
    return PaginatedEventListFromJSONTyped(json, false);
}
exports.PaginatedEventListFromJSON = PaginatedEventListFromJSON;
function PaginatedEventListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Event_1.EventFromJSON)),
    };
}
exports.PaginatedEventListFromJSONTyped = PaginatedEventListFromJSONTyped;
function PaginatedEventListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Event_1.EventToJSON)),
    };
}
exports.PaginatedEventListToJSON = PaginatedEventListToJSON;
