// @goauthentik/api/dist/models/PaginatedPlexSourceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPlexSourceListToJSON = exports.PaginatedPlexSourceListFromJSONTyped = exports.PaginatedPlexSourceListFromJSON = exports.instanceOfPaginatedPlexSourceList = void 0;
const Pagination_1 = require("./Pagination");
const PlexSource_1 = require("./PlexSource");
/**
 * Check if a given object implements the PaginatedPlexSourceList interface.
 */
function instanceOfPaginatedPlexSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPlexSourceList = instanceOfPaginatedPlexSourceList;
function PaginatedPlexSourceListFromJSON(json) {
    return PaginatedPlexSourceListFromJSONTyped(json, false);
}
exports.PaginatedPlexSourceListFromJSON = PaginatedPlexSourceListFromJSON;
function PaginatedPlexSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PlexSource_1.PlexSourceFromJSON)),
    };
}
exports.PaginatedPlexSourceListFromJSONTyped = PaginatedPlexSourceListFromJSONTyped;
function PaginatedPlexSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PlexSource_1.PlexSourceToJSON)),
    };
}
exports.PaginatedPlexSourceListToJSON = PaginatedPlexSourceListToJSON;
