// @goauthentik/api/dist/models/PaginatedRadiusProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedRadiusProviderListToJSON = exports.PaginatedRadiusProviderListFromJSONTyped = exports.PaginatedRadiusProviderListFromJSON = exports.instanceOfPaginatedRadiusProviderList = void 0;
const Pagination_1 = require("./Pagination");
const RadiusProvider_1 = require("./RadiusProvider");
/**
 * Check if a given object implements the PaginatedRadiusProviderList interface.
 */
function instanceOfPaginatedRadiusProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedRadiusProviderList = instanceOfPaginatedRadiusProviderList;
function PaginatedRadiusProviderListFromJSON(json) {
    return PaginatedRadiusProviderListFromJSONTyped(json, false);
}
exports.PaginatedRadiusProviderListFromJSON = PaginatedRadiusProviderListFromJSON;
function PaginatedRadiusProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(RadiusProvider_1.RadiusProviderFromJSON)),
    };
}
exports.PaginatedRadiusProviderListFromJSONTyped = PaginatedRadiusProviderListFromJSONTyped;
function PaginatedRadiusProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(RadiusProvider_1.RadiusProviderToJSON)),
    };
}
exports.PaginatedRadiusProviderListToJSON = PaginatedRadiusProviderListToJSON;
