// @goauthentik/api/dist/models/PaginatedSCIMMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSCIMMappingListToJSON = exports.PaginatedSCIMMappingListFromJSONTyped = exports.PaginatedSCIMMappingListFromJSON = exports.instanceOfPaginatedSCIMMappingList = void 0;
const Pagination_1 = require("./Pagination");
const SCIMMapping_1 = require("./SCIMMapping");
/**
 * Check if a given object implements the PaginatedSCIMMappingList interface.
 */
function instanceOfPaginatedSCIMMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSCIMMappingList = instanceOfPaginatedSCIMMappingList;
function PaginatedSCIMMappingListFromJSON(json) {
    return PaginatedSCIMMappingListFromJSONTyped(json, false);
}
exports.PaginatedSCIMMappingListFromJSON = PaginatedSCIMMappingListFromJSON;
function PaginatedSCIMMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SCIMMapping_1.SCIMMappingFromJSON)),
    };
}
exports.PaginatedSCIMMappingListFromJSONTyped = PaginatedSCIMMappingListFromJSONTyped;
function PaginatedSCIMMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SCIMMapping_1.SCIMMappingToJSON)),
    };
}
exports.PaginatedSCIMMappingListToJSON = PaginatedSCIMMappingListToJSON;
