// @goauthentik/api/dist/models/PaginatedExpressionPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedExpressionPolicyListToJSON = exports.PaginatedExpressionPolicyListFromJSONTyped = exports.PaginatedExpressionPolicyListFromJSON = exports.instanceOfPaginatedExpressionPolicyList = void 0;
const ExpressionPolicy_1 = require("./ExpressionPolicy");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedExpressionPolicyList interface.
 */
function instanceOfPaginatedExpressionPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedExpressionPolicyList = instanceOfPaginatedExpressionPolicyList;
function PaginatedExpressionPolicyListFromJSON(json) {
    return PaginatedExpressionPolicyListFromJSONTyped(json, false);
}
exports.PaginatedExpressionPolicyListFromJSON = PaginatedExpressionPolicyListFromJSON;
function PaginatedExpressionPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ExpressionPolicy_1.ExpressionPolicyFromJSON)),
    };
}
exports.PaginatedExpressionPolicyListFromJSONTyped = PaginatedExpressionPolicyListFromJSONTyped;
function PaginatedExpressionPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ExpressionPolicy_1.ExpressionPolicyToJSON)),
    };
}
exports.PaginatedExpressionPolicyListToJSON = PaginatedExpressionPolicyListToJSON;
