// @goauthentik/api/dist/models/PaginatedRadiusOutpostConfigList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedRadiusOutpostConfigListToJSON = exports.PaginatedRadiusOutpostConfigListFromJSONTyped = exports.PaginatedRadiusOutpostConfigListFromJSON = exports.instanceOfPaginatedRadiusOutpostConfigList = void 0;
const Pagination_1 = require("./Pagination");
const RadiusOutpostConfig_1 = require("./RadiusOutpostConfig");
/**
 * Check if a given object implements the PaginatedRadiusOutpostConfigList interface.
 */
function instanceOfPaginatedRadiusOutpostConfigList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedRadiusOutpostConfigList = instanceOfPaginatedRadiusOutpostConfigList;
function PaginatedRadiusOutpostConfigListFromJSON(json) {
    return PaginatedRadiusOutpostConfigListFromJSONTyped(json, false);
}
exports.PaginatedRadiusOutpostConfigListFromJSON = PaginatedRadiusOutpostConfigListFromJSON;
function PaginatedRadiusOutpostConfigListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(RadiusOutpostConfig_1.RadiusOutpostConfigFromJSON)),
    };
}
exports.PaginatedRadiusOutpostConfigListFromJSONTyped = PaginatedRadiusOutpostConfigListFromJSONTyped;
function PaginatedRadiusOutpostConfigListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(RadiusOutpostConfig_1.RadiusOutpostConfigToJSON)),
    };
}
exports.PaginatedRadiusOutpostConfigListToJSON = PaginatedRadiusOutpostConfigListToJSON;
