// @goauthentik/api/dist/models/PaginatedSAMLSourceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSAMLSourceListToJSON = exports.PaginatedSAMLSourceListFromJSONTyped = exports.PaginatedSAMLSourceListFromJSON = exports.instanceOfPaginatedSAMLSourceList = void 0;
const Pagination_1 = require("./Pagination");
const SAMLSource_1 = require("./SAMLSource");
/**
 * Check if a given object implements the PaginatedSAMLSourceList interface.
 */
function instanceOfPaginatedSAMLSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSAMLSourceList = instanceOfPaginatedSAMLSourceList;
function PaginatedSAMLSourceListFromJSON(json) {
    return PaginatedSAMLSourceListFromJSONTyped(json, false);
}
exports.PaginatedSAMLSourceListFromJSON = PaginatedSAMLSourceListFromJSON;
function PaginatedSAMLSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SAMLSource_1.SAMLSourceFromJSON)),
    };
}
exports.PaginatedSAMLSourceListFromJSONTyped = PaginatedSAMLSourceListFromJSONTyped;
function PaginatedSAMLSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SAMLSource_1.SAMLSourceToJSON)),
    };
}
exports.PaginatedSAMLSourceListToJSON = PaginatedSAMLSourceListToJSON;
