// @goauthentik/api/dist/models/Pagination.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginationToJSON = exports.PaginationFromJSONTyped = exports.PaginationFromJSON = exports.instanceOfPagination = void 0;
/**
 * Check if a given object implements the Pagination interface.
 */
function instanceOfPagination(value) {
    let isInstance = true;
    isInstance = isInstance && "next" in value;
    isInstance = isInstance && "previous" in value;
    isInstance = isInstance && "count" in value;
    isInstance = isInstance && "current" in value;
    isInstance = isInstance && "totalPages" in value;
    isInstance = isInstance && "startIndex" in value;
    isInstance = isInstance && "endIndex" in value;
    return isInstance;
}
exports.instanceOfPagination = instanceOfPagination;
function PaginationFromJSON(json) {
    return PaginationFromJSONTyped(json, false);
}
exports.PaginationFromJSON = PaginationFromJSON;
function PaginationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'next': json['next'],
        'previous': json['previous'],
        'count': json['count'],
        'current': json['current'],
        'totalPages': json['total_pages'],
        'startIndex': json['start_index'],
        'endIndex': json['end_index'],
    };
}
exports.PaginationFromJSONTyped = PaginationFromJSONTyped;
function PaginationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'next': value.next,
        'previous': value.previous,
        'count': value.count,
        'current': value.current,
        'total_pages': value.totalPages,
        'start_index': value.startIndex,
        'end_index': value.endIndex,
    };
}
exports.PaginationToJSON = PaginationToJSON;
