// @goauthentik/api/dist/models/PaginatedLDAPSourceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedLDAPSourceListToJSON = exports.PaginatedLDAPSourceListFromJSONTyped = exports.PaginatedLDAPSourceListFromJSON = exports.instanceOfPaginatedLDAPSourceList = void 0;
const LDAPSource_1 = require("./LDAPSource");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedLDAPSourceList interface.
 */
function instanceOfPaginatedLDAPSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedLDAPSourceList = instanceOfPaginatedLDAPSourceList;
function PaginatedLDAPSourceListFromJSON(json) {
    return PaginatedLDAPSourceListFromJSONTyped(json, false);
}
exports.PaginatedLDAPSourceListFromJSON = PaginatedLDAPSourceListFromJSON;
function PaginatedLDAPSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(LDAPSource_1.LDAPSourceFromJSON)),
    };
}
exports.PaginatedLDAPSourceListFromJSONTyped = PaginatedLDAPSourceListFromJSONTyped;
function PaginatedLDAPSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(LDAPSource_1.LDAPSourceToJSON)),
    };
}
exports.PaginatedLDAPSourceListToJSON = PaginatedLDAPSourceListToJSON;
