// @goauthentik/api/dist/models/PaginatedInvitationStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedInvitationStageListToJSON = exports.PaginatedInvitationStageListFromJSONTyped = exports.PaginatedInvitationStageListFromJSON = exports.instanceOfPaginatedInvitationStageList = void 0;
const InvitationStage_1 = require("./InvitationStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedInvitationStageList interface.
 */
function instanceOfPaginatedInvitationStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedInvitationStageList = instanceOfPaginatedInvitationStageList;
function PaginatedInvitationStageListFromJSON(json) {
    return PaginatedInvitationStageListFromJSONTyped(json, false);
}
exports.PaginatedInvitationStageListFromJSON = PaginatedInvitationStageListFromJSON;
function PaginatedInvitationStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(InvitationStage_1.InvitationStageFromJSON)),
    };
}
exports.PaginatedInvitationStageListFromJSONTyped = PaginatedInvitationStageListFromJSONTyped;
function PaginatedInvitationStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(InvitationStage_1.InvitationStageToJSON)),
    };
}
exports.PaginatedInvitationStageListToJSON = PaginatedInvitationStageListToJSON;
