// @goauthentik/api/dist/models/PaginatedProxyOutpostConfigList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedProxyOutpostConfigListToJSON = exports.PaginatedProxyOutpostConfigListFromJSONTyped = exports.PaginatedProxyOutpostConfigListFromJSON = exports.instanceOfPaginatedProxyOutpostConfigList = void 0;
const Pagination_1 = require("./Pagination");
const ProxyOutpostConfig_1 = require("./ProxyOutpostConfig");
/**
 * Check if a given object implements the PaginatedProxyOutpostConfigList interface.
 */
function instanceOfPaginatedProxyOutpostConfigList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedProxyOutpostConfigList = instanceOfPaginatedProxyOutpostConfigList;
function PaginatedProxyOutpostConfigListFromJSON(json) {
    return PaginatedProxyOutpostConfigListFromJSONTyped(json, false);
}
exports.PaginatedProxyOutpostConfigListFromJSON = PaginatedProxyOutpostConfigListFromJSON;
function PaginatedProxyOutpostConfigListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ProxyOutpostConfig_1.ProxyOutpostConfigFromJSON)),
    };
}
exports.PaginatedProxyOutpostConfigListFromJSONTyped = PaginatedProxyOutpostConfigListFromJSONTyped;
function PaginatedProxyOutpostConfigListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ProxyOutpostConfig_1.ProxyOutpostConfigToJSON)),
    };
}
exports.PaginatedProxyOutpostConfigListToJSON = PaginatedProxyOutpostConfigListToJSON;
