// @goauthentik/api/dist/models/PaginatedExtraUserObjectPermissionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedExtraUserObjectPermissionListToJSON = exports.PaginatedExtraUserObjectPermissionListFromJSONTyped = exports.PaginatedExtraUserObjectPermissionListFromJSON = exports.instanceOfPaginatedExtraUserObjectPermissionList = void 0;
const ExtraUserObjectPermission_1 = require("./ExtraUserObjectPermission");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedExtraUserObjectPermissionList interface.
 */
function instanceOfPaginatedExtraUserObjectPermissionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedExtraUserObjectPermissionList = instanceOfPaginatedExtraUserObjectPermissionList;
function PaginatedExtraUserObjectPermissionListFromJSON(json) {
    return PaginatedExtraUserObjectPermissionListFromJSONTyped(json, false);
}
exports.PaginatedExtraUserObjectPermissionListFromJSON = PaginatedExtraUserObjectPermissionListFromJSON;
function PaginatedExtraUserObjectPermissionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ExtraUserObjectPermission_1.ExtraUserObjectPermissionFromJSON)),
    };
}
exports.PaginatedExtraUserObjectPermissionListFromJSONTyped = PaginatedExtraUserObjectPermissionListFromJSONTyped;
function PaginatedExtraUserObjectPermissionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ExtraUserObjectPermission_1.ExtraUserObjectPermissionToJSON)),
    };
}
exports.PaginatedExtraUserObjectPermissionListToJSON = PaginatedExtraUserObjectPermissionListToJSON;
