// @goauthentik/api/dist/models/PaginatedProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedProviderListToJSON = exports.PaginatedProviderListFromJSONTyped = exports.PaginatedProviderListFromJSON = exports.instanceOfPaginatedProviderList = void 0;
const Pagination_1 = require("./Pagination");
const Provider_1 = require("./Provider");
/**
 * Check if a given object implements the PaginatedProviderList interface.
 */
function instanceOfPaginatedProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedProviderList = instanceOfPaginatedProviderList;
function PaginatedProviderListFromJSON(json) {
    return PaginatedProviderListFromJSONTyped(json, false);
}
exports.PaginatedProviderListFromJSON = PaginatedProviderListFromJSON;
function PaginatedProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Provider_1.ProviderFromJSON)),
    };
}
exports.PaginatedProviderListFromJSONTyped = PaginatedProviderListFromJSONTyped;
function PaginatedProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Provider_1.ProviderToJSON)),
    };
}
exports.PaginatedProviderListToJSON = PaginatedProviderListToJSON;
