// @goauthentik/api/dist/models/PaginatedDomainList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDomainListToJSON = exports.PaginatedDomainListFromJSONTyped = exports.PaginatedDomainListFromJSON = exports.instanceOfPaginatedDomainList = void 0;
const Domain_1 = require("./Domain");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDomainList interface.
 */
function instanceOfPaginatedDomainList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDomainList = instanceOfPaginatedDomainList;
function PaginatedDomainListFromJSON(json) {
    return PaginatedDomainListFromJSONTyped(json, false);
}
exports.PaginatedDomainListFromJSON = PaginatedDomainListFromJSON;
function PaginatedDomainListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Domain_1.DomainFromJSON)),
    };
}
exports.PaginatedDomainListFromJSONTyped = PaginatedDomainListFromJSONTyped;
function PaginatedDomainListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Domain_1.DomainToJSON)),
    };
}
exports.PaginatedDomainListToJSON = PaginatedDomainListToJSON;
