// @goauthentik/api/dist/models/PaginatedOutpostList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedOutpostListToJSON = exports.PaginatedOutpostListFromJSONTyped = exports.PaginatedOutpostListFromJSON = exports.instanceOfPaginatedOutpostList = void 0;
const Outpost_1 = require("./Outpost");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedOutpostList interface.
 */
function instanceOfPaginatedOutpostList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedOutpostList = instanceOfPaginatedOutpostList;
function PaginatedOutpostListFromJSON(json) {
    return PaginatedOutpostListFromJSONTyped(json, false);
}
exports.PaginatedOutpostListFromJSON = PaginatedOutpostListFromJSON;
function PaginatedOutpostListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Outpost_1.OutpostFromJSON)),
    };
}
exports.PaginatedOutpostListFromJSONTyped = PaginatedOutpostListFromJSONTyped;
function PaginatedOutpostListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Outpost_1.OutpostToJSON)),
    };
}
exports.PaginatedOutpostListToJSON = PaginatedOutpostListToJSON;
