// @goauthentik/api/dist/models/PaginatedSCIMProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSCIMProviderListToJSON = exports.PaginatedSCIMProviderListFromJSONTyped = exports.PaginatedSCIMProviderListFromJSON = exports.instanceOfPaginatedSCIMProviderList = void 0;
const Pagination_1 = require("./Pagination");
const SCIMProvider_1 = require("./SCIMProvider");
/**
 * Check if a given object implements the PaginatedSCIMProviderList interface.
 */
function instanceOfPaginatedSCIMProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSCIMProviderList = instanceOfPaginatedSCIMProviderList;
function PaginatedSCIMProviderListFromJSON(json) {
    return PaginatedSCIMProviderListFromJSONTyped(json, false);
}
exports.PaginatedSCIMProviderListFromJSON = PaginatedSCIMProviderListFromJSON;
function PaginatedSCIMProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SCIMProvider_1.SCIMProviderFromJSON)),
    };
}
exports.PaginatedSCIMProviderListFromJSONTyped = PaginatedSCIMProviderListFromJSONTyped;
function PaginatedSCIMProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SCIMProvider_1.SCIMProviderToJSON)),
    };
}
exports.PaginatedSCIMProviderListToJSON = PaginatedSCIMProviderListToJSON;
