// @goauthentik/api/dist/models/PaginatedSourceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSourceListToJSON = exports.PaginatedSourceListFromJSONTyped = exports.PaginatedSourceListFromJSON = exports.instanceOfPaginatedSourceList = void 0;
const Pagination_1 = require("./Pagination");
const Source_1 = require("./Source");
/**
 * Check if a given object implements the PaginatedSourceList interface.
 */
function instanceOfPaginatedSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSourceList = instanceOfPaginatedSourceList;
function PaginatedSourceListFromJSON(json) {
    return PaginatedSourceListFromJSONTyped(json, false);
}
exports.PaginatedSourceListFromJSON = PaginatedSourceListFromJSON;
function PaginatedSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Source_1.SourceFromJSON)),
    };
}
exports.PaginatedSourceListFromJSONTyped = PaginatedSourceListFromJSONTyped;
function PaginatedSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Source_1.SourceToJSON)),
    };
}
exports.PaginatedSourceListToJSON = PaginatedSourceListToJSON;
