// @goauthentik/api/dist/models/PaginatedPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPolicyListToJSON = exports.PaginatedPolicyListFromJSONTyped = exports.PaginatedPolicyListFromJSON = exports.instanceOfPaginatedPolicyList = void 0;
const Pagination_1 = require("./Pagination");
const Policy_1 = require("./Policy");
/**
 * Check if a given object implements the PaginatedPolicyList interface.
 */
function instanceOfPaginatedPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPolicyList = instanceOfPaginatedPolicyList;
function PaginatedPolicyListFromJSON(json) {
    return PaginatedPolicyListFromJSONTyped(json, false);
}
exports.PaginatedPolicyListFromJSON = PaginatedPolicyListFromJSON;
function PaginatedPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Policy_1.PolicyFromJSON)),
    };
}
exports.PaginatedPolicyListFromJSONTyped = PaginatedPolicyListFromJSONTyped;
function PaginatedPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Policy_1.PolicyToJSON)),
    };
}
exports.PaginatedPolicyListToJSON = PaginatedPolicyListToJSON;
