// @goauthentik/api/dist/models/PaginatedUserSAMLSourceConnectionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserSAMLSourceConnectionListToJSON = exports.PaginatedUserSAMLSourceConnectionListFromJSONTyped = exports.PaginatedUserSAMLSourceConnectionListFromJSON = exports.instanceOfPaginatedUserSAMLSourceConnectionList = void 0;
const Pagination_1 = require("./Pagination");
const UserSAMLSourceConnection_1 = require("./UserSAMLSourceConnection");
/**
 * Check if a given object implements the PaginatedUserSAMLSourceConnectionList interface.
 */
function instanceOfPaginatedUserSAMLSourceConnectionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserSAMLSourceConnectionList = instanceOfPaginatedUserSAMLSourceConnectionList;
function PaginatedUserSAMLSourceConnectionListFromJSON(json) {
    return PaginatedUserSAMLSourceConnectionListFromJSONTyped(json, false);
}
exports.PaginatedUserSAMLSourceConnectionListFromJSON = PaginatedUserSAMLSourceConnectionListFromJSON;
function PaginatedUserSAMLSourceConnectionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserSAMLSourceConnection_1.UserSAMLSourceConnectionFromJSON)),
    };
}
exports.PaginatedUserSAMLSourceConnectionListFromJSONTyped = PaginatedUserSAMLSourceConnectionListFromJSONTyped;
function PaginatedUserSAMLSourceConnectionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserSAMLSourceConnection_1.UserSAMLSourceConnectionToJSON)),
    };
}
exports.PaginatedUserSAMLSourceConnectionListToJSON = PaginatedUserSAMLSourceConnectionListToJSON;
