// @goauthentik/api/dist/models/PaginatedWebAuthnDeviceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedWebAuthnDeviceListToJSON = exports.PaginatedWebAuthnDeviceListFromJSONTyped = exports.PaginatedWebAuthnDeviceListFromJSON = exports.instanceOfPaginatedWebAuthnDeviceList = void 0;
const Pagination_1 = require("./Pagination");
const WebAuthnDevice_1 = require("./WebAuthnDevice");
/**
 * Check if a given object implements the PaginatedWebAuthnDeviceList interface.
 */
function instanceOfPaginatedWebAuthnDeviceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedWebAuthnDeviceList = instanceOfPaginatedWebAuthnDeviceList;
function PaginatedWebAuthnDeviceListFromJSON(json) {
    return PaginatedWebAuthnDeviceListFromJSONTyped(json, false);
}
exports.PaginatedWebAuthnDeviceListFromJSON = PaginatedWebAuthnDeviceListFromJSON;
function PaginatedWebAuthnDeviceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(WebAuthnDevice_1.WebAuthnDeviceFromJSON)),
    };
}
exports.PaginatedWebAuthnDeviceListFromJSONTyped = PaginatedWebAuthnDeviceListFromJSONTyped;
function PaginatedWebAuthnDeviceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(WebAuthnDevice_1.WebAuthnDeviceToJSON)),
    };
}
exports.PaginatedWebAuthnDeviceListToJSON = PaginatedWebAuthnDeviceListToJSON;
