// @goauthentik/api/dist/models/PaginatedSAMLPropertyMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSAMLPropertyMappingListToJSON = exports.PaginatedSAMLPropertyMappingListFromJSONTyped = exports.PaginatedSAMLPropertyMappingListFromJSON = exports.instanceOfPaginatedSAMLPropertyMappingList = void 0;
const Pagination_1 = require("./Pagination");
const SAMLPropertyMapping_1 = require("./SAMLPropertyMapping");
/**
 * Check if a given object implements the PaginatedSAMLPropertyMappingList interface.
 */
function instanceOfPaginatedSAMLPropertyMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSAMLPropertyMappingList = instanceOfPaginatedSAMLPropertyMappingList;
function PaginatedSAMLPropertyMappingListFromJSON(json) {
    return PaginatedSAMLPropertyMappingListFromJSONTyped(json, false);
}
exports.PaginatedSAMLPropertyMappingListFromJSON = PaginatedSAMLPropertyMappingListFromJSON;
function PaginatedSAMLPropertyMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SAMLPropertyMapping_1.SAMLPropertyMappingFromJSON)),
    };
}
exports.PaginatedSAMLPropertyMappingListFromJSONTyped = PaginatedSAMLPropertyMappingListFromJSONTyped;
function PaginatedSAMLPropertyMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SAMLPropertyMapping_1.SAMLPropertyMappingToJSON)),
    };
}
exports.PaginatedSAMLPropertyMappingListToJSON = PaginatedSAMLPropertyMappingListToJSON;
