// @goauthentik/api/dist/models/PaginatedPolicyBindingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPolicyBindingListToJSON = exports.PaginatedPolicyBindingListFromJSONTyped = exports.PaginatedPolicyBindingListFromJSON = exports.instanceOfPaginatedPolicyBindingList = void 0;
const Pagination_1 = require("./Pagination");
const PolicyBinding_1 = require("./PolicyBinding");
/**
 * Check if a given object implements the PaginatedPolicyBindingList interface.
 */
function instanceOfPaginatedPolicyBindingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPolicyBindingList = instanceOfPaginatedPolicyBindingList;
function PaginatedPolicyBindingListFromJSON(json) {
    return PaginatedPolicyBindingListFromJSONTyped(json, false);
}
exports.PaginatedPolicyBindingListFromJSON = PaginatedPolicyBindingListFromJSON;
function PaginatedPolicyBindingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PolicyBinding_1.PolicyBindingFromJSON)),
    };
}
exports.PaginatedPolicyBindingListFromJSONTyped = PaginatedPolicyBindingListFromJSONTyped;
function PaginatedPolicyBindingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PolicyBinding_1.PolicyBindingToJSON)),
    };
}
exports.PaginatedPolicyBindingListToJSON = PaginatedPolicyBindingListToJSON;
