// @goauthentik/api/dist/models/PaginatedAuthenticatorValidateStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticatorValidateStageListToJSON = exports.PaginatedAuthenticatorValidateStageListFromJSONTyped = exports.PaginatedAuthenticatorValidateStageListFromJSON = exports.instanceOfPaginatedAuthenticatorValidateStageList = void 0;
const AuthenticatorValidateStage_1 = require("./AuthenticatorValidateStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticatorValidateStageList interface.
 */
function instanceOfPaginatedAuthenticatorValidateStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticatorValidateStageList = instanceOfPaginatedAuthenticatorValidateStageList;
function PaginatedAuthenticatorValidateStageListFromJSON(json) {
    return PaginatedAuthenticatorValidateStageListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticatorValidateStageListFromJSON = PaginatedAuthenticatorValidateStageListFromJSON;
function PaginatedAuthenticatorValidateStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticatorValidateStage_1.AuthenticatorValidateStageFromJSON)),
    };
}
exports.PaginatedAuthenticatorValidateStageListFromJSONTyped = PaginatedAuthenticatorValidateStageListFromJSONTyped;
function PaginatedAuthenticatorValidateStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticatorValidateStage_1.AuthenticatorValidateStageToJSON)),
    };
}
exports.PaginatedAuthenticatorValidateStageListToJSON = PaginatedAuthenticatorValidateStageListToJSON;
