// @goauthentik/api/dist/models/PaginatedTokenModelList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedTokenModelListToJSON = exports.PaginatedTokenModelListFromJSONTyped = exports.PaginatedTokenModelListFromJSON = exports.instanceOfPaginatedTokenModelList = void 0;
const Pagination_1 = require("./Pagination");
const TokenModel_1 = require("./TokenModel");
/**
 * Check if a given object implements the PaginatedTokenModelList interface.
 */
function instanceOfPaginatedTokenModelList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedTokenModelList = instanceOfPaginatedTokenModelList;
function PaginatedTokenModelListFromJSON(json) {
    return PaginatedTokenModelListFromJSONTyped(json, false);
}
exports.PaginatedTokenModelListFromJSON = PaginatedTokenModelListFromJSON;
function PaginatedTokenModelListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(TokenModel_1.TokenModelFromJSON)),
    };
}
exports.PaginatedTokenModelListFromJSONTyped = PaginatedTokenModelListFromJSONTyped;
function PaginatedTokenModelListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(TokenModel_1.TokenModelToJSON)),
    };
}
exports.PaginatedTokenModelListToJSON = PaginatedTokenModelListToJSON;
