// @goauthentik/api/dist/models/PaginatedScopeMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedScopeMappingListToJSON = exports.PaginatedScopeMappingListFromJSONTyped = exports.PaginatedScopeMappingListFromJSON = exports.instanceOfPaginatedScopeMappingList = void 0;
const Pagination_1 = require("./Pagination");
const ScopeMapping_1 = require("./ScopeMapping");
/**
 * Check if a given object implements the PaginatedScopeMappingList interface.
 */
function instanceOfPaginatedScopeMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedScopeMappingList = instanceOfPaginatedScopeMappingList;
function PaginatedScopeMappingListFromJSON(json) {
    return PaginatedScopeMappingListFromJSONTyped(json, false);
}
exports.PaginatedScopeMappingListFromJSON = PaginatedScopeMappingListFromJSON;
function PaginatedScopeMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ScopeMapping_1.ScopeMappingFromJSON)),
    };
}
exports.PaginatedScopeMappingListFromJSONTyped = PaginatedScopeMappingListFromJSONTyped;
function PaginatedScopeMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ScopeMapping_1.ScopeMappingToJSON)),
    };
}
exports.PaginatedScopeMappingListToJSON = PaginatedScopeMappingListToJSON;
