// @goauthentik/api/dist/models/PaginatedExtAuthStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedExtAuthStageListToJSON = exports.PaginatedExtAuthStageListFromJSONTyped = exports.PaginatedExtAuthStageListFromJSON = exports.instanceOfPaginatedExtAuthStageList = void 0;
const ExtAuthStage_1 = require("./ExtAuthStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedExtAuthStageList interface.
 */
function instanceOfPaginatedExtAuthStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedExtAuthStageList = instanceOfPaginatedExtAuthStageList;
function PaginatedExtAuthStageListFromJSON(json) {
    return PaginatedExtAuthStageListFromJSONTyped(json, false);
}
exports.PaginatedExtAuthStageListFromJSON = PaginatedExtAuthStageListFromJSON;
function PaginatedExtAuthStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ExtAuthStage_1.ExtAuthStageFromJSON)),
    };
}
exports.PaginatedExtAuthStageListFromJSONTyped = PaginatedExtAuthStageListFromJSONTyped;
function PaginatedExtAuthStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ExtAuthStage_1.ExtAuthStageToJSON)),
    };
}
exports.PaginatedExtAuthStageListToJSON = PaginatedExtAuthStageListToJSON;
