// @goauthentik/api/dist/models/PaginatedEndpointList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedEndpointListToJSON = exports.PaginatedEndpointListFromJSONTyped = exports.PaginatedEndpointListFromJSON = exports.instanceOfPaginatedEndpointList = void 0;
const Endpoint_1 = require("./Endpoint");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedEndpointList interface.
 */
function instanceOfPaginatedEndpointList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedEndpointList = instanceOfPaginatedEndpointList;
function PaginatedEndpointListFromJSON(json) {
    return PaginatedEndpointListFromJSONTyped(json, false);
}
exports.PaginatedEndpointListFromJSON = PaginatedEndpointListFromJSON;
function PaginatedEndpointListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Endpoint_1.EndpointFromJSON)),
    };
}
exports.PaginatedEndpointListFromJSONTyped = PaginatedEndpointListFromJSONTyped;
function PaginatedEndpointListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Endpoint_1.EndpointToJSON)),
    };
}
exports.PaginatedEndpointListToJSON = PaginatedEndpointListToJSON;
