// @goauthentik/api/dist/models/PaginatedPasswordStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPasswordStageListToJSON = exports.PaginatedPasswordStageListFromJSONTyped = exports.PaginatedPasswordStageListFromJSON = exports.instanceOfPaginatedPasswordStageList = void 0;
const Pagination_1 = require("./Pagination");
const PasswordStage_1 = require("./PasswordStage");
/**
 * Check if a given object implements the PaginatedPasswordStageList interface.
 */
function instanceOfPaginatedPasswordStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPasswordStageList = instanceOfPaginatedPasswordStageList;
function PaginatedPasswordStageListFromJSON(json) {
    return PaginatedPasswordStageListFromJSONTyped(json, false);
}
exports.PaginatedPasswordStageListFromJSON = PaginatedPasswordStageListFromJSON;
function PaginatedPasswordStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PasswordStage_1.PasswordStageFromJSON)),
    };
}
exports.PaginatedPasswordStageListFromJSONTyped = PaginatedPasswordStageListFromJSONTyped;
function PaginatedPasswordStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PasswordStage_1.PasswordStageToJSON)),
    };
}
exports.PaginatedPasswordStageListToJSON = PaginatedPasswordStageListToJSON;
