// @goauthentik/api/dist/models/PaginatedAuthenticatorStaticStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticatorStaticStageListToJSON = exports.PaginatedAuthenticatorStaticStageListFromJSONTyped = exports.PaginatedAuthenticatorStaticStageListFromJSON = exports.instanceOfPaginatedAuthenticatorStaticStageList = void 0;
const AuthenticatorStaticStage_1 = require("./AuthenticatorStaticStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticatorStaticStageList interface.
 */
function instanceOfPaginatedAuthenticatorStaticStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticatorStaticStageList = instanceOfPaginatedAuthenticatorStaticStageList;
function PaginatedAuthenticatorStaticStageListFromJSON(json) {
    return PaginatedAuthenticatorStaticStageListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticatorStaticStageListFromJSON = PaginatedAuthenticatorStaticStageListFromJSON;
function PaginatedAuthenticatorStaticStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticatorStaticStage_1.AuthenticatorStaticStageFromJSON)),
    };
}
exports.PaginatedAuthenticatorStaticStageListFromJSONTyped = PaginatedAuthenticatorStaticStageListFromJSONTyped;
function PaginatedAuthenticatorStaticStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticatorStaticStage_1.AuthenticatorStaticStageToJSON)),
    };
}
exports.PaginatedAuthenticatorStaticStageListToJSON = PaginatedAuthenticatorStaticStageListToJSON;
