// @goauthentik/api/dist/models/PaginatedReputationList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedReputationListToJSON = exports.PaginatedReputationListFromJSONTyped = exports.PaginatedReputationListFromJSON = exports.instanceOfPaginatedReputationList = void 0;
const Pagination_1 = require("./Pagination");
const Reputation_1 = require("./Reputation");
/**
 * Check if a given object implements the PaginatedReputationList interface.
 */
function instanceOfPaginatedReputationList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedReputationList = instanceOfPaginatedReputationList;
function PaginatedReputationListFromJSON(json) {
    return PaginatedReputationListFromJSONTyped(json, false);
}
exports.PaginatedReputationListFromJSON = PaginatedReputationListFromJSON;
function PaginatedReputationListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Reputation_1.ReputationFromJSON)),
    };
}
exports.PaginatedReputationListFromJSONTyped = PaginatedReputationListFromJSONTyped;
function PaginatedReputationListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Reputation_1.ReputationToJSON)),
    };
}
exports.PaginatedReputationListToJSON = PaginatedReputationListToJSON;
