// @goauthentik/api/dist/models/PaginatedEmailStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedEmailStageListToJSON = exports.PaginatedEmailStageListFromJSONTyped = exports.PaginatedEmailStageListFromJSON = exports.instanceOfPaginatedEmailStageList = void 0;
const EmailStage_1 = require("./EmailStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedEmailStageList interface.
 */
function instanceOfPaginatedEmailStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedEmailStageList = instanceOfPaginatedEmailStageList;
function PaginatedEmailStageListFromJSON(json) {
    return PaginatedEmailStageListFromJSONTyped(json, false);
}
exports.PaginatedEmailStageListFromJSON = PaginatedEmailStageListFromJSON;
function PaginatedEmailStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(EmailStage_1.EmailStageFromJSON)),
    };
}
exports.PaginatedEmailStageListFromJSONTyped = PaginatedEmailStageListFromJSONTyped;
function PaginatedEmailStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(EmailStage_1.EmailStageToJSON)),
    };
}
exports.PaginatedEmailStageListToJSON = PaginatedEmailStageListToJSON;
