// @goauthentik/api/dist/models/PaginatedGroupList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedGroupListToJSON = exports.PaginatedGroupListFromJSONTyped = exports.PaginatedGroupListFromJSON = exports.instanceOfPaginatedGroupList = void 0;
const Group_1 = require("./Group");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedGroupList interface.
 */
function instanceOfPaginatedGroupList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedGroupList = instanceOfPaginatedGroupList;
function PaginatedGroupListFromJSON(json) {
    return PaginatedGroupListFromJSONTyped(json, false);
}
exports.PaginatedGroupListFromJSON = PaginatedGroupListFromJSON;
function PaginatedGroupListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Group_1.GroupFromJSON)),
    };
}
exports.PaginatedGroupListFromJSONTyped = PaginatedGroupListFromJSONTyped;
function PaginatedGroupListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Group_1.GroupToJSON)),
    };
}
exports.PaginatedGroupListToJSON = PaginatedGroupListToJSON;
