// @goauthentik/api/dist/models/PaginatedCaptchaStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedCaptchaStageListToJSON = exports.PaginatedCaptchaStageListFromJSONTyped = exports.PaginatedCaptchaStageListFromJSON = exports.instanceOfPaginatedCaptchaStageList = void 0;
const CaptchaStage_1 = require("./CaptchaStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedCaptchaStageList interface.
 */
function instanceOfPaginatedCaptchaStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedCaptchaStageList = instanceOfPaginatedCaptchaStageList;
function PaginatedCaptchaStageListFromJSON(json) {
    return PaginatedCaptchaStageListFromJSONTyped(json, false);
}
exports.PaginatedCaptchaStageListFromJSON = PaginatedCaptchaStageListFromJSON;
function PaginatedCaptchaStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(CaptchaStage_1.CaptchaStageFromJSON)),
    };
}
exports.PaginatedCaptchaStageListFromJSONTyped = PaginatedCaptchaStageListFromJSONTyped;
function PaginatedCaptchaStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(CaptchaStage_1.CaptchaStageToJSON)),
    };
}
exports.PaginatedCaptchaStageListToJSON = PaginatedCaptchaStageListToJSON;
