// @goauthentik/api/dist/models/PaginatedAuthenticatedSessionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticatedSessionListToJSON = exports.PaginatedAuthenticatedSessionListFromJSONTyped = exports.PaginatedAuthenticatedSessionListFromJSON = exports.instanceOfPaginatedAuthenticatedSessionList = void 0;
const AuthenticatedSession_1 = require("./AuthenticatedSession");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticatedSessionList interface.
 */
function instanceOfPaginatedAuthenticatedSessionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticatedSessionList = instanceOfPaginatedAuthenticatedSessionList;
function PaginatedAuthenticatedSessionListFromJSON(json) {
    return PaginatedAuthenticatedSessionListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticatedSessionListFromJSON = PaginatedAuthenticatedSessionListFromJSON;
function PaginatedAuthenticatedSessionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticatedSession_1.AuthenticatedSessionFromJSON)),
    };
}
exports.PaginatedAuthenticatedSessionListFromJSONTyped = PaginatedAuthenticatedSessionListFromJSONTyped;
function PaginatedAuthenticatedSessionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticatedSession_1.AuthenticatedSessionToJSON)),
    };
}
exports.PaginatedAuthenticatedSessionListToJSON = PaginatedAuthenticatedSessionListToJSON;
