// @goauthentik/api/dist/models/PaginatedConsentStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedConsentStageListToJSON = exports.PaginatedConsentStageListFromJSONTyped = exports.PaginatedConsentStageListFromJSON = exports.instanceOfPaginatedConsentStageList = void 0;
const ConsentStage_1 = require("./ConsentStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedConsentStageList interface.
 */
function instanceOfPaginatedConsentStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedConsentStageList = instanceOfPaginatedConsentStageList;
function PaginatedConsentStageListFromJSON(json) {
    return PaginatedConsentStageListFromJSONTyped(json, false);
}
exports.PaginatedConsentStageListFromJSON = PaginatedConsentStageListFromJSON;
function PaginatedConsentStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ConsentStage_1.ConsentStageFromJSON)),
    };
}
exports.PaginatedConsentStageListFromJSONTyped = PaginatedConsentStageListFromJSONTyped;
function PaginatedConsentStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ConsentStage_1.ConsentStageToJSON)),
    };
}
exports.PaginatedConsentStageListToJSON = PaginatedConsentStageListToJSON;
