// @goauthentik/api/dist/models/PaginatedUserWriteStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserWriteStageListToJSON = exports.PaginatedUserWriteStageListFromJSONTyped = exports.PaginatedUserWriteStageListFromJSON = exports.instanceOfPaginatedUserWriteStageList = void 0;
const Pagination_1 = require("./Pagination");
const UserWriteStage_1 = require("./UserWriteStage");
/**
 * Check if a given object implements the PaginatedUserWriteStageList interface.
 */
function instanceOfPaginatedUserWriteStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserWriteStageList = instanceOfPaginatedUserWriteStageList;
function PaginatedUserWriteStageListFromJSON(json) {
    return PaginatedUserWriteStageListFromJSONTyped(json, false);
}
exports.PaginatedUserWriteStageListFromJSON = PaginatedUserWriteStageListFromJSON;
function PaginatedUserWriteStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserWriteStage_1.UserWriteStageFromJSON)),
    };
}
exports.PaginatedUserWriteStageListFromJSONTyped = PaginatedUserWriteStageListFromJSONTyped;
function PaginatedUserWriteStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserWriteStage_1.UserWriteStageToJSON)),
    };
}
exports.PaginatedUserWriteStageListToJSON = PaginatedUserWriteStageListToJSON;
