// @goauthentik/api/dist/models/PaginatedAuthenticatorDuoStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticatorDuoStageListToJSON = exports.PaginatedAuthenticatorDuoStageListFromJSONTyped = exports.PaginatedAuthenticatorDuoStageListFromJSON = exports.instanceOfPaginatedAuthenticatorDuoStageList = void 0;
const AuthenticatorDuoStage_1 = require("./AuthenticatorDuoStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticatorDuoStageList interface.
 */
function instanceOfPaginatedAuthenticatorDuoStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticatorDuoStageList = instanceOfPaginatedAuthenticatorDuoStageList;
function PaginatedAuthenticatorDuoStageListFromJSON(json) {
    return PaginatedAuthenticatorDuoStageListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticatorDuoStageListFromJSON = PaginatedAuthenticatorDuoStageListFromJSON;
function PaginatedAuthenticatorDuoStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticatorDuoStage_1.AuthenticatorDuoStageFromJSON)),
    };
}
exports.PaginatedAuthenticatorDuoStageListFromJSONTyped = PaginatedAuthenticatorDuoStageListFromJSONTyped;
function PaginatedAuthenticatorDuoStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticatorDuoStage_1.AuthenticatorDuoStageToJSON)),
    };
}
exports.PaginatedAuthenticatorDuoStageListToJSON = PaginatedAuthenticatorDuoStageListToJSON;
