// @goauthentik/api/dist/models/PaginatedUserConsentList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserConsentListToJSON = exports.PaginatedUserConsentListFromJSONTyped = exports.PaginatedUserConsentListFromJSON = exports.instanceOfPaginatedUserConsentList = void 0;
const Pagination_1 = require("./Pagination");
const UserConsent_1 = require("./UserConsent");
/**
 * Check if a given object implements the PaginatedUserConsentList interface.
 */
function instanceOfPaginatedUserConsentList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserConsentList = instanceOfPaginatedUserConsentList;
function PaginatedUserConsentListFromJSON(json) {
    return PaginatedUserConsentListFromJSONTyped(json, false);
}
exports.PaginatedUserConsentListFromJSON = PaginatedUserConsentListFromJSON;
function PaginatedUserConsentListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserConsent_1.UserConsentFromJSON)),
    };
}
exports.PaginatedUserConsentListFromJSONTyped = PaginatedUserConsentListFromJSONTyped;
function PaginatedUserConsentListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserConsent_1.UserConsentToJSON)),
    };
}
exports.PaginatedUserConsentListToJSON = PaginatedUserConsentListToJSON;
