// @goauthentik/api/dist/models/PaginatedUserLoginStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserLoginStageListToJSON = exports.PaginatedUserLoginStageListFromJSONTyped = exports.PaginatedUserLoginStageListFromJSON = exports.instanceOfPaginatedUserLoginStageList = void 0;
const Pagination_1 = require("./Pagination");
const UserLoginStage_1 = require("./UserLoginStage");
/**
 * Check if a given object implements the PaginatedUserLoginStageList interface.
 */
function instanceOfPaginatedUserLoginStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserLoginStageList = instanceOfPaginatedUserLoginStageList;
function PaginatedUserLoginStageListFromJSON(json) {
    return PaginatedUserLoginStageListFromJSONTyped(json, false);
}
exports.PaginatedUserLoginStageListFromJSON = PaginatedUserLoginStageListFromJSON;
function PaginatedUserLoginStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserLoginStage_1.UserLoginStageFromJSON)),
    };
}
exports.PaginatedUserLoginStageListFromJSONTyped = PaginatedUserLoginStageListFromJSONTyped;
function PaginatedUserLoginStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserLoginStage_1.UserLoginStageToJSON)),
    };
}
exports.PaginatedUserLoginStageListToJSON = PaginatedUserLoginStageListToJSON;
