// @goauthentik/api/dist/models/PaginatedTenantList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedTenantListToJSON = exports.PaginatedTenantListFromJSONTyped = exports.PaginatedTenantListFromJSON = exports.instanceOfPaginatedTenantList = void 0;
const Pagination_1 = require("./Pagination");
const Tenant_1 = require("./Tenant");
/**
 * Check if a given object implements the PaginatedTenantList interface.
 */
function instanceOfPaginatedTenantList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedTenantList = instanceOfPaginatedTenantList;
function PaginatedTenantListFromJSON(json) {
    return PaginatedTenantListFromJSONTyped(json, false);
}
exports.PaginatedTenantListFromJSON = PaginatedTenantListFromJSON;
function PaginatedTenantListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Tenant_1.TenantFromJSON)),
    };
}
exports.PaginatedTenantListFromJSONTyped = PaginatedTenantListFromJSONTyped;
function PaginatedTenantListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Tenant_1.TenantToJSON)),
    };
}
exports.PaginatedTenantListToJSON = PaginatedTenantListToJSON;
