// @goauthentik/api/dist/models/PaginatedPlexSourceConnectionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPlexSourceConnectionListToJSON = exports.PaginatedPlexSourceConnectionListFromJSONTyped = exports.PaginatedPlexSourceConnectionListFromJSON = exports.instanceOfPaginatedPlexSourceConnectionList = void 0;
const Pagination_1 = require("./Pagination");
const PlexSourceConnection_1 = require("./PlexSourceConnection");
/**
 * Check if a given object implements the PaginatedPlexSourceConnectionList interface.
 */
function instanceOfPaginatedPlexSourceConnectionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPlexSourceConnectionList = instanceOfPaginatedPlexSourceConnectionList;
function PaginatedPlexSourceConnectionListFromJSON(json) {
    return PaginatedPlexSourceConnectionListFromJSONTyped(json, false);
}
exports.PaginatedPlexSourceConnectionListFromJSON = PaginatedPlexSourceConnectionListFromJSON;
function PaginatedPlexSourceConnectionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PlexSourceConnection_1.PlexSourceConnectionFromJSON)),
    };
}
exports.PaginatedPlexSourceConnectionListFromJSONTyped = PaginatedPlexSourceConnectionListFromJSONTyped;
function PaginatedPlexSourceConnectionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PlexSourceConnection_1.PlexSourceConnectionToJSON)),
    };
}
exports.PaginatedPlexSourceConnectionListToJSON = PaginatedPlexSourceConnectionListToJSON;
