// @goauthentik/api/dist/models/PaginatedUserOAuthSourceConnectionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserOAuthSourceConnectionListToJSON = exports.PaginatedUserOAuthSourceConnectionListFromJSONTyped = exports.PaginatedUserOAuthSourceConnectionListFromJSON = exports.instanceOfPaginatedUserOAuthSourceConnectionList = void 0;
const Pagination_1 = require("./Pagination");
const UserOAuthSourceConnection_1 = require("./UserOAuthSourceConnection");
/**
 * Check if a given object implements the PaginatedUserOAuthSourceConnectionList interface.
 */
function instanceOfPaginatedUserOAuthSourceConnectionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserOAuthSourceConnectionList = instanceOfPaginatedUserOAuthSourceConnectionList;
function PaginatedUserOAuthSourceConnectionListFromJSON(json) {
    return PaginatedUserOAuthSourceConnectionListFromJSONTyped(json, false);
}
exports.PaginatedUserOAuthSourceConnectionListFromJSON = PaginatedUserOAuthSourceConnectionListFromJSON;
function PaginatedUserOAuthSourceConnectionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserOAuthSourceConnection_1.UserOAuthSourceConnectionFromJSON)),
    };
}
exports.PaginatedUserOAuthSourceConnectionListFromJSONTyped = PaginatedUserOAuthSourceConnectionListFromJSONTyped;
function PaginatedUserOAuthSourceConnectionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserOAuthSourceConnection_1.UserOAuthSourceConnectionToJSON)),
    };
}
exports.PaginatedUserOAuthSourceConnectionListToJSON = PaginatedUserOAuthSourceConnectionListToJSON;
