// @goauthentik/api/dist/models/PaginatedReputationPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedReputationPolicyListToJSON = exports.PaginatedReputationPolicyListFromJSONTyped = exports.PaginatedReputationPolicyListFromJSON = exports.instanceOfPaginatedReputationPolicyList = void 0;
const Pagination_1 = require("./Pagination");
const ReputationPolicy_1 = require("./ReputationPolicy");
/**
 * Check if a given object implements the PaginatedReputationPolicyList interface.
 */
function instanceOfPaginatedReputationPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedReputationPolicyList = instanceOfPaginatedReputationPolicyList;
function PaginatedReputationPolicyListFromJSON(json) {
    return PaginatedReputationPolicyListFromJSONTyped(json, false);
}
exports.PaginatedReputationPolicyListFromJSON = PaginatedReputationPolicyListFromJSON;
function PaginatedReputationPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ReputationPolicy_1.ReputationPolicyFromJSON)),
    };
}
exports.PaginatedReputationPolicyListFromJSONTyped = PaginatedReputationPolicyListFromJSONTyped;
function PaginatedReputationPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ReputationPolicy_1.ReputationPolicyToJSON)),
    };
}
exports.PaginatedReputationPolicyListToJSON = PaginatedReputationPolicyListToJSON;
