// @goauthentik/api/dist/models/PaginatedPromptStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPromptStageListToJSON = exports.PaginatedPromptStageListFromJSONTyped = exports.PaginatedPromptStageListFromJSON = exports.instanceOfPaginatedPromptStageList = void 0;
const Pagination_1 = require("./Pagination");
const PromptStage_1 = require("./PromptStage");
/**
 * Check if a given object implements the PaginatedPromptStageList interface.
 */
function instanceOfPaginatedPromptStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPromptStageList = instanceOfPaginatedPromptStageList;
function PaginatedPromptStageListFromJSON(json) {
    return PaginatedPromptStageListFromJSONTyped(json, false);
}
exports.PaginatedPromptStageListFromJSON = PaginatedPromptStageListFromJSON;
function PaginatedPromptStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PromptStage_1.PromptStageFromJSON)),
    };
}
exports.PaginatedPromptStageListFromJSONTyped = PaginatedPromptStageListFromJSONTyped;
function PaginatedPromptStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PromptStage_1.PromptStageToJSON)),
    };
}
exports.PaginatedPromptStageListToJSON = PaginatedPromptStageListToJSON;
