// @goauthentik/api/dist/models/PaginatedTokenList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedTokenListToJSON = exports.PaginatedTokenListFromJSONTyped = exports.PaginatedTokenListFromJSON = exports.instanceOfPaginatedTokenList = void 0;
const Pagination_1 = require("./Pagination");
const Token_1 = require("./Token");
/**
 * Check if a given object implements the PaginatedTokenList interface.
 */
function instanceOfPaginatedTokenList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedTokenList = instanceOfPaginatedTokenList;
function PaginatedTokenListFromJSON(json) {
    return PaginatedTokenListFromJSONTyped(json, false);
}
exports.PaginatedTokenListFromJSON = PaginatedTokenListFromJSON;
function PaginatedTokenListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Token_1.TokenFromJSON)),
    };
}
exports.PaginatedTokenListFromJSONTyped = PaginatedTokenListFromJSONTyped;
function PaginatedTokenListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Token_1.TokenToJSON)),
    };
}
exports.PaginatedTokenListToJSON = PaginatedTokenListToJSON;
