// @goauthentik/api/dist/models/PaginatedLDAPPropertyMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedLDAPPropertyMappingListToJSON = exports.PaginatedLDAPPropertyMappingListFromJSONTyped = exports.PaginatedLDAPPropertyMappingListFromJSON = exports.instanceOfPaginatedLDAPPropertyMappingList = void 0;
const LDAPPropertyMapping_1 = require("./LDAPPropertyMapping");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedLDAPPropertyMappingList interface.
 */
function instanceOfPaginatedLDAPPropertyMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedLDAPPropertyMappingList = instanceOfPaginatedLDAPPropertyMappingList;
function PaginatedLDAPPropertyMappingListFromJSON(json) {
    return PaginatedLDAPPropertyMappingListFromJSONTyped(json, false);
}
exports.PaginatedLDAPPropertyMappingListFromJSON = PaginatedLDAPPropertyMappingListFromJSON;
function PaginatedLDAPPropertyMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(LDAPPropertyMapping_1.LDAPPropertyMappingFromJSON)),
    };
}
exports.PaginatedLDAPPropertyMappingListFromJSONTyped = PaginatedLDAPPropertyMappingListFromJSONTyped;
function PaginatedLDAPPropertyMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(LDAPPropertyMapping_1.LDAPPropertyMappingToJSON)),
    };
}
exports.PaginatedLDAPPropertyMappingListToJSON = PaginatedLDAPPropertyMappingListToJSON;
