// @goauthentik/api/dist/models/PaginatedNotificationTransportList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedNotificationTransportListToJSON = exports.PaginatedNotificationTransportListFromJSONTyped = exports.PaginatedNotificationTransportListFromJSON = exports.instanceOfPaginatedNotificationTransportList = void 0;
const NotificationTransport_1 = require("./NotificationTransport");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedNotificationTransportList interface.
 */
function instanceOfPaginatedNotificationTransportList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedNotificationTransportList = instanceOfPaginatedNotificationTransportList;
function PaginatedNotificationTransportListFromJSON(json) {
    return PaginatedNotificationTransportListFromJSONTyped(json, false);
}
exports.PaginatedNotificationTransportListFromJSON = PaginatedNotificationTransportListFromJSON;
function PaginatedNotificationTransportListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(NotificationTransport_1.NotificationTransportFromJSON)),
    };
}
exports.PaginatedNotificationTransportListFromJSONTyped = PaginatedNotificationTransportListFromJSONTyped;
function PaginatedNotificationTransportListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(NotificationTransport_1.NotificationTransportToJSON)),
    };
}
exports.PaginatedNotificationTransportListToJSON = PaginatedNotificationTransportListToJSON;
