// @goauthentik/api/dist/models/PaginatedPromptList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPromptListToJSON = exports.PaginatedPromptListFromJSONTyped = exports.PaginatedPromptListFromJSON = exports.instanceOfPaginatedPromptList = void 0;
const Pagination_1 = require("./Pagination");
const Prompt_1 = require("./Prompt");
/**
 * Check if a given object implements the PaginatedPromptList interface.
 */
function instanceOfPaginatedPromptList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPromptList = instanceOfPaginatedPromptList;
function PaginatedPromptListFromJSON(json) {
    return PaginatedPromptListFromJSONTyped(json, false);
}
exports.PaginatedPromptListFromJSON = PaginatedPromptListFromJSON;
function PaginatedPromptListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Prompt_1.PromptFromJSON)),
    };
}
exports.PaginatedPromptListFromJSONTyped = PaginatedPromptListFromJSONTyped;
function PaginatedPromptListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Prompt_1.PromptToJSON)),
    };
}
exports.PaginatedPromptListToJSON = PaginatedPromptListToJSON;
