// @goauthentik/api/dist/models/PaginatedInvitationList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedInvitationListToJSON = exports.PaginatedInvitationListFromJSONTyped = exports.PaginatedInvitationListFromJSON = exports.instanceOfPaginatedInvitationList = void 0;
const Invitation_1 = require("./Invitation");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedInvitationList interface.
 */
function instanceOfPaginatedInvitationList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedInvitationList = instanceOfPaginatedInvitationList;
function PaginatedInvitationListFromJSON(json) {
    return PaginatedInvitationListFromJSONTyped(json, false);
}
exports.PaginatedInvitationListFromJSON = PaginatedInvitationListFromJSON;
function PaginatedInvitationListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Invitation_1.InvitationFromJSON)),
    };
}
exports.PaginatedInvitationListFromJSONTyped = PaginatedInvitationListFromJSONTyped;
function PaginatedInvitationListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Invitation_1.InvitationToJSON)),
    };
}
exports.PaginatedInvitationListToJSON = PaginatedInvitationListToJSON;
