// @goauthentik/api/dist/models/PaginatedCertificateKeyPairList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedCertificateKeyPairListToJSON = exports.PaginatedCertificateKeyPairListFromJSONTyped = exports.PaginatedCertificateKeyPairListFromJSON = exports.instanceOfPaginatedCertificateKeyPairList = void 0;
const CertificateKeyPair_1 = require("./CertificateKeyPair");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedCertificateKeyPairList interface.
 */
function instanceOfPaginatedCertificateKeyPairList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedCertificateKeyPairList = instanceOfPaginatedCertificateKeyPairList;
function PaginatedCertificateKeyPairListFromJSON(json) {
    return PaginatedCertificateKeyPairListFromJSONTyped(json, false);
}
exports.PaginatedCertificateKeyPairListFromJSON = PaginatedCertificateKeyPairListFromJSON;
function PaginatedCertificateKeyPairListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(CertificateKeyPair_1.CertificateKeyPairFromJSON)),
    };
}
exports.PaginatedCertificateKeyPairListFromJSONTyped = PaginatedCertificateKeyPairListFromJSONTyped;
function PaginatedCertificateKeyPairListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(CertificateKeyPair_1.CertificateKeyPairToJSON)),
    };
}
exports.PaginatedCertificateKeyPairListToJSON = PaginatedCertificateKeyPairListToJSON;
