// @goauthentik/api/dist/models/PaginatedSystemTaskList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSystemTaskListToJSON = exports.PaginatedSystemTaskListFromJSONTyped = exports.PaginatedSystemTaskListFromJSON = exports.instanceOfPaginatedSystemTaskList = void 0;
const Pagination_1 = require("./Pagination");
const SystemTask_1 = require("./SystemTask");
/**
 * Check if a given object implements the PaginatedSystemTaskList interface.
 */
function instanceOfPaginatedSystemTaskList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSystemTaskList = instanceOfPaginatedSystemTaskList;
function PaginatedSystemTaskListFromJSON(json) {
    return PaginatedSystemTaskListFromJSONTyped(json, false);
}
exports.PaginatedSystemTaskListFromJSON = PaginatedSystemTaskListFromJSON;
function PaginatedSystemTaskListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SystemTask_1.SystemTaskFromJSON)),
    };
}
exports.PaginatedSystemTaskListFromJSONTyped = PaginatedSystemTaskListFromJSONTyped;
function PaginatedSystemTaskListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SystemTask_1.SystemTaskToJSON)),
    };
}
exports.PaginatedSystemTaskListToJSON = PaginatedSystemTaskListToJSON;
