// @goauthentik/api/dist/models/PaginatedNotificationRuleList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedNotificationRuleListToJSON = exports.PaginatedNotificationRuleListFromJSONTyped = exports.PaginatedNotificationRuleListFromJSON = exports.instanceOfPaginatedNotificationRuleList = void 0;
const NotificationRule_1 = require("./NotificationRule");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedNotificationRuleList interface.
 */
function instanceOfPaginatedNotificationRuleList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedNotificationRuleList = instanceOfPaginatedNotificationRuleList;
function PaginatedNotificationRuleListFromJSON(json) {
    return PaginatedNotificationRuleListFromJSONTyped(json, false);
}
exports.PaginatedNotificationRuleListFromJSON = PaginatedNotificationRuleListFromJSON;
function PaginatedNotificationRuleListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(NotificationRule_1.NotificationRuleFromJSON)),
    };
}
exports.PaginatedNotificationRuleListFromJSONTyped = PaginatedNotificationRuleListFromJSONTyped;
function PaginatedNotificationRuleListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(NotificationRule_1.NotificationRuleToJSON)),
    };
}
exports.PaginatedNotificationRuleListToJSON = PaginatedNotificationRuleListToJSON;
