// @goauthentik/api/dist/models/PaginatedDockerServiceConnectionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDockerServiceConnectionListToJSON = exports.PaginatedDockerServiceConnectionListFromJSONTyped = exports.PaginatedDockerServiceConnectionListFromJSON = exports.instanceOfPaginatedDockerServiceConnectionList = void 0;
const DockerServiceConnection_1 = require("./DockerServiceConnection");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDockerServiceConnectionList interface.
 */
function instanceOfPaginatedDockerServiceConnectionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDockerServiceConnectionList = instanceOfPaginatedDockerServiceConnectionList;
function PaginatedDockerServiceConnectionListFromJSON(json) {
    return PaginatedDockerServiceConnectionListFromJSONTyped(json, false);
}
exports.PaginatedDockerServiceConnectionListFromJSON = PaginatedDockerServiceConnectionListFromJSON;
function PaginatedDockerServiceConnectionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(DockerServiceConnection_1.DockerServiceConnectionFromJSON)),
    };
}
exports.PaginatedDockerServiceConnectionListFromJSONTyped = PaginatedDockerServiceConnectionListFromJSONTyped;
function PaginatedDockerServiceConnectionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(DockerServiceConnection_1.DockerServiceConnectionToJSON)),
    };
}
exports.PaginatedDockerServiceConnectionListToJSON = PaginatedDockerServiceConnectionListToJSON;
