// @goauthentik/api/dist/models/PaginatedUserDeleteStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserDeleteStageListToJSON = exports.PaginatedUserDeleteStageListFromJSONTyped = exports.PaginatedUserDeleteStageListFromJSON = exports.instanceOfPaginatedUserDeleteStageList = void 0;
const Pagination_1 = require("./Pagination");
const UserDeleteStage_1 = require("./UserDeleteStage");
/**
 * Check if a given object implements the PaginatedUserDeleteStageList interface.
 */
function instanceOfPaginatedUserDeleteStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserDeleteStageList = instanceOfPaginatedUserDeleteStageList;
function PaginatedUserDeleteStageListFromJSON(json) {
    return PaginatedUserDeleteStageListFromJSONTyped(json, false);
}
exports.PaginatedUserDeleteStageListFromJSON = PaginatedUserDeleteStageListFromJSON;
function PaginatedUserDeleteStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(UserDeleteStage_1.UserDeleteStageFromJSON)),
    };
}
exports.PaginatedUserDeleteStageListFromJSONTyped = PaginatedUserDeleteStageListFromJSONTyped;
function PaginatedUserDeleteStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(UserDeleteStage_1.UserDeleteStageToJSON)),
    };
}
exports.PaginatedUserDeleteStageListToJSON = PaginatedUserDeleteStageListToJSON;
