// @goauthentik/api/dist/models/PaginatedAuthenticateWebAuthnStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticateWebAuthnStageListToJSON = exports.PaginatedAuthenticateWebAuthnStageListFromJSONTyped = exports.PaginatedAuthenticateWebAuthnStageListFromJSON = exports.instanceOfPaginatedAuthenticateWebAuthnStageList = void 0;
const AuthenticateWebAuthnStage_1 = require("./AuthenticateWebAuthnStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticateWebAuthnStageList interface.
 */
function instanceOfPaginatedAuthenticateWebAuthnStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticateWebAuthnStageList = instanceOfPaginatedAuthenticateWebAuthnStageList;
function PaginatedAuthenticateWebAuthnStageListFromJSON(json) {
    return PaginatedAuthenticateWebAuthnStageListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticateWebAuthnStageListFromJSON = PaginatedAuthenticateWebAuthnStageListFromJSON;
function PaginatedAuthenticateWebAuthnStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticateWebAuthnStage_1.AuthenticateWebAuthnStageFromJSON)),
    };
}
exports.PaginatedAuthenticateWebAuthnStageListFromJSONTyped = PaginatedAuthenticateWebAuthnStageListFromJSONTyped;
function PaginatedAuthenticateWebAuthnStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticateWebAuthnStage_1.AuthenticateWebAuthnStageToJSON)),
    };
}
exports.PaginatedAuthenticateWebAuthnStageListToJSON = PaginatedAuthenticateWebAuthnStageListToJSON;
