// @goauthentik/api/dist/models/PaginatedLDAPProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedLDAPProviderListToJSON = exports.PaginatedLDAPProviderListFromJSONTyped = exports.PaginatedLDAPProviderListFromJSON = exports.instanceOfPaginatedLDAPProviderList = void 0;
const LDAPProvider_1 = require("./LDAPProvider");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedLDAPProviderList interface.
 */
function instanceOfPaginatedLDAPProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedLDAPProviderList = instanceOfPaginatedLDAPProviderList;
function PaginatedLDAPProviderListFromJSON(json) {
    return PaginatedLDAPProviderListFromJSONTyped(json, false);
}
exports.PaginatedLDAPProviderListFromJSON = PaginatedLDAPProviderListFromJSON;
function PaginatedLDAPProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(LDAPProvider_1.LDAPProviderFromJSON)),
    };
}
exports.PaginatedLDAPProviderListFromJSONTyped = PaginatedLDAPProviderListFromJSONTyped;
function PaginatedLDAPProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(LDAPProvider_1.LDAPProviderToJSON)),
    };
}
exports.PaginatedLDAPProviderListToJSON = PaginatedLDAPProviderListToJSON;
