// @goauthentik/api/dist/models/PaginatedDummyStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDummyStageListToJSON = exports.PaginatedDummyStageListFromJSONTyped = exports.PaginatedDummyStageListFromJSON = exports.instanceOfPaginatedDummyStageList = void 0;
const DummyStage_1 = require("./DummyStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDummyStageList interface.
 */
function instanceOfPaginatedDummyStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDummyStageList = instanceOfPaginatedDummyStageList;
function PaginatedDummyStageListFromJSON(json) {
    return PaginatedDummyStageListFromJSONTyped(json, false);
}
exports.PaginatedDummyStageListFromJSON = PaginatedDummyStageListFromJSON;
function PaginatedDummyStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(DummyStage_1.DummyStageFromJSON)),
    };
}
exports.PaginatedDummyStageListFromJSONTyped = PaginatedDummyStageListFromJSONTyped;
function PaginatedDummyStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(DummyStage_1.DummyStageToJSON)),
    };
}
exports.PaginatedDummyStageListToJSON = PaginatedDummyStageListToJSON;
