// @goauthentik/api/dist/models/PaginatedPropertyMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPropertyMappingListToJSON = exports.PaginatedPropertyMappingListFromJSONTyped = exports.PaginatedPropertyMappingListFromJSON = exports.instanceOfPaginatedPropertyMappingList = void 0;
const Pagination_1 = require("./Pagination");
const PropertyMapping_1 = require("./PropertyMapping");
/**
 * Check if a given object implements the PaginatedPropertyMappingList interface.
 */
function instanceOfPaginatedPropertyMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPropertyMappingList = instanceOfPaginatedPropertyMappingList;
function PaginatedPropertyMappingListFromJSON(json) {
    return PaginatedPropertyMappingListFromJSONTyped(json, false);
}
exports.PaginatedPropertyMappingListFromJSON = PaginatedPropertyMappingListFromJSON;
function PaginatedPropertyMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PropertyMapping_1.PropertyMappingFromJSON)),
    };
}
exports.PaginatedPropertyMappingListFromJSONTyped = PaginatedPropertyMappingListFromJSONTyped;
function PaginatedPropertyMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PropertyMapping_1.PropertyMappingToJSON)),
    };
}
exports.PaginatedPropertyMappingListToJSON = PaginatedPropertyMappingListToJSON;
