// @goauthentik/api/dist/models/PaginatedNotificationWebhookMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedNotificationWebhookMappingListToJSON = exports.PaginatedNotificationWebhookMappingListFromJSONTyped = exports.PaginatedNotificationWebhookMappingListFromJSON = exports.instanceOfPaginatedNotificationWebhookMappingList = void 0;
const NotificationWebhookMapping_1 = require("./NotificationWebhookMapping");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedNotificationWebhookMappingList interface.
 */
function instanceOfPaginatedNotificationWebhookMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedNotificationWebhookMappingList = instanceOfPaginatedNotificationWebhookMappingList;
function PaginatedNotificationWebhookMappingListFromJSON(json) {
    return PaginatedNotificationWebhookMappingListFromJSONTyped(json, false);
}
exports.PaginatedNotificationWebhookMappingListFromJSON = PaginatedNotificationWebhookMappingListFromJSON;
function PaginatedNotificationWebhookMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(NotificationWebhookMapping_1.NotificationWebhookMappingFromJSON)),
    };
}
exports.PaginatedNotificationWebhookMappingListFromJSONTyped = PaginatedNotificationWebhookMappingListFromJSONTyped;
function PaginatedNotificationWebhookMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(NotificationWebhookMapping_1.NotificationWebhookMappingToJSON)),
    };
}
exports.PaginatedNotificationWebhookMappingListToJSON = PaginatedNotificationWebhookMappingListToJSON;
