// @goauthentik/api/dist/models/PaginatedTOTPDeviceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedTOTPDeviceListToJSON = exports.PaginatedTOTPDeviceListFromJSONTyped = exports.PaginatedTOTPDeviceListFromJSON = exports.instanceOfPaginatedTOTPDeviceList = void 0;
const Pagination_1 = require("./Pagination");
const TOTPDevice_1 = require("./TOTPDevice");
/**
 * Check if a given object implements the PaginatedTOTPDeviceList interface.
 */
function instanceOfPaginatedTOTPDeviceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedTOTPDeviceList = instanceOfPaginatedTOTPDeviceList;
function PaginatedTOTPDeviceListFromJSON(json) {
    return PaginatedTOTPDeviceListFromJSONTyped(json, false);
}
exports.PaginatedTOTPDeviceListFromJSON = PaginatedTOTPDeviceListFromJSON;
function PaginatedTOTPDeviceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(TOTPDevice_1.TOTPDeviceFromJSON)),
    };
}
exports.PaginatedTOTPDeviceListFromJSONTyped = PaginatedTOTPDeviceListFromJSONTyped;
function PaginatedTOTPDeviceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(TOTPDevice_1.TOTPDeviceToJSON)),
    };
}
exports.PaginatedTOTPDeviceListToJSON = PaginatedTOTPDeviceListToJSON;
