// @goauthentik/api/dist/models/PaginatedFlowList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedFlowListToJSON = exports.PaginatedFlowListFromJSONTyped = exports.PaginatedFlowListFromJSON = exports.instanceOfPaginatedFlowList = void 0;
const Flow_1 = require("./Flow");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedFlowList interface.
 */
function instanceOfPaginatedFlowList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedFlowList = instanceOfPaginatedFlowList;
function PaginatedFlowListFromJSON(json) {
    return PaginatedFlowListFromJSONTyped(json, false);
}
exports.PaginatedFlowListFromJSON = PaginatedFlowListFromJSON;
function PaginatedFlowListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Flow_1.FlowFromJSON)),
    };
}
exports.PaginatedFlowListFromJSONTyped = PaginatedFlowListFromJSONTyped;
function PaginatedFlowListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Flow_1.FlowToJSON)),
    };
}
exports.PaginatedFlowListToJSON = PaginatedFlowListToJSON;
