// @goauthentik/api/dist/models/PaginatedNotificationList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedNotificationListToJSON = exports.PaginatedNotificationListFromJSONTyped = exports.PaginatedNotificationListFromJSON = exports.instanceOfPaginatedNotificationList = void 0;
const Notification_1 = require("./Notification");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedNotificationList interface.
 */
function instanceOfPaginatedNotificationList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedNotificationList = instanceOfPaginatedNotificationList;
function PaginatedNotificationListFromJSON(json) {
    return PaginatedNotificationListFromJSONTyped(json, false);
}
exports.PaginatedNotificationListFromJSON = PaginatedNotificationListFromJSON;
function PaginatedNotificationListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Notification_1.NotificationFromJSON)),
    };
}
exports.PaginatedNotificationListFromJSONTyped = PaginatedNotificationListFromJSONTyped;
function PaginatedNotificationListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Notification_1.NotificationToJSON)),
    };
}
exports.PaginatedNotificationListToJSON = PaginatedNotificationListToJSON;
