// @goauthentik/api/dist/models/PaginatedExtraRoleObjectPermissionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedExtraRoleObjectPermissionListToJSON = exports.PaginatedExtraRoleObjectPermissionListFromJSONTyped = exports.PaginatedExtraRoleObjectPermissionListFromJSON = exports.instanceOfPaginatedExtraRoleObjectPermissionList = void 0;
const ExtraRoleObjectPermission_1 = require("./ExtraRoleObjectPermission");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedExtraRoleObjectPermissionList interface.
 */
function instanceOfPaginatedExtraRoleObjectPermissionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedExtraRoleObjectPermissionList = instanceOfPaginatedExtraRoleObjectPermissionList;
function PaginatedExtraRoleObjectPermissionListFromJSON(json) {
    return PaginatedExtraRoleObjectPermissionListFromJSONTyped(json, false);
}
exports.PaginatedExtraRoleObjectPermissionListFromJSON = PaginatedExtraRoleObjectPermissionListFromJSON;
function PaginatedExtraRoleObjectPermissionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ExtraRoleObjectPermission_1.ExtraRoleObjectPermissionFromJSON)),
    };
}
exports.PaginatedExtraRoleObjectPermissionListFromJSONTyped = PaginatedExtraRoleObjectPermissionListFromJSONTyped;
function PaginatedExtraRoleObjectPermissionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ExtraRoleObjectPermission_1.ExtraRoleObjectPermissionToJSON)),
    };
}
exports.PaginatedExtraRoleObjectPermissionListToJSON = PaginatedExtraRoleObjectPermissionListToJSON;
