// @goauthentik/api/dist/models/PaginatedAuthenticatorSMSStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedAuthenticatorSMSStageListToJSON = exports.PaginatedAuthenticatorSMSStageListFromJSONTyped = exports.PaginatedAuthenticatorSMSStageListFromJSON = exports.instanceOfPaginatedAuthenticatorSMSStageList = void 0;
const AuthenticatorSMSStage_1 = require("./AuthenticatorSMSStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedAuthenticatorSMSStageList interface.
 */
function instanceOfPaginatedAuthenticatorSMSStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedAuthenticatorSMSStageList = instanceOfPaginatedAuthenticatorSMSStageList;
function PaginatedAuthenticatorSMSStageListFromJSON(json) {
    return PaginatedAuthenticatorSMSStageListFromJSONTyped(json, false);
}
exports.PaginatedAuthenticatorSMSStageListFromJSON = PaginatedAuthenticatorSMSStageListFromJSON;
function PaginatedAuthenticatorSMSStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(AuthenticatorSMSStage_1.AuthenticatorSMSStageFromJSON)),
    };
}
exports.PaginatedAuthenticatorSMSStageListFromJSONTyped = PaginatedAuthenticatorSMSStageListFromJSONTyped;
function PaginatedAuthenticatorSMSStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(AuthenticatorSMSStage_1.AuthenticatorSMSStageToJSON)),
    };
}
exports.PaginatedAuthenticatorSMSStageListToJSON = PaginatedAuthenticatorSMSStageListToJSON;
