// @goauthentik/api/dist/models/PaginatedDummyPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDummyPolicyListToJSON = exports.PaginatedDummyPolicyListFromJSONTyped = exports.PaginatedDummyPolicyListFromJSON = exports.instanceOfPaginatedDummyPolicyList = void 0;
const DummyPolicy_1 = require("./DummyPolicy");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDummyPolicyList interface.
 */
function instanceOfPaginatedDummyPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDummyPolicyList = instanceOfPaginatedDummyPolicyList;
function PaginatedDummyPolicyListFromJSON(json) {
    return PaginatedDummyPolicyListFromJSONTyped(json, false);
}
exports.PaginatedDummyPolicyListFromJSON = PaginatedDummyPolicyListFromJSON;
function PaginatedDummyPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(DummyPolicy_1.DummyPolicyFromJSON)),
    };
}
exports.PaginatedDummyPolicyListFromJSONTyped = PaginatedDummyPolicyListFromJSONTyped;
function PaginatedDummyPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(DummyPolicy_1.DummyPolicyToJSON)),
    };
}
exports.PaginatedDummyPolicyListToJSON = PaginatedDummyPolicyListToJSON;
