// @goauthentik/api/dist/models/PaginatedRoleList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedRoleListToJSON = exports.PaginatedRoleListFromJSONTyped = exports.PaginatedRoleListFromJSON = exports.instanceOfPaginatedRoleList = void 0;
const Pagination_1 = require("./Pagination");
const Role_1 = require("./Role");
/**
 * Check if a given object implements the PaginatedRoleList interface.
 */
function instanceOfPaginatedRoleList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedRoleList = instanceOfPaginatedRoleList;
function PaginatedRoleListFromJSON(json) {
    return PaginatedRoleListFromJSONTyped(json, false);
}
exports.PaginatedRoleListFromJSON = PaginatedRoleListFromJSON;
function PaginatedRoleListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Role_1.RoleFromJSON)),
    };
}
exports.PaginatedRoleListFromJSONTyped = PaginatedRoleListFromJSONTyped;
function PaginatedRoleListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Role_1.RoleToJSON)),
    };
}
exports.PaginatedRoleListToJSON = PaginatedRoleListToJSON;
