// @goauthentik/api/dist/models/PaginatedEventMatcherPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedEventMatcherPolicyListToJSON = exports.PaginatedEventMatcherPolicyListFromJSONTyped = exports.PaginatedEventMatcherPolicyListFromJSON = exports.instanceOfPaginatedEventMatcherPolicyList = void 0;
const EventMatcherPolicy_1 = require("./EventMatcherPolicy");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedEventMatcherPolicyList interface.
 */
function instanceOfPaginatedEventMatcherPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedEventMatcherPolicyList = instanceOfPaginatedEventMatcherPolicyList;
function PaginatedEventMatcherPolicyListFromJSON(json) {
    return PaginatedEventMatcherPolicyListFromJSONTyped(json, false);
}
exports.PaginatedEventMatcherPolicyListFromJSON = PaginatedEventMatcherPolicyListFromJSON;
function PaginatedEventMatcherPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(EventMatcherPolicy_1.EventMatcherPolicyFromJSON)),
    };
}
exports.PaginatedEventMatcherPolicyListFromJSONTyped = PaginatedEventMatcherPolicyListFromJSONTyped;
function PaginatedEventMatcherPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(EventMatcherPolicy_1.EventMatcherPolicyToJSON)),
    };
}
exports.PaginatedEventMatcherPolicyListToJSON = PaginatedEventMatcherPolicyListToJSON;
