// @goauthentik/api/dist/models/PaginatedSAMLProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedSAMLProviderListToJSON = exports.PaginatedSAMLProviderListFromJSONTyped = exports.PaginatedSAMLProviderListFromJSON = exports.instanceOfPaginatedSAMLProviderList = void 0;
const Pagination_1 = require("./Pagination");
const SAMLProvider_1 = require("./SAMLProvider");
/**
 * Check if a given object implements the PaginatedSAMLProviderList interface.
 */
function instanceOfPaginatedSAMLProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedSAMLProviderList = instanceOfPaginatedSAMLProviderList;
function PaginatedSAMLProviderListFromJSON(json) {
    return PaginatedSAMLProviderListFromJSONTyped(json, false);
}
exports.PaginatedSAMLProviderListFromJSON = PaginatedSAMLProviderListFromJSON;
function PaginatedSAMLProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(SAMLProvider_1.SAMLProviderFromJSON)),
    };
}
exports.PaginatedSAMLProviderListFromJSONTyped = PaginatedSAMLProviderListFromJSONTyped;
function PaginatedSAMLProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(SAMLProvider_1.SAMLProviderToJSON)),
    };
}
exports.PaginatedSAMLProviderListToJSON = PaginatedSAMLProviderListToJSON;
