// @goauthentik/api/dist/models/PaginatedFlowStageBindingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedFlowStageBindingListToJSON = exports.PaginatedFlowStageBindingListFromJSONTyped = exports.PaginatedFlowStageBindingListFromJSON = exports.instanceOfPaginatedFlowStageBindingList = void 0;
const FlowStageBinding_1 = require("./FlowStageBinding");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedFlowStageBindingList interface.
 */
function instanceOfPaginatedFlowStageBindingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedFlowStageBindingList = instanceOfPaginatedFlowStageBindingList;
function PaginatedFlowStageBindingListFromJSON(json) {
    return PaginatedFlowStageBindingListFromJSONTyped(json, false);
}
exports.PaginatedFlowStageBindingListFromJSON = PaginatedFlowStageBindingListFromJSON;
function PaginatedFlowStageBindingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(FlowStageBinding_1.FlowStageBindingFromJSON)),
    };
}
exports.PaginatedFlowStageBindingListFromJSONTyped = PaginatedFlowStageBindingListFromJSONTyped;
function PaginatedFlowStageBindingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(FlowStageBinding_1.FlowStageBindingToJSON)),
    };
}
exports.PaginatedFlowStageBindingListToJSON = PaginatedFlowStageBindingListToJSON;
