// @goauthentik/api/dist/models/PaginatedRACPropertyMappingList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedRACPropertyMappingListToJSON = exports.PaginatedRACPropertyMappingListFromJSONTyped = exports.PaginatedRACPropertyMappingListFromJSON = exports.instanceOfPaginatedRACPropertyMappingList = void 0;
const Pagination_1 = require("./Pagination");
const RACPropertyMapping_1 = require("./RACPropertyMapping");
/**
 * Check if a given object implements the PaginatedRACPropertyMappingList interface.
 */
function instanceOfPaginatedRACPropertyMappingList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedRACPropertyMappingList = instanceOfPaginatedRACPropertyMappingList;
function PaginatedRACPropertyMappingListFromJSON(json) {
    return PaginatedRACPropertyMappingListFromJSONTyped(json, false);
}
exports.PaginatedRACPropertyMappingListFromJSON = PaginatedRACPropertyMappingListFromJSON;
function PaginatedRACPropertyMappingListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(RACPropertyMapping_1.RACPropertyMappingFromJSON)),
    };
}
exports.PaginatedRACPropertyMappingListFromJSONTyped = PaginatedRACPropertyMappingListFromJSONTyped;
function PaginatedRACPropertyMappingListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(RACPropertyMapping_1.RACPropertyMappingToJSON)),
    };
}
exports.PaginatedRACPropertyMappingListToJSON = PaginatedRACPropertyMappingListToJSON;
