// @goauthentik/api/dist/models/PaginatedOAuth2ProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedOAuth2ProviderListToJSON = exports.PaginatedOAuth2ProviderListFromJSONTyped = exports.PaginatedOAuth2ProviderListFromJSON = exports.instanceOfPaginatedOAuth2ProviderList = void 0;
const OAuth2Provider_1 = require("./OAuth2Provider");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedOAuth2ProviderList interface.
 */
function instanceOfPaginatedOAuth2ProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedOAuth2ProviderList = instanceOfPaginatedOAuth2ProviderList;
function PaginatedOAuth2ProviderListFromJSON(json) {
    return PaginatedOAuth2ProviderListFromJSONTyped(json, false);
}
exports.PaginatedOAuth2ProviderListFromJSON = PaginatedOAuth2ProviderListFromJSON;
function PaginatedOAuth2ProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(OAuth2Provider_1.OAuth2ProviderFromJSON)),
    };
}
exports.PaginatedOAuth2ProviderListFromJSONTyped = PaginatedOAuth2ProviderListFromJSONTyped;
function PaginatedOAuth2ProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(OAuth2Provider_1.OAuth2ProviderToJSON)),
    };
}
exports.PaginatedOAuth2ProviderListToJSON = PaginatedOAuth2ProviderListToJSON;
