// @goauthentik/api/dist/models/PaginatedPermissionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPermissionListToJSON = exports.PaginatedPermissionListFromJSONTyped = exports.PaginatedPermissionListFromJSON = exports.instanceOfPaginatedPermissionList = void 0;
const Pagination_1 = require("./Pagination");
const Permission_1 = require("./Permission");
/**
 * Check if a given object implements the PaginatedPermissionList interface.
 */
function instanceOfPaginatedPermissionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPermissionList = instanceOfPaginatedPermissionList;
function PaginatedPermissionListFromJSON(json) {
    return PaginatedPermissionListFromJSONTyped(json, false);
}
exports.PaginatedPermissionListFromJSON = PaginatedPermissionListFromJSON;
function PaginatedPermissionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Permission_1.PermissionFromJSON)),
    };
}
exports.PaginatedPermissionListFromJSONTyped = PaginatedPermissionListFromJSONTyped;
function PaginatedPermissionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Permission_1.PermissionToJSON)),
    };
}
exports.PaginatedPermissionListToJSON = PaginatedPermissionListToJSON;
