// @goauthentik/api/dist/models/PaginatedDenyStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDenyStageListToJSON = exports.PaginatedDenyStageListFromJSONTyped = exports.PaginatedDenyStageListFromJSON = exports.instanceOfPaginatedDenyStageList = void 0;
const DenyStage_1 = require("./DenyStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDenyStageList interface.
 */
function instanceOfPaginatedDenyStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDenyStageList = instanceOfPaginatedDenyStageList;
function PaginatedDenyStageListFromJSON(json) {
    return PaginatedDenyStageListFromJSONTyped(json, false);
}
exports.PaginatedDenyStageListFromJSON = PaginatedDenyStageListFromJSON;
function PaginatedDenyStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(DenyStage_1.DenyStageFromJSON)),
    };
}
exports.PaginatedDenyStageListFromJSONTyped = PaginatedDenyStageListFromJSONTyped;
function PaginatedDenyStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(DenyStage_1.DenyStageToJSON)),
    };
}
exports.PaginatedDenyStageListToJSON = PaginatedDenyStageListToJSON;
