// @goauthentik/api/dist/models/PaginatedUserList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedUserListToJSON = exports.PaginatedUserListFromJSONTyped = exports.PaginatedUserListFromJSON = exports.instanceOfPaginatedUserList = void 0;
const Pagination_1 = require("./Pagination");
const User_1 = require("./User");
/**
 * Check if a given object implements the PaginatedUserList interface.
 */
function instanceOfPaginatedUserList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedUserList = instanceOfPaginatedUserList;
function PaginatedUserListFromJSON(json) {
    return PaginatedUserListFromJSONTyped(json, false);
}
exports.PaginatedUserListFromJSON = PaginatedUserListFromJSON;
function PaginatedUserListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(User_1.UserFromJSON)),
    };
}
exports.PaginatedUserListFromJSONTyped = PaginatedUserListFromJSONTyped;
function PaginatedUserListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(User_1.UserToJSON)),
    };
}
exports.PaginatedUserListToJSON = PaginatedUserListToJSON;
