// @goauthentik/api/dist/models/PaginatedDuoDeviceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedDuoDeviceListToJSON = exports.PaginatedDuoDeviceListFromJSONTyped = exports.PaginatedDuoDeviceListFromJSON = exports.instanceOfPaginatedDuoDeviceList = void 0;
const DuoDevice_1 = require("./DuoDevice");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedDuoDeviceList interface.
 */
function instanceOfPaginatedDuoDeviceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedDuoDeviceList = instanceOfPaginatedDuoDeviceList;
function PaginatedDuoDeviceListFromJSON(json) {
    return PaginatedDuoDeviceListFromJSONTyped(json, false);
}
exports.PaginatedDuoDeviceListFromJSON = PaginatedDuoDeviceListFromJSON;
function PaginatedDuoDeviceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(DuoDevice_1.DuoDeviceFromJSON)),
    };
}
exports.PaginatedDuoDeviceListFromJSONTyped = PaginatedDuoDeviceListFromJSONTyped;
function PaginatedDuoDeviceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(DuoDevice_1.DuoDeviceToJSON)),
    };
}
exports.PaginatedDuoDeviceListToJSON = PaginatedDuoDeviceListToJSON;
