// @goauthentik/api/dist/models/PaginatedProxyProviderList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedProxyProviderListToJSON = exports.PaginatedProxyProviderListFromJSONTyped = exports.PaginatedProxyProviderListFromJSON = exports.instanceOfPaginatedProxyProviderList = void 0;
const Pagination_1 = require("./Pagination");
const ProxyProvider_1 = require("./ProxyProvider");
/**
 * Check if a given object implements the PaginatedProxyProviderList interface.
 */
function instanceOfPaginatedProxyProviderList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedProxyProviderList = instanceOfPaginatedProxyProviderList;
function PaginatedProxyProviderListFromJSON(json) {
    return PaginatedProxyProviderListFromJSONTyped(json, false);
}
exports.PaginatedProxyProviderListFromJSON = PaginatedProxyProviderListFromJSON;
function PaginatedProxyProviderListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ProxyProvider_1.ProxyProviderFromJSON)),
    };
}
exports.PaginatedProxyProviderListFromJSONTyped = PaginatedProxyProviderListFromJSONTyped;
function PaginatedProxyProviderListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ProxyProvider_1.ProxyProviderToJSON)),
    };
}
exports.PaginatedProxyProviderListToJSON = PaginatedProxyProviderListToJSON;
