// @goauthentik/api/dist/models/PaginatedStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedStageListToJSON = exports.PaginatedStageListFromJSONTyped = exports.PaginatedStageListFromJSON = exports.instanceOfPaginatedStageList = void 0;
const Pagination_1 = require("./Pagination");
const Stage_1 = require("./Stage");
/**
 * Check if a given object implements the PaginatedStageList interface.
 */
function instanceOfPaginatedStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedStageList = instanceOfPaginatedStageList;
function PaginatedStageListFromJSON(json) {
    return PaginatedStageListFromJSONTyped(json, false);
}
exports.PaginatedStageListFromJSON = PaginatedStageListFromJSON;
function PaginatedStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(Stage_1.StageFromJSON)),
    };
}
exports.PaginatedStageListFromJSONTyped = PaginatedStageListFromJSONTyped;
function PaginatedStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(Stage_1.StageToJSON)),
    };
}
exports.PaginatedStageListToJSON = PaginatedStageListToJSON;
