// @goauthentik/api/dist/models/PaginatedIdentificationStageList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedIdentificationStageListToJSON = exports.PaginatedIdentificationStageListFromJSONTyped = exports.PaginatedIdentificationStageListFromJSON = exports.instanceOfPaginatedIdentificationStageList = void 0;
const IdentificationStage_1 = require("./IdentificationStage");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedIdentificationStageList interface.
 */
function instanceOfPaginatedIdentificationStageList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedIdentificationStageList = instanceOfPaginatedIdentificationStageList;
function PaginatedIdentificationStageListFromJSON(json) {
    return PaginatedIdentificationStageListFromJSONTyped(json, false);
}
exports.PaginatedIdentificationStageListFromJSON = PaginatedIdentificationStageListFromJSON;
function PaginatedIdentificationStageListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(IdentificationStage_1.IdentificationStageFromJSON)),
    };
}
exports.PaginatedIdentificationStageListFromJSONTyped = PaginatedIdentificationStageListFromJSONTyped;
function PaginatedIdentificationStageListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(IdentificationStage_1.IdentificationStageToJSON)),
    };
}
exports.PaginatedIdentificationStageListToJSON = PaginatedIdentificationStageListToJSON;
