// @goauthentik/api/dist/models/PaginatedConnectionTokenList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedConnectionTokenListToJSON = exports.PaginatedConnectionTokenListFromJSONTyped = exports.PaginatedConnectionTokenListFromJSON = exports.instanceOfPaginatedConnectionTokenList = void 0;
const ConnectionToken_1 = require("./ConnectionToken");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedConnectionTokenList interface.
 */
function instanceOfPaginatedConnectionTokenList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedConnectionTokenList = instanceOfPaginatedConnectionTokenList;
function PaginatedConnectionTokenListFromJSON(json) {
    return PaginatedConnectionTokenListFromJSONTyped(json, false);
}
exports.PaginatedConnectionTokenListFromJSON = PaginatedConnectionTokenListFromJSON;
function PaginatedConnectionTokenListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ConnectionToken_1.ConnectionTokenFromJSON)),
    };
}
exports.PaginatedConnectionTokenListFromJSONTyped = PaginatedConnectionTokenListFromJSONTyped;
function PaginatedConnectionTokenListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ConnectionToken_1.ConnectionTokenToJSON)),
    };
}
exports.PaginatedConnectionTokenListToJSON = PaginatedConnectionTokenListToJSON;
