// @goauthentik/api/dist/models/PaginatedOAuthSourceList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedOAuthSourceListToJSON = exports.PaginatedOAuthSourceListFromJSONTyped = exports.PaginatedOAuthSourceListFromJSON = exports.instanceOfPaginatedOAuthSourceList = void 0;
const OAuthSource_1 = require("./OAuthSource");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedOAuthSourceList interface.
 */
function instanceOfPaginatedOAuthSourceList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedOAuthSourceList = instanceOfPaginatedOAuthSourceList;
function PaginatedOAuthSourceListFromJSON(json) {
    return PaginatedOAuthSourceListFromJSONTyped(json, false);
}
exports.PaginatedOAuthSourceListFromJSON = PaginatedOAuthSourceListFromJSON;
function PaginatedOAuthSourceListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(OAuthSource_1.OAuthSourceFromJSON)),
    };
}
exports.PaginatedOAuthSourceListFromJSONTyped = PaginatedOAuthSourceListFromJSONTyped;
function PaginatedOAuthSourceListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(OAuthSource_1.OAuthSourceToJSON)),
    };
}
exports.PaginatedOAuthSourceListToJSON = PaginatedOAuthSourceListToJSON;
