// @goauthentik/api/dist/models/PaginatedLicenseList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedLicenseListToJSON = exports.PaginatedLicenseListFromJSONTyped = exports.PaginatedLicenseListFromJSON = exports.instanceOfPaginatedLicenseList = void 0;
const License_1 = require("./License");
const Pagination_1 = require("./Pagination");
/**
 * Check if a given object implements the PaginatedLicenseList interface.
 */
function instanceOfPaginatedLicenseList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedLicenseList = instanceOfPaginatedLicenseList;
function PaginatedLicenseListFromJSON(json) {
    return PaginatedLicenseListFromJSONTyped(json, false);
}
exports.PaginatedLicenseListFromJSON = PaginatedLicenseListFromJSON;
function PaginatedLicenseListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(License_1.LicenseFromJSON)),
    };
}
exports.PaginatedLicenseListFromJSONTyped = PaginatedLicenseListFromJSONTyped;
function PaginatedLicenseListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(License_1.LicenseToJSON)),
    };
}
exports.PaginatedLicenseListToJSON = PaginatedLicenseListToJSON;
