// @goauthentik/api/dist/models/PaginatedPasswordPolicyList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedPasswordPolicyListToJSON = exports.PaginatedPasswordPolicyListFromJSONTyped = exports.PaginatedPasswordPolicyListFromJSON = exports.instanceOfPaginatedPasswordPolicyList = void 0;
const Pagination_1 = require("./Pagination");
const PasswordPolicy_1 = require("./PasswordPolicy");
/**
 * Check if a given object implements the PaginatedPasswordPolicyList interface.
 */
function instanceOfPaginatedPasswordPolicyList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedPasswordPolicyList = instanceOfPaginatedPasswordPolicyList;
function PaginatedPasswordPolicyListFromJSON(json) {
    return PaginatedPasswordPolicyListFromJSONTyped(json, false);
}
exports.PaginatedPasswordPolicyListFromJSON = PaginatedPasswordPolicyListFromJSON;
function PaginatedPasswordPolicyListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(PasswordPolicy_1.PasswordPolicyFromJSON)),
    };
}
exports.PaginatedPasswordPolicyListFromJSONTyped = PaginatedPasswordPolicyListFromJSONTyped;
function PaginatedPasswordPolicyListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(PasswordPolicy_1.PasswordPolicyToJSON)),
    };
}
exports.PaginatedPasswordPolicyListToJSON = PaginatedPasswordPolicyListToJSON;
