// @goauthentik/api/dist/models/PaginatedServiceConnectionList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedServiceConnectionListToJSON = exports.PaginatedServiceConnectionListFromJSONTyped = exports.PaginatedServiceConnectionListFromJSON = exports.instanceOfPaginatedServiceConnectionList = void 0;
const Pagination_1 = require("./Pagination");
const ServiceConnection_1 = require("./ServiceConnection");
/**
 * Check if a given object implements the PaginatedServiceConnectionList interface.
 */
function instanceOfPaginatedServiceConnectionList(value) {
    let isInstance = true;
    isInstance = isInstance && "pagination" in value;
    isInstance = isInstance && "results" in value;
    return isInstance;
}
exports.instanceOfPaginatedServiceConnectionList = instanceOfPaginatedServiceConnectionList;
function PaginatedServiceConnectionListFromJSON(json) {
    return PaginatedServiceConnectionListFromJSONTyped(json, false);
}
exports.PaginatedServiceConnectionListFromJSON = PaginatedServiceConnectionListFromJSON;
function PaginatedServiceConnectionListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pagination': (0, Pagination_1.PaginationFromJSON)(json['pagination']),
        'results': (json['results'].map(ServiceConnection_1.ServiceConnectionFromJSON)),
    };
}
exports.PaginatedServiceConnectionListFromJSONTyped = PaginatedServiceConnectionListFromJSONTyped;
function PaginatedServiceConnectionListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pagination': (0, Pagination_1.PaginationToJSON)(value.pagination),
        'results': (value.results.map(ServiceConnection_1.ServiceConnectionToJSON)),
    };
}
exports.PaginatedServiceConnectionListToJSON = PaginatedServiceConnectionListToJSON;
