// @goauthentik/api/dist/models/PasswordExpiryPolicy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordExpiryPolicyToJSON = exports.PasswordExpiryPolicyFromJSONTyped = exports.PasswordExpiryPolicyFromJSON = exports.instanceOfPasswordExpiryPolicy = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PasswordExpiryPolicy interface.
 */
function instanceOfPasswordExpiryPolicy(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "boundTo" in value;
    isInstance = isInstance && "days" in value;
    return isInstance;
}
exports.instanceOfPasswordExpiryPolicy = instanceOfPasswordExpiryPolicy;
function PasswordExpiryPolicyFromJSON(json) {
    return PasswordExpiryPolicyFromJSONTyped(json, false);
}
exports.PasswordExpiryPolicyFromJSON = PasswordExpiryPolicyFromJSON;
function PasswordExpiryPolicyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'boundTo': json['bound_to'],
        'days': json['days'],
        'denyOnly': !(0, runtime_1.exists)(json, 'deny_only') ? undefined : json['deny_only'],
    };
}
exports.PasswordExpiryPolicyFromJSONTyped = PasswordExpiryPolicyFromJSONTyped;
function PasswordExpiryPolicyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
exports.PasswordExpiryPolicyToJSON = PasswordExpiryPolicyToJSON;
