// @goauthentik/api/dist/models/PasswordChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordChallengeResponseRequestToJSON = exports.PasswordChallengeResponseRequestFromJSONTyped = exports.PasswordChallengeResponseRequestFromJSON = exports.instanceOfPasswordChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PasswordChallengeResponseRequest interface.
 */
function instanceOfPasswordChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "password" in value;
    return isInstance;
}
exports.instanceOfPasswordChallengeResponseRequest = instanceOfPasswordChallengeResponseRequest;
function PasswordChallengeResponseRequestFromJSON(json) {
    return PasswordChallengeResponseRequestFromJSONTyped(json, false);
}
exports.PasswordChallengeResponseRequestFromJSON = PasswordChallengeResponseRequestFromJSON;
function PasswordChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'password': json['password'],
    };
}
exports.PasswordChallengeResponseRequestFromJSONTyped = PasswordChallengeResponseRequestFromJSONTyped;
function PasswordChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'password': value.password,
    };
}
exports.PasswordChallengeResponseRequestToJSON = PasswordChallengeResponseRequestToJSON;
