// @goauthentik/api/dist/models/PasswordExpiryPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordExpiryPolicyRequestToJSON = exports.PasswordExpiryPolicyRequestFromJSONTyped = exports.PasswordExpiryPolicyRequestFromJSON = exports.instanceOfPasswordExpiryPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PasswordExpiryPolicyRequest interface.
 */
function instanceOfPasswordExpiryPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "days" in value;
    return isInstance;
}
exports.instanceOfPasswordExpiryPolicyRequest = instanceOfPasswordExpiryPolicyRequest;
function PasswordExpiryPolicyRequestFromJSON(json) {
    return PasswordExpiryPolicyRequestFromJSONTyped(json, false);
}
exports.PasswordExpiryPolicyRequestFromJSON = PasswordExpiryPolicyRequestFromJSON;
function PasswordExpiryPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'days': json['days'],
        'denyOnly': !(0, runtime_1.exists)(json, 'deny_only') ? undefined : json['deny_only'],
    };
}
exports.PasswordExpiryPolicyRequestFromJSONTyped = PasswordExpiryPolicyRequestFromJSONTyped;
function PasswordExpiryPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
exports.PasswordExpiryPolicyRequestToJSON = PasswordExpiryPolicyRequestToJSON;
