// @goauthentik/api/dist/models/PasswordPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordPolicyRequestToJSON = exports.PasswordPolicyRequestFromJSONTyped = exports.PasswordPolicyRequestFromJSON = exports.instanceOfPasswordPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PasswordPolicyRequest interface.
 */
function instanceOfPasswordPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfPasswordPolicyRequest = instanceOfPasswordPolicyRequest;
function PasswordPolicyRequestFromJSON(json) {
    return PasswordPolicyRequestFromJSONTyped(json, false);
}
exports.PasswordPolicyRequestFromJSON = PasswordPolicyRequestFromJSON;
function PasswordPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'passwordField': !(0, runtime_1.exists)(json, 'password_field') ? undefined : json['password_field'],
        'amountDigits': !(0, runtime_1.exists)(json, 'amount_digits') ? undefined : json['amount_digits'],
        'amountUppercase': !(0, runtime_1.exists)(json, 'amount_uppercase') ? undefined : json['amount_uppercase'],
        'amountLowercase': !(0, runtime_1.exists)(json, 'amount_lowercase') ? undefined : json['amount_lowercase'],
        'amountSymbols': !(0, runtime_1.exists)(json, 'amount_symbols') ? undefined : json['amount_symbols'],
        'lengthMin': !(0, runtime_1.exists)(json, 'length_min') ? undefined : json['length_min'],
        'symbolCharset': !(0, runtime_1.exists)(json, 'symbol_charset') ? undefined : json['symbol_charset'],
        'errorMessage': !(0, runtime_1.exists)(json, 'error_message') ? undefined : json['error_message'],
        'checkStaticRules': !(0, runtime_1.exists)(json, 'check_static_rules') ? undefined : json['check_static_rules'],
        'checkHaveIBeenPwned': !(0, runtime_1.exists)(json, 'check_have_i_been_pwned') ? undefined : json['check_have_i_been_pwned'],
        'checkZxcvbn': !(0, runtime_1.exists)(json, 'check_zxcvbn') ? undefined : json['check_zxcvbn'],
        'hibpAllowedCount': !(0, runtime_1.exists)(json, 'hibp_allowed_count') ? undefined : json['hibp_allowed_count'],
        'zxcvbnScoreThreshold': !(0, runtime_1.exists)(json, 'zxcvbn_score_threshold') ? undefined : json['zxcvbn_score_threshold'],
    };
}
exports.PasswordPolicyRequestFromJSONTyped = PasswordPolicyRequestFromJSONTyped;
function PasswordPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'password_field': value.passwordField,
        'amount_digits': value.amountDigits,
        'amount_uppercase': value.amountUppercase,
        'amount_lowercase': value.amountLowercase,
        'amount_symbols': value.amountSymbols,
        'length_min': value.lengthMin,
        'symbol_charset': value.symbolCharset,
        'error_message': value.errorMessage,
        'check_static_rules': value.checkStaticRules,
        'check_have_i_been_pwned': value.checkHaveIBeenPwned,
        'check_zxcvbn': value.checkZxcvbn,
        'hibp_allowed_count': value.hibpAllowedCount,
        'zxcvbn_score_threshold': value.zxcvbnScoreThreshold,
    };
}
exports.PasswordPolicyRequestToJSON = PasswordPolicyRequestToJSON;
