// @goauthentik/api/dist/models/PasswordStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PasswordStageToJSON = exports.PasswordStageFromJSONTyped = exports.PasswordStageFromJSON = exports.instanceOfPasswordStage = void 0;
const runtime_1 = require("../runtime");
const BackendsEnum_1 = require("./BackendsEnum");
const FlowSet_1 = require("./FlowSet");
/**
 * Check if a given object implements the PasswordStage interface.
 */
function instanceOfPasswordStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "backends" in value;
    return isInstance;
}
exports.instanceOfPasswordStage = instanceOfPasswordStage;
function PasswordStageFromJSON(json) {
    return PasswordStageFromJSONTyped(json, false);
}
exports.PasswordStageFromJSON = PasswordStageFromJSON;
function PasswordStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'backends': (json['backends'].map(BackendsEnum_1.BackendsEnumFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'failedAttemptsBeforeCancel': !(0, runtime_1.exists)(json, 'failed_attempts_before_cancel') ? undefined : json['failed_attempts_before_cancel'],
    };
}
exports.PasswordStageFromJSONTyped = PasswordStageFromJSONTyped;
function PasswordStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'backends': (value.backends.map(BackendsEnum_1.BackendsEnumToJSON)),
        'configure_flow': value.configureFlow,
        'failed_attempts_before_cancel': value.failedAttemptsBeforeCancel,
    };
}
exports.PasswordStageToJSON = PasswordStageToJSON;
