// @goauthentik/api/dist/models/PatchedStaticDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedStaticDeviceRequestToJSON = exports.PatchedStaticDeviceRequestFromJSONTyped = exports.PatchedStaticDeviceRequestFromJSON = exports.instanceOfPatchedStaticDeviceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedStaticDeviceRequest interface.
 */
function instanceOfPatchedStaticDeviceRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedStaticDeviceRequest = instanceOfPatchedStaticDeviceRequest;
function PatchedStaticDeviceRequestFromJSON(json) {
    return PatchedStaticDeviceRequestFromJSONTyped(json, false);
}
exports.PatchedStaticDeviceRequestFromJSON = PatchedStaticDeviceRequestFromJSON;
function PatchedStaticDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.PatchedStaticDeviceRequestFromJSONTyped = PatchedStaticDeviceRequestFromJSONTyped;
function PatchedStaticDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.PatchedStaticDeviceRequestToJSON = PatchedStaticDeviceRequestToJSON;
