// @goauthentik/api/dist/models/PatchedSettingsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedSettingsRequestToJSON = exports.PatchedSettingsRequestFromJSONTyped = exports.PatchedSettingsRequestFromJSON = exports.instanceOfPatchedSettingsRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedSettingsRequest interface.
 */
function instanceOfPatchedSettingsRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedSettingsRequest = instanceOfPatchedSettingsRequest;
function PatchedSettingsRequestFromJSON(json) {
    return PatchedSettingsRequestFromJSONTyped(json, false);
}
exports.PatchedSettingsRequestFromJSON = PatchedSettingsRequestFromJSON;
function PatchedSettingsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'avatars': !(0, runtime_1.exists)(json, 'avatars') ? undefined : json['avatars'],
        'defaultUserChangeName': !(0, runtime_1.exists)(json, 'default_user_change_name') ? undefined : json['default_user_change_name'],
        'defaultUserChangeEmail': !(0, runtime_1.exists)(json, 'default_user_change_email') ? undefined : json['default_user_change_email'],
        'defaultUserChangeUsername': !(0, runtime_1.exists)(json, 'default_user_change_username') ? undefined : json['default_user_change_username'],
        'eventRetention': !(0, runtime_1.exists)(json, 'event_retention') ? undefined : json['event_retention'],
        'footerLinks': !(0, runtime_1.exists)(json, 'footer_links') ? undefined : json['footer_links'],
        'gdprCompliance': !(0, runtime_1.exists)(json, 'gdpr_compliance') ? undefined : json['gdpr_compliance'],
        'impersonation': !(0, runtime_1.exists)(json, 'impersonation') ? undefined : json['impersonation'],
    };
}
exports.PatchedSettingsRequestFromJSONTyped = PatchedSettingsRequestFromJSONTyped;
function PatchedSettingsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'avatars': value.avatars,
        'default_user_change_name': value.defaultUserChangeName,
        'default_user_change_email': value.defaultUserChangeEmail,
        'default_user_change_username': value.defaultUserChangeUsername,
        'event_retention': value.eventRetention,
        'footer_links': value.footerLinks,
        'gdpr_compliance': value.gdprCompliance,
        'impersonation': value.impersonation,
    };
}
exports.PatchedSettingsRequestToJSON = PatchedSettingsRequestToJSON;
