// @goauthentik/api/dist/models/PatchedOAuth2ProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedOAuth2ProviderRequestToJSON = exports.PatchedOAuth2ProviderRequestFromJSONTyped = exports.PatchedOAuth2ProviderRequestFromJSON = exports.instanceOfPatchedOAuth2ProviderRequest = void 0;
const runtime_1 = require("../runtime");
const ClientTypeEnum_1 = require("./ClientTypeEnum");
const IssuerModeEnum_1 = require("./IssuerModeEnum");
const SubModeEnum_1 = require("./SubModeEnum");
/**
 * Check if a given object implements the PatchedOAuth2ProviderRequest interface.
 */
function instanceOfPatchedOAuth2ProviderRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedOAuth2ProviderRequest = instanceOfPatchedOAuth2ProviderRequest;
function PatchedOAuth2ProviderRequestFromJSON(json) {
    return PatchedOAuth2ProviderRequestFromJSONTyped(json, false);
}
exports.PatchedOAuth2ProviderRequestFromJSON = PatchedOAuth2ProviderRequestFromJSON;
function PatchedOAuth2ProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': !(0, runtime_1.exists)(json, 'authorization_flow') ? undefined : json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'clientType': !(0, runtime_1.exists)(json, 'client_type') ? undefined : (0, ClientTypeEnum_1.ClientTypeEnumFromJSON)(json['client_type']),
        'clientId': !(0, runtime_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !(0, runtime_1.exists)(json, 'client_secret') ? undefined : json['client_secret'],
        'accessCodeValidity': !(0, runtime_1.exists)(json, 'access_code_validity') ? undefined : json['access_code_validity'],
        'accessTokenValidity': !(0, runtime_1.exists)(json, 'access_token_validity') ? undefined : json['access_token_validity'],
        'refreshTokenValidity': !(0, runtime_1.exists)(json, 'refresh_token_validity') ? undefined : json['refresh_token_validity'],
        'includeClaimsInIdToken': !(0, runtime_1.exists)(json, 'include_claims_in_id_token') ? undefined : json['include_claims_in_id_token'],
        'signingKey': !(0, runtime_1.exists)(json, 'signing_key') ? undefined : json['signing_key'],
        'redirectUris': !(0, runtime_1.exists)(json, 'redirect_uris') ? undefined : json['redirect_uris'],
        'subMode': !(0, runtime_1.exists)(json, 'sub_mode') ? undefined : (0, SubModeEnum_1.SubModeEnumFromJSON)(json['sub_mode']),
        'issuerMode': !(0, runtime_1.exists)(json, 'issuer_mode') ? undefined : (0, IssuerModeEnum_1.IssuerModeEnumFromJSON)(json['issuer_mode']),
        'jwksSources': !(0, runtime_1.exists)(json, 'jwks_sources') ? undefined : json['jwks_sources'],
    };
}
exports.PatchedOAuth2ProviderRequestFromJSONTyped = PatchedOAuth2ProviderRequestFromJSONTyped;
function PatchedOAuth2ProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'client_type': (0, ClientTypeEnum_1.ClientTypeEnumToJSON)(value.clientType),
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'access_code_validity': value.accessCodeValidity,
        'access_token_validity': value.accessTokenValidity,
        'refresh_token_validity': value.refreshTokenValidity,
        'include_claims_in_id_token': value.includeClaimsInIdToken,
        'signing_key': value.signingKey,
        'redirect_uris': value.redirectUris,
        'sub_mode': (0, SubModeEnum_1.SubModeEnumToJSON)(value.subMode),
        'issuer_mode': (0, IssuerModeEnum_1.IssuerModeEnumToJSON)(value.issuerMode),
        'jwks_sources': value.jwksSources,
    };
}
exports.PatchedOAuth2ProviderRequestToJSON = PatchedOAuth2ProviderRequestToJSON;
