// @goauthentik/api/dist/models/PatchedPlexSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPlexSourceConnectionRequestToJSON = exports.PatchedPlexSourceConnectionRequestFromJSONTyped = exports.PatchedPlexSourceConnectionRequestFromJSON = exports.instanceOfPatchedPlexSourceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedPlexSourceConnectionRequest interface.
 */
function instanceOfPatchedPlexSourceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPlexSourceConnectionRequest = instanceOfPatchedPlexSourceConnectionRequest;
function PatchedPlexSourceConnectionRequestFromJSON(json) {
    return PatchedPlexSourceConnectionRequestFromJSONTyped(json, false);
}
exports.PatchedPlexSourceConnectionRequestFromJSON = PatchedPlexSourceConnectionRequestFromJSON;
function PatchedPlexSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'identifier': !(0, runtime_1.exists)(json, 'identifier') ? undefined : json['identifier'],
        'plexToken': !(0, runtime_1.exists)(json, 'plex_token') ? undefined : json['plex_token'],
    };
}
exports.PatchedPlexSourceConnectionRequestFromJSONTyped = PatchedPlexSourceConnectionRequestFromJSONTyped;
function PatchedPlexSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
exports.PatchedPlexSourceConnectionRequestToJSON = PatchedPlexSourceConnectionRequestToJSON;
