// @goauthentik/api/dist/models/PatchedExtAuthStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedExtAuthStageRequestToJSON = exports.PatchedExtAuthStageRequestFromJSONTyped = exports.PatchedExtAuthStageRequestFromJSON = exports.instanceOfPatchedExtAuthStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedExtAuthStageRequest interface.
 */
function instanceOfPatchedExtAuthStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedExtAuthStageRequest = instanceOfPatchedExtAuthStageRequest;
function PatchedExtAuthStageRequestFromJSON(json) {
    return PatchedExtAuthStageRequestFromJSONTyped(json, false);
}
exports.PatchedExtAuthStageRequestFromJSON = PatchedExtAuthStageRequestFromJSON;
function PatchedExtAuthStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
    };
}
exports.PatchedExtAuthStageRequestFromJSONTyped = PatchedExtAuthStageRequestFromJSONTyped;
function PatchedExtAuthStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
    };
}
exports.PatchedExtAuthStageRequestToJSON = PatchedExtAuthStageRequestToJSON;
