// @goauthentik/api/dist/models/PatchedPromptStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPromptStageRequestToJSON = exports.PatchedPromptStageRequestFromJSONTyped = exports.PatchedPromptStageRequestFromJSON = exports.instanceOfPatchedPromptStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedPromptStageRequest interface.
 */
function instanceOfPatchedPromptStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPromptStageRequest = instanceOfPatchedPromptStageRequest;
function PatchedPromptStageRequestFromJSON(json) {
    return PatchedPromptStageRequestFromJSONTyped(json, false);
}
exports.PatchedPromptStageRequestFromJSON = PatchedPromptStageRequestFromJSON;
function PatchedPromptStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'fields': !(0, runtime_1.exists)(json, 'fields') ? undefined : json['fields'],
        'validationPolicies': !(0, runtime_1.exists)(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}
exports.PatchedPromptStageRequestFromJSONTyped = PatchedPromptStageRequestFromJSONTyped;
function PatchedPromptStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
exports.PatchedPromptStageRequestToJSON = PatchedPromptStageRequestToJSON;
