// @goauthentik/api/dist/models/PatchedDenyStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedDenyStageRequestToJSON = exports.PatchedDenyStageRequestFromJSONTyped = exports.PatchedDenyStageRequestFromJSON = exports.instanceOfPatchedDenyStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedDenyStageRequest interface.
 */
function instanceOfPatchedDenyStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedDenyStageRequest = instanceOfPatchedDenyStageRequest;
function PatchedDenyStageRequestFromJSON(json) {
    return PatchedDenyStageRequestFromJSONTyped(json, false);
}
exports.PatchedDenyStageRequestFromJSON = PatchedDenyStageRequestFromJSON;
function PatchedDenyStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'denyMessage': !(0, runtime_1.exists)(json, 'deny_message') ? undefined : json['deny_message'],
    };
}
exports.PatchedDenyStageRequestFromJSONTyped = PatchedDenyStageRequestFromJSONTyped;
function PatchedDenyStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'deny_message': value.denyMessage,
    };
}
exports.PatchedDenyStageRequestToJSON = PatchedDenyStageRequestToJSON;
