// @goauthentik/api/dist/models/PatchedTokenRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedTokenRequestToJSON = exports.PatchedTokenRequestFromJSONTyped = exports.PatchedTokenRequestFromJSON = exports.instanceOfPatchedTokenRequest = void 0;
const runtime_1 = require("../runtime");
const IntentEnum_1 = require("./IntentEnum");
/**
 * Check if a given object implements the PatchedTokenRequest interface.
 */
function instanceOfPatchedTokenRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedTokenRequest = instanceOfPatchedTokenRequest;
function PatchedTokenRequestFromJSON(json) {
    return PatchedTokenRequestFromJSONTyped(json, false);
}
exports.PatchedTokenRequestFromJSON = PatchedTokenRequestFromJSON;
function PatchedTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'identifier': !(0, runtime_1.exists)(json, 'identifier') ? undefined : json['identifier'],
        'intent': !(0, runtime_1.exists)(json, 'intent') ? undefined : (0, IntentEnum_1.IntentEnumFromJSON)(json['intent']),
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'expiring': !(0, runtime_1.exists)(json, 'expiring') ? undefined : json['expiring'],
    };
}
exports.PatchedTokenRequestFromJSONTyped = PatchedTokenRequestFromJSONTyped;
function PatchedTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'identifier': value.identifier,
        'intent': (0, IntentEnum_1.IntentEnumToJSON)(value.intent),
        'user': value.user,
        'description': value.description,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'expiring': value.expiring,
    };
}
exports.PatchedTokenRequestToJSON = PatchedTokenRequestToJSON;
