// @goauthentik/api/dist/models/PatchedAuthenticatorDuoStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedAuthenticatorDuoStageRequestToJSON = exports.PatchedAuthenticatorDuoStageRequestFromJSONTyped = exports.PatchedAuthenticatorDuoStageRequestFromJSON = exports.instanceOfPatchedAuthenticatorDuoStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedAuthenticatorDuoStageRequest interface.
 */
function instanceOfPatchedAuthenticatorDuoStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedAuthenticatorDuoStageRequest = instanceOfPatchedAuthenticatorDuoStageRequest;
function PatchedAuthenticatorDuoStageRequestFromJSON(json) {
    return PatchedAuthenticatorDuoStageRequestFromJSONTyped(json, false);
}
exports.PatchedAuthenticatorDuoStageRequestFromJSON = PatchedAuthenticatorDuoStageRequestFromJSON;
function PatchedAuthenticatorDuoStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'clientId': !(0, runtime_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !(0, runtime_1.exists)(json, 'client_secret') ? undefined : json['client_secret'],
        'apiHostname': !(0, runtime_1.exists)(json, 'api_hostname') ? undefined : json['api_hostname'],
        'adminIntegrationKey': !(0, runtime_1.exists)(json, 'admin_integration_key') ? undefined : json['admin_integration_key'],
        'adminSecretKey': !(0, runtime_1.exists)(json, 'admin_secret_key') ? undefined : json['admin_secret_key'],
    };
}
exports.PatchedAuthenticatorDuoStageRequestFromJSONTyped = PatchedAuthenticatorDuoStageRequestFromJSONTyped;
function PatchedAuthenticatorDuoStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
        'api_hostname': value.apiHostname,
        'admin_integration_key': value.adminIntegrationKey,
        'admin_secret_key': value.adminSecretKey,
    };
}
exports.PatchedAuthenticatorDuoStageRequestToJSON = PatchedAuthenticatorDuoStageRequestToJSON;
