// @goauthentik/api/dist/models/PatchedIdentificationStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedIdentificationStageRequestToJSON = exports.PatchedIdentificationStageRequestFromJSONTyped = exports.PatchedIdentificationStageRequestFromJSON = exports.instanceOfPatchedIdentificationStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
const UserFieldsEnum_1 = require("./UserFieldsEnum");
/**
 * Check if a given object implements the PatchedIdentificationStageRequest interface.
 */
function instanceOfPatchedIdentificationStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedIdentificationStageRequest = instanceOfPatchedIdentificationStageRequest;
function PatchedIdentificationStageRequestFromJSON(json) {
    return PatchedIdentificationStageRequestFromJSONTyped(json, false);
}
exports.PatchedIdentificationStageRequestFromJSON = PatchedIdentificationStageRequestFromJSON;
function PatchedIdentificationStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'userFields': !(0, runtime_1.exists)(json, 'user_fields') ? undefined : (json['user_fields'].map(UserFieldsEnum_1.UserFieldsEnumFromJSON)),
        'passwordStage': !(0, runtime_1.exists)(json, 'password_stage') ? undefined : json['password_stage'],
        'caseInsensitiveMatching': !(0, runtime_1.exists)(json, 'case_insensitive_matching') ? undefined : json['case_insensitive_matching'],
        'showMatchedUser': !(0, runtime_1.exists)(json, 'show_matched_user') ? undefined : json['show_matched_user'],
        'enrollmentFlow': !(0, runtime_1.exists)(json, 'enrollment_flow') ? undefined : json['enrollment_flow'],
        'recoveryFlow': !(0, runtime_1.exists)(json, 'recovery_flow') ? undefined : json['recovery_flow'],
        'passwordlessFlow': !(0, runtime_1.exists)(json, 'passwordless_flow') ? undefined : json['passwordless_flow'],
        'sources': !(0, runtime_1.exists)(json, 'sources') ? undefined : json['sources'],
        'showSourceLabels': !(0, runtime_1.exists)(json, 'show_source_labels') ? undefined : json['show_source_labels'],
        'pretendUserExists': !(0, runtime_1.exists)(json, 'pretend_user_exists') ? undefined : json['pretend_user_exists'],
    };
}
exports.PatchedIdentificationStageRequestFromJSONTyped = PatchedIdentificationStageRequestFromJSONTyped;
function PatchedIdentificationStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'user_fields': value.userFields === undefined ? undefined : (value.userFields.map(UserFieldsEnum_1.UserFieldsEnumToJSON)),
        'password_stage': value.passwordStage,
        'case_insensitive_matching': value.caseInsensitiveMatching,
        'show_matched_user': value.showMatchedUser,
        'enrollment_flow': value.enrollmentFlow,
        'recovery_flow': value.recoveryFlow,
        'passwordless_flow': value.passwordlessFlow,
        'sources': value.sources,
        'show_source_labels': value.showSourceLabels,
        'pretend_user_exists': value.pretendUserExists,
    };
}
exports.PatchedIdentificationStageRequestToJSON = PatchedIdentificationStageRequestToJSON;
