// @goauthentik/api/dist/models/PatchedPasswordExpiryPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPasswordExpiryPolicyRequestToJSON = exports.PatchedPasswordExpiryPolicyRequestFromJSONTyped = exports.PatchedPasswordExpiryPolicyRequestFromJSON = exports.instanceOfPatchedPasswordExpiryPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedPasswordExpiryPolicyRequest interface.
 */
function instanceOfPatchedPasswordExpiryPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPasswordExpiryPolicyRequest = instanceOfPatchedPasswordExpiryPolicyRequest;
function PatchedPasswordExpiryPolicyRequestFromJSON(json) {
    return PatchedPasswordExpiryPolicyRequestFromJSONTyped(json, false);
}
exports.PatchedPasswordExpiryPolicyRequestFromJSON = PatchedPasswordExpiryPolicyRequestFromJSON;
function PatchedPasswordExpiryPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'days': !(0, runtime_1.exists)(json, 'days') ? undefined : json['days'],
        'denyOnly': !(0, runtime_1.exists)(json, 'deny_only') ? undefined : json['deny_only'],
    };
}
exports.PatchedPasswordExpiryPolicyRequestFromJSONTyped = PatchedPasswordExpiryPolicyRequestFromJSONTyped;
function PatchedPasswordExpiryPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'days': value.days,
        'deny_only': value.denyOnly,
    };
}
exports.PatchedPasswordExpiryPolicyRequestToJSON = PatchedPasswordExpiryPolicyRequestToJSON;
