// @goauthentik/api/dist/models/PatchedNotificationTransportRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedNotificationTransportRequestToJSON = exports.PatchedNotificationTransportRequestFromJSONTyped = exports.PatchedNotificationTransportRequestFromJSON = exports.instanceOfPatchedNotificationTransportRequest = void 0;
const runtime_1 = require("../runtime");
const NotificationTransportModeEnum_1 = require("./NotificationTransportModeEnum");
/**
 * Check if a given object implements the PatchedNotificationTransportRequest interface.
 */
function instanceOfPatchedNotificationTransportRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedNotificationTransportRequest = instanceOfPatchedNotificationTransportRequest;
function PatchedNotificationTransportRequestFromJSON(json) {
    return PatchedNotificationTransportRequestFromJSONTyped(json, false);
}
exports.PatchedNotificationTransportRequestFromJSON = PatchedNotificationTransportRequestFromJSON;
function PatchedNotificationTransportRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, NotificationTransportModeEnum_1.NotificationTransportModeEnumFromJSON)(json['mode']),
        'webhookUrl': !(0, runtime_1.exists)(json, 'webhook_url') ? undefined : json['webhook_url'],
        'webhookMapping': !(0, runtime_1.exists)(json, 'webhook_mapping') ? undefined : json['webhook_mapping'],
        'sendOnce': !(0, runtime_1.exists)(json, 'send_once') ? undefined : json['send_once'],
    };
}
exports.PatchedNotificationTransportRequestFromJSONTyped = PatchedNotificationTransportRequestFromJSONTyped;
function PatchedNotificationTransportRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'mode': (0, NotificationTransportModeEnum_1.NotificationTransportModeEnumToJSON)(value.mode),
        'webhook_url': value.webhookUrl,
        'webhook_mapping': value.webhookMapping,
        'send_once': value.sendOnce,
    };
}
exports.PatchedNotificationTransportRequestToJSON = PatchedNotificationTransportRequestToJSON;
