// @goauthentik/api/dist/models/PatchedInvitationStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedInvitationStageRequestToJSON = exports.PatchedInvitationStageRequestFromJSONTyped = exports.PatchedInvitationStageRequestFromJSON = exports.instanceOfPatchedInvitationStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedInvitationStageRequest interface.
 */
function instanceOfPatchedInvitationStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedInvitationStageRequest = instanceOfPatchedInvitationStageRequest;
function PatchedInvitationStageRequestFromJSON(json) {
    return PatchedInvitationStageRequestFromJSONTyped(json, false);
}
exports.PatchedInvitationStageRequestFromJSON = PatchedInvitationStageRequestFromJSON;
function PatchedInvitationStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'continueFlowWithoutInvitation': !(0, runtime_1.exists)(json, 'continue_flow_without_invitation') ? undefined : json['continue_flow_without_invitation'],
    };
}
exports.PatchedInvitationStageRequestFromJSONTyped = PatchedInvitationStageRequestFromJSONTyped;
function PatchedInvitationStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'continue_flow_without_invitation': value.continueFlowWithoutInvitation,
    };
}
exports.PatchedInvitationStageRequestToJSON = PatchedInvitationStageRequestToJSON;
