// @goauthentik/api/dist/models/PatchedConsentStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedConsentStageRequestToJSON = exports.PatchedConsentStageRequestFromJSONTyped = exports.PatchedConsentStageRequestFromJSON = exports.instanceOfPatchedConsentStageRequest = void 0;
const runtime_1 = require("../runtime");
const ConsentStageModeEnum_1 = require("./ConsentStageModeEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedConsentStageRequest interface.
 */
function instanceOfPatchedConsentStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedConsentStageRequest = instanceOfPatchedConsentStageRequest;
function PatchedConsentStageRequestFromJSON(json) {
    return PatchedConsentStageRequestFromJSONTyped(json, false);
}
exports.PatchedConsentStageRequestFromJSON = PatchedConsentStageRequestFromJSON;
function PatchedConsentStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, ConsentStageModeEnum_1.ConsentStageModeEnumFromJSON)(json['mode']),
        'consentExpireIn': !(0, runtime_1.exists)(json, 'consent_expire_in') ? undefined : json['consent_expire_in'],
    };
}
exports.PatchedConsentStageRequestFromJSONTyped = PatchedConsentStageRequestFromJSONTyped;
function PatchedConsentStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'mode': (0, ConsentStageModeEnum_1.ConsentStageModeEnumToJSON)(value.mode),
        'consent_expire_in': value.consentExpireIn,
    };
}
exports.PatchedConsentStageRequestToJSON = PatchedConsentStageRequestToJSON;
