// @goauthentik/api/dist/models/PatchedUserSAMLSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserSAMLSourceConnectionRequestToJSON = exports.PatchedUserSAMLSourceConnectionRequestFromJSONTyped = exports.PatchedUserSAMLSourceConnectionRequestFromJSON = exports.instanceOfPatchedUserSAMLSourceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedUserSAMLSourceConnectionRequest interface.
 */
function instanceOfPatchedUserSAMLSourceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserSAMLSourceConnectionRequest = instanceOfPatchedUserSAMLSourceConnectionRequest;
function PatchedUserSAMLSourceConnectionRequestFromJSON(json) {
    return PatchedUserSAMLSourceConnectionRequestFromJSONTyped(json, false);
}
exports.PatchedUserSAMLSourceConnectionRequestFromJSON = PatchedUserSAMLSourceConnectionRequestFromJSON;
function PatchedUserSAMLSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'identifier': !(0, runtime_1.exists)(json, 'identifier') ? undefined : json['identifier'],
    };
}
exports.PatchedUserSAMLSourceConnectionRequestFromJSONTyped = PatchedUserSAMLSourceConnectionRequestFromJSONTyped;
function PatchedUserSAMLSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
    };
}
exports.PatchedUserSAMLSourceConnectionRequestToJSON = PatchedUserSAMLSourceConnectionRequestToJSON;
