// @goauthentik/api/dist/models/PatchedSAMLPropertyMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedSAMLPropertyMappingRequestToJSON = exports.PatchedSAMLPropertyMappingRequestFromJSONTyped = exports.PatchedSAMLPropertyMappingRequestFromJSON = exports.instanceOfPatchedSAMLPropertyMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedSAMLPropertyMappingRequest interface.
 */
function instanceOfPatchedSAMLPropertyMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedSAMLPropertyMappingRequest = instanceOfPatchedSAMLPropertyMappingRequest;
function PatchedSAMLPropertyMappingRequestFromJSON(json) {
    return PatchedSAMLPropertyMappingRequestFromJSONTyped(json, false);
}
exports.PatchedSAMLPropertyMappingRequestFromJSON = PatchedSAMLPropertyMappingRequestFromJSON;
function PatchedSAMLPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
        'samlName': !(0, runtime_1.exists)(json, 'saml_name') ? undefined : json['saml_name'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}
exports.PatchedSAMLPropertyMappingRequestFromJSONTyped = PatchedSAMLPropertyMappingRequestFromJSONTyped;
function PatchedSAMLPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
exports.PatchedSAMLPropertyMappingRequestToJSON = PatchedSAMLPropertyMappingRequestToJSON;
