// @goauthentik/api/dist/models/PatchedSMSDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedSMSDeviceRequestToJSON = exports.PatchedSMSDeviceRequestFromJSONTyped = exports.PatchedSMSDeviceRequestFromJSON = exports.instanceOfPatchedSMSDeviceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedSMSDeviceRequest interface.
 */
function instanceOfPatchedSMSDeviceRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedSMSDeviceRequest = instanceOfPatchedSMSDeviceRequest;
function PatchedSMSDeviceRequestFromJSON(json) {
    return PatchedSMSDeviceRequestFromJSONTyped(json, false);
}
exports.PatchedSMSDeviceRequestFromJSON = PatchedSMSDeviceRequestFromJSON;
function PatchedSMSDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.PatchedSMSDeviceRequestFromJSONTyped = PatchedSMSDeviceRequestFromJSONTyped;
function PatchedSMSDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.PatchedSMSDeviceRequestToJSON = PatchedSMSDeviceRequestToJSON;
