// @goauthentik/api/dist/models/PatchedCaptchaStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedCaptchaStageRequestToJSON = exports.PatchedCaptchaStageRequestFromJSONTyped = exports.PatchedCaptchaStageRequestFromJSON = exports.instanceOfPatchedCaptchaStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedCaptchaStageRequest interface.
 */
function instanceOfPatchedCaptchaStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedCaptchaStageRequest = instanceOfPatchedCaptchaStageRequest;
function PatchedCaptchaStageRequestFromJSON(json) {
    return PatchedCaptchaStageRequestFromJSONTyped(json, false);
}
exports.PatchedCaptchaStageRequestFromJSON = PatchedCaptchaStageRequestFromJSON;
function PatchedCaptchaStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'publicKey': !(0, runtime_1.exists)(json, 'public_key') ? undefined : json['public_key'],
        'privateKey': !(0, runtime_1.exists)(json, 'private_key') ? undefined : json['private_key'],
        'jsUrl': !(0, runtime_1.exists)(json, 'js_url') ? undefined : json['js_url'],
        'apiUrl': !(0, runtime_1.exists)(json, 'api_url') ? undefined : json['api_url'],
    };
}
exports.PatchedCaptchaStageRequestFromJSONTyped = PatchedCaptchaStageRequestFromJSONTyped;
function PatchedCaptchaStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'public_key': value.publicKey,
        'private_key': value.privateKey,
        'js_url': value.jsUrl,
        'api_url': value.apiUrl,
    };
}
exports.PatchedCaptchaStageRequestToJSON = PatchedCaptchaStageRequestToJSON;
