// @goauthentik/api/dist/models/PatchedLicenseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedLicenseRequestToJSON = exports.PatchedLicenseRequestFromJSONTyped = exports.PatchedLicenseRequestFromJSON = exports.instanceOfPatchedLicenseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedLicenseRequest interface.
 */
function instanceOfPatchedLicenseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedLicenseRequest = instanceOfPatchedLicenseRequest;
function PatchedLicenseRequestFromJSON(json) {
    return PatchedLicenseRequestFromJSONTyped(json, false);
}
exports.PatchedLicenseRequestFromJSON = PatchedLicenseRequestFromJSON;
function PatchedLicenseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': !(0, runtime_1.exists)(json, 'key') ? undefined : json['key'],
    };
}
exports.PatchedLicenseRequestFromJSONTyped = PatchedLicenseRequestFromJSONTyped;
function PatchedLicenseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
    };
}
exports.PatchedLicenseRequestToJSON = PatchedLicenseRequestToJSON;
