// @goauthentik/api/dist/models/PatchedUserLogoutStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserLogoutStageRequestToJSON = exports.PatchedUserLogoutStageRequestFromJSONTyped = exports.PatchedUserLogoutStageRequestFromJSON = exports.instanceOfPatchedUserLogoutStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedUserLogoutStageRequest interface.
 */
function instanceOfPatchedUserLogoutStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserLogoutStageRequest = instanceOfPatchedUserLogoutStageRequest;
function PatchedUserLogoutStageRequestFromJSON(json) {
    return PatchedUserLogoutStageRequestFromJSONTyped(json, false);
}
exports.PatchedUserLogoutStageRequestFromJSON = PatchedUserLogoutStageRequestFromJSON;
function PatchedUserLogoutStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
    };
}
exports.PatchedUserLogoutStageRequestFromJSONTyped = PatchedUserLogoutStageRequestFromJSONTyped;
function PatchedUserLogoutStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
    };
}
exports.PatchedUserLogoutStageRequestToJSON = PatchedUserLogoutStageRequestToJSON;
