// @goauthentik/api/dist/models/PatchedLDAPPropertyMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedLDAPPropertyMappingRequestToJSON = exports.PatchedLDAPPropertyMappingRequestFromJSONTyped = exports.PatchedLDAPPropertyMappingRequestFromJSON = exports.instanceOfPatchedLDAPPropertyMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedLDAPPropertyMappingRequest interface.
 */
function instanceOfPatchedLDAPPropertyMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedLDAPPropertyMappingRequest = instanceOfPatchedLDAPPropertyMappingRequest;
function PatchedLDAPPropertyMappingRequestFromJSON(json) {
    return PatchedLDAPPropertyMappingRequestFromJSONTyped(json, false);
}
exports.PatchedLDAPPropertyMappingRequestFromJSON = PatchedLDAPPropertyMappingRequestFromJSON;
function PatchedLDAPPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
        'objectField': !(0, runtime_1.exists)(json, 'object_field') ? undefined : json['object_field'],
    };
}
exports.PatchedLDAPPropertyMappingRequestFromJSONTyped = PatchedLDAPPropertyMappingRequestFromJSONTyped;
function PatchedLDAPPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'object_field': value.objectField,
    };
}
exports.PatchedLDAPPropertyMappingRequestToJSON = PatchedLDAPPropertyMappingRequestToJSON;
