// @goauthentik/api/dist/models/PatchedAuthenticatorValidateStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedAuthenticatorValidateStageRequestToJSON = exports.PatchedAuthenticatorValidateStageRequestFromJSONTyped = exports.PatchedAuthenticatorValidateStageRequestFromJSON = exports.instanceOfPatchedAuthenticatorValidateStageRequest = void 0;
const runtime_1 = require("../runtime");
const DeviceClassesEnum_1 = require("./DeviceClassesEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
const NotConfiguredActionEnum_1 = require("./NotConfiguredActionEnum");
const UserVerificationEnum_1 = require("./UserVerificationEnum");
/**
 * Check if a given object implements the PatchedAuthenticatorValidateStageRequest interface.
 */
function instanceOfPatchedAuthenticatorValidateStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedAuthenticatorValidateStageRequest = instanceOfPatchedAuthenticatorValidateStageRequest;
function PatchedAuthenticatorValidateStageRequestFromJSON(json) {
    return PatchedAuthenticatorValidateStageRequestFromJSONTyped(json, false);
}
exports.PatchedAuthenticatorValidateStageRequestFromJSON = PatchedAuthenticatorValidateStageRequestFromJSON;
function PatchedAuthenticatorValidateStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'notConfiguredAction': !(0, runtime_1.exists)(json, 'not_configured_action') ? undefined : (0, NotConfiguredActionEnum_1.NotConfiguredActionEnumFromJSON)(json['not_configured_action']),
        'deviceClasses': !(0, runtime_1.exists)(json, 'device_classes') ? undefined : (json['device_classes'].map(DeviceClassesEnum_1.DeviceClassesEnumFromJSON)),
        'configurationStages': !(0, runtime_1.exists)(json, 'configuration_stages') ? undefined : json['configuration_stages'],
        'lastAuthThreshold': !(0, runtime_1.exists)(json, 'last_auth_threshold') ? undefined : json['last_auth_threshold'],
        'webauthnUserVerification': !(0, runtime_1.exists)(json, 'webauthn_user_verification') ? undefined : (0, UserVerificationEnum_1.UserVerificationEnumFromJSON)(json['webauthn_user_verification']),
    };
}
exports.PatchedAuthenticatorValidateStageRequestFromJSONTyped = PatchedAuthenticatorValidateStageRequestFromJSONTyped;
function PatchedAuthenticatorValidateStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'not_configured_action': (0, NotConfiguredActionEnum_1.NotConfiguredActionEnumToJSON)(value.notConfiguredAction),
        'device_classes': value.deviceClasses === undefined ? undefined : (value.deviceClasses.map(DeviceClassesEnum_1.DeviceClassesEnumToJSON)),
        'configuration_stages': value.configurationStages,
        'last_auth_threshold': value.lastAuthThreshold,
        'webauthn_user_verification': (0, UserVerificationEnum_1.UserVerificationEnumToJSON)(value.webauthnUserVerification),
    };
}
exports.PatchedAuthenticatorValidateStageRequestToJSON = PatchedAuthenticatorValidateStageRequestToJSON;
