// @goauthentik/api/dist/models/PatchedDomainRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedDomainRequestToJSON = exports.PatchedDomainRequestFromJSONTyped = exports.PatchedDomainRequestFromJSON = exports.instanceOfPatchedDomainRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedDomainRequest interface.
 */
function instanceOfPatchedDomainRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedDomainRequest = instanceOfPatchedDomainRequest;
function PatchedDomainRequestFromJSON(json) {
    return PatchedDomainRequestFromJSONTyped(json, false);
}
exports.PatchedDomainRequestFromJSON = PatchedDomainRequestFromJSON;
function PatchedDomainRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'domain': !(0, runtime_1.exists)(json, 'domain') ? undefined : json['domain'],
        'isPrimary': !(0, runtime_1.exists)(json, 'is_primary') ? undefined : json['is_primary'],
        'tenant': !(0, runtime_1.exists)(json, 'tenant') ? undefined : json['tenant'],
    };
}
exports.PatchedDomainRequestFromJSONTyped = PatchedDomainRequestFromJSONTyped;
function PatchedDomainRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'domain': value.domain,
        'is_primary': value.isPrimary,
        'tenant': value.tenant,
    };
}
exports.PatchedDomainRequestToJSON = PatchedDomainRequestToJSON;
