// @goauthentik/api/dist/models/PatchedNotificationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedNotificationRequestToJSON = exports.PatchedNotificationRequestFromJSONTyped = exports.PatchedNotificationRequestFromJSON = exports.instanceOfPatchedNotificationRequest = void 0;
const runtime_1 = require("../runtime");
const EventRequest_1 = require("./EventRequest");
/**
 * Check if a given object implements the PatchedNotificationRequest interface.
 */
function instanceOfPatchedNotificationRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedNotificationRequest = instanceOfPatchedNotificationRequest;
function PatchedNotificationRequestFromJSON(json) {
    return PatchedNotificationRequestFromJSONTyped(json, false);
}
exports.PatchedNotificationRequestFromJSON = PatchedNotificationRequestFromJSON;
function PatchedNotificationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'event': !(0, runtime_1.exists)(json, 'event') ? undefined : (0, EventRequest_1.EventRequestFromJSON)(json['event']),
        'seen': !(0, runtime_1.exists)(json, 'seen') ? undefined : json['seen'],
    };
}
exports.PatchedNotificationRequestFromJSONTyped = PatchedNotificationRequestFromJSONTyped;
function PatchedNotificationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'event': (0, EventRequest_1.EventRequestToJSON)(value.event),
        'seen': value.seen,
    };
}
exports.PatchedNotificationRequestToJSON = PatchedNotificationRequestToJSON;
