// @goauthentik/api/dist/models/PatchedRoleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedRoleRequestToJSON = exports.PatchedRoleRequestFromJSONTyped = exports.PatchedRoleRequestFromJSON = exports.instanceOfPatchedRoleRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedRoleRequest interface.
 */
function instanceOfPatchedRoleRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedRoleRequest = instanceOfPatchedRoleRequest;
function PatchedRoleRequestFromJSON(json) {
    return PatchedRoleRequestFromJSONTyped(json, false);
}
exports.PatchedRoleRequestFromJSON = PatchedRoleRequestFromJSON;
function PatchedRoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.PatchedRoleRequestFromJSONTyped = PatchedRoleRequestFromJSONTyped;
function PatchedRoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.PatchedRoleRequestToJSON = PatchedRoleRequestToJSON;
