// @goauthentik/api/dist/models/PatchedBrandRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedBrandRequestToJSON = exports.PatchedBrandRequestFromJSONTyped = exports.PatchedBrandRequestFromJSON = exports.instanceOfPatchedBrandRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedBrandRequest interface.
 */
function instanceOfPatchedBrandRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedBrandRequest = instanceOfPatchedBrandRequest;
function PatchedBrandRequestFromJSON(json) {
    return PatchedBrandRequestFromJSONTyped(json, false);
}
exports.PatchedBrandRequestFromJSON = PatchedBrandRequestFromJSON;
function PatchedBrandRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'domain': !(0, runtime_1.exists)(json, 'domain') ? undefined : json['domain'],
        '_default': !(0, runtime_1.exists)(json, 'default') ? undefined : json['default'],
        'brandingTitle': !(0, runtime_1.exists)(json, 'branding_title') ? undefined : json['branding_title'],
        'brandingLogo': !(0, runtime_1.exists)(json, 'branding_logo') ? undefined : json['branding_logo'],
        'brandingFavicon': !(0, runtime_1.exists)(json, 'branding_favicon') ? undefined : json['branding_favicon'],
        'flowAuthentication': !(0, runtime_1.exists)(json, 'flow_authentication') ? undefined : json['flow_authentication'],
        'flowInvalidation': !(0, runtime_1.exists)(json, 'flow_invalidation') ? undefined : json['flow_invalidation'],
        'flowRecovery': !(0, runtime_1.exists)(json, 'flow_recovery') ? undefined : json['flow_recovery'],
        'flowUnenrollment': !(0, runtime_1.exists)(json, 'flow_unenrollment') ? undefined : json['flow_unenrollment'],
        'flowUserSettings': !(0, runtime_1.exists)(json, 'flow_user_settings') ? undefined : json['flow_user_settings'],
        'flowDeviceCode': !(0, runtime_1.exists)(json, 'flow_device_code') ? undefined : json['flow_device_code'],
        'webCertificate': !(0, runtime_1.exists)(json, 'web_certificate') ? undefined : json['web_certificate'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
    };
}
exports.PatchedBrandRequestFromJSONTyped = PatchedBrandRequestFromJSONTyped;
function PatchedBrandRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'domain': value.domain,
        'default': value._default,
        'branding_title': value.brandingTitle,
        'branding_logo': value.brandingLogo,
        'branding_favicon': value.brandingFavicon,
        'flow_authentication': value.flowAuthentication,
        'flow_invalidation': value.flowInvalidation,
        'flow_recovery': value.flowRecovery,
        'flow_unenrollment': value.flowUnenrollment,
        'flow_user_settings': value.flowUserSettings,
        'flow_device_code': value.flowDeviceCode,
        'web_certificate': value.webCertificate,
        'attributes': value.attributes,
    };
}
exports.PatchedBrandRequestToJSON = PatchedBrandRequestToJSON;
