// @goauthentik/api/dist/models/PatchedWebAuthnDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedWebAuthnDeviceRequestToJSON = exports.PatchedWebAuthnDeviceRequestFromJSONTyped = exports.PatchedWebAuthnDeviceRequestFromJSON = exports.instanceOfPatchedWebAuthnDeviceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedWebAuthnDeviceRequest interface.
 */
function instanceOfPatchedWebAuthnDeviceRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedWebAuthnDeviceRequest = instanceOfPatchedWebAuthnDeviceRequest;
function PatchedWebAuthnDeviceRequestFromJSON(json) {
    return PatchedWebAuthnDeviceRequestFromJSONTyped(json, false);
}
exports.PatchedWebAuthnDeviceRequestFromJSON = PatchedWebAuthnDeviceRequestFromJSON;
function PatchedWebAuthnDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.PatchedWebAuthnDeviceRequestFromJSONTyped = PatchedWebAuthnDeviceRequestFromJSONTyped;
function PatchedWebAuthnDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.PatchedWebAuthnDeviceRequestToJSON = PatchedWebAuthnDeviceRequestToJSON;
