// @goauthentik/api/dist/models/PatchedPromptRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPromptRequestToJSON = exports.PatchedPromptRequestFromJSONTyped = exports.PatchedPromptRequestFromJSON = exports.instanceOfPatchedPromptRequest = void 0;
const runtime_1 = require("../runtime");
const PromptTypeEnum_1 = require("./PromptTypeEnum");
const StageRequest_1 = require("./StageRequest");
/**
 * Check if a given object implements the PatchedPromptRequest interface.
 */
function instanceOfPatchedPromptRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPromptRequest = instanceOfPatchedPromptRequest;
function PatchedPromptRequestFromJSON(json) {
    return PatchedPromptRequestFromJSONTyped(json, false);
}
exports.PatchedPromptRequestFromJSON = PatchedPromptRequestFromJSON;
function PatchedPromptRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'fieldKey': !(0, runtime_1.exists)(json, 'field_key') ? undefined : json['field_key'],
        'label': !(0, runtime_1.exists)(json, 'label') ? undefined : json['label'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : (0, PromptTypeEnum_1.PromptTypeEnumFromJSON)(json['type']),
        'required': !(0, runtime_1.exists)(json, 'required') ? undefined : json['required'],
        'placeholder': !(0, runtime_1.exists)(json, 'placeholder') ? undefined : json['placeholder'],
        'initialValue': !(0, runtime_1.exists)(json, 'initial_value') ? undefined : json['initial_value'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'promptstageSet': !(0, runtime_1.exists)(json, 'promptstage_set') ? undefined : (json['promptstage_set'].map(StageRequest_1.StageRequestFromJSON)),
        'subText': !(0, runtime_1.exists)(json, 'sub_text') ? undefined : json['sub_text'],
        'placeholderExpression': !(0, runtime_1.exists)(json, 'placeholder_expression') ? undefined : json['placeholder_expression'],
        'initialValueExpression': !(0, runtime_1.exists)(json, 'initial_value_expression') ? undefined : json['initial_value_expression'],
    };
}
exports.PatchedPromptRequestFromJSONTyped = PatchedPromptRequestFromJSONTyped;
function PatchedPromptRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'field_key': value.fieldKey,
        'label': value.label,
        'type': (0, PromptTypeEnum_1.PromptTypeEnumToJSON)(value.type),
        'required': value.required,
        'placeholder': value.placeholder,
        'initial_value': value.initialValue,
        'order': value.order,
        'promptstage_set': value.promptstageSet === undefined ? undefined : (value.promptstageSet.map(StageRequest_1.StageRequestToJSON)),
        'sub_text': value.subText,
        'placeholder_expression': value.placeholderExpression,
        'initial_value_expression': value.initialValueExpression,
    };
}
exports.PatchedPromptRequestToJSON = PatchedPromptRequestToJSON;
