// @goauthentik/api/dist/models/PatchedConnectionTokenRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedConnectionTokenRequestToJSON = exports.PatchedConnectionTokenRequestFromJSONTyped = exports.PatchedConnectionTokenRequestFromJSON = exports.instanceOfPatchedConnectionTokenRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedConnectionTokenRequest interface.
 */
function instanceOfPatchedConnectionTokenRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedConnectionTokenRequest = instanceOfPatchedConnectionTokenRequest;
function PatchedConnectionTokenRequestFromJSON(json) {
    return PatchedConnectionTokenRequestFromJSONTyped(json, false);
}
exports.PatchedConnectionTokenRequestFromJSON = PatchedConnectionTokenRequestFromJSON;
function PatchedConnectionTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'provider': !(0, runtime_1.exists)(json, 'provider') ? undefined : json['provider'],
    };
}
exports.PatchedConnectionTokenRequestFromJSONTyped = PatchedConnectionTokenRequestFromJSONTyped;
function PatchedConnectionTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'provider': value.provider,
    };
}
exports.PatchedConnectionTokenRequestToJSON = PatchedConnectionTokenRequestToJSON;
