// @goauthentik/api/dist/models/PatchedDockerServiceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedDockerServiceConnectionRequestToJSON = exports.PatchedDockerServiceConnectionRequestFromJSONTyped = exports.PatchedDockerServiceConnectionRequestFromJSON = exports.instanceOfPatchedDockerServiceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedDockerServiceConnectionRequest interface.
 */
function instanceOfPatchedDockerServiceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedDockerServiceConnectionRequest = instanceOfPatchedDockerServiceConnectionRequest;
function PatchedDockerServiceConnectionRequestFromJSON(json) {
    return PatchedDockerServiceConnectionRequestFromJSONTyped(json, false);
}
exports.PatchedDockerServiceConnectionRequestFromJSON = PatchedDockerServiceConnectionRequestFromJSON;
function PatchedDockerServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'url': !(0, runtime_1.exists)(json, 'url') ? undefined : json['url'],
        'tlsVerification': !(0, runtime_1.exists)(json, 'tls_verification') ? undefined : json['tls_verification'],
        'tlsAuthentication': !(0, runtime_1.exists)(json, 'tls_authentication') ? undefined : json['tls_authentication'],
    };
}
exports.PatchedDockerServiceConnectionRequestFromJSONTyped = PatchedDockerServiceConnectionRequestFromJSONTyped;
function PatchedDockerServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'url': value.url,
        'tls_verification': value.tlsVerification,
        'tls_authentication': value.tlsAuthentication,
    };
}
exports.PatchedDockerServiceConnectionRequestToJSON = PatchedDockerServiceConnectionRequestToJSON;
