// @goauthentik/api/dist/models/PatchedEventMatcherPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedEventMatcherPolicyRequestToJSON = exports.PatchedEventMatcherPolicyRequestFromJSONTyped = exports.PatchedEventMatcherPolicyRequestFromJSON = exports.instanceOfPatchedEventMatcherPolicyRequest = void 0;
const runtime_1 = require("../runtime");
const AppEnum_1 = require("./AppEnum");
const EventActions_1 = require("./EventActions");
const ModelEnum_1 = require("./ModelEnum");
/**
 * Check if a given object implements the PatchedEventMatcherPolicyRequest interface.
 */
function instanceOfPatchedEventMatcherPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedEventMatcherPolicyRequest = instanceOfPatchedEventMatcherPolicyRequest;
function PatchedEventMatcherPolicyRequestFromJSON(json) {
    return PatchedEventMatcherPolicyRequestFromJSONTyped(json, false);
}
exports.PatchedEventMatcherPolicyRequestFromJSON = PatchedEventMatcherPolicyRequestFromJSON;
function PatchedEventMatcherPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'action': !(0, runtime_1.exists)(json, 'action') ? undefined : (0, EventActions_1.EventActionsFromJSON)(json['action']),
        'clientIp': !(0, runtime_1.exists)(json, 'client_ip') ? undefined : json['client_ip'],
        'app': !(0, runtime_1.exists)(json, 'app') ? undefined : (0, AppEnum_1.AppEnumFromJSON)(json['app']),
        'model': !(0, runtime_1.exists)(json, 'model') ? undefined : (0, ModelEnum_1.ModelEnumFromJSON)(json['model']),
    };
}
exports.PatchedEventMatcherPolicyRequestFromJSONTyped = PatchedEventMatcherPolicyRequestFromJSONTyped;
function PatchedEventMatcherPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'action': (0, EventActions_1.EventActionsToJSON)(value.action),
        'client_ip': value.clientIp,
        'app': (0, AppEnum_1.AppEnumToJSON)(value.app),
        'model': (0, ModelEnum_1.ModelEnumToJSON)(value.model),
    };
}
exports.PatchedEventMatcherPolicyRequestToJSON = PatchedEventMatcherPolicyRequestToJSON;
