// @goauthentik/api/dist/models/PatchedFlowStageBindingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedFlowStageBindingRequestToJSON = exports.PatchedFlowStageBindingRequestFromJSONTyped = exports.PatchedFlowStageBindingRequestFromJSON = exports.instanceOfPatchedFlowStageBindingRequest = void 0;
const runtime_1 = require("../runtime");
const InvalidResponseActionEnum_1 = require("./InvalidResponseActionEnum");
const PolicyEngineMode_1 = require("./PolicyEngineMode");
/**
 * Check if a given object implements the PatchedFlowStageBindingRequest interface.
 */
function instanceOfPatchedFlowStageBindingRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedFlowStageBindingRequest = instanceOfPatchedFlowStageBindingRequest;
function PatchedFlowStageBindingRequestFromJSON(json) {
    return PatchedFlowStageBindingRequestFromJSONTyped(json, false);
}
exports.PatchedFlowStageBindingRequestFromJSON = PatchedFlowStageBindingRequestFromJSON;
function PatchedFlowStageBindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'target': !(0, runtime_1.exists)(json, 'target') ? undefined : json['target'],
        'stage': !(0, runtime_1.exists)(json, 'stage') ? undefined : json['stage'],
        'evaluateOnPlan': !(0, runtime_1.exists)(json, 'evaluate_on_plan') ? undefined : json['evaluate_on_plan'],
        'reEvaluatePolicies': !(0, runtime_1.exists)(json, 're_evaluate_policies') ? undefined : json['re_evaluate_policies'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'policyEngineMode': !(0, runtime_1.exists)(json, 'policy_engine_mode') ? undefined : (0, PolicyEngineMode_1.PolicyEngineModeFromJSON)(json['policy_engine_mode']),
        'invalidResponseAction': !(0, runtime_1.exists)(json, 'invalid_response_action') ? undefined : (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumFromJSON)(json['invalid_response_action']),
    };
}
exports.PatchedFlowStageBindingRequestFromJSONTyped = PatchedFlowStageBindingRequestFromJSONTyped;
function PatchedFlowStageBindingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'target': value.target,
        'stage': value.stage,
        'evaluate_on_plan': value.evaluateOnPlan,
        're_evaluate_policies': value.reEvaluatePolicies,
        'order': value.order,
        'policy_engine_mode': (0, PolicyEngineMode_1.PolicyEngineModeToJSON)(value.policyEngineMode),
        'invalid_response_action': (0, InvalidResponseActionEnum_1.InvalidResponseActionEnumToJSON)(value.invalidResponseAction),
    };
}
exports.PatchedFlowStageBindingRequestToJSON = PatchedFlowStageBindingRequestToJSON;
