// @goauthentik/api/dist/models/PatchedUserWriteStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserWriteStageRequestToJSON = exports.PatchedUserWriteStageRequestFromJSONTyped = exports.PatchedUserWriteStageRequestFromJSON = exports.instanceOfPatchedUserWriteStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
const UserCreationModeEnum_1 = require("./UserCreationModeEnum");
const UserTypeEnum_1 = require("./UserTypeEnum");
/**
 * Check if a given object implements the PatchedUserWriteStageRequest interface.
 */
function instanceOfPatchedUserWriteStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserWriteStageRequest = instanceOfPatchedUserWriteStageRequest;
function PatchedUserWriteStageRequestFromJSON(json) {
    return PatchedUserWriteStageRequestFromJSONTyped(json, false);
}
exports.PatchedUserWriteStageRequestFromJSON = PatchedUserWriteStageRequestFromJSON;
function PatchedUserWriteStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'userCreationMode': !(0, runtime_1.exists)(json, 'user_creation_mode') ? undefined : (0, UserCreationModeEnum_1.UserCreationModeEnumFromJSON)(json['user_creation_mode']),
        'createUsersAsInactive': !(0, runtime_1.exists)(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !(0, runtime_1.exists)(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !(0, runtime_1.exists)(json, 'user_type') ? undefined : (0, UserTypeEnum_1.UserTypeEnumFromJSON)(json['user_type']),
        'userPathTemplate': !(0, runtime_1.exists)(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}
exports.PatchedUserWriteStageRequestFromJSONTyped = PatchedUserWriteStageRequestFromJSONTyped;
function PatchedUserWriteStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'user_creation_mode': (0, UserCreationModeEnum_1.UserCreationModeEnumToJSON)(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': (0, UserTypeEnum_1.UserTypeEnumToJSON)(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
exports.PatchedUserWriteStageRequestToJSON = PatchedUserWriteStageRequestToJSON;
