// @goauthentik/api/dist/models/PatchedUserRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserRequestToJSON = exports.PatchedUserRequestFromJSONTyped = exports.PatchedUserRequestFromJSON = exports.instanceOfPatchedUserRequest = void 0;
const runtime_1 = require("../runtime");
const UserTypeEnum_1 = require("./UserTypeEnum");
/**
 * Check if a given object implements the PatchedUserRequest interface.
 */
function instanceOfPatchedUserRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserRequest = instanceOfPatchedUserRequest;
function PatchedUserRequestFromJSON(json) {
    return PatchedUserRequestFromJSONTyped(json, false);
}
exports.PatchedUserRequestFromJSON = PatchedUserRequestFromJSON;
function PatchedUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'isActive': !(0, runtime_1.exists)(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !(0, runtime_1.exists)(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'groups': !(0, runtime_1.exists)(json, 'groups') ? undefined : json['groups'],
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : (0, UserTypeEnum_1.UserTypeEnumFromJSON)(json['type']),
    };
}
exports.PatchedUserRequestFromJSONTyped = PatchedUserRequestFromJSONTyped;
function PatchedUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'groups': value.groups,
        'email': value.email,
        'attributes': value.attributes,
        'path': value.path,
        'type': (0, UserTypeEnum_1.UserTypeEnumToJSON)(value.type),
    };
}
exports.PatchedUserRequestToJSON = PatchedUserRequestToJSON;
