// @goauthentik/api/dist/models/PatchedExpressionPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedExpressionPolicyRequestToJSON = exports.PatchedExpressionPolicyRequestFromJSONTyped = exports.PatchedExpressionPolicyRequestFromJSON = exports.instanceOfPatchedExpressionPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedExpressionPolicyRequest interface.
 */
function instanceOfPatchedExpressionPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedExpressionPolicyRequest = instanceOfPatchedExpressionPolicyRequest;
function PatchedExpressionPolicyRequestFromJSON(json) {
    return PatchedExpressionPolicyRequestFromJSONTyped(json, false);
}
exports.PatchedExpressionPolicyRequestFromJSON = PatchedExpressionPolicyRequestFromJSON;
function PatchedExpressionPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
    };
}
exports.PatchedExpressionPolicyRequestFromJSONTyped = PatchedExpressionPolicyRequestFromJSONTyped;
function PatchedExpressionPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'expression': value.expression,
    };
}
exports.PatchedExpressionPolicyRequestToJSON = PatchedExpressionPolicyRequestToJSON;
