// @goauthentik/api/dist/models/PatchedUserLoginStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserLoginStageRequestToJSON = exports.PatchedUserLoginStageRequestFromJSONTyped = exports.PatchedUserLoginStageRequestFromJSON = exports.instanceOfPatchedUserLoginStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
const GeoipBindingEnum_1 = require("./GeoipBindingEnum");
const NetworkBindingEnum_1 = require("./NetworkBindingEnum");
/**
 * Check if a given object implements the PatchedUserLoginStageRequest interface.
 */
function instanceOfPatchedUserLoginStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserLoginStageRequest = instanceOfPatchedUserLoginStageRequest;
function PatchedUserLoginStageRequestFromJSON(json) {
    return PatchedUserLoginStageRequestFromJSONTyped(json, false);
}
exports.PatchedUserLoginStageRequestFromJSON = PatchedUserLoginStageRequestFromJSON;
function PatchedUserLoginStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'sessionDuration': !(0, runtime_1.exists)(json, 'session_duration') ? undefined : json['session_duration'],
        'terminateOtherSessions': !(0, runtime_1.exists)(json, 'terminate_other_sessions') ? undefined : json['terminate_other_sessions'],
        'rememberMeOffset': !(0, runtime_1.exists)(json, 'remember_me_offset') ? undefined : json['remember_me_offset'],
        'networkBinding': !(0, runtime_1.exists)(json, 'network_binding') ? undefined : (0, NetworkBindingEnum_1.NetworkBindingEnumFromJSON)(json['network_binding']),
        'geoipBinding': !(0, runtime_1.exists)(json, 'geoip_binding') ? undefined : (0, GeoipBindingEnum_1.GeoipBindingEnumFromJSON)(json['geoip_binding']),
    };
}
exports.PatchedUserLoginStageRequestFromJSONTyped = PatchedUserLoginStageRequestFromJSONTyped;
function PatchedUserLoginStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'session_duration': value.sessionDuration,
        'terminate_other_sessions': value.terminateOtherSessions,
        'remember_me_offset': value.rememberMeOffset,
        'network_binding': (0, NetworkBindingEnum_1.NetworkBindingEnumToJSON)(value.networkBinding),
        'geoip_binding': (0, GeoipBindingEnum_1.GeoipBindingEnumToJSON)(value.geoipBinding),
    };
}
exports.PatchedUserLoginStageRequestToJSON = PatchedUserLoginStageRequestToJSON;
