// @goauthentik/api/dist/models/PatchedUserOAuthSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedUserOAuthSourceConnectionRequestToJSON = exports.PatchedUserOAuthSourceConnectionRequestFromJSONTyped = exports.PatchedUserOAuthSourceConnectionRequestFromJSON = exports.instanceOfPatchedUserOAuthSourceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedUserOAuthSourceConnectionRequest interface.
 */
function instanceOfPatchedUserOAuthSourceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedUserOAuthSourceConnectionRequest = instanceOfPatchedUserOAuthSourceConnectionRequest;
function PatchedUserOAuthSourceConnectionRequestFromJSON(json) {
    return PatchedUserOAuthSourceConnectionRequestFromJSONTyped(json, false);
}
exports.PatchedUserOAuthSourceConnectionRequestFromJSON = PatchedUserOAuthSourceConnectionRequestFromJSON;
function PatchedUserOAuthSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'identifier': !(0, runtime_1.exists)(json, 'identifier') ? undefined : json['identifier'],
        'accessToken': !(0, runtime_1.exists)(json, 'access_token') ? undefined : json['access_token'],
    };
}
exports.PatchedUserOAuthSourceConnectionRequestFromJSONTyped = PatchedUserOAuthSourceConnectionRequestFromJSONTyped;
function PatchedUserOAuthSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
        'access_token': value.accessToken,
    };
}
exports.PatchedUserOAuthSourceConnectionRequestToJSON = PatchedUserOAuthSourceConnectionRequestToJSON;
