// @goauthentik/api/dist/models/PatchedScopeMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedScopeMappingRequestToJSON = exports.PatchedScopeMappingRequestFromJSONTyped = exports.PatchedScopeMappingRequestFromJSON = exports.instanceOfPatchedScopeMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedScopeMappingRequest interface.
 */
function instanceOfPatchedScopeMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedScopeMappingRequest = instanceOfPatchedScopeMappingRequest;
function PatchedScopeMappingRequestFromJSON(json) {
    return PatchedScopeMappingRequestFromJSONTyped(json, false);
}
exports.PatchedScopeMappingRequestFromJSON = PatchedScopeMappingRequestFromJSON;
function PatchedScopeMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
        'scopeName': !(0, runtime_1.exists)(json, 'scope_name') ? undefined : json['scope_name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
    };
}
exports.PatchedScopeMappingRequestFromJSONTyped = PatchedScopeMappingRequestFromJSONTyped;
function PatchedScopeMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
exports.PatchedScopeMappingRequestToJSON = PatchedScopeMappingRequestToJSON;
