// @goauthentik/api/dist/models/PatchedNotificationRuleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedNotificationRuleRequestToJSON = exports.PatchedNotificationRuleRequestFromJSONTyped = exports.PatchedNotificationRuleRequestFromJSON = exports.instanceOfPatchedNotificationRuleRequest = void 0;
const runtime_1 = require("../runtime");
const SeverityEnum_1 = require("./SeverityEnum");
/**
 * Check if a given object implements the PatchedNotificationRuleRequest interface.
 */
function instanceOfPatchedNotificationRuleRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedNotificationRuleRequest = instanceOfPatchedNotificationRuleRequest;
function PatchedNotificationRuleRequestFromJSON(json) {
    return PatchedNotificationRuleRequestFromJSONTyped(json, false);
}
exports.PatchedNotificationRuleRequestFromJSON = PatchedNotificationRuleRequestFromJSON;
function PatchedNotificationRuleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'transports': !(0, runtime_1.exists)(json, 'transports') ? undefined : json['transports'],
        'severity': !(0, runtime_1.exists)(json, 'severity') ? undefined : (0, SeverityEnum_1.SeverityEnumFromJSON)(json['severity']),
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
    };
}
exports.PatchedNotificationRuleRequestFromJSONTyped = PatchedNotificationRuleRequestFromJSONTyped;
function PatchedNotificationRuleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'transports': value.transports,
        'severity': (0, SeverityEnum_1.SeverityEnumToJSON)(value.severity),
        'group': value.group,
    };
}
exports.PatchedNotificationRuleRequestToJSON = PatchedNotificationRuleRequestToJSON;
