// @goauthentik/api/dist/models/PatchedTOTPDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedTOTPDeviceRequestToJSON = exports.PatchedTOTPDeviceRequestFromJSONTyped = exports.PatchedTOTPDeviceRequestFromJSON = exports.instanceOfPatchedTOTPDeviceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedTOTPDeviceRequest interface.
 */
function instanceOfPatchedTOTPDeviceRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedTOTPDeviceRequest = instanceOfPatchedTOTPDeviceRequest;
function PatchedTOTPDeviceRequestFromJSON(json) {
    return PatchedTOTPDeviceRequestFromJSONTyped(json, false);
}
exports.PatchedTOTPDeviceRequestFromJSON = PatchedTOTPDeviceRequestFromJSON;
function PatchedTOTPDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.PatchedTOTPDeviceRequestFromJSONTyped = PatchedTOTPDeviceRequestFromJSONTyped;
function PatchedTOTPDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.PatchedTOTPDeviceRequestToJSON = PatchedTOTPDeviceRequestToJSON;
