// @goauthentik/api/dist/models/PatchedDummyPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedDummyPolicyRequestToJSON = exports.PatchedDummyPolicyRequestFromJSONTyped = exports.PatchedDummyPolicyRequestFromJSON = exports.instanceOfPatchedDummyPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedDummyPolicyRequest interface.
 */
function instanceOfPatchedDummyPolicyRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedDummyPolicyRequest = instanceOfPatchedDummyPolicyRequest;
function PatchedDummyPolicyRequestFromJSON(json) {
    return PatchedDummyPolicyRequestFromJSONTyped(json, false);
}
exports.PatchedDummyPolicyRequestFromJSON = PatchedDummyPolicyRequestFromJSON;
function PatchedDummyPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'result': !(0, runtime_1.exists)(json, 'result') ? undefined : json['result'],
        'waitMin': !(0, runtime_1.exists)(json, 'wait_min') ? undefined : json['wait_min'],
        'waitMax': !(0, runtime_1.exists)(json, 'wait_max') ? undefined : json['wait_max'],
    };
}
exports.PatchedDummyPolicyRequestFromJSONTyped = PatchedDummyPolicyRequestFromJSONTyped;
function PatchedDummyPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'result': value.result,
        'wait_min': value.waitMin,
        'wait_max': value.waitMax,
    };
}
exports.PatchedDummyPolicyRequestToJSON = PatchedDummyPolicyRequestToJSON;
