// @goauthentik/api/dist/models/PatchedTenantRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedTenantRequestToJSON = exports.PatchedTenantRequestFromJSONTyped = exports.PatchedTenantRequestFromJSON = exports.instanceOfPatchedTenantRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedTenantRequest interface.
 */
function instanceOfPatchedTenantRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedTenantRequest = instanceOfPatchedTenantRequest;
function PatchedTenantRequestFromJSON(json) {
    return PatchedTenantRequestFromJSONTyped(json, false);
}
exports.PatchedTenantRequestFromJSON = PatchedTenantRequestFromJSON;
function PatchedTenantRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'schemaName': !(0, runtime_1.exists)(json, 'schema_name') ? undefined : json['schema_name'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'ready': !(0, runtime_1.exists)(json, 'ready') ? undefined : json['ready'],
    };
}
exports.PatchedTenantRequestFromJSONTyped = PatchedTenantRequestFromJSONTyped;
function PatchedTenantRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
exports.PatchedTenantRequestToJSON = PatchedTenantRequestToJSON;
