// @goauthentik/api/dist/models/PatchedSCIMProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedSCIMProviderRequestToJSON = exports.PatchedSCIMProviderRequestFromJSONTyped = exports.PatchedSCIMProviderRequestFromJSON = exports.instanceOfPatchedSCIMProviderRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedSCIMProviderRequest interface.
 */
function instanceOfPatchedSCIMProviderRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedSCIMProviderRequest = instanceOfPatchedSCIMProviderRequest;
function PatchedSCIMProviderRequestFromJSON(json) {
    return PatchedSCIMProviderRequestFromJSONTyped(json, false);
}
exports.PatchedSCIMProviderRequestFromJSON = PatchedSCIMProviderRequestFromJSON;
function PatchedSCIMProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !(0, runtime_1.exists)(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'url': !(0, runtime_1.exists)(json, 'url') ? undefined : json['url'],
        'token': !(0, runtime_1.exists)(json, 'token') ? undefined : json['token'],
        'excludeUsersServiceAccount': !(0, runtime_1.exists)(json, 'exclude_users_service_account') ? undefined : json['exclude_users_service_account'],
        'filterGroup': !(0, runtime_1.exists)(json, 'filter_group') ? undefined : json['filter_group'],
    };
}
exports.PatchedSCIMProviderRequestFromJSONTyped = PatchedSCIMProviderRequestFromJSONTyped;
function PatchedSCIMProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
        'url': value.url,
        'token': value.token,
        'exclude_users_service_account': value.excludeUsersServiceAccount,
        'filter_group': value.filterGroup,
    };
}
exports.PatchedSCIMProviderRequestToJSON = PatchedSCIMProviderRequestToJSON;
