// @goauthentik/api/dist/models/PatchedPasswordStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPasswordStageRequestToJSON = exports.PatchedPasswordStageRequestFromJSONTyped = exports.PatchedPasswordStageRequestFromJSON = exports.instanceOfPatchedPasswordStageRequest = void 0;
const runtime_1 = require("../runtime");
const BackendsEnum_1 = require("./BackendsEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedPasswordStageRequest interface.
 */
function instanceOfPatchedPasswordStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPasswordStageRequest = instanceOfPatchedPasswordStageRequest;
function PatchedPasswordStageRequestFromJSON(json) {
    return PatchedPasswordStageRequestFromJSONTyped(json, false);
}
exports.PatchedPasswordStageRequestFromJSON = PatchedPasswordStageRequestFromJSON;
function PatchedPasswordStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'backends': !(0, runtime_1.exists)(json, 'backends') ? undefined : (json['backends'].map(BackendsEnum_1.BackendsEnumFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'failedAttemptsBeforeCancel': !(0, runtime_1.exists)(json, 'failed_attempts_before_cancel') ? undefined : json['failed_attempts_before_cancel'],
    };
}
exports.PatchedPasswordStageRequestFromJSONTyped = PatchedPasswordStageRequestFromJSONTyped;
function PatchedPasswordStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'backends': value.backends === undefined ? undefined : (value.backends.map(BackendsEnum_1.BackendsEnumToJSON)),
        'configure_flow': value.configureFlow,
        'failed_attempts_before_cancel': value.failedAttemptsBeforeCancel,
    };
}
exports.PatchedPasswordStageRequestToJSON = PatchedPasswordStageRequestToJSON;
