// @goauthentik/api/dist/models/PatchedLDAPProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedLDAPProviderRequestToJSON = exports.PatchedLDAPProviderRequestFromJSONTyped = exports.PatchedLDAPProviderRequestFromJSON = exports.instanceOfPatchedLDAPProviderRequest = void 0;
const runtime_1 = require("../runtime");
const LDAPAPIAccessMode_1 = require("./LDAPAPIAccessMode");
/**
 * Check if a given object implements the PatchedLDAPProviderRequest interface.
 */
function instanceOfPatchedLDAPProviderRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedLDAPProviderRequest = instanceOfPatchedLDAPProviderRequest;
function PatchedLDAPProviderRequestFromJSON(json) {
    return PatchedLDAPProviderRequestFromJSONTyped(json, false);
}
exports.PatchedLDAPProviderRequestFromJSON = PatchedLDAPProviderRequestFromJSON;
function PatchedLDAPProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': !(0, runtime_1.exists)(json, 'authorization_flow') ? undefined : json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'baseDn': !(0, runtime_1.exists)(json, 'base_dn') ? undefined : json['base_dn'],
        'searchGroup': !(0, runtime_1.exists)(json, 'search_group') ? undefined : json['search_group'],
        'certificate': !(0, runtime_1.exists)(json, 'certificate') ? undefined : json['certificate'],
        'tlsServerName': !(0, runtime_1.exists)(json, 'tls_server_name') ? undefined : json['tls_server_name'],
        'uidStartNumber': !(0, runtime_1.exists)(json, 'uid_start_number') ? undefined : json['uid_start_number'],
        'gidStartNumber': !(0, runtime_1.exists)(json, 'gid_start_number') ? undefined : json['gid_start_number'],
        'searchMode': !(0, runtime_1.exists)(json, 'search_mode') ? undefined : (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeFromJSON)(json['search_mode']),
        'bindMode': !(0, runtime_1.exists)(json, 'bind_mode') ? undefined : (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeFromJSON)(json['bind_mode']),
        'mfaSupport': !(0, runtime_1.exists)(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}
exports.PatchedLDAPProviderRequestFromJSONTyped = PatchedLDAPProviderRequestFromJSONTyped;
function PatchedLDAPProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'base_dn': value.baseDn,
        'search_group': value.searchGroup,
        'certificate': value.certificate,
        'tls_server_name': value.tlsServerName,
        'uid_start_number': value.uidStartNumber,
        'gid_start_number': value.gidStartNumber,
        'search_mode': (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeToJSON)(value.searchMode),
        'bind_mode': (0, LDAPAPIAccessMode_1.LDAPAPIAccessModeToJSON)(value.bindMode),
        'mfa_support': value.mfaSupport,
    };
}
exports.PatchedLDAPProviderRequestToJSON = PatchedLDAPProviderRequestToJSON;
