// @goauthentik/api/dist/models/PatchedBlueprintInstanceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedBlueprintInstanceRequestToJSON = exports.PatchedBlueprintInstanceRequestFromJSONTyped = exports.PatchedBlueprintInstanceRequestFromJSON = exports.instanceOfPatchedBlueprintInstanceRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedBlueprintInstanceRequest interface.
 */
function instanceOfPatchedBlueprintInstanceRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedBlueprintInstanceRequest = instanceOfPatchedBlueprintInstanceRequest;
function PatchedBlueprintInstanceRequestFromJSON(json) {
    return PatchedBlueprintInstanceRequestFromJSONTyped(json, false);
}
exports.PatchedBlueprintInstanceRequestFromJSON = PatchedBlueprintInstanceRequestFromJSON;
function PatchedBlueprintInstanceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'context': !(0, runtime_1.exists)(json, 'context') ? undefined : json['context'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'content': !(0, runtime_1.exists)(json, 'content') ? undefined : json['content'],
    };
}
exports.PatchedBlueprintInstanceRequestFromJSONTyped = PatchedBlueprintInstanceRequestFromJSONTyped;
function PatchedBlueprintInstanceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'path': value.path,
        'context': value.context,
        'enabled': value.enabled,
        'content': value.content,
    };
}
exports.PatchedBlueprintInstanceRequestToJSON = PatchedBlueprintInstanceRequestToJSON;
