// @goauthentik/api/dist/models/PatchedCertificateKeyPairRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedCertificateKeyPairRequestToJSON = exports.PatchedCertificateKeyPairRequestFromJSONTyped = exports.PatchedCertificateKeyPairRequestFromJSON = exports.instanceOfPatchedCertificateKeyPairRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedCertificateKeyPairRequest interface.
 */
function instanceOfPatchedCertificateKeyPairRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedCertificateKeyPairRequest = instanceOfPatchedCertificateKeyPairRequest;
function PatchedCertificateKeyPairRequestFromJSON(json) {
    return PatchedCertificateKeyPairRequestFromJSONTyped(json, false);
}
exports.PatchedCertificateKeyPairRequestFromJSON = PatchedCertificateKeyPairRequestFromJSON;
function PatchedCertificateKeyPairRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'certificateData': !(0, runtime_1.exists)(json, 'certificate_data') ? undefined : json['certificate_data'],
        'keyData': !(0, runtime_1.exists)(json, 'key_data') ? undefined : json['key_data'],
    };
}
exports.PatchedCertificateKeyPairRequestFromJSONTyped = PatchedCertificateKeyPairRequestFromJSONTyped;
function PatchedCertificateKeyPairRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'certificate_data': value.certificateData,
        'key_data': value.keyData,
    };
}
exports.PatchedCertificateKeyPairRequestToJSON = PatchedCertificateKeyPairRequestToJSON;
