// @goauthentik/api/dist/models/PatchedEndpointRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedEndpointRequestToJSON = exports.PatchedEndpointRequestFromJSONTyped = exports.PatchedEndpointRequestFromJSON = exports.instanceOfPatchedEndpointRequest = void 0;
const runtime_1 = require("../runtime");
const AuthModeEnum_1 = require("./AuthModeEnum");
const ProtocolEnum_1 = require("./ProtocolEnum");
/**
 * Check if a given object implements the PatchedEndpointRequest interface.
 */
function instanceOfPatchedEndpointRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedEndpointRequest = instanceOfPatchedEndpointRequest;
function PatchedEndpointRequestFromJSON(json) {
    return PatchedEndpointRequestFromJSONTyped(json, false);
}
exports.PatchedEndpointRequestFromJSON = PatchedEndpointRequestFromJSON;
function PatchedEndpointRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'provider': !(0, runtime_1.exists)(json, 'provider') ? undefined : json['provider'],
        'protocol': !(0, runtime_1.exists)(json, 'protocol') ? undefined : (0, ProtocolEnum_1.ProtocolEnumFromJSON)(json['protocol']),
        'host': !(0, runtime_1.exists)(json, 'host') ? undefined : json['host'],
        'settings': !(0, runtime_1.exists)(json, 'settings') ? undefined : json['settings'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'authMode': !(0, runtime_1.exists)(json, 'auth_mode') ? undefined : (0, AuthModeEnum_1.AuthModeEnumFromJSON)(json['auth_mode']),
        'maximumConnections': !(0, runtime_1.exists)(json, 'maximum_connections') ? undefined : json['maximum_connections'],
    };
}
exports.PatchedEndpointRequestFromJSONTyped = PatchedEndpointRequestFromJSONTyped;
function PatchedEndpointRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'provider': value.provider,
        'protocol': (0, ProtocolEnum_1.ProtocolEnumToJSON)(value.protocol),
        'host': value.host,
        'settings': value.settings,
        'property_mappings': value.propertyMappings,
        'auth_mode': (0, AuthModeEnum_1.AuthModeEnumToJSON)(value.authMode),
        'maximum_connections': value.maximumConnections,
    };
}
exports.PatchedEndpointRequestToJSON = PatchedEndpointRequestToJSON;
