// @goauthentik/api/dist/models/PatchedNotificationWebhookMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedNotificationWebhookMappingRequestToJSON = exports.PatchedNotificationWebhookMappingRequestFromJSONTyped = exports.PatchedNotificationWebhookMappingRequestFromJSON = exports.instanceOfPatchedNotificationWebhookMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedNotificationWebhookMappingRequest interface.
 */
function instanceOfPatchedNotificationWebhookMappingRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedNotificationWebhookMappingRequest = instanceOfPatchedNotificationWebhookMappingRequest;
function PatchedNotificationWebhookMappingRequestFromJSON(json) {
    return PatchedNotificationWebhookMappingRequestFromJSONTyped(json, false);
}
exports.PatchedNotificationWebhookMappingRequestFromJSON = PatchedNotificationWebhookMappingRequestFromJSON;
function PatchedNotificationWebhookMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
    };
}
exports.PatchedNotificationWebhookMappingRequestFromJSONTyped = PatchedNotificationWebhookMappingRequestFromJSONTyped;
function PatchedNotificationWebhookMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expression': value.expression,
    };
}
exports.PatchedNotificationWebhookMappingRequestToJSON = PatchedNotificationWebhookMappingRequestToJSON;
