// @goauthentik/api/dist/models/PatchedPermissionAssignRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedPermissionAssignRequestToJSON = exports.PatchedPermissionAssignRequestFromJSONTyped = exports.PatchedPermissionAssignRequestFromJSON = exports.instanceOfPatchedPermissionAssignRequest = void 0;
const runtime_1 = require("../runtime");
const ModelEnum_1 = require("./ModelEnum");
/**
 * Check if a given object implements the PatchedPermissionAssignRequest interface.
 */
function instanceOfPatchedPermissionAssignRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedPermissionAssignRequest = instanceOfPatchedPermissionAssignRequest;
function PatchedPermissionAssignRequestFromJSON(json) {
    return PatchedPermissionAssignRequestFromJSONTyped(json, false);
}
exports.PatchedPermissionAssignRequestFromJSON = PatchedPermissionAssignRequestFromJSON;
function PatchedPermissionAssignRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissions': !(0, runtime_1.exists)(json, 'permissions') ? undefined : json['permissions'],
        'model': !(0, runtime_1.exists)(json, 'model') ? undefined : (0, ModelEnum_1.ModelEnumFromJSON)(json['model']),
        'objectPk': !(0, runtime_1.exists)(json, 'object_pk') ? undefined : json['object_pk'],
    };
}
exports.PatchedPermissionAssignRequestFromJSONTyped = PatchedPermissionAssignRequestFromJSONTyped;
function PatchedPermissionAssignRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': value.permissions,
        'model': (0, ModelEnum_1.ModelEnumToJSON)(value.model),
        'object_pk': value.objectPk,
    };
}
exports.PatchedPermissionAssignRequestToJSON = PatchedPermissionAssignRequestToJSON;
