// @goauthentik/api/dist/models/PatchedEmailStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedEmailStageRequestToJSON = exports.PatchedEmailStageRequestFromJSONTyped = exports.PatchedEmailStageRequestFromJSON = exports.instanceOfPatchedEmailStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedEmailStageRequest interface.
 */
function instanceOfPatchedEmailStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedEmailStageRequest = instanceOfPatchedEmailStageRequest;
function PatchedEmailStageRequestFromJSON(json) {
    return PatchedEmailStageRequestFromJSONTyped(json, false);
}
exports.PatchedEmailStageRequestFromJSON = PatchedEmailStageRequestFromJSON;
function PatchedEmailStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'useGlobalSettings': !(0, runtime_1.exists)(json, 'use_global_settings') ? undefined : json['use_global_settings'],
        'host': !(0, runtime_1.exists)(json, 'host') ? undefined : json['host'],
        'port': !(0, runtime_1.exists)(json, 'port') ? undefined : json['port'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
        'useTls': !(0, runtime_1.exists)(json, 'use_tls') ? undefined : json['use_tls'],
        'useSsl': !(0, runtime_1.exists)(json, 'use_ssl') ? undefined : json['use_ssl'],
        'timeout': !(0, runtime_1.exists)(json, 'timeout') ? undefined : json['timeout'],
        'fromAddress': !(0, runtime_1.exists)(json, 'from_address') ? undefined : json['from_address'],
        'tokenExpiry': !(0, runtime_1.exists)(json, 'token_expiry') ? undefined : json['token_expiry'],
        'subject': !(0, runtime_1.exists)(json, 'subject') ? undefined : json['subject'],
        'template': !(0, runtime_1.exists)(json, 'template') ? undefined : json['template'],
        'activateUserOnSuccess': !(0, runtime_1.exists)(json, 'activate_user_on_success') ? undefined : json['activate_user_on_success'],
    };
}
exports.PatchedEmailStageRequestFromJSONTyped = PatchedEmailStageRequestFromJSONTyped;
function PatchedEmailStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'use_global_settings': value.useGlobalSettings,
        'host': value.host,
        'port': value.port,
        'username': value.username,
        'password': value.password,
        'use_tls': value.useTls,
        'use_ssl': value.useSsl,
        'timeout': value.timeout,
        'from_address': value.fromAddress,
        'token_expiry': value.tokenExpiry,
        'subject': value.subject,
        'template': value.template,
        'activate_user_on_success': value.activateUserOnSuccess,
    };
}
exports.PatchedEmailStageRequestToJSON = PatchedEmailStageRequestToJSON;
