// @goauthentik/api/dist/models/PatchedInvitationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedInvitationRequestToJSON = exports.PatchedInvitationRequestFromJSONTyped = exports.PatchedInvitationRequestFromJSON = exports.instanceOfPatchedInvitationRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedInvitationRequest interface.
 */
function instanceOfPatchedInvitationRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedInvitationRequest = instanceOfPatchedInvitationRequest;
function PatchedInvitationRequestFromJSON(json) {
    return PatchedInvitationRequestFromJSONTyped(json, false);
}
exports.PatchedInvitationRequestFromJSON = PatchedInvitationRequestFromJSON;
function PatchedInvitationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'fixedData': !(0, runtime_1.exists)(json, 'fixed_data') ? undefined : json['fixed_data'],
        'singleUse': !(0, runtime_1.exists)(json, 'single_use') ? undefined : json['single_use'],
        'flow': !(0, runtime_1.exists)(json, 'flow') ? undefined : json['flow'],
    };
}
exports.PatchedInvitationRequestFromJSONTyped = PatchedInvitationRequestFromJSONTyped;
function PatchedInvitationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'fixed_data': value.fixedData,
        'single_use': value.singleUse,
        'flow': value.flow,
    };
}
exports.PatchedInvitationRequestToJSON = PatchedInvitationRequestToJSON;
