// @goauthentik/api/dist/models/PatchedAuthenticatorTOTPStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedAuthenticatorTOTPStageRequestToJSON = exports.PatchedAuthenticatorTOTPStageRequestFromJSONTyped = exports.PatchedAuthenticatorTOTPStageRequestFromJSON = exports.instanceOfPatchedAuthenticatorTOTPStageRequest = void 0;
const runtime_1 = require("../runtime");
const DigitsEnum_1 = require("./DigitsEnum");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PatchedAuthenticatorTOTPStageRequest interface.
 */
function instanceOfPatchedAuthenticatorTOTPStageRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedAuthenticatorTOTPStageRequest = instanceOfPatchedAuthenticatorTOTPStageRequest;
function PatchedAuthenticatorTOTPStageRequestFromJSON(json) {
    return PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json, false);
}
exports.PatchedAuthenticatorTOTPStageRequestFromJSON = PatchedAuthenticatorTOTPStageRequestFromJSON;
function PatchedAuthenticatorTOTPStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'configureFlow': !(0, runtime_1.exists)(json, 'configure_flow') ? undefined : json['configure_flow'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
        'digits': !(0, runtime_1.exists)(json, 'digits') ? undefined : (0, DigitsEnum_1.DigitsEnumFromJSON)(json['digits']),
    };
}
exports.PatchedAuthenticatorTOTPStageRequestFromJSONTyped = PatchedAuthenticatorTOTPStageRequestFromJSONTyped;
function PatchedAuthenticatorTOTPStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'configure_flow': value.configureFlow,
        'friendly_name': value.friendlyName,
        'digits': (0, DigitsEnum_1.DigitsEnumToJSON)(value.digits),
    };
}
exports.PatchedAuthenticatorTOTPStageRequestToJSON = PatchedAuthenticatorTOTPStageRequestToJSON;
