// @goauthentik/api/dist/models/PatchedGroupRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedGroupRequestToJSON = exports.PatchedGroupRequestFromJSONTyped = exports.PatchedGroupRequestFromJSON = exports.instanceOfPatchedGroupRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedGroupRequest interface.
 */
function instanceOfPatchedGroupRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedGroupRequest = instanceOfPatchedGroupRequest;
function PatchedGroupRequestFromJSON(json) {
    return PatchedGroupRequestFromJSONTyped(json, false);
}
exports.PatchedGroupRequestFromJSON = PatchedGroupRequestFromJSON;
function PatchedGroupRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'isSuperuser': !(0, runtime_1.exists)(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !(0, runtime_1.exists)(json, 'parent') ? undefined : json['parent'],
        'users': !(0, runtime_1.exists)(json, 'users') ? undefined : json['users'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
        'roles': !(0, runtime_1.exists)(json, 'roles') ? undefined : json['roles'],
    };
}
exports.PatchedGroupRequestFromJSONTyped = PatchedGroupRequestFromJSONTyped;
function PatchedGroupRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'users': value.users,
        'attributes': value.attributes,
        'roles': value.roles,
    };
}
exports.PatchedGroupRequestToJSON = PatchedGroupRequestToJSON;
