// @goauthentik/api/dist/models/PatchedOutpostRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedOutpostRequestToJSON = exports.PatchedOutpostRequestFromJSONTyped = exports.PatchedOutpostRequestFromJSON = exports.instanceOfPatchedOutpostRequest = void 0;
const runtime_1 = require("../runtime");
const OutpostTypeEnum_1 = require("./OutpostTypeEnum");
/**
 * Check if a given object implements the PatchedOutpostRequest interface.
 */
function instanceOfPatchedOutpostRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedOutpostRequest = instanceOfPatchedOutpostRequest;
function PatchedOutpostRequestFromJSON(json) {
    return PatchedOutpostRequestFromJSONTyped(json, false);
}
exports.PatchedOutpostRequestFromJSON = PatchedOutpostRequestFromJSON;
function PatchedOutpostRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : (0, OutpostTypeEnum_1.OutpostTypeEnumFromJSON)(json['type']),
        'providers': !(0, runtime_1.exists)(json, 'providers') ? undefined : json['providers'],
        'serviceConnection': !(0, runtime_1.exists)(json, 'service_connection') ? undefined : json['service_connection'],
        'config': !(0, runtime_1.exists)(json, 'config') ? undefined : json['config'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
    };
}
exports.PatchedOutpostRequestFromJSONTyped = PatchedOutpostRequestFromJSONTyped;
function PatchedOutpostRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': (0, OutpostTypeEnum_1.OutpostTypeEnumToJSON)(value.type),
        'providers': value.providers,
        'service_connection': value.serviceConnection,
        'config': value.config,
        'managed': value.managed,
    };
}
exports.PatchedOutpostRequestToJSON = PatchedOutpostRequestToJSON;
