// @goauthentik/api/dist/models/PatchedKubernetesServiceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchedKubernetesServiceConnectionRequestToJSON = exports.PatchedKubernetesServiceConnectionRequestFromJSONTyped = exports.PatchedKubernetesServiceConnectionRequestFromJSON = exports.instanceOfPatchedKubernetesServiceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PatchedKubernetesServiceConnectionRequest interface.
 */
function instanceOfPatchedKubernetesServiceConnectionRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPatchedKubernetesServiceConnectionRequest = instanceOfPatchedKubernetesServiceConnectionRequest;
function PatchedKubernetesServiceConnectionRequestFromJSON(json) {
    return PatchedKubernetesServiceConnectionRequestFromJSONTyped(json, false);
}
exports.PatchedKubernetesServiceConnectionRequestFromJSON = PatchedKubernetesServiceConnectionRequestFromJSON;
function PatchedKubernetesServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'kubeconfig': !(0, runtime_1.exists)(json, 'kubeconfig') ? undefined : json['kubeconfig'],
        'verifySsl': !(0, runtime_1.exists)(json, 'verify_ssl') ? undefined : json['verify_ssl'],
    };
}
exports.PatchedKubernetesServiceConnectionRequestFromJSONTyped = PatchedKubernetesServiceConnectionRequestFromJSONTyped;
function PatchedKubernetesServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
        'kubeconfig': value.kubeconfig,
        'verify_ssl': value.verifySsl,
    };
}
exports.PatchedKubernetesServiceConnectionRequestToJSON = PatchedKubernetesServiceConnectionRequestToJSON;
