// @goauthentik/api/dist/models/Permission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionToJSON = exports.PermissionFromJSONTyped = exports.PermissionFromJSON = exports.instanceOfPermission = void 0;
/**
 * Check if a given object implements the Permission interface.
 */
function instanceOfPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "appLabelVerbose" in value;
    isInstance = isInstance && "modelVerbose" in value;
    return isInstance;
}
exports.instanceOfPermission = instanceOfPermission;
function PermissionFromJSON(json) {
    return PermissionFromJSONTyped(json, false);
}
exports.PermissionFromJSON = PermissionFromJSON;
function PermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'name': json['name'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'appLabelVerbose': json['app_label_verbose'],
        'modelVerbose': json['model_verbose'],
    };
}
exports.PermissionFromJSONTyped = PermissionFromJSONTyped;
function PermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'codename': value.codename,
    };
}
exports.PermissionToJSON = PermissionToJSON;
