// @goauthentik/api/dist/models/PermissionAssignRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionAssignRequestToJSON = exports.PermissionAssignRequestFromJSONTyped = exports.PermissionAssignRequestFromJSON = exports.instanceOfPermissionAssignRequest = void 0;
const runtime_1 = require("../runtime");
const ModelEnum_1 = require("./ModelEnum");
/**
 * Check if a given object implements the PermissionAssignRequest interface.
 */
function instanceOfPermissionAssignRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "permissions" in value;
    return isInstance;
}
exports.instanceOfPermissionAssignRequest = instanceOfPermissionAssignRequest;
function PermissionAssignRequestFromJSON(json) {
    return PermissionAssignRequestFromJSONTyped(json, false);
}
exports.PermissionAssignRequestFromJSON = PermissionAssignRequestFromJSON;
function PermissionAssignRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissions': json['permissions'],
        'model': !(0, runtime_1.exists)(json, 'model') ? undefined : (0, ModelEnum_1.ModelEnumFromJSON)(json['model']),
        'objectPk': !(0, runtime_1.exists)(json, 'object_pk') ? undefined : json['object_pk'],
    };
}
exports.PermissionAssignRequestFromJSONTyped = PermissionAssignRequestFromJSONTyped;
function PermissionAssignRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': value.permissions,
        'model': (0, ModelEnum_1.ModelEnumToJSON)(value.model),
        'object_pk': value.objectPk,
    };
}
exports.PermissionAssignRequestToJSON = PermissionAssignRequestToJSON;
