// @goauthentik/api/dist/models/PlexAuthenticationChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlexAuthenticationChallengeResponseRequestToJSON = exports.PlexAuthenticationChallengeResponseRequestFromJSONTyped = exports.PlexAuthenticationChallengeResponseRequestFromJSON = exports.instanceOfPlexAuthenticationChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PlexAuthenticationChallengeResponseRequest interface.
 */
function instanceOfPlexAuthenticationChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPlexAuthenticationChallengeResponseRequest = instanceOfPlexAuthenticationChallengeResponseRequest;
function PlexAuthenticationChallengeResponseRequestFromJSON(json) {
    return PlexAuthenticationChallengeResponseRequestFromJSONTyped(json, false);
}
exports.PlexAuthenticationChallengeResponseRequestFromJSON = PlexAuthenticationChallengeResponseRequestFromJSON;
function PlexAuthenticationChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
    };
}
exports.PlexAuthenticationChallengeResponseRequestFromJSONTyped = PlexAuthenticationChallengeResponseRequestFromJSONTyped;
function PlexAuthenticationChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
    };
}
exports.PlexAuthenticationChallengeResponseRequestToJSON = PlexAuthenticationChallengeResponseRequestToJSON;
