// @goauthentik/api/dist/models/PlexSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlexSourceConnectionRequestToJSON = exports.PlexSourceConnectionRequestFromJSONTyped = exports.PlexSourceConnectionRequestFromJSON = exports.instanceOfPlexSourceConnectionRequest = void 0;
/**
 * Check if a given object implements the PlexSourceConnectionRequest interface.
 */
function instanceOfPlexSourceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "plexToken" in value;
    return isInstance;
}
exports.instanceOfPlexSourceConnectionRequest = instanceOfPlexSourceConnectionRequest;
function PlexSourceConnectionRequestFromJSON(json) {
    return PlexSourceConnectionRequestFromJSONTyped(json, false);
}
exports.PlexSourceConnectionRequestFromJSON = PlexSourceConnectionRequestFromJSON;
function PlexSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'identifier': json['identifier'],
        'plexToken': json['plex_token'],
    };
}
exports.PlexSourceConnectionRequestFromJSONTyped = PlexSourceConnectionRequestFromJSONTyped;
function PlexSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'identifier': value.identifier,
        'plex_token': value.plexToken,
    };
}
exports.PlexSourceConnectionRequestToJSON = PlexSourceConnectionRequestToJSON;
