// @goauthentik/api/dist/models/PlexTokenRedeemRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlexTokenRedeemRequestToJSON = exports.PlexTokenRedeemRequestFromJSONTyped = exports.PlexTokenRedeemRequestFromJSON = exports.instanceOfPlexTokenRedeemRequest = void 0;
/**
 * Check if a given object implements the PlexTokenRedeemRequest interface.
 */
function instanceOfPlexTokenRedeemRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "plexToken" in value;
    return isInstance;
}
exports.instanceOfPlexTokenRedeemRequest = instanceOfPlexTokenRedeemRequest;
function PlexTokenRedeemRequestFromJSON(json) {
    return PlexTokenRedeemRequestFromJSONTyped(json, false);
}
exports.PlexTokenRedeemRequestFromJSON = PlexTokenRedeemRequestFromJSON;
function PlexTokenRedeemRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'plexToken': json['plex_token'],
    };
}
exports.PlexTokenRedeemRequestFromJSONTyped = PlexTokenRedeemRequestFromJSONTyped;
function PlexTokenRedeemRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'plex_token': value.plexToken,
    };
}
exports.PlexTokenRedeemRequestToJSON = PlexTokenRedeemRequestToJSON;
