// @goauthentik/api/dist/models/PolicyTestResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTestResultToJSON = exports.PolicyTestResultFromJSONTyped = exports.PolicyTestResultFromJSON = exports.instanceOfPolicyTestResult = void 0;
/**
 * Check if a given object implements the PolicyTestResult interface.
 */
function instanceOfPolicyTestResult(value) {
    let isInstance = true;
    isInstance = isInstance && "passing" in value;
    isInstance = isInstance && "messages" in value;
    isInstance = isInstance && "logMessages" in value;
    return isInstance;
}
exports.instanceOfPolicyTestResult = instanceOfPolicyTestResult;
function PolicyTestResultFromJSON(json) {
    return PolicyTestResultFromJSONTyped(json, false);
}
exports.PolicyTestResultFromJSON = PolicyTestResultFromJSON;
function PolicyTestResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'passing': json['passing'],
        'messages': json['messages'],
        'logMessages': json['log_messages'],
    };
}
exports.PolicyTestResultFromJSONTyped = PolicyTestResultFromJSONTyped;
function PolicyTestResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'passing': value.passing,
    };
}
exports.PolicyTestResultToJSON = PolicyTestResultToJSON;
