// @goauthentik/api/dist/models/PolicyBinding.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyBindingToJSON = exports.PolicyBindingFromJSONTyped = exports.PolicyBindingFromJSON = exports.instanceOfPolicyBinding = void 0;
const runtime_1 = require("../runtime");
const Group_1 = require("./Group");
const Policy_1 = require("./Policy");
const User_1 = require("./User");
/**
 * Check if a given object implements the PolicyBinding interface.
 */
function instanceOfPolicyBinding(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "policyObj" in value;
    isInstance = isInstance && "groupObj" in value;
    isInstance = isInstance && "userObj" in value;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "order" in value;
    return isInstance;
}
exports.instanceOfPolicyBinding = instanceOfPolicyBinding;
function PolicyBindingFromJSON(json) {
    return PolicyBindingFromJSONTyped(json, false);
}
exports.PolicyBindingFromJSON = PolicyBindingFromJSON;
function PolicyBindingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'policy': !(0, runtime_1.exists)(json, 'policy') ? undefined : json['policy'],
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'policyObj': (0, Policy_1.PolicyFromJSON)(json['policy_obj']),
        'groupObj': (0, Group_1.GroupFromJSON)(json['group_obj']),
        'userObj': (0, User_1.UserFromJSON)(json['user_obj']),
        'target': json['target'],
        'negate': !(0, runtime_1.exists)(json, 'negate') ? undefined : json['negate'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'order': json['order'],
        'timeout': !(0, runtime_1.exists)(json, 'timeout') ? undefined : json['timeout'],
        'failureResult': !(0, runtime_1.exists)(json, 'failure_result') ? undefined : json['failure_result'],
    };
}
exports.PolicyBindingFromJSONTyped = PolicyBindingFromJSONTyped;
function PolicyBindingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'policy': value.policy,
        'group': value.group,
        'user': value.user,
        'target': value.target,
        'negate': value.negate,
        'enabled': value.enabled,
        'order': value.order,
        'timeout': value.timeout,
        'failure_result': value.failureResult,
    };
}
exports.PolicyBindingToJSON = PolicyBindingToJSON;
