// @goauthentik/api/dist/models/PolicyTestRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTestRequestToJSON = exports.PolicyTestRequestFromJSONTyped = exports.PolicyTestRequestFromJSON = exports.instanceOfPolicyTestRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PolicyTestRequest interface.
 */
function instanceOfPolicyTestRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfPolicyTestRequest = instanceOfPolicyTestRequest;
function PolicyTestRequestFromJSON(json) {
    return PolicyTestRequestFromJSONTyped(json, false);
}
exports.PolicyTestRequestFromJSON = PolicyTestRequestFromJSON;
function PolicyTestRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'context': !(0, runtime_1.exists)(json, 'context') ? undefined : json['context'],
    };
}
exports.PolicyTestRequestFromJSONTyped = PolicyTestRequestFromJSONTyped;
function PolicyTestRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'context': value.context,
    };
}
exports.PolicyTestRequestToJSON = PolicyTestRequestToJSON;
