// @goauthentik/api/dist/models/PolicyBindingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyBindingRequestToJSON = exports.PolicyBindingRequestFromJSONTyped = exports.PolicyBindingRequestFromJSON = exports.instanceOfPolicyBindingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PolicyBindingRequest interface.
 */
function instanceOfPolicyBindingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "target" in value;
    isInstance = isInstance && "order" in value;
    return isInstance;
}
exports.instanceOfPolicyBindingRequest = instanceOfPolicyBindingRequest;
function PolicyBindingRequestFromJSON(json) {
    return PolicyBindingRequestFromJSONTyped(json, false);
}
exports.PolicyBindingRequestFromJSON = PolicyBindingRequestFromJSON;
function PolicyBindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'policy': !(0, runtime_1.exists)(json, 'policy') ? undefined : json['policy'],
        'group': !(0, runtime_1.exists)(json, 'group') ? undefined : json['group'],
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'target': json['target'],
        'negate': !(0, runtime_1.exists)(json, 'negate') ? undefined : json['negate'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
        'order': json['order'],
        'timeout': !(0, runtime_1.exists)(json, 'timeout') ? undefined : json['timeout'],
        'failureResult': !(0, runtime_1.exists)(json, 'failure_result') ? undefined : json['failure_result'],
    };
}
exports.PolicyBindingRequestFromJSONTyped = PolicyBindingRequestFromJSONTyped;
function PolicyBindingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'policy': value.policy,
        'group': value.group,
        'user': value.user,
        'target': value.target,
        'negate': value.negate,
        'enabled': value.enabled,
        'order': value.order,
        'timeout': value.timeout,
        'failure_result': value.failureResult,
    };
}
exports.PolicyBindingRequestToJSON = PolicyBindingRequestToJSON;
