// @goauthentik/api/dist/models/PolicyEngineMode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyEngineModeToJSON = exports.PolicyEngineModeFromJSONTyped = exports.PolicyEngineModeFromJSON = exports.PolicyEngineMode = void 0;
/**
 * * `all` - all, all policies must pass
 * * `any` - any, any policy must pass
 * @export
 */
exports.PolicyEngineMode = {
    All: 'all',
    Any: 'any',
    UnknownDefaultOpenApi: '11184809'
};
function PolicyEngineModeFromJSON(json) {
    return PolicyEngineModeFromJSONTyped(json, false);
}
exports.PolicyEngineModeFromJSON = PolicyEngineModeFromJSON;
function PolicyEngineModeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PolicyEngineModeFromJSONTyped = PolicyEngineModeFromJSONTyped;
function PolicyEngineModeToJSON(value) {
    return value;
}
exports.PolicyEngineModeToJSON = PolicyEngineModeToJSON;
