// @goauthentik/api/dist/models/PolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyRequestToJSON = exports.PolicyRequestFromJSONTyped = exports.PolicyRequestFromJSON = exports.instanceOfPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PolicyRequest interface.
 */
function instanceOfPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfPolicyRequest = instanceOfPolicyRequest;
function PolicyRequestFromJSON(json) {
    return PolicyRequestFromJSONTyped(json, false);
}
exports.PolicyRequestFromJSON = PolicyRequestFromJSON;
function PolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
    };
}
exports.PolicyRequestFromJSONTyped = PolicyRequestFromJSONTyped;
function PolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
    };
}
exports.PolicyRequestToJSON = PolicyRequestToJSON;
