// @goauthentik/api/dist/models/PromptChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromptChallengeResponseRequestToJSON = exports.PromptChallengeResponseRequestFromJSONTyped = exports.PromptChallengeResponseRequestFromJSON = exports.instanceOfPromptChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the PromptChallengeResponseRequest interface.
 */
function instanceOfPromptChallengeResponseRequest(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfPromptChallengeResponseRequest = instanceOfPromptChallengeResponseRequest;
function PromptChallengeResponseRequestFromJSON(json) {
    return PromptChallengeResponseRequestFromJSONTyped(json, false);
}
exports.PromptChallengeResponseRequestFromJSON = PromptChallengeResponseRequestFromJSON;
function PromptChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return Object.assign(Object.assign({}, json), { 'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'] });
}
exports.PromptChallengeResponseRequestFromJSONTyped = PromptChallengeResponseRequestFromJSONTyped;
function PromptChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return Object.assign(Object.assign({}, value), { 'component': value.component });
}
exports.PromptChallengeResponseRequestToJSON = PromptChallengeResponseRequestToJSON;
