// @goauthentik/api/dist/models/PromptStageRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromptStageRequestToJSON = exports.PromptStageRequestFromJSONTyped = exports.PromptStageRequestFromJSON = exports.instanceOfPromptStageRequest = void 0;
const runtime_1 = require("../runtime");
const FlowSetRequest_1 = require("./FlowSetRequest");
/**
 * Check if a given object implements the PromptStageRequest interface.
 */
function instanceOfPromptStageRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "fields" in value;
    return isInstance;
}
exports.instanceOfPromptStageRequest = instanceOfPromptStageRequest;
function PromptStageRequestFromJSON(json) {
    return PromptStageRequestFromJSONTyped(json, false);
}
exports.PromptStageRequestFromJSON = PromptStageRequestFromJSON;
function PromptStageRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSetRequest_1.FlowSetRequestFromJSON)),
        'fields': json['fields'],
        'validationPolicies': !(0, runtime_1.exists)(json, 'validation_policies') ? undefined : json['validation_policies'],
    };
}
exports.PromptStageRequestFromJSONTyped = PromptStageRequestFromJSONTyped;
function PromptStageRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSetRequest_1.FlowSetRequestToJSON)),
        'fields': value.fields,
        'validation_policies': value.validationPolicies,
    };
}
exports.PromptStageRequestToJSON = PromptStageRequestToJSON;
