// @goauthentik/api/dist/models/PromptTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromptTypeEnumToJSON = exports.PromptTypeEnumFromJSONTyped = exports.PromptTypeEnumFromJSON = exports.PromptTypeEnum = void 0;
/**
 * * `text` - Text: Simple Text input
 * * `text_area` - Text area: Multiline Text Input.
 * * `text_read_only` - Text (read-only): Simple Text input, but cannot be edited.
 * * `text_area_read_only` - Text area (read-only): Multiline Text input, but cannot be edited.
 * * `username` - Username: Same as Text input, but checks for and prevents duplicate usernames.
 * * `email` - Email: Text field with Email type.
 * * `password` - Password: Masked input, multiple inputs of this type on the same prompt need to be identical.
 * * `number` - Number
 * * `checkbox` - Checkbox
 * * `radio-button-group` - Fixed choice field rendered as a group of radio buttons.
 * * `dropdown` - Fixed choice field rendered as a dropdown.
 * * `date` - Date
 * * `date-time` - Date Time
 * * `file` - File: File upload for arbitrary files. File content will be available in flow context as data-URI
 * * `separator` - Separator: Static Separator Line
 * * `hidden` - Hidden: Hidden field, can be used to insert data into form.
 * * `static` - Static: Static value, displayed as-is.
 * * `ak-locale` - authentik: Selection of locales authentik supports
 * @export
 */
exports.PromptTypeEnum = {
    Text: 'text',
    TextArea: 'text_area',
    TextReadOnly: 'text_read_only',
    TextAreaReadOnly: 'text_area_read_only',
    Username: 'username',
    Email: 'email',
    Password: 'password',
    Number: 'number',
    Checkbox: 'checkbox',
    RadioButtonGroup: 'radio-button-group',
    Dropdown: 'dropdown',
    Date: 'date',
    DateTime: 'date-time',
    File: 'file',
    Separator: 'separator',
    Hidden: 'hidden',
    Static: 'static',
    AkLocale: 'ak-locale',
    UnknownDefaultOpenApi: '11184809'
};
function PromptTypeEnumFromJSON(json) {
    return PromptTypeEnumFromJSONTyped(json, false);
}
exports.PromptTypeEnumFromJSON = PromptTypeEnumFromJSON;
function PromptTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PromptTypeEnumFromJSONTyped = PromptTypeEnumFromJSONTyped;
function PromptTypeEnumToJSON(value) {
    return value;
}
exports.PromptTypeEnumToJSON = PromptTypeEnumToJSON;
