// @goauthentik/api/dist/models/PropertyMappingTestResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyMappingTestResultToJSON = exports.PropertyMappingTestResultFromJSONTyped = exports.PropertyMappingTestResultFromJSON = exports.instanceOfPropertyMappingTestResult = void 0;
/**
 * Check if a given object implements the PropertyMappingTestResult interface.
 */
function instanceOfPropertyMappingTestResult(value) {
    let isInstance = true;
    isInstance = isInstance && "result" in value;
    isInstance = isInstance && "successful" in value;
    return isInstance;
}
exports.instanceOfPropertyMappingTestResult = instanceOfPropertyMappingTestResult;
function PropertyMappingTestResultFromJSON(json) {
    return PropertyMappingTestResultFromJSONTyped(json, false);
}
exports.PropertyMappingTestResultFromJSON = PropertyMappingTestResultFromJSON;
function PropertyMappingTestResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': json['result'],
        'successful': json['successful'],
    };
}
exports.PropertyMappingTestResultFromJSONTyped = PropertyMappingTestResultFromJSONTyped;
function PropertyMappingTestResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.PropertyMappingTestResultToJSON = PropertyMappingTestResultToJSON;
