// @goauthentik/api/dist/models/ProtocolEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtocolEnumToJSON = exports.ProtocolEnumFromJSONTyped = exports.ProtocolEnumFromJSON = exports.ProtocolEnum = void 0;
/**
 * * `rdp` - Rdp
 * * `vnc` - Vnc
 * * `ssh` - Ssh
 * @export
 */
exports.ProtocolEnum = {
    Rdp: 'rdp',
    Vnc: 'vnc',
    Ssh: 'ssh',
    UnknownDefaultOpenApi: '11184809'
};
function ProtocolEnumFromJSON(json) {
    return ProtocolEnumFromJSONTyped(json, false);
}
exports.ProtocolEnumFromJSON = ProtocolEnumFromJSON;
function ProtocolEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ProtocolEnumFromJSONTyped = ProtocolEnumFromJSONTyped;
function ProtocolEnumToJSON(value) {
    return value;
}
exports.ProtocolEnumToJSON = ProtocolEnumToJSON;
