// @goauthentik/api/dist/models/ProviderEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderEnumToJSON = exports.ProviderEnumFromJSONTyped = exports.ProviderEnumFromJSON = exports.ProviderEnum = void 0;
/**
 * * `twilio` - Twilio
 * * `generic` - Generic
 * @export
 */
exports.ProviderEnum = {
    Twilio: 'twilio',
    Generic: 'generic',
    UnknownDefaultOpenApi: '11184809'
};
function ProviderEnumFromJSON(json) {
    return ProviderEnumFromJSONTyped(json, false);
}
exports.ProviderEnumFromJSON = ProviderEnumFromJSON;
function ProviderEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ProviderEnumFromJSONTyped = ProviderEnumFromJSONTyped;
function ProviderEnumToJSON(value) {
    return value;
}
exports.ProviderEnumToJSON = ProviderEnumToJSON;
