// @goauthentik/api/dist/models/ProviderTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderTypeEnumToJSON = exports.ProviderTypeEnumFromJSONTyped = exports.ProviderTypeEnumFromJSON = exports.ProviderTypeEnum = void 0;
/**
 * * `apple` - Apple
 * * `openidconnect` - OpenID Connect
 * * `azuread` - Azure AD
 * * `discord` - Discord
 * * `facebook` - Facebook
 * * `github` - GitHub
 * * `google` - Google
 * * `mailcow` - Mailcow
 * * `okta` - Okta
 * * `patreon` - Patreon
 * * `reddit` - Reddit
 * * `twitch` - Twitch
 * * `twitter` - Twitter
 * @export
 */
exports.ProviderTypeEnum = {
    Apple: 'apple',
    Openidconnect: 'openidconnect',
    Azuread: 'azuread',
    Discord: 'discord',
    Facebook: 'facebook',
    Github: 'github',
    Google: 'google',
    Mailcow: 'mailcow',
    Okta: 'okta',
    Patreon: 'patreon',
    Reddit: 'reddit',
    Twitch: 'twitch',
    Twitter: 'twitter',
    UnknownDefaultOpenApi: '11184809'
};
function ProviderTypeEnumFromJSON(json) {
    return ProviderTypeEnumFromJSONTyped(json, false);
}
exports.ProviderTypeEnumFromJSON = ProviderTypeEnumFromJSON;
function ProviderTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ProviderTypeEnumFromJSONTyped = ProviderTypeEnumFromJSONTyped;
function ProviderTypeEnumToJSON(value) {
    return value;
}
exports.ProviderTypeEnumToJSON = ProviderTypeEnumToJSON;
