// @goauthentik/api/dist/models/ProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderRequestToJSON = exports.ProviderRequestFromJSONTyped = exports.ProviderRequestFromJSON = exports.instanceOfProviderRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ProviderRequest interface.
 */
function instanceOfProviderRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "authorizationFlow" in value;
    return isInstance;
}
exports.instanceOfProviderRequest = instanceOfProviderRequest;
function ProviderRequestFromJSON(json) {
    return ProviderRequestFromJSONTyped(json, false);
}
exports.ProviderRequestFromJSON = ProviderRequestFromJSON;
function ProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
    };
}
exports.ProviderRequestFromJSONTyped = ProviderRequestFromJSONTyped;
function ProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
    };
}
exports.ProviderRequestToJSON = ProviderRequestToJSON;
