// @goauthentik/api/dist/models/ProxyMode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyModeToJSON = exports.ProxyModeFromJSONTyped = exports.ProxyModeFromJSON = exports.ProxyMode = void 0;
/**
 * * `proxy` - Proxy
 * * `forward_single` - Forward Single
 * * `forward_domain` - Forward Domain
 * @export
 */
exports.ProxyMode = {
    Proxy: 'proxy',
    ForwardSingle: 'forward_single',
    ForwardDomain: 'forward_domain',
    UnknownDefaultOpenApi: '11184809'
};
function ProxyModeFromJSON(json) {
    return ProxyModeFromJSONTyped(json, false);
}
exports.ProxyModeFromJSON = ProxyModeFromJSON;
function ProxyModeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ProxyModeFromJSONTyped = ProxyModeFromJSONTyped;
function ProxyModeToJSON(value) {
    return value;
}
exports.ProxyModeToJSON = ProxyModeToJSON;
