// @goauthentik/api/dist/models/ProxyProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyProviderRequestToJSON = exports.ProxyProviderRequestFromJSONTyped = exports.ProxyProviderRequestFromJSON = exports.instanceOfProxyProviderRequest = void 0;
const runtime_1 = require("../runtime");
const ProxyMode_1 = require("./ProxyMode");
/**
 * Check if a given object implements the ProxyProviderRequest interface.
 */
function instanceOfProxyProviderRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "authorizationFlow" in value;
    isInstance = isInstance && "externalHost" in value;
    return isInstance;
}
exports.instanceOfProxyProviderRequest = instanceOfProxyProviderRequest;
function ProxyProviderRequestFromJSON(json) {
    return ProxyProviderRequestFromJSONTyped(json, false);
}
exports.ProxyProviderRequestFromJSON = ProxyProviderRequestFromJSON;
function ProxyProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'internalHost': !(0, runtime_1.exists)(json, 'internal_host') ? undefined : json['internal_host'],
        'externalHost': json['external_host'],
        'internalHostSslValidation': !(0, runtime_1.exists)(json, 'internal_host_ssl_validation') ? undefined : json['internal_host_ssl_validation'],
        'certificate': !(0, runtime_1.exists)(json, 'certificate') ? undefined : json['certificate'],
        'skipPathRegex': !(0, runtime_1.exists)(json, 'skip_path_regex') ? undefined : json['skip_path_regex'],
        'basicAuthEnabled': !(0, runtime_1.exists)(json, 'basic_auth_enabled') ? undefined : json['basic_auth_enabled'],
        'basicAuthPasswordAttribute': !(0, runtime_1.exists)(json, 'basic_auth_password_attribute') ? undefined : json['basic_auth_password_attribute'],
        'basicAuthUserAttribute': !(0, runtime_1.exists)(json, 'basic_auth_user_attribute') ? undefined : json['basic_auth_user_attribute'],
        'mode': !(0, runtime_1.exists)(json, 'mode') ? undefined : (0, ProxyMode_1.ProxyModeFromJSON)(json['mode']),
        'interceptHeaderAuth': !(0, runtime_1.exists)(json, 'intercept_header_auth') ? undefined : json['intercept_header_auth'],
        'cookieDomain': !(0, runtime_1.exists)(json, 'cookie_domain') ? undefined : json['cookie_domain'],
        'jwksSources': !(0, runtime_1.exists)(json, 'jwks_sources') ? undefined : json['jwks_sources'],
        'accessTokenValidity': !(0, runtime_1.exists)(json, 'access_token_validity') ? undefined : json['access_token_validity'],
        'refreshTokenValidity': !(0, runtime_1.exists)(json, 'refresh_token_validity') ? undefined : json['refresh_token_validity'],
    };
}
exports.ProxyProviderRequestFromJSONTyped = ProxyProviderRequestFromJSONTyped;
function ProxyProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'internal_host': value.internalHost,
        'external_host': value.externalHost,
        'internal_host_ssl_validation': value.internalHostSslValidation,
        'certificate': value.certificate,
        'skip_path_regex': value.skipPathRegex,
        'basic_auth_enabled': value.basicAuthEnabled,
        'basic_auth_password_attribute': value.basicAuthPasswordAttribute,
        'basic_auth_user_attribute': value.basicAuthUserAttribute,
        'mode': (0, ProxyMode_1.ProxyModeToJSON)(value.mode),
        'intercept_header_auth': value.interceptHeaderAuth,
        'cookie_domain': value.cookieDomain,
        'jwks_sources': value.jwksSources,
        'access_token_validity': value.accessTokenValidity,
        'refresh_token_validity': value.refreshTokenValidity,
    };
}
exports.ProxyProviderRequestToJSON = ProxyProviderRequestToJSON;
