// @goauthentik/api/dist/models/RACPropertyMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RACPropertyMappingRequestToJSON = exports.RACPropertyMappingRequestFromJSONTyped = exports.RACPropertyMappingRequestFromJSON = exports.instanceOfRACPropertyMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the RACPropertyMappingRequest interface.
 */
function instanceOfRACPropertyMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "staticSettings" in value;
    return isInstance;
}
exports.instanceOfRACPropertyMappingRequest = instanceOfRACPropertyMappingRequest;
function RACPropertyMappingRequestFromJSON(json) {
    return RACPropertyMappingRequestFromJSONTyped(json, false);
}
exports.RACPropertyMappingRequestFromJSON = RACPropertyMappingRequestFromJSON;
function RACPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': !(0, runtime_1.exists)(json, 'expression') ? undefined : json['expression'],
        'staticSettings': json['static_settings'],
    };
}
exports.RACPropertyMappingRequestFromJSONTyped = RACPropertyMappingRequestFromJSONTyped;
function RACPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'static_settings': value.staticSettings,
    };
}
exports.RACPropertyMappingRequestToJSON = RACPropertyMappingRequestToJSON;
