// @goauthentik/api/dist/models/RadiusProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadiusProviderRequestToJSON = exports.RadiusProviderRequestFromJSONTyped = exports.RadiusProviderRequestFromJSON = exports.instanceOfRadiusProviderRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the RadiusProviderRequest interface.
 */
function instanceOfRadiusProviderRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "authorizationFlow" in value;
    return isInstance;
}
exports.instanceOfRadiusProviderRequest = instanceOfRadiusProviderRequest;
function RadiusProviderRequestFromJSON(json) {
    return RadiusProviderRequestFromJSONTyped(json, false);
}
exports.RadiusProviderRequestFromJSON = RadiusProviderRequestFromJSON;
function RadiusProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'authenticationFlow': !(0, runtime_1.exists)(json, 'authentication_flow') ? undefined : json['authentication_flow'],
        'authorizationFlow': json['authorization_flow'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'clientNetworks': !(0, runtime_1.exists)(json, 'client_networks') ? undefined : json['client_networks'],
        'sharedSecret': !(0, runtime_1.exists)(json, 'shared_secret') ? undefined : json['shared_secret'],
        'mfaSupport': !(0, runtime_1.exists)(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}
exports.RadiusProviderRequestFromJSONTyped = RadiusProviderRequestFromJSONTyped;
function RadiusProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'authentication_flow': value.authenticationFlow,
        'authorization_flow': value.authorizationFlow,
        'property_mappings': value.propertyMappings,
        'client_networks': value.clientNetworks,
        'shared_secret': value.sharedSecret,
        'mfa_support': value.mfaSupport,
    };
}
exports.RadiusProviderRequestToJSON = RadiusProviderRequestToJSON;
