// @goauthentik/api/dist/models/RadiusOutpostConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadiusOutpostConfigToJSON = exports.RadiusOutpostConfigFromJSONTyped = exports.RadiusOutpostConfigFromJSON = exports.instanceOfRadiusOutpostConfig = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the RadiusOutpostConfig interface.
 */
function instanceOfRadiusOutpostConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "applicationSlug" in value;
    isInstance = isInstance && "authFlowSlug" in value;
    return isInstance;
}
exports.instanceOfRadiusOutpostConfig = instanceOfRadiusOutpostConfig;
function RadiusOutpostConfigFromJSON(json) {
    return RadiusOutpostConfigFromJSONTyped(json, false);
}
exports.RadiusOutpostConfigFromJSON = RadiusOutpostConfigFromJSON;
function RadiusOutpostConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'applicationSlug': json['application_slug'],
        'authFlowSlug': json['auth_flow_slug'],
        'clientNetworks': !(0, runtime_1.exists)(json, 'client_networks') ? undefined : json['client_networks'],
        'sharedSecret': !(0, runtime_1.exists)(json, 'shared_secret') ? undefined : json['shared_secret'],
        'mfaSupport': !(0, runtime_1.exists)(json, 'mfa_support') ? undefined : json['mfa_support'],
    };
}
exports.RadiusOutpostConfigFromJSONTyped = RadiusOutpostConfigFromJSONTyped;
function RadiusOutpostConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'application_slug': value.applicationSlug,
        'auth_flow_slug': value.authFlowSlug,
        'client_networks': value.clientNetworks,
        'shared_secret': value.sharedSecret,
        'mfa_support': value.mfaSupport,
    };
}
exports.RadiusOutpostConfigToJSON = RadiusOutpostConfigToJSON;
