// @goauthentik/api/dist/models/RedirectChallenge.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedirectChallengeToJSON = exports.RedirectChallengeFromJSONTyped = exports.RedirectChallengeFromJSON = exports.instanceOfRedirectChallenge = void 0;
const runtime_1 = require("../runtime");
const ChallengeChoices_1 = require("./ChallengeChoices");
const ContextualFlowInfo_1 = require("./ContextualFlowInfo");
/**
 * Check if a given object implements the RedirectChallenge interface.
 */
function instanceOfRedirectChallenge(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "to" in value;
    return isInstance;
}
exports.instanceOfRedirectChallenge = instanceOfRedirectChallenge;
function RedirectChallengeFromJSON(json) {
    return RedirectChallengeFromJSONTyped(json, false);
}
exports.RedirectChallengeFromJSON = RedirectChallengeFromJSON;
function RedirectChallengeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesFromJSON)(json['type']),
        'flowInfo': !(0, runtime_1.exists)(json, 'flow_info') ? undefined : (0, ContextualFlowInfo_1.ContextualFlowInfoFromJSON)(json['flow_info']),
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'responseErrors': !(0, runtime_1.exists)(json, 'response_errors') ? undefined : json['response_errors'],
        'to': json['to'],
    };
}
exports.RedirectChallengeFromJSONTyped = RedirectChallengeFromJSONTyped;
function RedirectChallengeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': (0, ChallengeChoices_1.ChallengeChoicesToJSON)(value.type),
        'flow_info': (0, ContextualFlowInfo_1.ContextualFlowInfoToJSON)(value.flowInfo),
        'component': value.component,
        'response_errors': value.responseErrors,
        'to': value.to,
    };
}
exports.RedirectChallengeToJSON = RedirectChallengeToJSON;
