// @goauthentik/api/dist/models/ReputationPolicyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReputationPolicyRequestToJSON = exports.ReputationPolicyRequestFromJSONTyped = exports.ReputationPolicyRequestFromJSON = exports.instanceOfReputationPolicyRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ReputationPolicyRequest interface.
 */
function instanceOfReputationPolicyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfReputationPolicyRequest = instanceOfReputationPolicyRequest;
function ReputationPolicyRequestFromJSON(json) {
    return ReputationPolicyRequestFromJSONTyped(json, false);
}
exports.ReputationPolicyRequestFromJSON = ReputationPolicyRequestFromJSON;
function ReputationPolicyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'executionLogging': !(0, runtime_1.exists)(json, 'execution_logging') ? undefined : json['execution_logging'],
        'checkIp': !(0, runtime_1.exists)(json, 'check_ip') ? undefined : json['check_ip'],
        'checkUsername': !(0, runtime_1.exists)(json, 'check_username') ? undefined : json['check_username'],
        'threshold': !(0, runtime_1.exists)(json, 'threshold') ? undefined : json['threshold'],
    };
}
exports.ReputationPolicyRequestFromJSONTyped = ReputationPolicyRequestFromJSONTyped;
function ReputationPolicyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'execution_logging': value.executionLogging,
        'check_ip': value.checkIp,
        'check_username': value.checkUsername,
        'threshold': value.threshold,
    };
}
exports.ReputationPolicyRequestToJSON = ReputationPolicyRequestToJSON;
