// @goauthentik/api/dist/models/Reputation.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReputationToJSON = exports.ReputationFromJSONTyped = exports.ReputationFromJSON = exports.instanceOfReputation = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the Reputation interface.
 */
function instanceOfReputation(value) {
    let isInstance = true;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "ip" in value;
    isInstance = isInstance && "updated" in value;
    return isInstance;
}
exports.instanceOfReputation = instanceOfReputation;
function ReputationFromJSON(json) {
    return ReputationFromJSONTyped(json, false);
}
exports.ReputationFromJSON = ReputationFromJSON;
function ReputationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': !(0, runtime_1.exists)(json, 'pk') ? undefined : json['pk'],
        'identifier': json['identifier'],
        'ip': json['ip'],
        'ipGeoData': !(0, runtime_1.exists)(json, 'ip_geo_data') ? undefined : json['ip_geo_data'],
        'ipAsnData': !(0, runtime_1.exists)(json, 'ip_asn_data') ? undefined : json['ip_asn_data'],
        'score': !(0, runtime_1.exists)(json, 'score') ? undefined : json['score'],
        'updated': (new Date(json['updated'])),
    };
}
exports.ReputationFromJSONTyped = ReputationFromJSONTyped;
function ReputationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pk': value.pk,
        'identifier': value.identifier,
        'ip': value.ip,
        'ip_geo_data': value.ipGeoData,
        'ip_asn_data': value.ipAsnData,
        'score': value.score,
    };
}
exports.ReputationToJSON = ReputationToJSON;
