// @goauthentik/api/dist/models/ResidentKeyRequirementEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResidentKeyRequirementEnumToJSON = exports.ResidentKeyRequirementEnumFromJSONTyped = exports.ResidentKeyRequirementEnumFromJSON = exports.ResidentKeyRequirementEnum = void 0;
/**
 * * `discouraged` - Discouraged
 * * `preferred` - Preferred
 * * `required` - Required
 * @export
 */
exports.ResidentKeyRequirementEnum = {
    Discouraged: 'discouraged',
    Preferred: 'preferred',
    Required: 'required',
    UnknownDefaultOpenApi: '11184809'
};
function ResidentKeyRequirementEnumFromJSON(json) {
    return ResidentKeyRequirementEnumFromJSONTyped(json, false);
}
exports.ResidentKeyRequirementEnumFromJSON = ResidentKeyRequirementEnumFromJSON;
function ResidentKeyRequirementEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ResidentKeyRequirementEnumFromJSONTyped = ResidentKeyRequirementEnumFromJSONTyped;
function ResidentKeyRequirementEnumToJSON(value) {
    return value;
}
exports.ResidentKeyRequirementEnumToJSON = ResidentKeyRequirementEnumToJSON;
