// @goauthentik/api/dist/models/RoleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleRequestToJSON = exports.RoleRequestFromJSONTyped = exports.RoleRequestFromJSON = exports.instanceOfRoleRequest = void 0;
/**
 * Check if a given object implements the RoleRequest interface.
 */
function instanceOfRoleRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfRoleRequest = instanceOfRoleRequest;
function RoleRequestFromJSON(json) {
    return RoleRequestFromJSONTyped(json, false);
}
exports.RoleRequestFromJSON = RoleRequestFromJSON;
function RoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.RoleRequestFromJSONTyped = RoleRequestFromJSONTyped;
function RoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.RoleRequestToJSON = RoleRequestToJSON;
