// @goauthentik/api/dist/models/RoleAssignedObjectPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleAssignedObjectPermissionToJSON = exports.RoleAssignedObjectPermissionFromJSONTyped = exports.RoleAssignedObjectPermissionFromJSON = exports.instanceOfRoleAssignedObjectPermission = void 0;
const RoleObjectPermission_1 = require("./RoleObjectPermission");
/**
 * Check if a given object implements the RoleAssignedObjectPermission interface.
 */
function instanceOfRoleAssignedObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "rolePk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "permissions" in value;
    return isInstance;
}
exports.instanceOfRoleAssignedObjectPermission = instanceOfRoleAssignedObjectPermission;
function RoleAssignedObjectPermissionFromJSON(json) {
    return RoleAssignedObjectPermissionFromJSONTyped(json, false);
}
exports.RoleAssignedObjectPermissionFromJSON = RoleAssignedObjectPermissionFromJSON;
function RoleAssignedObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'rolePk': json['role_pk'],
        'name': json['name'],
        'permissions': (json['permissions'].map(RoleObjectPermission_1.RoleObjectPermissionFromJSON)),
    };
}
exports.RoleAssignedObjectPermissionFromJSONTyped = RoleAssignedObjectPermissionFromJSONTyped;
function RoleAssignedObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': (value.permissions.map(RoleObjectPermission_1.RoleObjectPermissionToJSON)),
    };
}
exports.RoleAssignedObjectPermissionToJSON = RoleAssignedObjectPermissionToJSON;
