// @goauthentik/api/dist/models/SAMLMetadata.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLMetadataToJSON = exports.SAMLMetadataFromJSONTyped = exports.SAMLMetadataFromJSON = exports.instanceOfSAMLMetadata = void 0;
/**
 * Check if a given object implements the SAMLMetadata interface.
 */
function instanceOfSAMLMetadata(value) {
    let isInstance = true;
    isInstance = isInstance && "metadata" in value;
    isInstance = isInstance && "downloadUrl" in value;
    return isInstance;
}
exports.instanceOfSAMLMetadata = instanceOfSAMLMetadata;
function SAMLMetadataFromJSON(json) {
    return SAMLMetadataFromJSONTyped(json, false);
}
exports.SAMLMetadataFromJSON = SAMLMetadataFromJSON;
function SAMLMetadataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'metadata': json['metadata'],
        'downloadUrl': json['download_url'],
    };
}
exports.SAMLMetadataFromJSONTyped = SAMLMetadataFromJSONTyped;
function SAMLMetadataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.SAMLMetadataToJSON = SAMLMetadataToJSON;
