// @goauthentik/api/dist/models/SAMLPropertyMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLPropertyMappingRequestToJSON = exports.SAMLPropertyMappingRequestFromJSONTyped = exports.SAMLPropertyMappingRequestFromJSON = exports.instanceOfSAMLPropertyMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the SAMLPropertyMappingRequest interface.
 */
function instanceOfSAMLPropertyMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "samlName" in value;
    return isInstance;
}
exports.instanceOfSAMLPropertyMappingRequest = instanceOfSAMLPropertyMappingRequest;
function SAMLPropertyMappingRequestFromJSON(json) {
    return SAMLPropertyMappingRequestFromJSONTyped(json, false);
}
exports.SAMLPropertyMappingRequestFromJSON = SAMLPropertyMappingRequestFromJSON;
function SAMLPropertyMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'samlName': json['saml_name'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}
exports.SAMLPropertyMappingRequestFromJSONTyped = SAMLPropertyMappingRequestFromJSONTyped;
function SAMLPropertyMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
exports.SAMLPropertyMappingRequestToJSON = SAMLPropertyMappingRequestToJSON;
