// @goauthentik/api/dist/models/SAMLPropertyMapping.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLPropertyMappingToJSON = exports.SAMLPropertyMappingFromJSONTyped = exports.SAMLPropertyMappingFromJSON = exports.instanceOfSAMLPropertyMapping = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the SAMLPropertyMapping interface.
 */
function instanceOfSAMLPropertyMapping(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "samlName" in value;
    return isInstance;
}
exports.instanceOfSAMLPropertyMapping = instanceOfSAMLPropertyMapping;
function SAMLPropertyMappingFromJSON(json) {
    return SAMLPropertyMappingFromJSONTyped(json, false);
}
exports.SAMLPropertyMappingFromJSON = SAMLPropertyMappingFromJSON;
function SAMLPropertyMappingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'samlName': json['saml_name'],
        'friendlyName': !(0, runtime_1.exists)(json, 'friendly_name') ? undefined : json['friendly_name'],
    };
}
exports.SAMLPropertyMappingFromJSONTyped = SAMLPropertyMappingFromJSONTyped;
function SAMLPropertyMappingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'saml_name': value.samlName,
        'friendly_name': value.friendlyName,
    };
}
exports.SAMLPropertyMappingToJSON = SAMLPropertyMappingToJSON;
