// @goauthentik/api/dist/models/SCIMProviderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCIMProviderRequestToJSON = exports.SCIMProviderRequestFromJSONTyped = exports.SCIMProviderRequestFromJSON = exports.instanceOfSCIMProviderRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the SCIMProviderRequest interface.
 */
function instanceOfSCIMProviderRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfSCIMProviderRequest = instanceOfSCIMProviderRequest;
function SCIMProviderRequestFromJSON(json) {
    return SCIMProviderRequestFromJSONTyped(json, false);
}
exports.SCIMProviderRequestFromJSON = SCIMProviderRequestFromJSON;
function SCIMProviderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'propertyMappings': !(0, runtime_1.exists)(json, 'property_mappings') ? undefined : json['property_mappings'],
        'propertyMappingsGroup': !(0, runtime_1.exists)(json, 'property_mappings_group') ? undefined : json['property_mappings_group'],
        'url': json['url'],
        'token': json['token'],
        'excludeUsersServiceAccount': !(0, runtime_1.exists)(json, 'exclude_users_service_account') ? undefined : json['exclude_users_service_account'],
        'filterGroup': !(0, runtime_1.exists)(json, 'filter_group') ? undefined : json['filter_group'],
    };
}
exports.SCIMProviderRequestFromJSONTyped = SCIMProviderRequestFromJSONTyped;
function SCIMProviderRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'property_mappings': value.propertyMappings,
        'property_mappings_group': value.propertyMappingsGroup,
        'url': value.url,
        'token': value.token,
        'exclude_users_service_account': value.excludeUsersServiceAccount,
        'filter_group': value.filterGroup,
    };
}
exports.SCIMProviderRequestToJSON = SCIMProviderRequestToJSON;
