// @goauthentik/api/dist/models/SCIMMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCIMMappingRequestToJSON = exports.SCIMMappingRequestFromJSONTyped = exports.SCIMMappingRequestFromJSON = exports.instanceOfSCIMMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the SCIMMappingRequest interface.
 */
function instanceOfSCIMMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    return isInstance;
}
exports.instanceOfSCIMMappingRequest = instanceOfSCIMMappingRequest;
function SCIMMappingRequestFromJSON(json) {
    return SCIMMappingRequestFromJSONTyped(json, false);
}
exports.SCIMMappingRequestFromJSON = SCIMMappingRequestFromJSON;
function SCIMMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
    };
}
exports.SCIMMappingRequestFromJSONTyped = SCIMMappingRequestFromJSONTyped;
function SCIMMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
    };
}
exports.SCIMMappingRequestToJSON = SCIMMappingRequestToJSON;
