// @goauthentik/api/dist/models/SCIMSyncStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCIMSyncStatusToJSON = exports.SCIMSyncStatusFromJSONTyped = exports.SCIMSyncStatusFromJSON = exports.instanceOfSCIMSyncStatus = void 0;
const SystemTask_1 = require("./SystemTask");
/**
 * Check if a given object implements the SCIMSyncStatus interface.
 */
function instanceOfSCIMSyncStatus(value) {
    let isInstance = true;
    isInstance = isInstance && "isRunning" in value;
    isInstance = isInstance && "tasks" in value;
    return isInstance;
}
exports.instanceOfSCIMSyncStatus = instanceOfSCIMSyncStatus;
function SCIMSyncStatusFromJSON(json) {
    return SCIMSyncStatusFromJSONTyped(json, false);
}
exports.SCIMSyncStatusFromJSON = SCIMSyncStatusFromJSON;
function SCIMSyncStatusFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'isRunning': json['is_running'],
        'tasks': (json['tasks'].map(SystemTask_1.SystemTaskFromJSON)),
    };
}
exports.SCIMSyncStatusFromJSONTyped = SCIMSyncStatusFromJSONTyped;
function SCIMSyncStatusToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.SCIMSyncStatusToJSON = SCIMSyncStatusToJSON;
