// @goauthentik/api/dist/models/ScopeMappingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeMappingRequestToJSON = exports.ScopeMappingRequestFromJSONTyped = exports.ScopeMappingRequestFromJSON = exports.instanceOfScopeMappingRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ScopeMappingRequest interface.
 */
function instanceOfScopeMappingRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "scopeName" in value;
    return isInstance;
}
exports.instanceOfScopeMappingRequest = instanceOfScopeMappingRequest;
function ScopeMappingRequestFromJSON(json) {
    return ScopeMappingRequestFromJSONTyped(json, false);
}
exports.ScopeMappingRequestFromJSON = ScopeMappingRequestFromJSON;
function ScopeMappingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'scopeName': json['scope_name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
    };
}
exports.ScopeMappingRequestFromJSONTyped = ScopeMappingRequestFromJSONTyped;
function ScopeMappingRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
exports.ScopeMappingRequestToJSON = ScopeMappingRequestToJSON;
