// @goauthentik/api/dist/models/ScopeMapping.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeMappingToJSON = exports.ScopeMappingFromJSONTyped = exports.ScopeMappingFromJSON = exports.instanceOfScopeMapping = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ScopeMapping interface.
 */
function instanceOfScopeMapping(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "expression" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    isInstance = isInstance && "scopeName" in value;
    return isInstance;
}
exports.instanceOfScopeMapping = instanceOfScopeMapping;
function ScopeMappingFromJSON(json) {
    return ScopeMappingFromJSONTyped(json, false);
}
exports.ScopeMappingFromJSON = ScopeMappingFromJSON;
function ScopeMappingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'name': json['name'],
        'expression': json['expression'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'scopeName': json['scope_name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
    };
}
exports.ScopeMappingFromJSONTyped = ScopeMappingFromJSONTyped;
function ScopeMappingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'name': value.name,
        'expression': value.expression,
        'scope_name': value.scopeName,
        'description': value.description,
    };
}
exports.ScopeMappingToJSON = ScopeMappingToJSON;
