// @goauthentik/api/dist/models/ServiceConnection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceConnectionToJSON = exports.ServiceConnectionFromJSONTyped = exports.ServiceConnectionFromJSON = exports.instanceOfServiceConnection = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ServiceConnection interface.
 */
function instanceOfServiceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfServiceConnection = instanceOfServiceConnection;
function ServiceConnectionFromJSON(json) {
    return ServiceConnectionFromJSONTyped(json, false);
}
exports.ServiceConnectionFromJSON = ServiceConnectionFromJSON;
function ServiceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
    };
}
exports.ServiceConnectionFromJSONTyped = ServiceConnectionFromJSONTyped;
function ServiceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
    };
}
exports.ServiceConnectionToJSON = ServiceConnectionToJSON;
