// @goauthentik/api/dist/models/ServiceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceConnectionRequestToJSON = exports.ServiceConnectionRequestFromJSONTyped = exports.ServiceConnectionRequestFromJSON = exports.instanceOfServiceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ServiceConnectionRequest interface.
 */
function instanceOfServiceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfServiceConnectionRequest = instanceOfServiceConnectionRequest;
function ServiceConnectionRequestFromJSON(json) {
    return ServiceConnectionRequestFromJSONTyped(json, false);
}
exports.ServiceConnectionRequestFromJSON = ServiceConnectionRequestFromJSON;
function ServiceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'local': !(0, runtime_1.exists)(json, 'local') ? undefined : json['local'],
    };
}
exports.ServiceConnectionRequestFromJSONTyped = ServiceConnectionRequestFromJSONTyped;
function ServiceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'local': value.local,
    };
}
exports.ServiceConnectionRequestToJSON = ServiceConnectionRequestToJSON;
