// @goauthentik/api/dist/models/ServiceConnectionState.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceConnectionStateToJSON = exports.ServiceConnectionStateFromJSONTyped = exports.ServiceConnectionStateFromJSON = exports.instanceOfServiceConnectionState = void 0;
/**
 * Check if a given object implements the ServiceConnectionState interface.
 */
function instanceOfServiceConnectionState(value) {
    let isInstance = true;
    isInstance = isInstance && "healthy" in value;
    isInstance = isInstance && "version" in value;
    return isInstance;
}
exports.instanceOfServiceConnectionState = instanceOfServiceConnectionState;
function ServiceConnectionStateFromJSON(json) {
    return ServiceConnectionStateFromJSONTyped(json, false);
}
exports.ServiceConnectionStateFromJSON = ServiceConnectionStateFromJSON;
function ServiceConnectionStateFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'healthy': json['healthy'],
        'version': json['version'],
    };
}
exports.ServiceConnectionStateFromJSONTyped = ServiceConnectionStateFromJSONTyped;
function ServiceConnectionStateToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.ServiceConnectionStateToJSON = ServiceConnectionStateToJSON;
