// @goauthentik/api/dist/models/SessionUser.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionUserToJSON = exports.SessionUserFromJSONTyped = exports.SessionUserFromJSON = exports.instanceOfSessionUser = void 0;
const runtime_1 = require("../runtime");
const UserSelf_1 = require("./UserSelf");
/**
 * Check if a given object implements the SessionUser interface.
 */
function instanceOfSessionUser(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfSessionUser = instanceOfSessionUser;
function SessionUserFromJSON(json) {
    return SessionUserFromJSONTyped(json, false);
}
exports.SessionUserFromJSON = SessionUserFromJSON;
function SessionUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, UserSelf_1.UserSelfFromJSON)(json['user']),
        'original': !(0, runtime_1.exists)(json, 'original') ? undefined : (0, UserSelf_1.UserSelfFromJSON)(json['original']),
    };
}
exports.SessionUserFromJSONTyped = SessionUserFromJSONTyped;
function SessionUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, UserSelf_1.UserSelfToJSON)(value.user),
        'original': (0, UserSelf_1.UserSelfToJSON)(value.original),
    };
}
exports.SessionUserToJSON = SessionUserToJSON;
