// @goauthentik/api/dist/models/SeverityEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeverityEnumToJSON = exports.SeverityEnumFromJSONTyped = exports.SeverityEnumFromJSON = exports.SeverityEnum = void 0;
/**
 * * `notice` - Notice
 * * `warning` - Warning
 * * `alert` - Alert
 * @export
 */
exports.SeverityEnum = {
    Notice: 'notice',
    Warning: 'warning',
    Alert: 'alert',
    UnknownDefaultOpenApi: '11184809'
};
function SeverityEnumFromJSON(json) {
    return SeverityEnumFromJSONTyped(json, false);
}
exports.SeverityEnumFromJSON = SeverityEnumFromJSON;
function SeverityEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.SeverityEnumFromJSONTyped = SeverityEnumFromJSONTyped;
function SeverityEnumToJSON(value) {
    return value;
}
exports.SeverityEnumToJSON = SeverityEnumToJSON;
