// @goauthentik/api/dist/models/SignatureAlgorithmEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignatureAlgorithmEnumToJSON = exports.SignatureAlgorithmEnumFromJSONTyped = exports.SignatureAlgorithmEnumFromJSON = exports.SignatureAlgorithmEnum = void 0;
/**
 * * `http://www.w3.org/2000/09/xmldsig#rsa-sha1` - RSA-SHA1
 * * `http://www.w3.org/2001/04/xmldsig-more#rsa-sha256` - RSA-SHA256
 * * `http://www.w3.org/2001/04/xmldsig-more#rsa-sha384` - RSA-SHA384
 * * `http://www.w3.org/2001/04/xmldsig-more#rsa-sha512` - RSA-SHA512
 * * `http://www.w3.org/2000/09/xmldsig#dsa-sha1` - DSA-SHA1
 * @export
 */
exports.SignatureAlgorithmEnum = {
    _200009XmldsigrsaSha1: 'http://www.w3.org/2000/09/xmldsig#rsa-sha1',
    _200104XmldsigMorersaSha256: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256',
    _200104XmldsigMorersaSha384: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384',
    _200104XmldsigMorersaSha512: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512',
    _200009XmldsigdsaSha1: 'http://www.w3.org/2000/09/xmldsig#dsa-sha1',
    UnknownDefaultOpenApi: '11184809'
};
function SignatureAlgorithmEnumFromJSON(json) {
    return SignatureAlgorithmEnumFromJSONTyped(json, false);
}
exports.SignatureAlgorithmEnumFromJSON = SignatureAlgorithmEnumFromJSON;
function SignatureAlgorithmEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.SignatureAlgorithmEnumFromJSONTyped = SignatureAlgorithmEnumFromJSONTyped;
function SignatureAlgorithmEnumToJSON(value) {
    return value;
}
exports.SignatureAlgorithmEnumToJSON = SignatureAlgorithmEnumToJSON;
