// @goauthentik/api/dist/models/SMSDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSDeviceRequestToJSON = exports.SMSDeviceRequestFromJSONTyped = exports.SMSDeviceRequestFromJSON = exports.instanceOfSMSDeviceRequest = void 0;
/**
 * Check if a given object implements the SMSDeviceRequest interface.
 */
function instanceOfSMSDeviceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfSMSDeviceRequest = instanceOfSMSDeviceRequest;
function SMSDeviceRequestFromJSON(json) {
    return SMSDeviceRequestFromJSONTyped(json, false);
}
exports.SMSDeviceRequestFromJSON = SMSDeviceRequestFromJSON;
function SMSDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.SMSDeviceRequestFromJSONTyped = SMSDeviceRequestFromJSONTyped;
function SMSDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.SMSDeviceRequestToJSON = SMSDeviceRequestToJSON;
