// @goauthentik/api/dist/models/SMSDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSDeviceToJSON = exports.SMSDeviceFromJSONTyped = exports.SMSDeviceFromJSON = exports.instanceOfSMSDevice = void 0;
/**
 * Check if a given object implements the SMSDevice interface.
 */
function instanceOfSMSDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "phoneNumber" in value;
    return isInstance;
}
exports.instanceOfSMSDevice = instanceOfSMSDevice;
function SMSDeviceFromJSON(json) {
    return SMSDeviceFromJSONTyped(json, false);
}
exports.SMSDeviceFromJSON = SMSDeviceFromJSON;
function SMSDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'pk': json['pk'],
        'phoneNumber': json['phone_number'],
    };
}
exports.SMSDeviceFromJSONTyped = SMSDeviceFromJSONTyped;
function SMSDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.SMSDeviceToJSON = SMSDeviceToJSON;
