// @goauthentik/api/dist/models/SpBindingEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpBindingEnumToJSON = exports.SpBindingEnumFromJSONTyped = exports.SpBindingEnumFromJSON = exports.SpBindingEnum = void 0;
/**
 * * `redirect` - Redirect
 * * `post` - Post
 * @export
 */
exports.SpBindingEnum = {
    Redirect: 'redirect',
    Post: 'post',
    UnknownDefaultOpenApi: '11184809'
};
function SpBindingEnumFromJSON(json) {
    return SpBindingEnumFromJSONTyped(json, false);
}
exports.SpBindingEnumFromJSON = SpBindingEnumFromJSON;
function SpBindingEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.SpBindingEnumFromJSONTyped = SpBindingEnumFromJSONTyped;
function SpBindingEnumToJSON(value) {
    return value;
}
exports.SpBindingEnumToJSON = SpBindingEnumToJSON;
