// @goauthentik/api/dist/models/StaticDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticDeviceToJSON = exports.StaticDeviceFromJSONTyped = exports.StaticDeviceFromJSON = exports.instanceOfStaticDevice = void 0;
const StaticDeviceToken_1 = require("./StaticDeviceToken");
/**
 * Check if a given object implements the StaticDevice interface.
 */
function instanceOfStaticDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "tokenSet" in value;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
exports.instanceOfStaticDevice = instanceOfStaticDevice;
function StaticDeviceFromJSON(json) {
    return StaticDeviceFromJSONTyped(json, false);
}
exports.StaticDeviceFromJSON = StaticDeviceFromJSON;
function StaticDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'tokenSet': (json['token_set'].map(StaticDeviceToken_1.StaticDeviceTokenFromJSON)),
        'pk': json['pk'],
    };
}
exports.StaticDeviceFromJSONTyped = StaticDeviceFromJSONTyped;
function StaticDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.StaticDeviceToJSON = StaticDeviceToJSON;
