// @goauthentik/api/dist/models/StaticDeviceTokenRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticDeviceTokenRequestToJSON = exports.StaticDeviceTokenRequestFromJSONTyped = exports.StaticDeviceTokenRequestFromJSON = exports.instanceOfStaticDeviceTokenRequest = void 0;
/**
 * Check if a given object implements the StaticDeviceTokenRequest interface.
 */
function instanceOfStaticDeviceTokenRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfStaticDeviceTokenRequest = instanceOfStaticDeviceTokenRequest;
function StaticDeviceTokenRequestFromJSON(json) {
    return StaticDeviceTokenRequestFromJSONTyped(json, false);
}
exports.StaticDeviceTokenRequestFromJSON = StaticDeviceTokenRequestFromJSON;
function StaticDeviceTokenRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'token': json['token'],
    };
}
exports.StaticDeviceTokenRequestFromJSONTyped = StaticDeviceTokenRequestFromJSONTyped;
function StaticDeviceTokenRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token': value.token,
    };
}
exports.StaticDeviceTokenRequestToJSON = StaticDeviceTokenRequestToJSON;
