// @goauthentik/api/dist/models/StaticDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticDeviceRequestToJSON = exports.StaticDeviceRequestFromJSONTyped = exports.StaticDeviceRequestFromJSON = exports.instanceOfStaticDeviceRequest = void 0;
/**
 * Check if a given object implements the StaticDeviceRequest interface.
 */
function instanceOfStaticDeviceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfStaticDeviceRequest = instanceOfStaticDeviceRequest;
function StaticDeviceRequestFromJSON(json) {
    return StaticDeviceRequestFromJSONTyped(json, false);
}
exports.StaticDeviceRequestFromJSON = StaticDeviceRequestFromJSON;
function StaticDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.StaticDeviceRequestFromJSONTyped = StaticDeviceRequestFromJSONTyped;
function StaticDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.StaticDeviceRequestToJSON = StaticDeviceRequestToJSON;
