// @goauthentik/api/dist/models/SystemInfoRuntime.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemInfoRuntimeToJSON = exports.SystemInfoRuntimeFromJSONTyped = exports.SystemInfoRuntimeFromJSON = exports.instanceOfSystemInfoRuntime = void 0;
/**
 * Check if a given object implements the SystemInfoRuntime interface.
 */
function instanceOfSystemInfoRuntime(value) {
    let isInstance = true;
    isInstance = isInstance && "pythonVersion" in value;
    isInstance = isInstance && "gunicornVersion" in value;
    isInstance = isInstance && "environment" in value;
    isInstance = isInstance && "architecture" in value;
    isInstance = isInstance && "platform" in value;
    isInstance = isInstance && "uname" in value;
    return isInstance;
}
exports.instanceOfSystemInfoRuntime = instanceOfSystemInfoRuntime;
function SystemInfoRuntimeFromJSON(json) {
    return SystemInfoRuntimeFromJSONTyped(json, false);
}
exports.SystemInfoRuntimeFromJSON = SystemInfoRuntimeFromJSON;
function SystemInfoRuntimeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pythonVersion': json['python_version'],
        'gunicornVersion': json['gunicorn_version'],
        'environment': json['environment'],
        'architecture': json['architecture'],
        'platform': json['platform'],
        'uname': json['uname'],
    };
}
exports.SystemInfoRuntimeFromJSONTyped = SystemInfoRuntimeFromJSONTyped;
function SystemInfoRuntimeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'python_version': value.pythonVersion,
        'gunicorn_version': value.gunicornVersion,
        'environment': value.environment,
        'architecture': value.architecture,
        'platform': value.platform,
        'uname': value.uname,
    };
}
exports.SystemInfoRuntimeToJSON = SystemInfoRuntimeToJSON;
