// @goauthentik/api/dist/models/SystemTaskStatusEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemTaskStatusEnumToJSON = exports.SystemTaskStatusEnumFromJSONTyped = exports.SystemTaskStatusEnumFromJSON = exports.SystemTaskStatusEnum = void 0;
/**
 * * `unknown` - UNKNOWN
 * * `successful` - SUCCESSFUL
 * * `warning` - WARNING
 * * `error` - ERROR
 * @export
 */
exports.SystemTaskStatusEnum = {
    Unknown: 'unknown',
    Successful: 'successful',
    Warning: 'warning',
    Error: 'error',
    UnknownDefaultOpenApi: '11184809'
};
function SystemTaskStatusEnumFromJSON(json) {
    return SystemTaskStatusEnumFromJSONTyped(json, false);
}
exports.SystemTaskStatusEnumFromJSON = SystemTaskStatusEnumFromJSON;
function SystemTaskStatusEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.SystemTaskStatusEnumFromJSONTyped = SystemTaskStatusEnumFromJSONTyped;
function SystemTaskStatusEnumToJSON(value) {
    return value;
}
exports.SystemTaskStatusEnumToJSON = SystemTaskStatusEnumToJSON;
