// @goauthentik/api/dist/models/SystemTask.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemTaskToJSON = exports.SystemTaskFromJSONTyped = exports.SystemTaskFromJSON = exports.instanceOfSystemTask = void 0;
const runtime_1 = require("../runtime");
const SystemTaskStatusEnum_1 = require("./SystemTaskStatusEnum");
/**
 * Check if a given object implements the SystemTask interface.
 */
function instanceOfSystemTask(value) {
    let isInstance = true;
    isInstance = isInstance && "uuid" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "fullName" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "startTimestamp" in value;
    isInstance = isInstance && "finishTimestamp" in value;
    isInstance = isInstance && "duration" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "messages" in value;
    return isInstance;
}
exports.instanceOfSystemTask = instanceOfSystemTask;
function SystemTaskFromJSON(json) {
    return SystemTaskFromJSONTyped(json, false);
}
exports.SystemTaskFromJSON = SystemTaskFromJSON;
function SystemTaskFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'uuid': json['uuid'],
        'name': json['name'],
        'fullName': json['full_name'],
        'uid': !(0, runtime_1.exists)(json, 'uid') ? undefined : json['uid'],
        'description': json['description'],
        'startTimestamp': (new Date(json['start_timestamp'])),
        'finishTimestamp': (new Date(json['finish_timestamp'])),
        'duration': json['duration'],
        'status': (0, SystemTaskStatusEnum_1.SystemTaskStatusEnumFromJSON)(json['status']),
        'messages': json['messages'],
    };
}
exports.SystemTaskFromJSONTyped = SystemTaskFromJSONTyped;
function SystemTaskToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'uid': value.uid,
        'description': value.description,
        'status': (0, SystemTaskStatusEnum_1.SystemTaskStatusEnumToJSON)(value.status),
        'messages': value.messages,
    };
}
exports.SystemTaskToJSON = SystemTaskToJSON;
