// @goauthentik/api/dist/models/TenantRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantRequestToJSON = exports.TenantRequestFromJSONTyped = exports.TenantRequestFromJSON = exports.instanceOfTenantRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TenantRequest interface.
 */
function instanceOfTenantRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "schemaName" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfTenantRequest = instanceOfTenantRequest;
function TenantRequestFromJSON(json) {
    return TenantRequestFromJSONTyped(json, false);
}
exports.TenantRequestFromJSON = TenantRequestFromJSON;
function TenantRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'schemaName': json['schema_name'],
        'name': json['name'],
        'ready': !(0, runtime_1.exists)(json, 'ready') ? undefined : json['ready'],
    };
}
exports.TenantRequestFromJSONTyped = TenantRequestFromJSONTyped;
function TenantRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
exports.TenantRequestToJSON = TenantRequestToJSON;
