// @goauthentik/api/dist/models/Tenant.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantToJSON = exports.TenantFromJSONTyped = exports.TenantFromJSON = exports.instanceOfTenant = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the Tenant interface.
 */
function instanceOfTenant(value) {
    let isInstance = true;
    isInstance = isInstance && "tenantUuid" in value;
    isInstance = isInstance && "schemaName" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfTenant = instanceOfTenant;
function TenantFromJSON(json) {
    return TenantFromJSONTyped(json, false);
}
exports.TenantFromJSON = TenantFromJSON;
function TenantFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tenantUuid': json['tenant_uuid'],
        'schemaName': json['schema_name'],
        'name': json['name'],
        'ready': !(0, runtime_1.exists)(json, 'ready') ? undefined : json['ready'],
    };
}
exports.TenantFromJSONTyped = TenantFromJSONTyped;
function TenantToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'schema_name': value.schemaName,
        'name': value.name,
        'ready': value.ready,
    };
}
exports.TenantToJSON = TenantToJSON;
