// @goauthentik/api/dist/models/TenantAdminGroupRequestRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantAdminGroupRequestRequestToJSON = exports.TenantAdminGroupRequestRequestFromJSONTyped = exports.TenantAdminGroupRequestRequestFromJSON = exports.instanceOfTenantAdminGroupRequestRequest = void 0;
/**
 * Check if a given object implements the TenantAdminGroupRequestRequest interface.
 */
function instanceOfTenantAdminGroupRequestRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfTenantAdminGroupRequestRequest = instanceOfTenantAdminGroupRequestRequest;
function TenantAdminGroupRequestRequestFromJSON(json) {
    return TenantAdminGroupRequestRequestFromJSONTyped(json, false);
}
exports.TenantAdminGroupRequestRequestFromJSON = TenantAdminGroupRequestRequestFromJSON;
function TenantAdminGroupRequestRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
    };
}
exports.TenantAdminGroupRequestRequestFromJSONTyped = TenantAdminGroupRequestRequestFromJSONTyped;
function TenantAdminGroupRequestRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
    };
}
exports.TenantAdminGroupRequestRequestToJSON = TenantAdminGroupRequestRequestToJSON;
