// @goauthentik/api/dist/models/TenantRecoveryKeyRequestRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantRecoveryKeyRequestRequestToJSON = exports.TenantRecoveryKeyRequestRequestFromJSONTyped = exports.TenantRecoveryKeyRequestRequestFromJSON = exports.instanceOfTenantRecoveryKeyRequestRequest = void 0;
/**
 * Check if a given object implements the TenantRecoveryKeyRequestRequest interface.
 */
function instanceOfTenantRecoveryKeyRequestRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "durationDays" in value;
    return isInstance;
}
exports.instanceOfTenantRecoveryKeyRequestRequest = instanceOfTenantRecoveryKeyRequestRequest;
function TenantRecoveryKeyRequestRequestFromJSON(json) {
    return TenantRecoveryKeyRequestRequestFromJSONTyped(json, false);
}
exports.TenantRecoveryKeyRequestRequestFromJSON = TenantRecoveryKeyRequestRequestFromJSON;
function TenantRecoveryKeyRequestRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'durationDays': json['duration_days'],
    };
}
exports.TenantRecoveryKeyRequestRequestFromJSONTyped = TenantRecoveryKeyRequestRequestFromJSONTyped;
function TenantRecoveryKeyRequestRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'duration_days': value.durationDays,
    };
}
exports.TenantRecoveryKeyRequestRequestToJSON = TenantRecoveryKeyRequestRequestToJSON;
