// @goauthentik/api/dist/models/TenantRecoveryKeyResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantRecoveryKeyResponseToJSON = exports.TenantRecoveryKeyResponseFromJSONTyped = exports.TenantRecoveryKeyResponseFromJSON = exports.instanceOfTenantRecoveryKeyResponse = void 0;
/**
 * Check if a given object implements the TenantRecoveryKeyResponse interface.
 */
function instanceOfTenantRecoveryKeyResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "expiry" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfTenantRecoveryKeyResponse = instanceOfTenantRecoveryKeyResponse;
function TenantRecoveryKeyResponseFromJSON(json) {
    return TenantRecoveryKeyResponseFromJSONTyped(json, false);
}
exports.TenantRecoveryKeyResponseFromJSON = TenantRecoveryKeyResponseFromJSON;
function TenantRecoveryKeyResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'expiry': (new Date(json['expiry'])),
        'url': json['url'],
    };
}
exports.TenantRecoveryKeyResponseFromJSONTyped = TenantRecoveryKeyResponseFromJSONTyped;
function TenantRecoveryKeyResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'expiry': (value.expiry.toISOString()),
        'url': value.url,
    };
}
exports.TenantRecoveryKeyResponseToJSON = TenantRecoveryKeyResponseToJSON;
