// @goauthentik/api/dist/models/Token.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenToJSON = exports.TokenFromJSONTyped = exports.TokenFromJSON = exports.instanceOfToken = void 0;
const runtime_1 = require("../runtime");
const IntentEnum_1 = require("./IntentEnum");
const User_1 = require("./User");
/**
 * Check if a given object implements the Token interface.
 */
function instanceOfToken(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "identifier" in value;
    isInstance = isInstance && "userObj" in value;
    return isInstance;
}
exports.instanceOfToken = instanceOfToken;
function TokenFromJSON(json) {
    return TokenFromJSONTyped(json, false);
}
exports.TokenFromJSON = TokenFromJSON;
function TokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'managed': !(0, runtime_1.exists)(json, 'managed') ? undefined : json['managed'],
        'identifier': json['identifier'],
        'intent': !(0, runtime_1.exists)(json, 'intent') ? undefined : (0, IntentEnum_1.IntentEnumFromJSON)(json['intent']),
        'user': !(0, runtime_1.exists)(json, 'user') ? undefined : json['user'],
        'userObj': (0, User_1.UserFromJSON)(json['user_obj']),
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'expiring': !(0, runtime_1.exists)(json, 'expiring') ? undefined : json['expiring'],
    };
}
exports.TokenFromJSONTyped = TokenFromJSONTyped;
function TokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'managed': value.managed,
        'identifier': value.identifier,
        'intent': (0, IntentEnum_1.IntentEnumToJSON)(value.intent),
        'user': value.user,
        'description': value.description,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'expiring': value.expiring,
    };
}
exports.TokenToJSON = TokenToJSON;
