// @goauthentik/api/dist/models/TokenView.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenViewToJSON = exports.TokenViewFromJSONTyped = exports.TokenViewFromJSON = exports.instanceOfTokenView = void 0;
/**
 * Check if a given object implements the TokenView interface.
 */
function instanceOfTokenView(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
exports.instanceOfTokenView = instanceOfTokenView;
function TokenViewFromJSON(json) {
    return TokenViewFromJSONTyped(json, false);
}
exports.TokenViewFromJSON = TokenViewFromJSON;
function TokenViewFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
exports.TokenViewFromJSONTyped = TokenViewFromJSONTyped;
function TokenViewToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.TokenViewToJSON = TokenViewToJSON;
