// @goauthentik/api/dist/models/TokenSetKeyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenSetKeyRequestToJSON = exports.TokenSetKeyRequestFromJSONTyped = exports.TokenSetKeyRequestFromJSON = exports.instanceOfTokenSetKeyRequest = void 0;
/**
 * Check if a given object implements the TokenSetKeyRequest interface.
 */
function instanceOfTokenSetKeyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
exports.instanceOfTokenSetKeyRequest = instanceOfTokenSetKeyRequest;
function TokenSetKeyRequestFromJSON(json) {
    return TokenSetKeyRequestFromJSONTyped(json, false);
}
exports.TokenSetKeyRequestFromJSON = TokenSetKeyRequestFromJSON;
function TokenSetKeyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': json['key'],
    };
}
exports.TokenSetKeyRequestFromJSONTyped = TokenSetKeyRequestFromJSONTyped;
function TokenSetKeyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
    };
}
exports.TokenSetKeyRequestToJSON = TokenSetKeyRequestToJSON;
