// @goauthentik/api/dist/models/TOTPDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TOTPDeviceToJSON = exports.TOTPDeviceFromJSONTyped = exports.TOTPDeviceFromJSON = exports.instanceOfTOTPDevice = void 0;
/**
 * Check if a given object implements the TOTPDevice interface.
 */
function instanceOfTOTPDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
exports.instanceOfTOTPDevice = instanceOfTOTPDevice;
function TOTPDeviceFromJSON(json) {
    return TOTPDeviceFromJSONTyped(json, false);
}
exports.TOTPDeviceFromJSON = TOTPDeviceFromJSON;
function TOTPDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'pk': json['pk'],
    };
}
exports.TOTPDeviceFromJSONTyped = TOTPDeviceFromJSONTyped;
function TOTPDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.TOTPDeviceToJSON = TOTPDeviceToJSON;
