// @goauthentik/api/dist/models/TOTPDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TOTPDeviceRequestToJSON = exports.TOTPDeviceRequestFromJSONTyped = exports.TOTPDeviceRequestFromJSON = exports.instanceOfTOTPDeviceRequest = void 0;
/**
 * Check if a given object implements the TOTPDeviceRequest interface.
 */
function instanceOfTOTPDeviceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfTOTPDeviceRequest = instanceOfTOTPDeviceRequest;
function TOTPDeviceRequestFromJSON(json) {
    return TOTPDeviceRequestFromJSONTyped(json, false);
}
exports.TOTPDeviceRequestFromJSON = TOTPDeviceRequestFromJSON;
function TOTPDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.TOTPDeviceRequestFromJSONTyped = TOTPDeviceRequestFromJSONTyped;
function TOTPDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.TOTPDeviceRequestToJSON = TOTPDeviceRequestToJSON;
