// @goauthentik/api/dist/models/TransactionApplicationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionApplicationRequestToJSON = exports.TransactionApplicationRequestFromJSONTyped = exports.TransactionApplicationRequestFromJSON = exports.instanceOfTransactionApplicationRequest = void 0;
const ApplicationRequest_1 = require("./ApplicationRequest");
const ModelRequest_1 = require("./ModelRequest");
const ProviderModelEnum_1 = require("./ProviderModelEnum");
/**
 * Check if a given object implements the TransactionApplicationRequest interface.
 */
function instanceOfTransactionApplicationRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "app" in value;
    isInstance = isInstance && "providerModel" in value;
    isInstance = isInstance && "provider" in value;
    return isInstance;
}
exports.instanceOfTransactionApplicationRequest = instanceOfTransactionApplicationRequest;
function TransactionApplicationRequestFromJSON(json) {
    return TransactionApplicationRequestFromJSONTyped(json, false);
}
exports.TransactionApplicationRequestFromJSON = TransactionApplicationRequestFromJSON;
function TransactionApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'app': (0, ApplicationRequest_1.ApplicationRequestFromJSON)(json['app']),
        'providerModel': (0, ProviderModelEnum_1.ProviderModelEnumFromJSON)(json['provider_model']),
        'provider': (0, ModelRequest_1.ModelRequestFromJSON)(json['provider']),
    };
}
exports.TransactionApplicationRequestFromJSONTyped = TransactionApplicationRequestFromJSONTyped;
function TransactionApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'app': (0, ApplicationRequest_1.ApplicationRequestToJSON)(value.app),
        'provider_model': (0, ProviderModelEnum_1.ProviderModelEnumToJSON)(value.providerModel),
        'provider': (0, ModelRequest_1.ModelRequestToJSON)(value.provider),
    };
}
exports.TransactionApplicationRequestToJSON = TransactionApplicationRequestToJSON;
