// @goauthentik/api/dist/models/TypeCreate.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypeCreateToJSON = exports.TypeCreateFromJSONTyped = exports.TypeCreateFromJSON = exports.instanceOfTypeCreate = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TypeCreate interface.
 */
function instanceOfTypeCreate(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "modelName" in value;
    return isInstance;
}
exports.instanceOfTypeCreate = instanceOfTypeCreate;
function TypeCreateFromJSON(json) {
    return TypeCreateFromJSONTyped(json, false);
}
exports.TypeCreateFromJSON = TypeCreateFromJSON;
function TypeCreateFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': json['description'],
        'component': json['component'],
        'modelName': json['model_name'],
        'requiresEnterprise': !(0, runtime_1.exists)(json, 'requires_enterprise') ? undefined : json['requires_enterprise'],
    };
}
exports.TypeCreateFromJSONTyped = TypeCreateFromJSONTyped;
function TypeCreateToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'component': value.component,
        'model_name': value.modelName,
        'requires_enterprise': value.requiresEnterprise,
    };
}
exports.TypeCreateToJSON = TypeCreateToJSON;
