// @goauthentik/api/dist/models/UiThemeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UiThemeEnumToJSON = exports.UiThemeEnumFromJSONTyped = exports.UiThemeEnumFromJSON = exports.UiThemeEnum = void 0;
/**
 * * `automatic` - Automatic
 * * `light` - Light
 * * `dark` - Dark
 * @export
 */
exports.UiThemeEnum = {
    Automatic: 'automatic',
    Light: 'light',
    Dark: 'dark',
    UnknownDefaultOpenApi: '11184809'
};
function UiThemeEnumFromJSON(json) {
    return UiThemeEnumFromJSONTyped(json, false);
}
exports.UiThemeEnumFromJSON = UiThemeEnumFromJSON;
function UiThemeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UiThemeEnumFromJSONTyped = UiThemeEnumFromJSONTyped;
function UiThemeEnumToJSON(value) {
    return value;
}
exports.UiThemeEnumToJSON = UiThemeEnumToJSON;
