// @goauthentik/api/dist/models/UsedByActionEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsedByActionEnumToJSON = exports.UsedByActionEnumFromJSONTyped = exports.UsedByActionEnumFromJSON = exports.UsedByActionEnum = void 0;
/**
 * * `cascade` - CASCADE
 * * `cascade_many` - CASCADE_MANY
 * * `set_null` - SET_NULL
 * * `set_default` - SET_DEFAULT
 * @export
 */
exports.UsedByActionEnum = {
    Cascade: 'cascade',
    CascadeMany: 'cascade_many',
    SetNull: 'set_null',
    SetDefault: 'set_default',
    UnknownDefaultOpenApi: '11184809'
};
function UsedByActionEnumFromJSON(json) {
    return UsedByActionEnumFromJSONTyped(json, false);
}
exports.UsedByActionEnumFromJSON = UsedByActionEnumFromJSON;
function UsedByActionEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UsedByActionEnumFromJSONTyped = UsedByActionEnumFromJSONTyped;
function UsedByActionEnumToJSON(value) {
    return value;
}
exports.UsedByActionEnumToJSON = UsedByActionEnumToJSON;
