// @goauthentik/api/dist/models/UserAssignedObjectPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssignedObjectPermissionToJSON = exports.UserAssignedObjectPermissionFromJSONTyped = exports.UserAssignedObjectPermissionFromJSON = exports.instanceOfUserAssignedObjectPermission = void 0;
const runtime_1 = require("../runtime");
const UserObjectPermission_1 = require("./UserObjectPermission");
/**
 * Check if a given object implements the UserAssignedObjectPermission interface.
 */
function instanceOfUserAssignedObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "permissions" in value;
    isInstance = isInstance && "isSuperuser" in value;
    return isInstance;
}
exports.instanceOfUserAssignedObjectPermission = instanceOfUserAssignedObjectPermission;
function UserAssignedObjectPermissionFromJSON(json) {
    return UserAssignedObjectPermissionFromJSONTyped(json, false);
}
exports.UserAssignedObjectPermissionFromJSON = UserAssignedObjectPermissionFromJSON;
function UserAssignedObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !(0, runtime_1.exists)(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !(0, runtime_1.exists)(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
        'permissions': (json['permissions'].map(UserObjectPermission_1.UserObjectPermissionFromJSON)),
        'isSuperuser': json['is_superuser'],
    };
}
exports.UserAssignedObjectPermissionFromJSONTyped = UserAssignedObjectPermissionFromJSONTyped;
function UserAssignedObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'email': value.email,
        'attributes': value.attributes,
        'permissions': (value.permissions.map(UserObjectPermission_1.UserObjectPermissionToJSON)),
        'is_superuser': value.isSuperuser,
    };
}
exports.UserAssignedObjectPermissionToJSON = UserAssignedObjectPermissionToJSON;
