// @goauthentik/api/dist/models/UserAccountRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAccountRequestToJSON = exports.UserAccountRequestFromJSONTyped = exports.UserAccountRequestFromJSON = exports.instanceOfUserAccountRequest = void 0;
/**
 * Check if a given object implements the UserAccountRequest interface.
 */
function instanceOfUserAccountRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
exports.instanceOfUserAccountRequest = instanceOfUserAccountRequest;
function UserAccountRequestFromJSON(json) {
    return UserAccountRequestFromJSONTyped(json, false);
}
exports.UserAccountRequestFromJSON = UserAccountRequestFromJSON;
function UserAccountRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
    };
}
exports.UserAccountRequestFromJSONTyped = UserAccountRequestFromJSONTyped;
function UserAccountRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'pk': value.pk,
    };
}
exports.UserAccountRequestToJSON = UserAccountRequestToJSON;
