// @goauthentik/api/dist/models/UserSourceConnection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSourceConnectionToJSON = exports.UserSourceConnectionFromJSONTyped = exports.UserSourceConnectionFromJSON = exports.instanceOfUserSourceConnection = void 0;
const Source_1 = require("./Source");
/**
 * Check if a given object implements the UserSourceConnection interface.
 */
function instanceOfUserSourceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "created" in value;
    return isInstance;
}
exports.instanceOfUserSourceConnection = instanceOfUserSourceConnection;
function UserSourceConnectionFromJSON(json) {
    return UserSourceConnectionFromJSONTyped(json, false);
}
exports.UserSourceConnectionFromJSON = UserSourceConnectionFromJSON;
function UserSourceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': json['user'],
        'source': (0, Source_1.SourceFromJSON)(json['source']),
        'created': (new Date(json['created'])),
    };
}
exports.UserSourceConnectionFromJSONTyped = UserSourceConnectionFromJSONTyped;
function UserSourceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.UserSourceConnectionToJSON = UserSourceConnectionToJSON;
