// @goauthentik/api/dist/models/UserTypeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTypeEnumToJSON = exports.UserTypeEnumFromJSONTyped = exports.UserTypeEnumFromJSON = exports.UserTypeEnum = void 0;
/**
 * * `internal` - Internal
 * * `external` - External
 * * `service_account` - Service Account
 * * `internal_service_account` - Internal Service Account
 * @export
 */
exports.UserTypeEnum = {
    Internal: 'internal',
    External: 'external',
    ServiceAccount: 'service_account',
    InternalServiceAccount: 'internal_service_account',
    UnknownDefaultOpenApi: '11184809'
};
function UserTypeEnumFromJSON(json) {
    return UserTypeEnumFromJSONTyped(json, false);
}
exports.UserTypeEnumFromJSON = UserTypeEnumFromJSON;
function UserTypeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserTypeEnumFromJSONTyped = UserTypeEnumFromJSONTyped;
function UserTypeEnumToJSON(value) {
    return value;
}
exports.UserTypeEnumToJSON = UserTypeEnumToJSON;
