// @goauthentik/api/dist/models/UserFieldsEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserFieldsEnumToJSON = exports.UserFieldsEnumFromJSONTyped = exports.UserFieldsEnumFromJSON = exports.UserFieldsEnum = void 0;
/**
 * * `email` - E Mail
 * * `username` - Username
 * * `upn` - Upn
 * @export
 */
exports.UserFieldsEnum = {
    Email: 'email',
    Username: 'username',
    Upn: 'upn',
    UnknownDefaultOpenApi: '11184809'
};
function UserFieldsEnumFromJSON(json) {
    return UserFieldsEnumFromJSONTyped(json, false);
}
exports.UserFieldsEnumFromJSON = UserFieldsEnumFromJSON;
function UserFieldsEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserFieldsEnumFromJSONTyped = UserFieldsEnumFromJSONTyped;
function UserFieldsEnumToJSON(value) {
    return value;
}
exports.UserFieldsEnumToJSON = UserFieldsEnumToJSON;
