// @goauthentik/api/dist/models/UserSAMLSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSAMLSourceConnectionRequestToJSON = exports.UserSAMLSourceConnectionRequestFromJSONTyped = exports.UserSAMLSourceConnectionRequestFromJSON = exports.instanceOfUserSAMLSourceConnectionRequest = void 0;
/**
 * Check if a given object implements the UserSAMLSourceConnectionRequest interface.
 */
function instanceOfUserSAMLSourceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "identifier" in value;
    return isInstance;
}
exports.instanceOfUserSAMLSourceConnectionRequest = instanceOfUserSAMLSourceConnectionRequest;
function UserSAMLSourceConnectionRequestFromJSON(json) {
    return UserSAMLSourceConnectionRequestFromJSONTyped(json, false);
}
exports.UserSAMLSourceConnectionRequestFromJSON = UserSAMLSourceConnectionRequestFromJSON;
function UserSAMLSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'identifier': json['identifier'],
    };
}
exports.UserSAMLSourceConnectionRequestFromJSONTyped = UserSAMLSourceConnectionRequestFromJSONTyped;
function UserSAMLSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
    };
}
exports.UserSAMLSourceConnectionRequestToJSON = UserSAMLSourceConnectionRequestToJSON;
