// @goauthentik/api/dist/models/UserLoginStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLoginStageToJSON = exports.UserLoginStageFromJSONTyped = exports.UserLoginStageFromJSON = exports.instanceOfUserLoginStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
const GeoipBindingEnum_1 = require("./GeoipBindingEnum");
const NetworkBindingEnum_1 = require("./NetworkBindingEnum");
/**
 * Check if a given object implements the UserLoginStage interface.
 */
function instanceOfUserLoginStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfUserLoginStage = instanceOfUserLoginStage;
function UserLoginStageFromJSON(json) {
    return UserLoginStageFromJSONTyped(json, false);
}
exports.UserLoginStageFromJSON = UserLoginStageFromJSON;
function UserLoginStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'sessionDuration': !(0, runtime_1.exists)(json, 'session_duration') ? undefined : json['session_duration'],
        'terminateOtherSessions': !(0, runtime_1.exists)(json, 'terminate_other_sessions') ? undefined : json['terminate_other_sessions'],
        'rememberMeOffset': !(0, runtime_1.exists)(json, 'remember_me_offset') ? undefined : json['remember_me_offset'],
        'networkBinding': !(0, runtime_1.exists)(json, 'network_binding') ? undefined : (0, NetworkBindingEnum_1.NetworkBindingEnumFromJSON)(json['network_binding']),
        'geoipBinding': !(0, runtime_1.exists)(json, 'geoip_binding') ? undefined : (0, GeoipBindingEnum_1.GeoipBindingEnumFromJSON)(json['geoip_binding']),
    };
}
exports.UserLoginStageFromJSONTyped = UserLoginStageFromJSONTyped;
function UserLoginStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'session_duration': value.sessionDuration,
        'terminate_other_sessions': value.terminateOtherSessions,
        'remember_me_offset': value.rememberMeOffset,
        'network_binding': (0, NetworkBindingEnum_1.NetworkBindingEnumToJSON)(value.networkBinding),
        'geoip_binding': (0, GeoipBindingEnum_1.GeoipBindingEnumToJSON)(value.geoipBinding),
    };
}
exports.UserLoginStageToJSON = UserLoginStageToJSON;
