// @goauthentik/api/dist/models/UserObjectPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserObjectPermissionToJSON = exports.UserObjectPermissionFromJSONTyped = exports.UserObjectPermissionFromJSON = exports.instanceOfUserObjectPermission = void 0;
/**
 * Check if a given object implements the UserObjectPermission interface.
 */
function instanceOfUserObjectPermission(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "codename" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "appLabel" in value;
    isInstance = isInstance && "objectPk" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfUserObjectPermission = instanceOfUserObjectPermission;
function UserObjectPermissionFromJSON(json) {
    return UserObjectPermissionFromJSONTyped(json, false);
}
exports.UserObjectPermissionFromJSON = UserObjectPermissionFromJSON;
function UserObjectPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'codename': json['codename'],
        'model': json['model'],
        'appLabel': json['app_label'],
        'objectPk': json['object_pk'],
        'name': json['name'],
    };
}
exports.UserObjectPermissionFromJSONTyped = UserObjectPermissionFromJSONTyped;
function UserObjectPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.UserObjectPermissionToJSON = UserObjectPermissionToJSON;
