// @goauthentik/api/dist/models/UserServiceAccountRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserServiceAccountRequestToJSON = exports.UserServiceAccountRequestFromJSONTyped = exports.UserServiceAccountRequestFromJSON = exports.instanceOfUserServiceAccountRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserServiceAccountRequest interface.
 */
function instanceOfUserServiceAccountRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfUserServiceAccountRequest = instanceOfUserServiceAccountRequest;
function UserServiceAccountRequestFromJSON(json) {
    return UserServiceAccountRequestFromJSONTyped(json, false);
}
exports.UserServiceAccountRequestFromJSON = UserServiceAccountRequestFromJSON;
function UserServiceAccountRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'createGroup': !(0, runtime_1.exists)(json, 'create_group') ? undefined : json['create_group'],
        'expiring': !(0, runtime_1.exists)(json, 'expiring') ? undefined : json['expiring'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
    };
}
exports.UserServiceAccountRequestFromJSONTyped = UserServiceAccountRequestFromJSONTyped;
function UserServiceAccountRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'create_group': value.createGroup,
        'expiring': value.expiring,
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
    };
}
exports.UserServiceAccountRequestToJSON = UserServiceAccountRequestToJSON;
