// @goauthentik/api/dist/models/User.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserToJSON = exports.UserFromJSONTyped = exports.UserFromJSON = exports.instanceOfUser = void 0;
const runtime_1 = require("../runtime");
const UserGroup_1 = require("./UserGroup");
const UserTypeEnum_1 = require("./UserTypeEnum");
/**
 * Check if a given object implements the User interface.
 */
function instanceOfUser(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isSuperuser" in value;
    isInstance = isInstance && "groupsObj" in value;
    isInstance = isInstance && "avatar" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "uuid" in value;
    return isInstance;
}
exports.instanceOfUser = instanceOfUser;
function UserFromJSON(json) {
    return UserFromJSONTyped(json, false);
}
exports.UserFromJSON = UserFromJSON;
function UserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': !(0, runtime_1.exists)(json, 'is_active') ? undefined : json['is_active'],
        'lastLogin': !(0, runtime_1.exists)(json, 'last_login') ? undefined : (json['last_login'] === null ? null : new Date(json['last_login'])),
        'isSuperuser': json['is_superuser'],
        'groups': !(0, runtime_1.exists)(json, 'groups') ? undefined : json['groups'],
        'groupsObj': (json['groups_obj'].map(UserGroup_1.UserGroupFromJSON)),
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'avatar': json['avatar'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
        'uid': json['uid'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : (0, UserTypeEnum_1.UserTypeEnumFromJSON)(json['type']),
        'uuid': json['uuid'],
    };
}
exports.UserFromJSONTyped = UserFromJSONTyped;
function UserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'is_active': value.isActive,
        'last_login': value.lastLogin === undefined ? undefined : (value.lastLogin === null ? null : value.lastLogin.toISOString()),
        'groups': value.groups,
        'email': value.email,
        'attributes': value.attributes,
        'path': value.path,
        'type': (0, UserTypeEnum_1.UserTypeEnumToJSON)(value.type),
    };
}
exports.UserToJSON = UserToJSON;
