// @goauthentik/api/dist/models/UserSelf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSelfToJSON = exports.UserSelfFromJSONTyped = exports.UserSelfFromJSON = exports.instanceOfUserSelf = void 0;
const runtime_1 = require("../runtime");
const UserSelfGroups_1 = require("./UserSelfGroups");
const UserTypeEnum_1 = require("./UserTypeEnum");
/**
 * Check if a given object implements the UserSelf interface.
 */
function instanceOfUserSelf(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isActive" in value;
    isInstance = isInstance && "isSuperuser" in value;
    isInstance = isInstance && "groups" in value;
    isInstance = isInstance && "avatar" in value;
    isInstance = isInstance && "uid" in value;
    isInstance = isInstance && "settings" in value;
    isInstance = isInstance && "systemPermissions" in value;
    return isInstance;
}
exports.instanceOfUserSelf = instanceOfUserSelf;
function UserSelfFromJSON(json) {
    return UserSelfFromJSONTyped(json, false);
}
exports.UserSelfFromJSON = UserSelfFromJSON;
function UserSelfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'username': json['username'],
        'name': json['name'],
        'isActive': json['is_active'],
        'isSuperuser': json['is_superuser'],
        'groups': (json['groups'].map(UserSelfGroups_1.UserSelfGroupsFromJSON)),
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'avatar': json['avatar'],
        'uid': json['uid'],
        'settings': json['settings'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : (0, UserTypeEnum_1.UserTypeEnumFromJSON)(json['type']),
        'systemPermissions': json['system_permissions'],
    };
}
exports.UserSelfFromJSONTyped = UserSelfFromJSONTyped;
function UserSelfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'name': value.name,
        'email': value.email,
        'type': (0, UserTypeEnum_1.UserTypeEnumToJSON)(value.type),
    };
}
exports.UserSelfToJSON = UserSelfToJSON;
