// @goauthentik/api/dist/models/UserSetting.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSettingToJSON = exports.UserSettingFromJSONTyped = exports.UserSettingFromJSON = exports.instanceOfUserSetting = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserSetting interface.
 */
function instanceOfUserSetting(value) {
    let isInstance = true;
    isInstance = isInstance && "objectUid" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "title" in value;
    return isInstance;
}
exports.instanceOfUserSetting = instanceOfUserSetting;
function UserSettingFromJSON(json) {
    return UserSettingFromJSONTyped(json, false);
}
exports.UserSettingFromJSON = UserSettingFromJSON;
function UserSettingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'objectUid': json['object_uid'],
        'component': json['component'],
        'title': json['title'],
        'configureUrl': !(0, runtime_1.exists)(json, 'configure_url') ? undefined : json['configure_url'],
        'iconUrl': !(0, runtime_1.exists)(json, 'icon_url') ? undefined : json['icon_url'],
    };
}
exports.UserSettingFromJSONTyped = UserSettingFromJSONTyped;
function UserSettingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'object_uid': value.objectUid,
        'component': value.component,
        'title': value.title,
        'configure_url': value.configureUrl,
        'icon_url': value.iconUrl,
    };
}
exports.UserSettingToJSON = UserSettingToJSON;
