// @goauthentik/api/dist/models/UserPasswordSetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPasswordSetRequestToJSON = exports.UserPasswordSetRequestFromJSONTyped = exports.UserPasswordSetRequestFromJSON = exports.instanceOfUserPasswordSetRequest = void 0;
/**
 * Check if a given object implements the UserPasswordSetRequest interface.
 */
function instanceOfUserPasswordSetRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "password" in value;
    return isInstance;
}
exports.instanceOfUserPasswordSetRequest = instanceOfUserPasswordSetRequest;
function UserPasswordSetRequestFromJSON(json) {
    return UserPasswordSetRequestFromJSONTyped(json, false);
}
exports.UserPasswordSetRequestFromJSON = UserPasswordSetRequestFromJSON;
function UserPasswordSetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'password': json['password'],
    };
}
exports.UserPasswordSetRequestFromJSONTyped = UserPasswordSetRequestFromJSONTyped;
function UserPasswordSetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'password': value.password,
    };
}
exports.UserPasswordSetRequestToJSON = UserPasswordSetRequestToJSON;
