// @goauthentik/api/dist/models/UserGroupRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserGroupRequestToJSON = exports.UserGroupRequestFromJSONTyped = exports.UserGroupRequestFromJSON = exports.instanceOfUserGroupRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserGroupRequest interface.
 */
function instanceOfUserGroupRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfUserGroupRequest = instanceOfUserGroupRequest;
function UserGroupRequestFromJSON(json) {
    return UserGroupRequestFromJSONTyped(json, false);
}
exports.UserGroupRequestFromJSON = UserGroupRequestFromJSON;
function UserGroupRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'isSuperuser': !(0, runtime_1.exists)(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !(0, runtime_1.exists)(json, 'parent') ? undefined : json['parent'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
    };
}
exports.UserGroupRequestFromJSONTyped = UserGroupRequestFromJSONTyped;
function UserGroupRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'attributes': value.attributes,
    };
}
exports.UserGroupRequestToJSON = UserGroupRequestToJSON;
