// @goauthentik/api/dist/models/UserCreationModeEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserCreationModeEnumToJSON = exports.UserCreationModeEnumFromJSONTyped = exports.UserCreationModeEnumFromJSON = exports.UserCreationModeEnum = void 0;
/**
 * * `never_create` - Never Create
 * * `create_when_required` - Create When Required
 * * `always_create` - Always Create
 * @export
 */
exports.UserCreationModeEnum = {
    NeverCreate: 'never_create',
    CreateWhenRequired: 'create_when_required',
    AlwaysCreate: 'always_create',
    UnknownDefaultOpenApi: '11184809'
};
function UserCreationModeEnumFromJSON(json) {
    return UserCreationModeEnumFromJSONTyped(json, false);
}
exports.UserCreationModeEnumFromJSON = UserCreationModeEnumFromJSON;
function UserCreationModeEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserCreationModeEnumFromJSONTyped = UserCreationModeEnumFromJSONTyped;
function UserCreationModeEnumToJSON(value) {
    return value;
}
exports.UserCreationModeEnumToJSON = UserCreationModeEnumToJSON;
