// @goauthentik/api/dist/models/UserConsent.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserConsentToJSON = exports.UserConsentFromJSONTyped = exports.UserConsentFromJSON = exports.instanceOfUserConsent = void 0;
const runtime_1 = require("../runtime");
const Application_1 = require("./Application");
const User_1 = require("./User");
/**
 * Check if a given object implements the UserConsent interface.
 */
function instanceOfUserConsent(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "application" in value;
    return isInstance;
}
exports.instanceOfUserConsent = instanceOfUserConsent;
function UserConsentFromJSON(json) {
    return UserConsentFromJSONTyped(json, false);
}
exports.UserConsentFromJSON = UserConsentFromJSON;
function UserConsentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'expires': !(0, runtime_1.exists)(json, 'expires') ? undefined : (new Date(json['expires'])),
        'expiring': !(0, runtime_1.exists)(json, 'expiring') ? undefined : json['expiring'],
        'user': (0, User_1.UserFromJSON)(json['user']),
        'application': (0, Application_1.ApplicationFromJSON)(json['application']),
        'permissions': !(0, runtime_1.exists)(json, 'permissions') ? undefined : json['permissions'],
    };
}
exports.UserConsentFromJSONTyped = UserConsentFromJSONTyped;
function UserConsentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'expires': value.expires === undefined ? undefined : (value.expires.toISOString()),
        'expiring': value.expiring,
        'user': (0, User_1.UserToJSON)(value.user),
        'application': (0, Application_1.ApplicationToJSON)(value.application),
        'permissions': value.permissions,
    };
}
exports.UserConsentToJSON = UserConsentToJSON;
