// @goauthentik/api/dist/models/UserOAuthSourceConnectionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserOAuthSourceConnectionRequestToJSON = exports.UserOAuthSourceConnectionRequestFromJSONTyped = exports.UserOAuthSourceConnectionRequestFromJSON = exports.instanceOfUserOAuthSourceConnectionRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserOAuthSourceConnectionRequest interface.
 */
function instanceOfUserOAuthSourceConnectionRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "identifier" in value;
    return isInstance;
}
exports.instanceOfUserOAuthSourceConnectionRequest = instanceOfUserOAuthSourceConnectionRequest;
function UserOAuthSourceConnectionRequestFromJSON(json) {
    return UserOAuthSourceConnectionRequestFromJSONTyped(json, false);
}
exports.UserOAuthSourceConnectionRequestFromJSON = UserOAuthSourceConnectionRequestFromJSON;
function UserOAuthSourceConnectionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': json['user'],
        'identifier': json['identifier'],
        'accessToken': !(0, runtime_1.exists)(json, 'access_token') ? undefined : json['access_token'],
    };
}
exports.UserOAuthSourceConnectionRequestFromJSONTyped = UserOAuthSourceConnectionRequestFromJSONTyped;
function UserOAuthSourceConnectionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
        'access_token': value.accessToken,
    };
}
exports.UserOAuthSourceConnectionRequestToJSON = UserOAuthSourceConnectionRequestToJSON;
