// @goauthentik/api/dist/models/UserLoginChallengeResponseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLoginChallengeResponseRequestToJSON = exports.UserLoginChallengeResponseRequestFromJSONTyped = exports.UserLoginChallengeResponseRequestFromJSON = exports.instanceOfUserLoginChallengeResponseRequest = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserLoginChallengeResponseRequest interface.
 */
function instanceOfUserLoginChallengeResponseRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "rememberMe" in value;
    return isInstance;
}
exports.instanceOfUserLoginChallengeResponseRequest = instanceOfUserLoginChallengeResponseRequest;
function UserLoginChallengeResponseRequestFromJSON(json) {
    return UserLoginChallengeResponseRequestFromJSONTyped(json, false);
}
exports.UserLoginChallengeResponseRequestFromJSON = UserLoginChallengeResponseRequestFromJSON;
function UserLoginChallengeResponseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'rememberMe': json['remember_me'],
    };
}
exports.UserLoginChallengeResponseRequestFromJSONTyped = UserLoginChallengeResponseRequestFromJSONTyped;
function UserLoginChallengeResponseRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'remember_me': value.rememberMe,
    };
}
exports.UserLoginChallengeResponseRequestToJSON = UserLoginChallengeResponseRequestToJSON;
