// @goauthentik/api/dist/models/UserGroup.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserGroupToJSON = exports.UserGroupFromJSONTyped = exports.UserGroupFromJSON = exports.instanceOfUserGroup = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserGroup interface.
 */
function instanceOfUserGroup(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "numPk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "parentName" in value;
    return isInstance;
}
exports.instanceOfUserGroup = instanceOfUserGroup;
function UserGroupFromJSON(json) {
    return UserGroupFromJSONTyped(json, false);
}
exports.UserGroupFromJSON = UserGroupFromJSON;
function UserGroupFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'numPk': json['num_pk'],
        'name': json['name'],
        'isSuperuser': !(0, runtime_1.exists)(json, 'is_superuser') ? undefined : json['is_superuser'],
        'parent': !(0, runtime_1.exists)(json, 'parent') ? undefined : json['parent'],
        'parentName': json['parent_name'],
        'attributes': !(0, runtime_1.exists)(json, 'attributes') ? undefined : json['attributes'],
    };
}
exports.UserGroupFromJSONTyped = UserGroupFromJSONTyped;
function UserGroupToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'is_superuser': value.isSuperuser,
        'parent': value.parent,
        'attributes': value.attributes,
    };
}
exports.UserGroupToJSON = UserGroupToJSON;
