// @goauthentik/api/dist/models/UserSelfGroups.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSelfGroupsToJSON = exports.UserSelfGroupsFromJSONTyped = exports.UserSelfGroupsFromJSON = exports.instanceOfUserSelfGroups = void 0;
/**
 * Check if a given object implements the UserSelfGroups interface.
 */
function instanceOfUserSelfGroups(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "pk" in value;
    return isInstance;
}
exports.instanceOfUserSelfGroups = instanceOfUserSelfGroups;
function UserSelfGroupsFromJSON(json) {
    return UserSelfGroupsFromJSONTyped(json, false);
}
exports.UserSelfGroupsFromJSON = UserSelfGroupsFromJSON;
function UserSelfGroupsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'pk': json['pk'],
    };
}
exports.UserSelfGroupsFromJSONTyped = UserSelfGroupsFromJSONTyped;
function UserSelfGroupsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.UserSelfGroupsToJSON = UserSelfGroupsToJSON;
