// @goauthentik/api/dist/models/UserWriteStage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserWriteStageToJSON = exports.UserWriteStageFromJSONTyped = exports.UserWriteStageFromJSON = exports.instanceOfUserWriteStage = void 0;
const runtime_1 = require("../runtime");
const FlowSet_1 = require("./FlowSet");
const UserCreationModeEnum_1 = require("./UserCreationModeEnum");
const UserTypeEnum_1 = require("./UserTypeEnum");
/**
 * Check if a given object implements the UserWriteStage interface.
 */
function instanceOfUserWriteStage(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "component" in value;
    isInstance = isInstance && "verboseName" in value;
    isInstance = isInstance && "verboseNamePlural" in value;
    isInstance = isInstance && "metaModelName" in value;
    return isInstance;
}
exports.instanceOfUserWriteStage = instanceOfUserWriteStage;
function UserWriteStageFromJSON(json) {
    return UserWriteStageFromJSONTyped(json, false);
}
exports.UserWriteStageFromJSON = UserWriteStageFromJSON;
function UserWriteStageFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'component': json['component'],
        'verboseName': json['verbose_name'],
        'verboseNamePlural': json['verbose_name_plural'],
        'metaModelName': json['meta_model_name'],
        'flowSet': !(0, runtime_1.exists)(json, 'flow_set') ? undefined : (json['flow_set'].map(FlowSet_1.FlowSetFromJSON)),
        'userCreationMode': !(0, runtime_1.exists)(json, 'user_creation_mode') ? undefined : (0, UserCreationModeEnum_1.UserCreationModeEnumFromJSON)(json['user_creation_mode']),
        'createUsersAsInactive': !(0, runtime_1.exists)(json, 'create_users_as_inactive') ? undefined : json['create_users_as_inactive'],
        'createUsersGroup': !(0, runtime_1.exists)(json, 'create_users_group') ? undefined : json['create_users_group'],
        'userType': !(0, runtime_1.exists)(json, 'user_type') ? undefined : (0, UserTypeEnum_1.UserTypeEnumFromJSON)(json['user_type']),
        'userPathTemplate': !(0, runtime_1.exists)(json, 'user_path_template') ? undefined : json['user_path_template'],
    };
}
exports.UserWriteStageFromJSONTyped = UserWriteStageFromJSONTyped;
function UserWriteStageToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'flow_set': value.flowSet === undefined ? undefined : (value.flowSet.map(FlowSet_1.FlowSetToJSON)),
        'user_creation_mode': (0, UserCreationModeEnum_1.UserCreationModeEnumToJSON)(value.userCreationMode),
        'create_users_as_inactive': value.createUsersAsInactive,
        'create_users_group': value.createUsersGroup,
        'user_type': (0, UserTypeEnum_1.UserTypeEnumToJSON)(value.userType),
        'user_path_template': value.userPathTemplate,
    };
}
exports.UserWriteStageToJSON = UserWriteStageToJSON;
