// @goauthentik/api/dist/models/UserServiceAccountResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserServiceAccountResponseToJSON = exports.UserServiceAccountResponseFromJSONTyped = exports.UserServiceAccountResponseFromJSON = exports.instanceOfUserServiceAccountResponse = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserServiceAccountResponse interface.
 */
function instanceOfUserServiceAccountResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "username" in value;
    isInstance = isInstance && "token" in value;
    isInstance = isInstance && "userUid" in value;
    isInstance = isInstance && "userPk" in value;
    return isInstance;
}
exports.instanceOfUserServiceAccountResponse = instanceOfUserServiceAccountResponse;
function UserServiceAccountResponseFromJSON(json) {
    return UserServiceAccountResponseFromJSONTyped(json, false);
}
exports.UserServiceAccountResponseFromJSON = UserServiceAccountResponseFromJSON;
function UserServiceAccountResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'username': json['username'],
        'token': json['token'],
        'userUid': json['user_uid'],
        'userPk': json['user_pk'],
        'groupPk': !(0, runtime_1.exists)(json, 'group_pk') ? undefined : json['group_pk'],
    };
}
exports.UserServiceAccountResponseFromJSONTyped = UserServiceAccountResponseFromJSONTyped;
function UserServiceAccountResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'token': value.token,
        'user_uid': value.userUid,
        'user_pk': value.userPk,
        'group_pk': value.groupPk,
    };
}
exports.UserServiceAccountResponseToJSON = UserServiceAccountResponseToJSON;
