// @goauthentik/api/dist/models/UserVerificationEnum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserVerificationEnumToJSON = exports.UserVerificationEnumFromJSONTyped = exports.UserVerificationEnumFromJSON = exports.UserVerificationEnum = void 0;
/**
 * * `required` - Required
 * * `preferred` - Preferred
 * * `discouraged` - Discouraged
 * @export
 */
exports.UserVerificationEnum = {
    Required: 'required',
    Preferred: 'preferred',
    Discouraged: 'discouraged',
    UnknownDefaultOpenApi: '11184809'
};
function UserVerificationEnumFromJSON(json) {
    return UserVerificationEnumFromJSONTyped(json, false);
}
exports.UserVerificationEnumFromJSON = UserVerificationEnumFromJSON;
function UserVerificationEnumFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserVerificationEnumFromJSONTyped = UserVerificationEnumFromJSONTyped;
function UserVerificationEnumToJSON(value) {
    return value;
}
exports.UserVerificationEnumToJSON = UserVerificationEnumToJSON;
