// @goauthentik/api/dist/models/UserSAMLSourceConnection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSAMLSourceConnectionToJSON = exports.UserSAMLSourceConnectionFromJSONTyped = exports.UserSAMLSourceConnectionFromJSON = exports.instanceOfUserSAMLSourceConnection = void 0;
const Source_1 = require("./Source");
/**
 * Check if a given object implements the UserSAMLSourceConnection interface.
 */
function instanceOfUserSAMLSourceConnection(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "user" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "identifier" in value;
    return isInstance;
}
exports.instanceOfUserSAMLSourceConnection = instanceOfUserSAMLSourceConnection;
function UserSAMLSourceConnectionFromJSON(json) {
    return UserSAMLSourceConnectionFromJSONTyped(json, false);
}
exports.UserSAMLSourceConnectionFromJSON = UserSAMLSourceConnectionFromJSON;
function UserSAMLSourceConnectionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'user': json['user'],
        'source': (0, Source_1.SourceFromJSON)(json['source']),
        'identifier': json['identifier'],
    };
}
exports.UserSAMLSourceConnectionFromJSONTyped = UserSAMLSourceConnectionFromJSONTyped;
function UserSAMLSourceConnectionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': value.user,
        'identifier': value.identifier,
    };
}
exports.UserSAMLSourceConnectionToJSON = UserSAMLSourceConnectionToJSON;
