// @goauthentik/api/dist/models/UserPath.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPathToJSON = exports.UserPathFromJSONTyped = exports.UserPathFromJSON = exports.instanceOfUserPath = void 0;
/**
 * Check if a given object implements the UserPath interface.
 */
function instanceOfUserPath(value) {
    let isInstance = true;
    isInstance = isInstance && "paths" in value;
    return isInstance;
}
exports.instanceOfUserPath = instanceOfUserPath;
function UserPathFromJSON(json) {
    return UserPathFromJSONTyped(json, false);
}
exports.UserPathFromJSON = UserPathFromJSON;
function UserPathFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'paths': json['paths'],
    };
}
exports.UserPathFromJSONTyped = UserPathFromJSONTyped;
function UserPathToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.UserPathToJSON = UserPathToJSON;
