// @goauthentik/api/dist/models/ValidationError.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationErrorToJSON = exports.ValidationErrorFromJSONTyped = exports.ValidationErrorFromJSON = exports.instanceOfValidationError = void 0;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ValidationError interface.
 */
function instanceOfValidationError(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfValidationError = instanceOfValidationError;
function ValidationErrorFromJSON(json) {
    return ValidationErrorFromJSONTyped(json, false);
}
exports.ValidationErrorFromJSON = ValidationErrorFromJSON;
function ValidationErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return Object.assign(Object.assign({}, json), { 'nonFieldErrors': !(0, runtime_1.exists)(json, 'non_field_errors') ? undefined : json['non_field_errors'], 'code': !(0, runtime_1.exists)(json, 'code') ? undefined : json['code'] });
}
exports.ValidationErrorFromJSONTyped = ValidationErrorFromJSONTyped;
function ValidationErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return Object.assign(Object.assign({}, value), { 'non_field_errors': value.nonFieldErrors, 'code': value.code });
}
exports.ValidationErrorToJSON = ValidationErrorToJSON;
