// @goauthentik/api/dist/models/Version.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionToJSON = exports.VersionFromJSONTyped = exports.VersionFromJSON = exports.instanceOfVersion = void 0;
/**
 * Check if a given object implements the Version interface.
 */
function instanceOfVersion(value) {
    let isInstance = true;
    isInstance = isInstance && "versionCurrent" in value;
    isInstance = isInstance && "versionLatest" in value;
    isInstance = isInstance && "buildHash" in value;
    isInstance = isInstance && "outdated" in value;
    return isInstance;
}
exports.instanceOfVersion = instanceOfVersion;
function VersionFromJSON(json) {
    return VersionFromJSONTyped(json, false);
}
exports.VersionFromJSON = VersionFromJSON;
function VersionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'versionCurrent': json['version_current'],
        'versionLatest': json['version_latest'],
        'buildHash': json['build_hash'],
        'outdated': json['outdated'],
    };
}
exports.VersionFromJSONTyped = VersionFromJSONTyped;
function VersionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {};
}
exports.VersionToJSON = VersionToJSON;
