// @goauthentik/api/dist/models/WebAuthnDevice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAuthnDeviceToJSON = exports.WebAuthnDeviceFromJSONTyped = exports.WebAuthnDeviceFromJSON = exports.instanceOfWebAuthnDevice = void 0;
/**
 * Check if a given object implements the WebAuthnDevice interface.
 */
function instanceOfWebAuthnDevice(value) {
    let isInstance = true;
    isInstance = isInstance && "pk" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdOn" in value;
    return isInstance;
}
exports.instanceOfWebAuthnDevice = instanceOfWebAuthnDevice;
function WebAuthnDeviceFromJSON(json) {
    return WebAuthnDeviceFromJSONTyped(json, false);
}
exports.WebAuthnDeviceFromJSON = WebAuthnDeviceFromJSON;
function WebAuthnDeviceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'pk': json['pk'],
        'name': json['name'],
        'createdOn': (new Date(json['created_on'])),
    };
}
exports.WebAuthnDeviceFromJSONTyped = WebAuthnDeviceFromJSONTyped;
function WebAuthnDeviceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.WebAuthnDeviceToJSON = WebAuthnDeviceToJSON;
