// @goauthentik/api/dist/models/WebAuthnDeviceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAuthnDeviceRequestToJSON = exports.WebAuthnDeviceRequestFromJSONTyped = exports.WebAuthnDeviceRequestFromJSON = exports.instanceOfWebAuthnDeviceRequest = void 0;
/**
 * Check if a given object implements the WebAuthnDeviceRequest interface.
 */
function instanceOfWebAuthnDeviceRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfWebAuthnDeviceRequest = instanceOfWebAuthnDeviceRequest;
function WebAuthnDeviceRequestFromJSON(json) {
    return WebAuthnDeviceRequestFromJSONTyped(json, false);
}
exports.WebAuthnDeviceRequestFromJSON = WebAuthnDeviceRequestFromJSON;
function WebAuthnDeviceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
    };
}
exports.WebAuthnDeviceRequestFromJSONTyped = WebAuthnDeviceRequestFromJSONTyped;
function WebAuthnDeviceRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.WebAuthnDeviceRequestToJSON = WebAuthnDeviceRequestToJSON;
