// @goauthentik/api/dist/models/Workers.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * authentik
 * Making authentication simple.
 *
 * The version of the OpenAPI document: 2024.2.1
 * Contact: hello@goauthentik.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkersToJSON = exports.WorkersFromJSONTyped = exports.WorkersFromJSON = exports.instanceOfWorkers = void 0;
/**
 * Check if a given object implements the Workers interface.
 */
function instanceOfWorkers(value) {
    let isInstance = true;
    isInstance = isInstance && "count" in value;
    return isInstance;
}
exports.instanceOfWorkers = instanceOfWorkers;
function WorkersFromJSON(json) {
    return WorkersFromJSONTyped(json, false);
}
exports.WorkersFromJSON = WorkersFromJSON;
function WorkersFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': json['count'],
    };
}
exports.WorkersFromJSONTyped = WorkersFromJSONTyped;
function WorkersToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'count': value.count,
    };
}
exports.WorkersToJSON = WorkersToJSON;
